<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>Unmapped reads Annotations - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="Unmapped reads Annotations by Kraken2 and bracken"><meta property="og:type" content="blog"><meta property="og:title" content="Unmapped reads Annotations"><meta property="og:url" content="https://karobben.github.io/2022/08/30/Bioinfor/unmapped-reads/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="Unmapped reads Annotations by Kraken2 and bracken"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://cdn.pixabay.com/photo/2017/01/07/00/07/bacteria-1959386_1280.jpg"><meta property="article:published_time" content="2022-08-30T17:53:05.000Z"><meta property="article:modified_time" content="2023-06-06T21:42:32.126Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="Software"><meta property="article:tag" content="Bioinformatics"><meta property="article:tag" content="NGS"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://cdn.pixabay.com/photo/2017/01/07/00/07/bacteria-1959386_1280.jpg"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2022/08/30/Bioinfor/unmapped-reads/"},"headline":"Karobben","image":["https://cdn.pixabay.com/photo/2017/01/07/00/07/bacteria-1959386_1280.jpg"],"datePublished":"2022-08-30T17:53:05.000Z","dateModified":"2023-06-06T21:42:32.126Z","author":{"@type":"Person","name":"Karobben"},"description":"Unmapped reads Annotations by Kraken2 and bracken"}</script><link rel="canonical" href="https://karobben.github.io/2022/08/30/Bioinfor/unmapped-reads/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><div class="card-image"><span class="image is-7by3"><img class="fill" src="https://cdn.pixabay.com/photo/2017/01/07/00/07/bacteria-1959386_1280.jpg" alt="Unmapped reads Annotations"></span><span class="ccopy" style="position: absolute; bottom: 0; right: 0; padding-right:3px; padding-left:4px; background-color: rgba(255, 255, 255, 0.6)"><a target="_blank" rel="noopener" href="https://pixabay.com/illustrations/bacteria-germs-microbes-medical-1959386/">© Prawny  </a></span></div><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2022-08-30T17:53:05.000Z" title="8/30/2022, 12:53:05 PM">2022-08-30</time></span><span class="level-item">Updated&nbsp;<time dateTime="2023-06-06T21:42:32.126Z" title="6/6/2023, 4:42:32 PM">2023-06-06</time></span><span class="level-item"><a class="link-muted" href="/categories/Biology/">Biology</a><span> / </span><a class="link-muted" href="/categories/Biology/Bioinformatics/">Bioinformatics</a><span> / </span><a class="link-muted" href="/categories/Biology/Bioinformatics/Software/">Software</a><span> / </span><a class="link-muted" href="/categories/Biology/Bioinformatics/Software/De-nove/">De nove</a></span><span class="level-item">29 minutes read (About 4420 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">Unmapped reads Annotations</h1><div class="content"><h2 id="Why-Unmapped-Reads">Why Unmapped Reads</h2>
<p>Though we expect that the microbes are free in normal/healthy animal tissue, organ, or brain, we still get a few unmapped reads from NGS. And most of them are contributed by microbes. A Gouin, et al., 2014<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup> studied those unmapped reads from 33 plants and believed that those sequences may contribute to the divergence between host plant specialized biotypes.</p>
<p>To identify those unmapped reads, lots of pipelines and tools are developed. Mara Sangiovanni, 2019<sup class="footnote-ref"><a href="#fn2" id="fnref2">[2]</a></sup> published the <strong>DecontaMiner</strong> for investigating the presence of contamination. Zifan Zhu, 2019<sup class="footnote-ref"><a href="#fn3" id="fnref3">[3]</a></sup> released <strong>MicroPro</strong>  to study the know and unknown microbes based on the unmapped reads.</p>
<p>Not only in DNA-seq, but RNA-seq could also find lots of unmapped reads and they could be meaningful, too. Artur Gurgul, et al., 2022 <sup class="footnote-ref"><a href="#fn4" id="fnref4">[4]</a></sup> found those unmapped reads not only comes from contamination but also come from the hosts and could be used to refine the reference genome. For example, Majid Kazemian, 2015<sup class="footnote-ref"><a href="#fn5" id="fnref5">[5]</a></sup> find lots of new or abnormal transcripts from human cancer RNA-seq. A few de novo contigs from unmapped reads could not align to humans but other few kinds of Primates have positive effects on the tumor cell.</p>
<p>Possible sources of unmapped reads:</p>
<ul>
<li>Biology Issue
<ul>
<li>Human (From the technician)</li>
<li>symbionts/pathogens
<ul>
<li>Viruses</li>
<li>Bacteria</li>
<li>fungi</li>
<li>other parasites (exp, mites)</li>
</ul>
</li>
<li>repetitive elements (some alignment tools would ignore them)<sup class="footnote-ref"><a href="#fn4" id="fnref4:1">[4:1]</a></sup></li>
<li>genetic variation
<ul>
<li>SCGB1A1 gene(RNA-seq) <sup class="footnote-ref"><a href="#fn4" id="fnref4:2">[4:2]</a></sup></li>
<li>absent or misassembled in the reference genome<sup class="footnote-ref"><a href="#fn6" id="fnref6">[6]</a></sup></li>
</ul>
</li>
<li>Not well-identified Genes<sup class="footnote-ref"><a href="#fn4" id="fnref4:3">[4:3]</a></sup></li>
<li>non-coding RNA (RNA-seq)<sup class="footnote-ref"><a href="#fn4" id="fnref4:4">[4:4]</a></sup></li>
</ul>
</li>
<li>Technical Issue
<ul>
<li>primer-dimer formation<sup class="footnote-ref"><a href="#fn7" id="fnref7">[7]</a></sup></li>
<li>Cloning/Expression Vecters</li>
</ul>
</li>
</ul>
<p>Once we realized the existence and significance of unmapped reads, we need to find a way to annotate and analyze them. <strong>Kraken2</strong> and <strong>bracken</strong> could be good choose.</p>
<h2 id="Published-Pipeline">Published Pipeline</h2>
<h3 id="A-Gouin-2014">A Gouin, 2014<sup class="footnote-ref"><a href="#fn1" id="fnref1:1">[1:1]</a></sup>:</h3>
<p>Host: 33 pea aphid<br>
Host Genome; mitochondrial genome; primary bacterial symbiont; several secondary symbiont genomes reported for the pea aphid.</p>
<table>
<thead>
<tr>
<th style="text-align:left">Main Steps</th>
<th style="text-align:center">Software</th>
<th style="text-align:left">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Find unmapped reads</td>
<td style="text-align:center">Bowtie2</td>
<td style="text-align:left">align to Reference Genome</td>
</tr>
<tr>
<td style="text-align:left">Find unmapped reads</td>
<td style="text-align:center">Samtools</td>
<td style="text-align:left">extract unmaped reads</td>
</tr>
<tr>
<td style="text-align:left">Unmapped reads assembly</td>
<td style="text-align:center">Prinseq</td>
<td style="text-align:left">trimme the quality below 20 within a window of 10 nucleotides, and only sequences of at least 66 nucleotides in length were retained</td>
</tr>
<tr>
<td style="text-align:left">Unmapped reads Analysis</td>
<td style="text-align:center">Compareads</td>
<td style="text-align:left">Find similar reads between two sets of reads in an assembly-free manner</td>
</tr>
<tr>
<td style="text-align:left">Unmapped reads assembly</td>
<td style="text-align:center">ABySS</td>
<td style="text-align:left">Unmapped reads de novo assembling</td>
</tr>
<tr>
<td style="text-align:left">Unmapped reads assembly</td>
<td style="text-align:center">Bowtie2</td>
<td style="text-align:left">align unmapped reads to assembled Contigs</td>
</tr>
<tr>
<td style="text-align:left">Contigs analysis</td>
<td style="text-align:center">BLASTClust</td>
<td style="text-align:left">Found homologous large contigous.</td>
</tr>
<tr>
<td style="text-align:left">Contigs analysis</td>
<td style="text-align:center">BLASTn</td>
<td style="text-align:left">Found homologous contigous</td>
</tr>
<tr>
<td style="text-align:left">Unmapped reads assembly</td>
<td style="text-align:center">readFilter</td>
<td style="text-align:left">only reads for which 68% of the length was covered by 31-mers present at least 100 times in the data set were retained</td>
</tr>
<tr>
<td style="text-align:left">Unmapped reads assembly</td>
<td style="text-align:center">SPAdes</td>
<td style="text-align:left">De novo assembly</td>
</tr>
<tr>
<td style="text-align:left">Potentially divergent</td>
<td style="text-align:center">Mummer</td>
<td style="text-align:left">Global aligner, against reference: &gt;80% identity and &gt;500 bp were retained</td>
</tr>
<tr>
<td style="text-align:left">Contigs analysis</td>
<td style="text-align:center">GeneMarkS+</td>
<td style="text-align:left">Predict the protein from de novo contigs.</td>
</tr>
<tr>
<td style="text-align:left">Contigs analysis</td>
<td style="text-align:center">Blastp</td>
<td style="text-align:left">Predicted protein annotation</td>
</tr>
<tr>
<td style="text-align:left">Potentially divergent</td>
<td style="text-align:center">Stampy</td>
<td style="text-align:left">An aligner performs well when divergent is high.</td>
</tr>
<tr>
<td style="text-align:left">Potentially divergent</td>
<td style="text-align:center">GATK</td>
<td style="text-align:left">SNP calling</td>
</tr>
</tbody>
</table>
<p>PS:</p>
<ul>
<li><strong>Why Bowtie2</strong>: the percentage of unmapped reads was higher than for Bowtie2 (on average over the 33 individuals, 6.1% vs 3.7% for BWA and Bowtie2, respectively)</li>
</ul>
<h3 id="DecontaMiner-2019">DecontaMiner, 2019<sup class="footnote-ref"><a href="#fn2" id="fnref2:1">[2:1]</a></sup></h3>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fs12859-019-2684-x/MediaObjects/12859_2019_2684_Fig1_HTML.png?as=webp" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://doi.org/10.1186/s12859-019-2684-x">© Mara Sangiovanni, et al. 2019</a></td>
</tr>
</tbody>
</table>
<h3 id="MicroPro-2019">MicroPro, 2019<sup class="footnote-ref"><a href="#fn3" id="fnref3:1">[3:1]</a></sup></h3>
<ul>
<li>
<ol>
<li>Align reads to Reference-based known microbial by <strong>Centrifuge</strong></li>
</ol>
<ul>
<li>The average mapping rate for each dataset is about 35–40%.</li>
</ul>
</li>
<li>
<ol start="2">
<li>Cross-assembly by: <strong>Megahit</strong> and <strong>Minia 3</strong></li>
</ol>
<ul>
<li>Megahit performed better in real data analysis.</li>
<li><strong>MetaBAT 2.12.1</strong> to binning</li>
<li>Abundance estimate achived by <strong>BWA-aln</strong></li>
</ul>
</li>
<li>
<ol start="3">
<li>Predicting phenotypes using random forests</li>
</ol>
</li>
</ul>
<p>About viral:</p>
<ul>
<li>Step 1: Known viral abundance extraction</li>
<li>Step 2: Unknown viral feature detection</li>
<li>Step 3: Predicting phenotypes based on viral abundance</li>
</ul>
<p>PS: Corss-assembly:</p>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://s1.ax1x.com/2022/09/06/v7KmYF.png" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">Using reads from different samples to assemble: Increasing the coverage of contigs.<a target="_blank" rel="noopener" href="https://scholarworks.iu.edu/dspace/handle/2022/22512">B. Papudeshi; 2018</a></td>
</tr>
</tbody>
</table>
<h3 id="Unmapped-human-RNA-Seq-data-and-their-association-with-cancer">Unmapped human RNA-Seq data and their association with cancer</h3>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://www.embopress.org/cms/asset/350fd448-ae21-4aa2-ab37-41bde5cc69b8/msb156172-fig-0001-m.jpg" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://www.embopress.org/doi/full/10.15252/msb.156172">© Majid Kazemian, 2015</a></td>
</tr>
</tbody>
</table>
<h3 id="Exploring-the-unmapped-DNA-and-RNA-reads-in-a-songbird-genome">Exploring the unmapped DNA and RNA reads in a songbird genome</h3>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fs12864-018-5378-2/MediaObjects/12864_2018_5378_Fig4_HTML.png?as=webp" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-5378-2#Sec15">© Veronika N. Laine, 2015</a></td>
</tr>
</tbody>
</table>
<h3 id="About-Bracken">About Bracken</h3>
<ul>
<li><a target="_blank" rel="noopener" href="https://microbe.net/2017/04/27/why-use-bracken-instead-of-kraken/">Jen Lu; 2017</a>: If a 150bp read is 100% identical to two different species, Kraken will assign it to their lowest common ancestor (LCA), which could be at the genus level or higher. But some reads within a particular sample usually come from the unique (or species-specific) portion of the genome. Bracken uses this information, plus information about the similarity between the sister species, to “push down” reads from the genus level to the species level.</li>
<li>Other background information: <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41592-021-01141-3">Challenges in benchmarking metagenomic profilers; Nature Methods</a></li>
</ul>
<p>To be notices, Kraken2 and Bracken is DNA-to-DNA strategy. The abundance of the reads is relative to the nucleotide, not species individual.</p>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41592-021-01141-3/MediaObjects/41592_2021_1141_Fig1_HTML.png?as=webp" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41592-021-01141-3">© Zheng Sun, 2021</a></td>
</tr>
</tbody>
</table>
<h3 id="Why-Kraken">Why Kraken</h3>
<p>Lu J., et al. showed that comparing to QiiMe, Kraken2 has uncompetable speed in annotation and the results from kraken-bracken are even better than QiiMe<sup class="footnote-ref"><a href="#fn8" id="fnref8">[8]</a></sup>. I think it’s time for QiiMe move a room for Kraken2 now.</p>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fs40168-020-00900-2/MediaObjects/40168_2020_900_Fig1_HTML.png?as=webp" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1186/s40168-020-00900-2">© Lu J.</a></td>
</tr>
</tbody>
</table>
<h2 id="Unmapped-reads">Unmapped reads</h2>
<div class="admonition note"><p class="admonition-title">Prepare your working directories
</p><pre>
.
├── <font color="skytblue">Bam</font>
├── <font color="skytblue">FQ_unmapped</font>
├── <font color="skytblue">Parasites</font>
├── <font color="skytblue">Log</font>
├── Model.sh
└── <font color="skytblue">script</font>
</pre>
</div>
<ul>
<li><code>Bam</code>: Aligned bam files</li>
<li><code>FQ_unmapped</code>: Storing the unmapped reads from <code>.bam</code></li>
<li><code>Parasites</code>: Directory for store <code>Clean.fa</code> which contains some contamination seqs and cloning vectors.</li>
<li><code>Log</code>: Directory for logs</li>
<li><code>Model.sh</code>: A script with slurm parameters.</li>
<li><code>script</code>: script for</li>
</ul>
<p>A Quick Pipeline from <a target="_blank" rel="noopener" href="https://genomics.sschmeier.com/ngs-taxonomic-investigation/index.html#preface">Genomics Tutorial; 2020</a>:</p>
<h3 id="Prepare-Database">Prepare Database</h3>
<div class="admonition warning"><p class="admonition-title">Prepare your vector sequence in `vector.fa`
</p></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">mkdir Parasites<br><span class="hljs-built_in">cd</span> Parasites<br>wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/010/205/GCF_000010205.1_ASM1020v1/GCF_000010205.1_ASM1020v1_genomic.fna.gz<br>wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/239/435/GCF_000239435.1_ASM23943v1/GCF_000239435.1_ASM23943v1_genomic.fna.gz<br>wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/255/335/GCF_000255335.2_Mocc_1.1/GCF_000255335.2_Mocc_1.1_genomic.fna.gz<br><br>curl -OJX GET <span class="hljs-string">&quot;https://api.ncbi.nlm.nih.gov/datasets/v1/genome/accession/GCA_024678965.1/download?include_annotation_type=GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA&amp;filename=GCA_024678965.1.zip&quot;</span> -H <span class="hljs-string">&quot;Accept: application/zip&quot;</span><br><br>gunzip *<br>sed  <span class="hljs-string">&quot;s/&gt;/&gt;Mite_/&quot;</span> *.fna &gt; Clean.fa<br>unzip GCA_024678965.1.zip<br>sed  <span class="hljs-string">&quot;s/&gt;/&gt;nematodes_/&quot;</span> ncbi_dataset/data/GCA_024678965.1/GCA_024678965.1_AAFC-BrLi-01_v2_genomic.fna  &gt;&gt; Clean.fa<br>cat  vector.fa &gt;&gt; Clean.fa<br></code></pre></td></tr></table></figure></div>
<h3 id="Unmapped-reads-extraction">Unmapped reads extraction</h3>
<ul>
<li><mark><strong>Align to Reference Genome → Unmapped reads extract → Kraken2 annotation → Bracken Abundance estimation.</strong></mark></li>
</ul>
<p>Technically, you can find the unmapped reads by <code>samtools view *.bam| awk '$3==&quot;*&quot;'</code>. But I don’t know why the number of reads is less than the result from <code>samtools stats</code>. So, the easiest way to achieve this is:</p>
<ul>
<li>Single end: <code>samtools fastq -@ 8 -f 4 Bam/S10_aln_pe_bqsr.bam &gt; s.fq.gz</code></li>
<li>Paired end:<br>
<code>samtools fastq -@ 8 -f 4 Bam/10_aln_pe_bqsr.bam -1 1.fq.gz -2 2.fq.gz -s s.fq.gz</code></li>
</ul>
<p>In this code, reads would be split into 3 files: <code>1.fq.gz</code>, <code>2.fq.gz</code>, and <code>s.fq.gz</code>. The first two are paired reads and the 3rd is unpaired reads. <mark>In Kraken classification under paired parameter, unpaired reads in fq file would be ignored</mark>. So, we need to store them independently and run kraken twice.</p>
<p>If some samples are single ends, the result cannot be stored with this command. But the reads would be printed into log file: <code>Log/*.out</code>. So, we can compress it into <code>fq.gz</code></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">module load samtools<br><br>SUFFIX=_aln_pe_bqsr.bam<br>BAM=Bam<br>OUT_dir=FQ_unmapped<br><br>mkdir <span class="hljs-variable">$OUT_dir</span> script Log<br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$BAM</span>/*.bam);<br>  <span class="hljs-keyword">do</span> SAMPLE=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>| awk -F<span class="hljs-string">&quot;/&quot;</span> <span class="hljs-string">&#x27;&#123;print $NF&#125;&#x27;</span>| sed <span class="hljs-string">&quot;s/<span class="hljs-variable">$SUFFIX</span>//&quot;</span>);<br>    <span class="hljs-keyword">if</span> [ -f <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>\_1.fq.gz ];<br>        <span class="hljs-keyword">then</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> is <span class="hljs-keyword">done</span><br>    <span class="hljs-keyword">else</span><br>        sed <span class="hljs-string">&quot;s/Hi/<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>        <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;samtools fastq -@ 8 -f 4 <span class="hljs-variable">$i</span> -1 <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_1.fq.gz -2 <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_2.fq.gz -s <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_s.fq.gz&quot;</span> &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>        sbatch script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment"># Check for Single Ends</span><br><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(du Log/*.out | grep -v ^0|awk <span class="hljs-string">&#x27;&#123;print $2&#125;&#x27;</span>);<br><span class="hljs-keyword">do</span> SAMPLE=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>| sed <span class="hljs-string">&#x27;s=Log/==;s/.out//&#x27;</span>)<br>    sed <span class="hljs-string">&quot;s/Hi/Single_<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq_s.sh<br>    <span class="hljs-built_in">echo</span> gzip -c <span class="hljs-variable">$i</span> \&gt; <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>\_s.fq.gz &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq_s.sh<br>    sbatch script/<span class="hljs-variable">$SAMPLE</span>\_fq_s.sh<br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment"># Finally, remove emplty fq files</span><br>rm $(du <span class="hljs-variable">$OUT_dir</span>/*| grep <span class="hljs-string">&quot;^4&quot;</span>$<span class="hljs-string">&#x27;\t&#x27;</span>|awk <span class="hljs-string">&#x27;&#123;print $2&#125;&#x27;</span>)<br><br><span class="hljs-comment"># Check the number of samples is correct or not</span><br>ls <span class="hljs-variable">$OUT_dir</span>/*|sed <span class="hljs-string">&#x27;s/_[12s].fq.gz//&#x27;</span>|sort|uniq |wc -l<br></code></pre></td></tr></table></figure></div>
<pre><code>131
</code></pre>
<p>Here I have 131 samples in total. And result is 131 which means I didn’t miss any of them.</p>
<div class="admonition note"><p class="admonition-title">If you are not using slurm syste, this code may help
</p><p><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash"><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(du FQ_unmapped/<em>_s.fq.gz | grep ^4$<span class="hljs-string">'\t'</span>| awk -F<span class="hljs-string">&quot;/&quot;</span> <span class="hljs-string">'{print $NF}'</span>| sed <span class="hljs-string">'s/_s.fq.gz//'</span>);<br>    <span class="hljs-keyword">do</span> i=$(ls <span class="hljs-variable">$BAM</span>/</em>.bam| grep <span class="hljs-variable">$SAMPLE</span>)<br>    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> <span class="hljs-variable">$i</span><br>    samtools fastq -@ 8 -f 4 <span class="hljs-variable">$i</span> &gt; FQ_unmapped/<span class="hljs-variable">$SAMPLE</span>_s.fq.gz<br><span class="hljs-keyword">done</span><br></code></pre></td></tr></table></figure></div></p>
</div>
<h3 id="Align-into-Cleaning-database-and-counts">Align into Cleaning database and counts</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">cd</span> Parasites<br>bwa index Clean.fa<br><span class="hljs-built_in">cd</span> ..<br></code></pre></td></tr></table></figure></div>
<p>Here we already know that the fq ends as <code>1_fq.gz</code> is Paired reads, <code>s_fq.gz</code> is unpaired reads. So, we can do:</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">FQ_DB=FQ_unmapped<br>DB=Parasites<br>DB_FA=Parasites/Clean.fa<br>OUTPUT=Bam_clean<br><br>mkdir <span class="hljs-variable">$OUTPUT</span><br><br><span class="hljs-comment"># Paired</span><br><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$FQ_DB</span>/*_1.fq.gz| sed <span class="hljs-string">&quot;s=<span class="hljs-variable">$FQ_DB</span>/==;s/_[12].fq.gz//&quot;</span>); <span class="hljs-keyword">do</span><br>    <span class="hljs-keyword">if</span> [ -f <span class="hljs-variable">$SAMPLE</span>.P.sorted.bam ]; <span class="hljs-keyword">then</span><br>        <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> is <span class="hljs-keyword">done</span><br>    <span class="hljs-keyword">else</span><br>        <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> can not be find<br>        sed <span class="hljs-string">&quot;s/Hi/<span class="hljs-variable">$SAMPLE</span>\_P_<span class="hljs-variable">$DB</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_P_<span class="hljs-variable">$DB</span>.sh<br>        <span class="hljs-built_in">echo</span> bwa mem <span class="hljs-variable">$DB_FA</span> <span class="hljs-variable">$FQ_DB</span>/<span class="hljs-variable">$SAMPLE</span>\_1.fq.gz <span class="hljs-variable">$FQ_DB</span>/<span class="hljs-variable">$SAMPLE</span>\_1.fq.gz -t 64 \|samtools view -S -b -\|samtools sort \&gt; <span class="hljs-variable">$SAMPLE</span>.P.sorted.bam &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_P_<span class="hljs-variable">$DB</span>.sh<br>        <span class="hljs-built_in">echo</span> samtools index <span class="hljs-variable">$SAMPLE</span>.P.sorted.bam &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_P_<span class="hljs-variable">$DB</span>.sh<br>        sbatch script/<span class="hljs-variable">$SAMPLE</span>\_P_<span class="hljs-variable">$DB</span>.sh<br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment"># Single</span><br><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$FQ_DB</span>/*_s.fq.gz| sed <span class="hljs-string">&quot;s=<span class="hljs-variable">$FQ_DB</span>/==;s/_s.fq.gz//&quot;</span>); <span class="hljs-keyword">do</span><br>    <span class="hljs-keyword">if</span> [ -f <span class="hljs-variable">$SAMPLE</span>.S.sorted.bam ]; <span class="hljs-keyword">then</span><br>        <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> is <span class="hljs-keyword">done</span><br>    <span class="hljs-keyword">else</span><br>        <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> can not be find<br>        Clean<br>        <span class="hljs-built_in">echo</span> bwa mem <span class="hljs-variable">$DB_FA</span> <span class="hljs-variable">$FQ_DB</span>/<span class="hljs-variable">$SAMPLE</span>\_s.fq.gz -t 64 \|samtools view -S -b -\|samtools sort \&gt; <span class="hljs-variable">$SAMPLE</span>.S.sorted.bam &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_S_<span class="hljs-variable">$DB</span>.sh<br>        <span class="hljs-built_in">echo</span> samtools index <span class="hljs-variable">$SAMPLE</span>.S.sorted.bam &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_S_<span class="hljs-variable">$DB</span>.sh<br>        sbatch script/<span class="hljs-variable">$SAMPLE</span>\_S_<span class="hljs-variable">$DB</span>.sh<br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment">## Check unexpected size of file</span><br>rm $(du   *.sorted.bam | awk <span class="hljs-string">&#x27;$1&lt;=100&#x27;</span>| awk <span class="hljs-string">&#x27;&#123;print $2&#125;&#x27;</span>)<br><span class="hljs-comment">## After removed failed files, we can excute the for loop again</span><br><span class="hljs-comment">#.</span><br><span class="hljs-comment">#.</span><br><span class="hljs-comment">#.</span><br><span class="hljs-comment">## finally, we can mv all bam into another directory</span><br>mv *.sorted.* <span class="hljs-variable">$OUTPUT</span><br></code></pre></td></tr></table></figure></div>
<div class="admonition note"><p class="admonition-title">Counting results
</p><p>There are two bam files. One is paired result, another is single end result. For avoid counting them twice in paired result, we can samply <code>sort</code> and <code>uniq</code> them. The only problem is it would consume lots of memory and time.
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash"><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(ls Bam_clean/*.bam| sed <span class="hljs-string">'s/.[PS].sorted.bam//;s=Bam_clean/=='</span>| sort |uniq);<br>    <span class="hljs-keyword">do</span> sed <span class="hljs-string">&quot;s/Hi/<span class="hljs-variable">$SAMPLE</span>_Clean_count/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>.bam_count.sh<br>    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;cat Bam_clean/<span class="hljs-variable">$SAMPLE</span>.[PS].sorted.bam| samtools view| awk '{print $1,$3}'| awk -F&quot;</span>_<span class="hljs-string">&quot; '{print $1}'| sort|uniq| awk '{print $2}'|sort| uniq -c |sed 's/^ *//'&gt; Bam_Stats/<span class="hljs-variable">$SAMPLE</span>.Clean.csv&quot;</span> &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>.bam_count.sh<br>    sbatch script/<span class="hljs-variable">$SAMPLE</span>.bam_count.sh<br><span class="hljs-keyword">done</span><br></code></pre></td></tr></table></figure></div></p>
</div>
<div class="admonition info"><p class="admonition-title">extract unmapped reads again
</p><!--<details>-->
<details><summary>Codes folded for extract reads again</summary>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">module load samtools<br><br>BAM=Bam_clean<br>OUT_dir=FQ_split<br><br>mkdir <span class="hljs-variable">$OUT_dir</span> script Log<br><br><span class="hljs-comment">## paired-end reads</span><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$BAM</span>/*.P.sorted.bam);<br>  <span class="hljs-keyword">do</span> SAMPLE=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>| awk -F<span class="hljs-string">&quot;/&quot;</span> <span class="hljs-string">&#x27;&#123;print $NF&#125;&#x27;</span>| sed <span class="hljs-string">&quot;s/.P.sorted.bam//&quot;</span>);<br>    <span class="hljs-keyword">if</span> [ -f <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>\_1.fq.gz ];<br>        <span class="hljs-keyword">then</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> is <span class="hljs-keyword">done</span><br>    <span class="hljs-keyword">else</span><br>        sed <span class="hljs-string">&quot;s/Hi/<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>        <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;samtools fastq -@ 8 -f 4 <span class="hljs-variable">$i</span> -1 <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_1.fq.gz -2 <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_2.fq.gz -s <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_s.fq.gz&quot;</span> &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>        sbatch script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment"># Check for Single Ends</span><br><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$BAM</span>/*.S.sorted.bam);<br>  <span class="hljs-keyword">do</span> SAMPLE=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>| awk -F<span class="hljs-string">&quot;/&quot;</span> <span class="hljs-string">&#x27;&#123;print $NF&#125;&#x27;</span>| sed <span class="hljs-string">&quot;s/.S.sorted.bam//&quot;</span>);<br>    sed <span class="hljs-string">&quot;s/Hi/<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;samtools fastq -@ 8 -f 4 <span class="hljs-variable">$i</span>| gzip -f &gt; <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$SAMPLE</span>&quot;</span><span class="hljs-string">&quot;_s.fq.gz&quot;</span> &gt;&gt; script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br>    sbatch script/<span class="hljs-variable">$SAMPLE</span>\_fq.sh<br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment"># Finally, remove emplty fq files</span><br>rm $(du <span class="hljs-variable">$OUT_dir</span>/*| grep <span class="hljs-string">&quot;^4&quot;</span>$<span class="hljs-string">&#x27;\t&#x27;</span>|awk <span class="hljs-string">&#x27;&#123;print $2&#125;&#x27;</span>)<br><br><span class="hljs-comment"># Check the number of samples is correct or not</span><br>ls <span class="hljs-variable">$OUT_dir</span>/*|sed <span class="hljs-string">&#x27;s/_[12s].fq.gz//&#x27;</span>|sort|uniq |wc -l<br></code></pre></td></tr></table></figure></div>
</details>
<!--</details>-->
</div>
<h3 id="A-Quick-Pipeline-Example">A Quick Pipeline Example</h3>
<ol>
<li>Create a virtual environment and install tools</li>
<li>Prepare the test data (fastq)</li>
<li>Prepare the database (virus Kraken database)</li>
<li>Run the result</li>
<li>Bracken correction</li>
</ol>
<p>More annotation database could be found: <a target="_blank" rel="noopener" href="https://benlangmead.github.io/aws-indexes/k2">GitHub Kraken2 DataBase</a></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash"><span class="hljs-comment"># 1. Creat a virtual environment</span><br>conda create --yes -n kraken -c bioconda kraken2 bracken<br>conda activate kraken<br><br><span class="hljs-comment"># 2. Download Test data</span><br>wget -O mappings.tar.gz https://osf.io/g5at8/download<br>tar xvzf mappings.tar.gz<br><br>mkdir kraken<br><span class="hljs-built_in">cd</span> kraken<br><br><span class="hljs-comment"># 3. download annotation database</span><br>mkdir Viral<br><span class="hljs-built_in">cd</span> Viral<br>wget https://genome-idx.s3.amazonaws.com/kraken/k2_viral_20220607.tar.gz<br>tar -zxvf k2_viral_20220607.tar.gz<br><span class="hljs-built_in">cd</span> ..<br><br><span class="hljs-comment"># 4. Run the resutls</span><br>DB=Viral<br><br><span class="hljs-comment"># 5. Branken correction</span><br>kraken2 --use-names --threads 4 --db PlusPFP --fastq-input --report <span class="hljs-variable">$DB</span>.report --gzip-compressed --paired ../mappings/evol1.sorted.unmapped.R1.fastq.gz ../mappings/evol1.sorted.unmapped.R2.fastq.gz &gt; <span class="hljs-variable">$DB</span>.kraken<br><br>bracken -d <span class="hljs-variable">$DB</span> -i <span class="hljs-variable">$DB</span>.report -o <span class="hljs-variable">$DB</span>.bracken -r 300 -l S -t 2<br></code></pre></td></tr></table></figure></div>
<h3 id="Pipeline-for-Paired-reads">Pipeline for Paired reads</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">mkdir Kraken_result FQ_split<br><br>FQ=FQ_split<br>DB=PlusPFP<br>DB_L=../kraken/PlusPFP<br><span class="hljs-comment"># paired</span><br><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$FQ</span>/*_1.fq.gz| sed <span class="hljs-string">&quot;s/_1.fq.gz//;s=<span class="hljs-variable">$FQ</span>/==&quot;</span>); <span class="hljs-keyword">do</span><br>    <span class="hljs-keyword">if</span> [ -f Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_P.bracken ];<span class="hljs-keyword">then</span><br>        <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> is <span class="hljs-keyword">done</span><br>    <span class="hljs-keyword">else</span><br>        <span class="hljs-built_in">echo</span> Try <span class="hljs-variable">$SAMPLE</span><br>        sed <span class="hljs-string">&quot;s/Hi/Kra_<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/kraken_<span class="hljs-variable">$SAMPLE</span>\_P.sh<br>        <span class="hljs-built_in">echo</span> kraken2 --use-names --threads 64 --db <span class="hljs-variable">$DB_L</span> --fastq-input --report Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_P.report --gzip-compressed --paired <span class="hljs-variable">$FQ</span>/<span class="hljs-variable">$SAMPLE</span>\_1.fq.gz <span class="hljs-variable">$FQ</span>/<span class="hljs-variable">$SAMPLE</span>\_2.fq.gz \&gt; Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_P.kraken &gt;&gt; script/kraken_<span class="hljs-variable">$SAMPLE</span>\_P.sh<br>        <span class="hljs-built_in">echo</span> bracken -d <span class="hljs-variable">$DB_L</span> -i Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_P.report -o Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_P.bracken -r 300 -l S -t 2 &gt;&gt; script/kraken_<span class="hljs-variable">$SAMPLE</span>\_P.sh<br>        sbatch script/kraken_<span class="hljs-variable">$SAMPLE</span>\_P.sh<br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br><span class="hljs-comment"># sangle</span><br><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(ls <span class="hljs-variable">$FQ</span>/*_s.fq.gz| sed <span class="hljs-string">&quot;s/_s.fq.gz//;s=<span class="hljs-variable">$FQ</span>/==&quot;</span>); <span class="hljs-keyword">do</span><br>    <span class="hljs-keyword">if</span> [ -f Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_S.bracken ];<span class="hljs-keyword">then</span><br>        <span class="hljs-built_in">echo</span> <span class="hljs-variable">$SAMPLE</span> is <span class="hljs-keyword">done</span><br>    <span class="hljs-keyword">else</span><br>        <span class="hljs-built_in">echo</span> Try <span class="hljs-variable">$SAMPLE</span><br>        sed <span class="hljs-string">&quot;s/Hi/Kra_<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/kraken_<span class="hljs-variable">$SAMPLE</span>\_S.sh<br>        <span class="hljs-built_in">echo</span> kraken2 --use-names --threads 64 --db <span class="hljs-variable">$DB_L</span> --fastq-input --report Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_S.report --gzip-compressed --paired <span class="hljs-variable">$FQ</span>/<span class="hljs-variable">$SAMPLE</span>\_s.fq.gz <span class="hljs-variable">$FQ</span>/<span class="hljs-variable">$SAMPLE</span>\_s.fq.gz \&gt; Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_S.kraken &gt;&gt; script/kraken_<span class="hljs-variable">$SAMPLE</span>\_S.sh<br>        <span class="hljs-built_in">echo</span> bracken -d <span class="hljs-variable">$DB_L</span> -i Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_S.report -o Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_S.bracken -r 300 -l S -t 2 &gt;&gt; script/kraken_<span class="hljs-variable">$SAMPLE</span>\_S.sh<br>        sbatch script/kraken_<span class="hljs-variable">$SAMPLE</span>\_S.sh<br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br></code></pre></td></tr></table></figure></div>
<p>For other levels:<br>
Levels: D → K → P → C → O → F → G → S<br>
EXP:</p>
<pre>
D	Eukaryota
K	  Metazoa
P	   Chordata
C	    Mammalia
O	     Primates
F	      Hominidae
G	       Homo
S	        Homo sapiens
</pre>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">DB=PlusPFP<br>LEVEL=G<br><span class="hljs-keyword">for</span> SAMPLE <span class="hljs-keyword">in</span> $(ls FQ_split/*.fq.gz| sed <span class="hljs-string">&#x27;s/_[12].fq.gz//;s=FQ_split/==&#x27;</span>|sort |uniq); <span class="hljs-keyword">do</span><br>    bracken -d <span class="hljs-variable">$DB</span> -i Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>.report -o Kraken_result/<span class="hljs-variable">$SAMPLE</span>\_<span class="hljs-variable">$DB</span>\_<span class="hljs-variable">$LEVEL</span>.bracken -r 300 -l <span class="hljs-variable">$LEVEL</span> -t 2<br><span class="hljs-keyword">done</span><br></code></pre></td></tr></table></figure></div>
<p>Combine them into one csv:</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">grep . Kraken_result/*.bracken| head -n 1| sed <span class="hljs-string">&#x27;s/:/\t/;s=Kraken_result/==;s/_PlusPFP.bracken//&#x27;</span> &gt; bracken.csv<br>grep -v <span class="hljs-string">&quot;taxonomy_id&quot;</span> Kraken_result/*.bracken| sed <span class="hljs-string">&#x27;s/:/\t/;s=Kraken_result/==;s/_PlusPFP.bracken//&#x27;</span> &gt;&gt; bracken.csv<br><br>grep <span class="hljs-string">&quot;sequences unclassified&quot;</span> Log/Kra_*.err| grep -v nan| awk <span class="hljs-string">&#x27;&#123;print $1,$2,$NF&#125;&#x27;</span>| sed <span class="hljs-string">&#x27;s/[:/()%]//g;s/LogKra_//;s/.err//&#x27;</span> &gt; Unclassed.csv<br></code></pre></td></tr></table></figure></div>
<h3 id="Extract-unclassified-reads">Extract unclassified reads</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash"><span class="hljs-comment"># record unclassified reads</span><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls Kraken_result/*.kraken);<br>    <span class="hljs-keyword">do</span> OUT=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>| sed <span class="hljs-string">&#x27;s=Kraken_result/==;s/.kraken/.uc.list/&#x27;</span>)<br>    grep ^U <span class="hljs-variable">$i</span>| awk <span class="hljs-string">&#x27;&#123;print $2&#125;&#x27;</span> &gt; Log/<span class="hljs-variable">$OUT</span><br><span class="hljs-keyword">done</span><br><br>RED_dir=FQ_split<br>OUT_dir=FQ_split2<br>mkdir <span class="hljs-variable">$OUT_dir</span><br>mkdir <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$RED_dir</span><br><br><span class="hljs-comment"># extract unclassified reads with seqkit</span><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls Log/*.uc.list)<br>    <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>;<br>    <span class="hljs-keyword">if</span> [ $(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>| sed <span class="hljs-string">&#x27;s/_S.uc.list//&#x27;</span>) != <span class="hljs-variable">$i</span> ] ; <span class="hljs-keyword">then</span><br>        sed <span class="hljs-string">&quot;s/Hi/seqkit_S_<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_S.sh<br>        READ_1=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> | sed <span class="hljs-string">&quot;s=Log=<span class="hljs-variable">$RED_dir</span>=;s/_PlusPFP_P.uc.list/_s.fq.gz/&quot;</span>)<br>        seqkit grep -n -f <span class="hljs-variable">$i</span> <span class="hljs-variable">$READ_1</span> &gt; <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$READ_1</span><br>    <span class="hljs-keyword">else</span><br>        sed <span class="hljs-string">&quot;s/Hi/seqkit_S_<span class="hljs-variable">$SAMPLE</span>/&quot;</span> Model.sh &gt; script/<span class="hljs-variable">$SAMPLE</span>\_P.sh<br>        READ_1=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> | sed <span class="hljs-string">&quot;s=Log=<span class="hljs-variable">$RED_dir</span>=;s/_PlusPFP_P.uc.list/_1.fq.gz/&quot;</span>)<br>        READ_2=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> | sed <span class="hljs-string">&quot;s=Log=<span class="hljs-variable">$RED_dir</span>=;s/_PlusPFP_P.uc.list/_2.fq.gz/&quot;</span>)<br>        seqkit grep -n -f <span class="hljs-variable">$i</span> <span class="hljs-variable">$READ_1</span> &gt; <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$READ_1</span><br>        seqkit grep -n -f <span class="hljs-variable">$i</span> <span class="hljs-variable">$READ_2</span> &gt; <span class="hljs-variable">$OUT_dir</span>/<span class="hljs-variable">$READ_2</span><br>    <span class="hljs-keyword">fi</span><br><span class="hljs-keyword">done</span><br><br></code></pre></td></tr></table></figure></div>
<h3 id="Visualization-in-R">Visualization in R</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(ggplot2)<br>library(ggrepel)<br><br>TB &lt;- read.table(<span class="hljs-string">&quot;Unclassed.csv&quot;</span>)<br>TB$Total &lt;- TB$V2/TB$V3*<span class="hljs-number">100</span><br><br>ggplot(TB, aes(x=Total, y= V3)) + geom_point() +<br>    geom_text_repel(aes(label= V1)) + theme_bw() +<br>    coord_cartesian(ylim = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-number">100</span>), expand = <span class="hljs-built_in">F</span>)<br></code></pre></td></tr></table></figure></div>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://s1.ax1x.com/2022/09/20/xPH7VK.png" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">Ratio of unclassified Reads</td>
</tr>
</tbody>
</table>
<div class="admonition note"><p class="admonition-title">Download SuperKingdom table from NCBI
</p><p><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">wget ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxcat.tar.Z<br>tar -zxvf taxcat.tar.Z<br></code></pre></td></tr></table></figure></div></p>
</div>
<table>
<thead>
<tr>
<th style="text-align:left">Abbr.</th>
<th style="text-align:left">Kingdom</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">A</td>
<td style="text-align:left">Archaea</td>
</tr>
<tr>
<td style="text-align:left">B</td>
<td style="text-align:left">Bacteria</td>
</tr>
<tr>
<td style="text-align:left">E</td>
<td style="text-align:left">Eukaryota</td>
</tr>
<tr>
<td style="text-align:left">V</td>
<td style="text-align:left">Viruses and Viroids</td>
</tr>
<tr>
<td style="text-align:left">U</td>
<td style="text-align:left">Unclassified</td>
</tr>
<tr>
<td style="text-align:left">O</td>
<td style="text-align:left">Other (exp. Plasmids)</td>
</tr>
</tbody>
</table>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(ggplot2)<br>library(reshape2)<br>library(stringr)<br>library(pheatmap)<br>library(ggrepel)<br><br>TB &lt;- read.csv(<span class="hljs-string">&quot;bracken.csv&quot;</span>, sep= <span class="hljs-string">&#x27;\t&#x27;</span>, header = <span class="hljs-built_in">T</span>)<br>Kingdom &lt;- read.table(<span class="hljs-string">&quot;categories.dmp&quot;</span>)<br><br>TB$SK &lt;- Kingdom$V1[match(TB$taxonomy_id, Kingdom$V2)]<br>colnames(TB)[<span class="hljs-number">1</span>]=<span class="hljs-string">&quot;Sample&quot;</span><br><br>SAMPLE = head(unique(TB$Sample), <span class="hljs-number">10</span>)<br><span class="hljs-comment">#SAMPLE = unique(TB$Sample[-grep(&quot;[AT][TA]|[Hh][Oo][Ss][tT]|NICD|G17&quot;, TB$Sample)])</span><br>TMP &lt;-  TB[TB$Sample %in% SAMPLE,]<br>TMP$Sample &lt;- as.factor(<span class="hljs-built_in">as.character</span>(TMP$Sample))<br><span class="hljs-comment">#TMP$Sample &lt;- factor(TMP$Sample, levels = unique(TMP$Sample)[c(1:10,14:18,21,11:13,19:20)])</span><br><br><br><br><br>TB_w &lt;- reshape(TB[<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>,<span class="hljs-number">8</span>)], timevar =  <span class="hljs-string">&quot;Sample&quot;</span>,  idvar = <span class="hljs-string">&#x27;name&#x27;</span>, direction = <span class="hljs-string">&#x27;wide&#x27;</span>)<br>row.names(TB_w) &lt;- TB_w[[<span class="hljs-number">1</span>]]<br>TB_w &lt;- TB_w[-<span class="hljs-number">1</span>]<br>colnames(TB_w) &lt;- str_replace(colnames(TB_w), <span class="hljs-string">&quot;fraction_total_reads.&quot;</span>, <span class="hljs-string">&quot;&quot;</span>)<br>TB_w[<span class="hljs-built_in">is.na</span>(TB_w)] &lt;- <span class="hljs-number">0</span><br><br><br><br>TB_wS &lt;- as.data.frame(t(scale(t(TB_w))))<br>TB_wS&lt;- TB_wS[!<span class="hljs-built_in">is.na</span>(TB_wS[[<span class="hljs-number">1</span>]]),]<br>pheatmap(TB_wS)<br><br><span class="hljs-comment"># PCA</span><br>TB_wS.pca &lt;- prcomp(t(TB_wS),scale. = <span class="hljs-built_in">F</span>)<br>TB_wS.pca= data.frame(TB_wS.pca$x)<br><br>TB_wS.pca$Group = <span class="hljs-string">&quot;Other&quot;</span><br>TB_wS.pca$Group[grep(<span class="hljs-string">&quot;MH&quot;</span>,row.names(TB_wS.pca))] &lt;- <span class="hljs-string">&quot;MH&quot;</span><br>TB_wS.pca$Group[grep(<span class="hljs-string">&quot;FH&quot;</span>,row.names(TB_wS.pca))] &lt;- <span class="hljs-string">&quot;FH&quot;</span><br><br>ggplot(TB_wS.pca, aes(PC1,PC2, color=Group)) +<br>    geom_point()+ theme_bw() +<br>    geom_text_repel(aes(label=row.names(TB_wS.pca)))<br><br>ggplot(TB_wS.pca[TB_wS.pca$Group %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FH&quot;</span>, <span class="hljs-string">&quot;MH&quot;</span>),], aes(PC1,PC2, color=Group)) +<br>    geom_point()+ theme_bw() +<br>    geom_text_repel(aes(label=row.names(TB_wS.pca[TB_wS.pca$Group %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FH&quot;</span>, <span class="hljs-string">&quot;MH&quot;</span>),])))+<br>    stat_ellipse(lwd=<span class="hljs-number">1</span>,level = <span class="hljs-number">0.75</span>)<br><br><span class="hljs-comment">#ggplot(TB, aes(Sample, fraction_total_reads, fill=name)) + geom_bar(stat = &#x27;identity&#x27;, position = &#x27;fill&#x27;)</span><br></code></pre></td></tr></table></figure></div>
<table>
<thead>
<tr>
<th style="text-align:center">Type</th>
<th style="text-align:center">Graphics</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">Pheatmap</td>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/09/02/vICql4.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">PCA</td>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/09/02/vICL6J.png" alt=""></td>
</tr>
</tbody>
</table>
<div class="admonition note"><p class="admonition-title">Barplot
</p><p><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">ggplot(TB, aes(Sample, new_est_reads, fill=name)) + theme_bw()+<br>    geom_bar(stat = <span class="hljs-string">'identity'</span>, position = <span class="hljs-string">&quot;fill&quot;</span>, show.legend = <span class="hljs-built_in">F</span>)+<br>    theme(axis.text.x = element_text(angle = <span class="hljs-number">90</span>, hjust = <span class="hljs-number">1</span>, vjust = <span class="hljs-number">.5</span>))+<br>    facet_grid(SK~.)+ geom_text_repel(aes(label=name), color = <span class="hljs-string">&quot;black&quot;</span>,<br>    show.legend = <span class="hljs-built_in">F</span>, position = position_fill(vjust= <span class="hljs-number">.5</span>))<br></code></pre></td></tr></table></figure></div>
<img src="https://s1.ax1x.com/2022/09/21/xPLAun.png" alt="" /></p>
</div>
<h2 id="Relationship-with-unmapped-reads-and-unclassified-reads">Relationship with unmapped reads and unclassified reads</h2>
<pre>
.
├── Bam_stats
│   └── SAMPLE.stats.csv
└── Kraken_result
    ├── SAMPLE_PlusPFP.bracken
    ├── SAMPLE_PlusPFP_bracken.report
    ├── SAMPLE_PlusPFP.kraken
    └── SAMPLE_PlusPFP.report
</pre>
<p><code>Bams_stats</code> is generated by <code>samtools stats SAMPLE.bam &gt; SAMPLE.stats.csv</code></p>
<p>In Kraken, when you use paired reads, it would only show the number of the paired reads which means you only get a half size of the reads compared to the result from <code>samtools stats</code>. As a result, we need to multiply them by 2 during visualization.</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">echo</span> -e <span class="hljs-string">&quot;Sample\tTotal\tUnmapped\tClassified\tUnClassified&quot;</span> &gt; Counts_Ratio.csv<br><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls Bam_stats/| sed <span class="hljs-string">&#x27;s/.stats.csv//&#x27;</span>);<br>    <span class="hljs-keyword">do</span><br>    UNMAP=$(grep -w <span class="hljs-string">&quot;reads unmapped&quot;</span> Bam_stats/<span class="hljs-variable">$i</span>.stats.csv| awk -F<span class="hljs-string">&quot;\t&quot;</span> <span class="hljs-string">&#x27;&#123;print $3&#125;&#x27;</span>)<br>    TOTAL=$(grep -w <span class="hljs-string">&quot;raw total sequences:&quot;</span> Bam_stats/<span class="hljs-variable">$i</span>.stats.csv| awk -F<span class="hljs-string">&quot;\t&quot;</span> <span class="hljs-string">&#x27;&#123;print $3&#125;&#x27;</span>)<br>    CLASF=$(grep -c ^C Kraken_result/<span class="hljs-variable">$i</span>\_PlusPFP.kraken)<br>    UNCLF=$(grep -c ^U Kraken_result/<span class="hljs-variable">$i</span>\_PlusPFP.kraken)<br>    <span class="hljs-built_in">echo</span> -e <span class="hljs-string">&quot;<span class="hljs-variable">$i</span>\t<span class="hljs-variable">$TOTAL</span>\t<span class="hljs-variable">$UNMAP</span>\t<span class="hljs-variable">$CLASF</span>\t<span class="hljs-variable">$UNCLF</span>&quot;</span><br><span class="hljs-keyword">done</span>  &gt;&gt; Counts_Ratio.csv<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(ggplot2)<br>library(reshape2)<br>library(pheatmap)<br>library(ggplotify)<br>library(ggkaboom)<br><br>TB &lt;- read.table(<span class="hljs-string">&quot;Counts_Ratio.csv&quot;</span>, header =<span class="hljs-built_in">T</span>)<br><span class="hljs-comment">#TB$Sample &lt;- as.character(as.numeric(TB$Sample))</span><br>TB$Classified &lt;- TB$Classified*<span class="hljs-number">2</span><br>TB$UnClassified &lt;- TB$UnClassified*<span class="hljs-number">2</span><br><br>TB2 &lt;- TB<br>TB2[<span class="hljs-number">3</span>:<span class="hljs-number">5</span>] &lt;- TB2[<span class="hljs-number">3</span>:<span class="hljs-number">5</span>]/TB2[[<span class="hljs-number">2</span>]]<br>TB2[<span class="hljs-number">2</span>:<span class="hljs-number">5</span>] &lt;- scale(TB2[<span class="hljs-number">2</span>:<span class="hljs-number">5</span>])<br>row.names(TB2) &lt;- TB2[[<span class="hljs-number">1</span>]]<br>TB2 &lt;-TB2[-<span class="hljs-number">1</span>]<br>P &lt;- pheatmap(t(TB2))<br>as.ggplot(P)<br>ggsave(<span class="hljs-string">&quot;img/Number_Pheat.png&quot;</span>, h= <span class="hljs-number">7.5</span>, w = <span class="hljs-number">12</span>)<br><br><br><span class="hljs-comment"># for Bar fill plot</span><br>TB_bar &lt;- TB<br>TB_bar$Mapped &lt;- TB_bar$Total -TB_bar$ Unmapped<br>TB_bar &lt;- TB_bar[<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Sample&quot;</span>, <span class="hljs-string">&quot;Mapped&quot;</span>, <span class="hljs-string">&quot;Classified&quot;</span>, <span class="hljs-string">&quot;UnClassified&quot;</span>)]<br><span class="hljs-comment">#TB_bar_r &lt;- TB_bar[2:4]/TB$Total</span><br><span class="hljs-comment">#row.names(TB_bar_r) &lt;- TB$Sample</span><br><br>TB.hclust &lt;- hclust(dist(TB2))<br>TB_L &lt;- melt(TB_bar)<br>TB_L$Sample &lt;- factor(TB_L$Sample, levels=TB.hclust$labels[TB.hclust$order])<br>P &lt;- ggplot(TB_L, aes(Sample, value, fill= variable)) +<br>    geom_bar(stat = <span class="hljs-string">&#x27;identity&#x27;</span>, position = <span class="hljs-string">&#x27;fill&#x27;</span>)+theme_bw() +<br>    scale_fill_manual( values= <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;OliveDrab2&quot;</span>, <span class="hljs-string">&quot;deepskyblue&quot;</span>, <span class="hljs-string">&quot;salmon&quot;</span>)) +<br>    theme(axis.text.x = element_text(angle= <span class="hljs-number">270</span>, hjust=<span class="hljs-number">0</span>, vjust = <span class="hljs-number">.5</span>))<br><br>Kaboom_break(P, <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,  <span class="hljs-number">0.04</span>, <span class="hljs-number">0</span>, <span class="hljs-number">1</span>), R=<span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>, <span class="hljs-number">1</span>))<br>ggsave(<span class="hljs-string">&quot;img/Number_Bar.png&quot;</span>, h= <span class="hljs-number">7.5</span>, w = <span class="hljs-number">12</span>)<br><br><span class="hljs-comment"># Not Fill</span><br>TB_bar2 &lt;- melt(TB[<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Sample&quot;</span>, <span class="hljs-string">&quot;Total&quot;</span>, <span class="hljs-string">&quot;Unmapped&quot;</span>, <span class="hljs-string">&quot;UnClassified&quot;</span>)])<br>TB_bar2$Sample &lt;- factor(TB_bar2$Sample, levels=TB.hclust$labels[TB.hclust$order])<br><br>P &lt;- ggplot(TB_bar2, aes(Sample, value, fill= variable)) +<br>    geom_bar(stat = <span class="hljs-string">&#x27;identity&#x27;</span>, position = position_identity() )+<br>    scale_fill_brewer(palette = <span class="hljs-string">&quot;Set2&quot;</span>)+ theme_bw() +<br>    theme(axis.text.x = element_text(angle= <span class="hljs-number">270</span>, hjust=<span class="hljs-number">0</span>, vjust = <span class="hljs-number">.5</span>))<br>Kaboom_break(P, <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,  <span class="hljs-number">20000000</span>, <span class="hljs-number">0</span>, <span class="hljs-number">200000000</span>), R=<span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>, <span class="hljs-number">1</span>))<br>ggsave(<span class="hljs-string">&quot;img/Number_raw.png&quot;</span>, h= <span class="hljs-number">7.5</span>, w = <span class="hljs-number">12</span>)<br><br><span class="hljs-comment">## Unclassified</span><br>TB_L$Sample &lt;- factor(TB_L$Sample, levels=TB$Sample[order(TB$UnClassified/TB$Classified)])<br><br>P &lt;- ggplot(TB_L[TB_L$variable!=<span class="hljs-string">&quot;Mapped&quot;</span>,], aes(Sample, value, fill= variable)) +<br>    geom_bar(stat = <span class="hljs-string">&#x27;identity&#x27;</span>, position = <span class="hljs-string">&#x27;fill&#x27;</span>)+theme_bw() +<br>    scale_fill_manual( values= <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;deepskyblue&quot;</span>, <span class="hljs-string">&quot;salmon&quot;</span>)) +<br>    theme(axis.text.x = element_text(angle= <span class="hljs-number">270</span>, hjust=<span class="hljs-number">0</span>, vjust = <span class="hljs-number">.5</span>))<br><br>ggsave(<span class="hljs-string">&quot;img/Number_Bar_classR.png&quot;</span>, h= <span class="hljs-number">7.5</span>, w = <span class="hljs-number">12</span>)<br></code></pre></td></tr></table></figure></div>
<table>
<thead>
<tr>
<th style="text-align:left">Header One</th>
<th style="text-align:left">Header Two</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left"><img src="https://s1.ax1x.com/2022/09/15/vzutY9.png" alt=""></td>
<td style="text-align:left"><img src="https://s1.ax1x.com/2022/09/15/vzuYFJ.png" alt=""></td>
</tr>
<tr>
<td style="text-align:left"><img src="https://s1.ax1x.com/2022/09/15/vzKZnK.png" alt=""></td>
<td style="text-align:left"><img src="https://s1.ax1x.com/2022/09/15/vzuGo4.png" alt=""></td>
</tr>
</tbody>
</table>
<h2 id="Full-Taxonomic-information-based-on-ID">Full Taxonomic information based on ID</h2>
<p>Some times, we’d like to analyze the Eucariaotic or Viral independently. Them, we need to find the phlums of them based on taxonomic ID. <a target="_blank" rel="noopener" href="http://etetoolkit.org/download/">etetoolkit</a> Can help find the phylums efficiently.</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">conda create -n ete3 python=3<br>conda activate ete3<br>conda install -c etetoolkit ete3 ete_toolchain<br>ete3 build check<br><br>ete3 ncbiquery --search 1204725 2162 13000163 420247 --info<br></code></pre></td></tr></table></figure></div>
<p>By aquriring specific information, SuperKingdom here, we can also use “E-utilities” from NCBI</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">efetch -db taxonomy -id 9606,1234,81726 -format xml | \<br>xtract -pattern Taxon -tab <span class="hljs-string">&quot;,&quot;</span> -first TaxId ScientificName \<br>-group Taxon -KING <span class="hljs-string">&quot;(-)&quot;</span> -PHYL <span class="hljs-string">&quot;(-)&quot;</span> -CLSS <span class="hljs-string">&quot;(-)&quot;</span> -ORDR <span class="hljs-string">&quot;(-)&quot;</span> -FMLY <span class="hljs-string">&quot;(-)&quot;</span> -GNUS <span class="hljs-string">&quot;(-)&quot;</span> \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:superkingdom&quot;</span> -SKIN ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:kingdom&quot;</span> -KING ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:phylum&quot;</span> -PHYL ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:class&quot;</span> -CLSS ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:order&quot;</span> -ORDR ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:family&quot;</span> -FMLY ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:genus&quot;</span> -GNUS ScientificName \<br>-group Taxon -tab <span class="hljs-string">&quot;,&quot;</span> -element <span class="hljs-string">&quot;&amp;SKIN&quot;</span>  <span class="hljs-string">&quot;&amp;KING&quot;</span> <span class="hljs-string">&quot;&amp;PHYL&quot;</span> <span class="hljs-string">&quot;&amp;CLSS&quot;</span> <span class="hljs-string">&quot;&amp;ORDR&quot;</span> <span class="hljs-string">&quot;&amp;FMLY&quot;</span> <span class="hljs-string">&quot;&amp;GNUS&quot;</span><br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">ID=$(awk -F<span class="hljs-string">&quot;\t&quot;</span> <span class="hljs-string">&#x27;&#123;print $3&#125;&#x27;</span> bracken.csv | sort |uniq| grep -v tax| tr <span class="hljs-string">&#x27;\n&#x27;</span> <span class="hljs-string">&quot;,&quot;</span>| sed <span class="hljs-string">&#x27;s/,$/\n/&#x27;</span>)<br><br>efetch -db taxonomy -id <span class="hljs-variable">$ID</span> -format xml | \<br>xtract -pattern Taxon -tab <span class="hljs-string">&quot;,&quot;</span> -first TaxId ScientificName \<br>-group Taxon -KING <span class="hljs-string">&quot;(-)&quot;</span> -PHYL <span class="hljs-string">&quot;(-)&quot;</span> -CLSS <span class="hljs-string">&quot;(-)&quot;</span> -ORDR <span class="hljs-string">&quot;(-)&quot;</span> -FMLY <span class="hljs-string">&quot;(-)&quot;</span> -GNUS <span class="hljs-string">&quot;(-)&quot;</span> \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:superkingdom&quot;</span> -SKIN ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:kingdom&quot;</span> -KING ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:phylum&quot;</span> -PHYL ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:class&quot;</span> -CLSS ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:order&quot;</span> -ORDR ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:family&quot;</span> -FMLY ScientificName \<br>-block <span class="hljs-string">&quot;*/Taxon&quot;</span> -match <span class="hljs-string">&quot;Rank:genus&quot;</span> -GNUS ScientificName \<br>-group Taxon -tab <span class="hljs-string">&quot;\t&quot;</span> -element <span class="hljs-string">&quot;&amp;SKIN&quot;</span>  <span class="hljs-string">&quot;&amp;KING&quot;</span> <span class="hljs-string">&quot;&amp;PHYL&quot;</span> <span class="hljs-string">&quot;&amp;CLSS&quot;</span> <span class="hljs-string">&quot;&amp;ORDR&quot;</span> <span class="hljs-string">&quot;&amp;FMLY&quot;</span> <span class="hljs-string">&quot;&amp;GNUS&quot;</span> &gt; Taxonomy.csv<br><br></code></pre></td></tr></table></figure></div>
<h1>Other plots</h1>
<p><a target="_blank" rel="noopener" href="https://taylorreiter.github.io/2017-07-28-Taxonomy-from-Species-Name-in-R/">Taxonomy from species name in r</a></p>
<style>
pre {
  background-color:#38393d;
  color: #5fd381;
}
</style>
<hr class="footnotes-sep">
<section class="footnotes">
<ol class="footnotes-list">
<li id="fn1" class="footnote-item"><p>Gouin, A., Legeai, F., Nouhaud, P. et al. Whole-genome re-sequencing of non-model organisms: lessons from unmapped reads. Heredity 114, 494–501 (2015). <a target="_blank" rel="noopener" href="https://doi.org/10.1038/hdy.2014.85">https://doi.org/10.1038/hdy.2014.85</a> <a href="#fnref1" class="footnote-backref">↩︎</a> <a href="#fnref1:1" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn2" class="footnote-item"><p>Sangiovanni, M., Granata, I., Thind, A. et al. From trash to treasure: detecting unexpected contamination in unmapped NGS data. BMC Bioinformatics 20 (Suppl 4), 168 (2019). <a target="_blank" rel="noopener" href="https://doi.org/10.1186/s12859-019-2684-x">https://doi.org/10.1186/s12859-019-2684-x</a> <a href="#fnref2" class="footnote-backref">↩︎</a> <a href="#fnref2:1" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn3" class="footnote-item"><p>Zhu, Z., Ren, J., Michail, S. et al. MicroPro: using metagenomic unmapped reads to provide insights into human microbiota and disease associations. Genome Biol 20, 154 (2019). <a target="_blank" rel="noopener" href="https://doi.org/10.1186/s13059-019-1773-5">https://doi.org/10.1186/s13059-019-1773-5</a> <a href="#fnref3" class="footnote-backref">↩︎</a> <a href="#fnref3:1" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn4" class="footnote-item"><p>Gurgul, A., Szmatoła, T., Ocłe sources of unmapped reads are various: they may derive from characterized or unchoń, E. et al. Another lesson from unmapped reads: in-depth analysis of RNA-Seq reads from various horse tissues. J Appl Genetics 63, 571–581 (2022). <a target="_blank" rel="noopener" href="https://doi.org/10.1007/s13353-022-00705-z">https://doi.org/10.1007/s13353-022-00705-z</a> <a href="#fnref4" class="footnote-backref">↩︎</a> <a href="#fnref4:1" class="footnote-backref">↩︎</a> <a href="#fnref4:2" class="footnote-backref">↩︎</a> <a href="#fnref4:3" class="footnote-backref">↩︎</a> <a href="#fnref4:4" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn5" class="footnote-item"><p>Kazemian, Majid, et al. “Comprehensive assembly of novel transcripts from unmapped human RNA‐Seq data and their association with cancer.” Molecular systems biology 11.8 (2015): 826. <a target="_blank" rel="noopener" href="https://doi.org/10.15252/msb.156172">https://doi.org/10.15252/msb.156172</a> <a href="#fnref5" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn6" class="footnote-item"><p>Laine, V.N., Gossmann, T.I., van Oers, K. et al. Exploring the unmapped DNA and RNA reads in a songbird genome. BMC Genomics 20, 19 (2019). <a target="_blank" rel="noopener" href="https://doi.org/10.1186/s12864-018-5378-2">https://doi.org/10.1186/s12864-018-5378-2</a> <a href="#fnref6" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn7" class="footnote-item"><p>De Bourcy, Charles FA, et al. “A quantitative comparison of single-cell whole genome amplification methods.” PloS one 9.8 (2014): e105585. <a href="#fnref7" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn8" class="footnote-item"><p>Lu J, Salzberg S L. Ultrafast and accurate 16S rRNA microbial community analysis using Kraken 2[J]. Microbiome, 2020, 8(1): 1-11. <a href="#fnref8" class="footnote-backref">↩︎</a></p>
</li>
</ol>
</section>
</div><div class="article-licensing box"><div class="licensing-title"><p>Unmapped reads Annotations</p><p><a href="https://karobben.github.io/2022/08/30/Bioinfor/unmapped-reads/">https://karobben.github.io/2022/08/30/Bioinfor/unmapped-reads/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2022-08-30</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2023-06-06</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/Software/">Software</a><a class="link-muted mr-2" rel="tag" href="/tags/Bioinformatics/">Bioinformatics</a><a class="link-muted mr-2" rel="tag" href="/tags/NGS/">NGS</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2022/09/05/Blog/hexo-butterfly/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">Hexo theme: Butterfly</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2022/08/26/Bioinfor/DNA-seq/"><span class="level-item">DNA Seq Pipeline and General Infor statistics</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">517</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">144</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">158</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><ul class="menu-list"><li><a class="level is-mobile" href="#Why-Unmapped-Reads"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">Why Unmapped Reads</span></span></a></li><li><a class="level is-mobile" href="#Published-Pipeline"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">Published Pipeline</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#A-Gouin-2014"><span class="level-left"><span class="level-item">1.2.1</span><span class="level-item">A Gouin, 2014[1:1]:</span></span></a></li><li><a class="level is-mobile" href="#DecontaMiner-2019"><span class="level-left"><span class="level-item">1.2.2</span><span class="level-item">DecontaMiner, 2019[2:1]</span></span></a></li><li><a class="level is-mobile" href="#MicroPro-2019"><span class="level-left"><span class="level-item">1.2.3</span><span class="level-item">MicroPro, 2019[3:1]</span></span></a></li><li><a class="level is-mobile" href="#Unmapped-human-RNA-Seq-data-and-their-association-with-cancer"><span class="level-left"><span class="level-item">1.2.4</span><span class="level-item">Unmapped human RNA-Seq data and their association with cancer</span></span></a></li><li><a class="level is-mobile" href="#Exploring-the-unmapped-DNA-and-RNA-reads-in-a-songbird-genome"><span class="level-left"><span class="level-item">1.2.5</span><span class="level-item">Exploring the unmapped DNA and RNA reads in a songbird genome</span></span></a></li><li><a class="level is-mobile" href="#About-Bracken"><span class="level-left"><span class="level-item">1.2.6</span><span class="level-item">About Bracken</span></span></a></li><li><a class="level is-mobile" href="#Why-Kraken"><span class="level-left"><span class="level-item">1.2.7</span><span class="level-item">Why Kraken</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Unmapped-reads"><span class="level-left"><span class="level-item">1.3</span><span class="level-item">Unmapped reads</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Prepare-Database"><span class="level-left"><span class="level-item">1.3.1</span><span class="level-item">Prepare Database</span></span></a></li><li><a class="level is-mobile" href="#Unmapped-reads-extraction"><span class="level-left"><span class="level-item">1.3.2</span><span class="level-item">Unmapped reads extraction</span></span></a></li><li><a class="level is-mobile" href="#Align-into-Cleaning-database-and-counts"><span class="level-left"><span class="level-item">1.3.3</span><span class="level-item">Align into Cleaning database and counts</span></span></a></li><li><a class="level is-mobile" href="#A-Quick-Pipeline-Example"><span class="level-left"><span class="level-item">1.3.4</span><span class="level-item">A Quick Pipeline Example</span></span></a></li><li><a class="level is-mobile" href="#Pipeline-for-Paired-reads"><span class="level-left"><span class="level-item">1.3.5</span><span class="level-item">Pipeline for Paired reads</span></span></a></li><li><a class="level is-mobile" href="#Extract-unclassified-reads"><span class="level-left"><span class="level-item">1.3.6</span><span class="level-item">Extract unclassified reads</span></span></a></li><li><a class="level is-mobile" href="#Visualization-in-R"><span class="level-left"><span class="level-item">1.3.7</span><span class="level-item">Visualization in R</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Relationship-with-unmapped-reads-and-unclassified-reads"><span class="level-left"><span class="level-item">1.4</span><span class="level-item">Relationship with unmapped reads and unclassified reads</span></span></a></li><li><a class="level is-mobile" href="#Full-Taxonomic-information-based-on-ID"><span class="level-left"><span class="level-item">1.5</span><span class="level-item">Full Taxonomic information based on ID</span></span></a></li></ul><li><a class="level is-mobile" href="#"><span class="level-left"><span class="level-item">2</span><span class="level-item">Other plots</span></span></a></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Antibody/"><span class="tag">Antibody</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">72</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/CLI-Tools/"><span class="tag">CLI Tools</span><span class="tag">15</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data-Science/"><span class="tag">Data Science</span><span class="tag">11</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fasta/"><span class="tag">Fasta</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immunology/"><span class="tag">Immunology</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG-pathway/"><span class="tag">KEGG pathway</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">21</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matplotlib/"><span class="tag">Matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">27</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">62</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein/"><span class="tag">Protein</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyMol/"><span class="tag">PyMol</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">132</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Pyvista/"><span class="tag">Pyvista</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">86</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripting/"><span class="tag">Scripting</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Seaborn/"><span class="tag">Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sequencing/"><span class="tag">Sequencing</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">47</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI-lib/"><span class="tag">TUI lib</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/VCF/"><span class="tag">VCF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dash/"><span class="tag">dash</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/git/"><span class="tag">git</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/matplotlib/"><span class="tag">matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/vim/"><span class="tag">vim</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://camo.githubusercontent.com/fd83d8c9f8b8dbe85686812e2e186dceec728bbabad2d331aa44e0bfeee8628c/68747470733a2f2f70706f66666963652e6769746875622e696f2f6865786f2d7468656d652d6963617275732f67616c6c6572792f707265766965772e706e673f31" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">76</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">70</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protein-Structure/"><span class="level-start"><span class="level-item">Protein Structure</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fasta-q/"><span class="level-start"><span class="level-item">Fasta/q</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">41</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">11</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">139</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">13</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Immunity/"><span class="level-start"><span class="level-item">Immunity</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">81</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/AI/"><span class="level-start"><span class="level-item">AI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">12</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immunology/"><span class="level-start"><span class="level-item">Immunology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">56</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">39</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2024 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>