---
title: Paper
date: 2022/01/22
excerpt: "argparse for python"
toc: true
priority: 6
---
<span style="font-size:30px; font-weight: bold">Comprehensive physiological, transcriptomic analysis and intestinal microbiota analysis reveal the responses of hybrid grouper (Epinephelus fuscoguttatus♀×E.lanceolatus♂) to fishmeal substitution with black soldier fly larvae</span>


Running Title: effects of BSFL on hybrid grouper


### Abstract

Keywords: Hybrid grouper; Physiological analysis; Transcriptomic analysis; Intestinal microbiota analysis



## Introduction
In the past two decades, many literatures were looking on the alternative protein sources for fish meal (FM)---the optimal protein source in fish feeds. And now, these efforts are still continuing as the raise of the world population expected by 2050 and ﬁsheries production has stabilized over the last decades (FAO, 2016), in which the demand for fish products will increase with an additional 23 million tons of aquatic food by 2030 to maintain current per capita ﬁsh consumption.
Black soldier fly larvae (BSFL, Hermetia illucens) can provide high-value feedstuffs as they are ==rich in protein== (35% to 63%) and well-balanced proﬁle of essential amino acids (AA) profile than soybean meal (Barragan et al., 2017). In addition, they contain a ==greater amount of lipids== (15% to 49%), which can be isolated and used for the preparation of ==biodiesel==, while the rest of the defatted meal could be used as a protein rich source for the feed industry. Since the 1970s, this species (Hermetia illucens L.) has been used as a protein source in animal feed, mainly due to its ability to convert food waste (vegetable, fruit, factory waste, and animal tissues) into high-quality protein (Hale, 1973; Newton et al., 1977). The research and industrial-scale production of BSFL larvae as feed ingredients have been intensiﬁed the last few years (FAO, 2013; Wang and Shelomi, 2017).
Until now, the replacement of FM by BSFL in aquafeeds has not yet been as successful as hoped (Henry et al., 2015). Maximum dietary FM replacement level has ranged from 6 to 25%, depending on the ﬁsh species (Henry et al., 2015), with highest levels being attained for rainbow trout, Oncorhynchus mykiss (Sealey et al., 2011). High inclusion levels of BSFL reduce growth performance of channel catﬁsh, Ictalurus punctatus, rainbow trout, and turbot , Psetta maxima (Kroeckel et al.,2012; Newton et al., 2004; St-Hilaire et al., 2007). In salmon, Salmo salar, dietary supplementation with lysine and methionine allowed dietary BSFL level up to a maximum of 25% (Lock et al., 2016).
Hybrid grouper (Epinephelus fuscoguttatus♀×E.lanceolatus♂) is an important commercial and economic fish with rapid growth, strong disease resistance and high nutritions. This species has been widely cultured in China using land-based farm and sea-cage farm, and is mainly fed with formulated pellet diets. As a typical carnivorous species, hybrid grouper requires high protein, and is heavily dependent on high levels of FM in diet to meet its protein requirement, which leads to higher production costs. Previous reports have shown the nutrient requirements of protein, lipid, fatty acid in this species. But few studies were focused on the effects of substituting FM with BSFL on grouper. It is still unknown whether it is possible to replace FM with BSFL and how much proportion of BSFL was optimal for this species. In a addition, overing the optimal replacement, whether it will have negative effect on its growth performance and immunity and so on. For this purpose, BFSL was used as the source of protein to replace FM in the present study, and the physiological, transcriptomic analysis and intestinal microbiota analysis was observed.


## Materials and methods

### Diet formulation and preparation

Using FM and BSFL as protein sources, fish oil and lecithin as lipid sources and microcrystalline cellulose and pregelatinized starch as a carbohydrate source, six isonitrogenous and isocaloric practical diets was formulated to meet the protein and energy requirements of juvenile grouper. FM (crude protein: 65%) was replaced with BSFL (crude protein: 35%) in each diet (rang from 10% to 50% of FM in 10% interval) in equal protein. These diets was marked as CK(Control group), BSFL10, BSFL20, BSFL30, BSFL40 and BSFL50. Diets formulation is presented as follows (Table 1).
After shredding and passing through a 250 μm screen, all ingredients of each diet were weighed and mixed with fish oil and distilled water (30%, v/w) for a stiff dough. The dough was then pelletized to form 3-mm diameter pellets using the twin-screw extruder (F-26 (II), South China University of Technology, China), and dried at 60 ℃ in an oven. All diets were saved at a freezer (-20°C) until applied. Proximate composition and anti-nutrition factors (ANFS) of diets are showed in Table 1 and the amino acid composition of the diets is presented in Table 2.

### Feeding trial

Prior to the experiment, the use of animal was approved by the "Institutional Animal Care and Use Committee of Hainan Tropical Ocean University " and "Hainan Key Laboratory for Conservation and Utilization of Tropical Marine Fishery Resources"(20161111A1).
Hybrid grouper with the same batch (specific weight would be determined) was purchased from a commercial hatchery (Hainan, China). Before the trial, fish was moved to the experiment destination (Hainan Key Laboratory for Conservation and Utilization of Tropical Marine Fishery Resources, College of life science and ecology, Hainan Tropical Ocean University, Sanya, China), and stocked in indoor sea water circulating system (Recycling Water Aquaculture System Co., Ltd, Qingdao, China) for 2 weeks to adapt to the feeding environment. During this time, the commercial diet is to be used twice daily for fish satiation. In the start of a trial, after fasting for 24 h, 720 fish (specific weight was determined at the beginning of the experiment, average weight 56.49g）are randomly distributed in a 10 m3 indoor sea water circulating system (500 L per tanks, 40 fish per tank) with the triplicate of each diet. The fish are fed two times a day (08:00 am and 16:00 pm). Throughout the trial, feed intake of each diet and mortality of fish are recorded. With a portable multi-meter, water quality was monitored daily and almost remained at following value: temperature (30.15 ± 0.5°C), dissolved oxygen (7.20 ± 0.3 mg L−1), salinity (29.75 ± 0.5‰), pH (7.15 ± 0.24) and total ammonia nitrogen (0.3 ± 0.2 mg L−1). This study followed good laboratory practices (GLP), and the use of animal was approved by the Institutional Animal Care and Use Committee of Hainan Tropical Ocean University.

### Sample collection and calculation formula for growth performance
According to Wilson et al. (2009), rapid cooling and tricaine methanesulfonate (MS222) were used as the methods of euthanasia in grouper in our study [55]. At the beginning of the trial, 20 fish were euthanized, ground these fish into a homogeneous slurry, freeze-dried, reground, and froze immediately at -20°C for detecting whole body composition. After six weeks, all the fish were harvest and weighed after anaesthetizing with MS-222 (50 mg L−1, 3-aminobenzoic ethyl ester acid, Sigma, USA). Specific growth rate (SGR), survival, feed conversion ratio (FCR) and protein efficiency ratio (PER) were calculated. At the end of trial, ten fish each tank were euthanized, and freeze immediately at a froze (-20°C) for determination of whole-body composition.
Eight fish from each tank were anaesthetized, weighted and dissected to remove stomach, liver and intestine. The weight of liver and intraperitoneal fat was recorded to calculated the hepatosomatic index (HSI) and the intraperitoneal ratio (IPR). The dorsal muscle was collected for proximate composition analysis. Feed were recorded after removal of the extra and uneaten feed pellets per meal.
The computational formulas were as follows:
Average body weight gain (BWG, g) = (Total final weight − Total Initial weigh)/amount of fish;
WG (%) = 100 × [(Final weight − Initial weigh) / Initial weight];
SGR (%) = 100 × [(In Final weight) – (ln Initial weight) / total feeding days];
FCR (kg/kg) = Weight of diet / Weight of fish gained;
FCE (%) = 100× (Weight of fish gained / Weight of diet);
Survival rate (%) = 100 (Initial population – Number of dead fish) / Initial population
Daily feed intake (DFI, % d−1) = 100 × Dry feed intake/ [(Initial feed weight + Final weight)/2 × Duration of experiment (d)];
PER = Wet weight gain/Protein intake;
Condition factor (CF, g cm−3) = 100 × (Final weight / Average body length of fish (cm)3);
HSI (%) = 100 × (Wet weight of the liver/ Final weight);
IPR (%) = 100 × (Intraperitoneal fat weigh/ Final weight).

### Biochemical analysis

Dry matter, crude protein, crude lipid, crude ash, moisture contents were analyzed for all ingredients, experimental diets, and fish samples (AOAC 1995). The amino acids of ingredients and diets was determined by an amino acid analyzer (Hitachi 8900; Hitachi, Tokyo, Japan) after hydrolysis in 6N HCl for 22–24 h at 110 °C. For sulfur amino acids determination, oxidation in performic acid for 30 min at 55 °C was conducted before hydrolysis by 6N HCl. Methanol-chloroform method (1:2, V/V) was used to determine total lipids in tissues extraction (Bligh et al., 196). Fatty acids in lipid samples was prepared in a methanol solution containing 10% potassium at room temperature 1 h for methylation; Fatty acid methyl esters was identified by gas chromatography (Hewlett-Packard HP-5890) with conditions i.e. L = 25 m, ID = 0.32 min, DF = 0.25 umol/l, HP-Ultra 2, with nitrogen as carrier gas.
Plasma samples was subjected to various assays include aspartate aminotransferase (AST), alanine transaminase (ALT), alkaline phosphatase (ALP), gamma-glutamyl transpeptidase (GGT), triglyceride (TG), cholesterol (CHO), low-density lipoprotein cholesterol (LDL-C), high-density lipoprotein cholesterol (HDL-C), total protein (TP), albumin (ALB) and total bilirubin (T-BIL). All these assays was analyzed using an automatic biochemical analyzer (Hitachi 7020, Hitachi Science Systems, Lbaranki, Japan).  

### Histopathological study on fish tissues
After the feeding trial, 10 fish of each tank was randomly selected .The fresh tissues (head kidney, liver and intestinal) of these fish was fixed with 4% paraformaldehyde. Paraformaldehyde-fixed tissues was processed and embedded in paraffin according to routine histological techniques. Sections of 5-μm thickness was mounted on glass slides, stained with haematoxylins and eosin (H&E), and examined by light microscopy. Randomly oriented sections was cut through the tissues. In vivo alterations was studied in more detail. In this case, specimen’s position was adjusted to provide the greatest amount of information per alteration.

### Intestinal digestive enzyme
The following assays was performed for the liver and intestinal homogenate. Pepsin, trypsin, lipase, and amylase activities was assayed using commercial kits purchased from Nanjing Jiancheng Bioengineering Institute following the manufacturer's instructions. The protease activity is measured by the method of using Folin-phenol reagent. Lipase activity is determined according to a modified method of using 4-nitrophenyl caproate (4-NPC) dissolved in ethanol and amylase activity is quantified using a solution to reveal non-hydrolyzed starch. One unit of protease activity is defined as 1µg tyrosine liberated by hydrolyzing casein in 1 min at 37℃. One unit of lipase activity is defined as the amount of enzyme that catalyzed the release of 1 µg of fatty acids in 1 min at 37℃. One unit of amylase activity is defined as 10 mg amylum is hydrolyzed by 100 ml enzyme solution in 30 min at 37℃. Digestive enzyme activities was expressed as enzyme activity per gram intestine weight.

### Intestinal microbiota analysis

After 6 weeks of feeding trial, 5 fish of each tank was randomly selected and surface-disinfected with ethanol (75%). Under sterile conditions, the intestinal tracts was removed using sterile scissors and forceps. Incisions was made in the stomach, foregut (from the pyloric ceca to the middle of the intestine), hindgut (distal half of the intestine), and rectal samples. The contents of each digestive segment was gently squeezed out, and collected separately (Schrijver & Ollevier, 2000). The intestinal samples was then minced into fine pieces. The whole process is performed in a 2 ml sterile saline solution (NaCl, 0.85%) under sterile conditions. The processed GI samples was then kept in Petri dishes at – 20 °C until use.  

### DNA extraction, Illumina-sequencing, and assembly

The metagenomic DNA of the microbiota was extracted from the grouper intestinal tract samples using Bacterial DNA Kit (Omega, Shanghai, China). The DNA concentration was determined by means of a Nanodrop 1000 (Thermo Fisher Scientific, Wilmington, DE, USA) and stored at − 20 °C until further processing. The amplicon library preparation was performed by polymerase chain reaction (PCR) amplification of the V3–V4 region of the 16S rRNA gene. All libraries was sequenced by using an Illumina MiSeq platform (Illumina, San Diego, CA, USA) at Biomarker Technologies, Beijing, China. Paired reads was filtered for quality (Q30) and joined by FLASH version 1.2.11. Sequences that contained read lengths shorter than 400 bp was removed. The remaining sequences was then classified into taxa by blasting with the ribosomal database project (RDP) database at a 97% similarity threshold. Operational taxonomic units (OTUs), whose counts was more than 3 in at least one of the samples, was retained in the further analysis. The selected OTUs was normalized to the relative abundance for each sample. The diversity of the microbial communities was estimated by using the R program phyloseq package. For a deeper analysis of the diversity of the major evolutional clades in the microbiota, OTUs was filtered to acquire a relative abundance of at least 1% in at least one sample. The GTR model was used to construct the maximum likelihood (ML) trees. The R program ape package was used to plot the tree.
Ball mill MM 400 (RETSCH, Germany) was used to homogenize the intestinal before RNA extraction, with the parameters of 25 oscillations/s and 1 min. The homogenized tissue was incubated on ice and immediately used for RNA extraction. Total RNA was extracted by using the RNeasy Mini Kit (Qiagen, Shanghai, China) according to the manufacturer’s instructions. RNA was quantified by using a NanoDrop 1000 spectrophotometer, and its integrity was evaluated by using the RNA 6000 Assay Kit of the Agilent Bioanalyzer 2100 system (Agilent Technologies, CA, USA). High-quality RNA (RNA integrity number > 9.0) was processed by using NEB Next Ultra RNA Library Prep Kit (NEB, USA) following the manufacturer’s instruction. All libraries was sequenced via paired-end chemistry (PE125) on an Illumina HiSeq2500 platform (Illumina, San Diego, CA, USA) at Biomarker Technologies, Beijing, China.

### Quality control and De novo assembly

When the content of nitrogen exceeds 10% of the base number of any raw sequencing reads, the paired reads was removed. Then all clean reads was assembled into transcripts by Trinity software (Grabherr et al. 2011; Haas et al., 2013).
Transcriptional expression level analysis and coding potential prediction
The clean reads of different groups was used to mapping to the reliable reference transcripts. The RSEM method was used to examine the transcripts by the transcripts per million (TPM) (Li et al., 2011; Wagner et al., 2012). TPM > 0.5 in at least 1 sample was considered as a screening condition for highly reliable transcripts. These highly reliable transcripts not only contained the mRNA encoded protein, but also a large number of noncoding RNA. The CPAT (v.1.22) tool was used to predict the coding potential of RNA (Wang et al., 2013), and 4 model organisms is used to evaluate the prediction result. As long as 1 model supported the coding genes, they was judged to be the encoding genes.

### Functional annotation and analysis

The transdecoder software was used to analysis the encoding region of the transcripts, and the Blastp program (v2.2.29) was used to annotate gene function using non-redundant proteins (Nr), Swiss-Prot, Kyoto Encyclopaedia of Genes and Genomes (KEGG), evolutionary genealogy of genes and non-supervised orthologous groups (eggNOG), Gene ontology (GO) databases. GO level 2 classifications was used to analyze the GO functional class.

### Analysis of differentially expressed genes (DEGs)

The R language (v3.2.1) was used to identify the DEGs between every two groups. The transcripts with |log2FC| > 1 and q value <0.05 was defined as DEGs in this study. Moreover, the transcripts with false discovery rate (FDR) <0.05 and log2 Ratio >1 was regarded as significant abundance. Additionally, a hyper geometric distribution test was carried out to identify GO and KEGG with significantly enriched DEGs (p < 0.05), comparing to total background expressed genes. Enriched GO items was showed in the topological diagram by using software GO topnet (v.1.0).

### Statistical analysis

Data will be presented as the means ± standard deviation (SD, n = 3). After normality and heterogeneity of variances tests, one-way analysis of variance will be used for mean values comparison (SPSS19.0, SPSS Inc., Chicago, USA). A Statistically significant difference will be described at p < 0.05. The regression models and correlation analysis will be established using Origin 9.0 (© OriginLab Corporation, USA).


## Results


### Growth performance and biometry

![](/Morphology/BWG.png)
|Liner regression model| [Broken-line regression model](https://rdrr.io/cran/easyreg/man/bl.html)|
|:--:|:--:|
|![](Morphology/line_r.png)|![](Morphology/bl_r.png)|
||AAAAAAAAAAAAAAAAAAAAAAAA|
As is shown in **Table 3**, the body weight gain has not significantly changed among CK, BSFL10, and BSFL20. Interestingly, the BSFL10 diet was dramatically decreased when compared with the CK group though it only decreased 2.85% of BWG. When substitution ratio larger thant 20%, the BWG was decreased dramatically and negatively correlated with the increases of the substitution-ratio. Though decreased BWG is found in diet BFSL10, due to the minimal scale of BEG and had a good performance on BSFL20, BSFL substitution up to 20% is plausible. Survival ratio was not significantly different from all trails.
According to the linear regression model which p value is $2.214092e^{−12}$ and $R^2$ equals to 0.9988, 16.25% of BSFL substitute could reach the max BWG which was prediceted by linear regression model. According to the broken line model, the Maximum safe level of BSFL replacing ratio for grouper was 19.20%, which agrees with the result of BSFL20 treatment.


$
P = 2.214092e^{-12}
$
$
R^2 = 0.9987687
$

$
51.6666666666667+-22777.7777777781 * x^5+34861.1111111114 * x^4+-18055.5555555556 * x^3+3451.38888888888 * x^2+-210.499999999997 * x
$

### Nutritional composition

Muscle moistrue is higher in BSFL20 to BSFL50. Muscle crude protein significantly decreased in BSFL20, BSFL30, BSFL40, and BSFL50. the muscle crude lipid was not significant effect among all trials. Muscle Ash was in creased in BSFL10 and BSFL30. Whole body of Moisture is increased in BSFL40, BSFL50. Crude protein was decreased among all trails. curd lipid was significantly decreased in BSFL 10 and BSFL 30, but it is increased in BSFL 50. Ash was significantly increased in BSFL10, BSFL20, BSFL30, and BSLF50.

About the composition of the amino acid in muscle, the free amino acids were waved lager than a hydrolytic amino acid. ==One possible way that the hydrolytic amino acids are mainly served for the compositions and functions, and the free amino acids serve to backup resources. Hence that, the concentration of free amino acids is more sensitive to environmental change. (这两句瞎说的. 能找到文献嘛? 如果ok, 就做个统计图, 来体现这两种不同类型氨基酸所改变的尺度)==.
Among those amino acids, methionine had attracted the most attention due to it is an essential amino acid and deficient in plant-based protein. In this experiment, we find that the ratio of BSFL substitution could up to 60%, though, in the BSFL30, the ratio of methionine is significantly increased. And this similar pattern can be found in other amino acids, too. Those evidence suggested that the BSFL is a relatively high-quality alternative protein compared with plant-based protein.

### Digestive Enzyme

![](png/Enzyme/Enzyme_sig.png)
From BSFL10 to BSFL50, the liver and stomach amylase, stomach lipase were not significantly affected. The activity of intestine pepsin was increased in BSFL 30, BSFL 40, but lower in BSFL 50. Intestine amylase was significantly increased only in BSFL 20. Intestine trypsin was significantly increased in BSFL 10, but decreased in BSFL 20 to BSFL 50. Liver pepsin was increased significantly in all except BSFL 20. Liver Lipase only significantly increased in BSFL 10. Liver Trypsin significantly increased in BSFL 40 and BSFL 50. Stomach pepsin was significantly decreased among all treatments. While, stomach Trypsin was sifnificant increased in BSFL 10, BSFL 30, and BSFL 50, but lower in BSFL 20, and BSFL 40. Low level BSFL substitute doesn't significantly affect the activity of intestinal pepsin.

### Serum Biochemical

In our experiment, ALP (Alkaline phosphatase), ALB (albumin), and ALT (alanine transaminase), etc., were measured. We find that the concentration of ALT is waving among different diet protocols. AST was not significantly decreased in BSFL10, 20, and 40.

The concentration of the LZY and SOD has connected the immune responses of the host. LZY was firstly significantly increased in diet BSFL10, 20, and 30, but backed to normal-level and diet BSFL40 and 50. The SOD concentration is lower at firstly two diet degrees and had a sudden surge in diet BSFL30, which is not significantly different from the CK group. And then, it decreased to the same level as BSFL10. ==(又来瞎猜了:) That two-component sway in the different diet may reveal the statue switch of the immune system cause we find clear evidence of the inflammation responses when the substitution-ratio ups to 30%.==

### Transcriptomic analysis

#### GO enrichment

GO enrichment shows the change of the gene function group. Among the Intestinal group, the BSFL10 show most dramatically enriched in immune responses and serine-type endopeptidase activity, which may indicate the immune responses environment changed in fish. In BSFL20 and 30, the Go enrichment results show significance in extracellular component and ATPase activity. When it went to BSFL50, the immunity, and serine-type endopeptidase were significantly enriched again. In the liver group, molecule function was dominant the whole diet switch samples when compared with the CK group. In molecule function, heme binding, ATP binding, metal iron-binding were significantly enriched more than twice. The dramatic functional switch may potentially reveal the connection between those genes and insect-related protein assimilation. In muscle samples, iron-binding, DNA/RNA binding function, etc., was enriched. But noticeably, most ontology and enriched genes are significantly down-regulated, which indicates the function of muscle may impair to some degree. (==可以统计一下==)

#### KEGG enrichment

KEGG enrichment results are very helpful in focusing on specific functions. In this experiment, we can easily find that diseases, immunity, and cancer-related pathways are enrichment in the intestine, which indicates that the BSFL diet may evoke some diseases-responding genes or alter the behavior of immune system. Otherwise, due to the difference in the lipid composition between BFSL and fishmeal diet, steroid biosynthesis and pancreatic secretion pathway were enriched and detect in BSFL10 and BSFL50. Among liver samples, due to diet switch, we detected the alterations in metabolism-related pathways, protein digestion, and absorption and fat digestion and absorption for instance, in the BSFL10 diet. At the same time, some disease pathways and cancer-related pathways suggest that the inner immune environment was evoked again. This situation is continuing in BSFL20. When the diet goes to BSFL30, metabolic pathways were much dramatic than others. An almost similar number of up and down gene expression levels shown a complicated (==compensate mechanism to keeping the inner environment balancing..or..metabolic surge due to imbalanced nutrition diet and cause or effected by the inflomatoin==). When the diet substitution went to BSFL40 and BSFL50, the signaling pathways are more noticeable than others. Those singling pathways are the best evidence indicates that the balance of inner environment is close to the edge of collapse. Among muscle, cancer-related signaling pathways are most profound in BSFL10 and BSFL20. Cancer-related pathways like the Transcriptional misregulation pathway, MicroRNAs in Cancer are steal enriched. But at the same time, muscle function related pathway like, Tight junction, circadian rhythm are enriched. After that, cancer and immune-related pathways are still dominant in BSFL40 and BSFL50 groups, which may recall the inflammation and responsible for the slow rate of body weight gain.

### gene
- Intestine
IL-8 (gene-LOC117246332) was significantly increased in BSFL 40,
IRF-1 (gene-irf1b; interferon regulatory factor 1b)


### Intestinal microbiota analysis

From the PCA result, the CK group is hard to distinguish from BSL10, BSFL20, BSFL30, and BSFL40 but have major differences when against it to BSFL50. As to the PCoA plot, all CK, BSL10, BSFL20, BSFL30, and BSFL40 are quite scattering except BSFL50 that indicate BSFL50 is independent of the rest of the group. So as the NMDS result. Though, Anosim analysis shows no reliable significant difference between all groups. But the difference between CK and BSFL50 are still considerable, which had the greatest R-value and the minimal p-value among all comparisons. MRPP strengthens this suggestion, which shows that the difference between CK and BSFL50 is significant.

















## Discussion

### Growth Performance

Different species has different tolerance to the insect diet since some of them feed on insect by nature. And so, their ability of metabolic and catabolism is different. <u>Our experiment consistence with previous researches</u>, which suggests that substitute ratio from 20% to 50% have positive effects on BWG in different species[^Wing-Keong_2001][^St-Hilaire_2007][^Alegbeleye_2002][^Cummins_2017][^Katya_2017]. On the other hand, a relative higher of substitution could significantly reduce the BWG for some species. For example, the barramundi's (*Lates calcarifer*) of WG and SGR were significantly lower when the BSFL replacement ratio up tp 75% and 100% diets[^Katya_2017]. Replacing the Herring meal with BSFL lager than 30% would significantly decrease the palatability of the diet, protein digestibility, feed intake, and growth performance in juvenile turbot (*Psetta maxima*)[^Kroeckel_2012]. Similar results were also found in in channel catfish and rainbow trout (*Oncorhynchus mykiss*)[^Newton_2005][^St-Hilaire_2007], catfish, or blue tilapia[^Bondari_1987]. But for some species, Tilapia (*monosex tilapiawas*)[^Rana_2015] and Jian Carp(*Cyprinus carpio var. Jian*)[^Zhou_J_2018] for instance, which feed on the insect naturally, a total BSFL replacement doesn't show negative effects on 's itgrowth performance. This means that BSFL could be a perfect substitution for some fish. Not as sensitivity as BWG, <u>the survival ratio of fish was much stable. The survival of all our treatments were not significantly different</u>, which agrees with the previous researches from others in both marine and fresh water fishes[^Rawski_2020][^Cummins_2017][^Xiao_2018][^Zhou_J_2018]or Pacific white shrimp (*Litopenaeus vannamei*)[^Cummins_2017].

Growth Factors(==补文献==) were important to the growth performance of the fish. Epidermic growth factor playes an important role in the growth of animals. <u>In our experiment, gene *EGF-like-domain multiple 7*(*EGFL7*, gene-egfl7), *heparin-binding EGF-like growth factor a*(*hbegfa*, "gene-hbegfa"), *transforming growth factor beta-1-induced transcript 1 protein-like*("gene-LOC117247407"). Was significantly decreased in all traits compared with CK group. Though the GFs were significantly decreased, growth performance was significantly decreased until the substitution up to 20%. Other growth factor related genes like *btc*(betacellulin  epidermal growth factor, gene-btc), insulin-like growth factor binding protein 1a (igfbp1a, gene-igfbp1a) were not significantly decreased in BSFL 10 my explained that the substitution is up to 20% though lot's of GFs were down-regulate.</u>

![](png/GrowthF/GF.png)

<details><summary>More</summary>

  - the maximum inclusion could be as great as **20%** in the diets of rainbow trout[^Elia_2018].
  - 16% is not only appropriate, but beneficial to the immunity and gut histomorphology of ==rainbow trou==t[^Kumar_2021].
  - **BSF prepupae** meal could replace up to 45% in juvenile ==European seabass==[^Magalhães_2017]
  -  In ==rainbow trout (Oncorhynchus mykiss)==, **BSF prepupae** could be used to replace 25% of the FM and 38% of the fish oil components of a commercial diet without negative impacts on production[^St-Hilaire_2007]
  - **BSF prepupae** reared on dairy cattle manure and trout offal and used to replace up to **50%** of fish meal portion of a practical trout diet for 8 week did not impact sensory quality of resulting ==rainbow trout== fillets.[^Sealey_2011]
  - Low level: The 60 g/kg dietary inclusion of BSF prepupae in ==channel catfish== [^Newton_2005], 150 g/kg[^St-Hilaire_2007] or 180–360 g/kg inclusion of BSF prepupae in ==rainbow trout== [^Sealey_2011] and 50–250 g/kg inclusion of BSF prepupae in ==Atlantic salmon== (Lys and Met)[^Lock_2014] had shown a similar weight gain to that of fish fed fish meal.
</details>


### ***Muscle***

<details><summary>Gene Expression</summary>
  The sarcomeric structure is the basic contractile unit
  of skeletal muscle (Squire et al. 1997)

  troponin T, C and I ("gene-tnnt1", "gene-LOC117258686"， "gene-LOC117249055")
  tropomyosin alpha-3 chain (tropomyosin, gene-LOC117248496)
  myosin light chain 4-like (gene-LOC117269269)
  Composition of the sarcomeric structures.
  ==All decreased in all trails expected BSFL 30.==

  *eif4ebp* (gene-eif4ebp2) like above
  - essential or non-essential amino acid contents in muscle of Jian carp were not significantly different among five dietary groups[^Zhou_J_2018]
</details>

Muscle is one of the biggest tissue and demand large quantity of energy. So, it plays a crucial role in fatty acid and glucose homeostasis[^Chabowski_2005]. Protease is needed for muscle which could regulating the activity of leukocyte and inflammation after exercise[^Buford_2009] by attenuate soft tissue and may also helping the muscle healing[^Miller_2004]. Previously study shown that the production of reactive oxygen species (ROS), which could modulate the the force of skeletal muscle contraction, was supported by *Pla2* in mitochondria[^Nethery_2000]. Besides, *Pla2* was also take an important role in ROS increasing during repetitive and fatiguing contractions[^Nethery_1999].




<details>
  ```bash
  grep -E $(grep --color "growth factor" 18_RNA-seq_Report/HTSW_Report/Result/Expression/Gene/allSample.expr.xls | awk '{print $1}'| tr "\n" '|'|  \
  sed 's/|$/\n/') 18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/M_CK-vs-M_BSFL*.diffFilter.xls|  \
  sed 's=18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==; s=.DEGseq.diffFilter.xls:=\t='| awk '{print $2}'|sort|uniq| \
  tr '\n' ','|sed 's/,/", "/g;s/^/gene_List = c( "/;s/, "$/)\n/'

  grep --color  creatine 18_RNA-seq_Report/HTSW_Report/Result/Expression/Gene/allSample.expr.xls | grep --color kinase|  awk -F"\t" '{print $1"\t"$4"\t"$5"\t"$7}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | awk '{print $1}'| tr '\n'  ';'| sed 's/;/", "/g;s/^/gene_List= c("/;s/$/")\n/'
  ```
  ```R
  library(ggplot2)
  library(reshape2)

  Exp <- read.table("../../18_RNA-seq_Report/HTSW_Report//Result/ken//allSample.expr.csv", header=T)

  gene_List = c( "gene-egfl7", "gene-hbegfa", "gene-LOC117247407", "gene-btc", "gene-igfbp1a")
  Gene_plot2 <- function(gene_List, Organe='Intestine', Filter = 'Y'){
    tmp <- Exp[Exp$GeneID %in% gene_List,]
    if( Filter == 'Y'){
      tmp <- tmp[rowSums(tmp[,-1])> 100,]
    }
    Result = ""
    for(i in tmp$GeneID){
      Result = paste(Result, "'" ,i,"', ",sep='')
    }
    Result = paste('gene_List = c(',Result,")")
    print(Result)
    tmp_TB <- melt(tmp)
    tmp_TB$Organ = "Muscle"
    tmp_TB$Organ[grep('I_', tmp_TB$variable)] = "Intestine"
    tmp_TB$Organ[grep('L_', tmp_TB$variable)] = "Liver"
    tmp_TB$variable = factor(tmp_TB$variable, levels = c(levels(tmp_TB$variable)[c(6,1:5,12,7:11,18,13:17)]))
    # Significant
    tmp2 <- tmp
    tmp2[,2:19] =""
    Sample = colnames(tmp)[grep("_",colnames(tmp))]
    Sample = Sample[grep("CK", Sample, invert = T)]
    for(i in Sample){
      File = paste("../../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/",paste(strsplit(i,'_')[[1]][1],'CK',sep='_') ,"-vs-" ,i ,".DEGseq.diffFilter.xls",sep='')
      for( GeneID in tmp$GeneID){
        Sig_TB <- read.csv(File,sep = '\t')
        Pvalue <- Sig_TB$pvalue[Sig_TB$GeneID == GeneID]
        if(length(Pvalue) > 0){
          tmp2[tmp$GeneID==GeneID,i] = "*"
        }
      }
    }
    tmp2_tb <- melt(tmp2, id.vars = c("GeneID"))
    tmp2_tb$variable = factor(tmp2_tb$variable, levels = c(levels(tmp2_tb$variable)[c(6,1:5,12,7:11,18,13:17)]))
    tmp2_tb$Organ = "Muscle"
    tmp2_tb$Organ[grep('I_', tmp2_tb$variable)] = "Intestine"
    tmp2_tb$Organ[grep('L_', tmp2_tb$variable)] = "Liver"

    tmp_TB = tmp_TB[tmp_TB$Organ== Organe,]
    tmp2_tb = tmp2_tb[tmp2_tb$Organ== Organe,]
    ggplot(tmp_TB,aes(x= variable, y= value, group=GeneID)) + geom_line()+
        theme_light() + facet_wrap(GeneID~ ., scales = 'free')+
        theme(axis.text.x = element_text(angle=-45, hjust= 0))+
        geom_text(aes(label=tmp2_tb$value), color = 'salmon')
  }
  Gene_plot2(gene_List, "Muscle")

  # Muscle sarcomeric
  gene_List = c( "gene-LOC117249055", "gene-tnnt1", "gene-LOC117258686")
  Gene_plot2(gene_List, "Muscle")
  ggsave("Musle_sarcomeric.png", w= 7, h = 2.5)
  ```
</details>

### Body Composition

One main reason for BSFL be an appropriate meal to substitute the fishmeal is that it's relatively high concentration of protein. It comprised of about 40-45% crud protein and around 35% lipid[^Bondari_1987][^Yu_GH_2009][^Van_Huis_2013]. Study on Yellow catfish (*Pelteobagrus fulvidraco*)[^Xiao_2018] shows that significant differences in moisture, ash, crude lipid, or crude protein were found, <u>which in consistence to our experiment</u>. However, another study on barramundi (*Lates calcarifer*) shown that crude protein and moisture were not much affected by the treatments on[^Katya_2017]. In juvenile turbot (*Psetta maxima*), body protein was no effect on BSFL diet; However, body lipid content decreased with the increase of BSFL replacing ratio[^Kroeckel_2012]. The This different results could caused by the difference of species has different tolerance ratio, it may also caused the different composition of the BSFL, too. (==Try to insert something later?==)


### AA
Except arginine, other essential amino acids (EAA) is significantly lower in BSFL75 diet. which converse the result from  Katya ,et al., all essential amino acids were found to be higher in BSFL100 diet[^Katya_2017]. Zhou's research shows that essential or non-essential amino acid (NEAA) contents in muscle of Jian. On the other hand, carp were not significantly different among five trials[^Zhou_J_2018]. But from our result, though the ratio of hydrolytic EAA to NEAA is not significantly different, most content of AA was significantly effect by BSFL diets.  


### Serum

<details><summary>Result of Serum</summary>
  In our experiment, ALP, MDA, ALB, and UA were signaling improved in all 5 BSFL diets. In opponent, TBA was significantly decreased.

  r-GT, ALB/GLB, TC, or HDL were not significantly effected.

  ALT     higher in BSFL10, 20, and 40, but lower in BSFL30, 50.
  AST     lower  in 30, 50.
  AST/ALT higher in all but BSFL40 is not significant
  SOD     lower  in BSFL 10, 20, 40, 50
  LZY     higher in BSFL 10, 20, 30
  GSH-Px  higher in BSFL 20, 30, 50
  CAT     higher in BSFL 30, 40, 50, but lower in BSFL 10, 20
  TP      higher in all, but not significant in BSFL 20, 40
  GLB     higher in BSFL 10, 20, 30, 40, but lower in BSFL 50  
  LDL     higher in all, but not significant in BSFL 20, 30
  BUN     higher in BSFL 10, 30, 40


  *[ALB]: albumin
  *[TP]: total protein
  *[GLB]: globulin
  *[TC]: total cholesterol
  *[HDL]: high-density lipoprotein
  *[LDL]: low-density lipoprotein

  TP, ALB, GLB, TC, HDL, LDL, was not significantly changed in serum of Jian carp when feed with BSFL diet even in a high substitute ratio[^Zhou_J_2018].

  random: TP, GLB. LDL
  high: ALB,
  Not: TC, HDL,
  The significant of regression models[^Aarons_2004].
</details>



The serum MDA was significantly increased, witch in contrast with the result from Zhou J, et al.[^Zhou_J_2018], who shown that BSFL have no significant effects on Jian carp. Serum SOD was also significantly decreased and opposite to the yellow cat fish[^Xiao_X_2018].

Lysozyme is important for non-specific immune system of fish[^Chaklader_2019][^Saurabh_2008], which is important to against different kind of pathogens. In fish, lysozyme is distributed in mucus, plasma, and other body fluids[^Saurabh_2008]. <u>In our experiment, LZY was slightly but significant increased in Diet BSFL 10, BSFL 20, and BSFL 30, which shown a immune activity stimulated effect on BSFL diet.</u> Similar result could be found in Pelteobagrus fulvidraco, marron, Cherax cainii, and black carp, Mylopharyngodon piceus[^Yixiang_2013][^Foysal_2019] when they feed with BSFL or maggot diet. Similar dose-response of BSFL meal was found in yellow catfish which had a relative lower LZY activity in higher BSFL substitute diet[^Xiao_X_2018]. The experiment on defatted BSFL meal in Japanese seabass end with no sifnificant affect[^Wang_G_2019] may suggests that the activity of serum LZY may stipulated by fatty acids, lauric acid and medium-chain fatty acid for instance. Since BSFL with high share of lauric acid [^Oonincx_2015][^Lee_K_2020] and medium-chain fatty acid[^Weththasinghe_2021][^Spranghers_2017] which has antimicrobial effects against gram-positive bacteria[^Spranghers_2018][^Dayrit_2015][^Lieberman_2006][^Skrivanova_2006][^Skřivanová_2006]and viruses[^Dayrit_2015][^Lieberman_2006].


### Liver

Previous studies was identitfied that BSFL meal did not affect the expression of genes involved in stress response heat‐shock protein‐70 member gene (*hspa8b*, gene-hspa8b) and superoxide dismutase (*SOD*, gene-sod1)in the liver of pre-smolt salmon[^Belghit_2_2019] and clownfish (Amphiprion ocellaris)[^Vargas_2019], respectively. But in higher the increased expression of **heat‐shock protein‐70 gene** was observed in the hepatopancreas of Jian carp[^Li_S_2017] in 75% of substitute and rainbow trout[^Cardinaletti_2019] in 21% of BSFL substitute. <u>Not like them, we both *hspa8b* and *SOD* was down-regulated in liver. *SOD* gene was also down regulated on BSFL 20 and BSFL 40 which consistent with the results of plasma SOD concentration .</u> ALT and AST could be an appropriate indicator of liver health since they'll leak into the bloodstream upon liver cell damage[^Zou_H_2016]. Weththasinghe's[^Weththasinghe_2_2021] believe that BSFL meal might not defeat the health of liver since plasma AST, ALT and CK levels were not affected. A similar pattern was also found on Atlantic salmon[^Belghit_2019][^Belghit_2_2019]. <u>We detected a lower present of plasma ALT and AST which lead a similar result. And it could be explained that thought high level (50%) of BSFL substitute caused a few problems in liver, a few eosinophilic granulocyte filtered in, but no clear and heavy inflammation response was found in liver samples.</u> This consistent with the experiment on rainbow trou[^Elia_2018] which shows that BSFL diet was no damage to the liver though it caused lipid vacuolization in the liver.  As a result, the health of liver was not significantly affect by the BSFL diet by judging from its histology, stressing genes, and ALT nad AST in plasma.</u>


### Intestine


![Immune](png/In_immune/Immune_related.png)
<details><summary>Related Genes</summary>
  *CCR9* (C-C chemokine receptor type 9, gene-ccr9b), *TRAV*(T cell receptor alpha chain V region, gene-LOC117263065), *IGH*(immunoglobulin heavy chain, gene-LOC117268697).

  Besids, Gene *ITGB1*(integrin beta 1, gene-LOC117246171), *ITGB2*(integrin beta 2, "gene-LOC117245725", ), *CLDN*(claudin, gene-cldn34a)


  Infection related Genes:
  *CALM*(calmodulin, gene-calm2a), *C1QA*(complement C1q subcomponent subunit A, gene-LOC117269898), *ERK*(mitogen-activated protein kinase 1/3, gene-ypel3), *IRF1* (interferon regulatory factor 1, gene-irf1b), *ASC*(apoptosis-associated speck-like protein containing a CARD, gene-LOC117255006).
</details>

The gut is an important organ since it considered as the place of exposure to variety of nutrients and harmful antigens[^Martin_2017]. In a normal and healthy gut, no vacuolization in the enterocytes of the proximal intestine was expected[^Li_Y_2020]. And increased abnormal lipid droplet accumulation on enterocytes in the PC is called enterocyte steatosis, which may impaired lipoprotein synthesis[^Olsen_1999][^Olsen_2000] or lipid transport[^Gu_M_2014].
Previous study detected that lower degree of enterocyte steatosis in the proximal intestine on pre-smolt salmon fed diets with 60% of dechitinized BSFL meal[^Li_Y_2019][^Li_Y_2020]. In rainbow trout, 26.4% of BSFL could shorten the villi in the distal intestine[^Elia_2018]; immune related genes like *IL-8* was not affected, *IRF-1* was significantly increased which indicate an ==What?== effects on BSFL diet[^Kumar_2021]. <u>On our experiment, the high expression of *IL-8* in BSFL 40 diet, and *IRF-1* in BSFL 10, BSFL 20, BSFL 30, and BSFL 50. which make the BSFL40 been an unique pattern among BSFL diet treatment group. This is consistent with the histology result that BSFL40 has the less and shortest villi</u>.

### Pancreatic metabolism

##### result part
<details><summary>Pancreatic genes</summary>

  Protease:
  *Trypsin* (gene-LOC117265263); *CELA2* (pancreatic elastase II) (gene-LOC117271048), *CTRL*(chymotrypsin-like protease) (gene-LOC117265746), *Cap5* (carboxypeptidase A5),(gene-cpa5), *Cpb1* (carboxypeptidase B)(gene-cpb1); Lipase: *PLRP1* (pancreatic lipase-related protein 1) (gene-LOC117248582), *Cel* (bile salt-stimulated lipase)(gene-cel.2), *Pla2* (secretory phospholipase A2)(gene-LOC117252894),

  Low in Intestine on diet BSFL10 (except *CTRL*), BSFL20 (except *Cpb1*, *CTRL*, *Cel*, *Pla2*), BSFL50
  High in Liver on diet BSFL20

  And all of those genes are almost completely suppressed in muscle samples.

  ```R
  gene_List = c('gene-LOC117265263', 'gene-LOC117271048', 'gene-LOC117265746', 'gene-cpa5', "gene-cpb1", 'gene-LOC117248582', 'gene-cel.2', 'gene-LOC117252894')
  ```
</details>

![lipid catabolism](png/digest/Pancreatic_me.png)
![lipid catabolism](png/digest/Pancreatic_me_trend.png)

##### discussion part
==Significant of Pancreatic metabolism==
All pancreatic metabolism related genes are shown similar expression pattern in or organs. In intestine, they are suppressed in BSFL10, BSFL20, and BSFL50. In Liver, their expression level are surged to the top at BSFL20. When comes to muscle, they are all suppressed. Those significant changing pattern shown that both metabolism and catabolism pattern was affected. In intestine, they low expression protease and lipase could impair the ability of protein and lipid metabolic. In the liver, high expression on BSFL20 indicate that they a stress responses was evoked.

### Lipid Metabolism
![lipid catabolism](png/digest/lipid_catabolism.png)
Lipid metabolism is critical for the health of organisms. Takuya et al.[^Takuya1_2007] previously shown that the fish oil has significant reduce the effect on high lipid-diet inducted BWG in C57BL/6J Mice by regulating a group of genes which could lead to increase the lipid metabolism. In BSFL diet, Fish oil was reduced due to high crude lipid in BSFL was detected. So, it is interesting to see if the crude lipid in BSFL could have the similar effects or not. Previous[^Takuya1_2007] experiment shown that genes Medium-chain specific acyl-CoA dehydrogenase (Acadm), NADP-dependent malic enzyme (Me1). And Peroxisomal acyl-coenzyme A oxidase 1 (Acox1) Cytochrome P450 4A10 (Cyp4a10), Carnitine O-palmitoyltransferase 1(Cpt1a), NADP-dependent malic enzyme (Me1), Hydroxymethylglutaryl-CoA synthase (Hmgcs2) were up-regulated by fish oil diet. And Fatty acid synthase (Fasn), Acetyl-CoA carboxylase 1 (Acaca) were significantly down-regulated. <u>In our experiment, we found an reverse expression pattern which could. The expression of *Acadm* (gene-acadm), *Acox1* (gene-acox1) not significantly suppressed (p-value< 0.001 but Fold Change<2). *Fasn* (gene-fasn) and *Acaca* (gene-acaca) were significantly up-regulated in Liver on diet BSFL20, BSFL30, BSFL40 (except *Acaca*). *Me1* (gene-me1), *Pdk4* (gene-pdk4), *Cpt1a* (gene-LOC117247075) significantly down regulated. The Lipase activity in intestine and liver strengthens the idea that with the reduce of the fish oil ratio, the lipid metabolic was increased. *Fasn* and *Acaca* were increased to stimulating the fatty acid synthesis. Mean while, the lower expression of *Me1*, *Pdk4*, *Cpt1a* suggested that the lipid catabolism may be suppressed.</u> So, by partially opposing the result agreed with Mori's research on fish oil[^Takuya1_2007].


<details><summary>Gene Above</summary>

  ```R
  gene_List = c( 'gene-acadm', 'gene-acox1', 'gene-fasn', 'gene-acaca', 'gene-me1', 'gene-pdk4', 'gene-LOC117247075')
  ## Lipid related gene in Grouper
  ```

  `### fads`
  `grep  -i  'gene-fads' ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5"\t"$7}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG|  awk '{if($4 < 0.001) print $0}'`
  L_CK-vs-L_BSFL10	gene-fads2	3.586887128	5.27E-234	Up
  L_CK-vs-L_BSFL20	gene-fads6	-2.013712137	8.06E-34	Down
  L_CK-vs-L_BSFL20	gene-fads2	-6.43869676	2.10E-24	Down
  L_CK-vs-L_BSFL30	gene-fads2	2.411517849	2.08E-81	Up
  L_CK-vs-L_BSFL40	gene-fads2	-1.840885846	6.03E-11	Down
  L_CK-vs-L_BSFL50	gene-fads6	-1.382365608	4.43E-21	Down

  `### ACADM_MOUSE`
  `grep  -i -w gene-acadm ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG`
  I_CK-vs-I_BSFL10	gene-acadm	-0.177313265	7.99E-14
  I_CK-vs-I_BSFL20	gene-acadm	-0.359548766	1.54E-62
  I_CK-vs-I_BSFL30	gene-acadm	-0.779621794	5.81E-216
  I_CK-vs-I_BSFL40	gene-acadm	-0.800691506	1.41E-221
  I_CK-vs-I_BSFL50	gene-acadm	-0.378467326	7.38E-71
  L_CK-vs-L_BSFL10	gene-acadm	-0.388939734	3.37E-22
  L_CK-vs-L_BSFL20	gene-acadm	-0.344915169	1.20E-17
  L_CK-vs-L_BSFL30	gene-acadm	-1.051171573	6.43E-137
  L_CK-vs-L_BSFL40	gene-acadm	-0.090347104	0.017810419
  L_CK-vs-L_BSFL50	gene-acadm	-0.114704505	0.002661671

  `### ACOX1_MOUSE`
  `grep  -i -w gene-acox1 ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG`
  I_CK-vs-I_BSFL10	gene-acox1	0.163866601	6.26E-14
  I_CK-vs-I_BSFL20	gene-acox1	-0.047155984	0.017937873
  I_CK-vs-I_BSFL30	gene-acox1	-0.040400927	0.058421747
  I_CK-vs-I_BSFL40	gene-acox1	-0.20846255	6.01E-21
  I_CK-vs-I_BSFL50	gene-acox1	-0.309631915	1.79E-51
  L_CK-vs-L_BSFL10	gene-acox1	0.205809852	4.36E-15
  L_CK-vs-L_BSFL20	gene-acox1	-0.721778869	1.80E-118
  L_CK-vs-L_BSFL30	gene-acox1	-0.128497628	8.20E-07
  L_CK-vs-L_BSFL40	gene-acox1	-0.140249788	4.83E-07
  L_CK-vs-L_BSFL50	gene-acox1	-0.130758566	2.43E-06

  `###	FAS_MOUSE`
  `grep  -i -w fasn ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG`
  I_CK-vs-I_BSFL10	gene-fasn	0.868824964	1.57E-43
  I_CK-vs-I_BSFL20	gene-fasn	0.220767592	0.00035314
  I_CK-vs-I_BSFL30	gene-fasn	0.797496144	2.45E-40
  I_CK-vs-I_BSFL40	gene-fasn	-0.220515485	0.002281126
  I_CK-vs-I_BSFL50	gene-fasn	0.533576321	1.94E-20
  L_CK-vs-L_BSFL10	gene-fasn	3.417661218	0
  L_CK-vs-L_BSFL20	gene-fasn	3.592285129	0
  L_CK-vs-L_BSFL30	gene-fasn	5.359710673	0
  L_CK-vs-L_BSFL40	gene-fasn	2.820736053	0
  L_CK-vs-L_BSFL50	gene-fasn	0.952171495	4.70E-28

  `###	ACACA_MOUSE`
  `grep  -i -w gene-acaca ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG`
  I_CK-vs-I_BSFL10	gene-acaca	0.119324451	0.071719904
  I_CK-vs-I_BSFL20	gene-acaca	-0.103738978	0.086735491
  I_CK-vs-I_BSFL30	gene-acaca	-0.237043827	0.000363648
  I_CK-vs-I_BSFL40	gene-acaca	-1.124949878	9.06E-47
  I_CK-vs-I_BSFL50	gene-acaca	-0.476826909	6.33E-14
  L_CK-vs-L_BSFL10	gene-acaca	2.062568066	4.17E-192
  L_CK-vs-L_BSFL20	gene-acaca	1.497125008	2.41E-82
  L_CK-vs-L_BSFL30	gene-acaca	3.457026854	0
  L_CK-vs-L_BSFL40	gene-acaca	0.785823372	1.78E-19
  L_CK-vs-L_BSFL50	gene-acaca	-0.642542108	7.52E-09

  `### MAOX_MOUSE *Me1* NADP-dependent malic enzyme`
  `grep  -i -w 'malic enzyme' ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG`

  L_CK-vs-L_BSFL20	gene-me1	-3.216303169	0.003574798
  L_CK-vs-L_BSFL30	gene-me2	-2.003518837	8.96E-08

  I_CK-vs-I_BSFL10	gene-me1	-0.37074208	1.21E-12
  I_CK-vs-I_BSFL10	gene-me2	-0.448400414	4.51E-07
  I_CK-vs-I_BSFL10	gene-me3	-0.024125253	0.550858624
  I_CK-vs-I_BSFL20	gene-me3	-1.722497273	4.65E-285
  I_CK-vs-I_BSFL20	gene-me1	-0.843970905	2.24E-64
  I_CK-vs-I_BSFL20	gene-me2	-0.705832299	5.71E-18
  I_CK-vs-I_BSFL30	gene-me3	-1.139646007	8.21E-134
  I_CK-vs-I_BSFL30	gene-me2	-1.5170125	4.21E-51
  I_CK-vs-I_BSFL30	gene-me1	-0.572001479	3.27E-29
  I_CK-vs-I_BSFL40	gene-me3	-0.979511841	7.14E-103
  I_CK-vs-I_BSFL40	gene-me2	-1.873089982	6.57E-67
  I_CK-vs-I_BSFL40	gene-me1	-0.802784235	1.03E-50
  I_CK-vs-I_BSFL50	gene-me3	-0.837592848	5.66E-96
  I_CK-vs-I_BSFL50	gene-me2	-1.114633754	3.64E-38
  I_CK-vs-I_BSFL50	gene-me1	-0.400965071	1.02E-18
  L_CK-vs-L_BSFL10	gene-me2	-1.29186818	0.000256453
  L_CK-vs-L_BSFL10	gene-me3	-0.247474562	0.109516651
  L_CK-vs-L_BSFL10	gene-me1	-0.850358361	0.240058337
  L_CK-vs-L_BSFL20	gene-me2	-1.698455032	9.63E-06
  L_CK-vs-L_BSFL20	gene-me3	0.393244973	0.004994916
  L_CK-vs-L_BSFL30	gene-me1	0.733446884	0.154897504
  L_CK-vs-L_BSFL30	gene-me3	0.158368516	0.241121215
  L_CK-vs-L_BSFL40	gene-me2	-1.782468644	3.62E-06
  L_CK-vs-L_BSFL40	gene-me3	0.204700867	0.152338638
  L_CK-vs-L_BSFL40	gene-me1	0.092001189	0.881065019
  L_CK-vs-L_BSFL50	gene-me3	0.918892393	3.04E-13
  L_CK-vs-L_BSFL50	gene-me2	-1.282305861	0.000268108
  L_CK-vs-L_BSFL50	gene-me1	1.229594063	0.014589325

  `### PDK4_MOUSE`
  `grep  -i -w 'PDK4' ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG`

  I_CK-vs-I_BSFL10	gene-pdk4	-0.110855671	0.006421948
  I_CK-vs-I_BSFL20	gene-pdk4	0.078386753	0.023542564
  I_CK-vs-I_BSFL30	gene-pdk4	-0.154461794	6.47E-05
  I_CK-vs-I_BSFL40	gene-pdk4	-0.676911463	1.86E-56
  I_CK-vs-I_BSFL50	gene-pdk4	-0.504283157	7.44E-41
  L_CK-vs-L_BSFL10	gene-pdk4	-0.950831233	5.40E-15
  L_CK-vs-L_BSFL20	gene-pdk4	-1.145913816	2.30E-19
  L_CK-vs-L_BSFL30	gene-pdk4	-0.815252128	2.47E-13
  L_CK-vs-L_BSFL40	gene-pdk4	-1.782468387	7.17E-37
  L_CK-vs-L_BSFL50	gene-pdk4	-0.949730565	4.86E-15

  `### CP4AA_MOUSE, Cyp4a10, Cytochrome P450 4A10`

  `grep  -i -w 'Cytochrome P450' ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG|  awk '{if($4 < 0.001) print $0}'`

  I_CK-vs-I_BSFL10	gene-LOC117265993	3.38339284	5.80E-160
  I_CK-vs-I_BSFL10	gene-LOC117246222	-2.02635974	1.24E-11
  I_CK-vs-I_BSFL10	gene-LOC117262872	-5.497705781	6.04E-09
  I_CK-vs-I_BSFL10	gene-LOC117253312	5.074251694	0.000178879
  I_CK-vs-I_BSFL20	gene-LOC117246716	-2.098072195	1.36E-30
  I_CK-vs-I_BSFL20	gene-LOC117246222	-2.191894563	5.57E-16
  I_CK-vs-I_BSFL40	gene-LOC117265993	3.431396452	6.37E-184
  I_CK-vs-I_BSFL40	gene-LOC117262872	-2.3302968	7.51E-06
  I_CK-vs-I_BSFL50	gene-LOC117265993	3.82237422	0
  I_CK-vs-I_BSFL50	gene-LOC117262872	-4.174505189	1.37E-10
  L_CK-vs-L_BSFL10	gene-LOC117267658	2.150671545	2.41E-65
  L_CK-vs-L_BSFL10	gene-LOC117246222	-2.646217632	7.93E-28
  L_CK-vs-L_BSFL20	gene-LOC117260539	-3.331780513	7.36E-10
  L_CK-vs-L_BSFL20	gene-LOC117254060	-2.670869271	6.98E-08
  L_CK-vs-L_BSFL20	gene-LOC117258895	-4.333239433	0
  L_CK-vs-L_BSFL20	gene-LOC117251801	-3.542224484	0
  L_CK-vs-L_BSFL20	gene-LOC117259370	-6.287724838	0
  L_CK-vs-L_BSFL20	gene-LOC117254557	-2.442320232	0
  L_CK-vs-L_BSFL20	gene-LOC117253312	-2.191936807	0
  L_CK-vs-L_BSFL20	gene-LOC117265993	-3.01174805	0
  L_CK-vs-L_BSFL20	gene-LOC117262877	-2.93103889	0
  L_CK-vs-L_BSFL20	gene-LOC117253505	-2.511732152	0
  L_CK-vs-L_BSFL20	gene-LOC117254651	-2.923665275	0
  L_CK-vs-L_BSFL20	gene-LOC117253503	-2.051550687	1.40E-176
  L_CK-vs-L_BSFL20	gene-LOC117254162	-4.253973613	3.62E-99
  L_CK-vs-L_BSFL20	gene-LOC117250851	-4.505810416	3.33E-07
  L_CK-vs-L_BSFL20	gene-LOC117253315	-3.216303799	3.79E-05
  L_CK-vs-L_BSFL30	gene-LOC117258895	-3.414186535	0
  L_CK-vs-L_BSFL30	gene-LOC117259370	-3.864260758	0
  L_CK-vs-L_BSFL30	gene-LOC117251801	-2.848676908	0
  L_CK-vs-L_BSFL30	gene-LOC117254557	-2.220904505	0
  L_CK-vs-L_BSFL30	gene-LOC117271254	-2.476454392	2.37E-31
  L_CK-vs-L_BSFL30	gene-LOC117260539	-3.703958575	4.18E-12
  L_CK-vs-L_BSFL30	gene-LOC117253315	-4.173444003	1.00E-06
  L_CK-vs-L_BSFL30	gene-LOC117250851	-2.87798812	3.05E-06
  L_CK-vs-L_BSFL40	gene-LOC117251801	-2.472518565	0
  L_CK-vs-L_BSFL40	gene-LOC117254557	-2.085835952	0
  L_CK-vs-L_BSFL50	gene-LOC117250851	-2.130302709	0.000219023
  L_CK-vs-L_BSFL50	gene-LOC117251801	-3.636655333	0
  L_CK-vs-L_BSFL50	gene-LOC117254557	-2.817830942	0
  L_CK-vs-L_BSFL50	gene-LOC117258895	-2.211144943	0
  L_CK-vs-L_BSFL50	gene-LOC117259370	-2.452950251	0
  L_CK-vs-L_BSFL50	gene-LOC117265993	-2.296372541	0
  L_CK-vs-L_BSFL50	gene-LOC117267658	2.646102465	1.54E-116
  L_CK-vs-L_BSFL50	gene-LOC117250428	-2.165863668	1.43E-33
  L_CK-vs-L_BSFL50	gene-LOC117253315	-4.162723928	4.36E-06

  I_CK-vs-I_BSFL10	gene-LOC117259370	0.951712609	0
  I_CK-vs-I_BSFL10	gene-LOC117272518	0.388328554	2.65E-180
  I_CK-vs-I_BSFL10	gene-LOC117258895	-0.558269163	2.11E-169
  I_CK-vs-I_BSFL10	gene-LOC117254557	-0.288481655	3.46E-161
  I_CK-vs-I_BSFL10	gene-LOC117248526	-0.334960171	2.47E-44
  I_CK-vs-I_BSFL10	gene-LOC117262877	-0.389373931	3.45E-42
  I_CK-vs-I_BSFL10	gene-LOC117253503	-0.681067187	1.98E-35
  I_CK-vs-I_BSFL10	gene-LOC117248593	-0.279441284	6.29E-32
  I_CK-vs-I_BSFL10	gene-LOC117251801	0.559907162	1.17E-23
  I_CK-vs-I_BSFL10	gene-LOC117253313	1.642457197	9.11E-18
  I_CK-vs-I_BSFL10	gene-LOC117254162	-0.43497035	7.14E-17
  I_CK-vs-I_BSFL10	gene-LOC117254175	-0.275627883	6.11E-12
  I_CK-vs-I_BSFL10	gene-LOC117253314	0.395104109	2.07E-10
  I_CK-vs-I_BSFL10	gene-LOC117249917	0.38032331	1.39E-08
  I_CK-vs-I_BSFL10	gene-si:dkey-91i10.3	-0.212930005	2.18E-08
  I_CK-vs-I_BSFL10	gene-LOC117272492	0.664833337	4.32E-08
  I_CK-vs-I_BSFL10	gene-LOC117253504	-0.222245689	4.57E-08
  I_CK-vs-I_BSFL10	gene-LOC117248681	-1.711095198	4.69E-06
  I_CK-vs-I_BSFL10	gene-LOC117254651	0.105341132	0.000559799
  I_CK-vs-I_BSFL10	gene-LOC117248527	0.076394076	0.000717015
  I_CK-vs-I_BSFL10	gene-LOC117246716	-0.53199932	0.000770622
  I_CK-vs-I_BSFL20	gene-LOC117254557	-0.680964725	0
  I_CK-vs-I_BSFL20	gene-LOC117259370	0.66953779	5.26E-207
  I_CK-vs-I_BSFL20	gene-LOC117254651	-0.892363561	5.05E-172
  I_CK-vs-I_BSFL20	gene-LOC117253504	-0.679326487	5.37E-69
  I_CK-vs-I_BSFL20	gene-LOC117249917	0.779313878	1.41E-45
  I_CK-vs-I_BSFL20	gene-LOC117254162	-0.574509813	2.06E-34
  I_CK-vs-I_BSFL20	gene-LOC117251801	0.551324648	3.65E-29
  I_CK-vs-I_BSFL20	gene-LOC117253314	0.534540078	1.28E-23
  I_CK-vs-I_BSFL20	gene-LOC117258895	-0.158547423	7.81E-22
  I_CK-vs-I_BSFL20	gene-LOC117253505	-0.341252097	1.30E-18
  I_CK-vs-I_BSFL20	gene-LOC117272518	0.093880023	7.46E-14
  I_CK-vs-I_BSFL20	gene-LOC117248526	0.118003145	1.43E-09
  I_CK-vs-I_BSFL20	gene-LOC117254556	-0.162064411	4.51E-09
  I_CK-vs-I_BSFL20	gene-LOC117272492	0.551766181	4.00E-07
  I_CK-vs-I_BSFL20	gene-LOC117254770	1.536521097	3.14E-06
  I_CK-vs-I_BSFL20	gene-LOC117248681	-1.391203193	7.67E-06
  I_CK-vs-I_BSFL20	gene-LOC117254175	-0.154025687	8.08E-06
  I_CK-vs-I_BSFL20	gene-LOC117248593	0.087087011	8.96E-06
  I_CK-vs-I_BSFL20	gene-LOC117253503	0.170675856	4.08E-05
  I_CK-vs-I_BSFL20	gene-LOC117248527	0.0761488	0.000125865
  I_CK-vs-I_BSFL20	gene-LOC117265993	0.657497949	0.00020458
  I_CK-vs-I_BSFL20	gene-LOC117253313	-0.857964274	0.000945538
  I_CK-vs-I_BSFL30	gene-LOC117254557	-0.568528655	0
  I_CK-vs-I_BSFL30	gene-LOC117259370	1.006068484	0
  I_CK-vs-I_BSFL30	gene-LOC117258895	-0.375392677	1.32E-92
  I_CK-vs-I_BSFL30	gene-LOC117254651	-0.571593329	1.04E-69
  I_CK-vs-I_BSFL30	gene-LOC117253314	0.928500181	1.50E-67
  I_CK-vs-I_BSFL30	gene-LOC117253505	0.561285789	3.79E-56
  I_CK-vs-I_BSFL30	gene-LOC117248526	-0.349797794	1.17E-53
  I_CK-vs-I_BSFL30	gene-si:dkey-91i10.3	0.433367382	1.43E-41
  I_CK-vs-I_BSFL30	gene-LOC117265993	1.91503273	4.71E-36
  I_CK-vs-I_BSFL30	gene-LOC117272518	-0.165544648	6.76E-32
  I_CK-vs-I_BSFL30	gene-LOC117253504	-0.362499216	2.82E-20
  I_CK-vs-I_BSFL30	gene-LOC117254162	-0.384876793	2.90E-15
  I_CK-vs-I_BSFL30	gene-LOC117272492	0.83130605	7.88E-14
  I_CK-vs-I_BSFL30	gene-LOC117262877	-0.190750801	2.56E-13
  I_CK-vs-I_BSFL30	gene-LOC117254175	-0.252440817	2.03E-11
  I_CK-vs-I_BSFL30	gene-LOC117254556	-0.199798504	2.15E-11
  I_CK-vs-I_BSFL30	gene-LOC117259932	0.340881675	1.70E-09
  I_CK-vs-I_BSFL30	gene-LOC117246222	-1.316744771	1.31E-07
  I_CK-vs-I_BSFL30	gene-LOC117266621	0.785895531	2.00E-07
  I_CK-vs-I_BSFL30	gene-LOC117251801	0.244115323	1.15E-05
  I_CK-vs-I_BSFL30	gene-LOC117262872	1.180755058	2.53E-05
  I_CK-vs-I_BSFL30	gene-LOC117246716	-0.612166781	5.22E-05
  I_CK-vs-I_BSFL30	gene-LOC117249917	0.254890317	7.98E-05
  I_CK-vs-I_BSFL40	gene-LOC117254557	-1.038516134	0
  I_CK-vs-I_BSFL40	gene-LOC117258895	-1.063543936	0
  I_CK-vs-I_BSFL40	gene-si:dkey-91i10.3	1.080428588	2.22E-304
  I_CK-vs-I_BSFL40	gene-LOC117253314	1.72859702	6.24E-296
  I_CK-vs-I_BSFL40	gene-LOC117248526	-0.725095833	2.98E-193
  I_CK-vs-I_BSFL40	gene-LOC117262877	-0.803401835	1.19E-165
  I_CK-vs-I_BSFL40	gene-LOC117253504	-1.08957217	2.11E-129
  I_CK-vs-I_BSFL40	gene-LOC117259370	0.539360613	2.60E-110
  I_CK-vs-I_BSFL40	gene-LOC117248593	-0.483227446	2.75E-94
  I_CK-vs-I_BSFL40	gene-LOC117254556	-0.60440592	5.51E-78
  I_CK-vs-I_BSFL40	gene-LOC117254175	-0.771089423	1.21E-76
  I_CK-vs-I_BSFL40	gene-LOC117254162	-0.681938866	1.62E-39
  I_CK-vs-I_BSFL40	gene-LOC117253505	-0.521308266	2.38E-33
  I_CK-vs-I_BSFL40	gene-LOC117248527	-0.2279214	9.41E-24
  I_CK-vs-I_BSFL40	gene-LOC117253503	-0.493795724	1.62E-22
  I_CK-vs-I_BSFL40	gene-LOC117251801	-0.328016027	1.26E-07
  I_CK-vs-I_BSFL40	gene-LOC117266621	0.777344014	3.69E-07
  I_CK-vs-I_BSFL40	gene-LOC117246222	-1.178293701	1.68E-06
  I_CK-vs-I_BSFL40	gene-LOC117249917	0.270029593	3.35E-05
  I_CK-vs-I_BSFL40	gene-LOC117259932	0.207522268	0.000388593
  I_CK-vs-I_BSFL40	gene-LOC117267943	-1.593331155	0.000502659
  I_CK-vs-I_BSFL40	gene-LOC117246716	-0.503360848	0.000791008
  I_CK-vs-I_BSFL50	gene-LOC117254557	-1.135688728	0
  I_CK-vs-I_BSFL50	gene-LOC117258895	-1.08634408	0
  I_CK-vs-I_BSFL50	gene-LOC117272518	-0.528269328	0
  I_CK-vs-I_BSFL50	gene-LOC117248526	-0.77760681	6.07E-264
  I_CK-vs-I_BSFL50	gene-LOC117248593	-0.728113202	1.90E-235
  I_CK-vs-I_BSFL50	gene-LOC117254175	-1.363508177	1.34E-232
  I_CK-vs-I_BSFL50	gene-LOC117262877	-0.679378953	3.50E-150
  I_CK-vs-I_BSFL50	gene-LOC117253314	1.200533852	2.98E-145
  I_CK-vs-I_BSFL50	gene-LOC117254556	-0.735740706	1.53E-132
  I_CK-vs-I_BSFL50	gene-LOC117253505	-0.952484015	8.06E-112
  I_CK-vs-I_BSFL50	gene-LOC117253504	-0.781133931	2.26E-90
  I_CK-vs-I_BSFL50	gene-LOC117254162	-0.993352689	6.26E-89
  I_CK-vs-I_BSFL50	gene-LOC117249917	0.938712142	3.90E-71
  I_CK-vs-I_BSFL50	gene-LOC117253503	-0.870724089	1.41E-70
  I_CK-vs-I_BSFL50	gene-LOC117248527	-0.192496063	5.28E-21
  I_CK-vs-I_BSFL50	gene-LOC117254770	1.627688182	3.42E-07
  I_CK-vs-I_BSFL50	gene-LOC117246222	-1.004580506	3.58E-06
  I_CK-vs-I_BSFL50	gene-LOC117272492	0.499892407	3.73E-06
  I_CK-vs-I_BSFL50	gene-LOC117253313	-1.126570862	2.53E-05
  I_CK-vs-I_BSFL50	gene-LOC117251801	-0.212963824	0.000115701
  I_CK-vs-I_BSFL50	gene-LOC117269282	0.724117513	0.00028509
  I_CK-vs-I_BSFL50	gene-LOC117253502	-0.108935729	0.000488592
  I_CK-vs-I_BSFL50	gene-LOC117254154	-1.879049275	0.00077357
  L_CK-vs-L_BSFL10	gene-LOC117258895	-0.651345427	0
  L_CK-vs-L_BSFL10	gene-LOC117251801	-0.777693073	0
  L_CK-vs-L_BSFL10	gene-LOC117259370	-0.690471076	9.88E-282
  L_CK-vs-L_BSFL10	gene-LOC117265993	-0.684797987	9.25E-72
  L_CK-vs-L_BSFL10	gene-LOC117248593	0.503655631	5.92E-38
  L_CK-vs-L_BSFL10	gene-LOC117262872	0.455864848	2.36E-36
  L_CK-vs-L_BSFL10	gene-LOC117253312	0.210227053	5.36E-35
  L_CK-vs-L_BSFL10	gene-LOC117269282	-1.226529917	7.02E-35
  L_CK-vs-L_BSFL10	gene-LOC117253505	-0.506586766	2.03E-34
  L_CK-vs-L_BSFL10	gene-si:dkey-91i10.3	-0.426028421	2.13E-28
  L_CK-vs-L_BSFL10	gene-LOC117254651	-0.408512774	3.70E-20
  L_CK-vs-L_BSFL10	gene-LOC117254556	-0.420570611	4.29E-16
  L_CK-vs-L_BSFL10	gene-LOC117253503	-0.377675175	1.71E-12
  L_CK-vs-L_BSFL10	gene-LOC117262877	-0.206058361	1.84E-07
  L_CK-vs-L_BSFL10	gene-LOC117266621	-1.025445736	2.71E-07
  L_CK-vs-L_BSFL10	gene-LOC117255834	-0.224754842	1.39E-06
  L_CK-vs-L_BSFL10	gene-LOC117253314	-0.191786123	1.47E-06
  L_CK-vs-L_BSFL10	gene-LOC117254175	-0.319897992	3.75E-06
  L_CK-vs-L_BSFL10	gene-LOC117254557	-0.058844442	3.44E-05
  L_CK-vs-L_BSFL10	gene-LOC117250428	0.469560721	3.75E-05
  L_CK-vs-L_BSFL10	gene-LOC117253504	0.617365948	4.77E-05
  L_CK-vs-L_BSFL10	gene-LOC117253313	0.131303541	9.17E-05
  L_CK-vs-L_BSFL10	gene-LOC117272492	-0.960541666	0.0003452
  L_CK-vs-L_BSFL10	gene-LOC117249917	-0.340659919	0.000437652
  L_CK-vs-L_BSFL20	gene-LOC117253314	-1.802791842	9.72E-251
  L_CK-vs-L_BSFL20	gene-LOC117262872	-1.834931528	1.99E-248
  L_CK-vs-L_BSFL20	gene-LOC117253313	0.892242089	3.13E-197
  L_CK-vs-L_BSFL20	gene-LOC117255834	0.991053905	1.61E-148
  L_CK-vs-L_BSFL20	gene-LOC117254556	-1.686556395	1.68E-147
  L_CK-vs-L_BSFL20	gene-LOC117254175	-1.492238387	8.98E-68
  L_CK-vs-L_BSFL20	gene-LOC117248593	-0.710154498	7.10E-48
  L_CK-vs-L_BSFL20	gene-LOC117259932	-1.233894157	3.08E-45
  L_CK-vs-L_BSFL20	gene-LOC117250428	-2.569940149	1.43E-39
  L_CK-vs-L_BSFL20	gene-LOC117265841	-1.18869863	3.17E-39
  L_CK-vs-L_BSFL20	gene-LOC117269282	-1.314746585	8.68E-38
  L_CK-vs-L_BSFL20	gene-LOC117253502	-1.227274592	1.98E-19
  L_CK-vs-L_BSFL20	gene-LOC117248527	-1.469235399	3.15E-16
  L_CK-vs-L_BSFL20	gene-LOC117246222	-1.547276278	2.16E-14
  L_CK-vs-L_BSFL20	gene-LOC117249917	-0.55948361	3.95E-08
  L_CK-vs-L_BSFL20	gene-LOC117271254	-0.885441804	3.76E-07
  L_CK-vs-L_BSFL20	gene-LOC117248526	-0.430953924	7.20E-06
  L_CK-vs-L_BSFL20	gene-LOC117266214	-1.667866734	3.11E-05
  L_CK-vs-L_BSFL20	gene-LOC117253504	-0.818968029	3.25E-05
  L_CK-vs-L_BSFL20	gene-LOC117272492	-1.133841204	5.12E-05
  L_CK-vs-L_BSFL30	gene-LOC117265993	-1.719793674	0
  L_CK-vs-L_BSFL30	gene-LOC117254651	-1.703430993	2.41E-234
  L_CK-vs-L_BSFL30	gene-LOC117253505	-1.439792056	1.42E-218
  L_CK-vs-L_BSFL30	gene-LOC117254556	-1.661581708	4.13E-168
  L_CK-vs-L_BSFL30	gene-LOC117253312	-0.42391268	4.69E-125
  L_CK-vs-L_BSFL30	gene-LOC117253314	-0.990764986	4.16E-118
  L_CK-vs-L_BSFL30	gene-LOC117262877	-0.835121327	6.82E-91
  L_CK-vs-L_BSFL30	gene-LOC117255834	-0.97737691	5.62E-86
  L_CK-vs-L_BSFL30	gene-LOC117269282	-1.966726827	1.00E-76
  L_CK-vs-L_BSFL30	gene-LOC117253503	-1.035893178	8.88E-76
  L_CK-vs-L_BSFL30	gene-LOC117262872	-0.643123547	2.66E-55
  L_CK-vs-L_BSFL30	gene-LOC117265841	-1.180016458	6.04E-45
  L_CK-vs-L_BSFL30	gene-LOC117248593	-0.552273472	4.16E-36
  L_CK-vs-L_BSFL30	gene-LOC117254162	-1.072050517	1.79E-23
  L_CK-vs-L_BSFL30	gene-LOC117267658	1.265229313	3.92E-20
  L_CK-vs-L_BSFL30	gene-LOC117254175	-0.627716312	4.36E-20
  L_CK-vs-L_BSFL30	gene-LOC117249917	-0.838552355	2.44E-17
  L_CK-vs-L_BSFL30	gene-LOC117253313	0.206287831	3.10E-11
  L_CK-vs-L_BSFL30	gene-LOC117250428	-0.883224319	6.01E-11
  L_CK-vs-L_BSFL30	gene-LOC117248527	0.741224063	6.74E-11
  L_CK-vs-L_BSFL30	gene-LOC117253502	-0.758406347	6.82E-11
  L_CK-vs-L_BSFL30	gene-si:dkey-91i10.3	-0.203784222	4.58E-09
  L_CK-vs-L_BSFL30	gene-LOC117246222	-0.61762745	0.000109115
  L_CK-vs-L_BSFL40	gene-LOC117258895	-1.903075615	0
  L_CK-vs-L_BSFL40	gene-LOC117259370	-1.520447165	0
  L_CK-vs-L_BSFL40	gene-LOC117253313	1.025301751	1.98E-278
  L_CK-vs-L_BSFL40	gene-LOC117255834	1.150960167	6.92E-216
  L_CK-vs-L_BSFL40	gene-LOC117262877	-1.447101985	1.28E-191
  L_CK-vs-L_BSFL40	gene-LOC117265993	-1.212742683	3.42E-184
  L_CK-vs-L_BSFL40	gene-LOC117253312	-0.459163382	3.00E-127
  L_CK-vs-L_BSFL40	gene-si:dkey-91i10.3	0.682395744	4.74E-102
  L_CK-vs-L_BSFL40	gene-LOC117254556	-1.179038923	8.21E-89
  L_CK-vs-L_BSFL40	gene-LOC117254651	-0.93930189	7.52E-83
  L_CK-vs-L_BSFL40	gene-LOC117253505	-0.818070228	3.75E-78
  L_CK-vs-L_BSFL40	gene-LOC117253314	-0.749301088	2.28E-65
  L_CK-vs-L_BSFL40	gene-LOC117254162	-1.519227127	1.31E-34
  L_CK-vs-L_BSFL40	gene-LOC117253503	-0.589586831	2.52E-26
  L_CK-vs-L_BSFL40	gene-LOC117259932	0.519473143	3.68E-16
  L_CK-vs-L_BSFL40	gene-LOC117250428	-1.124132126	7.12E-14
  L_CK-vs-L_BSFL40	gene-LOC117254175	-0.537630395	7.25E-14
  L_CK-vs-L_BSFL40	gene-LOC117262872	-0.307351751	1.04E-13
  L_CK-vs-L_BSFL40	gene-LOC117265841	0.451896738	2.10E-11
  L_CK-vs-L_BSFL40	gene-LOC117248526	-0.633334355	1.16E-10
  L_CK-vs-L_BSFL40	gene-LOC117269282	-0.506332519	8.37E-09
  L_CK-vs-L_BSFL40	gene-LOC117249917	-0.581281538	8.74E-09
  L_CK-vs-L_BSFL40	gene-LOC117267658	-1.130391592	3.77E-07
  L_CK-vs-L_BSFL40	gene-LOC117266621	-0.97838879	7.59E-07
  L_CK-vs-L_BSFL40	gene-LOC117248527	-0.671196851	1.45E-05
  L_CK-vs-L_BSFL40	gene-LOC117246222	-0.661084076	0.000117092
  L_CK-vs-L_BSFL50	gene-LOC117253312	-1.202500871	0
  L_CK-vs-L_BSFL50	gene-LOC117253505	-1.803111945	1.70E-266
  L_CK-vs-L_BSFL50	gene-LOC117254651	-1.777461331	8.43E-220
  L_CK-vs-L_BSFL50	gene-LOC117254556	-1.994701019	2.18E-190
  L_CK-vs-L_BSFL50	gene-LOC117262872	-1.266343719	2.94E-149
  L_CK-vs-L_BSFL50	gene-LOC117262877	-1.188947632	8.42E-143
  L_CK-vs-L_BSFL50	gene-LOC117253313	0.714594713	2.70E-123
  L_CK-vs-L_BSFL50	gene-si:dkey-91i10.3	0.681515031	2.44E-102
  L_CK-vs-L_BSFL50	gene-LOC117255834	0.778542878	1.66E-88
  L_CK-vs-L_BSFL50	gene-LOC117253314	-0.8751185	2.89E-85
  L_CK-vs-L_BSFL50	gene-LOC117254162	-1.85124505	1.70E-45
  L_CK-vs-L_BSFL50	gene-LOC117253503	-0.759676696	1.79E-40
  L_CK-vs-L_BSFL50	gene-LOC117269282	-1.014252395	3.05E-26
  L_CK-vs-L_BSFL50	gene-LOC117254175	-0.756399401	2.82E-24
  L_CK-vs-L_BSFL50	gene-LOC117248593	-0.374306976	1.69E-16
  L_CK-vs-L_BSFL50	gene-LOC117249917	-0.716122017	3.80E-12
  L_CK-vs-L_BSFL50	gene-LOC117248526	-0.681739922	5.25E-12
  L_CK-vs-L_BSFL50	gene-LOC117272518	0.775875569	1.50E-09
  L_CK-vs-L_BSFL50	gene-LOC117253504	0.832870859	1.13E-08
  L_CK-vs-L_BSFL50	gene-LOC117265841	-0.376127563	1.27E-06
  L_CK-vs-L_BSFL50	gene-LOC117253502	0.472361724	2.28E-06
  L_CK-vs-L_BSFL50	gene-LOC117260539	-1.818769499	9.23E-06
  L_CK-vs-L_BSFL50	gene-LOC117248527	-0.666198971	1.56E-05
  L_CK-vs-L_BSFL50	gene-LOC117272492	-1.126065749	4.36E-05
  L_CK-vs-L_BSFL50	gene-LOC117254060	-1.617289854	6.97E-05
  L_CK-vs-L_BSFL50	gene-LOC117259932	0.23415356	0.000455006
  L_CK-vs-L_BSFL50	gene-LOC117266214	-1.227264712	0.000786681

  `### CPT1A_MOUSE`
  `grep  -i -w 'Carnitine O-palmitoyltransferase' ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG|  awk '{if($4 < 0.001) print $0}'`
  I_CK-vs-I_BSFL20	gene-LOC117262087	-0.357768953	2.56E-14
  I_CK-vs-I_BSFL20	gene-LOC117247075	0.364889583	0.000691643
  I_CK-vs-I_BSFL30	gene-LOC117262087	-0.42216275	1.16E-16
  I_CK-vs-I_BSFL40	gene-LOC117262087	0.920345989	8.29E-113
  L_CK-vs-L_BSFL10	gene-LOC117262087	-1.320102207	2.63E-25
  L_CK-vs-L_BSFL10	gene-LOC117247075	-1.785054828	2.25E-23
  L_CK-vs-L_BSFL20	gene-LOC117262087	1.648890204	4.01E-102
  L_CK-vs-L_BSFL20	gene-LOC117247075	-2.188152538	2.33E-29
  L_CK-vs-L_BSFL30	gene-LOC117262087	-2.452503684	2.68E-64
  L_CK-vs-L_BSFL30	gene-LOC117247075	-1.20942319	1.79E-15
  L_CK-vs-L_BSFL40	gene-LOC117247075	-2.666444927	2.44E-37
  L_CK-vs-L_BSFL50	gene-LOC117247075	-1.985709808	7.31E-27
  L_CK-vs-L_BSFL50	gene-LOC117262087	0.499799423	6.35E-08

  `### HMCS2_MOUSE`
  grep  -i -w 'gene-hmgcs1' ../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/*[IL]_CK-vs-*.xls|  awk -F"\t" '{print $1"\t"$4"\t"$5}'| sed 's=../18_RNA-seq_Report/HTSW_Report/Result/Diff/DEGseq/==;s/.DEGseq.diff.xls:/\t/' | grep -v DEG|  awk '{if($4 < 0.001) print $0}'
  I_CK-vs-I_BSFL10	gene-hmgcs1	0.359765461	2.34E-11
  I_CK-vs-I_BSFL20	gene-hmgcs1	0.490119231	3.32E-26
  I_CK-vs-I_BSFL30	gene-hmgcs1	0.38393702	3.15E-14
  I_CK-vs-I_BSFL40	gene-hmgcs1	-0.363325432	4.14E-10
  I_CK-vs-I_BSFL50	gene-hmgcs1	-0.203742303	7.27E-05
  L_CK-vs-L_BSFL10	gene-hmgcs1	-0.847035104	6.82E-22
  L_CK-vs-L_BSFL20	gene-hmgcs1	1.392817957	2.31E-117
  L_CK-vs-L_BSFL40	gene-hmgcs1	0.747315028	5.02E-29
  L_CK-vs-L_BSFL50	gene-hmgcs1	-0.284155366	0.00036255

  <pre>
  ## Lipid related Gene in Mice
  Liver	ACADM_MOUSE	high Medium-chain specific acyl-CoA dehydrogenase
  Liver	Acca1b	high
  Liver	ACOX1_MOUSE	high
  Liver	UCP2_MOUSE	high
  Liver	FAS_MOUSE	low
  Liver	ACACA_MOUSE	low
  Liver	MAOX_MOUSE	high
  Liver	PDK4_MOUSE	high
  Liver	CP4AA_MOUSE	high
  Intestine	CPT1A_MOUSE	high
  Intestine	Acca1b	high
  Intestine	MAOX_MOUSE	high
  Intestine	PDK4_MOUSE	high
  Intestine	HMCS2_MOUSE	high
  Intestine	CP4AA_MOUSE	high
  Intestine	ACOX1_MOUSE	high
  Intestine	FAS_MOUSE	low
  Intestine	ACACA_MOUSE	low
  </pre>

</details>


### Chitin

![Chia](png/digest/Chia.png)


Chitin[^Finke_2013][^Caligiani_2018] and antimicrobial peptides (AMP)[^Park_2014][^Park_2015] bioactive compounds are known to contain in BSFL, which have antioxidant and immunostimulatory properties in fish[^Esteban_2001][^Dong_X_2015][^Ringo_E_2012][^Gopalakannan_2006]. As a result, chitin could be used as an immunostimulant when supplemented in fish diets[^Henry_M_2015][^Ringo_E_2012]. Chitin and its derivatives such as chitosan, were also reported to have antioxidant properties[^Ngo_D_2014][^Khoushab_F_2010][^Dautremepuits_C_2004]. Previous studies shown that pathogen decrease or gastrointestinal tract histomorphological alteration have observed on Siberian sturgeon (Acipenser baerii Brandt) has observed[^Józefiak_2019]. Immune activity in common carp[^Gopalakannan_2006], seabream[^Esteban_2001], rainbow trout[^Sanchez_M_2014], Cirrhina mrigala[^Mari_L_2014] was increased in different of ways when chitin was added. However, Chitin could also decrease feed palatability, nutrient digestibility, and growth performance [^Lee_K_2020][^Gasco_2018][^Belghit_2018][^Li_S_2016][^Shiau_1999][^Sanchez_Muros_2015]. Noted worthy, some taxa of fish naturally feed with Chitin and had evolved a high tolerate of Chitin rich diet which makes Chitin had very limited effects on some fishes like juvenile lake sturgeon[^Volkman_2004], common carp (Cyprinus carpio)[^Gopalakannan_2006]. Further more, serum lysozyme activity and phagocytic activity of HK leukocytes in gilthead seabream (Sparus aurata) has no significant change with <1% chitin included diet[^Esteban_2001]. In our experiment, we detected two types of chitinase. One is *Chia* (chitinase, gene-chia.1), another is *AMCase* (acidic mammalian chitinase-like, gene-LOC117258721) which show an opposing expression pattern. The compensate expression pattern may indicate the content of marine-source chitin and insect-source chitin switch. On the other hand, the growth performance indicate that low contain of insect-source chitin has positive effect on grouper, and gut inflammation was related to a higher BSFL substit which contain more Chitin(==Stimulatory effects of chitinase on growth and immune defense of orange-spotted grouper (Epinephelus coioides)==). In a all, the role of chitin in fish nutrition was really depending on the dose[^Cardinaletti_2019][^Ringo_E_2012] and different from species to species. On the Other hand, size of the Chitin may also make things different. Lee CG, et al.[^Lee_CG_2008] fund that large chitin polymers are relatively biologically inert, while smaller fragments are pro-inflammatory, but even smaller fragments could stimulate the production of anti-inflammatory cytokine. In mice, chitin fragment size around 40–70 µm was fond it's able to trigger inflammation and cytokine production via the pattern recognition receptors[^Da_Silva_2008]. Hence that, The role of Chitin must contribute an important role in grouper.



<details>

  ```bash
  # Intestinal immune network for IgA production
  cat /media/ken/Data/Yan/2021/18_RNA-seq_Report/HTSW_Report/Result/Enrichment/Pathway/I_CK-vs-I_BSF*/I_CK-vs-I_BSFL*_map/map04672.html| sed 's/<ul>/=/g'| tr '=' '\n'|grep --color "<li>K"| tr '<>' '='| tr '=' '\n'| grep --color K|tr ' ' '\n' | grep -v [:\(]| sort|uniq | tr '\n' '=' | sed 's/=/", "/g; s/^/gene_List = c("/;s/, "$/)/'
  ```
  ```R
  gene_List = c("gene-ccr9b", "gene-LOC117263065", "gene-LOC117268697")
  ```

  ```bash
  # Leukocyte transendothelial migration
  cat /media/ken/Data/Yan/2021/18_RNA-seq_Report/HTSW_Report/Result/Enrichment/Pathway/I_CK-vs-I_BSF*/I_CK-vs-I_BSFL*_map/map04670.html| sed 's/<ul>/=/g'| tr '=' '\n'|grep --color "<li>K"| tr '<>' '='| tr '=' '\n'| grep --color K|tr ' ' '\n' | grep -v [:\(]| sort|uniq | tr '\n' '=' | sed 's/=/", "/g; s/^/gene_List = c("/;s/, "$/)/'

  File=/media/ken/Data/Yan/2021/18_RNA-seq_Report/HTSW_Report/Result/Enrichment/Pathway/I_CK-vs-*/[IL]_CK-vs-*_map/map04670.html
  for i in $(echo "gene-cldn34a", "gene-LOC117245725", "gene-LOC117246171"| sed 's/,//g');do RE=$(grep   --color $i $File| tr '>' '\n'| grep --color ^K.....:| awk '{print $1}'| sort|uniq); echo -e $RE $i;done| sort
  ```
  ```R
  gene_List = c( "gene-cldn34a", "gene-LOC117245725", "gene-LOC117246171")
  ```

  ```bash
  # Pertussis
  cat /media/ken/Data/Yan/2021/18_RNA-seq_Report/HTSW_Report/Result/Enrichment/Pathway/I_CK-vs-I_BSF*/I_CK-vs-I_BSFL*_map/map05133.html| sed 's/<ul>/=/g'| tr '=' '\n'|grep --color "<li>K"| tr '<>' '='| tr '=' '\n'| grep --color K|tr ' ' '\n' | grep -v [:\(]| sort|uniq | tr '\n' '=' | sed 's/=/", "/g; s/^/gene_List = c("/;s/, "$/)/'

  File=/media/ken/Data/Yan/2021/18_RNA-seq_Report/HTSW_Report/Result/Enrichment/Pathway/I_CK-vs-*/[IL]_CK-vs-*_map/map05133.html
  for i in $(echo 'gene-LOC117245725', 'gene-LOC117246256', 'gene-LOC117255006', 'gene-LOC117263950', 'gene-LOC117269898', 'gene-LOC117272684', 'gene-calm2a', 'gene-irf1b', 'gene-ypel3' | sed 's/,//g');do RE=$(grep   --color $i $File| tr '>' '\n'| grep --color ^K.....:| awk '{print $1}'| sort|uniq); echo -e $RE $i;done| sort
  ```

  ```R
  gene_List = c( 'gene-LOC117245725', 'gene-LOC117255006',  'gene-LOC117269898', 'gene-LOC117272684', 'gene-calm2a', 'gene-irf1b', 'gene-ypel3')
  ```

  ```bash
  # Intestinal  inflammation from GO
  grep --color "inflammatory" ../../18_RNA-seq_Report/HTSW_Report/Result/Enrichment/GO/I_CK-vs-I_BSFL*/enrichment_GO.xls | awk -F"\t" '{print $7}'| tr '\n'  ';'| sed 's/;/", "/g;s/^/gene_List= c("/;s/$/")\n/'
  ```
  ```R
  gene_List = c( 'gene-anxa1a', 'gene-elf3', 'gene-mpx', 'gene-mst1')
  ```
  ```bash
  # Immune from GO
  grep --color "immune" ../../18_RNA-seq_Report/HTSW_Report/Result/Enrichment/GO/I_CK-vs-I_BSFL*/enrichment_GO.xls | awk -F"\t" '{print $7}'| tr '\n'  ';'| sed 's/;/", "/g;s/^/gene_List= c("/;s/$/")\n/'
  ```
  ```R
  gene_List = c( 'gene-LOC117248453', 'gene-LOC117261627', 'gene-LOC117272390', 'gene-ccl20a.3', 'gene-ccl36.1', 'gene-ccl38a.4', 'gene-cd36', 'gene-endou', 'gene-ilf2', 'gene-irf1b', 'gene-psmb10',
    'gene-sting1', 'gene-tnfsf14', 'gene-vsir')
  # gene-LOC117248453 == CXCL20.2
  # gene-LOC117261627 == CCL3L1
  # gene-LOC117272390 == CCL20
  ```

  ```R
  # inflammation & Immune
  gene_List = c(  'gene-anxa1a', 'gene-elf3', 'gene-mpx', 'gene-mst1',
    'gene-LOC117248453', 'gene-LOC117261627', 'gene-LOC117272390', 'gene-ccl20a.3', 'gene-ccl36.1', 'gene-ccl38a.4', 'gene-cd36', 'gene-endou', 'gene-ilf2', 'gene-irf1b', 'gene-psmb10', 'gene-sting1', 'gene-tnfsf14', 'gene-vsir')
  ```

</details>













## References

[^Aarons_2004]: Aarons, Leon, Claire Baxter, and Suneel Gupta. "Pharmacodynamics of controlled release verapamil in patients with hypertension: an analysis using spline functions." Biopharmaceutics & drug disposition 25.5 (2004): 219-225.
[^Alegbeleye_2002]: Alegbeleye WO, Obasa SO, Olude OO, Otubu K, Jimoh W (2002) Preliminary evaluation of the nutritive value of the variegated grasshopper (Zonocerus variegatus L.) for African catfish Clarias gariepinus (Burchell. 1822) fingerlings. Aquac Res 43:412–420
[^Belghit_2018]: Belghit, I.; Liland, N.S.; Waagbø, R.; Biancarosa, I.; Pelusio, N.; Li, Y.; Krogdahl, Å.; Lock, E.J. Potential of insect-based diets for Atlantic salmon (Salmo salar). Aquaculture 2018, 491, 72–81. [Google Scholar] [CrossRef]
[^Belghit_2019]: Belghit, I., Liland, N. S., Gjesdal, P., Biancarosa, I., Menchetti, E., Li, Y., Waagbø, R., Krogdahl, Å., Lock, E.-J., 2019. Black soldier fly larvae meal can replace fish meal in diets of sea-water phase Atlantic salmon (Salmo salar). Aquaculture 503, 609-619.
[^Belghit_2_2019]: Belghit I, Waagbø R, Lock EJ, Liland NS. Insect-based diets high in lauric acid reduce liver lipids in freshwater Atlantic salmon. Aquac Nutr (2019) 25(2):343–57. doi: 10.1111/anu.12860
[^Bell_2001]: Bell, J. G., McEvoy, J., Tocher, D. R., McGhee, F., Campbell, P. J., & Sargent, J. R. (2001). Replacement offish oilwith rapeseed oil in diets ofAtlantic salmon (Salmo salar) affects tissue lipid compositions and hepatocyte fatty acid metabolism. Journal of Nutrition, 131, 1535–1543.
[^Bell_2003]: Bell, J. G., McGhee, F., Campbell, P. J., & Sargent, J. R. (2003). Rapeseed oil as an alternative to marine fish oil in diets of postsmolt Atlantic salmon (Salmo salar): Changes in flesh fatty acid composition and effectiveness of subsequent fish oil wash out. Aquaculture, 218, 515–528.
[^Bondari_1987]: Bondari K, Sheppard DC (1987) Soldier fly, Hermetia illucens L., larvae as feed for channel catfish, Ictalurus punctatus (Rafinesque), and blue tilapia, Oreochromis aureus (Steindachner). Aquac Res 18(3):209–220
[^Buford_2009]: Buford TW, Cooke MB, Redd LL, Hudson GM, Shelmadine BD, Willoughby DS. Protease supplementation improves muscle function after eccentric exercise. Med Sci Sports Exerc. 2009 Oct;41(10):1908-14. doi: 10.1249/MSS.0b013e3181a518f0. PMID: 19727022.
[^Burr_2012]: Burr, G. S., Wolters, R. W., Barrows, T. F., & Hardy, W. R. (2012). Replacing fish meal with blends of alternative proteins on growth performance of rainbow trout (Oncorhynchus mykiss), and early or late stage juvenile Atlantic salmon (Salmo salar). Aquaculture, 334–337, 110–116. https://doi.org/10.1016/j.aquaculture.2011.12.044
[^Caimi_2020]: Caimi, C.; Renna, M.; Lussiana, C.; Bonaldo, A.; Gariglio, M.; Meneguz, M.; Dabbou, S.; Schiavone, A.; Gai, F.; Elia, A.C.; et al. First insights on black soldier fly (Hermetia illucens L.) larvae meal dietary administration in Siberian sturgeon (Acipenser baerii Brandt) juveniles. Aquaculture 2020, 515, 734539.
[^Caligiani_2018]: Caligiani A, Marseglia A, Leni G, Baldassarre S, Maistrello L, Dossena A, et al. Composition of black soldier fly prepupae and systematic approaches for extraction and fractionation of proteins, lipids and chitin. Food Res Int (2018) 105:812–20. doi: 10.1016/j.foodres.2017.12.012
[^Cardinaletti_2019]: Cardinaletti G, Randazzo B, Messina M, Zarantoniello M, Giorgini E, Zimbelli A, et al. Effects of graded dietary inclusion level of full-fat Hermetia illucens prepupae meal in practical diets for rainbow trout (Oncorhynchus mykiss). Animals (2019) 9(5):251. doi: 10.3390/ani9050251
[^Chabowski_2005]: Chabowski, A., E. Barańczuk, and J. Gorski. "EFFECT OF ADRENALIN, INSULIN AND CONTRACTIONS." Journal of physiology and pharmacology 56.3 (2005): 381-390.
[^Chaklader_2019]: Chaklader, M. R., Siddik, M. A., Fotedar, R., Howieson, J., 2019. Insect larvae, Hermetia illucens in poultry by-product meal for barramundi, Lates calcarifer modulates histomorphology, immunity and resistance to Vibrio harveyi. Sci. Rep. 9, 16703.
[^Collet_2020]: Collet , B., Secombes, C.J., 2002. Type I-interferon signalling in fish. Fish Shellfish Immunol. 12, 389-397.
[^Cummins_2017]: Cummins VC Jr, Rawles SD, Thompson KR, Velasquez A, Kobayashi Y, Hager J, Webster CD (2017) Evaluation of black soldier fly (Hermetia illucens) larvae meal as partial or total replacement of marine fish meal in practical diets for Pacific white shrimp (Litopenaeus vannamei). Aquaculture 473:337–344
[^Da_Silva_2008]: Da Silva CA, Hartl D, Liu W, Lee CG, Elias JA. TLR-2 and IL-17A in chitin-induced macrophage activation and acute inflammation. J Immunol (2008) 181(6):4279–86. doi: 10.4049/jimmunol.181.6.4279
[^Dautremepuits_C_2004]: Dautremepuits C, Paris-Palacios S, Betoulle S, Vernet G. Modulation in hepatic and head kidney parameters of carp (Cyprinus carpio L.) induced by copper and chitosan. Comp Biochem Physiol C Toxicol Pharmacol (2004) 137(4):325–33. doi: 10.1016/j.cca.2004.03.005
[^Dayrit_2015]: Dayrit FM. The properties of lauric acid and their significance in coconut oil. J Am Oil Chem Soc (2015) 92:1–15. doi: 10.1007/s11746-014-2562-7
[^Dong_X_2015]: Dong X-Q, Zhang D-M, Chen Y-K, Wang Q-J, Yang Y-Y. Effects of antimicrobial peptides (AMPs) on blood biochemical parameters, antioxidase activity, and immune function in the common carp (Cyprinus carpio). Fish Shellfish Immunol (2015) 47(1):429–434. doi: 10.1016/j.fsi.2015.09.030
[^Dumas_2018]: Dumas, A., Raggi, T., Barkhouse, J., Lewis, E., Weltzien, E., 2018. The oil fraction and partially defatted meal of black soldier fly larvae (Hermetia illucens) affect differentially growth performance, feed efficiency, nutrient deposition, blood glucose and lipid digestibility of rainbow trout (Oncorhynchus mykiss). Aquaculture 492, 24-34.
[^Edagawa_1993]: Edagawa T, Murata M, Hattori M, Onuma M, Kodama H. Cell surface C-reactive protein of rainbow trout lymphocytes. Dev Comp Immunol (1993) 17(2):119–27. doi: 10.1016/0145-305X(93)90022-I
[^Elia_2018]: Elia, A.C., Capucchio, M.T., Caldaroni, B., Magara, G., Dörr, A.J.M., Biasato, I., Biasibetti, E., Righetti, M., Pastorino, P., Prearo, M., Gai, F., Schiavone, A., Gasco, L., 2018. Influence of Hermetia illucens meal dietary inclusion on the histological traits, gut mucin composition and the oxidative stress biomarkers in rainbow trout (Oncorhynchus mykiss). Aquaculture 496, 50-57.
[^Esteban_2001]: Esteban MA, Cuesta A, Ortuno J, Meseguer J. Immunomodulatory effects of dietary intake of chitin on gilthead seabream (Sparus aurata L.) innate immune system. Fish Shellfish Immunol (2001) 11(4):303–315. doi: 10.1006/fsim.2000.0315
[^Finke_2013]: Finke MD. Complete nutrient content of four species of feeder insects. Zoo Biol (2013) 32(1):27–36. doi: 10.1002/zoo.21012
[^Foysal_2019]: Foysal, M.J., Fotedar, R., Tay, C.Y., Gupta, S.K., 2019. Dietary supplementation of black soldier fly (Hermetica illucens) meal modulates gut microbiota, innate immune response and health status of marron (Cherax cainii, Austin 2002) fed poultry-by-product and fishmeal based diets. PeerJ, 7, 6891.
[^Gasco_2018]: Gasco, L.; Finke, M.; Van Huis, A. Can diets containing insects promote animal health? J. Insects Food Feed 2018, 4, 1–4. [Google Scholar] [CrossRef]
[^Gopalakannan_2006]: Gopalakannan, A., & Arul, V. (2006). Immunomodulatory effects of dietary intake of chitin, chitosan and levamisole on the immune system of Cyprinus carpio and control of Aeromonas hydrophila infection in ponds. Aquaculture, 255, 179–187.
[^Gu_M_2014]: Gu M, Kortner TM, Penn M, Hansen AK, Krogdahl Å. Effects of dietary plant meal and soya-saponin supplementation on intestinal and hepatic lipid droplet accumulation and lipoprotein and sterol metabolism in Atlantic salmon (Salmo salar L.). Br J Nutr (2014) 111(3):432–44. doi: 10.1017/S0007114513002717
[^Hansen_2020]: Hansen AKG, Kortner TM, Krasnov A, Björkhem I, Penn M, Krogdahl Å. Choline supplementation prevents diet induced gut mucosa lipid accumulation in post-smolt Atlantic salmon (Salmo salar L.). BMC Vet Res (2020) 16:32. doi: 10.1186/s12917-020-2252-7
[^Henry_M_2015]: Henry M, Gasco L, Piccolo G, Fountoulaki E. Review on the use of insects in the diet of farmed fish: Past and future. Anim Feed Sci Technol (2015) 203:1–22. doi: 10.1016/j.anifeedsci.2015.03.001
[^Huang_2007]: Huang, S. Y., Oo, A. N., Higgs, D. A., Brauner, C. J., & Satoh, S. (2007). Effect of dietary canola oil level on the growth performance and fatty acid composition of juvenile red sea bream, Pagrus major. Aquaculture, 271, 420–431.
[^Huyben_2019]: Huyben D, Vidaković A, Hallgren SW, Langeland M. High-throughput sequencing of gut microbiota in rainbow trout (Oncorhynchus mykiss) fed larval and pre-pupae stages of black soldier fly (Hermetia illucens). Aquaculture (2019) 500:485–91. doi: 10.1016/j.aquaculture.2018.10.034
[^Irungu_2018]: Irungu, F.; Mutungi, C.; Faraj, A.; Affognon, H.; Kibet, N.; Tanga, C.; Ekesi, S.; Nakimbugwe, D.; Fiaboe, K. Physico-chemical properties of extruded aquafeed pellets containing black soldier fly (Hermetia illucens) larvae and adult cricket (Acheta domesticus) meals. J. Insects Food Feed 2018, 4, 19–30.
[^Józefiak_2019]: Józefiak, A.; Nogales-Mérida, S.; Rawski, M.; Kierończyk, B.; Mazurkiewicz, J. Effects of insect diets on the gastrointestinal tract health and growth performance of Siberian sturgeon (Acipenser baerii Brandt, 1869). BMC Vet. Res. 2019, 15, 348. [Google Scholar] [CrossRef] [PubMed]
[^Kasumyan_1995]: Kasumyan, A.O.; Sidorov, S.S.; Kazhlaev, A.A.; Pashchenko, N.I. Behavioral responses of young stellate sturgeon to smell and taste of artificial feeds and their components. In Proceedings of the International Symposium on Stugeons, Moscow, Russia, 6–11 September 1993; Gershanovich, A.D., Smith, T.I.J., Eds.; VNIRO Publishing: Moscow, Russia, 1995; pp. 278–288.
[^Kasumyan_2016]: Kasumyan, A.O. Taste attractiveness of free amino acids and their physiochemical and biological properties (as exemplified by fishes). J. Evol. Biochem. Physiol. 2016, 52, 271–281. [Google Scholar] [CrossRef]
[^Kasumyan_2018]: Kasumyan, A.O. Olfaction and Gustation in Acipenseridae, with special references to the Siberian sturgeon. In The Siberian sturgeon (Acipenser baerii, Brandt 1869) Volume 1—Biology, 1st ed.; Williot, P., Nonnotte, G., Vizziano-Cantonnet, D., Chebanov, D., Eds.; Springer: Cham, Switzerland, 2018; Volume 1, pp. 173–205.
[^Katya_2017]: Katya, Kumar, et al. "Efficacy of insect larval meal to replace fish meal in juvenile barramundi, Lates calcarifer reared in freshwater." International Aquatic Research 9.4 (2017): 303-312.
[^Khoushab_F_2010]: Khoushab F, Yamabhai M. Chitin research revisited. Mar Drugs (2010) 8(7):1988–2012. doi: 10.3390/md8071988
[^Kierończyk_2018]: Kierończyk, B.; Rawski, M.; Pawełczyk, P.; Różyńska, J.; Golusik, J.; Mikołajczak, Z.; Józefiak, D. Do insects smell attractive to dogs? A comparison of dog reactions to insects and commercial feed aromas—A preelimnary study. Ann. Anim. Sci. 2018, 18, 795–800.
[^Kodama_2004]: Kodama H, Matsuoka Y, Tanaka Y, Liu Y, Iwasaki T, Watarai S. Changes of C-reactive protein levels in rainbow trout (Oncorhynchus mykiss) sera after exposure to anti-ectoparasitic chemicals used in aquaculture. Fish Shellfish Immunol (2004) 16(5):589–97. doi: 10.1016/j.fsi.2003.09.007
[^Kroeckel_2012]: Kroeckel, S., Harjes, A. G. E., Roth, I., Katz, H., Wuertz, S., Susenbeth, A., & Schulz, C. (2012). When a turbot catches a fly: Evaluation of a prepupae meal of the Black Soldier Fly (Hermetia illucens) as fish meal substitute-growth performance and chitin degradation in juvenile turbot (Psetta maxima). Aquaculture, 364–365, 345–352. https://doi.org/10.1016/j.aquaculture.2012.08.041
[^Krogdahl_2000]: Krogdahl Å, Bakke-McKellep AM, Roed KH, Baeverfjord G. Feeding Atlantic salmon Salmo salar L. soybean products: effects on disease resistance (furunculosis), and lysozyme and IgM levels in the intestinal mucosa. Aquac Nutr (2000) 6(2):77–84. doi: 10.1046/j.1365-2095.2000.00129.x
[^Krogdahl_2020]: Krogdahl Å, Hansen AKG, Kortner TM, Björkhem I, Krasnov A, Berge GM, et al. Choline and phosphatidylcholine, but not methionine, cysteine, taurine and taurocholate, eliminate excessive gut mucosal lipid accumulation in Atlantic salmon (Salmo salar L). Aquaculture (2020) 528:735552. doi: 10.1016/j.aquaculture.2020.735552
[^Kumar_2021]: Kumar, Vikas, et al. "Insect (black soldier fly, Hermetia illucens) meal supplementation prevents the soybean meal-induced intestinal enteritis in rainbow trout and health benefits of using insect oil." Fish & Shellfish Immunology 109 (2021): 116-124.
[^Lee_CG_2008]: Lee CG, Da Silva CA, Lee J-Y, Hartl D, Elias JA. Chitin regulation of immune responses: an old molecule with new roles. Curr Opin Immunol (2008) 20(6):684–9. doi: 10.1016/j.coi.2008.10.002
[^Lee_K_2020]: Lee, K.; Yun, E.; Goo, T. Antimicrobial activity of an extract of Hermetia illucens larvae immunized with Lactobacillus casei against Salmonella species. Insects 2020, 10, 704. [Google Scholar] [CrossRef] [PubMed]
[^Lieberman_2006]: Lieberman S, Enig MG, Preuss HG. A review of monolaurin and lauric acid: natural virucidal and bactericidal agents. Altern Complement Ther (2006) 12(6):310–4. doi: 10.1089/act.2006.12.310
[^Lim_2016]: , R., Tran, H. T., Liong, S., Barker, G., Lappas, M., 2016. The transcription factor interferon regulatory factor-1 (IRF1) plays a key role in the terminal effector pathways of human preterm labor. Biol. Reprod. 94(2), 32-1.
[^Li_S_2016]: Li, S.; Ji, H.; Zhang, B.; Tian, J.; Zhou, J.; Yu, H. Influence of black soldier fly (Hermetia illucens) larvae oil on growth performance in juvenile Jian Carp (Cyprinus carpio var. Jian). Aquaculture 2016, 465, 43–52. [Google Scholar] [CrossRef]
[^Li_S_2017]: Li, S., Ji, H., Zhang, B., Zhou, J., Yu, H., 2017. Defatted black soldier fly (Hermetia illucens) larvae meal in diets for juvenile Jian carp (Cyprinus carpio var. Jian): growth performance, antioxidant enzyme activities, digestive enzyme activities, intestine and hepatopancreas histological structure. Aquaculture. 477, 62-70
[^Liu_H_2018]: Liu H, Wang J, He T, Becker S, Zhang G, Li D, et al. Butyrate: a double-edged sword for health? Adv Nutr (2018) 9(1):21–9. doi: 10.1093/advances/nmx009
[^Liu_Y_2015]: Liu Y. Fatty acids, inflammation and intestinal health in pigs. J Anim Sci Biotechnol (2015) 6:41. doi: 10.1186/s40104-015-0040-1
[^Li_Y_2019]: Li Y, Kortner TM, Chikwati EM, Munang’andu HM, Lock E-J, Krogdahl Å. Gut health and vaccination response in pre-smolt Atlantic salmon (Salmo salar) fed black soldier fly (Hermetia illucens) larvae meal. Fish Shellfish Immunol (2019) 86:1106–13. doi: 10.1016/j.fsi.2018.12.057
[^Li_Y_2020]: Li Y, Kortner TM, Chikwati EM, Belghit I, Lock E-J, Krogdahl Å. Total replacement of fish meal with black soldier fly (Hermetia illucens) larvae meal does not compromise the gut health of Atlantic salmon (Salmo salar). Aquaculture (2020) 520:734967. doi: 10.1016/j.aquaculture.2020.734967
[^Lock_2014]: Lock, E-J., Arsiwalla, T., & Waagbo, R. (2014). Insect meal: A promising source of nutrients in the diet of Atlantic salmon (Salmon salar). In: P. Vantomme, C. Munke & A. vanHuis (Eds.), 1st International Conference“ Insects to Feed the World” (p. 67). Ede-Wageningen, the Netherlands: Wageningen University.
[^Magalhães_2017]: Magalhães, R., Sánchez-López, A., Leal, R. S., Martínez-Llorens, S., OlivaTeles, A., & Peres, H. (2017). Black soldier fly (Hermetia illucens) prepupae meal as a fish meal replacement in diets for European seabass (Dicentrarchus labrax). Aquaculture, 476, 79–85. https://doi.org/10.1016/j.aquaculture.2017.04.021
[^Mari_L_2014]: Mari LSS, Jagruthi C, Anbazahan SM, Yogeshwari G, Thirumurugan R, Arockiaraj J, et al. Protective effect of chitin and chitosan enriched diets on immunity and disease resistance in Cirrhina mrigala against Aphanomyces invadans. Fish Shellfish Immunol (2014) 39(2):378–85. doi: 10.1016/j.fsi.2014.05.027
[^Martin_2017]: Martin SAM, Król E. Nutrigenomics and immune function in fish: new insights from omics technologies. Dev Comp Immunol (2017) 75:86–98. doi: 10.1016/j.dci.2017.02.024
[^Miao_2018]: Miao, S., Zhao, C., Zhu, J., Hu, J., Dong, X., & Sun, L. (2018). Dietary soybean meal affects intestinal homoeostasis by altering the microbiota, morphology and inflammatory cytokine gene expression in northern snakehead. Sci. Rep. 8(1), 1-10.
[^Micallef_2017]: Micallef G, Cash P, Fernandes JMO, Rajan B, Tinsley JW, Bickerdike R, et al. Dietary yeast cell wall extract alters the proteome of the skin mucous barrier in Atlantic Salmon (Salmo salar): increased abundance and expression of a calreticulin-like protein. PLoS One (2017) 12(1):e0169075. doi: 10.1371/journal.pone.0169075
[^Miller_2004]: Miller P C, Bailey S P, Barnes M E, et al. The effects of protease supplementation on skeletal muscle function and DOMS following downhill running[J]. Journal of sports sciences, 2004, 22(4): 365-372.
[^Moretta_2020]: Moretta, A.; Salvia, R.; Scieuzo, C.; Di Somma, A.; Vogel, H.; Pucci, P.; Sgambato, A.; Wolff, M.; Falabella, P. A bioinformatic study on antimicrobial peptides identified in the black soldier fly (BSF) Hermetia illucens (Diptera: Stratiomyidae). Sci. Rep. 2020, 10, 16875. [Google Scholar] [CrossRef] [PubMed]
[^Murofushi_1981]: Murofushi, S.; Ina, K. Survey of feeding stimulants for the sea bream present in the dried pupae of silkworms. Agric. Biol. Chem. 1981, 45, 1501–1504.
[^Nethery_1999]: Nethery D, Stofan D, Callahan L, Dimarco A, and Supinski G. Formation of reactive oxygen species by the contracting diaphragm is PLA2 dependent. J Appl Physiol 87: 792–800, 1999.
[^Nethery_2000]: Nethery D, Callahan LA, Stofan D, Mattera R, Dimarco A, and Supinski G. PLA2 dependence of diaphragm mitochondrial formation of reactive oxygen species. J Appl Physiol 89: 72–80, 2000.
[^Newton_2005]: Newton, G. L., Sheppard, D. C., Watson, D. W., Burtle, G. J., Dove, C. R., Tomberlin, J. K., & Thelen, E. E. (2005) The black soldier fly, Hermetia illucens, as a manure management/resource recovery tool. In: Symposium on the State of the Science of Animal Manure and Waste Management, San Antonio, TX, USA
[^Ngo_D_2014]: Ngo D-H, Kim S-K. “Antioxidant effects of chitin, chitosan, and their derivatives”. In: Kim S-K, editor. Adv Food Nutr Res ((2014) 73:15–31. doi: 10.1016/B978-0-12-800268-1.00002-0
[^Nogales_2019]: Nogales-Mérida, S.; Gobbi, P.; Józefiak, D.; Mazurkiewicz, J.; Dudek, K.; Rawski, M.; Kierończyk, B.; Józefiak, A. Insect meals in fish nutrition. Rev. Aquac. 2019, 11, 1080–1103.
[^Olsen_1999]: Olsen RE, Myklebust R, Kaino T, Ringø E. Lipid digestibility and ultrastructural changes in the enterocytes of Arctic char (Salvelinus alpinus L.) fed linseed oil and soybean lecithin. Fish Physiol Biochem (1999) 21:35–44. doi: 10.1023/A:1007726615889
[^Olsen_2000]: Olsen RE, Myklebust R, Ringø E, Mayhew TM. The influences of dietary linseed oil and saturated fatty acids on caecal enterocytes in Arctic char (Salvelinus alpinus L.): a quantitative ultrastructural study. Fish Physiol Biochem (2000) 22:207–16. doi: 10.1023/A:1007879127182
[^Oonincx_2015]: Oonincx, D.G.A.B., van Broekhoven, S., van Huis, A., van Loon, J.J.A., 2015. Feed conversion, survival and development and composition of four insect species on diets composed of food by-products. PLoS One 10, e0144601.
[^Park_2014]: Park SI, Chang BS, Yoe SM. Detection of antimicrobial substances from larvae of the black soldier fly, Hermetia illucens (Diptera: Stratiomyidae). Entomol Res (2014) 44(2):58–64. doi: 10.1111/1748-5967.12050
[^Park_2015]: Park, S.; Kim, J.; Yoe, S. Purification and characterization of a novel antibacterial peptide from black soldier fly (Hermetia illucens) larvae. Dev. Comp. Immunol. 2015, 52, 98–106. [Google Scholar] [CrossRef]
[^Rana_2015]: Rana, KM Shakil, et al. “Development of black soldier fly larvae production technique as an alternate fish feed.” International Journal of Research in Fisheries and Aquaculture 5.1 (2015): 41-47.
[^Rawski_2020]: Rawski, Mateusz, et al. "Black Soldier Fly Full-Fat Larvae Meal as an Alternative to Fish Meal and Fish Oil in Siberian Sturgeon Nutrition: The Effects on Physical Properties of the Feed, Animal Growth Performance, and Feed Acceptance and Utilization." Animals 10.11 (2020): 2119. ==(PS ( CK, H5, H10, H15, H20, H25, H30: 0%, 10.3%, 20.3%, 30.7%, 40.6%, 50.9%, and 61.3% of fishmeal replacement.  0%, 15.4%, 32.3%, 47.7%, 61.3%, 78.5%, and 95.4% of fish oil replacement))==
[^review_immune]: Sanchez-Muros, M. J., Barroso, F. G., & Manzano-Agugliaro, F. (2014). Insect meal as renewable source of food for animal feeding: A review. Journal of Cleaner Production, 65, 16–27. https://doi.org/10.1016/j.jclepro.2013.11.068
[^Rimoldi_2019]: Rimoldi S, Gini E, Iannini F, Gasco L, Terova G. The effects of dietary insect meal from Hermetia illucens prepupae on autochthonous gut microbiota of rainbow trout (Oncorhynchus mykiss). Animals (2019) 9(4):143. doi: 10.3390/ani9040143
[^Ringo_E_2012]: Ringø E, Zhou Z, Olsen RE, Song SK. Use of chitin and krill in aquaculture–the effect on gut microbiota and the immune system: a review. Aquac Nutr (2012) 18(2):117–131. doi: 10.1111/j.1365-2095.2011.00919.x
[^Roy_S_2017]: Roy S, Kumar V, Kumar V, Behera BK. Acute phase proteins and their potential role as an indicator for fish health and in diagnosis of fish diseases. Protein Pept Lett (2017) 24(1):78–89. doi: 10.2174/0929866524666161121142221
[^Sanchez_M_2014]: Sanchez-Muros, M. J., Barroso, F. G., & Manzano-Agugliaro, F. (2014). Insect meal as renewable source of food for animal feeding: A review. Journal of Cleaner Production, 65, 16–27. https://doi.org/10.1016/j.jclepro.2013.11.068
[^Sanchez_Muros_2015]: Muros MJS, De Haro C, Sanz A, Villareces S, Barroso FG (2015) Nutritional evaluation of Tenebrio molitor meal as fishmeal substitute for Tilapia (Oreochromis niloticus) diet. Aquac Nutr 22(5):943–955
[^Saurabh_2008]: Saurabh S, Sahoo PK. Lysozyme: an important defence molecule of fish innate immune system. Aquac Res (2008) 39(3):223–39. doi: 10.1111/j.1365-2109.2007.01883.x
[^Sealey_2011]: Sealey, W. M., Gaylord, T. G., Barrows, F. T., Tomberlin, J. K., McGuire, M. A., Ross, C., & St-Hilaire, S. (2011). Sensory analysis of rainbow trout, Oncorhynchus mykiss, fed enriched black soldier fly prepupae, Hermetia illucens. Journal of the World Aquaculture Society, 42, 34–45. https://doi.org/10.1111/j.1749-7345.2010.00441.x
[^Shiau_1999]: Shiau, S.Y.; Yu, Y.P. Dietary supplementation of chitin and chitosan depresses growth in tilapia, Oreochromis niloticus × O. aureus. Aquaculture 1999, 179, 439–446. [Google Scholar] [CrossRef]
[^Skrivanova_2006]: Skrivanova ̌ ́ E, Marounek M, Benda V, Brezina P. Susceptibility of ̌ Escherichia coli, Salmonella sp and Clostridium perfringens to organic acids and monolaurin. Vet Med (Praha) (2006) 51(3):81–8. doi: 10.17221/5524-VETMED
[^Skřivanová_2006]: Skřivanová E, Marounek M, Benda V, Březina P. Susceptibility of Escherichia coli, Salmonella sp and Clostridium perfringens to organic acids and monolaurin. Vet Med (Praha) (2006) 51(3):81–8. doi: 10.17221/5524-VETMED
[^Spranghers_2017]: Spranghers T, Ottoboni M, Klootwijk C, Ovyn A, Deboosere S, De Meulenaer B, et al. Nutritional composition of black soldier fly (Hermetia illucens) prepupae reared on different organic waste substrates. J Sci Food Agric (2017) 97(8):2594–600. doi: 10.1002/jsfa.8081
[^Spranghers_2018]: Spranghers T, Michiels J, Vrancx J, Ovyn A, Eeckhout M, De Clercq P, et al. Gut antimicrobial effects and nutritional value of black soldier fly (Hermetia illucens L.) prepupae for weaned piglets. Anim Feed Sci Technol (2018) 235:33–42. doi: 10.1016/j.anifeedsci.2017.08.012
[^Stenberg_2019]: Stenberg OK, Holen E, Piemontese L, Liland NS, Lock E-J, Espe M, et al. Effect of dietary replacement of fish meal with insect meal on in vitro bacterial and viral induced gene response in Atlantic salmon (Salmo salar) head kidney leukocytes. Fish Shellfish Immunol (2019) 91:223–32. doi: 10.1016/j.fsi.2019.05.042
[^St-Hilaire_2007]: St-Hilaire S, Sheppard C, Tomberlin JK, Irving S, Newton L, McGuire MA, Mosley E, Hardy RW, Sealey W (2007) Fly prepupae as a feedstuff for rainbow trout, Oncorhynchus mykiss. J World Aquac Soc 38:59–67
[^St-Hilaire_C_2007]: St-Hilaire, S., Cranfill, K., McGuire, M. A., Mosley, E. E., Tomberlin, J. K., Newton, L., … Irving, S. (2007). Fish offal recycling by the black soldier fly produces a foodstuff high in omega-3 fatty acids. Journal of the World Aquaculture Society, 38, 309–313.
[^St-Onge_2002]: St-Onge, M.P., Jones, P.J.H., 2002. Physiological effects of medium-chain triglycerides: potential agents in the prevention of obesity. J. Nutr. 132, 329-332..
[^St-Onge_2008]: St-Onge, M.P., Bosarge, A., Goree, L.L.T., Darnell, B., 2008. Medium chain triglyceride oil consumption as part of a weight loss diet does not lead to an adverse metabolic profile when compared to olive oil. J. Amer. College Nutr. 27, 547-552.
[^Takuya1_2007]: Mori, Takuya, et al. "Dietary fish oil upregulates intestinal lipid metabolism and reduces body weight gain in C57BL/6J mice." The Journal of nutrition 137.12 (2007): 2629-2634.
[^Terova_2019]: Terova G, Rimoldi S, Ascione C, Gini E, Ceccotti C, Gasco L. Rainbow trout (Oncorhynchus mykiss) gut microbiota is modulated by insect meal from Hermetia illucens prepupae in the diet. Rev Fish Biol Fish (2019) 29:465–86. doi: 10.1007/s11160-019-09558-y
[^Torstensen_2005]: Torstensen, B. E., Bell, J. G., Rosenlund, G., Henderson, R. J., Graff, I. E., Tocher, D. R., … Sargent, J. R. (2005). Tailoring of Atlantic salmon (Salmo salar L.) flesh lipid composition and sensory quality by replacing fish oil with a vegetable oil blend. Journal of Agriculture and Food Chemistry, 53, 10166–10178.
[^Valdenegro_2014]: Valdenegro-Vega VA, Crosbie P, Bridle A, Leef M, Wilson R, Nowak BF. Differentially expressed proteins in gill and skin mucus of Atlantic salmon (Salmo salar) affected by amoebic gill disease. Fish Shellfish Immunol (2014) 40(1):69–77. doi: 10.1016/j.fsi.2014.06.025
[^Van_Huis_2013]: Van Huis A (2013) Potential of insects as food and feed in assuring food security. Annu Revis Entomol 58:563–583
[^Vargas_2019]: Vargas-Abúndez AJ, Randazzo B, Foddai M, Sanchini L, Truzzi C, Giorgini E, et al. Insect meal based diets for clownfish: Biometric, histological, spectroscopic, biochemical and molecular implications. Aquaculture (2019) 498:1–11. doi: 10.1016/j.aquaculture.2018.08.018
[^Vasanth_2015]: Vasanth G, Kiron V, Kulkarni A, Dahle D, Lokesh J, Kitani Y. A microbial feed additive abates intestinal inflammation in Atlantic salmon. Front Immunol (2015) 6:409. doi: 10.3389/fimmu.2015.00409
[^Venold_2012]: Venold, F. F., Penn, M. H., Krogdahl, Å., Overturf, K., 2012. Severity of soybean meal induced distal intestinal inflammation, enterocyte proliferation rate, and fatty acid binding protein (Fabp2) level differ between strains of rainbow trout (Oncorhynchus mykiss). Aquaculture 364, 281-292.
[^Volkman_2004]: Volkman, E.T.; Pangle, K.L.; Rajchel, D.A.; Sutton, T.M. Hatchery performance attributes of juvenile lake sturgeon fed two natural food types. N. Am. J. Aquac. 2004, 66, 105–112.
[^Wagner_2004]: Wagner T, Congleton JL. Blood chemistry correlates of nutritional condition, tissue damage, and stress in migrating juvenile chinook salmon (Oncorhynchus tshawytscha). Can J Fish Aquat Sci (2004) 61(7):1066–74. doi: 10.1139/f04-050
[^Wang_G_2019]: Wang G, Peng K, Hu J, Yi C, Chen X, Wu H, et al. Evaluation of defatted black soldier fly (Hermetia illucens L.) larvae meal as an alternative protein ingredient for juvenile Japanese seabass (Lateolabrax japonicus) diets. Aquaculture (2019) 507:144–54. doi: 10.1016/j.aquaculture.2019.04.023
[^Weththasinghe_2021]:  Weththasinghe P, Hansen JØ, Nøkland D, Lagos L, Rawski M, Øverland M. Full-fat black soldier fly larvae (Hermetia illucens) meal and paste in extruded diets for Atlantic salmon (Salmo salar): Effect on physical pellet quality, nutrient digestibility, nutrient utilization and growth performances. Aquaculture (2021) 530:735785. doi: 10.1016/j.aquaculture.2020.735785
[^Weththasinghe_2_2021]: ththasinghe, Pabodha, et al. “Dietary Inclusion of Black Soldier Fly (Hermetia Illucens) Larvae Meal and Paste Improved Gut Health but Had Minor Effects on Skin Mucus Proteome and Immune Response in Atlantic Salmon (Salmo Salar).” Frontiers in Immunology 12 (2021): 114.
[^Whyte_2007]: Whyte SK. The innate immune response of finfish–a review of current knowledge. Fish Shellfish Immunol (2007) 23(6):1127–51. doi: 10.1016/j.fsi.2007.06.005
[^Wing-Keong_2001]: Wing-Keong N, Liew F-L, Ang LP, Wong KW (2001) Potential of mealworm (Tenebrio molitor) as an alternative protein source in practical diets for African catfish, Clarias gariepinus. Aquac Res 32:273–280
[^Wu_J_2020]: Wu J, Ma N, Johnston LJ, Ma X. Dietary nutrients mediate intestinal host defense peptide expression. Adv Nutr (2020) 11(1):92–102. doi: 10.1093/advances/nmz057
[^Xiang_2020]: Xiang, J., Qin, L., Zhao, D., Xiong, F., Wang, G., Zou, H., Li, W., Li, M., Song, K., Wu,
[^Xiao_2018]: Xiao, Xiaopeng, et al. "Effects of black soldier fly (Hermetia illucens) larvae meal protein as a fishmeal replacement on the growth and immune index of yellow catfish (Pelteobagrus fulvidraco)." Aquaculture research 49.4 (2018): 1569-1577.
[^Xiao_X_2018]: Xiao, Xiaopeng, et al. "Effects of black soldier fly (Hermetia illucens) larvae meal protein as a fishmeal replacement on the growth and immune index of yellow catfish (Pelteobagrus fulvidraco)." Aquaculture research 49.4 (2018): 1569-1577.
[^Xu_X_2020]: Xu, X., Ji, H., Belight, I., Sun, J., 2020. Black soldier fly larvae as a better lipid source than yellow mealworm or silkworm oils for juvenile mirror carp (Cyprinus carpio var. specularis). Aquaculture. 527 (2020) 7354532
[^Xu_X_2_2020]: Xu X, Ji H, Yu H, Zhou J. Influence of dietary black soldier fly (Hermetia illucens Linnaeus) pulp on growth performance, antioxidant capacity and intestinal health of juvenile mirror carp (Cyprinus carpio var. specularis). Aquac Nutr (2020) 26(2):432–43. doi: 10.1111/anu.13005
[^Yixiang_2013]: Yixiang, M., Jianhua, Y., Jinyun, Z., Xianping, Y., Xia, W., Chenglong, S., Pei, L., 2013. The influence of maggot meal and l-carnitine on growth, immunity, antioxidant indices and disease resistance of black carp (Mylopharyngodon piceus). J. Chinese Cer. Oils Assoc. 2, 1-9.
[^Yu_GH_2009]: Yu GH, Chen YH (2009) Research progression on the larvae and prepupae of black soldier fly Hermetia illucens used as animal feedstuff. Chin Bull Entomol 46(1):41–45
[^Zhou_J_2018]: Zhou, J. S., et al. "Effect of replacing dietary fish meal with black soldier fly larvae meal on growth and fatty acid composition of Jian carp (Cyprinus carpio var. Jian)." Aquaculture Nutrition 24.1 (2018): 424-433.
[^Zou_H_2016]: Zou H, Bai X, Feng Y, Zhang Y, Wang Y, Lu W. Influence of long (16L: 8D) and short (8L: 16D) photoperiods on blood metabolites and hepatic metabolism in Olive flounder, Paralichthys olivaceus. Springerplus (2016) 5:924. doi: 10.1186/s40064-016-2614-6
