// waque/lib/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const command_1 = require("@oclif/command");
const signale = tslib_1.__importStar(require("signale"));
const User_1 = tslib_1.__importDefault(require("./User"));
const Config_1 = require("./Config");
class default_1 extends command_1.Command {
    async init() {
        // do some initialization
        const { args, flags } = this.parse();
        this.flags = flags;
        this.args = args;
        if (this.id !== 'login') {
            await this.loadUser();
        }
        if (['export', 'upload'].includes(this.id)) {
            this.loadConfig();
        }
    }
    loadConfig() {
        this.config.lark = Config_1.load();
    }
    async loadUser() {
        const user = new User_1.default(this.config, this.flags.token);
        await user.load();
        if (!user.token) {
            signale.error('请先使用 waque login 登录语雀');
            process.exit(1);
        }
        this.config.currentUser = user;
    }
}
default_1.flags = {
    token: command_1.flags.string({
        char: 't',
        env: 'YUQUE_TOKEN',
    }),
};
exports.default = default_1;
