<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>WGCNA: an R package for weighted correlation network analysis | Karobben</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="WGCNA: an R package for weighted correlation network analysis">
<meta property="og:type" content="website">
<meta property="og:title" content="WGCNA: an R package for weighted correlation network analysis">
<meta property="og:url" content="https://karobben.github.io/LearnNotes/paper_WGCNA.html">
<meta property="og:site_name" content="Karobben">
<meta property="og:description" content="WGCNA: an R package for weighted correlation network analysis">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://i.loli.net/2020/06/08/UWBD5CLEcsvd6Fl.png">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig4_HTML.jpg?as=webp">
<meta property="article:published_time" content="2020-07-16T07:13:33.357Z">
<meta property="article:modified_time" content="2020-07-16T07:13:33.357Z">
<meta property="article:author" content="Karobben">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://i.loli.net/2020/06/08/UWBD5CLEcsvd6Fl.png">
  
    <link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml">
  
  
    <link rel="icon" href="img/tag.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

  
<link rel="stylesheet" href="/css/search.css">

  <script type="text/javascript"
   src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>

  <style>
  pre {
    color: salmon;
  }
  </style>
<meta name="generator" content="Hexo 4.2.1"></head>

<script>
  var _hmt = _hmt || [];
  (function() {
    var hm = document.createElement("script");
    hm.src = "https://hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hm, s);
  })();
</script>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header"; style="background:#1B414B;height:150px">
  <!--div id="banner"></div>-->
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Karobben</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">Karobben Blog</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
          <a id="main-nav-toggle" class="nav-icon"></a>
          <!-- Header Section Start -->
  <header id="slider-area">
    <nav class="navbar navbar-expand-md fixed-top scrolling-navbar bg-white">
      <div class="container">
        <div class="collapse navbar-collapse" id="navbarCollapse">
          <ul class="navbar-nav mr-auto w-100 justify-content-end">
              <a class="main-nav-link" href="../index.html">Home</a>
              <a class="main-nav-link" href="../R/R-index.html">R</a>
              <a class="main-nav-link" href="../Python/index.html">Python</a>
              <a class="main-nav-link" href="../Linux/index.html">Linux</a>
              <a class="main-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
              <a class="main-nav-link" href="../LearnNotes/index.html">Notes</a>
              <a class="main-nav-link" href="../Blog/index.html">Others</a>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <!-- Header Section End -->

      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Looking for Something?"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://karobben.github.io"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><div align='right', style='padding-right:160px;padding-top:1px'>
  
  <span class="local-search local-search-google local-search-plugin" style=";position:absolute;z-index:2;">
      <input type="search" placeholder="Looking for Something?" id="local-search-input" class="local-search-input-cls" style="">
      <div id="local-search-result" class="local-search-result-cls"></div>
  </span>


</div>

<article  id="page-" class="article article-type-page" itemscope itemprop="blogPost">
<div id="tipue_search_content" style="display: none"></div>
  <div class="article-meta">
    <a href="/LearnNotes/paper_WGCNA.html" class="article-date">
  <time datetime="2020-07-16T07:13:33.357Z" itemprop="datePublished">2020-07-16</time>
</a>
    
  </div>
  <div  class="article-inner">
    
    
      <header class="article-header">
        <!--
  
    <h1 class="article-title" itemprop="name">
      WGCNA: an R package for weighted correlation network analysis
    </h1>
  
 -->
      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Go To Top Link -->
        <div title="Back to Top" class="back-to-top" style="background:salmon">
  <a href="#"  >
    <i >Top</i>
  </a>
</div>
<div class="title_index">
  <ul id="nav">
   <li>
     Titles
     <ul>
       <!-- Table of Contents -->

  <div id="toc" class="toc-article">
    <strong class="toc-title">Titles</strong>
    <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#WGCNA-an-R-package-for-weighted-correlation-network-analysis"><span class="toc-number">1.</span> <span class="toc-text">WGCNA: an R package for weighted correlation network analysis</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Abstract"><span class="toc-number">2.</span> <span class="toc-text">Abstract</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Background"><span class="toc-number">3.</span> <span class="toc-text">Background</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Results"><span class="toc-number">4.</span> <span class="toc-text">Results</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-Gene-Cluster"><span class="toc-number">4.1.</span> <span class="toc-text">1. Gene Cluster</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-Module-Detection"><span class="toc-number">4.2.</span> <span class="toc-text">2. Module Detection</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#2-1-Algorithm"><span class="toc-number">4.2.1.</span> <span class="toc-text">2.1 Algorithm</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-2-Biological-Meaning"><span class="toc-number">4.2.2.</span> <span class="toc-text">2.2 Biological Meaning</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-Module-and-Gene-Selection"><span class="toc-number">4.3.</span> <span class="toc-text">3. Module and Gene Selection</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-Topological-Properties"><span class="toc-number">4.4.</span> <span class="toc-text">4. Topological Properties</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#…Skip"><span class="toc-number">4.5.</span> <span class="toc-text">…Skip</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Mouse-Data-Application"><span class="toc-number">5.</span> <span class="toc-text">Mouse Data Application</span></a></li></ol>
  </div>

<!-- Table of Contents Down-->

     </ul>
    </li>
   </ul>
</div>


        <h1 id="WGCNA-an-R-package-for-weighted-correlation-network-analysis"><a href="#WGCNA-an-R-package-for-weighted-correlation-network-analysis" class="headerlink" title="WGCNA: an R package for weighted correlation network analysis"></a>WGCNA: an R package for weighted correlation network analysis</h1><p>Peter Langfelder1 and Steve Horvath*2</p>
<h1 id="Abstract"><a href="#Abstract" class="headerlink" title="Abstract"></a>Abstract</h1><p>Weighted gene co-expression network analysis is a method for describing the <strong>correlation patterns among genes across samples</strong>.<br>WGCNA can find <strong>clusters</strong> (modules) of high correlated genes.<br>Those gene clusters are called as <strong><em>module eigengene</em></strong> or <strong><em>intramodular hub gene</em></strong>.<br>And WGCNA can masure the <strong>correlation of modules to one another or external traits</strong>.<br>It can be used to identify candidate <strong>biomarkers</strong> or <strong>therapeutic targets</strong>.</p>
<h1 id="Background"><a href="#Background" class="headerlink" title="Background"></a>Background</h1><p><strong>Functions of WGCNA</strong>:</p>
<ol>
<li><span style="background:lightgreen">Cluster co-expression genes</span></li>
<li><span style="background:lightgreen">Correlation of modules to traits identification</span></li>
<li><span style="background:lightgreen">Significant modules identification</span></li>
<li><span style="background:lightgreen">Module annotation</span></li>
<li><span style="background:lightgreen">Define the network neighborhood</span></li>
<li><span style="background:lightgreen">Screen of nodes</span></li>
<li><span style="background:lightgreen">Contract network</span></li>
</ol>
<h1 id="Results"><a href="#Results" class="headerlink" title="Results"></a>Results</h1><p><strong>Overview of typical analysis steps and the rational behind them:</strong></p>
<p><img src="https://i.loli.net/2020/06/08/UWBD5CLEcsvd6Fl.png" alt="DeepinScreenshot_select-area_20200608133456"><br>© Peter Langfelder 2008</p>
<ol>
<li>co-expression</li>
<li>Using dynamic tree cut to idnetify modules.</li>
<li>Correlation with traits:<ul>
<li>Clinical data, SNPs, proteomics</li>
<li>ontology, functional enrichment</li>
<li>find biology interest modules</li>
</ul>
</li>
<li>Module relationships</li>
<li>Find the Key drivers in <strong><em>interesting</em></strong> modules</li>
</ol>
<h2 id="1-Gene-Cluster"><a href="#1-Gene-Cluster" class="headerlink" title="1. Gene Cluster"></a>1. Gene Cluster</h2><p><span style="background:red;color:white">Question: Soft Threadhold power????</span></p>
<h2 id="2-Module-Detection"><a href="#2-Module-Detection" class="headerlink" title="2. Module Detection"></a>2. Module Detection</h2><h3 id="2-1-Algorithm"><a href="#2-1-Algorithm" class="headerlink" title="2.1 Algorithm"></a>2.1 Algorithm</h3><p><span style="background:salmon">Modules are defined as clusters of <strong>densely interconnected genes</strong>.</span><br><strong><em>hierarhical cluster</em></strong> is used to cluster the genes.<br><span style="background:lightgreen">Short Coming</span> of this algorithm:</p>
<ul>
<li>Difficult determines how many clusters present in the data set.<br><a href="#paper1">About how to determines the numbers of cluster [1]</a></li>
</ul>
<h3 id="2-2-Biological-Meaning"><a href="#2-2-Biological-Meaning" class="headerlink" title="2.2 Biological Meaning"></a>2.2 Biological Meaning</h3><p>It could reflect:</p>
<ol>
<li>Biological signal</li>
<li>Noise</li>
</ol>
<p>So, gene ontology information can be used.</p>
<p>Algorithms of Modules detection</p>
<ul>
<li>Fuzzy measure of module membership</li>
<li>Automatic block-wise module detection</li>
<li>Consensus module detection</li>
</ul>
<h2 id="3-Module-and-Gene-Selection"><a href="#3-Module-and-Gene-Selection" class="headerlink" title="3. Module and Gene Selection"></a>3. Module and Gene Selection</h2><p>ummmm….<br> = =</p>
<h2 id="4-Topological-Properties"><a href="#4-Topological-Properties" class="headerlink" title="4. Topological Properties"></a>4. Topological Properties</h2><p>To study about network concept.</p>
<ol>
<li>Whole Network Connectivity (degree)</li>
<li>Intramodular Connectivity</li>
<li>Topological Overlap</li>
<li>Clustering Coefficient</li>
</ol>
<h2 id="…Skip"><a href="#…Skip" class="headerlink" title="…Skip"></a>…Skip</h2><h1 id="Mouse-Data-Application"><a href="#Mouse-Data-Application" class="headerlink" title="Mouse Data Application"></a>Mouse Data Application</h1><p>For computational reason, only 3600 most related genes are selected.<br>18 modules…<br><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig4_HTML.jpg?as=webp" alt="img"><br>© Peter Langfelder 2008</p>
<p>As is how in Graph D above, weight is mostly correlated to brown, red, and salmon.<br>By GO enrichment result, we can find that brown is significantly enriched in categories “glycoprotein” and “signal”, red is enriched in “cell cycle”, and “salmon” is enriched in “chromosome”. Overall, it is biological meaningful.</p>
<p>Figure E shows body weight between genes significant…</p>
<hr>
 <li id="paper1">1. Dudoit S, Fridlyand J: A prediction-based resampling method for estimating the number of clusters in a dataset. Genome Biol 2002, 3(7):RESEARCH0036.

<hr>
<p> github: <a href="https://github.com/Karobben" target="_blank" rel="noopener">https://github.com/Karobben</a><br> blog: <a href="http://Karobben.github.io">Karobben.github.io</a><br> R 语言画图索引: <a href="https://karobben.github.io/R/R-index.html">https://karobben.github.io/R/R-index.html</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://karobben.github.io/LearnNotes/paper_WGCNA.html" data-id="ckcogiwgy00432bfc0xwm2cts" class="article-share-link">Share</a>
      
      
    </footer>
  </div>
  
    
  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    

  
    
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/06/">June 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2020/06/30/tmp/">test</a>
          </li>
        
      </ul>
    </div>
  </div>

  
  <!-- Weather -->
<div id="weather-view-he" ></div>
<script>
WIDGET = {ID: 'HfP68rcshb'};
</script>
<script type="text/javascript" src="https://apip.weatherdt.com/view/static/js/r.js?v=1111"></script>

</aside>

          <!-- Weather Plug -->
        
        <!--git talk-->
        <section id="main">
          <div style='padding-top:10px'>
  <div style="background:white">
    <script src="https://utteranc.es/client.js"
            repo="Karobben/utterances"
            issue-term="pathname"
            theme="preferred-color-scheme"
            crossorigin="anonymous"
            async>
    </script>
  </div>
</div>
<!--git talk-->
  <!-- Link Gitalk 的支持文件  -->
  <!--
    <div>
      <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
      <script src="https://unpkg.com/gitalk@latest/dist/gitalk.min.js"></script>
      <div id="gitalk-container"></div>     <script type="text/javascript">
          var gitalk = new Gitalk({

          // gitalk的主要参数
              clientID: '96c208ade65e80e3aaac',
              clientSecret: '86e9240674a3815b0ddc31dababdb4998d5562f4',
              repo: 'OAuth-Application',
              owner: 'Karobben',
              admin: ['Karobben'],
              id:window.location.pathname,
          });
          gitalk.render('gitalk-container');
      </script>
    </div>
<!-- Gitalk end -->

        </section>
        <!-- Gitalk end -->

      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      <span style="float:right; list-style:none">
        <a style="padding: 10px" href="https://karobben.github.io/">Blog</a>
        <a style="padding: 10px" href="https://github.com/Karobben" target="_blank" rel="noopener">GitHub</a>
        <br>
        <a style="padding: 10px" href="https://space.bilibili.com/393056819" target="_blank" rel="noopener">BiliBili</a>
        <a href="mailto:a591465908@outlook.com">Mail me</a>
      </span>
      &copy; Karobben 2021<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
  <span>
    <p align = "center">
      <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
      <span id="busuanzi_container_site_pv">~<span id="busuanzi_value_site_pv"></span>~</span>
    </p>
  </span>
</footer>

    </div>
    <nav id="mobile-nav">
  <!--`
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  -->
  <a class="mobile-nav-link" href="../index.html">Home</a>
  <a class="mobile-nav-link" href="../R/R-index.html">R</a>
  <a class="mobile-nav-link" href="../Python/index.html">Python</a>
  <a class="mobile-nav-link" href="../Linux/index.html">Linux</a>
  <a class="mobile-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
  <a class="mobile-nav-link" href="../LearnNotes/index.html">Notes</a>
  <a class="mobile-nav-link" href="../Blog/index.html">Others</a>
</nav>

    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




<!--mermaid module-->

  <script src='https://unpkg.com/mermaid@7.1.2/dist/mermaid.min.js'></script>
  <script>
    if (window.mermaid) {
      mermaid.initialize({theme: 'forest'});
    }
  </script>



<!-- Searche Module-->
<script src="/js/search.js"></script>
<script src="/js/load.js"></script>
<!-- 这里可能有一些其他原来文件里的代码，也可以没有 -->

<!-- mathjax config similar to math.stackexchange -->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

  </div>
<script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body>
</html>
