<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>Edger - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="Codes for hot to running EdgeR to call differential Expression Genes from RSEM result. (More details you can find in Trinity wiki)"><meta property="og:type" content="blog"><meta property="og:title" content="Edger"><meta property="og:url" content="https://karobben.github.io/2019/08/13/R/edgeR/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="Codes for hot to running EdgeR to call differential Expression Genes from RSEM result. (More details you can find in Trinity wiki)"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://tse3-mm.cn.bing.net/th/id/OIP.hrqFhdEKQNHjEho5wb4tFQHaE8?pid=Api&amp;rs=1"><meta property="article:published_time" content="2019-08-13T05:00:00.000Z"><meta property="article:modified_time" content="2024-01-12T00:29:16.315Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="Bioinformatics"><meta property="article:tag" content="R"><meta property="article:tag" content="RNA-Seq"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://tse3-mm.cn.bing.net/th/id/OIP.hrqFhdEKQNHjEho5wb4tFQHaE8?pid=Api&amp;rs=1"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2019/08/13/R/edgeR/"},"headline":"Karobben","image":[],"datePublished":"2019-08-13T05:00:00.000Z","dateModified":"2024-01-12T00:29:16.315Z","author":{"@type":"Person","name":"Karobben"},"description":"Codes for hot to running EdgeR to call differential Expression Genes from RSEM result. (More details you can find in Trinity wiki)"}</script><link rel="canonical" href="https://karobben.github.io/2019/08/13/R/edgeR/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><div class="card-image"><span class="image is-7by3"><img class="fill" src="https://tse3-mm.cn.bing.net/th/id/OIP.hrqFhdEKQNHjEho5wb4tFQHaE8?pid=Api&amp;rs=1" alt="Edger"></span><span class="ccopy" style="position: absolute; bottom: 0; right: 0; padding-right:3px; padding-left:4px; background-color: rgba(255, 255, 255, 0.6)"></span></div><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2019-08-13T05:00:00.000Z" title="8/13/2019, 12:00:00 AM">2019-08-13</time></span><span class="level-item">Updated&nbsp;<time dateTime="2024-01-12T00:29:16.315Z" title="1/11/2024, 6:29:16 PM">2024-01-11</time></span><span class="level-item"><a class="link-muted" href="/categories/R/">R</a><span> / </span><a class="link-muted" href="/categories/R/Bio/">Bio</a><span> / </span><a class="link-muted" href="/categories/R/Bio/DEG/">DEG</a></span><span class="level-item">37 minutes read (About 5558 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">Edger</h1><div class="content"><h2 id="Edger">Edger</h2>
<h2 id="Example-form-Trinity">Example form Trinity</h2>
<h3 id="Expression-Differential">Expression Differential</h3>
<p>Attention:</p>
<ul>
<li>Make sure Trinity is installed because this scripts needs few functions from re-packaged scripts.</li>
<li>If you didn’t install the Trinity, please download <em>heatmap.3.R</em> <em>misc_rnaseq_funcs</em> <em>pairs3.R</em> <em>vioplot2.R</em> from <a target="_blank" rel="noopener" href="https://github.com/trinityrnaseq/trinityrnaseq/tree/master/Analysis/DifferentialExpression/R">Github</a>, or paste them to your codes instead of source them. (I paste this three codes at the end of the page)</li>
<li>This script is running the differential expression <strong>counts</strong>. So, make sure you are handling an appropriate matrix.</li>
</ul>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs R">library(cluster)<br>library(Biobase)<br>library(qvalue)<br>library(fastcluster)<br>options(stringsAsFactors = <span class="hljs-literal">FALSE</span>)<br>NO_REUSE = <span class="hljs-built_in">F</span><br><br><span class="hljs-comment">## try to reuse earlier-loaded data if possible</span><br><span class="hljs-keyword">if</span> (file.exists(<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.RData&quot;</span>) &amp;&amp; ! NO_REUSE) &#123;<br>    print(<span class="hljs-string">&#x27;RESTORING DATA FROM EARLIER ANALYSIS&#x27;</span>)<br>    load(<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.RData&quot;</span>)<br>&#125; <span class="hljs-keyword">else</span> &#123;<br>  <span class="hljs-comment"># loading Count matrix</span><br>    print(<span class="hljs-string">&#x27;Reading matrix file.&#x27;</span>)<br>    primary_data = read.table(<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix&quot;</span>, header=<span class="hljs-built_in">T</span>, com=<span class="hljs-string">&#x27;&#x27;</span>, row.names=<span class="hljs-number">1</span>, check.names=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&#x27;\t&#x27;</span>)<br>    primary_data = as.matrix(primary_data)<br>&#125;<br><br><span class="hljs-comment">## reading repackaged R function from scripts</span><br><span class="hljs-comment">## similar like import File/lib.py in python</span><br>source(<span class="hljs-string">&quot;/home/ken/Biosoft/trinityrnaseq-Trinity-v2.8.4/Analysis/DifferentialExpression/R/heatmap.3.R&quot;</span>)<br>source(<span class="hljs-string">&quot;/home/ken/Biosoft/trinityrnaseq-Trinity-v2.8.4/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R&quot;</span>)<br>source(<span class="hljs-string">&quot;/home/ken/Biosoft/trinityrnaseq-Trinity-v2.8.4/Analysis/DifferentialExpression/R/pairs3.R&quot;</span>)<br>source(<span class="hljs-string">&quot;/home/ken/Biosoft/trinityrnaseq-Trinity-v2.8.4/Analysis/DifferentialExpression/R/vioplot2.R&quot;</span>)<br><br>data = primary_data<br>myheatcol = colorpanel(<span class="hljs-number">75</span>, <span class="hljs-string">&#x27;purple&#x27;</span>,<span class="hljs-string">&#x27;black&#x27;</span>,<span class="hljs-string">&#x27;yellow&#x27;</span>)<br>sample_types = colnames(data)<br>nsamples = <span class="hljs-built_in">length</span>(sample_types)<br>sample_colors = rainbow(nsamples)<br>sample_type_list = <span class="hljs-built_in">list</span>()<br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nsamples) &#123;<br>    sample_type_list[[sample_types[i]]] = sample_types[i]<br>&#125;<br>sample_factoring = colnames(data)<br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nsamples) &#123;<br>    sample_type = sample_types[i]<br>    replicates_want = sample_type_list[[sample_type]]<br>    sample_factoring[ colnames(data) %in% replicates_want ] = sample_type<br>&#125;<br>initial_matrix = data <span class="hljs-comment"># store before doing various data transformations</span><br>data = log2(data+<span class="hljs-number">1</span>)<br>sample_factoring = colnames(data)<br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nsamples) &#123;<br>    sample_type = sample_types[i]<br>    replicates_want = sample_type_list[[sample_type]]<br>    sample_factoring[ colnames(data) %in% replicates_want ] = sample_type<br>&#125;<br>sampleAnnotations = matrix(ncol=ncol(data),nrow=nsamples)<br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nsamples) &#123;<br>  sampleAnnotations[i,] = colnames(data) %in% sample_type_list[[sample_types[i]]]<br>&#125;<br>sampleAnnotations = apply(sampleAnnotations, <span class="hljs-number">1</span>:<span class="hljs-number">2</span>, <span class="hljs-keyword">function</span>(x) <span class="hljs-built_in">as.logical</span>(x))<br>sampleAnnotations = sample_matrix_to_color_assignments(sampleAnnotations, col=sample_colors)<br>rownames(sampleAnnotations) = as.vector(sample_types)<br>colnames(sampleAnnotations) = colnames(data)<br>data = as.matrix(data) <span class="hljs-comment"># convert to matrix</span><br><br><span class="hljs-comment">## Centering rows</span><br>data = t(scale(t(data), scale=<span class="hljs-built_in">F</span>))<br><br>write.table(data, file=<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.log2.centered.dat&quot;</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&#x27;  &#x27;</span>);<br><span class="hljs-keyword">if</span> (nrow(data) &lt; <span class="hljs-number">2</span>) &#123; stop(<span class="hljs-string">&quot;</span><br><span class="hljs-string"></span><br><span class="hljs-string">**** Sorry, at least two rows are required for this matrix.</span><br><span class="hljs-string"></span><br><span class="hljs-string">&quot;</span>);&#125;<br><span class="hljs-keyword">if</span> (ncol(data) &lt; <span class="hljs-number">2</span>) &#123; stop(<span class="hljs-string">&quot;</span><br><span class="hljs-string"></span><br><span class="hljs-string">**** Sorry, at least two columns are required for this matrix.</span><br><span class="hljs-string"></span><br><span class="hljs-string">&quot;</span>);&#125;<br>sample_cor = cor(data, method=<span class="hljs-string">&#x27;pearson&#x27;</span>, use=<span class="hljs-string">&#x27;pairwise.complete.obs&#x27;</span>)<br>write.table(sample_cor, file=<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.log2.centered.sample_cor.dat&quot;</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&#x27;  &#x27;</span>)<br>sample_dist = dist(t(data), method=<span class="hljs-string">&#x27;euclidean&#x27;</span>)<br>hc_samples = hclust(sample_dist, method=<span class="hljs-string">&#x27;complete&#x27;</span>)<br>pdf(<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.log2.centered.sample_cor_matrix.pdf&quot;</span>)<br>sample_cor_for_plot = sample_cor<br>heatmap.3(sample_cor_for_plot, dendrogram=<span class="hljs-string">&#x27;both&#x27;</span>, Rowv=as.dendrogram(hc_samples), Colv=as.dendrogram(hc_samples), col = myheatcol, scale=<span class="hljs-string">&#x27;none&#x27;</span>, symm=<span class="hljs-literal">TRUE</span>, key=<span class="hljs-literal">TRUE</span>,density.info=<span class="hljs-string">&#x27;none&#x27;</span>, trace=<span class="hljs-string">&#x27;none&#x27;</span>, symkey=<span class="hljs-literal">FALSE</span>, symbreaks=<span class="hljs-built_in">F</span>, margins=<span class="hljs-built_in">c</span>(<span class="hljs-number">10</span>,<span class="hljs-number">10</span>), cexCol=<span class="hljs-number">1</span>, cexRow=<span class="hljs-number">1</span>, cex.main=<span class="hljs-number">0.75</span>, main=paste(<span class="hljs-string">&quot;sample correlation matrix</span><br><span class="hljs-string">&quot;</span>, <span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.log2.centered&quot;</span>) )<br>dev.off()<br>gene_cor = <span class="hljs-literal">NULL</span><br>gene_dist = dist(data, method=<span class="hljs-string">&#x27;euclidean&#x27;</span>)<br><span class="hljs-keyword">if</span> (nrow(data) &lt;= <span class="hljs-number">1</span>) &#123; message(<span class="hljs-string">&#x27;Too few genes to generate heatmap&#x27;</span>); quit(status=<span class="hljs-number">0</span>); &#125;<br>hc_genes = hclust(gene_dist, method=<span class="hljs-string">&#x27;complete&#x27;</span>)<br>heatmap_data = data<br>pdf(<span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf&quot;</span>)<br>heatmap.3(heatmap_data, dendrogram=<span class="hljs-string">&#x27;both&#x27;</span>, Rowv=as.dendrogram(hc_genes), Colv=as.dendrogram(hc_samples), col=myheatcol, scale=<span class="hljs-string">&quot;none&quot;</span>, density.info=<span class="hljs-string">&quot;none&quot;</span>, trace=<span class="hljs-string">&quot;none&quot;</span>, key=<span class="hljs-literal">TRUE</span>, keysize=<span class="hljs-number">1.2</span>, cexCol=<span class="hljs-number">1</span>, margins=<span class="hljs-built_in">c</span>(<span class="hljs-number">10</span>,<span class="hljs-number">10</span>), cex.main=<span class="hljs-number">0.75</span>, main=paste(<span class="hljs-string">&quot;samples vs. features</span><br><span class="hljs-string">&quot;</span>, <span class="hljs-string">&quot;diffExpr.P1e-5_C2.matrix.log2.centered&quot;</span> ) )<br>dev.off()<br><span class="hljs-comment">##save(list=ls(all=TRUE), file=&quot;diffExpr.P1e-5_C2.matrix.RData&quot;)</span><br></code></pre></td></tr></table></figure></div>
<h3 id="Test">Test</h3>
<h2 id="edgeR-out-of-date">edgeR(out of date)</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">~/Biosoft/trinityrnaseq-Trinity-v2.8.4/Analysis/DifferentialExpression/run_DE_analysis.pl  --matrix Intest.table  --method edgeR --dispersion <span class="hljs-number">0.1</span><br><br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br><br><br><br>x &lt;- read.delim(<span class="hljs-string">&quot;3.isoforms.results&quot;</span>)<br>y &lt;- estimateCommonDisp(y)<br>y &lt;- DGEList(counts=<span class="hljs-number">5</span>,group=<span class="hljs-number">1</span>)<br>y &lt;- calcNormFactors(y)<br>y &lt;- estimateTagwiseDisp(y)<br>y &lt;- estimateCommonDisp(y)<br><br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;3.isoforms.results&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">3</span>:<span class="hljs-number">8</span>], genes=rawdata[,<span class="hljs-number">1</span>:<span class="hljs-number">2</span>])<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">8</span>,<span class="hljs-number">8</span>,<span class="hljs-number">33</span>,<span class="hljs-number">33</span>,<span class="hljs-number">51</span>,<span class="hljs-number">51</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br>write.csv(topTags(lrt)$table, file=<span class="hljs-string">&quot;qqq&quot;</span>)<br><span class="hljs-comment">##############################################</span><br>��Դ����������ģ�� edgeR�������裺https://www.tanboyu.com/edger-users-guide.html<br><span class="hljs-comment">######################################################################################</span><br><br><span class="hljs-comment">#####################################################################################</span><br>��׼��<br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br><br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;3.isoforms.results&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">2</span>:<span class="hljs-number">6</span>], genes=rawdata[,<span class="hljs-number">1</span>])<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">8</span>,<span class="hljs-number">8</span>,<span class="hljs-number">51</span>,<span class="hljs-number">51</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br><br><br><span class="hljs-comment">##########################################################</span><br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br><br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;3.isoforms.results&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">5</span>:<span class="hljs-number">8</span>], genes=rawdata[,<span class="hljs-number">1</span>])<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>,<span class="hljs-number">51</span>,<span class="hljs-number">51</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br><br><br><span class="hljs-comment">############################################################</span><br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br><br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;3.isoforms.results&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">5</span>], genes=rawdata[,<span class="hljs-number">1</span>])<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br><br>------------------<br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;6.matrix&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">2</span>:<span class="hljs-number">7</span>], genes=rawdata[,<span class="hljs-number">1</span>])<br>head(rawdata)<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>,<span class="hljs-number">2</span>,<span class="hljs-number">3</span>,<span class="hljs-number">3</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br>write.table(topTags(lrt), file = <span class="hljs-string">&quot;6.txt&quot;</span>,append =<span class="hljs-literal">FALSE</span>, <span class="hljs-built_in">quote</span> = <span class="hljs-literal">TRUE</span>, sep = <span class="hljs-string">&quot; &quot;</span>,eol = <span class="hljs-string">&quot;\n&quot;</span>, na = <span class="hljs-string">&quot;NA&quot;</span>, dec = <span class="hljs-string">&quot;.&quot;</span>,row.names = <span class="hljs-literal">TRUE</span>,col.names = <span class="hljs-literal">TRUE</span>, qmethod = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;escape&quot;</span>, <span class="hljs-string">&quot;double&quot;</span>),fileEncoding = <span class="hljs-string">&quot;&quot;</span>)<br><br>-------------------------------<br>10matrix<br><br><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;Intest.table&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">2</span>:<span class="hljs-number">101</span>], genes=rawdata[,<span class="hljs-number">1</span>])<br>head(rawdata)<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>,<span class="hljs-number">2</span>,<span class="hljs-number">3</span>,<span class="hljs-number">3</span>,<span class="hljs-number">4</span>,<span class="hljs-number">4</span>,<span class="hljs-number">5</span>,<span class="hljs-number">5</span>,<span class="hljs-number">6</span>,<span class="hljs-number">6</span>,<span class="hljs-number">7</span>,<span class="hljs-number">7</span>,<span class="hljs-number">8</span>,<span class="hljs-number">8</span>,<span class="hljs-number">9</span>,<span class="hljs-number">9</span>,<span class="hljs-number">10</span>,<span class="hljs-number">10</span>,<span class="hljs-number">11</span>,<span class="hljs-number">11</span>,<span class="hljs-number">12</span>,<span class="hljs-number">12</span>,<span class="hljs-number">13</span>,<span class="hljs-number">13</span>,<span class="hljs-number">14</span>,<span class="hljs-number">14</span>,<span class="hljs-number">15</span>,<span class="hljs-number">15</span>,<span class="hljs-number">16</span>,<span class="hljs-number">16</span>,<span class="hljs-number">17</span>,<span class="hljs-number">17</span>,<span class="hljs-number">18</span>,<span class="hljs-number">18</span>,<span class="hljs-number">19</span>,<span class="hljs-number">19</span>,<span class="hljs-number">20</span>,<span class="hljs-number">20</span>,<span class="hljs-number">21</span>,<span class="hljs-number">21</span>,<span class="hljs-number">22</span>,<span class="hljs-number">22</span>,<span class="hljs-number">23</span>,<span class="hljs-number">23</span>,<span class="hljs-number">24</span>,<span class="hljs-number">24</span>,<span class="hljs-number">25</span>,<span class="hljs-number">25</span>,<span class="hljs-number">26</span>,<span class="hljs-number">26</span>,<span class="hljs-number">27</span>,<span class="hljs-number">27</span>,<span class="hljs-number">28</span>,<span class="hljs-number">28</span>,<span class="hljs-number">29</span>,<span class="hljs-number">29</span>,<span class="hljs-number">30</span>,<span class="hljs-number">30</span>,<span class="hljs-number">31</span>,<span class="hljs-number">31</span>,<span class="hljs-number">32</span>,<span class="hljs-number">32</span>,<span class="hljs-number">33</span>,<span class="hljs-number">33</span>,<span class="hljs-number">34</span>,<span class="hljs-number">34</span>,<span class="hljs-number">35</span>,<span class="hljs-number">35</span>,<span class="hljs-number">36</span>,<span class="hljs-number">36</span>,<span class="hljs-number">37</span>,<span class="hljs-number">37</span>,<span class="hljs-number">38</span>,<span class="hljs-number">38</span>,<span class="hljs-number">39</span>,<span class="hljs-number">39</span>,<span class="hljs-number">40</span>,<span class="hljs-number">40</span>,<span class="hljs-number">41</span>,<span class="hljs-number">41</span>,<span class="hljs-number">42</span>,<span class="hljs-number">42</span>,<span class="hljs-number">43</span>,<span class="hljs-number">43</span>,<span class="hljs-number">44</span>,<span class="hljs-number">44</span>,<span class="hljs-number">45</span>,<span class="hljs-number">45</span>,<span class="hljs-number">46</span>,<span class="hljs-number">46</span>,<span class="hljs-number">47</span>,<span class="hljs-number">47</span>,<span class="hljs-number">48</span>,<span class="hljs-number">48</span>,<span class="hljs-number">49</span>,<span class="hljs-number">49</span>,<span class="hljs-number">50</span>,<span class="hljs-number">50</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br>write.table(topTags(lrt), file = <span class="hljs-string">&quot;a40.txt&quot;</span>,append =<span class="hljs-literal">FALSE</span>, <span class="hljs-built_in">quote</span> = <span class="hljs-literal">TRUE</span>, sep = <span class="hljs-string">&quot; &quot;</span>,eol = <span class="hljs-string">&quot;\n&quot;</span>, na = <span class="hljs-string">&quot;NA&quot;</span>, dec = <span class="hljs-string">&quot;.&quot;</span>,row.names = <span class="hljs-literal">TRUE</span>,col.names = <span class="hljs-literal">TRUE</span>, qmethod = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;escape&quot;</span>, <span class="hljs-string">&quot;double&quot;</span>),fileEncoding = <span class="hljs-string">&quot;&quot;</span>)<br><br><br><br><span class="hljs-comment">#################################</span><br>3line<br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(<span class="hljs-string">&quot;limma&quot;</span>)<br>library(<span class="hljs-string">&quot;edgeR&quot;</span>)<br>rawdata &lt;- read.delim(<span class="hljs-string">&quot;Intest.table&quot;</span>, check.names=<span class="hljs-literal">FALSE</span>, stringsAsFactors=<span class="hljs-literal">FALSE</span>)<br>y &lt;- DGEList(counts=rawdata[,<span class="hljs-number">2</span>:<span class="hljs-number">4</span>], genes=rawdata[,<span class="hljs-number">1</span>])<br>head(rawdata)<br>y$samples$lib.size &lt;- colSums(y$counts)<br>y &lt;- calcNormFactors(y)<br>Patient &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>,<span class="hljs-number">3</span>))<br>Tissue &lt;- factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;N&quot;</span>,<span class="hljs-string">&quot;T&quot;</span>,<span class="hljs-string">&quot;T2&quot;</span>))<br>data.frame(Sample=colnames(y),Patient,Tissue)<br>design &lt;- model.matrix(~Patient+Tissue)<br>rownames(design) &lt;- colnames(y)<br>y &lt;- estimateGLMCommonDisp(y, design, verbose=<span class="hljs-literal">TRUE</span>)<br>y &lt;- estimateGLMTrendedDisp(y, design)<br>y &lt;- estimateGLMTagwiseDisp(y, design)<br>fit &lt;- glmFit(y, design)<br>lrt &lt;- glmLRT(fit); topTags(lrt)<br>topTags(lrt)<br>write.table(topTags(lrt), file = <span class="hljs-string">&quot;a40.txt&quot;</span>,append =<span class="hljs-literal">FALSE</span>, <span class="hljs-built_in">quote</span> = <span class="hljs-literal">TRUE</span>, sep = <span class="hljs-string">&quot; &quot;</span>,eol = <span class="hljs-string">&quot;\n&quot;</span>, na = <span class="hljs-string">&quot;NA&quot;</span>, dec = <span class="hljs-string">&quot;.&quot;</span>,row.names = <span class="hljs-literal">TRUE</span>,col.names = <span class="hljs-literal">TRUE</span>, qmethod = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;escape&quot;</span>, <span class="hljs-string">&quot;double&quot;</span>),fileEncoding = <span class="hljs-string">&quot;&quot;</span>)<br><br></code></pre></td></tr></table></figure></div>
<hr>
<h2 id="Attachment">Attachment</h2>
<h3 id="heatmap-3-R">heatmap.3.R</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs R"><span class="hljs-comment">## From: https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Analysis/DifferentialExpression/R/heatmap.3.R</span><br><span class="hljs-comment">### pulled from here, and then tweaked slightly: http://www.biostars.org/p/18211/</span><br><span class="hljs-comment">## CODE</span><br>heatmap.3 &lt;- <span class="hljs-keyword">function</span>(x,<br>                      Rowv = <span class="hljs-literal">TRUE</span>, Colv = <span class="hljs-keyword">if</span> (symm) <span class="hljs-string">&quot;Rowv&quot;</span> <span class="hljs-keyword">else</span> <span class="hljs-literal">TRUE</span>,<br>                      distfun = dist,<br>                      hclustfun = hclust,<br>                      dendrogram = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>,<span class="hljs-string">&quot;row&quot;</span>, <span class="hljs-string">&quot;column&quot;</span>, <span class="hljs-string">&quot;none&quot;</span>),<br>                      symm = <span class="hljs-literal">FALSE</span>,<br>                      scale = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;none&quot;</span>,<span class="hljs-string">&quot;row&quot;</span>, <span class="hljs-string">&quot;column&quot;</span>),<br>                      na.rm = <span class="hljs-literal">TRUE</span>,<br>                      revC = identical(Colv,<span class="hljs-string">&quot;Rowv&quot;</span>),<br>                      add.expr,<br>                      breaks,<br>                      symbreaks = <span class="hljs-built_in">max</span>(x &lt; <span class="hljs-number">0</span>, na.rm = <span class="hljs-literal">TRUE</span>) || scale != <span class="hljs-string">&quot;none&quot;</span>,<br>                      col = <span class="hljs-string">&quot;heat.colors&quot;</span>,<br>                      colsep,<br>                      rowsep,<br>                      sepcolor = <span class="hljs-string">&quot;white&quot;</span>,<br>                      sepwidth = <span class="hljs-built_in">c</span>(<span class="hljs-number">0.05</span>, <span class="hljs-number">0.05</span>),<br>                      cellnote,<br>                      notecex = <span class="hljs-number">1</span>,<br>                      notecol = <span class="hljs-string">&quot;cyan&quot;</span>,<br>                      na.color = par(<span class="hljs-string">&quot;bg&quot;</span>),<br>                      trace = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;none&quot;</span>, <span class="hljs-string">&quot;column&quot;</span>,<span class="hljs-string">&quot;row&quot;</span>, <span class="hljs-string">&quot;both&quot;</span>),<br>                      tracecol = <span class="hljs-string">&quot;cyan&quot;</span>,<br>                      hline = median(breaks),<br>                      vline = median(breaks),<br>                      linecol = tracecol,<br>                      margins = <span class="hljs-built_in">c</span>(<span class="hljs-number">5</span>,<span class="hljs-number">5</span>),<br>                      ColSideColors,<br>                      RowSideColors,<br>                      side.height.fraction=<span class="hljs-number">0.1</span>,<br>                      <span class="hljs-comment">#cexRow = 0.2 + 1/log10(max(nr,2)),</span><br>                      <span class="hljs-comment">#cexCol = 0.2 + 1/log10(max(nc,2)),</span><br>        cexRow = <span class="hljs-number">0.2</span>,<br>        cexCol = <span class="hljs-number">0.2</span>,                  <br><br>        scaleRangeMin,<br>        scaleRangeMax,<br><br><br>    cex.main = <span class="hljs-number">1</span>,<br>                      labRow = <span class="hljs-literal">NULL</span>,<br>                      labCol = <span class="hljs-literal">NULL</span>,<br>                      key = <span class="hljs-literal">TRUE</span>,<br>                      keysize = <span class="hljs-number">1.5</span>,<br>                      density.info = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;none&quot;</span>, <span class="hljs-string">&quot;histogram&quot;</span>, <span class="hljs-string">&quot;density&quot;</span>),<br>                      denscol = tracecol,<br>                      symkey = <span class="hljs-built_in">max</span>(x &lt; <span class="hljs-number">0</span>, na.rm = <span class="hljs-literal">TRUE</span>) || symbreaks,<br>                      densadj = <span class="hljs-number">0.25</span>,<br>                      main = <span class="hljs-literal">NULL</span>,<br>                      xlab = <span class="hljs-literal">NULL</span>,<br>                      ylab = <span class="hljs-literal">NULL</span>,<br>                      lmat = <span class="hljs-literal">NULL</span>,<br>                      lhei = <span class="hljs-literal">NULL</span>,<br>                      lwid = <span class="hljs-literal">NULL</span>,<br>                      NumColSideColors = <span class="hljs-number">1</span>,<br>                      NumRowSideColors = <span class="hljs-number">1</span>,<br>                      KeyValueName=<span class="hljs-string">&quot;Value&quot;</span>,...)&#123;<br><br>    invalid &lt;- <span class="hljs-keyword">function</span> (x) &#123;<br>      <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(x) || <span class="hljs-built_in">is.null</span>(x) || <span class="hljs-built_in">length</span>(x) == <span class="hljs-number">0</span>)<br>          <span class="hljs-built_in">return</span>(<span class="hljs-literal">TRUE</span>)<br>      <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.list</span>(x))<br>          <span class="hljs-built_in">return</span>(<span class="hljs-built_in">all</span>(sapply(x, invalid)))<br>      <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (is.vector(x))<br>          <span class="hljs-built_in">return</span>(<span class="hljs-built_in">all</span>(<span class="hljs-built_in">is.na</span>(x)))<br>      <span class="hljs-keyword">else</span> <span class="hljs-built_in">return</span>(<span class="hljs-literal">FALSE</span>)<br>    &#125;<br><br><br><br>    x &lt;- as.matrix(x)<br>    scale01 &lt;- <span class="hljs-keyword">function</span>(x, low = <span class="hljs-built_in">min</span>(x), high = <span class="hljs-built_in">max</span>(x)) &#123;<br>        x &lt;- (x - low)/(high - low)<br>        x<br>    &#125;<br><br>    retval &lt;- <span class="hljs-built_in">list</span>()<br><br><br>    scale &lt;- <span class="hljs-keyword">if</span> (symm &amp;&amp; <span class="hljs-built_in">missing</span>(scale))<br>        <span class="hljs-string">&quot;none&quot;</span><br>    <span class="hljs-keyword">else</span> match.arg(scale)<br><br>    dendrogram &lt;- match.arg(dendrogram)<br><br>    trace &lt;- match.arg(trace)<br><br>    density.info &lt;- match.arg(density.info)<br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(col) == <span class="hljs-number">1</span> &amp;&amp; <span class="hljs-built_in">is.character</span>(col))<br>        col &lt;- get(col, mode = <span class="hljs-string">&quot;function&quot;</span>)<br><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(breaks) &amp;&amp; (scale != <span class="hljs-string">&quot;none&quot;</span>))<br>        warning(<span class="hljs-string">&quot;Using scale=\&quot;row\&quot; or scale=\&quot;column\&quot; when breaks are&quot;</span>,<br>            <span class="hljs-string">&quot;specified can produce unpredictable results.&quot;</span>, <span class="hljs-string">&quot;Please consider using only one or the other.&quot;</span>)<br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(Rowv) || <span class="hljs-built_in">is.na</span>(Rowv))<br>        Rowv &lt;- <span class="hljs-literal">FALSE</span><br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(Colv) || <span class="hljs-built_in">is.na</span>(Colv))<br>        Colv &lt;- <span class="hljs-literal">FALSE</span><br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (Colv == <span class="hljs-string">&quot;Rowv&quot;</span> &amp;&amp; !isTRUE(Rowv))<br>        Colv &lt;- <span class="hljs-literal">FALSE</span><br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(di &lt;- <span class="hljs-built_in">dim</span>(x)) != <span class="hljs-number">2</span> || !<span class="hljs-built_in">is.numeric</span>(x))<br>        stop(<span class="hljs-string">&quot;`x&#x27; must be a numeric matrix&quot;</span>)<br><br>    nr &lt;- di[<span class="hljs-number">1</span>]<br>    nc &lt;- di[<span class="hljs-number">2</span>]<br><br>    <span class="hljs-keyword">if</span> (nr &lt;= <span class="hljs-number">1</span> || nc &lt;= <span class="hljs-number">1</span>)<br>        stop(<span class="hljs-string">&quot;`x&#x27; must have at least 2 rows and 2 columns&quot;</span>)<br>    <span class="hljs-comment">#print(paste(&quot;nr:&quot;, nr, &quot;nc:&quot;, nc, &quot;cexCol:&quot;, cexCol, &quot;cexRow:&quot;, cexRow))</span><br>    <span class="hljs-comment">#stop(&quot;debug&quot;)</span><br><br><br><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.numeric</span>(margins) || <span class="hljs-built_in">length</span>(margins) != <span class="hljs-number">2</span>)<br>        stop(<span class="hljs-string">&quot;`margins&#x27; must be a numeric vector of length 2&quot;</span>)<br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(cellnote))<br>        cellnote &lt;- matrix(<span class="hljs-string">&quot;&quot;</span>, ncol = ncol(x), nrow = nrow(x))<br><br>    <span class="hljs-keyword">if</span> (!inherits(Rowv, <span class="hljs-string">&quot;dendrogram&quot;</span>)) &#123;<br>        <span class="hljs-keyword">if</span> (((!isTRUE(Rowv)) || (<span class="hljs-built_in">is.null</span>(Rowv))) &amp;&amp; (dendrogram %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>, <span class="hljs-string">&quot;row&quot;</span>))) &#123;<br>            <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.logical</span>(Colv) &amp;&amp; (Colv))<br>                dendrogram &lt;- <span class="hljs-string">&quot;column&quot;</span><br>            <span class="hljs-keyword">else</span> dedrogram &lt;- <span class="hljs-string">&quot;none&quot;</span><br><br>            warning(<span class="hljs-string">&quot;Discrepancy: Rowv is FALSE, while dendrogram is `&quot;</span>,<br>                dendrogram, <span class="hljs-string">&quot;&#x27;. Omitting row dendogram.&quot;</span>)<br>        &#125;<br>    &#125;<br><br>    <span class="hljs-keyword">if</span> (!inherits(Colv, <span class="hljs-string">&quot;dendrogram&quot;</span>)) &#123;<br>        <span class="hljs-keyword">if</span> (((!isTRUE(Colv)) || (<span class="hljs-built_in">is.null</span>(Colv))) &amp;&amp; (dendrogram %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>, <span class="hljs-string">&quot;column&quot;</span>))) &#123;<br>            <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.logical</span>(Rowv) &amp;&amp; (Rowv))<br>                dendrogram &lt;- <span class="hljs-string">&quot;row&quot;</span><br>            <span class="hljs-keyword">else</span> dendrogram &lt;- <span class="hljs-string">&quot;none&quot;</span><br><br>            warning(<span class="hljs-string">&quot;Discrepancy: Colv is FALSE, while dendrogram is `&quot;</span>,<br>                dendrogram, <span class="hljs-string">&quot;&#x27;. Omitting column dendogram.&quot;</span>)<br>        &#125;<br>    &#125;<br><br>   <span class="hljs-keyword">if</span> (inherits(Rowv, <span class="hljs-string">&quot;dendrogram&quot;</span>)) &#123;<br>        ddr &lt;- Rowv<br>        rowInd &lt;- order.dendrogram(ddr)<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.integer</span>(Rowv)) &#123;<br>        hcr &lt;- hclustfun(distfun(x))<br>        ddr &lt;- as.dendrogram(hcr)<br>        ddr &lt;- reorder(ddr, Rowv)<br>        rowInd &lt;- order.dendrogram(ddr)<br>        <span class="hljs-keyword">if</span> (nr != <span class="hljs-built_in">length</span>(rowInd))<br>            stop(<span class="hljs-string">&quot;row dendrogram ordering gave index of wrong length&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (isTRUE(Rowv)) &#123;<br>        Rowv &lt;- rowMeans(x, na.rm = na.rm)<br>        hcr &lt;- hclustfun(distfun(x))<br>        ddr &lt;- as.dendrogram(hcr)<br>        ddr &lt;- reorder(ddr, Rowv)<br>        rowInd &lt;- order.dendrogram(ddr)<br>        <span class="hljs-keyword">if</span> (nr != <span class="hljs-built_in">length</span>(rowInd))<br>            stop(<span class="hljs-string">&quot;row dendrogram ordering gave index of wrong length&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>        rowInd &lt;- nr:<span class="hljs-number">1</span><br>    &#125;<br><br>   <span class="hljs-keyword">if</span> (inherits(Colv, <span class="hljs-string">&quot;dendrogram&quot;</span>)) &#123;<br>        ddc &lt;- Colv<br>        colInd &lt;- order.dendrogram(ddc)<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (identical(Colv, <span class="hljs-string">&quot;Rowv&quot;</span>)) &#123;<br>        <span class="hljs-keyword">if</span> (nr != nc)<br>            stop(<span class="hljs-string">&quot;Colv = \&quot;Rowv\&quot; but nrow(x) != ncol(x)&quot;</span>)<br>        <span class="hljs-keyword">if</span> (exists(<span class="hljs-string">&quot;ddr&quot;</span>)) &#123;<br>            ddc &lt;- ddr<br>            colInd &lt;- order.dendrogram(ddc)<br>        &#125;<br>        <span class="hljs-keyword">else</span> colInd &lt;- rowInd<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.integer</span>(Colv)) &#123;<br>        hcc &lt;- hclustfun(distfun(<span class="hljs-keyword">if</span> (symm)<br>            x<br>        <span class="hljs-keyword">else</span> t(x)))<br>        ddc &lt;- as.dendrogram(hcc)<br>        ddc &lt;- reorder(ddc, Colv)<br>        colInd &lt;- order.dendrogram(ddc)<br>        <span class="hljs-keyword">if</span> (nc != <span class="hljs-built_in">length</span>(colInd))<br>            stop(<span class="hljs-string">&quot;column dendrogram ordering gave index of wrong length&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (isTRUE(Colv)) &#123;<br>        Colv &lt;- colMeans(x, na.rm = na.rm)<br>        hcc &lt;- hclustfun(distfun(<span class="hljs-keyword">if</span> (symm)<br>            x<br>        <span class="hljs-keyword">else</span> t(x)))<br>        ddc &lt;- as.dendrogram(hcc)<br>        ddc &lt;- reorder(ddc, Colv)<br>        colInd &lt;- order.dendrogram(ddc)<br>        <span class="hljs-keyword">if</span> (nc != <span class="hljs-built_in">length</span>(colInd))<br>            stop(<span class="hljs-string">&quot;column dendrogram ordering gave index of wrong length&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>        colInd &lt;- 1:nc<br>    &#125;<br><br>    retval$rowInd &lt;- rowInd<br>    retval$colInd &lt;- colInd<br>    retval$call &lt;- match.call()<br><br>    x &lt;- x[rowInd, colInd]  <span class="hljs-comment"># rearrange matrix according to dendrograms</span><br>    x.unscaled &lt;- x<br><br>    cellnote &lt;- cellnote[rowInd, colInd]  <span class="hljs-comment"># also rearrange the cellnotes</span><br><br>    <span class="hljs-comment"># get labels</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(labRow))<br>        labRow &lt;- <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(rownames(x)))<br>            (<span class="hljs-number">1</span>:nr)[rowInd]<br>        <span class="hljs-keyword">else</span> rownames(x)<br>    <span class="hljs-keyword">else</span> labRow &lt;- labRow[rowInd]<br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(labCol))<br>        labCol &lt;- <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(colnames(x)))<br>            (<span class="hljs-number">1</span>:nc)[colInd]<br>        <span class="hljs-keyword">else</span> colnames(x)<br>    <span class="hljs-keyword">else</span> labCol &lt;- labCol[colInd]<br><br><br>    <span class="hljs-comment">## do scaling of matrix according to Z-scores</span><br>    <span class="hljs-keyword">if</span> (scale == <span class="hljs-string">&quot;row&quot;</span>) &#123;<br>        retval$rowMeans &lt;- rm &lt;- rowMeans(x, na.rm = na.rm)<br>        x &lt;- sweep(x, <span class="hljs-number">1</span>, rm)<br>        retval$rowSDs &lt;- sx &lt;- apply(x, <span class="hljs-number">1</span>, sd, na.rm = na.rm)<br>        x &lt;- sweep(x, <span class="hljs-number">1</span>, sx, <span class="hljs-string">&quot;/&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (scale == <span class="hljs-string">&quot;column&quot;</span>) &#123;<br>        retval$colMeans &lt;- rm &lt;- colMeans(x, na.rm = na.rm)<br>        x &lt;- sweep(x, <span class="hljs-number">2</span>, rm)<br>        retval$colSDs &lt;- sx &lt;- apply(x, <span class="hljs-number">2</span>, sd, na.rm = na.rm)<br>        x &lt;- sweep(x, <span class="hljs-number">2</span>, sx, <span class="hljs-string">&quot;/&quot;</span>)<br>    &#125;<br><br>    <span class="hljs-comment"># number of breaks</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(breaks) || <span class="hljs-built_in">is.null</span>(breaks) || <span class="hljs-built_in">length</span>(breaks) &lt; <span class="hljs-number">1</span>) &#123;<br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(col) || <span class="hljs-built_in">is.function</span>(col))<br>            breaks &lt;- 16<br>        <span class="hljs-keyword">else</span> breaks &lt;- <span class="hljs-built_in">length</span>(col) + <span class="hljs-number">1</span><br>    &#125;<br><br>    <span class="hljs-comment"># set breakpoints</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(breaks) == <span class="hljs-number">1</span>) &#123;<br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(scaleRangeMin))<br>            scaleRangeMin = <span class="hljs-built_in">min</span>(x, na.rm=na.rm)<br><br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(scaleRangeMax))<br>            scaleRangeMax = <span class="hljs-built_in">max</span>(x, na.rm=na.rm)<br><br><br>        <span class="hljs-keyword">if</span> (!symbreaks) &#123;<br>            breaks &lt;- seq(scaleRangeMin, scaleRangeMax, <span class="hljs-built_in">length</span>=breaks);<br>        &#125; <span class="hljs-keyword">else</span> &#123;<br>            <span class="hljs-comment">#extreme &lt;- max(abs(x), na.rm = TRUE)</span><br>            extreme = <span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(<span class="hljs-built_in">c</span>(scaleRangeMin,scaleRangeMax)), na.rm=na.rm)<br>            breaks &lt;- seq(-extreme, extreme, <span class="hljs-built_in">length</span> = breaks)<br>        &#125;<br>    &#125;<br><br>    nbr &lt;- <span class="hljs-built_in">length</span>(breaks)<br>    ncol &lt;- <span class="hljs-built_in">length</span>(breaks) - <span class="hljs-number">1</span><br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">class</span>(col) == <span class="hljs-string">&quot;function&quot;</span>)<br>        col &lt;- col(ncol)<br><br>    min.breaks &lt;- <span class="hljs-built_in">min</span>(breaks)<br>    max.breaks &lt;- <span class="hljs-built_in">max</span>(breaks)<br><br>    <span class="hljs-comment"># adjust for out-of-range given break settings</span><br>    x[x &lt; min.breaks] &lt;- min.breaks<br>    x[x &gt; max.breaks] &lt;- max.breaks<br><br>    <span class="hljs-comment"># layout height</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(lhei) || <span class="hljs-built_in">is.null</span>(lhei))<br>        lhei &lt;- <span class="hljs-built_in">c</span>(keysize, <span class="hljs-number">4</span>)<br><br>    <span class="hljs-comment"># layout width</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(lwid) || <span class="hljs-built_in">is.null</span>(lwid))<br>        lwid &lt;- <span class="hljs-built_in">c</span>(keysize, <span class="hljs-number">4</span>)<br><br>    <span class="hljs-comment"># define the layout</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(lmat) || <span class="hljs-built_in">is.null</span>(lmat)) &#123;<br>        lmat &lt;- rbind(<span class="hljs-number">4</span>:<span class="hljs-number">3</span>, <span class="hljs-number">2</span>:<span class="hljs-number">1</span>)<br><br>        <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(ColSideColors)) &#123;<br>            <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.character</span>(ColSideColors) || ncol(ColSideColors) != nc)<br>                stop(<span class="hljs-string">&quot;&#x27;ColSideColors&#x27; must be a matrix of ncol(x) &quot;</span>, nc, <span class="hljs-string">&quot; columns&quot;</span>)<br>            lmat &lt;- rbind(lmat[<span class="hljs-number">1</span>, ] + <span class="hljs-number">1</span>, <span class="hljs-built_in">c</span>(<span class="hljs-literal">NA</span>, <span class="hljs-number">1</span>), lmat[<span class="hljs-number">2</span>, ] + <span class="hljs-number">1</span>)<br>            <span class="hljs-comment">#lhei=c(lhei[1], side.height.fraction*NumColSideColors, lhei[2])</span><br>            side_height = <span class="hljs-built_in">min</span>(side.height.fraction*nrow(ColSideColors), <span class="hljs-number">1</span>);<br>            lhei=<span class="hljs-built_in">c</span>(lhei[<span class="hljs-number">1</span>], side_height, lhei[<span class="hljs-number">2</span>])<br>        &#125;<br><br>        <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(RowSideColors)) &#123;<br>            <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.character</span>(RowSideColors) || nrow(RowSideColors) != nr)<br>                stop(<span class="hljs-string">&quot;&#x27;RowSideColors&#x27; must be a matrix of nrow(x) &quot;</span>, nr, <span class="hljs-string">&quot; rows.  It currently has &quot;</span>, nrow(RowSideColors), <span class="hljs-string">&quot; rows.&quot;</span>)<br>            lmat &lt;- cbind(lmat[, <span class="hljs-number">1</span>] + <span class="hljs-number">1</span>, <span class="hljs-built_in">c</span>(<span class="hljs-built_in">rep</span>(<span class="hljs-literal">NA</span>, nrow(lmat) - <span class="hljs-number">1</span>), <span class="hljs-number">1</span>), lmat[,<span class="hljs-number">2</span>] + <span class="hljs-number">1</span>)<br>            <span class="hljs-comment">#lwid &lt;- c(lwid[1], side.height.fraction*NumRowSideColors, lwid[2])</span><br>            side_width = <span class="hljs-built_in">min</span>(side.height.fraction*ncol(RowSideColors), <span class="hljs-number">1</span>);<br>      lwid &lt;- <span class="hljs-built_in">c</span>(lwid[<span class="hljs-number">1</span>], side_width, lwid[<span class="hljs-number">2</span>])<br>        &#125;<br>        lmat[<span class="hljs-built_in">is.na</span>(lmat)] &lt;- <span class="hljs-number">0</span><br>    &#125;<br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(lhei) != nrow(lmat))<br>        stop(<span class="hljs-string">&quot;lhei must have length = nrow(lmat) = &quot;</span>, nrow(lmat))<br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(lwid) != ncol(lmat))<br>        stop(<span class="hljs-string">&quot;lwid must have length = ncol(lmat) =&quot;</span>, ncol(lmat))<br><br><br>  op &lt;- par(no.readonly = <span class="hljs-literal">TRUE</span>)<br>    <span class="hljs-built_in">on.exit</span>(par(op))<br><br>    layout(lmat, widths = lwid, heights = lhei, respect = <span class="hljs-literal">FALSE</span>)<br><br>  <span class="hljs-comment">###########################################</span><br>  <span class="hljs-comment">## Draw the colorbars for the annotations:</span><br>  <span class="hljs-comment">###########################################</span><br><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(RowSideColors)) &#123;<br>        <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.matrix</span>(RowSideColors))&#123;<br>                par(mar = <span class="hljs-built_in">c</span>(margins[<span class="hljs-number">1</span>], <span class="hljs-number">0</span>, <span class="hljs-number">0</span>, <span class="hljs-number">0.5</span>))<br>                image(rbind(<span class="hljs-number">1</span>:nr), col = RowSideColors[rowInd], axes = <span class="hljs-literal">FALSE</span>)<br>        &#125; <span class="hljs-keyword">else</span> &#123;<br>            par(mar = <span class="hljs-built_in">c</span>(margins[<span class="hljs-number">1</span>], <span class="hljs-number">0</span>, <span class="hljs-number">0</span>, <span class="hljs-number">0.5</span>))<br>            rsc = t(RowSideColors[rowInd, , drop=<span class="hljs-built_in">F</span>])<br>            rsc.colors = matrix()<br>            rsc.names = <span class="hljs-built_in">names</span>(table(rsc))<br>            rsc.i = <span class="hljs-number">1</span><br>            <span class="hljs-keyword">for</span> (rsc.name <span class="hljs-keyword">in</span> rsc.names) &#123;<br>                rsc.colors[rsc.i] = rsc.name<br>                rsc[rsc == rsc.name] = rsc.i<br>                rsc.i = rsc.i + <span class="hljs-number">1</span><br>            &#125;<br>            <span class="hljs-comment"># print(rsc)</span><br>            rsc = matrix(<span class="hljs-built_in">as.numeric</span>(rsc), nrow = <span class="hljs-built_in">dim</span>(rsc)[<span class="hljs-number">1</span>])<br>            <span class="hljs-comment">#print(&quot;RSC: &quot;, rsc)</span><br>            <span class="hljs-comment">#print(rsc.colors)    </span><br>            image(<span class="hljs-number">1</span>:nrow(rsc), <span class="hljs-number">1</span>:ncol(rsc), rsc, col = as.vector(rsc.colors), axes = <span class="hljs-literal">FALSE</span>, xlab=<span class="hljs-string">&quot;&quot;</span>, ylab=<span class="hljs-string">&quot;&quot;</span>)<br><br>      <span class="hljs-comment"># add labels</span><br>            <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(colnames(RowSideColors)) &gt; <span class="hljs-number">0</span>) &#123;  <br>                <span class="hljs-comment">#axis(1, 0:(dim(rsc)[2] - 1)/(dim(rsc)[2] - 1), rownames(RowSideColors), las = 2, tick = FALSE)</span><br>        <span class="hljs-comment">#axis(1, 0:(nrow(rsc)-1), colnames(RowSideColors), las = 2, tick = T) # ncol because transposed</span><br>              axis(<span class="hljs-number">1</span>, <span class="hljs-number">1</span>:ncol(RowSideColors), labels=colnames(RowSideColors), las=<span class="hljs-number">2</span>, cex.axis=<span class="hljs-number">0.5</span>, tick=<span class="hljs-built_in">F</span>, xlab=<span class="hljs-string">&quot;&quot;</span>, ylab=<span class="hljs-string">&quot;&quot;</span>)<br><br>      &#125;<br>        &#125;<br>    &#125;<br><br><br><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(ColSideColors)) &#123;<br><br>        <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.matrix</span>(ColSideColors))&#123;<br>            par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">0.5</span>, <span class="hljs-number">0</span>, <span class="hljs-number">0</span>, margins[<span class="hljs-number">2</span>]))<br>            image(cbind(<span class="hljs-number">1</span>:nc), col = ColSideColors[colInd], axes = <span class="hljs-literal">FALSE</span>)<br>        &#125; <span class="hljs-keyword">else</span> &#123;<br>            par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">0.5</span>, <span class="hljs-number">0</span>, <span class="hljs-number">0</span>, margins[<span class="hljs-number">2</span>]))<br>            csc = ColSideColors[, colInd, drop=<span class="hljs-built_in">F</span>]<br>            csc.colors = matrix()<br>            csc.names = <span class="hljs-built_in">names</span>(table(csc))<br>            csc.i = <span class="hljs-number">1</span><br>            <span class="hljs-keyword">for</span> (csc.name <span class="hljs-keyword">in</span> csc.names) &#123;<br>                csc.colors[csc.i] = csc.name<br>                csc[csc == csc.name] = csc.i<br>                csc.i = csc.i + <span class="hljs-number">1</span><br>            &#125;<br>            csc = matrix(<span class="hljs-built_in">as.numeric</span>(csc), nrow = <span class="hljs-built_in">dim</span>(csc)[<span class="hljs-number">1</span>])<br>            <span class="hljs-comment">#print(csc)</span><br>            image(<span class="hljs-number">1</span>:nrow(t(csc)), <span class="hljs-number">1</span>:ncol(t(csc)), t(csc), col = as.vector(csc.colors), axes = <span class="hljs-literal">FALSE</span>, xlab=<span class="hljs-string">&quot;&quot;</span>, ylab=<span class="hljs-string">&quot;&quot;</span>)<br><br>      <span class="hljs-comment"># add labels</span><br>            <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(rownames(ColSideColors)) &gt; <span class="hljs-number">0</span>) &#123;<br>                <span class="hljs-comment">#axis(2, 0:(dim(csc)[2] - 1)/max(1,(dim(csc)[2] - 1)), colnames(ColSideColors), las = 2, tick = FALSE)</span><br>        axis(<span class="hljs-number">2</span>, <span class="hljs-number">1</span>:(nrow(ColSideColors)), labels=rownames(ColSideColors), las = <span class="hljs-number">2</span>, tick = <span class="hljs-literal">FALSE</span>, cex.axis=<span class="hljs-number">0.5</span>)<br>            &#125;<br>        &#125;<br>    &#125;<br><br><br><br>    par(mar = <span class="hljs-built_in">c</span>(margins[<span class="hljs-number">1</span>], <span class="hljs-number">0</span>, <span class="hljs-number">0</span>, margins[<span class="hljs-number">2</span>]))<br>    x &lt;- t(x)<br>    cellnote &lt;- t(cellnote)<br>    <span class="hljs-keyword">if</span> (revC) &#123;<br>        iy &lt;- nr:<span class="hljs-number">1</span><br>        <span class="hljs-keyword">if</span> (exists(<span class="hljs-string">&quot;ddr&quot;</span>))<br>            ddr &lt;- rev(ddr)<br>        x &lt;- x[, iy]<br>        cellnote &lt;- cellnote[, iy]<br>    &#125;<br>    <span class="hljs-keyword">else</span> iy &lt;- 1:nr<br><br>  <span class="hljs-comment"># draw the central heatmap</span><br>  image(<span class="hljs-number">1</span>:nc, <span class="hljs-number">1</span>:nr, x, xlim = <span class="hljs-number">0.5</span> + <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, nc), ylim = <span class="hljs-number">0.5</span> + <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, nr), axes = <span class="hljs-literal">FALSE</span>, xlab = <span class="hljs-string">&quot;&quot;</span>, ylab = <span class="hljs-string">&quot;&quot;</span>, col = col, breaks = breaks, ...)<br><br>  <span class="hljs-comment"># store the matrix drawn</span><br>  retval$carpet &lt;- x<br><br>  <span class="hljs-comment"># store the dendrograms</span><br>  <span class="hljs-keyword">if</span> (exists(<span class="hljs-string">&quot;ddr&quot;</span>))<br>        retval$rowDendrogram &lt;- ddr<br>    <span class="hljs-keyword">if</span> (exists(<span class="hljs-string">&quot;ddc&quot;</span>))<br>        retval$colDendrogram &lt;- ddc<br><br>  <span class="hljs-comment"># store the breaks</span><br>  retval$breaks &lt;- breaks<br><br>  <span class="hljs-comment"># store the colormap used</span><br>  retval$col &lt;- col<br><br>  <span class="hljs-comment"># specially color in the na values</span><br>    <span class="hljs-keyword">if</span> (!invalid(na.color) &amp; <span class="hljs-built_in">any</span>(<span class="hljs-built_in">is.na</span>(x))) &#123; <span class="hljs-comment"># load library(gplots)</span><br>        mmat &lt;- ifelse(<span class="hljs-built_in">is.na</span>(x), <span class="hljs-number">1</span>, <span class="hljs-literal">NA</span>)<br>        image(<span class="hljs-number">1</span>:nc, <span class="hljs-number">1</span>:nr, mmat, axes = <span class="hljs-literal">FALSE</span>, xlab = <span class="hljs-string">&quot;&quot;</span>, ylab = <span class="hljs-string">&quot;&quot;</span>, col = na.color, add = <span class="hljs-literal">TRUE</span>)<br>    &#125;<br><br>  <span class="hljs-comment"># X-axis column labels</span><br>    axis(<span class="hljs-number">1</span>, <span class="hljs-number">1</span>:nc, labels = labCol, las = <span class="hljs-number">2</span>, line = -<span class="hljs-number">0.5</span>, tick = <span class="hljs-number">0</span>, cex.axis = cexCol)<br><br>  <span class="hljs-comment"># X-axis title</span><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(xlab))<br>        mtext(xlab, side = <span class="hljs-number">1</span>, line = margins[<span class="hljs-number">1</span>] - <span class="hljs-number">1.25</span>)<br><br>  <span class="hljs-comment"># Y-axis row labeling</span><br>    axis(<span class="hljs-number">4</span>, iy, labels = labRow, las = <span class="hljs-number">2</span>, line = -<span class="hljs-number">0.5</span>, tick = <span class="hljs-number">0</span>,<br>        cex.axis = cexRow)<br><br>  <span class="hljs-comment"># Y-axis title</span><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(ylab))<br>        mtext(ylab, side = <span class="hljs-number">4</span>, line = margins[<span class="hljs-number">2</span>] - <span class="hljs-number">1.25</span>)<br><br>     <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(add.expr))<br>        eval(<span class="hljs-built_in">substitute</span>(add.expr))<br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(colsep))<br>        <span class="hljs-keyword">for</span> (csep <span class="hljs-keyword">in</span> colsep) rect(xleft = csep + <span class="hljs-number">0.5</span>, ybottom = <span class="hljs-built_in">rep</span>(<span class="hljs-number">0</span>, <span class="hljs-built_in">length</span>(csep)), xright = csep + <span class="hljs-number">0.5</span> + sepwidth[<span class="hljs-number">1</span>], ytop = <span class="hljs-built_in">rep</span>(ncol(x) + <span class="hljs-number">1</span>, csep), lty = <span class="hljs-number">1</span>, lwd = <span class="hljs-number">1</span>, col = sepcolor, border = sepcolor)<br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(rowsep))<br>        <span class="hljs-keyword">for</span> (rsep <span class="hljs-keyword">in</span> rowsep) rect(xleft = <span class="hljs-number">0</span>, ybottom = (ncol(x) + <span class="hljs-number">1</span> - rsep) - <span class="hljs-number">0.5</span>, xright = nrow(x) + <span class="hljs-number">1</span>, ytop = (ncol(x) + <span class="hljs-number">1</span> - rsep) - <span class="hljs-number">0.5</span> - sepwidth[<span class="hljs-number">2</span>], lty = <span class="hljs-number">1</span>, lwd = <span class="hljs-number">1</span>, col = sepcolor, border = sepcolor)<br><br><br>  min.scale &lt;- <span class="hljs-built_in">min</span>(breaks)<br>    max.scale &lt;- <span class="hljs-built_in">max</span>(breaks)<br>    x.scaled &lt;- scale01(t(x), min.scale, max.scale)<br><br>  <span class="hljs-comment"># column trace</span><br>  <span class="hljs-keyword">if</span> (trace %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>, <span class="hljs-string">&quot;column&quot;</span>)) &#123;<br>        retval$vline &lt;- vline<br>        vline.vals &lt;- scale01(vline, min.scale, max.scale)<br>        <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> colInd) &#123;<br>            <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(vline)) &#123;<br>                abline(v = i - <span class="hljs-number">0.5</span> + vline.vals, col = linecol, lty = <span class="hljs-number">2</span>)<br>            &#125;<br>            xv &lt;- <span class="hljs-built_in">rep</span>(i, nrow(x.scaled)) + x.scaled[, i] - <span class="hljs-number">0.5</span><br>            xv &lt;- <span class="hljs-built_in">c</span>(xv[<span class="hljs-number">1</span>], xv)<br>            yv &lt;- 1:<span class="hljs-built_in">length</span>(xv) - <span class="hljs-number">0.5</span><br>            lines(x = xv, y = yv, lwd = <span class="hljs-number">1</span>, col = tracecol, type = <span class="hljs-string">&quot;s&quot;</span>)<br>        &#125;<br>    &#125;<br><br>  <span class="hljs-comment"># row trace</span><br>    <span class="hljs-keyword">if</span> (trace %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>, <span class="hljs-string">&quot;row&quot;</span>)) &#123;<br>        retval$hline &lt;- hline<br>        hline.vals &lt;- scale01(hline, min.scale, max.scale)<br>        <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> rowInd) &#123;<br>            <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(hline)) &#123;<br>                abline(h = i + hline, col = linecol, lty = <span class="hljs-number">2</span>)<br>            &#125;<br>            yv &lt;- <span class="hljs-built_in">rep</span>(i, ncol(x.scaled)) + x.scaled[i, ] - <span class="hljs-number">0.5</span><br>            yv &lt;- rev(<span class="hljs-built_in">c</span>(yv[<span class="hljs-number">1</span>], yv))<br>            xv &lt;- <span class="hljs-built_in">length</span>(yv):<span class="hljs-number">1</span> - <span class="hljs-number">0.5</span><br>            lines(x = xv, y = yv, lwd = <span class="hljs-number">1</span>, col = tracecol, type = <span class="hljs-string">&quot;s&quot;</span>)<br>        &#125;<br>    &#125;<br><br>  <span class="hljs-comment"># add cell labels</span><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">missing</span>(cellnote))<br>        text(x = <span class="hljs-built_in">c</span>(row(cellnote)), y = <span class="hljs-built_in">c</span>(col(cellnote)), labels = <span class="hljs-built_in">c</span>(cellnote), col = notecol, cex = notecex)<br><br>  <span class="hljs-comment">###########################</span><br>  <span class="hljs-comment">## Plot the row dendrogram</span><br>  <span class="hljs-comment">###########################</span><br><br>    par(mar = <span class="hljs-built_in">c</span>(margins[<span class="hljs-number">1</span>], <span class="hljs-number">0</span>, <span class="hljs-number">0</span>, <span class="hljs-number">0</span>))<br>    <span class="hljs-keyword">if</span> (dendrogram %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>, <span class="hljs-string">&quot;row&quot;</span>)) &#123;<br>        plot(ddr, horiz = <span class="hljs-literal">TRUE</span>, axes = <span class="hljs-literal">FALSE</span>, yaxs = <span class="hljs-string">&quot;i&quot;</span>, leaflab = <span class="hljs-string">&quot;none&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> plot.new()<br><br>  <span class="hljs-comment">#############################</span><br>  <span class="hljs-comment">## Plot the column dendrogram</span><br>  <span class="hljs-comment">#############################</span><br><br>    par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-number">0</span>, <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(main)) <span class="hljs-number">5</span> <span class="hljs-keyword">else</span> <span class="hljs-number">0</span>, margins[<span class="hljs-number">2</span>]))<br>    <span class="hljs-keyword">if</span> (dendrogram %in% <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;both&quot;</span>, <span class="hljs-string">&quot;column&quot;</span>)) &#123;<br>        plot(ddc, axes = <span class="hljs-literal">FALSE</span>, xaxs = <span class="hljs-string">&quot;i&quot;</span>, leaflab = <span class="hljs-string">&quot;none&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> plot.new()<br><br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(main))<br>        title(main, cex.main=cex.main) <span class="hljs-comment">#cex.main = 1.5 * op[[&quot;cex.main&quot;]])</span><br><br><br>  <span class="hljs-comment">############################</span><br>  <span class="hljs-comment">## Add the Color Chart</span><br>  <span class="hljs-comment">############################</span><br><br>    <span class="hljs-keyword">if</span> (key) &#123;<br>        par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">5</span>, <span class="hljs-number">4</span>, <span class="hljs-number">2</span>, <span class="hljs-number">1</span>), cex = <span class="hljs-number">0.75</span>)<br>        tmpbreaks &lt;- breaks<br>        <span class="hljs-keyword">if</span> (symkey) &#123;<br>            max.raw &lt;- <span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(<span class="hljs-built_in">c</span>(x, breaks)), na.rm = <span class="hljs-literal">TRUE</span>)<br>            min.raw &lt;- -max.raw<br>            tmpbreaks[<span class="hljs-number">1</span>] &lt;- -<span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(x), na.rm = <span class="hljs-literal">TRUE</span>)<br>            tmpbreaks[<span class="hljs-built_in">length</span>(tmpbreaks)] &lt;- <span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(x), na.rm = <span class="hljs-literal">TRUE</span>)<br>        &#125;<br>        <span class="hljs-keyword">else</span> &#123;<br>            min.raw &lt;- <span class="hljs-built_in">min</span>(<span class="hljs-built_in">c</span>(x,breaks), na.rm = <span class="hljs-literal">TRUE</span>)<br>            max.raw &lt;- <span class="hljs-built_in">max</span>(<span class="hljs-built_in">c</span>(x,breaks), na.rm = <span class="hljs-literal">TRUE</span>)<br>        &#125;<br><br>        message(<span class="hljs-string">&#x27;for plotting:: min.raw: &#x27;</span>, min.raw, <span class="hljs-string">&#x27; max.raw: &#x27;</span>, max.raw);<br><br>        z &lt;- seq(min.raw, max.raw, <span class="hljs-built_in">length</span> = <span class="hljs-built_in">length</span>(col))<br>        image(z = matrix(z, ncol = <span class="hljs-number">1</span>), col = col, breaks = tmpbreaks,<br>            xaxt = <span class="hljs-string">&quot;n&quot;</span>, yaxt = <span class="hljs-string">&quot;n&quot;</span>)<br>        par(usr = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">0</span>, <span class="hljs-number">1</span>))<br>        lv &lt;- pretty(breaks)<br>        xv &lt;- scale01(<span class="hljs-built_in">as.numeric</span>(lv), min.raw, max.raw)<br>        axis(<span class="hljs-number">1</span>, at = xv, labels = lv)<br>        <span class="hljs-keyword">if</span> (scale == <span class="hljs-string">&quot;row&quot;</span>)<br>            mtext(side = <span class="hljs-number">1</span>, <span class="hljs-string">&quot;Row Z-Score&quot;</span>, line = <span class="hljs-number">2</span>)<br>        <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (scale == <span class="hljs-string">&quot;column&quot;</span>)<br>            mtext(side = <span class="hljs-number">1</span>, <span class="hljs-string">&quot;Column Z-Score&quot;</span>, line = <span class="hljs-number">2</span>)<br>        <span class="hljs-keyword">else</span> mtext(side = <span class="hljs-number">1</span>, KeyValueName, line = <span class="hljs-number">2</span>)<br>        <span class="hljs-keyword">if</span> (density.info == <span class="hljs-string">&quot;density&quot;</span>) &#123;<br>            dens &lt;- density(x, adjust = densadj, na.rm = <span class="hljs-literal">TRUE</span>)<br>            omit &lt;- dens$x &lt; <span class="hljs-built_in">min</span>(breaks) | dens$x &gt; <span class="hljs-built_in">max</span>(breaks)<br>            dens$x &lt;- dens$x[-omit]<br>            dens$y &lt;- dens$y[-omit]<br>            dens$x &lt;- scale01(dens$x, min.raw, max.raw)<br>            lines(dens$x, dens$y/<span class="hljs-built_in">max</span>(dens$y) * <span class="hljs-number">0.95</span>, col = denscol,<br>                lwd = <span class="hljs-number">1</span>)<br>            axis(<span class="hljs-number">2</span>, at = pretty(dens$y)/<span class="hljs-built_in">max</span>(dens$y) * <span class="hljs-number">0.95</span>, pretty(dens$y))<br>            title(<span class="hljs-string">&quot;Color Key\nand Density Plot&quot;</span>)<br>            par(cex = <span class="hljs-number">0.5</span>)<br>            mtext(side = <span class="hljs-number">2</span>, <span class="hljs-string">&quot;Density&quot;</span>, line = <span class="hljs-number">2</span>)<br>        &#125;<br>        <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (density.info == <span class="hljs-string">&quot;histogram&quot;</span>) &#123;<br>            h &lt;- hist(x, plot = <span class="hljs-literal">FALSE</span>, breaks = breaks)<br>            hx &lt;- scale01(breaks, min.raw, max.raw)<br>            hy &lt;- <span class="hljs-built_in">c</span>(h$counts, h$counts[<span class="hljs-built_in">length</span>(h$counts)])<br>            lines(hx, hy/<span class="hljs-built_in">max</span>(hy) * <span class="hljs-number">0.95</span>, lwd = <span class="hljs-number">1</span>, type = <span class="hljs-string">&quot;s&quot;</span>,<br>                col = denscol)<br>            axis(<span class="hljs-number">2</span>, at = pretty(hy)/<span class="hljs-built_in">max</span>(hy) * <span class="hljs-number">0.95</span>, pretty(hy))<br>            title(<span class="hljs-string">&quot;Color Key\nand Histogram&quot;</span>)<br>            par(cex = <span class="hljs-number">0.5</span>)<br>            mtext(side = <span class="hljs-number">2</span>, <span class="hljs-string">&quot;Count&quot;</span>, line = <span class="hljs-number">2</span>)<br>        &#125;<br>        <span class="hljs-keyword">else</span> title(<span class="hljs-string">&quot;Color Key&quot;</span>)<br>    &#125;<br>    <span class="hljs-keyword">else</span> plot.new()<br><br>    retval$colorTable &lt;- data.frame(low = retval$breaks[-<span class="hljs-built_in">length</span>(retval$breaks)], high = retval$breaks[-<span class="hljs-number">1</span>], color = retval$col)<br><br>    <span class="hljs-built_in">invisible</span>(retval)<br>&#125;<br><br><br><br><span class="hljs-comment">## EXAMPLE USAGE</span><br><br><span class="hljs-comment">## example of colsidecolors rowsidecolors (single column, single row)</span><br><span class="hljs-comment">##mat &lt;- matrix(1:100, byrow=T, nrow=10)</span><br><span class="hljs-comment">##column_annotation &lt;- sample(c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;), 10, replace=T)</span><br><span class="hljs-comment">##column_annotation &lt;- as.matrix(column_annotation)</span><br><span class="hljs-comment">##colnames(column_annotation) &lt;- c(&quot;Variable X&quot;)</span><br><br><span class="hljs-comment">##row_annotation &lt;- sample(c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;), 10, replace=T)</span><br><span class="hljs-comment">##row_annotation &lt;- as.matrix(t(row_annotation))</span><br><span class="hljs-comment">##rownames(row_annotation) &lt;- c(&quot;Variable Y&quot;)</span><br><br><span class="hljs-comment">##heatmap.3(mat, RowSideColors=row_annotation, ColSideColors=column_annotation)</span><br><br><span class="hljs-comment">## multiple column and row</span><br><span class="hljs-comment">##mat &lt;- matrix(1:100, byrow=T, nrow=10)</span><br><span class="hljs-comment">##column_annotation &lt;- matrix(sample(c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;), 20, replace=T), ncol=2)</span><br><span class="hljs-comment">##colnames(column_annotation) &lt;- c(&quot;Variable X1&quot;, &quot;Variable X2&quot;)</span><br><br><span class="hljs-comment">##row_annotation &lt;- matrix(sample(c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;), 20, replace=T), nrow=2)</span><br><span class="hljs-comment">##rownames(row_annotation) &lt;- c(&quot;Variable Y1&quot;, &quot;Variable Y2&quot;)</span><br><br><span class="hljs-comment">##heatmap.3(mat, RowSideColors=row_annotation, ColSideColors=column_annotation)</span><br><br><br><br><span class="hljs-comment">##---------------------------------------------------------------------------------</span><br><span class="hljs-comment">##---------------------------------------------------------------------------------</span><br><span class="hljs-comment">##---------------------------------------------------------------------------------</span><br><br><span class="hljs-comment">## Borrowing the following from gplots  (gplots isn&#x27;t compatible with R 3.0 (yet), and so bypassing it for now).</span><br><br>colorpanel = <span class="hljs-keyword">function</span> (n, low, mid, high)<br>&#123;<br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(mid) || <span class="hljs-built_in">missing</span>(high)) &#123;<br>        low &lt;- col2rgb(low)<br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(high))<br>            high &lt;- col2rgb(mid)<br>        <span class="hljs-keyword">else</span> high &lt;- col2rgb(high)<br>        red &lt;- seq(low[<span class="hljs-number">1</span>, <span class="hljs-number">1</span>], high[<span class="hljs-number">1</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = n)/<span class="hljs-number">255</span><br>        green &lt;- seq(low[<span class="hljs-number">3</span>, <span class="hljs-number">1</span>], high[<span class="hljs-number">3</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = n)/<span class="hljs-number">255</span><br>        blue &lt;- seq(low[<span class="hljs-number">2</span>, <span class="hljs-number">1</span>], high[<span class="hljs-number">2</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = n)/<span class="hljs-number">255</span><br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>        isodd &lt;- odd(n)<br>        <span class="hljs-keyword">if</span> (isodd) &#123;<br>            n &lt;- n + <span class="hljs-number">1</span><br>        &#125;<br>        low &lt;- col2rgb(low)<br>        mid &lt;- col2rgb(mid)<br>        high &lt;- col2rgb(high)<br>        lower &lt;- <span class="hljs-built_in">floor</span>(n/<span class="hljs-number">2</span>)<br>        upper &lt;- n - lower<br>        red &lt;- <span class="hljs-built_in">c</span>(seq(low[<span class="hljs-number">1</span>, <span class="hljs-number">1</span>], mid[<span class="hljs-number">1</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = lower), seq(mid[<span class="hljs-number">1</span>,<br>            <span class="hljs-number">1</span>], high[<span class="hljs-number">1</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = upper))/<span class="hljs-number">255</span><br>        green &lt;- <span class="hljs-built_in">c</span>(seq(low[<span class="hljs-number">3</span>, <span class="hljs-number">1</span>], mid[<span class="hljs-number">3</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = lower),<br>            seq(mid[<span class="hljs-number">3</span>, <span class="hljs-number">1</span>], high[<span class="hljs-number">3</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = upper))/<span class="hljs-number">255</span><br>        blue &lt;- <span class="hljs-built_in">c</span>(seq(low[<span class="hljs-number">2</span>, <span class="hljs-number">1</span>], mid[<span class="hljs-number">2</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = lower),<br>            seq(mid[<span class="hljs-number">2</span>, <span class="hljs-number">1</span>], high[<span class="hljs-number">2</span>, <span class="hljs-number">1</span>], <span class="hljs-built_in">length</span> = upper))/<span class="hljs-number">255</span><br>        <span class="hljs-keyword">if</span> (isodd) &#123;<br>            red &lt;- red[-(lower + <span class="hljs-number">1</span>)]<br>            green &lt;- green[-(lower + <span class="hljs-number">1</span>)]<br>            blue &lt;- blue[-(lower + <span class="hljs-number">1</span>)]<br>        &#125;<br>    &#125;<br>    rgb(red, blue, green)<br>&#125;<br><br><br>greenred = <span class="hljs-keyword">function</span> (n)  &#123;<br>    colorpanel(n, <span class="hljs-string">&quot;green&quot;</span>, <span class="hljs-string">&quot;black&quot;</span>, <span class="hljs-string">&quot;red&quot;</span>)<br>&#125;<br><br>odd = <span class="hljs-keyword">function</span> (x) &#123;<br>    x%%2 == <span class="hljs-number">1</span><br>&#125;<br><br>even = <span class="hljs-keyword">function</span> (x) &#123;<br>    x%%2 == <span class="hljs-number">0</span><br>&#125;<br></code></pre></td></tr></table></figure></div>
<h3 id="misc-rnaseq-funcs-R">misc_rnaseq_funcs.R</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs R"><span class="hljs-comment">## from https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Analysis/DifferentialExpression/R/misc_rnaseq_funcs.R</span><br><br>plot_counts_matrix_log2_dist = <span class="hljs-keyword">function</span>(matrix_file) &#123;<br><br><br>  data = read.table(file=matrix_file, com=<span class="hljs-string">&#x27;&#x27;</span>, row.names=<span class="hljs-number">1</span>, header=<span class="hljs-built_in">T</span>)<br><br>  conditions = colnames(data)<br>  colors = rainbow(<span class="hljs-built_in">length</span>(conditions))<br><br><br>  plot(density(log2(data[,<span class="hljs-number">1</span>])), col=colors[<span class="hljs-number">1</span>], main=matrix_file, xlab=<span class="hljs-string">&#x27;log2(frag_counts)&#x27;</span>, ylab=<span class="hljs-string">&#x27;density&#x27;</span>)<br><br>  <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">2</span>:<span class="hljs-built_in">length</span>(data[<span class="hljs-number">1</span>,])) &#123;<br><br>    points(density(log2(data[,i])), type=<span class="hljs-string">&#x27;l&#x27;</span>, col=colors[i])<br><br>  &#125;<br><br>  legend(<span class="hljs-string">&#x27;topright&#x27;</span>, conditions, col=colors, pch=<span class="hljs-number">15</span>)<br><br>&#125;<br><br><br>matrix_to_color_assignments = <span class="hljs-keyword">function</span>(matrix_m, col=<span class="hljs-literal">NULL</span>, by=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;matrix&quot;</span>, <span class="hljs-string">&quot;row&quot;</span>, <span class="hljs-string">&quot;col&quot;</span>)) &#123;<br><br>  <span class="hljs-keyword">if</span> (! <span class="hljs-built_in">is.matrix</span>(matrix_m))<br>    stop(<span class="hljs-string">&quot;Error, matrix_to_color_assignments() requires a matrix as parameter.&quot;</span>)<br>  num_colors = <span class="hljs-number">0</span><br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(col)) &#123;<br>        num_colors = <span class="hljs-built_in">min</span>(nrow(matrix_m), ncol(matrix_m))<br>        col = rainbow(num_colors)<br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>        num_colors = <span class="hljs-built_in">length</span>(col)<br>    &#125;<br><br>    by = match.arg(by)<br><br>    <span class="hljs-keyword">if</span> (by == <span class="hljs-string">&quot;matrix&quot;</span>) &#123;<br><br>        min_val = <span class="hljs-built_in">min</span>(matrix_m, na.rm=<span class="hljs-built_in">T</span>)<br>      matrix_m = matrix_m - min_val<br>      max_val = <span class="hljs-built_in">max</span>(matrix_m, na.rm=<span class="hljs-built_in">T</span>)<br>      matrix_m = matrix_m / max_val * num_colors<br>        <span class="hljs-comment">#print(matrix_m)</span><br>         matrix_m = apply(matrix_m, <span class="hljs-number">1</span>:<span class="hljs-number">2</span>, <span class="hljs-keyword">function</span>(x) ifelse (x&lt;<span class="hljs-number">1</span>, <span class="hljs-built_in">as.character</span>(col[<span class="hljs-number">1</span>]), <span class="hljs-built_in">as.character</span>(col[x])));<br><br>        matrix_m = matrix(<span class="hljs-built_in">as.character</span>(matrix_m), nrow=<span class="hljs-built_in">dim</span>(matrix_m)[<span class="hljs-number">1</span>])<br>  &#125;<br>  <span class="hljs-keyword">else</span> &#123;<br><br>    row_or_col_only_color_selector_func = <span class="hljs-keyword">function</span>(x) &#123;<br>        a = <span class="hljs-built_in">min</span>(x, na.rm=<span class="hljs-built_in">T</span>);<br>        b = <span class="hljs-built_in">max</span>(x, na.rm=<span class="hljs-built_in">T</span>);<br>        <span class="hljs-built_in">c</span> = (x-a)/(b-a) * num_colors;<br>                <span class="hljs-built_in">c</span> = <span class="hljs-built_in">round</span>(<span class="hljs-built_in">c</span>);<br>        <span class="hljs-built_in">c</span> = ifelse (<span class="hljs-built_in">c</span>&lt;<span class="hljs-number">1</span>, <span class="hljs-number">1</span>, <span class="hljs-built_in">c</span>);<br>                <span class="hljs-comment">#print(paste(c(&quot;color selection: (a)&quot;, a, &quot; (b)&quot;, b, &quot; (c)&quot;, paste(c, sep=&#x27;,&#x27;))));</span><br>                colors = <span class="hljs-built_in">as.character</span>(col[<span class="hljs-built_in">c</span>]);<br>                <span class="hljs-built_in">return</span>(colors);<br>    &#125;<br><br>    <span class="hljs-keyword">if</span> (by == <span class="hljs-string">&quot;row&quot;</span>) &#123;<br>            matrix_m = apply(matrix_m, <span class="hljs-number">1</span>, row_or_col_only_color_selector_func);<br>            print(matrix_m)<br>            print(<span class="hljs-string">&quot;dim matrix_m after apply&quot;</span>); print(<span class="hljs-built_in">dim</span>(matrix_m))<br>            matrix_m = t(matrix_m);<br>            print(<span class="hljs-string">&quot;dim matrix_m after transpose: &quot;</span>); print(<span class="hljs-built_in">dim</span>(matrix_m))<br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>      <span class="hljs-comment"># by column</span><br>      matrix_m = apply(matrix_m, <span class="hljs-number">2</span>, row_or_col_only_color_selector_func);<br>    &#125;<br>  &#125;<br><br>  <span class="hljs-comment">#print(matrix_m)</span><br>  <span class="hljs-built_in">return</span>(matrix_m)<br>&#125;<br><br>sample_matrix_to_color_assignments = <span class="hljs-keyword">function</span>(sampleAnnotationsMatrix, colors) &#123;<br><br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(colors))<br>    colors = rainbow(nrow(sampleAnnotationsMatrix))<br><br>  nsamples = nrow(sampleAnnotationsMatrix);<br><br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(colors) &lt; nrow(sampleAnnotationsMatrix))<br>    stop(<span class="hljs-string">&quot;Error, only &quot;</span>, <span class="hljs-built_in">length</span>(colors), <span class="hljs-string">&quot; colors specified, but have &quot;</span>, nsamples, <span class="hljs-string">&quot; samples&quot;</span>);<br><br>  <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nrow(sampleAnnotationsMatrix)) &#123;<br>    <span class="hljs-built_in">c</span> = colors[i]<br>    sampleAnnotationsMatrix[i,] = sapply(sampleAnnotationsMatrix[i,], <span class="hljs-keyword">function</span>(x) ifelse( x, <span class="hljs-built_in">as.character</span>(<span class="hljs-built_in">c</span>), <span class="hljs-string">&#x27;white&#x27;</span>))<br>  &#125;<br><br>  <span class="hljs-built_in">return</span>(sampleAnnotationsMatrix);<br><br>&#125;<br></code></pre></td></tr></table></figure></div>
<h3 id="pairs3-R">pairs3.R</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs R"><span class="hljs-comment">## from: https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Analysis/DifferentialExpression/R/pairs3.R</span><br><span class="hljs-comment">### snagged from: http://stackoverflow.com/questions/9680783/how-can-i-change-the-axis-position-for-pairs</span><br><br><span class="hljs-comment">## modified to include function for modifying X/Y values.</span><br><br><br>pairs3 &lt;-<br>  <span class="hljs-keyword">function</span> (x, labels, XY_convert_fun = <span class="hljs-literal">NULL</span>, CustomColorFun = <span class="hljs-literal">NULL</span>, panel = points, ..., lower.panel = panel,<br>            upper.panel = panel, diag.panel = <span class="hljs-literal">NULL</span>, text.panel = textPanel,<br>            label.pos = <span class="hljs-number">0.5</span> + has.diag/<span class="hljs-number">3</span>, cex.labels = <span class="hljs-literal">NULL</span>, font.labels = <span class="hljs-number">1</span>,<br>            row1attop = <span class="hljs-literal">TRUE</span>, gap = <span class="hljs-number">1</span>)<br>  &#123;<br>    textPanel &lt;- <span class="hljs-keyword">function</span>(x = <span class="hljs-number">0.5</span>, y = <span class="hljs-number">0.5</span>, txt, cex, font) text(x,<br>                                                                 y, txt, cex = cex, font = font)<br>    localAxis &lt;- <span class="hljs-keyword">function</span>(side, x, y, xpd, bg, col = <span class="hljs-literal">NULL</span>, main,<br>                          oma, ...) &#123;<br>      <span class="hljs-keyword">if</span> (side%%2 == <span class="hljs-number">1</span>)<br>        Axis(x, side = side, xpd = <span class="hljs-literal">NA</span>, ...)<br>      <span class="hljs-keyword">else</span> Axis(y, side = side, xpd = <span class="hljs-literal">NA</span>, ...)<br>    &#125;<br>    localPlot &lt;- <span class="hljs-keyword">function</span>(..., main, oma, font.main, cex.main) &#123;<br>        plot(...)<br>    &#125;<br>    localLowerPanel &lt;- <span class="hljs-keyword">function</span>(..., main, oma, font.main, cex.main) lower.panel(...)<br>    localUpperPanel &lt;- <span class="hljs-keyword">function</span>(..., main, oma, font.main, cex.main) upper.panel(...)<br>    localDiagPanel &lt;- <span class="hljs-keyword">function</span>(..., main, oma, font.main, cex.main) diag.panel(...)<br>    dots &lt;- <span class="hljs-built_in">list</span>(...)<br>    nmdots &lt;- <span class="hljs-built_in">names</span>(dots)<br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.matrix</span>(x)) &#123;<br>      x &lt;- as.data.frame(x)<br>      <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-built_in">seq_along</span>(<span class="hljs-built_in">names</span>(x))) &#123;<br>        <span class="hljs-keyword">if</span> (is.factor(x[[i]]) || <span class="hljs-built_in">is.logical</span>(x[[i]]))<br>          x[[i]] &lt;- <span class="hljs-built_in">as.numeric</span>(x[[i]])<br>        <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.numeric</span>(<span class="hljs-built_in">unclass</span>(x[[i]])))<br>          stop(<span class="hljs-string">&quot;non-numeric argument to &#x27;pairs&#x27;&quot;</span>)<br>      &#125;<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.numeric</span>(x))<br>      stop(<span class="hljs-string">&quot;non-numeric argument to &#x27;pairs&#x27;&quot;</span>)<br>    panel &lt;- match.fun(panel)<br>    <span class="hljs-keyword">if</span> ((has.lower &lt;- !<span class="hljs-built_in">is.null</span>(lower.panel)) &amp;&amp; !<span class="hljs-built_in">missing</span>(lower.panel))<br>      lower.panel &lt;- match.fun(lower.panel)<br>    <span class="hljs-keyword">if</span> ((has.upper &lt;- !<span class="hljs-built_in">is.null</span>(upper.panel)) &amp;&amp; !<span class="hljs-built_in">missing</span>(upper.panel))<br>      upper.panel &lt;- match.fun(upper.panel)<br>    <span class="hljs-keyword">if</span> ((has.diag &lt;- !<span class="hljs-built_in">is.null</span>(diag.panel)) &amp;&amp; !<span class="hljs-built_in">missing</span>(diag.panel))<br>      diag.panel &lt;- match.fun(diag.panel)<br>    <span class="hljs-keyword">if</span> (row1attop) &#123;<br>      tmp &lt;- lower.panel<br>      lower.panel &lt;- upper.panel<br>      upper.panel &lt;- tmp<br>      tmp &lt;- has.lower<br>      has.lower &lt;- has.upper<br>      has.upper &lt;- tmp<br>    &#125;<br>    nc &lt;- ncol(x)<br>    <span class="hljs-keyword">if</span> (nc &lt; <span class="hljs-number">2</span>)<br>      stop(<span class="hljs-string">&quot;only one column in the argument to &#x27;pairs&#x27;&quot;</span>)<br>    has.labs &lt;- <span class="hljs-literal">TRUE</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(labels)) &#123;<br>      labels &lt;- colnames(x)<br>      <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(labels))<br>        labels &lt;- paste(<span class="hljs-string">&quot;var&quot;</span>, <span class="hljs-number">1L</span>:nc)<br>    &#125;<br>    <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(labels))<br>      has.labs &lt;- <span class="hljs-literal">FALSE</span><br>    oma &lt;- <span class="hljs-keyword">if</span> (<span class="hljs-string">&quot;oma&quot;</span> %in% nmdots)<br>      dots$oma<br>    <span class="hljs-keyword">else</span> <span class="hljs-literal">NULL</span><br>    main &lt;- <span class="hljs-keyword">if</span> (<span class="hljs-string">&quot;main&quot;</span> %in% nmdots)<br>      dots$main<br>    <span class="hljs-keyword">else</span> <span class="hljs-literal">NULL</span><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(oma)) &#123;<br>      oma &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">4</span>, <span class="hljs-number">4</span>, <span class="hljs-number">4</span>, <span class="hljs-number">4</span>)<br>      <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(main))<br>        oma[<span class="hljs-number">3L</span>] &lt;- <span class="hljs-number">6</span><br>    &#125;<br>    opar &lt;- par(mfrow = <span class="hljs-built_in">c</span>(nc, nc), mar = rep.int(gap/<span class="hljs-number">2</span>, <span class="hljs-number">4</span>), oma = oma)<br>    <span class="hljs-built_in">on.exit</span>(par(opar))<br>    dev.hold()<br>    <span class="hljs-built_in">on.exit</span>(dev.flush(), add = <span class="hljs-literal">TRUE</span>)<br>    <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-keyword">if</span> (row1attop)<br>      <span class="hljs-number">1L</span>:nc<br>         <span class="hljs-keyword">else</span> nc:<span class="hljs-number">1L</span>) <span class="hljs-keyword">for</span> (j <span class="hljs-keyword">in</span> <span class="hljs-number">1L</span>:nc) &#123;<br><br>      <span class="hljs-comment">#print(paste(i,&quot;vs&quot;, j));</span><br>      xvals = x[,j]<br>      yvals = x[,i]      <br><br>      <span class="hljs-keyword">if</span> ( (i != j) &amp;&amp; ! <span class="hljs-built_in">is.null</span>(XY_convert_fun)) &#123;<br><br>        res = XY_convert_fun(xvals, yvals);<br>        xvals = res[[<span class="hljs-string">&#x27;x&#x27;</span>]]<br>        yvals = res[[<span class="hljs-string">&#x27;y&#x27;</span>]]<br>      &#125;<br><br>            col=<span class="hljs-string">&#x27;black&#x27;</span> <span class="hljs-comment"># default</span><br>            <span class="hljs-keyword">if</span> (! <span class="hljs-built_in">is.null</span>(CustomColorFun))<br>                col=CustomColorFun(xvals,yvals)<br><br>       <span class="hljs-comment">#     get_list_from_ellipsis2 &lt;- function(...) as.list(substitute(list(...)))[-1L]</span><br>       <span class="hljs-comment"># print(get_list_from_ellipsis2(...));stop();</span><br><br>           localPlot(xvals, yvals, xlab = <span class="hljs-string">&quot;&quot;</span>, ylab = <span class="hljs-string">&quot;&quot;</span>,axes = <span class="hljs-built_in">F</span>, <span class="hljs-comment">#col=ifelse(abs(xvals-yvals)&gt;2, &#x27;red&#x27;, &#x27;black&#x27;),</span><br>                     type = <span class="hljs-string">&quot;n&quot;</span>, ...)<br>           <span class="hljs-keyword">if</span> (i == j || (i &lt; j &amp;&amp; has.lower) || (i &gt; j &amp;&amp; has.upper)) &#123;<br>             box()<br>             <span class="hljs-comment"># edited here...</span><br>             <span class="hljs-comment">#           if (i == 1 &amp;&amp; (!(j%%2) || !has.upper || !has.lower))</span><br>             <span class="hljs-comment">#           localAxis(1 + 2 * row1attop, x[, j], x[, i],</span><br>             <span class="hljs-comment">#                       ...)</span><br>             <span class="hljs-comment"># draw x-axis</span><br>             <span class="hljs-keyword">if</span> (i != j) <span class="hljs-comment">#(i == nc &amp; j != nc)</span><br>               localAxis(<span class="hljs-number">1</span>, xvals, yvals,<br>                         ...)<br>             <span class="hljs-comment"># draw y-axis</span><br>             <span class="hljs-keyword">if</span> (j != i) <span class="hljs-comment">#(j == 1 &amp; i != 1)</span><br>               localAxis(<span class="hljs-number">2</span>, xvals, yvals, ...)<br>             <span class="hljs-comment">#           if (j == nc &amp;&amp; (i%%2 || !has.upper || !has.lower))</span><br>             <span class="hljs-comment">#             localAxis(4, x[, j], x[, i], ...)</span><br>             mfg &lt;- par(<span class="hljs-string">&quot;mfg&quot;</span>)<br>             <span class="hljs-keyword">if</span> (i == j) &#123;<br>               <span class="hljs-keyword">if</span> (has.diag)<br>                 localDiagPanel(as.vector(yvals), ...)<br>               <span class="hljs-keyword">if</span> (has.labs) &#123;<br>                 par(usr = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">0</span>, <span class="hljs-number">1</span>))<br>                 <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(cex.labels)) &#123;<br>                   l.wid &lt;- strwidth(labels, <span class="hljs-string">&quot;user&quot;</span>)<br>                   cex.labels &lt;- <span class="hljs-built_in">max</span>(<span class="hljs-number">0.8</span>, <span class="hljs-built_in">min</span>(<span class="hljs-number">2</span>, <span class="hljs-number">0.9</span>/<span class="hljs-built_in">max</span>(l.wid)))<br>                 &#125;<br>                 text.panel(<span class="hljs-number">0.5</span>, label.pos, labels[i], cex = cex.labels,<br>                            font = font.labels)<br>               &#125;<br>             &#125;<br>             <span class="hljs-keyword">else</span> <span class="hljs-keyword">if</span> (i &lt; j)<br>               localLowerPanel(as.vector(xvals), as.vector(yvals), col=col, ...)<br>             <span class="hljs-keyword">else</span> localUpperPanel(as.vector(xvals), as.vector(yvals), col=col, ...)<br>             <span class="hljs-keyword">if</span> (<span class="hljs-built_in">any</span>(par(<span class="hljs-string">&quot;mfg&quot;</span>) != mfg))<br>               stop(<span class="hljs-string">&quot;the &#x27;panel&#x27; function made a new plot&quot;</span>)<br>           &#125;<br>           <span class="hljs-keyword">else</span> par(new = <span class="hljs-literal">FALSE</span>)<br>         &#125;<br>    <span class="hljs-keyword">if</span> (!<span class="hljs-built_in">is.null</span>(main)) &#123;<br>      font.main &lt;- <span class="hljs-keyword">if</span> (<span class="hljs-string">&quot;font.main&quot;</span> %in% nmdots)<br>        dots$font.main<br>      <span class="hljs-keyword">else</span> par(<span class="hljs-string">&quot;font.main&quot;</span>)<br>      cex.main &lt;- <span class="hljs-keyword">if</span> (<span class="hljs-string">&quot;cex.main&quot;</span> %in% nmdots)<br>        dots$cex.main<br>      <span class="hljs-keyword">else</span> par(<span class="hljs-string">&quot;cex.main&quot;</span>)<br>      mtext(main, <span class="hljs-number">3</span>, <span class="hljs-number">3</span>, <span class="hljs-literal">TRUE</span>, <span class="hljs-number">0.5</span>, cex = cex.main, font = font.main)<br>    &#125;<br>    <span class="hljs-built_in">invisible</span>(<span class="hljs-literal">NULL</span>)<br>  &#125;<br><br><span class="hljs-comment">### Demo it:</span><br>demo_pairs3 = <span class="hljs-keyword">function</span> (custom_fun=<span class="hljs-built_in">F</span>) &#123;<br>    data(iris)<br><br>  xy_conv = <span class="hljs-keyword">function</span>(x,y) &#123;<br><br>    res = <span class="hljs-built_in">list</span>();<br>    res[[<span class="hljs-string">&#x27;x&#x27;</span>]] = (x+y)/<span class="hljs-number">2</span>;<br>    res[[<span class="hljs-string">&#x27;y&#x27;</span>]] = x - y;<br><br>    <span class="hljs-built_in">return</span>(res);<br>  &#125;<br><br>  <span class="hljs-keyword">if</span> (custom_fun) &#123;<br>    pairs3(iris[<span class="hljs-number">1</span>:<span class="hljs-number">4</span>], XY_convert_fun = xy_conv, main = <span class="hljs-string">&quot;Anderson&#x27;s Iris Data -- 3 species&quot;</span>,pch = <span class="hljs-number">21</span>, bg = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;red&quot;</span>, <span class="hljs-string">&quot;green3&quot;</span>, <span class="hljs-string">&quot;blue&quot;</span>)[<span class="hljs-built_in">unclass</span>(iris$Species)])<br>  &#125;<br>  <span class="hljs-keyword">else</span> &#123;<br>    pairs3(iris[<span class="hljs-number">1</span>:<span class="hljs-number">4</span>], main = <span class="hljs-string">&quot;Anderson&#x27;s Iris Data -- 3 species&quot;</span>,pch = <span class="hljs-number">21</span>, bg = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;red&quot;</span>, <span class="hljs-string">&quot;green3&quot;</span>, <span class="hljs-string">&quot;blue&quot;</span>)[<span class="hljs-built_in">unclass</span>(iris$Species)])<br>  &#125;<br>&#125;<br></code></pre></td></tr></table></figure></div>
<h3 id="vioplot2-R">vioplot2.R</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs R"><span class="hljs-comment">## from: https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Analysis/DifferentialExpression/R/vioplot2.R</span><br><br><span class="hljs-comment">###################################################################################</span><br><span class="hljs-comment">### From:  https://stackoverflow.com/questions/22410606/violin-plot-with-list-input</span><br><span class="hljs-comment">### and slightly modified to my liking here.</span><br><br>vioplot2&lt;-<span class="hljs-keyword">function</span> (x, ..., <span class="hljs-built_in">range</span> = <span class="hljs-number">1.5</span>, h = <span class="hljs-literal">NULL</span>, ylim = <span class="hljs-literal">NULL</span>, <span class="hljs-built_in">names</span> = <span class="hljs-literal">NULL</span>,<br>                    horizontal = <span class="hljs-literal">FALSE</span>, col = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;magenta&quot;</span>), border = <span class="hljs-string">&quot;black&quot;</span>, lty = <span class="hljs-number">1</span>,<br>                    lwd = <span class="hljs-number">1</span>, rectCol = <span class="hljs-string">&quot;black&quot;</span>, colMed = <span class="hljs-string">&quot;white&quot;</span>, pchMed = <span class="hljs-number">19</span>,<br>                    at, add = <span class="hljs-literal">FALSE</span>, wex = <span class="hljs-number">1</span>, drawRect = <span class="hljs-literal">TRUE</span>)<br>&#123;<br><br>    library(sm)<br><br><br>    <span class="hljs-keyword">if</span>(!<span class="hljs-built_in">is.list</span>(x))&#123;<br>        datas &lt;- <span class="hljs-built_in">list</span>(x, ...)<br>    &#125; <span class="hljs-keyword">else</span>&#123;<br>        datas&lt;-x<br>    &#125;<br>    n &lt;- <span class="hljs-built_in">length</span>(datas)<br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">missing</span>(at))<br>        at &lt;- 1:n<br>    upper &lt;- vector(mode = <span class="hljs-string">&quot;numeric&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    lower &lt;- vector(mode = <span class="hljs-string">&quot;numeric&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    q1 &lt;- vector(mode = <span class="hljs-string">&quot;numeric&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    q3 &lt;- vector(mode = <span class="hljs-string">&quot;numeric&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    med &lt;- vector(mode = <span class="hljs-string">&quot;numeric&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    base &lt;- vector(mode = <span class="hljs-string">&quot;list&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    height &lt;- vector(mode = <span class="hljs-string">&quot;list&quot;</span>, <span class="hljs-built_in">length</span> = n)<br>    baserange &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-literal">Inf</span>, -<span class="hljs-literal">Inf</span>)<br>    args &lt;- <span class="hljs-built_in">list</span>(display = <span class="hljs-string">&quot;none&quot;</span>)<br>    <span class="hljs-keyword">if</span> (!(<span class="hljs-built_in">is.null</span>(h)))<br>        args &lt;- <span class="hljs-built_in">c</span>(args, h = h)<br>    <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:n) &#123;<br>        data &lt;- datas[[i]]<br>        data.min &lt;- <span class="hljs-built_in">min</span>(data)<br>        data.max &lt;- <span class="hljs-built_in">max</span>(data)<br>        q1[i] &lt;- quantile(data, <span class="hljs-number">0.25</span>)<br>        q3[i] &lt;- quantile(data, <span class="hljs-number">0.75</span>)<br>        med[i] &lt;- median(data)<br>        iqd &lt;- q3[i] - q1[i]<br>        upper[i] &lt;- <span class="hljs-built_in">min</span>(q3[i] + <span class="hljs-built_in">range</span> * iqd, data.max)<br>        lower[i] &lt;- <span class="hljs-built_in">max</span>(q1[i] - <span class="hljs-built_in">range</span> * iqd, data.min)<br>        est.xlim &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-built_in">min</span>(lower[i], data.min), <span class="hljs-built_in">max</span>(upper[i],<br>                                                   data.max))<br>        smout &lt;- do.call(<span class="hljs-string">&quot;sm.density&quot;</span>, <span class="hljs-built_in">c</span>(<span class="hljs-built_in">list</span>(data, xlim = est.xlim),<br>                                         args))<br>        hscale &lt;- 0.4/<span class="hljs-built_in">max</span>(smout$estimate) * wex<br>        base[[i]] &lt;- smout$eval.points<br>        height[[i]] &lt;- smout$estimate * hscale<br>        t &lt;- <span class="hljs-built_in">range</span>(base[[i]])<br>        baserange[<span class="hljs-number">1</span>] &lt;- <span class="hljs-built_in">min</span>(baserange[<span class="hljs-number">1</span>], t[<span class="hljs-number">1</span>])<br>        baserange[<span class="hljs-number">2</span>] &lt;- <span class="hljs-built_in">max</span>(baserange[<span class="hljs-number">2</span>], t[<span class="hljs-number">2</span>])<br>    &#125;<br>    <span class="hljs-keyword">if</span> (!add) &#123;<br>        xlim &lt;- <span class="hljs-keyword">if</span> (n == <span class="hljs-number">1</span>)<br>                    at + <span class="hljs-built_in">c</span>(-<span class="hljs-number">0.5</span>, <span class="hljs-number">0.5</span>)<br>                <span class="hljs-keyword">else</span> <span class="hljs-built_in">range</span>(at) + <span class="hljs-built_in">min</span>(diff(at))/<span class="hljs-number">2</span> * <span class="hljs-built_in">c</span>(-<span class="hljs-number">1</span>, <span class="hljs-number">1</span>)<br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(ylim)) &#123;<br>            ylim &lt;- baserange<br>        &#125;<br>    &#125;<br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(<span class="hljs-built_in">names</span>)) &#123;<br>        label &lt;- 1:n<br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>        label &lt;- <span class="hljs-built_in">names</span><br>    &#125;<br>    boxwidth &lt;- 0.05 * wex<br>    <span class="hljs-keyword">if</span> (!add)<br>        plot.new()<br>    <span class="hljs-keyword">if</span> (!horizontal) &#123;<br>        <span class="hljs-keyword">if</span> (!add) &#123;<br>            plot.window(xlim = xlim, ylim = ylim)<br>            axis(<span class="hljs-number">2</span>)<br>            axis(<span class="hljs-number">1</span>, at = at, label = label)<br>        &#125;<br>        box()<br>        <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:n) &#123;<br>            polygon(<span class="hljs-built_in">c</span>(at[i] - height[[i]], rev(at[i] + height[[i]])),<br>                    <span class="hljs-built_in">c</span>(base[[i]], rev(base[[i]])), col = col[i], border = border,<br>                    lty = lty, lwd = lwd)<br>            <span class="hljs-keyword">if</span> (drawRect) &#123;<br>                lines(at[<span class="hljs-built_in">c</span>(i, i)], <span class="hljs-built_in">c</span>(lower[i], upper[i]), lwd = lwd,<br>                      lty = lty)<br>                rect(at[i] - boxwidth/<span class="hljs-number">2</span>, q1[i], at[i] + boxwidth/<span class="hljs-number">2</span>,<br>                     q3[i], col = rectCol)<br>                points(at[i], med[i], pch = pchMed, col = colMed)<br>            &#125;<br>        &#125;<br>    &#125;<br>    <span class="hljs-keyword">else</span> &#123;<br>        <span class="hljs-keyword">if</span> (!add) &#123;<br>            plot.window(xlim = ylim, ylim = xlim)<br>            axis(<span class="hljs-number">1</span>)<br>            axis(<span class="hljs-number">2</span>, at = at, label = label)<br>        &#125;<br>        box()<br>        <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:n) &#123;<br>            polygon(<span class="hljs-built_in">c</span>(base[[i]], rev(base[[i]])), <span class="hljs-built_in">c</span>(at[i] - height[[i]],<br>                                                    rev(at[i] + height[[i]])), col = col[i], border = border,<br>                    lty = lty, lwd = lwd)<br>            <span class="hljs-keyword">if</span> (drawRect) &#123;<br>                lines(<span class="hljs-built_in">c</span>(lower[i], upper[i]), at[<span class="hljs-built_in">c</span>(i, i)], lwd = lwd,<br>                      lty = lty)<br>                rect(q1[i], at[i] - boxwidth/<span class="hljs-number">2</span>, q3[i], at[i] +<br>                                                       boxwidth/<span class="hljs-number">2</span>, col = rectCol)<br>                points(med[i], at[i], pch = pchMed, col = colMed)<br>            &#125;<br>        &#125;<br>    &#125;<br>    <span class="hljs-built_in">invisible</span>(<span class="hljs-built_in">list</span>(upper = upper, lower = lower, median = med,<br>                   q1 = q1, q3 = q3))<br>&#125;<br></code></pre></td></tr></table></figure></div>
</div><div class="article-licensing box"><div class="licensing-title"><p>Edger</p><p><a href="https://karobben.github.io/2019/08/13/R/edgeR/">https://karobben.github.io/2019/08/13/R/edgeR/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2019-08-13</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2024-01-11</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/Bioinformatics/">Bioinformatics</a><a class="link-muted mr-2" rel="tag" href="/tags/R/">R</a><a class="link-muted mr-2" rel="tag" href="/tags/RNA-Seq/">RNA-Seq</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2020/01/22/R/XGBoost/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">XGBoost (out of date)</span></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">534</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">151</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">157</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#Edger"><span class="level-left"><span class="level-item">1</span><span class="level-item">Edger</span></span></a></li><li><a class="level is-mobile" href="#Example-form-Trinity"><span class="level-left"><span class="level-item">2</span><span class="level-item">Example form Trinity</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Expression-Differential"><span class="level-left"><span class="level-item">2.1</span><span class="level-item">Expression Differential</span></span></a></li><li><a class="level is-mobile" href="#Test"><span class="level-left"><span class="level-item">2.2</span><span class="level-item">Test</span></span></a></li></ul></li><li><a class="level is-mobile" href="#edgeR-out-of-date"><span class="level-left"><span class="level-item">3</span><span class="level-item">edgeR(out of date)</span></span></a></li><li><a class="level is-mobile" href="#Attachment"><span class="level-left"><span class="level-item">4</span><span class="level-item">Attachment</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#heatmap-3-R"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">heatmap.3.R</span></span></a></li><li><a class="level is-mobile" href="#misc-rnaseq-funcs-R"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">misc_rnaseq_funcs.R</span></span></a></li><li><a class="level is-mobile" href="#pairs3-R"><span class="level-left"><span class="level-item">4.3</span><span class="level-item">pairs3.R</span></span></a></li><li><a class="level is-mobile" href="#vioplot2-R"><span class="level-left"><span class="level-item">4.4</span><span class="level-item">vioplot2.R</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/AI/"><span class="tag">AI</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Algorithm/"><span class="tag">Algorithm</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Antibody/"><span class="tag">Antibody</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">79</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">44</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/CLI-Tools/"><span class="tag">CLI Tools</span><span class="tag">15</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data-Science/"><span class="tag">Data Science</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fasta/"><span class="tag">Fasta</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immunology/"><span class="tag">Immunology</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">27</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">33</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">63</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein/"><span class="tag">Protein</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein-Structure/"><span class="tag">Protein Structure</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyMol/"><span class="tag">PyMol</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">131</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python-Plot/"><span class="tag">Python Plot</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python-Seaborn/"><span class="tag">Python Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Pyvista/"><span class="tag">Pyvista</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">86</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripting/"><span class="tag">Scripting</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sequencing/"><span class="tag">Sequencing</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">48</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/VCF/"><span class="tag">VCF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGS/"><span class="tag">WGS</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/git/"><span class="tag">git</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://imgur.com/cm7mDjj.png" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">83</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">77</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protein-Structure/"><span class="level-start"><span class="level-item">Protein Structure</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">29</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fasta-q/"><span class="level-start"><span class="level-item">Fasta/q</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/WGS/"><span class="level-start"><span class="level-item">WGS</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">41</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">11</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">4</span></span></a><ul><li><a class="level is-mobile" href="/categories/Machine-Learning/LM/"><span class="level-start"><span class="level-item">LM</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Machine-Learning/LM/Protein/"><span class="level-start"><span class="level-item">Protein</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Machine-Learning/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Machine-Learning/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">145</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">15</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Immunity/"><span class="level-start"><span class="level-item">Immunity</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/AI/"><span class="level-start"><span class="level-item">AI</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">12</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/other/"><span class="level-start"><span class="level-item">other</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immunology/"><span class="level-start"><span class="level-item">Immunology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">56</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">39</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2024 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>