---
toc: true
url: tulane_biochem_14
covercopy: <a href="http://www.bx.psu.edu/~ross/workmg/Struc_Nucleic_Acids_Chpt2_files/">© psu.edu</a>
priority: 10000
date: 2021-11-26 08:39:32
title: "Chromosome Structure"
ytitle: "染色体结构|生化 14| 杜兰"
description: "Class notes of Graduate Biochemistry in Tulane"
excerpt: "Let's explore the function of proteins"
tags: [Classes, Biochemistry, Tulane Classes]
category: [Notes, Class, Tulane, Biochemistry]
cover: "http://www.bx.psu.edu/~ross/workmg/Struc_Nucleic_Acids_Chpt2_files/image112.png"
thumbnail: "https://static-s.aa-cdn.net/img/ios/1535372364/8bea5e7d0cecf8c51488133d2c6a4f48?v=1"
---

### DNA fiber and its packaging

Introduction:
E.coli, is 4.6 million base pairs (approximately 1.1 mm, if cut and stretched out (0.5 uM in width and 2 uM in length)
Each human cells contains several meters of DNA (3.2 x 10 9 nucleosides ) if stretched end-to-end, the **nucleus** of a human cell, which contains the DNA, is only about 6 μm in diameter. This is geometrically equivalent to packing 40 km (24 miles) of extremely fine thread into a tennis ball!


==In bacteria, DNA gyrase aids in DNA packaging by causing an accumulation of negatively supercoiled (underwound) DNA.==
Some proteins are known to be involved in the supercoiling; other proteins and enzymes such as ==DNA gyrase== (Topoisomerase)help in maintaining the supercoiled structure.

|![Prokaryotic Chromosome](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnrg3375/MediaObjects/41576_2013_Article_BFnrg3375_Fig1_HTML.jpg?as=webp)|
|:-:|
|[© Xindan Wang, 2013](https://www.researchgate.net/publication/224856333_Structure_of_Metaphase_Chromosomes_A_Role_for_Effects_of_Macromolecular_Crowding)|

### Supercoiled DNA

#### Winding

|![Supercoiled plasma](http://www.bx.psu.edu/~ross/workmg/Struc_Nucleic_Acids_Chpt2_files/image112.png)|
|:-:|
|[© psu.edu](http://www.bx.psu.edu/~ross/workmg/Struc_Nucleic_Acids_Chpt2_files/)|

Supercoiled DNA can acquire different conformations or shapes (topologies). Excessively unwound DNA molecules exist as topological isomers with negatively supercoiled (plectonemic or solenoidal) forms.

#### Unwinding

|![DNA unwiding](https://z3.ax1x.com/2021/11/27/oeQpGt.png)|
|:-:|
|Unwinding a DNA molecule without allowing it to rotate creates supercoils|


### ==topoisomerase I==

Type I topos cleave single strand through use of covalent Tyr-DNA intermediate
- Type IA: relax negative supercoil; Tyr-5’-P-DNA (free 3’-OH)
- Type IB: relax pos. or neg. supercoil; Tyr-3’-DNA (free 5’-OH)

|![Topoisomerase I](https://cdn.rcsb.org/images/structures/ec/1ecl/1ecl_assembly-1.jpeg)|![Topoisomerase II](https://cdn.rcsb.org/images/structures/fo/3fof/3fof_assembly-1.jpeg)|
|:-:|:-:|
|[© PDB, Topoisomerase I](https://www.rcsb.org/structure/1ecl)|[© PDB, Topoisomerase II](https://www.rcsb.org/structure/3FOF)|
|==Type I topos cleave single strand through use of covalent Tyr-DNA intermediate== <li>Type IA: relax negative supercoil; Tyr-5’-P-DNA (free 3’-OH)<li>Type IB: relax pos. or neg. supercoil; Tyr-3’-DNA (free 5’-OH)|==All Type II topos can relax pos. and neg. supercoils and cleave double strand==: DNA gyrase (prokaryotic) is the only enzyme that can introduce neg. supercoils|

|![Topoisomerase II](https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-018-05005-x/MediaObjects/41467_2018_5005_Fig1_HTML.jpg?as=webp)|
|:-:|
|[© Ivan Laponogov, 2018](https://www.nature.com/articles/s41467-018-05005-x)|

### Topoisomerase Inhibitors

Topoisomerase Inhibitors are Useful Anti-biotic & -cancer Therapeutics

|![Topoisomerase Inhibitors](https://z3.ax1x.com/2021/11/27/oe11UJ.png)|
|:-:|
|Unknow Source|

### Eukaryotic Chromosome

|![Eukaryotic Chromosome](https://www.researchgate.net/publication/224856333/figure/fig6/AS:667174356140032@1536078170039/Images-by-transmission-electron-microscopy-of-chromosomes-released-in-12-PEG-Sections.ppm)|
|:-:|
|[© Ronald Hancock, 2012](https://www.researchgate.net/publication/224856333_Structure_of_Metaphase_Chromosomes_A_Role_for_Effects_of_Macromolecular_Crowding)|


| ![DNA Compase](https://s3-us-west-2.amazonaws.com/courses-images/wp-content/uploads/sites/110/2016/05/03204339/Figure_10_01_03.jpg) | ![Chromosome Structure](https://microbenotes.com/wp-content/uploads/2018/09/Chromosomes-Structure.png)     |
| :------------- | :------------- |
| [© lumenlearning](https://courses.lumenlearning.com/wm-biology1/chapter/reading-chromosome-structure/)       | [© microbenotes, 2021](https://microbenotes.com/chromosome-structure-types-and-functions/)       |

<iframe width="600" height="335" src="https://www.youtube.com/embed/gbSIBhFwQ4s" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

==Compositions==:
- DNA
- Histones: H1, H2A, H2B, H3 and H4
- Topoisomerase II

### Histon Genes
|![Histon Genes located on human chromosome](https://www.researchgate.net/profile/Erik-Ferlanti/publication/13440973/figure/fig1/AS:341498395676672@1458430970625/Chromosomal-localization-of-the-human-histone-gene-complement-Map-loci-were-collected.png)|
|:-:|
|[© Izabela Makalowska, 1999](https://www.researchgate.net/publication/13440973_Histone_Sequence_Database_sequences_structures_post-translational_modifications_and_genetic_loci)|

Histones Genes:
- No introns
- Multigene compound clusters
- Genes are duplicated (H2A, H2b, H3, H4)
- Highly conserved evolutionarily (except H1)
- Histone variants (CENP-A and H2AZ, etc.)

Histone Core and the Nucleosome

Nucleosome:
1.  Nucleosome Histone Octamer Core:
		H2A, H2B, H3, H4
2. DNA:Protein complex:			8 histone core proteins		~146 bp of wrapped DNA (twice)	spacer region (~90 bp)
3. Forms 11 nm nucleosome  chromatin fiber “beads on a string” (~6-7X compaction of DNA).
4. Stabilized through electrostatic interactions:				DNA phosphate (-) charge	Histones (+) charge (Arg, Lys)


==DNA wrapped around histones==

|[© Anthony T. Annunziato, 2008](https://www.nature.com/scitable/topicpage/dna-packaging-nucleosomes-and-chromatin-310/)|
|:-:|
|![Chromosomes are composed of DNA ](https://www.nature.com/scitable/content/ne0000/ne0000/ne0000/ne0000/113158606/18847_6.jpg)|
|![Nucleosome core particle](https://www.nature.com/scitable/content/7826/10.1038_38444-f1a_large_2.jpg)|

| ![Electron micrograph of chromatin](https://www.nature.com/scitable/content/30318/10.1038_nrm1225-f3a_mid.jpg) | ![Electron micrograph of chromatin](https://www.nature.com/scitable/content/30340/10.1038_nrm1225-f3c_full.jpg)   |![](https://z3.ax1x.com/2021/11/27/oeGqMt.png)|
| :-: | :-: | :-:|


### Chromatin Structure

30 nm Chromatin Fiber
1. Histone 1 (H1) bind DNA as it wraps nucleosome core.
2. Six nucleosomes per helical turn.
3. Nucleosomes stacked on top of one another in a zigzag forming (~100X compaction).
4.  Inactive chromatin in 30 nm fiber (or higher order).

- Higher Order Structure
1. 30 nm fiber begins to loop.
2. Forms a rosette arrangement constructed upon nuclear scaffold proteins.
3. A coil forms of repeated rosettes.
4. Each chromatid consists of numerous coils.

However, more evidence suggests a lack of the existence of 30-nm chromatin.
1. Fussner, E. et al. Open and closed domains in the mouse genome are configured as 10-nm chromatin fibres. EMBO Rep. 13, 992–996 (2012).
2. Nishino, Y. et al. Human mitotic chromosomes consist predominantly of irregularly folded nucleosome fibres without a 30-nm chromatin structure. EMBO J. 31, 1644–1653 (2012).
3. Hansen, J. C. et al. The 10-nm chromatin fiber and its relationship to interphase chromosome organization. Biochem. Soc. Trans. 46, 67–76 (2018).

Instead, 11-nm fibers form DNA-loops
|![DNA-loops](https://www.seas.harvard.edu/sites/default/files/images/news/fig-6f_650web.jpg)|
|:-:|
|[© harvard.edu](https://www.seas.harvard.edu/news/2014/12/scientists-map-human-loop-ome)|

### ==Interphase Chromatin==

|![Interphase chromatin dynamics ](https://www.pnas.org/content/pnas/115/45/11442/F1.large.jpg?width=800&height=600&carousel=1)|
|:-:|
|[© David Saintillan, 2018](https://www.pnas.org/content/115/45/11442)|

1. Euchromatin (lightly staining)
     - 11 nm active chromatin.
	- Gene expression "on".
	- DNA replication of S phase.

2. Heterochromatin (darkly staining)
    - Condensed, inactive chromatin.
    - Gene expression "off"

### ==Q&A==

Q: How to read genetic information from highly packed chromatin?
A: Basic principle: Loosen packed chromatin
1. Post-translational modification on histone tails (enzymes)
2. Chromatin remodeling (chromatin remodeling complex)

### ==Histone Tails Modifications==

|![Histone Tails Modifications](https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/attachment/0f65dc0b-0f94-448a-ad55-c74ac24138e5/gr1.jpg)|
|:-:|
|[© Jochen Erler, 2014](https://www.cell.com/biophysj/biophysj/supplemental/S0006-3495(14)01156-4)|
==Occur on exposed histone tails (dashed lines).==

==3 Types of Modifications :==
	==Acetylation:  decondensation.==
	==Methylation:  condensation → prevents acetylation.==  
	==Phosphorylation:  decondensation== → creates (-) charge.  			Oddly H3^Ser10^ phos. → condensation.

**Acetylation**:  Neutralizes histone (+) charge and electrostatic attraction to DNA.
	Opens chromatin → beads on string.
	==HATs== – Histone acetyltransferases on lysines e-			amino groups (H3^lys9^).

**Deacetylation**:  Maintains (+) charge and electrostatic attraction to DNA.
	Closes chromatin – 30 nm fiber
	==HDACs== – Histone deacetylases
	Constituitively assoc. with silent genes
	Chromatin less sensitive to DNAse.  

### Regulation: Chromatin Unwinding/Winding

|![](https://z3.ax1x.com/2021/11/28/onpbkD.png)|
|:-:|
|Unknow Source|

==Activators (SWI/SNF) and repressors (H1) promote unwinding or winding.==

1. SWI/SNF complex binds enhancers and begin unwinding.
2. Attracts HATs to acetylate histone tails.
3. Further action continues to open chromatin.
4. Equilibrium balance between HATs and HDACs to maintain unwinding/winding.

### Centromere and Telomere

Telomere: TTAGGG Minisatellite
Centromere: Various satellite components
Hypervariable: ministatellites
Structures see previous picture

- 0.1-20 kb of 6-64 bp repeated units.
- Several MB in length of tandemly repeated 5-170 bp sequences.
- <100 bp repeats dispersed in chromosomes.
- Also Megasatellite: 100s of 3-5 kb repeats at different locations of some chromosomes.

### Genome-Wide Repeat Sequences: Transposons

|![Compositions of Chromosome](https://z3.ax1x.com/2021/11/28/on9mn0.png)|
|:-:|
|Lehninger, Fig 24-8|

1. Retrotransposons:  replicative (or copy) transposition. LINEs, SINEs and LTRs.
2.	DNA transposons: Conservative transposition.  Cut and paste mechanism.
Autonomous vs. nonautonomous transposition.


### ==Telomeres and The End Replication Problem==

- Problem? Lagging strand end will shorten by ~ 1 primer length every genome duplication
- Solution? Telomerase!

==Ribonucleoprotein Complex That Catalyzes 3’ Telomere Extension==

|![](https://z3.ax1x.com/2021/11/28/on9I3j.png)|
|:-:|
|Source Unknow|

==Telomere functions, cancer and aging==
1. The primary role of telomeres is to protect chromosome ends from recombination, fusion, and from being recognized as damaged DNA. 
2. Maintaining Telomere length by telomerase is crucial for the survival of cancer cells in the vast majority of tumors. (Highly expressing telomerase can immortalize normal cells)
3. Telomere length shortens with age. Progressive shortening of telomeres leads to senescence and apoptosis, affecting the health and lifespan of an individual.

### Centromeres across species

|![](https://z3.ax1x.com/2021/11/28/on9XUU.png)|
|:-:|
|![](https://z3.ax1x.com/2021/11/28/onCS29.png)|
|Source Unknow|

### ==Functions of centromeres and kinetochores during mitosis: chromosome segregation==

|![](https://z3.ax1x.com/2021/11/28/oniSn1.png)|
|:-:|

### Genomic Instability and Cancer

|![](https://z3.ax1x.com/2021/11/28/oniitO.png)|
|:-:|
|© myeloma.uams.edu|
