---
toc: true
url: tulane_cellbio_1
covercopy: <a href="https://www.researchgate.net/figure/Electrophoresis-1-volt-5-cm-of-ethidium-bromide-stained-agarose-gel-2-of-PCR_fig3_346328038">© Dhifar Raa’d Al-Guranie</a>
priority: 10000
date: 2021-08-24 13:24:57
title: "1 Methods in Biology|Advanced Cell Biology|Tulane"
ytitle: "1 实验方法|高等细胞生物学|杜兰"
description: "Classes notes of the Advanced Cell Biology"
excerpt: "Classes 1, Reach Methods "
tags: [Classes, Cell Biology, Tulane Classes]
category: [Notes, Class, Tulane, Cell Biology]
cover: "https://www.researchgate.net/profile/Sareaa-Al-Mayahie/publication/346328038/figure/fig3/AS:967122800898053@1607591452404/Electrophoresis-1-volt-5-cm-of-ethidium-bromide-stained-agarose-gel-2-of-PCR.png"
thumbnail: "https://www.researchgate.net/profile/Callixte-Yadufashije/publication/324543378/figure/fig26/AS:616049695285249@1523889101090/Onion-cell-before-staining.png"
---

|Co-Author: Haoyang liang|
|:-:|


## Cell Biology Methods
Comment Methods:
- Immunohistochemistry-immunofluorescence
- Western blotting
- Immunoprecipitation
- Southern blotting
- Northern blotting
- in suit hybridization
- allele-specific oligonucleotides
- DNA microarray
- Polymerase chain reaction (PCR)
- DNA sequencing

## Immunohistochemistry-immunofluorescence

1. Fixation:(antigen denaturation)
    - Formalin: kill and cell fixation
    - Paraffin: paraffin-embedded tissue section
2. Antigen retrieval
    - [Antigen retrieval is an effective method of unmasking antigenic epitopes on the surface of formalin-fixed paraffin-embedded (FFPE) tissue sections. The antigen retrieval technique breaks the methylene bridges between **epitopes** and unrelated proteins to ==expose antigenic sites for antibody binding== (1)](https://biogenex.com/antigen-retrieval-and-how-does-it-work/?v=7516fd43adaa)
3. Permeabilization
    - Exp: Non-ion detergent so the antibody could enter the cell

*[Ab]: Antibody
*[Ab]: Antibody

4. Blocking
    - Incubation
      - Unspecific binding by using blocking buffer
        - exp: milk, BSA (Bovine Serum Albumin), gelatin, or casein blocking agents.
      - For the secondary ***Ab*** specific binding to primary ***Ab***

5. Primary antibody
    - high stringency (exp: mouse)
    - if 1st Ab~goat~, then 2nd Anti-goat
6. Second antibody
    - large quantity (Donkey)
7. Counterstain
    - Image the histology structures
8. Mounting
    - [adhere a coverslip to a tissue section or cell smear](https://www.abcam.com/reagents/mounting-media  )

Control: nonspecific antibody?


Common Dyes:  
- 5-ethynyl-2′-deoxyuridine (EdU)

**Click Reaction**

|![Labeling of DNA with EdU followed by a fluorogenic “click” reaction](https://www.future-science.com/cms/10.2144/000113463/asset/images/medium/figure1.jpg)|
|:-:|
|[&copy; Kai Li, et al.](https://www.future-science.com/doi/10.2144/000113463)|


## Western
> Fractionation of proteins by gel electrophoresis in polyacrylamide gels followed by transfer to solid support. After blocking the residual binding capacity of the membrane, the filter is probed with a specific antibody. The bound antibody is detected with a secondary reagent that is labeled.

> 1. **Gel containing separated proteins**
>  Transfer of entire protein pattern by electrophoresis.
> 2. **Membrane with blotted proteins**
>  Blocking of residual binding sites and incubation with the first antibody
> 3. **Membrane with the first antibody bound**
>  Incubation with secondary enzyme-linked antibody and substrate
> 4. **Developed immunoblot**


Why protein migrated by its weight:
1. Denatured:
  - DTT: [DL-Dithiothreitol; disruption of protein and (DNA) disulfide bonds](https://www.thermofisher.com/order/catalog/product/R0861)
  - SDS: sodium dodecyl sulfate
    - Ionic detergent; Amphipathic surfactant
    - Function: Denatures & binds to proteins to make them uniformly negatively charged, aiming to be separated by mass

2. Technically, protein is negatively charged.

**Positive Control**: Protein from housekeeping genes like ribosome protein


## Immunoprecipitation


|![](https://ie.vwr.com/stibo/bigweb/std.lang.all/12/46/18541246.jpg)|
|:-:|
|[&copy; avantor; Classic IP kit](https://ie.vwr.com/store/product/18501264/classic-ip-kit-piercetm)|

**Two important steps**:
Spin: remove the beads.
Wash: remove the Unspecific bindings.
Protein A/G has a high affinity for the antibody

**Two "A" factors**
1. **Abundance** of antigen solution
2. **Affinity** of the  antibody to antigen

## Co-immunoprecipitation

- protein interacting with antigen (interactive~protein~ + Antigen)

(Boiled) denatured and separated by electrophoresis.


## DNA/RNA

> - Probe: A nucleic acid that is radiolabeled or tagged with another detectable tracer that is used to identify complementary sequences by hybridization
> - Hybridization: The process of pairing two complementary single-stranded nucleic acids. Hydrogen bonds are formed between the bases to become a single double-stranded molecule

DNA denature: Heating and high pH(11.3)


[formamide](https://en.wikipedia.org/wiki/Formamide): a 42$^{\circ} C$
- Maintains RNA denaturing stats
- Lower the Tm of DNA

> Tm:
> 1. GC%
> 2. Length
> 3. % Homology
> 4. Salt concentration

> The hybridization stringency is adjusted to take these factors into account.

## Southern & Northern blot

N -> RNA
S -> DNA

RNA should be treated with denaturing detergent (Formamide/paraformaldehyde) to remove **the secondary structure**.
  - Exp: Hairpin, etc.
DNA doesn't need this move after being digested (restriction nucleases).
  - Digestion: smaller linear fragments.
  - Denaturation: NaOH (sodium hydroxide)

## PCR
> Polymerase Chain Reaction
> -  Uses thermostable polymerase to synthesize a region of target DNA defined at each end by specific primers.
>      - **Denaturation** at high temperature;
>      - **annealing** at lower temperature;
>      - **extension** at moderate temperature;
>      - **Repeat** for many cycles.
> - Primer: A short nucleic acid that is paired to another nucleic acid template. The primer provides a free 3’-OH that can be extended by polymerases along the template.

$\uparrow$ salt $\to \ \downarrow$stringency
$\downarrow$ low salt $\to \ \uparrow$ stringency


## RFLP
Restriction Fragment Length Polymorphism - RFLP
> Based on:
> - Linkage of the genetic disease with the RFLP
> - Polymorphic distribution of restriction enzyme sites
>   - The normal variation in DNA sequence between two unrelated individuals is about one base out of 400. This variation can create or destroy restriction enzyme sites.
> - Structural alteration in the gene or flanking sequences

> **Linkage**
> Clues to the location of a gene can come from comparing the inheritance of a mutant gene with the inheritance of markers of a known chromosomal location within a family
> Coinheritance, or genetic linkage, of a disease gene and a marker suggests that they are physically close together on the chromosome
> Linkage is determined by analyzing the pattern of inheritance of a gene and a marker in suitable families

> Because the RFLP is inherited just like a gene the individual chromosomes can be followed as they pass from generation to generation by tracing the inheritance of the marker fragments

## Variable Number of Tandem Repeats
> A hypervariable locus that consists of a variable number of identical sequences joined together in tandem.
> There are many VNTR loci in the genome; therefore, the pattern of fragments from the VNTR loci in one individual is essentially unique for that individual.
> When DNA is digested with a restriction endonuclease that cuts the sequence flanking a VNTR locus, the lengths of the DNA fragments produced in different individuals depend on the number of repeats at the locus - a DNA fingerprint

## DNA Fingerprinting
> Analyses of a set of polymorphic loci that are chosen so that the probability that two individual DNA samples with identical haplotypes could by chance have come from two different individuals is low.
> The most useful polymorphisms for this purpose occur at hypervariable loci.

## Allele Specific Oligonucleotide Hybridization

> - Oligonucleotide probes are used that hybridize selectively to the normal or the mutant allele. These allele-specific probes can be used for any disorder where the nucleotide sequence of the mutant and normal alleles are known.
>   - Example: cystic fibrosis
> - The ASO hybridization method exploits selective hybridization to distinguish between the normal and mutant alleles

## DNA Chip Technology

> Miniaturized silicone chips with densely packed arrays of oligonucleotides have been developed. The chips can be hybridized to a fluorescently labeled sample. Laser scanning of the chip can detect localized areas of fluorescence.

## DNA Microarray:
> - The Genome Project made available cDNA clones and DNA sequences representing the entire genome
> - Robotics allows precise immobilization of DNA onto microarrays
> - Well-developed knowledge of nucleic acid hybridization
> - Computer-based bioinformatics to analyze massive amount of data.


> In most instances cluster analyses of microarrays confirms pathological classification and staging, but new tumor entities have also been revealed.
> Response to chemotherapy has been predicted by the expression profile.
> Matrix-assisted laser desorption ionization (MALDI) mass spetrometry is a complementary means to analyze the proteome.

## Target amplified nucleic acid-based techniques couple hybridization with target nucleic acid and signal amplification techniques.

> The assays require exacting assay conditions (i.e. control for molecular contamination of reactions by amplified products leading to false positive results and using quality assurance measures and experimental controls that are carefully designed and executed) and careful interpretation.
