---
toc: true
url: tulane_biochem_6
covercopy: <a href="https://www.researchgate.net/publication/334738832_Specific_purification_of_a_single_protein_from_a_cell_broth_mixture_using_molecularly_imprinted_membranes_for_the_biopharmaceutical_industry">© Wenyuan Xie, et al</a>
priority: 10000
date: 2021-09-24 08:31:54
title: "Protein Analysis|Graduate Biochemistry 6| Tulane"
ytitle: "蛋白研究方法|生化 6|杜兰"
description: "Class notes of Graduate Biochemistry in Tulane"
excerpt: "Let's explore the function of proteins"
tags: [Classes, Biochemistry, Tulane Classes]
category: [Notes, Class, Tulane, Biochemistry]
cover: "https://www.researchgate.net/publication/334738832/figure/fig1/AS:786014943584257@1564411974204/Schematic-diagram-of-preparation-and-separation-of-the-protein-imprinted-membrane.png"
thumbnail: "https://static-s.aa-cdn.net/img/ios/1535372364/8bea5e7d0cecf8c51488133d2c6a4f48?v=1"
---

## Purification and Analysis Overview
> - Methods of protein separation and purification
> - Protein solubility
> - Isoelectric focusing
> - Principles of chromatography
>   - Polyacrylamide gel electrophoresis
>   - Gel filtration
>   - Ion exchange/Hydrophobic interaction
>   - Affinity Chromatography
> - Ultracentrifugation
> ***Methods of protein analysis***
> - Summary of primary information
> - Protein sequencing by Edman degradation
> - Mass spectrometry
> - Methods of determining protein three dimensional structure

## Methods of Macromolecule Separation

|Method of Separation|Physical Basis|sensitivity | Specifity|
|:-|:-|:-|:-|
|[Precipitation](#Protein-Solubility)|Solubility (charge, isoelectric point)|μg - kg | Very low|
|[Mass Spectrometry](#Mass-Spectrometry)|Molecular Mass|fg - ug | Very High|
|***Nondenaturing PAGE***|***Charge/mass, shape***|***μg - mg***|***High***|
|[***SDS PAGE***](#SDS-PAGE)|***Molecular Weight***|***μg - mg***|***High***|
|[***Isoelectric focusing***](#Isoelectric-focusing)|***Isoelectric point***|***μg - mg***|***Moderate***|
|***Capillary Electrophoresis***|***Charge/mass***|***ng - μg***|***High***|
|==[Gel Filtration](#Gel-Filtration-Chromatography)==|==Size and shape==|==ng - mg==|==High==|
|==[Ion Exchange](#Ion-Exchange-Chromatography)==|==Net charge==|==ng - mg==|==Moderate High==|
|==[Hydrophobic interaction Chromatography](#Hydrophobic-Interaction-Chromatography)==|==Hydrophobicity==|==ng - mg==|==High==|
|==[Affinity Chromatography](#Affinity-Chromatography)==|==Specific molecular interactoin==|==ng - mg==|==Very High==|
|[**Density Gradient Ultracentrifugation**](#Density-Gradient-Ultracentrifugation)|**Molecular weight, density and shape**|**μg - g**|**Moderate**|
|[**Equlibrium Density Ultracentrifugation**](#Ultracentrifugation)|**Density**|**μg - g**|**Moderate**|


Legend:
- ***Electrophoresis techniques***
- ==Chromatography techniques==
- **Ultracentrifugation techniques**

### Protein Solubility in Salt

> ***Salting in***
> - Increase in solubility with increasing salt at low ionic strength
> - Proteins unfold at low ionic strength and irreversibly aggregate
>
> ***Salting out***
> -  Protein precipitation at high ionic strength driven by ionic charge shielding and solvent effects
> -  Solubility is minimum at isoelectric point (ionic effect)
> -  Effectiveness of salting out in **Hoffmeister series** (solvent effect):
>     - Anions: SO4>H2PO4>CH3COO>Br>I>ClO4>SCN
>     - Cations: NH4,Cs,K,Na>Li>Mg>Ca>Ba


PS: Increase the salt, you shield the charge surface of the protein

| ![Salting In and Salting Out](https://wou.edu/chemistry/files/2019/08/Salting-in-out-2-768x599.png) | ![Ammonium sulfate solubility curve](https://www.researchgate.net/profile/Richard-Burgess-5/publication/229422294/figure/fig1/AS:669381373722633@1536604364653/Ammonium-sulfate-solubility-curve-for-a-hypothetical-protein-This-represents-the-log.png)     |
| :--: | :----: |
| [© Michel Awkal 2020](https://commons.wikimedia.org/wiki/File:Saltin_in_%26_Salting_out.png)      | [© Richard R Burgess 2009](https://www.researchgate.net/publication/229422294_Chapter_20_Protein_Precipitation_Techniques)|

Solting in: The unsaturated ions on the surface of the protein can interacted with other ions to increase the soluability of the protein.

### Isoelectric focusing

| ![Isoelectric focusing](https://www.researchgate.net/profile/Blake-Meyers/publication/6626845/figure/fig6/AS:667197747757081@1536083747895/Fractionation-of-proteins-by-preparative-isoelectric-focusing-A-Protein-samples.jpg) |
| :-------------: |
| [© Blake C Meyers 2007](https://www.researchgate.net/publication/6626845_Use_of_Massively_Parallel_Signature_Sequencing_to_Study_Genes_Expressed_During_the_Plant_Defense_Response)|

> Isoelectric point (pI) is the pH at which positive and negative charges are equally abundant on a molecule.
> pI for the **G** is ~6;
> pI for **D** is ~3.5
> pI for **K** is ~ 9.5

> - Isoelectric focusing is equilibrium electrophoresis
> - Ampholytes - Mixtures of buffer molecules covering a range of pKa values stabilize the pH gradient in an electric field
> - Charged molecules, including the ampholytes, will move toward the location where their net charge is zero. This is the isoelectric point, or pI.

### Chromatography

> ***Principle of Chromatography***
> - Molecules in a mobile medium move through a stationary phase with which they can interact.
> - Differential interactions with the stationary phase are the basis for physical separation of molecules
>
> ***Some stationary phases***
> - Paper (paper Chromatography)
> - Bare silica (Thin layer chromatography)
> - Functionalized Silica (Column Chromatography)
> - Various polymers (acrylamide, dextran, agarose, and others)
>
> Mobile phase is usually liquid

### SDS PAGE

Sodium Dodecyl Sulfate Polyacrylamide Gel Electrophoresis

> Acrylamide gel:
> - Polymerized acrylamide
> - Macromolecule movement is hindered
> - Movement driven by electric field
> - Mobility is determined by charge, size and shape

> SDS: A detergent that denatures proteins


- with/withough reducing agent:
  - breaking the disulfide bond

> Proteins denatured in SDS:
> - 3 SDS bound per amino acid
> - Constant charge/mass
> - Mobility α ln(mass)

- SDS can also solubilize hydrophobic AA
- Useful but may lots some information.


### Gel Filtration Chromatography

| ![Gel Filtration Chromatography](https://www.researchgate.net/profile/Marcell-Wolf/publication/305215444/figure/fig5/AS:614360984612865@1523486481230/Size-exclusion-chromatography-also-called-gel-filtration-separates-proteins-according.png)     | ![Gel Filtration Beets](https://www.researchgate.net/profile/Rufika-Abidin/publication/313851127/figure/fig7/AS:590027151138817@1517684843651/Principles-of-size-exclusion-chromatography-A-Magnified-diagram-of-a-media-bead-and-its.png)    |
| :------------- | :------------- |
| [© Marcell Wolf, 2015](https://www.researchgate.net/publication/305215444_Effective_interactions_in_liquid-liquid_phase_separated_protein_solutions_induced_by_multivalent_ions)       |  [© Rufika Shari Abidin](https://www.researchgate.net/publication/313851127_Murine_polyomavirus_VLP_as_a_platform_for_cytotoxic_T_cell-based_influenza_vaccine_candidates)     |

> Vo = Void Volume, Excluded Volume
> V~i~ = Included Volume
> V~T~ (Total solvent volume) = V~o~+V~i~
>
> For each mocromolecule
> V~p~ = Penetrable volume
> V~e~ (Protein Elution Volume) = V~o~+V~p~
>
> σ (fractional retention) = V~p~/V~i~
> V~p~/V~i~ = (V~e~ - V~o~)/(V~T~ - V~o~)

- Polymer pours beets.
- separated the protein by its size
- You can't tell the protein is monomor, dimer or etc.
- small proteins randomly filtered inside of the beets and causing transfer delay.

### Ion Exchange Chromatography

> - **Retention** is due to **ionic interactions** with fixed charges on stationary phase
> - **Counterions** compete for binding sites
>
> - Three phases: binding, elution, regeneration
>
> Binding:
> - Cation exchange: fixed anions
> - Anion exchange: fixed cations
>
> *Typical Ionic groups*
> - Diaminoethylamino (DEAE)
> - Carboxymethyl (caboxylate)
> - Sulfoethyl/Sulfobenzyl (sulfate)

The material is charged:
  - Same charge: not interacted
  - opposite Charge: Reacted

### Hydrophobic Interaction Chromatography

> - Binding to stationary phase is hydrophobic
> - Elution is with organic solvents (Acetonitrile, methanol, isopropanol, hexane)
> - Often used in High Pressure Liquid Chromatography (HPLC).

### Affinity Chromatography

> - Retention is due to specific interaction
>
> - ***Typical affinity molecules:***
>     - Antibody
>     - Ligand/substrate/inhibitor
>     - Lectin (carbohydrates)

> - Nickel-Nitrilotriacetate or Ni-NTA Binds to engineered poly Histidine

PS: Some specific interact between the protein and the ?agent

### Example of Enzyme Purification

|Purification Step|Total Acitivity |% recovery |Specific Activity|
|:-|:-:|:-:|:-:|
| Crude Extract | 31800 |100 | 1|
| Acid Precipitate| 59900| 189 |10.1|
| Amm. Sulfate Precipitate | 57200| 175 | 27.2|
| GTP ( affinity) | 37300 | 117| 3962|
| Mono Q (ion exchange) | 14300 | 45 | 6304|
| Gel filtration | 11000| 35 | 11620|


{% echarts 400 '85%' %}

option = {
  tooltip: {
    trigger: 'axis',
    axisPointer: {
      type: 'shadow'
    }
  },
  legend: {
    data:['Total Acitivity', '% recovery', 'Specific Activity']
  },
  grid: {
    left: '3%',
    right: '4%',
    bottom: '3%',
    containLabel: true
  },
  xAxis: [
    {
      type: 'category',
      data: ['Crude Extract', 'Acid Precipitate', 'Amm. Sulfate Precipitate ', 'GTP ( affinity)',
      'Mono Q (ion exchange)', 'Gel filtration'],
      axisLabel: { interval: 0, rotate: 15 },
    }
  ],
  yAxis: [
    {
      type: 'value'
    }
  ],
  series: [
    {
      name: 'Total Acitivity',
      type: 'bar',
      label: {
        show: true,
        rotate: 90,
        align: 'left',
        verticalAlign: 'middle',
        position:  'insideBottom',
        formatter: '{c} | {a}',
        distance: 15
      },
      emphasis: {
        focus: 'series'
      },
      data: [31800 , 59900, 57200, 37300 , 14300 , 11000]
    },
    {
      name: '% recovery',
      type: 'bar',
      label: {
        show: true,
        rotate: 90,
        align: 'left',
        verticalAlign: 'middle',
        position:  'insideBottom',
        formatter: '{c} | {a}',
        distance: 15
      },
      emphasis: {
        focus: 'series'
      },
      data: [100 , 189 , 175 , 117, 45 , 35]
    },
    {
      name: 'Specific Activity',
      type: 'bar',
      label: {
        show: true,
        formatter: '{c} | {a}',
      },
      emphasis: {
        focus: 'series'
      },
      data: [1,10.1, 27.2, 3962, 6304, 11620]
    },
  ]
};

{% endecharts %}


### Ultracentrifugation

> ***Bouyant*** force depends on:
> - Density Difference between molecule and solvent

> ***Frictional*** force depends on:
> - Molecular shape
> - Solvent viscosity

> ***Centrifugal*** force depends on:
> - Molecular weight
> - Rotation speed
> - Distance from center

> ***Ultracentrifugation***
> - Can be used to obtain information on molecular weight, molecular shape and intermolecular interactions
> - Sedimentation velocity (S=Svedbergs)
> - Sedimentation equilibrium
>

#### ***Density Gradient Ultracentrifugation***
>
> - Sucrose gradient:
>   - Gradient is man-made and impermanent
>   - Sample applied at top
>   - Separation is by sedimentation velocity
>
> CsCl density Gradient:
>   - Gradient is formed by centrifugal force
>   - Equilibrium position of macromolecule determined by its density

### Protein  Sequence Determination

#### Primary Information
> - Number of chains
>   - Dansyl Chloride reaction (Voet, Voet & Pratt pg 109)
>   - One cycle of Edman degradation
>   - SDS PAGE (under reducing conditions)
> - Primary Sequence
>   - Fragmentation, purification of fragments and Edman sequencing or mass spectroscopy
>   - Requires overlapping fragments
>     - Common fragmentation methods:
>       - Cyanogen Bromide (cleaves C-term side of Met)
>       - ==Trypsin== (cleaves C-term side of Arg, Lys)
>       - Chymotrypsin (cleaves C-term side of Trp, Phe, Tyr)
> - Disulfide linkages
>   - SDS PAGE (reduced and non-reduced)
>   - Fragmentation, purification and partial sequencing with disulfide bonds oxidized/reduced

## Protein Sequencing

> ***Principles of Edman degradation:***
> - Cyclic, sequential removal and identification of N-terminal amino acids
> - Identification is done by High Pressure Liquid Chromatography (HPLC)
> - Can sequence 50-100 residues
> - Sequence determination requires overlapping fragments

1. Derivitization: Phenylisothiocyanate (PITC)
2. Acid cleavage: PTC polypeptide terminal remove
3. Acid reduction: Polypeptide like of residue
4. Separation and Identification
> (Back to 1.)
> cite: Voet, Voet & Pratt pg 113

### Mass Spectrometry
> ***Principle of Mass Spectrometry***
> - Acceleration of a molecule in an electric field depends on the mass/charge ratio

> ***Protein Mass Spectrometry***
> MALDI TOF - Matrix assisted laser desorption - Time of flight
> ESI - Electrospray ionization
>
> Both methods can be used to determine the molecular weight of proteins up to ~ 105 Daltons
>
> Molecular weight determination is unambiguous

### Proteomics - by mass spectrometry

> ***Genomics***: The study of the an organism’s genetic information.
> ***Proteomics***: The study of large scale protein expression pattern in a cell type or tissue.
>
> Genomic information is fixed. Proteomic information depends on life cycle, environment, cell type etc.

> Below: Two-dimensional gel electrophoresis of an E. coli whole cell extract.
> Modern MALDI/SELDI mass spectrometers can identify these proteins
> Often done by cutting out spots, digesting proteins with trypsin and getting the mass of the fragments by mass spec.

### Determination of Protein Structure

> - Two methods are used to determine protein structure:
>   - X-ray crystallography
>   - Two-dimensional Nuclear Magnetic Resonance (NMR) spectroscopy

> ***X-ray crystallography***
> - purify protein
> - grow crystals
> - collect X-ray diffraction patterns
> - determine electron density in three dimensions
> - solve protein structure
>
> ***Advantages:***
>   - Can solve very large proteins/complexes
>   - Gives atomic-level resolution
>
> ***Disadvantages:***
>   - crystallization is often difficult or impossible
>   - crystal contacts can distort structure

### Determination of Protein Structure

> ***2D NMR Spectroscopy***
>   - purify protein
>   - collect multidimensional NMR spectra in solution
>   - assign peaks and crosspeaks to specific residues
>   - calculate distances between residues from the strength of the crosspeaks
>    - solve structure using distance constraints
> ***Advantages:***
>    - no crystallization required
>    - structure is a solution structure
> ***Disadvantages:***
>    - Only small proteins (<200 residues) can be solved
>    - Resolution is not at atomic level
