---
toc: true
url: tulane_biochem_13
covercopy: © Karobben
priority: 10000
date: 2021-10-22 08:39:32
title: "Nuclear"
ytitle: "核酸|生化 13| 杜兰"
description: "Class notes of Graduate Biochemistry in Tulane"
excerpt: "Let's explore the function of proteins"
tags: [Classes, Biochemistry, Tulane Classes]
category: [Notes, Class, Tulane, Biochemistry]
cover: ""
thumbnail: "https://static-s.aa-cdn.net/img/ios/1535372364/8bea5e7d0cecf8c51488133d2c6a4f48?v=1"
---

## The central Dogma of Molecular Biology

### DAN
### Puring; Pyrimidine

|![Puring; Pyrimidine](https://z3.ax1x.com/2021/11/25/oAgyCV.png)|
|:-:|
|© Zac Pursell|
Nucleotides

## ==Biosynthesis of the DNA & RAN purines==
> ==Exam: Nucleotide synthesis==

### OverView

|![Puring; Pyrimidine](https://z3.ax1x.com/2021/11/25/oA2ydA.png)|
|:-:|
|© Zac Pursell|

==Adenine and guanine nucleotides originate from inosinic acid (IMP)==, which derives its atoms from phosphoribosyl pyrophosphate (==PRPP=), amino acids, formate and CO~2~

|![IMP](https://upload.wikimedia.org/wikipedia/commons/thumb/6/63/Inosinic_acid_structure.svg/800px-Inosinic_acid_structure.svg.png)|![PRPP](https://upload.wikimedia.org/wikipedia/commons/thumb/2/26/Phosphoribosyl_pyrophosphate.svg/1920px-Phosphoribosyl_pyrophosphate.svg.png)|
|:-:|:-:|
|[© wikipedia](https://en.wikipedia.org/wiki/Inosinic_acid)|[© wikipedia](https://en.wikipedia.org/wiki/Phosphoribosyl_pyrophosphate)|

<details><summary>

  ==Click to see from PRPP to IMP==
  </summary>

  ![© Biochemistry](http://www.columbia.edu/cu/biology/courses/w3034/Larry/readings/PurineChapter/8883n27_03.jpg)
</details><br>

==AMP== synthesis requires ==GTP hydrolysis== and aspartate==
==GPM== syntehsis requries ==ATP and glutamine==

|![](http://www.columbia.edu/cu/biology/courses/w3034/Larry/readings/PurineChapter/8883n27_06.jpg)|
|:-:|
|[© Biochemistry; Garrett & Grisham](http://www.columbia.edu/cu/biology/courses/w3034/Larry/readings/PurineChapter/PurineChapter.html)|

## Biosynthesis of the RNA Purine Ribonucleotides

$$
AMP+ATP \overset{NMK}{\leftrightharpoons}  2 ADP + 2NTP \overset{NDK}{\leftrightharpoons} 2ATP + 2NDP
$$

$$
GMP+GTP \overset{NMK}{\leftrightharpoons} 2 GDP + 2NTP \overset{NDK}{\leftrightharpoons} 2GTP + 2NDP  
$$

NMK: Nuclioside Monophosphate Kinase
NDK: Nuclioside Diphosphate Kinase



### Regulation of the Nucleotide Synthesis

Nucleotide Synthesis is Highly Regulated: Through **Feedback Inhibition**, **Feedforward Activation** and **Coordinate Regulation**

|![](https://z3.ax1x.com/2021/11/25/oALCuR.png)|
|:-:|
|[© Fundamentals of Biochemistry, 3rd edition](https://www.amazon.com/Fundamentals-Biochemistry-Life-Molecular-Level/dp/0470129301/)|


|![ribonucleotide reductase](http://www.columbia.edu/cu/biology/courses/w3034/Larry/readings/PurineChapter/8883n27_22.jpg)|
|:-:|
|[© Biochemistry; Garrett & Grisham](http://www.columbia.edu/cu/biology/courses/w3034/Larry/readings/PurineChapter/PurineChapter.html)|

Activity-determining effector (Hexomerization site)

Ribonucleotide reductase is a Allosteric enzyme. Schematic diagram of the quaternary structure. The enzyme consists of two identical pairs of subunits, R1~2~ and R2~2~. Each R2 subunit contains a binuclear Fe(III) complex that generates a phenoxy radical at Tyr 122. The R1 subunits each contain three different allosteric effector sites and five catalytically important Cys residues. The enzyme’s two active sites occur at the interface between the R1 and R2 subunits.

### Ribonucleotide Reductases Reduce NDPs to dNDPs

$$
CDP \& UDP \underset{dTTP}{\overset{ATP}{\longrightarrow}} dCDP \& dUDP
$$
$$
GDP \overset{dTTP}{\longrightarrow} dGDP
$$
$$
NDP \overset{dATP}{\longrightarrow} dNDP
$$

### dTMP Is Synthesized from dUMP

==Not TDP → dTDP reduction==

$UDP \overset{RNR}{\longrightarrow} dUDP \overset{NDP\ kinase}{\longrightarrow} dUTP \overset{dUTPase}{\longrightarrow} dUMP \overset{thymidylate\ synthesis}{\longrightarrow} dTMP  → dTTP$

$dUMP + THF \overset{thymidylate\ synthase}{\longrightarrow} dTMP + DHF$

dUTP intermediates are quickly broken down into dUMP to avoid mutagenic dUTP incorporation. The concentration of DHF was quickly rescued by DHFR:

$DHF  \overset{DHFR}{\longrightarrow} THF$

## FdU and Methotrexate are anticancer compounds thath disrupt dTMP Biosynthehsis

Cancer require large amount of dTTP than do normal cells.

| ![Fluorodeoxyuridylate](https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/5-Fluorodeoxyuridine_monophosphate.svg/715px-5-Fluorodeoxyuridine_monophosphate.svg.png) | ![Methotrexate](https://upload.wikimedia.org/wikipedia/commons/thumb/3/30/Methotrexate_skeletal.svg/1280px-Methotrexate_skeletal.svg.png)     |
| :-------------: | :-------------: |
| [© wikipedia](https://en.wikipedia.org/wiki/Fluorodeoxyuridylate)       | [© wikipedia](https://en.wikipedia.org/wiki/Methotrexate)|
|5-Fluorodeoxyuridylate(FdUMP): ==locks== dTMP synthesis by binding and irreversibly inhibiting thymidylate synthase|Methotrexate: ==DHF analog==; binds tightly and inhibits DHFR|

### Purines Salvage

Unlike de novo purine syntehsis, ==purine salvage is NOT conserved== across species.

==Mammal==:
$Adenosine + PRPP \overset{APRT}{\longrightarrow} AMP + PP_i$
$Hypoxanthine/Guanine + PRPP\overset{HPRT}{\longrightarrow} AMP + PP_i$

### Animal purin catabolism

|![purin catabolism](https://z3.ax1x.com/2021/11/26/oEZzTK.png)|
|:-:|
|[© Fundamentals of Biochemistry, 3rd edition](https://www.amazon.com/Fundamentals-Biochemistry-Life-Molecular-Level/dp/0470129301/)|


## DNA structures and metabolism

### The B-DNA Molecule

|![Structure of the DNA](https://microbenotes.com/wp-content/uploads/2019/04/Different-forms-of-DNA.jpg)|
|:-:|
|[© Sagar Aryal, 2019](https://microbenotes.com/different-forms-of-dna-b-form-a-form-z-form/)|

- Right Handed;
- 20 Å in diameter with two grooves.
- The distance between the bases (rise) is 3.4 Å, 10 bp
- Major groove is ~1.2nm wide, 0.6~0.8nm deep.
- minor groove is ~0.7nm Wide.

B-form by far most common
dsDNA under dehydrating conditions can exist as A-form

### Metabolic Instability of RNA

|![RNA self-hydrolysis](https://www.researchgate.net/profile/Shingo-Nakamura-2/publication/10654215/figure/fig3/AS:601657004011524@1520457616208/Mechanism-of-RNA-cleavage-under-mildly-acidic-conditions-In-the-range-of-approximately.png)|
|:-:|
|[© Gail Mitchell Emilsson, et al; 2003](https://www.researchgate.net/publication/10654215_Ribozyme_speed_limits)|

==RNA is intrinsically unstable== in aqueous media and undergoes ==self-hydrolysis== due to nucleophylic attack of the 2 ’OH on the adjacent phosphodiester linkage. This instability ==increases at alkaline pH==.

### Base pairing

|![DNA Base pairing](https://www.researchgate.net/profile/William-Brown-21/publication/334974633/figure/fig2/AS:788647141986305@1565039539540/a-standard-base-pairing-arrangements-of-the-canonical-nucleotide-isomers-and-b.png)|
|:-:|
|[© William Brown](https://www.researchgate.net/publication/334974633_Unified_Physics_and_the_Entanglement_Nexus_of_Awareness)|


### Syn- and anti-configurations: the basis for Z-DNA formation

|![purin catabolism](https://z3.ax1x.com/2021/11/26/oEeFld.png)|
|:-:|
|[© Fundamentals of Biochemistry, 3rd edition](https://www.amazon.com/Fundamentals-Biochemistry-Life-Molecular-Level/dp/0470129301/)|


Purine nucleotides, particularly guanine nucleotides, have a a tendency to exist in the “syn” configuration. A repeats of GC sequence causes a distortion in the double helix with the phosphates zig-zagging in the backbone.

### Sugar-base configurations that affect the conformation of DNA and RNA molecules

|![Ribose ring in C2′endo and C3′ endo conformation](https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/Ribose_ring_in_C2%E2%80%B2endo_and_C3%E2%80%B2_endo_conformation.svg/530px-Ribose_ring_in_C2%E2%80%B2endo_and_C3%E2%80%B2_endo_conformation.svg.png)|
|:-:|
|[© wikipedia](https://commons.wikimedia.org/wiki/File:Ribose_ring_in_C2%E2%80%B2endo_and_C3%E2%80%B2_endo_conformation.svg)|

C-2' endo (South): Distance of phosphates: 7.0 Å; Favored in B-Form DNA
C-3' endo (North): Distance of phosphates: 5.9 Å; Favored in RNA and some DNA forms

### helical conformations

Polynucleotides have a natural tendency to form helical conformations due to stacking interactions between the adjacent hydrophobic bases of the polymer


Hydrophobic forces on planar bases drive stacking interactions

Random coils (unstacked polymers) are favored in alkaline solution, in some organic solvents and at high temperatures

### Base Pair Geometry

|![Base Pair Geometry](https://z3.ax1x.com/2021/11/26/oEfSfI.png)|
|:-:|
|© Zac Pursell|

### Metabolic roles of cellular RNA

|![Metabolic roles of cellular RNA](https://z3.ax1x.com/2021/11/26/oEfknS.png)|
|:-:|
|© Zac Pursell|

### Structure of the RNA

RNA double-helices exist in a variety of shapes

base-pairing, stacking interactions, covalently modified bases, 3 base hydrogen bonding all contribute to compact and complex structure

### Ribosomal RNAs
high degree of base pairing

large numbers of "stems" and "loops"
unpaired bases can interact with ribosomal proteins or RNA bases from other RNAs
base-paired segments are mostly α-helical

### Some RNAs Capable of Carrying Out Catalysis

self-splicing and self-cleavage reactions in vivo engineered catalysis in vitro

### Genomes

- dsDNA, linear, e.g., eukaryotic nuclear DNA, many viruses.
- dsDNA, circular, e.g., bacterial chromosomes, mitochondrial DNA, plasmids, some viruses.
- ssDNA, circular or linear e.g., the genomes of some bacterial, plant and animal viruses.
- ssRNA, e.g., the genomes of many viruses.
- dsRNA, e.g., the genomes of many viruses.

### Size of Genome

|![Size of Genome](https://www.researchgate.net/profile/M-Golubovsky/publication/8154001/figure/tbl1/AS:669290105692175@1536582604279/Genome-size-gene-numbers-of-some-prokaryotes-and-eukaryotes.png)|
|:-:|
|[© M. D. Golubovsky, Kenneth Manton; 2005](https://www.researchgate.net/publication/8154001_Genome_organization_and_three_kinds_of_heritable_changes_General_description_and_stochastic_factors_a_review)|
