---
toc: true
url: tulane_cellbio_6
covercopy: <a href="https://pdb101.rcsb.org/motm/32"> © PDB </a>
priority: 10000
date: 2021-09-30 09:30:15
title: "6 Protein Folding and Degradation|Advanced Cell Biology|Tulane"
ytitle: "6 蛋白折叠|高等细胞生物学|杜兰"
description: "Classes notes of the Advanced Cell Biology"
excerpt: "Classes 5,  Cytoskeleton"
tags: [Classes, Cell Biology, Tulane Classes]
category: [Notes, Class, Tulane, Cell Biology]
cover: "https://z3.ax1x.com/2021/10/18/5aRadg.png"
thumbnail: "https://www.researchgate.net/profile/Callixte-Yadufashije/publication/324543378/figure/fig26/AS:616049695285249@1523889101090/Onion-cell-before-staining.png"
---

## Protein Folding and Degradation

> • Lodish MCB, 8th ed., pages 81-105 and 622-631.
> • 3.2 Protein Folding
> • 3.3 Protein Binding and Enzyme Catalysis
> • 3.4 Regulating Protein Function
> • 13.6 Transport Into and Out of the Nucleus


## Protein Folding

Principles of protein folding:
- AA Seq. → 3D Structure and Function.
- Assisted by: ATP-dependent chaperones and chaperonins  
- Misfolded/denatured proteins → Diseasecan
  - Exp: amyloid fibril: Alzheimer & Parkinson.

### Factors that Affect Protein Folding/Structure

Hypothetical protein folding pathway:
- Primary → Secondary → Tertiary
- Native conformation: lowest free energy

Forces that govern folding
> - Hydrophobic interaction (favors)
> - Conformational entropy (disfavors)
> - Steric hindrance limits dihedral angles off central α-carbon
> - van der Waals contacts (packing)
> - Hydrogen bonds
> - Electrostatic interactions

PS:
- Planar peptide bonds
  - φ and ψ: steric clash

### PS: How AA Seq determing the high dimensions information

For example, :
  - Steric effect: a large side chain, such as that of (**W**), might sterically block one region of the chain from packing closely against another region
  - Charge effect: a side chain with a positive charge (**A**) , might attract a segment of the polypeptide that has a complementary negatively charged side chain (e.g., **D**).
  - Secondary structure: Example we have already discussed is the effect of the aliphatic side chains in heptad repeats in promoting the association of helices and the consequent formation of coiled coils. Thus a polypeptide’s primary structure determines its secondary, tertiary, and quaternary structures.

### Steric hindrance and secondary structure

> - Values of dihedral angles &phi; and &psi; are excluded by contact between bulky groups - steric hindrance (clash)
> - "Allowed" regions of Ramachandran Plot correspond to major secondary structures



## Molecular Chaperone Families

*[Mito]: Mitochondria
*[mito]: Mitochondria
*[chlp]: Chloroplast
*[nuc]: nuclear
*[cyto]: cytoplasmic

> • stabilize unfolded or partly **folded** proteins and prevent **aggregation** during synthesis, membrane translocation, assembly, and degradation.
> • Molecular chaperones bind exposed segments and surfaces of unfolded proteins.
> • Hsp60-Hsp10 family chaperones (Mito and chlp) manage folding of enzymes and proteotoxic stress
> • Hsp70-Hsp40 family chaperones (in cyto, ER, nuc,mito, chlp) manage protein translocation across membranes and proteotoxic stress
> • Hsp90 family chaperones (cyto, nuc, ER, mito, chlp) regulate signaling and gene expression and manage proteotoxic stress
> • Hsp100 (AAA+) family chaperones (cyto, mito, chlp) unfold proteins for degradation and manage proteotoxic stress
> • Hsp25 family chaperones (in cyto) manage actin assembly and proteotoxic stress - do not utilize ATP


| Molecular Chaperones     | Chaperonins|
| :------------- | :------------- |
|  Small protein which ==bind== to a short segment of a target and stabilize unfold or part folded protein.       | Large protein which form a ==folding chamber== into which all or part of unfolded protein, give it time and an appropriate environment to fold properly. |
|<li>Hsp70; <li>Hsp40; <li>Hsp90; |<li> Hsp60; <li> Hsp10|

Functions:
- Preventing aggregation
- Fold newly made proteins
- disassmble ptentially toxic protein aggregates duet to ...

### Chaperones: Hsp60-Hsp10 ATP cycle

|![](https://ars.els-cdn.com/content/image/1-s2.0-S0968000498011931-gr1.gif)|
|:-:|
|[© Alexandra Richardsona, et al.](https://www.sciencedirect.com/science/article/abs/pii/S0968000498011931)|

Video:
- [3D Animation; 2014 Youtube](https://www.youtube.com/watch?v=--NcNeLc1mo)
- [Ibiology; 2014 Youtube](https://www.youtube.com/watch?v=VjJ2XL__788)

Hsp60: Huge cylindrical supramolecular assemblies

Two groups:
- Group I: The Prokaryotic Group
  - GroEL/GroES
- Group II: The Eukaryotic Group
  - TriC

1. ATP + GroEL → ADP:
    - unfolded protein in
    - lid of GroES sealed
2. ATP → ADP → protein fold
3. ATP → ADP → GroES and peptide release
4. Pepetide enter another chamber → Release/Repeats   

### Chaperonins: Hsp70-Hsp40 ATP-cycle

> 1. Hsp40-bound "client" protein binds to Hsp70
> 2. Hsp40 J-domain triggers ATP hydrolysis, Hsp70 closes, Hsp40 released
> 3. Waiting/folding
> 4. ADP/ATP exchange triggers opening of Hsp70
> 5. Release of client protein


#### Hsp70

Monomeric

Families of heat-shock protein Hsp70 homologs:
1. Hsp70 in Mito; 2. Bip in ER, 3. DnaK in

Heat-shock: large expressed after heat-shock in bacteria.

1. +ATP → Open conformation → Expose Hydrophobic Region  → Bond client.
2. ATP → ADP → Close conformation → helping folding
3. ADP Release → conformation change → Target release
4. ATP bond → binds another target
5. Check:
    - appropriate folded → Cannot bind again
    - inappropriate folded → bind again

#### Hsp40

Four main families of **nucleotide exchange factors**:
  - GrpE: bacteria
  - BAG; HspBP; Hsp110

Hsp40 (co-chaperone) facilitate Hsp70 processes.
- Increasing the ATP hydrolysis. ×100 ~ ×1000
-


### Hsp90 ATP cycle
> 1. Hs70-bound "client" protein binds to Hsp90
> 2. ATP binding triggers closing of Hsp90
> 3. Waiting/folding
> 4. ADP hydrolysis and release
> 5. Release of client protein Hsp70-bound Hsp70

Dimer protein

Usually recognizing **partially folded protein**, highly conserved among species.

Four families: two in cytosol, one in Mito, and one in ER.
**Important**: Helping miss folded protein under the stress, convert them into **active** form or held in a **functional conformation**.

1. open conformation → Bind target
2. +ATP → Closed conformation
3. ATP → ADP → target release

Else:
  - Cooperate with Hsp70
  - Influenced by covalent modification
  - Facilitate identification and degradation


### AAA+ Protein Family

> Aggregate disassembly
> - Hsp100 of yeast
> - ClpB of E. coli
>
>  Vesicular transport complex disassembly
> - NEM-sensitive factor for SNARE disassembly
> - ==Vps4== for ESCRT disassembly
>
> DNA-binding protein assembly
> - Clamp loader
>  Export of mRNA from nucleus
> - ==Dbp5==
> Protein translocation across membranes
> - p97 for ER-associated degradation (ERAD)
>  Protein degradation
> - ClpA, ClpX of E. coli
> - "Cap" of 26S proteasome (6 in 19S cap)
>  Molecular motor
> - Dynein for organellar transport along microtubules

> (ATPases associated with various cellular activities, homologous to hexameric nucleic-acid helicases)

## 3.4 Regulating Protein Function

1. At the level of **synthesis**, **degradation**, or through noncovalent or covalent **interactions**.
2. Proteins marked for **destruction** with a polyubiquitin tag by ubiquitin ligases are degraded in **proteasomes**.
3. **Allosteric mechanisms** act as switches, reversibly turning protein activity on and off.
4. Higher-order regulation includes the intracellular **compartmentation** of proteins.
5. Proline cis/trans **isomerizations**.
6. **Alters structure** of a protein SH2 domain.
7. **Can influence protein activity**.???
8. Proline **isomerase** may act as switch to regulate protein activity.

From Textbook:
1. ALtering the rate of protein synthesis and degradation
2. Change the intrinsic activity, Exp: active or in active conformation
3. Change the location and the concentration within the cell.


## Ubiquitin- and Proteasome-Mediated Proteolysis

|![26S Proteasome](https://z3.ax1x.com/2021/10/17/5YZdYQ.png)|
|:-:|
|[© Lodish, Molecular Cell Biology, Eight Edition, p98](https://www.amazon.com/Molecular-Cell-Biology-Harvey-Lodish/dp/1464183392)|

### Proteasome
Structures:
1. ==ATPas==: AAA+ family: Red
2. Ubiquitin recepor: Yellow
3. Deubiquitinase Enzyme: Green

### Ubiquitin and Formation

| Ubiquitin's type | Features     |
| :------------- | :------------- |
| Monoubiquitinylation  | Single Ubiquitin  |
| Multyubiquitinylation | Multiple, single ubiquitin  |
| Polyubiquitinylation  | A polymeric chain of ubiquitin  |


Degradation in Proteasome
1. ==Actication==: ATP + ubiquitin (Ub) → activate Enzyme E1
    - E1 - Ub
2. ==Ttransfers== Ub → E2 (step 2 )
    - E2 - Ub
3. ==Formation==: Ubiquitin ligase (E3) transfer Ub  →  Lysine–NH2 and forming an isopeptide bond
    - UB - Peptide
4. + more Ub by repeating steps 1 – 3
    - n * (Ub) - Pepide
5. polyubiquitinylatd targed → Ub receptors (19S cap)
    - went into proteasome
6. deubiquitinase enzyme (19S cap) → Ub remove
7. a. +ATP → unfold → the 20S core (
    b. coordinately with step 6: protein → short peptide → release (7c)

Degradation: Proteosome recognition, 1. de-Ub protein, 2. unfolds protein; protein transferred to core proteolysis chamber, protein cleaved into short peptides (2-24 aa) and released for further degradation by soluble proteases.


### Determination of polyubiquitin function by the lysine used for inter-ubiquitin isopeptide bonds

|![Polyubiquitinylation](https://z3.ax1x.com/2021/10/17/5YZI61.png)|
|:-:|
|[© Lodish, Molecular Cell Biology, Eight Edition, 104](https://www.amazon.com/Molecular-Cell-Biology-Harvey-Lodish/dp/1464183392)|



Signaling: Lys-63;
Proteasomes: Lys-48:Gly-76 isopeptide.
T-lymphocyte control: Lys-33
Cell division: Lys-11

## Misfolded Proteins can form Amyloid Aggregates
|![Misfolded Proteins](https://z3.ax1x.com/2021/10/17/5YZbTO.png)|
|:-:|
|[© Lodish, Molecular Cell Biology, Eight Edition, p88](https://www.amazon.com/Molecular-Cell-Biology-Harvey-Lodish/dp/1464183392)|

> - **Amyloid aggregate plaques form inside or outside many cell types.**
> - Flat sheets form protofilaments, assemble into thicker fibrils, and into larger aggregates or macroscopic plaques.
> - Cause neurodegenerative amyloidosis diseases, including Alzheimer’s disease, Parkinson’s disease, and transmissible spongiform encephalopathy ("mad cow" disease).


### Clinical Case: Alzheimer’s Disease
|![plaques and fibrillary tangles](https://z3.ax1x.com/2021/10/17/5YZLkD.png)|
|:-:|
|[© Lodish, Molecular Cell Biology, Eight Edition, p88](https://www.amazon.com/Molecular-Cell-Biology-Harvey-Lodish/dp/1464183392)|

> 68 yo male seen in ER with apparent confusion, disorientation of where he is or the present date.
> • Most common form of dementia.
> • Affects 1 in 10 people over 65 and 50% of people over 85.
> • Plaques (Aβ peptide) and tangles (tau microtubule-associated protein) results in gradual and progressive loss of neurons and cognitive function over a period of 4-15 years (memory lapse, language impairment, loss of motor functions).

[Alzheimer's Disease Fact Sheet; 2021 NCBI](www.nia.nih.gov/Alzheimers/Publications/adfact.htm)


## Prion Diseases & Transmissible Spongiform Encephalopathies
> TSE - spongiform neurodegeneration, astrocytic gliosis.
> Prion - proteinaceous infectious agent, considered heretical concept for transmission.
> Lehninger Box 4-6

### Transmissible Spongiform Encephalopathies
> 1. Creutzfeldt-Jakob Disease (CJD)
> 2. Gerstmann-Straussler-Scheinker syndrome (GSS)
> 3. Fatal Familial Insomnia (FFI)
> 4. Kuru: Fore tribe in New Guinea
> 5. Wild Animals and Livestock (bovine, mink, feline, elk, sheep)

### The Prion Hypothesis for TSEs

> PrP^Sc^ + PrP^c^ → 2PrP^Sc^

||Synfrome| Etiology|
|:-:|:-:|:-|
| Sporadic | sCJD| Spontaneous conversion of PrPc to PrPsc (usually not mutant) |
| Acquired | Kuru |Ritualistic endocannabilism|
| | iCJD |Exposure to contaminated growth hormone, gonadotrophins, dura mater, cornea or surgical instruments|
|| vCJD| Ingestion of BSE-contaminated food products|
| Familial | fCJD | Germline mutation in PRNP locus|
| | GSS; FFI| |
>From: Hilton (2006) J Pathol 208: 134, Table 1 Iatrogenic CJD 2004

*[sCJD]: sporadic CJD
*[iCJD]: iatrogenic CJD
*[vCJD]: variant CJD
*[BSE]: bovine spongiform encephalopathy
*[CJD]: familial CJD
*[FFI]: familial fatal insomnia
*[GSS]: Gerstmann-Straussler-Scheinker syndrome
*[Kuru]: Fore tribe in New Guinea
*[fCJD]: familial CJD

## Diseases involving Amyloid Formation: Know the normal and abnormal proteins, the disease, and specific mechanisms.

Disease|Type II|Alzheimer’s disease|Prion disease|Huntington’s disease|Parkinson’s disease|ALS|Spinal Cerebellar Ataxia
|:-|:-|:-|:-|:-|:-|:-|:-
Protein|Amylin|Amyloid Precursor Protein (APP)|PrPc|Huntingtin protein|α-synuclein|SOD|ataxin
Function|Glycemic regulation with insulin|ECM, migration during development|Unk.|Unclear, MAP, vesicle trafficking|Unclear, MAP, vesicle trafficking|Destroy oxygen radicals|Unclear
Abnormal protein|n/a|Aβ42(β-Amyloid) |PrPsc|mHTT, Trinucleotide repeats (polyQ)|Lewy bodies|n/a|Trinucleotide repeats (polyQ), and nonPolyQ
Effects|Apoptosis of β islet cells|Neuronal death|Neuronal death|Neuro-tramsmitter release, neuronal death|Neuro-transmitter release, dopaminergic neuronal death|Dementia neuronal atrophy|Neuronal deat


## 13.6 Transport Into and Out of the Nucleus

| ![](https://www.researchgate.net/profile/Jeroen-Vangindertael/publication/323073291/figure/fig41/AS:614206873284615@1523449738399/A-GP210-proteins-forming-nuclear-pore-complexes-were-labeled-with-Alexa647-and.png) | ![](https://www.researchgate.net/profile/Roderick-Lim/publication/7370146/figure/fig7/AS:670340590088219@1536833059661/maging-the-nuclear-pore-complex-NPC-by-various-microscopy-techniques-a-Thin-section-of.jpg)|
| :------------- | :------------- |
| [© Jeroen Vangindertael, et al; 2018](https://www.researchgate.net/publication/323073291_An_introduction_to_optical_super-resolution_microscopy_for_the_adventurous_biologist)| [© Roderick Lim, et al; 2006](https://www.researchgate.net/publication/7370146_From_the_trap_to_the_basket_Getting_to_the_bottom_of_the_nuclear_pore_complex)      |
|![](https://www.researchgate.net/profile/Mario_Tagliazucchi/publication/276836976/figure/fig10/AS:270235693482015@1441440618772/a-Scheme-of-the-nuclear-pore-complex-NPC-a-protein-complex-in-the-nuclear-membrane.png)| ![](https://www.researchgate.net/profile/Anita-Corbett/publication/8598881/figure/fig2/AS:285611622715392@1445106525506/Structure-of-the-nuclear-pore-complex-A-Schematic-diagram-of-the-nuclear-pore-complex.png)|
|[© Mario Tagliazucchi, et al; 2015 ]()|[© Anita H Corbett, et al; 2004](https://www.researchgate.net/publication/8598881_Hot_trends_erupting_in_the_nuclear_transport_field_Workshop_on_Mechanisms_of_Nuclear_Transport)

> • Unidirectional transport of a protein larger than 40 kDa through large, complex nuclear pore complexes requires a nuclear-localization or nuclear-export signal, nuclear transport receptors, Ran G-proteins, and localized Ran?GEFs and GAPs.
> • Other molecules, including mRNPs, are transported by a Ran-independent pathway.

Nucleoporins:
- Structural nucleoporins (circule the memrain)
- Membrane nucleoporins (Y complex)
- FG nucleoporins (matrix on the pour center)

*[Nu Env]: nuclear envelope
*[NPCs]: nuclear pore complexes
*[NPC]: nuclear pore complexes

Transport Into and Out of the Nucleus
- The Nu Env: NPCs, ×30 different nucleoporins.
  - FG-nucleoporins: multiple repeats of a short hydrophobic sequence (FG-repeats), line the ==central transporter channel== and play a role in the ==transporter==.

- \>40 kDa: nuclear transport receptors interact

- ==nuclear-localization signal (NLS) or a nuclear-export signal (NES)== on the proteins in/out.
  - NLS: Nucleus-restricted proteins
  - NLS+NES: proteins that shuttle between the nucleus and cytoplasm

- Each type of NESs/NLSs is thought to interact with a ==specific nuclear transport receptor==.

- Transient interactions: Protein-[NES/NLS] + receptor.
  - Transport receptors + FG-repeats: very rapid diffusion through the NPC,

- Ran: a monomeric G protein  
  - Localization of the Ran guanine (**Ran-GEF**) **in the nucleus** and of the Ran GTPase-activating protein (**Ran-GAP**) **in the cytoplasm** creates a gradient **with high concentrations of Ran⋅GTP in the nucleoplasm and of Ran⋅GDP in the cytoplasm**.
  - The interaction of a **cargo complex** with **Ran⋅GTP** in the **nucleoplasm** causes dissociation of the **complex**, releasing the cargo into the **nucleoplasm**, whereas the **assembly of an export cargo complex** is stimulated by interaction with **Ran⋅GTP in the nucleoplasm**.
- Most **mRNPs** are exported from the **nucleus** by binding to a **heterodimeric mRNP exporter**. **RNA helicase** associated with the **cytoplasmic filaments of the NPC** that **removes the heterodimeric mRNP** exporter once the transport complex has reached the cytoplasm.

## Nuclear-Localization Signals (NLSs) Direct Proteins to Cell Nucleus
|![Pyruvate kinase Experiment](https://z3.ax1x.com/2021/10/17/5YeJ39.png)|
|:-:|
|[© Kalderon D, et al; 1984](https://pubmed.ncbi.nlm.nih.gov/6096007/)|

*[NLSs]: Nuclear-Localization Signals
*[NTR]: Nuclear transport receptor.

NLSs is a 7 residue sequence rich in the terminal of the protein, exp: PLLLRLV

- All nuclear proteins come from cytosol.
- They all have NLSs
- Experiment:
   - (a) Pyruvate kinase: in the cytosol for glycolysis.
   - (b) Chimeric pyruvate kinase containing SV40 NLS at its N-terminus: transported into the nucleus.

### Nuclear Import

|![Nuclear Import](https://www.researchgate.net/profile/Terence-Allen-2/publication/12547805/figure/fig1/AS:341225950466050@1458366014093/Nuclear-import-A-schematic-representation-of-nuclear-import-through-the-NPC-The.png)|
|:-:|
|[© Terence David Allen, et al. 2000](https://www.researchgate.net/publication/12547805_The_nuclear_pore_complex_Mediator_of_translocation_between_nucleus_and_cytoplasm)|

Ran & nuclear transporter receptor was purified.

*[Ran-GAP]: GTPase-activating protein
*[Ran-GEF]: Ganine nucleotide exchange factor

In Cytosol:
1. Importin + NLS-[Cargo].
2. Importin-NLS-Cargo passively diffuses.

In nucleus:

3. Ran-GEF: Ran-GDP + GTP  → Ran-GTP + GDP
4. Importin binds Ran-GTP → Cargo release.
5. Importin-Ran-GTP diffuses through NPC.

Back in cytosol.

6. Ran-GTP reacts with Ran-GAP → part of cytoplasmic filament.
7. GTP hydrolysis → low affinity for Importin → releases Importin.



RanL a small monomeric G protein: GTP/GDP-bound conformation: supply the energy
NTR binds to the NLSs: Recognition

### Monomeric G Proteins: The GTPase Switch

|![GTPase](https://biologicmodels.com/wp-content/uploads/2019/09/710x528_23032935_12778728_1523454624.jpg)|
|:-:|
|[© biologicmodels](https://biologicmodels.com/product/h-ras-gtpase-3k8y/)|

- GTPase:
  - GTP-bound: Active "on" conformation → interacts with target proteins to regulate their activities.
  - GDP-bound: inactive "off" conformation → intrinsic GTPase activity → hydrolyzes GTP to GDP.
GEF: Guanine nucleotide exchange factor, stimulates exchange of GDP-(off) to GTP-bound (on) forms.
GAP: GTPase-activating protein, stimulates GTP (on) hydrolysis to GDP (off).

Examples: Ras, Ran, Rho, ARF

### Ran-Dependent Export

|![Nuclear Import](https://www.researchgate.net/profile/Terence-Allen-2/publication/12547805/figure/fig2/AS:341225950466051@1458366014114/Nuclear-export-A-schematic-representation-of-nuclear-export-through-the-NPC-The.png)|
|:-:|
|[© Terence David Allen, et al. 2000](https://www.researchgate.net/publication/12547805_The_nuclear_pore_complex_Mediator_of_translocation_between_nucleus_and_cytoplasm)|


In Nucleus:

1. [Ran-GTP]-[Exportin]-[Cargo] → passive diffusion.

In Cytosol:

2. Ran-GAP: [Ran-GTP]-[Exportin]-[Cargo] → Ran-GTP + Cargo + Exportin
3. Passively diffuse: Ran-GDP & Exportin → into Nucleus.

Back in nucleus:
4. Ran-GEF: Ran-GDP + GTP → Ran-GTP + GDP

### Ran-Independent mRNA Nuclear Export

*[mRNP]: Messenger Ribonuclear Protein Complex
*[NXF1]: Nuclear Export Factor 1
*[NXT1]: Nuclear Export Transport 1

NXF1: Nuclear Export Factor 1
NXT1: Nuclear Export Transport 1
mRNP: Messenger Ribonuclear Protein Complex


Nucleoplasm:
- Passive diffusion： Heterodimeric NXF1/NXT1 + mRNPs.
Cytosol
- Dissociation: RNA helicase (Dbp5) + ATP + [NXF1/NXT1]-[mRNP] →  NXF1 + NXT1 + mRNA + ADP + Pi
- Recycling: Ran-dependent import recycles NXF1 and NXT1



---












---

Vesicular Traffic

Class I Presentation to CO8⁺ T Cells [go to find the pic and make more notes.]
Cross presentaiton
P: 1110

Class II Presentation to CD4⁺ T Cells
P: 1112

Lodish MCB, 8th ed., pages 631-659.
## Techniques for Studying the Secretory Pathway

### Protein Transport through the Secretory Pathway

### Secretory and Endocytic Pathways

### Sec Mutants Identify 5 Stages of Secretory Pathway
==Exam==: Classes , what machine could be broken/mutant


### Coated Vesicles Involved in Protein Trafficking
==Coated Vesicles table, 1, 2, 4==

### Docking/Fusion of Vesicle w/Target Membrane

Coil-Coil structure is really staby, not easty to recycle.

### KDEL Receptor Retrieval of ER-Resident Luminal Proteins from the Golgi

slide difference pH between in cis and trans Golig: help the cargo binds and releaes (pH trigger).

### Dynamin Pinching Off of Clathrin-Coated Vesicles

High energy state to close the bud neck


14.2 Molecular Mechanisms of Vesicle Budding and Fusion
14.3 Early Stages of the Secretory Pathway
14.4 Later Stages of the Secretory Pathway

| ![Lineweaver-Burk plots](https://www.ncbi.nlm.nih.gov/books/NBK545242/bin/Lineweaver__Burke__Plot.jpg)|
| :-------------: |
| [© Keith P. Delaune; Khalid Alsayouri.](https://www.ncbi.nlm.nih.gov/books/NBK545242/)|


Exam:

Draw cell component
G protein regulation and fedality
