---
toc: true
url: tulane_biochem_5
covercopy: <a href="https://www.researchgate.net/publication/326440265_Guided_Folding_of_Life's_Proteins_in_Integrate_Cells_with_Holographic_Memory_and_GM-Biophysical_Steering">© Dirk K F Meijer, Hans J. H. Geesink</a>
priority: 10000
date: 2021-09-22 08:33:16
title: "Protein Folding|Graduate Biochemistry 5| Tulane"
ytitle: "蛋白质修饰和折叠|生化 5|杜兰"
description: "Class notes of Graduate Biochemistry in Tulane"
excerpt: "Let's explore the function of proteins"
tags: [Classes, Biochemistry, Tulane Classes]
category: [Notes, Class, Tulane, Biochemistry]
cover: "https://www.researchgate.net/profile/Dirk-Meijer-5/publication/326440265/figure/fig2/AS:649300770029571@1531816775478/A-Protein-folding-from-primary-to-quaternary-structure-B-Influence-of-phonons-on.png"
thumbnail: "https://static-s.aa-cdn.net/img/ios/1535372364/8bea5e7d0cecf8c51488133d2c6a4f48?v=1"
---

## Protein Stability and Folding

> Overview
> - Post translational modifications
> - The interactions that drive protein folding
> - The interactions that drive protein unfolding
> - The net stability of folded proteins
> - Free energies of protein folding
> - Mutation studies
> - The Levinthal paradox
> - Folding pathways
> - Molecular Chaperones
> - Degradation of proteins
> Reading: All of Chapter 6

## Proteolysis

- segments-cleave
- signal peptide cleaved in Membrane/secreted proteins
- proproteins: hormones; enzyems
- The final hormone is cleaved
> - Some proteases are synthesized as zymogens which are activated by cleavage of a segment from the protein

Significant: Table (half life), soluable  
Exp: Insulin


PS: Some maturation of proteins are involved with pro-peptides cleavage. [Insulin is an classical example.](https://karobben.github.io/2021/03/29/LearnNotes/edx-biochm-5/#Enzyme-Chymotrypsin)
Insulin has not much of structure amino acid, lack of an hydrophobic core.
6 Cystines and form 3 pair of disulfide bonds. Insulin's primary structure has not enough information to drive it to the native structure. As a result, people believe that insulin has a memory of each processes to fold.

|![Insulin](https://cdn.rcsb.org/images/structures/hi/2hiu/2hiu_assembly-1.jpeg)|
|:-:|
|[© PDB ID=2hiu](https://www.rcsb.org/structure/2HIU)|


## Amino Acid Modulation

| Modulation Type   | Added Group     |Location / targets    |Reversible / Not | General Features     |
| :------------- | :------------- | :------------- | :--- |:---------- |
| Glycosylation| carbohydrates|<li>-N-linkage (on **N**-R) (More common)<li>-O-linked (on **S**, **T**) || <li>Variety; long; branched; Charged; Ionized; Large|
| Hydroxylation |Carboxyl|<li>**L** & **P** are hydroxylated on —CH₂ groups<li>Requires Ascorbate (V~C~)<li> mainly in collagens(**Gly**-X-Y)<li> **P** - 3 or 4 hydroxy proline |Irreversible|<li>Lack of V~C~, cell depart|
| Methylation | Methyl |<li>**K** and **H** are methylated on –NH₃, NH₂ groups;<li>**K** can also be trimethylated (3 Methyl groups)|Reversible|<li>==Doesn't loose it's charge==. the Amine group still keep charge.<li>Spatial Increased and effect the interactions with other atoms.<li> DNA binding proteins (**K** phosphate the DNA)|
| Acetylation/Acylation | R-Carboxyl (myristate)|<li>Amino group of the R chain to for a peptide bond<li>**K**; **H**; Terminal AA|<li>R Chain are reversible;<li>Terminal are usually not reversible|<li>Charge is ==removed==.<li>More hydrophobic <li> Increase the interact with cell membrane.<li>Reversible; Highly Controlled|
| Carboxylation |Carboxyl|<li> The γ-carbon of **E** can be carboxylated||<li> The **E** have two carboxyl groups.<li>Mainly blood coagulation enzymes.<li> Requires V~K~|
| Sulfation | Sulfate |<li>the hydroxyl of **T**| Permenent |<li> more polar, give hydrogen bonds<li>large polar ionic sulfate group|
| Prenylation | Isoprenyl chains| <li>C-terminal of **C** residue through a thioether (C-S-C) linkage. || <li> Promoting interact with membranes<li> Signal transduction proteins are prenylated|
| Amidation | Amnie group| <li> C-terminal carboxyl group| | <li>polar but not ionizable.|
|Phosphorylation| Phosphate groups |<li> The hydroxyl group in the R chain. <li>**T**; **S**; **Y**<li>**H** Could be also be phosphorylated in some case|Reversible|<li>Large polar group|


| ![Prenylation](https://www.researchgate.net/profile/Axel-Schmidt-4/publication/329134617/figure/fig1/AS:960135673413644@1605925591623/Reactions-catalyzed-by-isoprenyl-diphosphates-synthases-Dimethylallyl-diphosphate.png)     | ![Isoprenyl Group](https://upload.wikimedia.org/wikipedia/commons/6/61/Prenyl_group.png)     |
| :-------------: | :-------------: |
| [© Raimund Nagel, et al](https://www.researchgate.net/publication/329134617_Isoprenyl_diphosphate_synthases_the_chain_length_determining_step_in_terpene_biosynthesis)       | [© Wiki](https://en.wikipedia.org/wiki/Prenylation)      |

## Protein Folding

### Interactions

> **The forces that stabilize folded proteins**
> - The hydrophobic effect
> - Hydrogen bonding
> - Ionic interactions
> - van der Waals interactions (London dispersion force)
> - Disulfide crosslinks
> cite: Kauzmann, W. (1959) “Some factors in the interpretation of protein unfolding” Adv. In Proteins Chemistry. 14:1.

> **The hydrophobic effect**
> - The unfavorable interaction between hydrophobic sidechains and water is reduced when a protein folds into its native structure
> - Up to 75% of hydrophobic residues are buried in folded proteins
> - Its contribution greatly favors folded state

> **Hydrogen bonding**
> - Every residue in a protein has at least one H-bond donor and at least one H-bond acceptor (the peptide bond)
> - Concentration of H-bonding groups in a protein is ~25M
> - Most potential H-bonding groups are, in fact, H-bonded in the interior of folded proteins
> - H-bonds define secondary structure geometry

> ***The forces that stabilize folded protein***
> - The hydrophobic effect
> - Hydrogen bonding
> - Ionic interactions
> - van der Waals interactions (London dispersion force)
> - Disulfide crosslinks

> ***Ionic interactions***
> - Salt bridges form between acidic and basic residues
> - Majority of salt bridges are on surface
> - Buried salt bridges are conserved

> ***van der Waals Interaction***
> - Contribution is from tight packing of protein core
> - Protein density is ~1.3 g/ml (H20=1.0 g/ml Ethanol=0.8 g/ml)


> ***Disulfide Crosslinking***
> - Covalent crosslinks between the sulfurs of cysteine residues
> - Generally nonlocal
> - Significant (but not dominant) contribution to stability


## ***The forces that destabilize folded protein***

> - Conformational entropy
> - Hydrogen bonds to water
> - Electrostatic repulsion

###  ***Conformational Entropy***

> - Backbone of folded protein has only one (or a few) conformations.
> - The backbone of an unfolded protein has 32n conformations (because there are 3 low energy φ and ψ torsion angles for each C~α~)
> The backbone entropy change favoring unfolding:
> ΔS = Rln(3^2n^/1) = 2nRln(3) = 1.4 kcal/mol/residue (5.4 kJ/mol/residue)

### ***Hydrogen bonds to water***

> - Every amino acid has at least two hydrogen-bonding groups
> - The concentration of bulk water is ~55 Molar and each water molecule has 4 hydrogen bonding groups. Therefore the favorable energetic contribution of hydrogen bonds to folding may be small
> However, the unfavorable cost of unmade hydrogen bonds in the interior of the protein is still high



## Hydrophobic amino acids are buried

|Residue| Exposed| Burued| Interm ediate|
|:-|:-:|:-:|:-:|
|S|0.70|0.20|0.10|
|T|0.71|0.16|0.13|
|A|0.48|0.35|0.17|
|G|0.51|0.36|0.13|
|P|0.78|0.13|0.09|
|==C==|0.32|0.54|0.14|
|D|0.81|0.09|==0.10==|
|E|0.93|0.04|==0.03==|
|Q|0.81|0.10|==0.09==|
|N|0.82|0.10|==0.08==|
|==L==|0.41|0.49|0.10|
|==I==|0.39|0.47|0.14|
|==V==|0.40|0.50|0.10|
|M|0.44|0.20|0.36|
|==F==|0.42|0.42|0.16|
|Y|0.67|0.20|0.13|
|==W==|0.49|0.44|0.07|
|K|0.93|0.02|==0.05==|
|R|0.84|0.05|==0.11==|
|H|0.66|0.19|0.15|

==C==: Buried Residues
V; L; I; F; W; C

==0.01==: Exposed Residues
D; E; N; Q; K; R

## Protein Unfolding

> - An unfolded protein has lost its tertiary/quaternary structure and its biological activity
> - Depending on the experimental conditions an unfolded protein may be stable or it may aggregate/precipitate

> ***Treatments that unfold proteins***:
> **High temperature**: many proteins unfold at 50-90 º C
> **pH**: some proteins will unfold when they gain a large net charge
> **Denaturants**: proteins unfold in detergents like sodium dodecyl sulfate (SDS) or in denaturants like urea and guanidine
> Unfolding curves are **sigmoidal**

|![Unfolding curves](https://www.researchgate.net/profile/Deepak-Kumar-25/post/Why-there-is-difference-in-delta-G-H20-of-a-protein-calculated-by-CD-and-fluorescence-spectroscopy/attachment/5de21914cfe4a777d4f5e9d5/AS%3A830838086324224%401575098644656/image/CD+vs+floure.tif)|
|:-:|
|[© Agnishwar Girigoswami](https://www.researchgate.net/post/Why-there-is-difference-in-delta-G-H20-of-a-protein-calculated-by-CD-and-fluorescence-spectroscopy)|

### Net Stability

> - Most proteins have a net stability of ~10 kcal/mol/protein
> Compare this to the individual contributions:
> - Hydrophobic effect ~ 1-2 kcal/mol/amino acid residue
> - Conformational Entropy ~1-2 kcal/mol/amino acid residue
> For a 200 residue protein these contributions can be 400 kcal/mol each but the overall stability will always be ~10 kcal/mol

### Mutational Studies of Protein Stability


|![Mutational Studies of Protein Stability](https://www.researchgate.net/publication/280033060/figure/fig2/AS:340154561646621@1458110575853/Ubiquilin-2-proteins-containing-ALS-mutation-have-increased-stability-and-they-slow.png)|
|:-:|
|[© Lydia Chang](https://www.researchgate.net/publication/280033060_Defective_Proteasome_Delivery_of_Polyubiquitinated_Proteins_by_Ubiquilin-2_Proteins_Containing_ALS_Mutations)|

> - Most mutations are destabilizing
>   - The ability to precisely predict the change in folding energy (ΔG) is not very good
> - Buried residues have a bigger effect than exposed residues


> [图] Predicting changes in the stability of proteins and protein complexes: a study of more than 1000 mutations. J Mol Biol. 2002 Jul 5;320(2):369-87.


## The Levinthal Paradox

> ***Does a protein find its native configuration by a random search?***
> Carbon-carbon bonds reorient each ~ 10-13 sec
> For a protein of n residues there are 32n
> conformations
> If n = 100, then 3^2n^ ~ 10^95^
> If 10^13^ conformations were sampled every second it would take 10^82^ seconds to sample them all…
> …but the age of the universe is only about 10^17^ sec!
> It is now known that proteins fold on time scales from milliseconds to minutes
> - Proteins fold via conserved, hierarchical pathways
> (This may explain the prevalence of the Greek Key motif)

PS:
**why 3^2n^**:
- 2: there are two tortion angles: φ and ψ
- n: n peptides
- 3: If each of these bond angles can be in one of three stable conformations[^wiki_lp]

[^wiki_lp]: [Wikipedia: Levinthal's paradox](https://en.wikipedia.org/wiki/Levinthal%27s_paradox)

### Diseases of protein misfolding

> Cystic Fibrosis: Alzheimer’s disease
> Mad Cow Disease: Retinitus pigmentosa

## Molecular Chaperones

|![The GroEL/GroES ATPase cycle](https://ars.els-cdn.com/content/image/1-s2.0-S0968000498011931-gr1.gif)|
|:-:|
|[© Alexandra Richardson, 1998](https://www.sciencedirect.com/science/article/pii/S0968000498011931?casa_token=7xufCboPWDgAAAAA:LMCj9M2w9u8jtBaqNG1NMgc6vulcOsdx8YSzyPY6kDFll-LMDRBoUUKDznaotVdSmzpGMOc)|

> "Heat shock proteins"
> Molecular Chaperones
> Chaperonins


> Chaperones assist protein folding in the cell
> - They bind unfolded chains
> - They use energy from ATP to bind and release protein chains in a repeating cycle
> - ATP turnover rate is slow 1 sec^-1^
> The GroEL/GroES chaperone form E. Coli is well understood



### Other Proteins
> Protein Disulfide Isomerase
> Peptidyl proline cis-trans isomerase

## Protein Degradation in the Cell

> There are two main pathways for protein degradation in eukaryotic cells
> ***Lysosomal Degradation***
> - The lysosome is a membrane-encapsulated organelle
> - It contains ~50 hydrolytic enzymes
> - Lysosomes have acidic pH
> - Has both selective and non-selective degradation processes
> ***Proteasomal Degradation***
> - The proteasome is a large protein machine (MW = 2x106)
> - It degrades proteins that have the small protein ubiquitin attached to them
> - Ubiquitination rate (degradation rate) is determined by the N-terminal residue
> - Ubiquitin (76 residues) is the most highly conserved protein known
