---
toc: true
url: tulane_biochem_13
covercopy: © Karobben
priority: 10000
date: 2021-10-22 08:39:32
title: "Nuclear"
ytitle: "核酸|生化 13| 杜兰"
description: "Class notes of Graduate Biochemistry in Tulane"
excerpt: "Let's explore the function of proteins"
tags: [Classes, Biochemistry, Tulane Classes]
category: [Notes, Class, Tulane, Biochemistry]
cover: ""
thumbnail: "https://static-s.aa-cdn.net/img/ios/1535372364/8bea5e7d0cecf8c51488133d2c6a4f48?v=1"
---

## The central Dogma of Molecular Biology

### DAN
### Puring; Pyrimidine

Nucleotides
> ==Exam: Nucleotide synthesis==

## ==Biosynthesis of the DNA & RAN purines==
AMP synthesis requires GTP hydrolysis and aspartate
GPM syntehsis requries ATP and glutamine

## ==Biosynthesis of the DNA & RAN Pyrimidine==

### Ribonucleotide reductases reduction
Tyrasine radical

### Ribonucleotide ruduction is highly regulated

## FdU and Methotrexate are anticancer compounds thath disrupt dTMP Biosynthehsis
Cancer require large amount of dTTP

### Purines Can Also Be Salvaged

Unlike de novo purine syntehsis, purine salvage is NOT conserved across species

### Animal purin catabolism
PRPP synthesis

Purube ubcreasm nuroue → = =

## Watson & Crick

--
## DNA polymerases

Some DNA polymerase have 3' → 5' syntehsis

### Inhibitors of DNA synthesis and their specificities

AZT and DDI: reverse transcriptase, chain terminate

nuceoside analogues of guanosine: phosphorylated by a viral kinase

### Polymerase processivity Enhanced by a Clamp

β clamp in bacteria
PCNA in eukayotes
loaded onto dsDNA on calmp

Clamp loader couples ATP Hydrolysis to Clamp Loading

Sliding clamp + ATP: open and bind DNA; ATP → ADP, DNA releases

### DNA ligases

Eukaryotic DAN ligase use ATP rather than NADP

### initiation od DNA replication in E. coli

origin region (OriC) ~245bp

Duplex unwinding by DNA helicase
- Exposed ssDNA allows SSB and helic ...


Initiation: OriC-DnaA intercation → Entry of SSB and DnaB (hexlicase) → ENtry of promase → RNA primers are synthesized and the elongation complexes for 2 DNA replication forks are assembled

Elongation:

## Prokaryotic DNA replication


### A typical DNA replicsome
synthesis of leading strand are coupled with lagging strand by clamploader complex.

### The e.coli Pol III holoenzyme ...

Pols enyzme subunit factors

### Eukaryotic DNA replication:

Pro: no cell cycle and start every time

Eu: onece anc only once in S phase; highly regulated

G1: Preparing
Checkpoint: end of the S phase

Time is variatd

### Eukaryotic chromaomse have multiple origins..

Pro: single initial site
Euk: multiple initial site

### Eukaryotic replication initiation is hihgly oreder.. well regulated and complex...
Helicase (Mcm2-7)
Sld2/3 phosphorylated by CDK


### Super Current Biochemical Reconstitution of Regulated replication origin firing

### Termination in prokaryotes

Ter sites, 2 converging new ds circular DNA molecules separated from each other by recombination and with assistance of

### Cell are constantly exposed to agents that can cause damage to their genome

Repair, tolerant, destroy the cell

### ==Types of mutation==
Transition: AT → G/C
Transversion: A/T → C/G
Frameshift: GGGGGG → GGGGG

## Base alteration
Oxidation:
8-oxodG(antl)-dC(antf)
8-oxodG(synl)-dA(antf)

The rotation of the bound which cause the shift of the side chain and attribute difference hydrogen bonds. 3 to 2, C to A

Alkylation; Cross-links

## Exogenous Damage: Exposure to ultraviolet radiation

Absorbed by adjacent pyrimidine bind and form double bond and cause mutation or lethal during replication/transcription


### Repair

1. Base Excision Repair
  - Single Nucleotide
2. Nucleotide Excision Repair
  - oligonucleotide
3. Mismatch Excision Repair
  - mismatched base

## Base Excision Repair
- how recognize
- AP site
- Donw string/ leading string doesn't matter
- ligates involved


## Nucleotide excission repair (NER)
- recognize Enzymes (XPC)
- recruit other factors
- unwinding, mutation is exposed
- Some protein was recruited and mistake was cut / rescued.

Recognition → resynthesis

## Xeroderma pigmentosum (XP)
Accelerated skin cancer; mortality likely due to metastatic melanoma or squamous cell carcinoma

## MMR pathway - E. Coli

MutS, MutL, MutH, ATP
MutH: cut the strand (down/leading)

methelation is critical here


---


## RNA polymerases

- Transcription using DNA to RAN
  - Bacterial: ON enzyme for all of RNA synthesis
  - Eukaryotic: At leased Three I II III

- Strangs 1:
  - Template;
  - Non-coding;
  - Anti-sense:
  - Completed strand

- Strands 2:
  - Sense strand: coding strange of the DNA
  - Non-completed strand

- Bacterial promoter consensus sequences
  - UP Element; -35; -10; RNA stra +1
  - -35; -10: promoters (similar to regulatories in Eukaryotic.)

Expression controlled by interactions of promoter elements with RNA polymerase and specific repressor/Activators


### Bacterial RNAP
- Holoenyme versis core enzyme
- Core subunits: α₂ββ'ω (400kDa)
- Holoenzyme: core subunits plus factors
- ...


Five σ factirs in E. coli gene expression

- Exp: (not important)
  - σ 70: Most genes
  - σ 32: Heat schock proteins
- Trascription Elongation
  - RANP trancks along the DNA template, ==synthesizes mRNA in te 5' to 3' direction== and unwinds and rewinds the DNA as it reads.
  - ==Transcription elongation causing DNA supercoiling, relaxed by topisomerases==
- Termination
  - Transcription Terminattion
    - A series of 4 - 10 base
    - Sequence depended: CGGCGCTTTTTT (CG rich region; AT rich region)
  - ρ factor-dependent
    - The amino-termina domain of ρ factor binds to hte RNA sequence of ρ utilization site
    - the carboxy-terminal domain of ρ hexameric ring closes
    - Ring closure propels ρ moving close to RNAP
    - ρ dis-engages RNA and RNAP from DNA

### Eukaryotic Transcriotion

- Three types of Nuclears RNAPs
  - I: rRNA (28S, 18S)
  - ==II==: mRNA; snRNAs; miRNAs
  - III: tRNAs; 5S rRNA; snRNA U6 7S RNA; other stable short RNAs
- Transcription
  - Subunit compositions of nueclear RNAPs
    - β'; β; αI; αII; ω
    - All yeast RNAP has five subunits similar to Bacteria
  - RNAP II
    - Yeast RNAP II strcture resembling bacterial RNAPs
      - A vrab claw-like strcuture
  - RNA biosynthesis
    - The clamp of the Rpb1 subunit moves down to trap DAN between the two claw
    - Unwind DNA at the active site
    - Wall of Rpb2 kinks the template by 90° out of the active site
    - One vase of the template points at the active site
    - This base is paired with the ribonucleotide
  - RNA translocation
  - Transcription promoters
    - Class I, II, II to RAN poly I, II, III
  - ==Mammalian Gene and proximal promoter (Class II)==
    - Proximal region (-200 ~ -30), within 200 bp of transcription start; similar to the -10; -35 in the bacteria's
    1. ==TATA Box (TATAAA)==
      - Recruits TATA binding protein ...
    2. Initiator region (Inr)
    3. Other activation elements
  - ==EPromoter and Activator proteins==
    1. ==Transcriptional Activators==: DNA binding domain; activation domain;
    2. Activator functions: Chromatin remodeling (acetylation-HAT); Mediator facilitates PIC assembly
  - Repressor
    - Bind UAS.enhancers → displace activators
    - Prevent mediator → no PIC formation
    - Attracts HDACs and HMTs → heterochromatin
  - PIC Assmbled
    - mediator facilitates TBP and TFIIB binding promoter
    - Other basal TFs and Pol II bind
    - phosphorylation of the CTD of Pol II by TFIIH → Transcription initiation
  - Six transcription factors for clas II promoters (Not important)
  - ==Phospho-code on RNAP II CTD
    - TFIH: Phosphate the Ser5 would initiated the transcription activity
    - P-TEFb: phospholate Ser2 to maintain the elongation state
  - Non-canonical functions of transcription
    - CENP-A defines centromeric chromatin
  - ==RNA..==
  - ==ChiP-Seq: Determine protein distributed on chromatins==
  - RNA-Seq: Expression profiles


### ==Transcription inhibition as a therapeutic target for cancer==

1. mRANs of many oncogenes
2. Transformed → RNA-directed agents: different sensitivity.

- Alpha-amanitin: from mushroom
  - directed binding to RNAP, adn block the bridge to repress the conformation change


## Bacterial Transcription

## Eukaryotic transcription

## RNAP II CTD phospho-code

## Non-canonical functions of transcription

## Assays to study transcription

---

## posttranscriptional Modification of Eukaryotic RNAs

- Role of 5' cap
  - Ribosomal recogintion during translation
- Cap structure
 -  7-methylguanosin (m⁷G)
 - joined to the mRNA first nucleotide
 - Via a 5'-5' tri-P brifge
- Involved Enzymes
  - RNA riphosphatase removing the γ-P of the mRNA's 5' site
  - mRNA guanlyltransferase (a capping enzyme) adding GMP. (Take GMP to the site)
  - Guanine-7-methyltranserase (second capping enzyme) methylating guanine
  - Capping enzymes **bind to RNAP-II**, which will switch RAN synthesis initiation to elongation ()

### posttransicriptional Modification of Eukaryotic RANs

- Poly A tails; ~250 nt
  - Cleavage and polyadenylation specificity factor (CPASF), cleaving up to 35 nt past the **AAUAAA** sequence
- Poly A polymerase generate poly A tail using ATP. (termination signals given to RNA-polymerase)
- CPSF binds to RNAP-II, coupling polyadenylation to transcription termination.
- Poly A tail binds to poly-binding protein, protecting from degradation, ==increasing mRNA stability==.

### Exons and Introns
- Precurso mRANs (pre-mRNAs) are processsed by the excision of introns and the splicing (joing) of exons.

- Exon splicing in Two-Stage Reactions
  - Invariant Sequences for splice junction
    - GU at the intron's 5'
    - AG at teh intron's 3
      - (graph), the number under the NT means the ration you are supposed to see them on the Intron
    - A branch point near the 3' splice site
    - Free G, not paire to the intron...
    1. A2'-5' P-diester bind between the tinron A (OH²') and the intron at teh 5' splice site, forming a "lariat" structure.
    2. The Exon 1 OH³' group at the 5' splice site from a 3'-5' P-diester bind with the Exon 2' at the 3' splice site, releasing the intron with the free OH³' group.
    3. The intron keeps the  lariat structure.
    - Note: Splicing proceeds w/o free energy lose, Cleavage of one P-diseter bond and formation of a new bond.

### ==Spliceosome-aided RNA splicing==

Splicing could be focilited by the protein Splicesome.60S spliceosome particle containing five small nuclear RNAs + Ribonucleoproteins (U 1~6, no 3)
1. U1 recognizes the 5' splice junction
2. U2 recognizes the branch point (intron A)
3. The binding of U4-U5-U6 forms spliceosome.
4. RAN cleavage at the 5' splice site
5. RNA cleavage at the 3' splice site
6. Intron ..

### Self-Splicing RNA
- Group II Intron
  - Exson 1 and Exon II aligned together
  - the giant rondant was splicesd by themselves
- Group I intron:
  - not Intron A to initiated the reaction
  - G , GTP, GDP, GMP... as the starting site
  1. The 3'-OH of ...
  2.
  3.

### ==Summary of spliceosome-aided Splicing and Self-Splicing==

- In goup I introns, aguanosine cofactor (G) tat is not part of the RNA chain associates with the active site. The 3'-hydroxyl group of this G attacks the 5' splice sit;
  - The reaction is similar to those involving the 2 hydroxyl groups of branch sites as in group II introns and RNA inrons spliced in spliceosomes.
- The subsequent trans-e...

### ==Interactions of RNAP-II with capping enzyme, splicing and CPSF==
Figure 16-5

### Aminoacyl-tRAN Synthetase

- Attach amino acids to tRNAs
- Two step reactions
  - Amino acid activation by the reaction with ATP
  (Aminoacyl-tRNA syntheiase)
  - Formation of an aminoacyl-tRNA

### Aminoacyl-tRAN SYnthetases
- The synthetase enzymes have an elongated shape.
- Binding the anticodon of tRNA near the one end of the enzyme.
- Binding the A.a. acceptor stem of tRNA near th other end

### Proofreading by Aminoacyl-tRAN Synthetase

Binding to the amino acid substrate pocket of tRNA Synthetase that consists of the zinc ion.L-Group: Megan, Ryan, Ka
- Threonine binds at this pocket b/c it can coordinate Zn^2+^ using it NH~2~ and OH groups

- tRNA^thr^ is tRNA for threonin.
  - Attaching ThR to tRNA^thr^ makes Thr-tRNA^thr^ (tRNA is correctly charged)
  - Attaching Ser to tRNA^thr^ makes Ser-tRNA^thr^ (tRAN is mischarged)
   
