---
toc: true
url: tulane_biochem_16
covercopy: <a href="https://www.sciencedirect.com/science/article/abs/pii/S1877117318300942">© Chang HoLee, 2018</a>
priority: 10000
date: 2021-11-27 12:39:32
title: "Catalytic RNAs: Splicing and Translation"
ytitle: "RNA 酶|生化 16| 杜兰"
description: "Class notes of Graduate Biochemistry in Tulane"
excerpt: "Let's explore the function of proteins"
tags: [Classes, Biochemistry, Tulane Classes]
category: [Notes, Class, Tulane, Biochemistry]
cover: "https://ars.els-cdn.com/content/image/1-s2.0-S1877117318300942-f04-01-9780128162354.jpg"
thumbnail: "https://static-s.aa-cdn.net/img/ios/1535372364/8bea5e7d0cecf8c51488133d2c6a4f48?v=1"
---

## Posttranscriptional Modification of Eukaryotic RNAs

Role of the 5’ cap
- Ribosomal recognition during translation

Cap structure
- 7-methylguanosine (m7G)
- Joined to the mRNA first nucleotide
- Via a 5’-5’ tri-P bridge

Involved enzymes
- RNA triphosphatase removing the γ-P of the mRNA’s 5’ site
- mRNA guanylyltransferase (a capping enzyme) adding GMP
- Guanine-7-methyltransferase (a capping enzyme) methylating guanine
- Capping enzymes bind to RNAP-II, which will switch RNA synthesis initiation to elongation

Poly(A) tails
- ~250 nt (~80 nt in yeast)
- Cleavage and polyadenylation specificity factor (CPSF), cleaving up to 35 nt past the AAUAAA sequence
- Poly(A) polymerase generate poly(A) tail using ATP.
- CPSF binds to RNAP-II, coupling polyadenylation to transcription termination.
- Poly(A) tail binds to poly(A)-binding protein, protecting from degradation, ==increasing mRNA stability==.


### Exons and Introns

Precursor mRNAs (pre-mRNAs) are processed by the excision
of introns and the splicing (joining) of exons

### Exon Splicing in Two-Stage Reactions

|![Mechanisms of recursive splicing](https://www.mdpi.com/molecules/molecules-26-02263/article_deploy/html/images/molecules-26-02263-g001.png)|
|:-:|
|[© Zhichao Tang, 2021](https://www.nature.com/articles/nature14524)|

Invariant sequences for splice junction:
- GU at the intron’s 5’
- AG at the intron’s 3’
- A branch point (Intron A) near the 3’ splice site
- Free guanosine, GMP, GDP, or GTP not part of the intron

1. A 2’-5’ P-diester bond between the intron A (OH2’) and the intron at the 5’ splice site, forming a "lariat" structure.
2. The Exon 1 OH3’ group at the 5’ splice site form a 3’-5’ P- diester bond with the Exon 2 at the 3’ splice site, releasing the intron with the free OH3’ group.
3. The intron keeps the lariat structure. Note: Splicing proceeds w/o free energy lose. Cleavage of one P-diester bond and formation of a new bond.

Note: Splicing proceeds w/o free energy lose. Cleavage of one P-diester bond and formation of a new bond.

### Spliceosome-aided RNA splicing

|  |    |
| :------------- | :-------------: |
| 60S spliceosome particle containing five small nuclear RNAs + ribonucleoproteins (U1, U2, U4, U5 and U6) <li>1.  U1 recognizes the 5’ splice junction <li>2.  U2 recognizes the branch point (intron A)<li>3.  The binding of U4-U5-U6 forms spliceosome.<li>4.  RNA cleavage at the 5’ splice site<li>5.  RNA cleavage at the 3’ splice site<li>6.  Intron degregradation| ![](http://www.zo.utexas.edu/faculty/sjasper/images/17.10.gif)<br>[© utexas.edu](http://www.bio.utexas.edu/faculty/sjasper/bio212/proteins.html)|


### Self-Splicing RNA

|![Group II intron](https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/attachment/2118908092/2086560908/gr1.jpg)|
|:-:|
|[© Chen Zhao, 2017](https://www.cell.com/trends/biochemical-sciences/fulltext/S0968-0004(17)30069-5)|
Group II intron
1. A bulged adenosine in Domain 6 attacks the 5’ splice site, forming a lariat intermediate (a 2’-5’ linkage)
2. The 5’ and 3’ exons are ligated and the lariat intron is released


|![Group I intron](https://ars.els-cdn.com/content/image/1-s2.0-S1877117318300942-f04-01-9780128162354.jpg)|
|:-:|
|[© Chang HoLee, 2018](https://www.sciencedirect.com/science/article/abs/pii/S1877117318300942)|

Group I intron: In the presence of guanosine, GMP, GDP, or GTP, the pre-rRNA is self- splicing.
1. The 3’-OH of guanosine attacks the 5’ splice site, forming a P-diester bond with the intron.
2. The 3’-OH of the exon 1’s 5’ end attacks the 3’ splice site of exon 2, joining the two exons.
3. The 3’-OH of the intron attacks the nucleotide 15 residues from the intron’s 5’ end, yielding a cyclic form.

### Summary of Spliceosome- aided Splicing and Self-Splicing

In group I introns, a guanosine cofactor (G) that is not part of the RNA chain associates with the active site. The 3’-hydroxyl group of this G attacks the 5’ splice site;
- the reaction is similar to those involving the 2’ hydroxyl groups of branch sites as in group II introns and RNA introns spliced in spliceosomes.

The subsequent trans-esterification that links the 5’ and 3’ exons is similar in all three splicing mechanisms.


### Interactions of RNAP-II with Capping enzyme, splicing and CPSF

![Interactions of RNAP-II with Capping enzyme, splicing and CPSF](https://z3.ax1x.com/2021/11/28/onMqqU.png)

### Aminoacyl-tRNA Synthetases

|![Aminoacyl-tRNA Synthetases](https://biology-forums.com/gallery/33_24_06_11_5_00_34.jpeg)|
|:-:|
|[© Bioforums](https://biology-forums.com/index.php?action=gallery;sa=view;id=430)|
- Attach amino acids to tRNAs
- Two step reactions
	- Amino acid activation by the reaction with ATP
	- Formation of an aminoacyl-tRNA

- The synthetase enzymes have an elongated shape.
-	Binding the anticodon of tRNA near the one end of the enzyme.
-	Binding the a.a. acceptor stem of tRNA near the other end.

The bases of the anticodon are unstacked and splay outward, binding in separate recognition pockets of the Synthetase.

### Proofreading by Aminoacyl-tRNA Synthetase (tRNA^thr^ Synthetase)

|![Justin C. Morse](https://www.pnas.org/content/pnas/117/7/3610/F1.large.jpg?width=800&height=600&carousel=1)|
|:-:|
|[© Justin C. Morse](https://www.pnas.org/content/117/7/3610)|

Binding to the amino acid substrate pocket of tRNA^thr^ Synthetase that consists of the zinc ion
- Threonine (attached at the 3’ of tRNA) binds at this pocket b/c it can coordinate Zn^2+^ using its NH~2~ and OH groups
- Serine binds b/c it can coordinate Zn^2+^ using its NH~2~ and OH groups!!
- But valine cannot bind b/c it lacks the OH group.
- tRNA^thr^ is tRNA for threonine.
 	- Attaching Thr to tRNA^thr^ makes Thr-tRNA^thr^ (tRNA is correctly charged)
	- Attaching Ser to tRNAthr makes Ser-tRNA^thr^ (tRNA is mischarged!)
	- Ser-tRNA^thr^ incubates with Thr-tRNA synthetase------>
 	- ==Rapid breakdown: Ser and free tRNA^thr^ !!==
	- Thr- tRNA^thr^ incubates with Thr-tRNA synthetase------>
		- No breakdown!

- The flexible aa acceptor stem of tRNAthr can move the attached serine residue between the activation/catalytic site and the editing site.
- Since serine fits well at the editing site, the residue is removed from tRNAthr .
- Threonine is too large to fit at the editing site, the residue remains on tRNAthr .

### Ribosomes

| Prokaryotic 70S    | Eukaryotic (80S)    |
| :------------- | :------------- |
| 30S subunit: 16S rRNA, 21 proteins ( bond mRNA (red))| 40S subunit: 18S rRNA, 33 proteins|
| 50S subunit: 5S and 23S rRNAs and 31 proteins [bound to three sites: exit (E), peptidyl (P), and aminoacyl (A)]| 60S subunit: 5S, 5.8S and 28S rRNAs and 49 proteins|

|Summary of the Composition and Mass of Ribosomes in Bacteria and Eukaryotes|![](https://z3.ax1x.com/2021/11/28/onQ6Y9.png)|
|:-|:-:|


### Translation (Protein biosynthesis)

|![](https://z3.ax1x.com/2021/11/28/onl9Yj.png)|
|:-:|
||

- Translation (Protein biosynthesis) Protein synthesis proceeds from N-term to C-term by a peptidyl transferase activity
- Chain elongation occurs by linking the growing peptide to the incoming tRNA’s aa.

- The growing peptide is transferred from the P site to the incoming aa-tRNA in the A site, resulting in the formation of a peptidyl-tRNA with one more aa residue in the A site.
- The peptidyl-tRNA is translocated to the P site to have an empty A site for a new aa-tRNA.
- The uncharged tRNA in the P site moves to the E site (not shown) for
exiting.



|![Translation and Tanscription](https://z3.ax1x.com/2021/11/28/onll11.png)|
|:-:|
|© 1970 American Association for the Advancement of Science. Miller, O. L. et al. Visualization of bacterial genes in action. Science 169, 392–395 (1970).|

Ribosomes read mRNA in the 5’ → 3’ direction.
Active translation occurs on polysomes
- Multiple ribosomes can bind to a single mRNA transcript
- Initiation
	- Initiation codon(AUG) in mRNA, N-formylmethionyl-tRNA, 30S rRNA, 50S rRNA, Initiation factors (IF-1, IF-2, IF-3) GTP, Mg
- Elongation
	- Functional 70S ribosome (initiation complex), aminoacyl-tRNAs specified by codons, elongation factors (EF-Tu, EF-Ts, EF-G), GTP, Mg
	- Peptidyl transferase activity not by rProtein, but by rRNA
-	Termination and release
	-	Termination codon in mRNA, polypeptide release factors (RF1, RF2, RF3), GTP

### Translation - Initiation

- Almost half of E. coli proteins begin with the uncommon N-formylated Met (fMet).
- The fMet has an amide bond, which can grow only to the C-term side.
- fMet is removed by deformylation, followed by the removal of resulting Met sometimes.

### Selectivity for the Translation Initiation Site

In E. coli, base pairing interactions of an **mRNA Shine-Dalgarno sequence** with the **16S rRNA** allows ribosome to select the proper initiation codon.

![Selectivity for the Translation Initiation Site](https://z3.ax1x.com/2021/11/28/on1E8A.png)

### Prokaryotic Translation - Initiation
|![Initiation](https://www.onlinebiologynotes.com/wp-content/uploads/2017/07/initiation.png)|
|:-:|
|[© Gaurab Karki, 2017](https://www.onlinebiologynotes.com/translation-in-prokaryotes/)|
1. IF-3 binds to the 30S subunit to dissociate the inactive 70S ribosome into 30S & 50S
2. mRNA and IF-2 forms a complex with GTP and fMet- tRNAfmet along with IF-1 bind to the 30S subunit.
3. IF-1 and IF-3 are released and the 50S subunit binds to the complex, stimulating GTP hydrolysis by IF-2. It is followed by IF-2 release.

“fMet-tRNA bound at the P site of 50S in the 70S initiation complex.”

### Prokaryotic Translation - Elongation

|![Elongation](https://www.onlinebiologynotes.com/wp-content/uploads/2017/07/elongation.png)|
|:-:|
|[© Gaurab Karki, 2017](https://www.onlinebiologynotes.com/translation-in-prokaryotes/)|
1. Decoding: Ribosome binds an aa-tRNA, whose anticodon is complementary to the mRNA codon in the A site.
2. Peptide bond formation: The peptidyl group in the P site is transferred to the aminoacyl group in the A site
3. Translocation: EF-GGTP binding to A translocates tRNAs from “P and A” to “E and P”, releasing the free tRNA at E.
4. GTP hydrolysis releases EF-GGDP

### Prokaryotic Translation – Termination

|![Termination](https://www.onlinebiologynotes.com/wp-content/uploads/2017/07/termination.jpg)|
|:-:|
|[© Gaurab Karki, 2017](https://www.onlinebiologynotes.com/translation-in-prokaryotes/)|
In E. coli, the stop codons (UAA, UGA, UAG) have no
corresponding tRNAs, recognized by release factors.
- RF-1 recognizes UAA and UAG.
- RF-2 recognizes UAA and UGA.

1. RF-1 (or RF-2) recognize corresponding
codons at the A site.
2. (1) The peptidyl group of tRNA at the P site is transferred to water, releasing the polypeptide.
(2) RF-3GTP binds to ribosome.
3. GTP hydrolysis releases RF-1 (or RF-2).
4. RF-3GDP is replaced by ribosome recycling factor (RRF), followed by EF-GGTP
5. Upon GTP hydrolysis, RRF, EF-GGDP, mRNA and tRNA at the P site dissociate, yielding an inactive 70S ribosome.



---
### Inhibitors of DNA synthesis and their specificities

AZT and DDI: reverse transcriptase, chain terminate

nuceoside analogues of guanosine: phosphorylated by a viral kinase

### Polymerase processivity Enhanced by a Clamp

β clamp in bacteria
PCNA in eukayotes
loaded onto dsDNA on calmp

Clamp loader couples ATP Hydrolysis to Clamp Loading

Sliding clamp + ATP: open and bind DNA; ATP → ADP, DNA releases


### Biological advantages of DNA

1. Provides stable, yet mutable storage of genetic information. Metabolic stability and availability of repair mechanisms ensure long-term survival under a variety of physiological conditions.
2. Serves as a template for accurate and adaptable replication of genomes. This biological role ensures the transmission of important physiological functions for multiple generations.
3. Serves as a template for the expression and regulation of genetic information. This biological role facilitates adaptation of genomes to changing physiological environments.
4. Amenable to diversification (i.e., can evolve). Mutation and genetic recombination (DNA rearrangements) are the two major driving forces in the evolution of genomic diversity and adaptation in nature.

---
### DNA ligases

Eukaryotic DAN ligase use ATP rather than NADP

### initiation od DNA replication in E. coli

origin region (OriC) ~245bp

Duplex unwinding by DNA helicase
- Exposed ssDNA allows SSB and helic ...


Initiation: OriC-DnaA intercation → Entry of SSB and DnaB (hexlicase) → ENtry of promase → RNA primers are synthesized and the elongation complexes for 2 DNA replication forks are assembled

Elongation:

## Prokaryotic DNA replication


### A typical DNA replicsome
synthesis of leading strand are coupled with lagging strand by clamploader complex.

### The e.coli Pol III holoenzyme ...

Pols enyzme subunit factors

### Eukaryotic DNA replication:

Pro: no cell cycle and start every time

Eu: onece anc only once in S phase; highly regulated

G1: Preparing
Checkpoint: end of the S phase

Time is variatd

### Eukaryotic chromaomse have multiple origins..

Pro: single initial site
Euk: multiple initial site

### Eukaryotic replication initiation is hihgly oreder.. well regulated and complex...
Helicase (Mcm2-7)
Sld2/3 phosphorylated by CDK


### Super Current Biochemical Reconstitution of Regulated replication origin firing

### Termination in prokaryotes

Ter sites, 2 converging new ds circular DNA molecules separated from each other by recombination and with assistance of

### Cell are constantly exposed to agents that can cause damage to their genome

Repair, tolerant, destroy the cell

### ==Types of mutation==
Transition: AT → G/C
Transversion: A/T → C/G
Frameshift: GGGGGG → GGGGG

## Base alteration
Oxidation:
8-oxodG(antl)-dC(antf)
8-oxodG(synl)-dA(antf)

The rotation of the bound which cause the shift of the side chain and attribute difference hydrogen bonds. 3 to 2, C to A

Alkylation; Cross-links

## Exogenous Damage: Exposure to ultraviolet radiation

Absorbed by adjacent pyrimidine bind and form double bond and cause mutation or lethal during replication/transcription


### Repair

1. Base Excision Repair
  - Single Nucleotide
2. Nucleotide Excision Repair
  - oligonucleotide
3. Mismatch Excision Repair
  - mismatched base

## Base Excision Repair
- how recognize
- AP site
- Donw string/ leading string doesn't matter
- ligates involved


## Nucleotide excission repair (NER)
- recognize Enzymes (XPC)
- recruit other factors
- unwinding, mutation is exposed
- Some protein was recruited and mistake was cut / rescued.

Recognition → resynthesis

## Xeroderma pigmentosum (XP)
Accelerated skin cancer; mortality likely due to metastatic melanoma or squamous cell carcinoma

## MMR pathway - E. Coli

MutS, MutL, MutH, ATP
MutH: cut the strand (down/leading)

methelation is critical here


---


## RNA polymerases

- Transcription using DNA to RAN
  - Bacterial: ON enzyme for all of RNA synthesis
  - Eukaryotic: At leased Three I II III

- Strangs 1:
  - Template;
  - Non-coding;
  - Anti-sense:
  - Completed strand

- Strands 2:
  - Sense strand: coding strange of the DNA
  - Non-completed strand

- Bacterial promoter consensus sequences
  - UP Element; -35; -10; RNA stra +1
  - -35; -10: promoters (similar to regulatories in Eukaryotic.)

Expression controlled by interactions of promoter elements with RNA polymerase and specific repressor/Activators


### Bacterial RNAP
- Holoenyme versis core enzyme
- Core subunits: α₂ββ'ω (400kDa)
- Holoenzyme: core subunits plus factors
- ...


Five σ factirs in E. coli gene expression

- Exp: (not important)
  - σ 70: Most genes
  - σ 32: Heat schock proteins
- Trascription Elongation
  - RANP trancks along the DNA template, ==synthesizes mRNA in te 5' to 3' direction== and unwinds and rewinds the DNA as it reads.
  - ==Transcription elongation causing DNA supercoiling, relaxed by topisomerases==
- Termination
  - Transcription Terminattion
    - A series of 4 - 10 base
    - Sequence depended: CGGCGCTTTTTT (CG rich region; AT rich region)
  - ρ factor-dependent
    - The amino-termina domain of ρ factor binds to hte RNA sequence of ρ utilization site
    - the carboxy-terminal domain of ρ hexameric ring closes
    - Ring closure propels ρ moving close to RNAP
    - ρ dis-engages RNA and RNAP from DNA

### Eukaryotic Transcriotion

- Three types of Nuclears RNAPs
  - I: rRNA (28S, 18S)
  - ==II==: mRNA; snRNAs; miRNAs
  - III: tRNAs; 5S rRNA; snRNA U6 7S RNA; other stable short RNAs
- Transcription
  - Subunit compositions of nueclear RNAPs
    - β'; β; αI; αII; ω
    - All yeast RNAP has five subunits similar to Bacteria
  - RNAP II
    - Yeast RNAP II strcture resembling bacterial RNAPs
      - A vrab claw-like strcuture
  - RNA biosynthesis
    - The clamp of the Rpb1 subunit moves down to trap DAN between the two claw
    - Unwind DNA at the active site
    - Wall of Rpb2 kinks the template by 90° out of the active site
    - One vase of the template points at the active site
    - This base is paired with the ribonucleotide
  - RNA translocation
  - Transcription promoters
    - Class I, II, II to RAN poly I, II, III
  - ==Mammalian Gene and proximal promoter (Class II)==
    - Proximal region (-200 ~ -30), within 200 bp of transcription start; similar to the -10; -35 in the bacteria's
    1. ==TATA Box (TATAAA)==
      - Recruits TATA binding protein ...
    2. Initiator region (Inr)
    3. Other activation elements
  - ==EPromoter and Activator proteins==
    1. ==Transcriptional Activators==: DNA binding domain; activation domain;
    2. Activator functions: Chromatin remodeling (acetylation-HAT); Mediator facilitates PIC assembly
  - Repressor
    - Bind UAS.enhancers → displace activators
    - Prevent mediator → no PIC formation
    - Attracts HDACs and HMTs → heterochromatin
  - PIC Assmbled
    - mediator facilitates TBP and TFIIB binding promoter
    - Other basal TFs and Pol II bind
    - phosphorylation of the CTD of Pol II by TFIIH → Transcription initiation
  - Six transcription factors for clas II promoters (Not important)
  - ==Phospho-code on RNAP II CTD
    - TFIH: Phosphate the Ser5 would initiated the transcription activity
    - P-TEFb: phospholate Ser2 to maintain the elongation state
  - Non-canonical functions of transcription
    - CENP-A defines centromeric chromatin
  - ==RNA..==
  - ==ChiP-Seq: Determine protein distributed on chromatins==
  - RNA-Seq: Expression profiles


### ==Transcription inhibition as a therapeutic target for cancer==

1. mRANs of many oncogenes
2. Transformed → RNA-directed agents: different sensitivity.

- Alpha-amanitin: from mushroom
  - directed binding to RNAP, adn block the bridge to repress the conformation change


## Bacterial Transcription

## Eukaryotic transcription

## RNAP II CTD phospho-code

## Non-canonical functions of transcription

## Assays to study transcription

---

## posttranscriptional Modification of Eukaryotic RNAs

- Role of 5' cap
  - Ribosomal recogintion during translation
- Cap structure
 -  7-methylguanosin (m⁷G)
 - joined to the mRNA first nucleotide
 - Via a 5'-5' tri-P brifge
- Involved Enzymes
  - RNA riphosphatase removing the γ-P of the mRNA's 5' site
  - mRNA guanlyltransferase (a capping enzyme) adding GMP. (Take GMP to the site)
  - Guanine-7-methyltranserase (second capping enzyme) methylating guanine
  - Capping enzymes **bind to RNAP-II**, which will switch RAN synthesis initiation to elongation ()

### posttransicriptional Modification of Eukaryotic RANs

- Poly A tails; ~250 nt
  - Cleavage and polyadenylation specificity factor (CPASF), cleaving up to 35 nt past the **AAUAAA** sequence
- Poly A polymerase generate poly A tail using ATP. (termination signals given to RNA-polymerase)
- CPSF binds to RNAP-II, coupling polyadenylation to transcription termination.
- Poly A tail binds to poly-binding protein, protecting from degradation, ==increasing mRNA stability==.

### Exons and Introns
- Precurso mRANs (pre-mRNAs) are processsed by the excision of introns and the splicing (joing) of exons.

- Exon splicing in Two-Stage Reactions
  - Invariant Sequences for splice junction
    - GU at the intron's 5'
    - AG at teh intron's 3
      - (graph), the number under the NT means the ration you are supposed to see them on the Intron
    - A branch point near the 3' splice site
    - Free G, not paire to the intron...
    1. A2'-5' P-diester bind between the tinron A (OH²') and the intron at teh 5' splice site, forming a "lariat" structure.
    2. The Exon 1 OH³' group at the 5' splice site from a 3'-5' P-diester bind with the Exon 2' at the 3' splice site, releasing the intron with the free OH³' group.
    3. The intron keeps the  lariat structure.
    - Note: Splicing proceeds w/o free energy lose, Cleavage of one P-diseter bond and formation of a new bond.

### ==Spliceosome-aided RNA splicing==

Splicing could be focilited by the protein Splicesome.60S spliceosome particle containing five small nuclear RNAs + Ribonucleoproteins (U 1~6, no 3)
1. U1 recognizes the 5' splice junction
2. U2 recognizes the branch point (intron A)
3. The binding of U4-U5-U6 forms spliceosome.
4. RAN cleavage at the 5' splice site
5. RNA cleavage at the 3' splice site
6. Intron ..

### Self-Splicing RNA
- Group II Intron
  - Exson 1 and Exon II aligned together
  - the giant rondant was splicesd by themselves
- Group I intron:
  - not Intron A to initiated the reaction
  - G , GTP, GDP, GMP... as the starting site
  1. The 3'-OH of ...
  2.
  3.

### ==Summary of spliceosome-aided Splicing and Self-Splicing==

- In goup I introns, aguanosine cofactor (G) tat is not part of the RNA chain associates with the active site. The 3'-hydroxyl group of this G attacks the 5' splice sit;
  - The reaction is similar to those involving the 2 hydroxyl groups of branch sites as in group II introns and RNA inrons spliced in spliceosomes.
- The subsequent trans-e...

### ==Interactions of RNAP-II with capping enzyme, splicing and CPSF==
Figure 16-5

### Aminoacyl-tRAN Synthetase

- Attach amino acids to tRNAs
- Two step reactions
  - Amino acid activation by the reaction with ATP
  (Aminoacyl-tRNA syntheiase)
  - Formation of an aminoacyl-tRNA

### Aminoacyl-tRAN SYnthetases
- The synthetase enzymes have an elongated shape.
- Binding the anticodon of tRNA near the one end of the enzyme.
- Binding the A.a. acceptor stem of tRNA near th other end

### Proofreading by Aminoacyl-tRAN Synthetase

Binding to the amino acid substrate pocket of tRNA Synthetase that consists of the zinc ion.L-Group: Megan, Ryan, Ka
- Threonine binds at this pocket b/c it can coordinate Zn^2+^ using it NH~2~ and OH groups

- tRNA^thr^ is tRNA for threonin.
  - Attaching ThR to tRNA^thr^ makes Thr-tRNA^thr^ (tRNA is correctly charged)
  - Attaching Ser to tRNA^thr^ makes Ser-tRNA^thr^ (tRAN is mischarged)




---

## Acetylation

- definition
- histone modification
1. Acetylation by HATs
2. Recognition by bromodomains
3. Deacetylation by lysine deacytelases (HDACs adn Sirtuins)
    - cofactors: Zinx, NAD
4. Methylation by histone methyltransefrases (HMTs)
    - The donor: S-adenosylmethionine (==SAM==)
    - ==SET== domains in SAM
5. Recognition by chromodomains and plant homeodomain (==PHD==) fingers
    - Aromatci cage: to recognize
    - Chroodomain-containing heterochromatin protein 1 (==HP1==) function: regulated transcription
6. Demethylation
    - Lysine-specific histone demethylases (==LSD1== and ==LSD2==)-FAD-dependent
    - Jumonji C-domain (==JMJC==) family members - F2a and α-ketoglutarate-dependent

## Chromatin structure

Chrmosome → chromatin fiber → Beads on a string DNA wound on nucleosomes → Histones + Doble helix

- ==Heteochromatin==: Highly condensed ==nonexpressing== DNA. (PS: Canot be transcripted because transcript machinary can't recognize and bind)
- ==Euchromatin==: Less condensed, Transcriptionally active DAN

H2A, H2B, H3, H4

### Histone Modification

Histone was circled in the center of the DNA, but with tail out of the stracture and cann't be visualized by cristal structure. But they are the ==key site for the Methylating or Acetylating==

acKL acetyl lysine
meR:
meK:
PS: phosphoryl serine
..

### Histone Code

Histone modification is the signals of transcription on/off.

### Histone Acetyltraseferases (HATs)

Histone + Me-CoA → CoA + Me-Histone

### Structure of the HAT domain from Tetrahymena thermophila

- The enzyme is deeply clefted
- The histone H3 peptide KSTGGK^14^APRK! and coenzyme A are bound at the deep cleft.

Catalysis appears to involve water-mediated proton extraction from the substrate lysine by a glutamic acid general base.


### Acetylated Lys recognition by Bromodomains

- Bromodomains specifically bind acetylated lysine residues on histones.
  - A deep hydrophobic pocket (==hole==) accommodates the acetyl-Lys side chain (PS, the hole recognize the acetyl-Lys specificity)

- TAF1: has two brpmodomains and dipart from each with ~25Å, seperated by 7 ~8 residues.
- A subunit of transcription factor TFIID (PDBid=1eqf)
- ...
- ...

### Role of TAF1 double Bromodomain
- TAF1 double bromodomain targets TFIID to promoters
- A refined proposal for transcription initiation
  1. A AHT-cotaining coactivator binds to upstream DAN binding protein (An activator)
  2. The HAT acetylates nearby histone tails
  3. TFIID is recruited to the site via the binding of the TAF1 double bromodomain to the acetyl-Lys residues of histones.
  4. followed by recruitment of other initiation factors and RNAP-II for transcription initiation.


### Histone Deacetylases (HDACs)

- 18 HDAC enzymes that deacetylate acetyl lysine substrates including histones.
  - Histone deacetylase ..
  - ...

Reaction mechanism:
    - Bound Zn^2+^ mediates the nucleophilic attack of water on acetylated lysine, forming a tetrahedral oxyanion intermidiate
    - The carbon-...

### Situins

NAD⁺-dependent reaction
1. nucleophilic addition to form a C1'-O-alkylamidate intermediate and free nicotinamide.
2. Teh 2'hydroxy group of the NAD⁺ ribose attacks the C1'-O-alkylamidate to form the 1',2'cyclic intermediate
3. The formation of deacetylasted lysine and 2'-O-acetyl-ADP ribose.

Thus, nicotinamide, a deacetylated lysine-containing histone, and 2'-O-acetyl-ADP

### Histone Methylation

- Methylation at Lys and Arg of ==H3 and H4 tends to silence== the genes, inducing heterochromatin formation
- However, ==trimetheylated Lys4 of H3 is associated with active genes==

(methylation can add one to 3 methyl in a time) (==give a exampel==)

- Histone methyltransferases (HMTs) use S-adenosylmethionine (SAM) as a methy donor
- The HMT enzymes have a SET domain containing the catalytic site for methylation.

### Structure of SET7/9 (PDB=1o9s)

SET7/9 mono-methylates Lys4 of H3
- SAM (SAH) adn the Lys4-containing peptide bind to oppsite sites of the protein
- The lys4 side chain is inserted through a narrow channel and positioned for methylation by SAM.

### Methylated Histone Recognition

- Chromodomains and plant homeodomain (PHD) fingers bind to mono, di- or tri-methylated Lys by an ==aromatic cage==.

1. Fits in the shape,
2. ==lots of electron to nutralize the positive charge of the methyl group???==

### Consequence of Methylated hisone Recognition

- The chromodomain heterochromatin protein 1 (HP1) bidns to methylated H3 Ly9, contributing to gene silencing,
- Bound HP1 recruitsthe HMT Suv39h, methylating nearby histone lysines (H3K9), thereby recruiting more HP1 and forming HP1 complexes.
- This mechanism explains how heterochromatin spread to silence neighboring genes.
- HP1 clust prefer to form a tide complex and so, silence the DNA transcription.

## Histone Demethylases

LSD1 & LSD2 demethylases
  - Flacin adenine dinucleotide (FAD)- dependent amine oxidation
  - Demethylate mono- and dimethylated substrates

The JMJC family
  - Dioxygenases dependent on Fe^2+^ and α-ketoglutarate
  - Demethylates mono-, di- and trimethylated substrates

==Focuse on the Chemical reaction==

## Understanding the demethylation reaction from the LSD1 Structure

The LSD1 is bound to a 21 amino acid H2 peptide with a mutation of Lys4 to Met.

Modeling predicts that methylated Lys3 binds in a solvent inaccessible area in front of the FAD cofactor (electron extraction).

### Understanding .. JMJD3 structure

THe JMJD3 is bound to a H3 pep with a trimetrhylated Lys27
- The cofactor analog N-oxylayglycine NOG not α-ketoglutarate and nickel not iron are used to prevetn teh reaction
The methylated Lys27 side chain insertes deep into the catalytic pocked, close to the cofactor analog and nickel for demethylation reaction.

### ==Chromatin Writers, Readers and Erases==

brif adn = =  



---

## Recombination

1. homologous reconbination, need small similar sequnces
2. Site-specific reconbination, nees specific DNA sequence
3. ranspositions, no Specific DNA
4. Non-homologous DNA end-joining (), specific proteins that repair dsDNA breaks


### Homologous reconbination
- Required for accurate chomosome segregation
- repairing and tolerating DNA damage
- Recovery of stalled or broken replication forks
- Defects in recombination can contribute to genome instability and cancer predispositions

### Biological roles
1. Genetic reassortments in gametogenesis
2. Repair of DNA damages
3.  Repair replication forks
4. laterl DNA transfer between ceels
5. therOther, deletions, inversions, translocations, etc.

### How HR produces deletions, insertions and inversions

1. Incersion:
  DNA fold in a revised direction, The DNA form a arch bridge and at the foot of the bridge connected
2. Deletion and insertion
  DNA fold in a same direction. A circle formed and the cross point was connected together and the circle was deleted. (Insertion happens the similar way.)

### Translocation that casue cancers
1. Chronic myeloid leukemia
BCR/ABL mRNA
p210 fusion protein

ABL are concertive hyregulated RTK, BRC is a higly active promoter which drive the high expression of the ABL which resbonsible to growth and differentiation.

### Holliday Model of Recombination

Nicking  → Strand Exchange → Branch Migration → Resoltion → Recombination molecules


### SOS response: Mutagenic Survival Pathway of Last Resort
1. DNA Damges → ssDNA or dsDNA break → No Transcription
  Normal LaxA turning of LexA-Regulated gene
2. Auto-cleavage of LaxA recruitment lots of RecA.

RacA: Pre-synapsis → Synapsis → ppost-synapsis
RacA complex driving homology search and alignment of complementary sequences.


### Site-specific reconbination: intergration of lambda phage DNA into the E.coli Chromosome
