<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>WGCNA Tutorial 2 - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="WGCNA Tutorial 2"><meta property="og:type" content="blog"><meta property="og:title" content="WGCNA Tutorial 2"><meta property="og:url" content="https://karobben.github.io/2020/06/10/R/WGCNA_2/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="WGCNA Tutorial 2"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://i.loli.net/2020/06/10/9Rw23KCBXcIOMNz.png"><meta property="article:published_time" content="2020-06-10T05:00:00.000Z"><meta property="article:modified_time" content="2024-01-12T00:29:16.314Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="Bioinformatics"><meta property="article:tag" content="R"><meta property="article:tag" content="WGCNA"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://i.loli.net/2020/06/10/9Rw23KCBXcIOMNz.png"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2020/06/10/R/WGCNA_2/"},"headline":"Karobben","image":["https://i.loli.net/2020/06/10/9Rw23KCBXcIOMNz.png"],"datePublished":"2020-06-10T05:00:00.000Z","dateModified":"2024-01-12T00:29:16.314Z","author":{"@type":"Person","name":"Karobben"},"description":"WGCNA Tutorial 2"}</script><link rel="canonical" href="https://karobben.github.io/2020/06/10/R/WGCNA_2/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><div class="card-image"><span class="image is-7by3"><img class="fill" src="https://i.loli.net/2020/06/10/9Rw23KCBXcIOMNz.png" alt="WGCNA Tutorial 2"></span><span class="ccopy" style="position: absolute; bottom: 0; right: 0; padding-right:3px; padding-left:4px; background-color: rgba(255, 255, 255, 0.6)"></span></div><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2020-06-10T05:00:00.000Z" title="6/10/2020, 12:00:00 AM">2020-06-10</time></span><span class="level-item">Updated&nbsp;<time dateTime="2024-01-12T00:29:16.314Z" title="1/11/2024, 6:29:16 PM">2024-01-11</time></span><span class="level-item"><a class="link-muted" href="/categories/R/">R</a><span> / </span><a class="link-muted" href="/categories/R/Bio/">Bio</a><span> / </span><a class="link-muted" href="/categories/R/Bio/WGCNA/">WGCNA</a></span><span class="level-item">16 minutes read (About 2459 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">WGCNA Tutorial 2</h1><div class="content"><h2 id="WGCNA-Tutorial-2">WGCNA Tutorial 2</h2>
<p><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/">Official Website</a><br>
Paper: <a target="_blank" rel="noopener" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559">Peter Langfelder, 2008</a></p>
<p>Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures.</p>
<p><a target="_blank" rel="noopener" href="https://www.youtube.com/playlist?list=PLAv3hOAKTXH1zpvHiXUbxTmzUi-ZL0AR5">Play list</a> for Network Analysis</p>
<h2 id="Install">Install</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Mirrors</span><br><span class="hljs-comment">## options(&quot;repos&quot; = c(CRAN=&quot;https://mirrors.tuna.tsinghua.edu.cn/CRAN/&quot;))</span><br><span class="hljs-comment">## options(BioC_mirror=&quot;https://mirrors.tuna.tsinghua.edu.cn/bioconductor&quot;)</span><br>install.packages(<span class="hljs-string">&quot;BiocManager&quot;</span>)<br>BiocManager::install(<span class="hljs-string">&quot;WGCNA&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<p>Data:</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip">Female data</a></li>
<li><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/MaleLiver-Data.zip">Male data</a></li>
</ul>
<h2 id="1-Data-Preparation">1. Data Preparation</h2>
<h3 id="1-1-Loading-DataSet">1.1 Loading DataSet</h3>
<p>The data sets contain roughly 130 samples each. Note that each row corresponds to a gene and column to a<br>
sample or auxiliary information.</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(WGCNA)<br><br><span class="hljs-comment">## Reading Data</span><br>options(stringsAsFactors = <span class="hljs-literal">FALSE</span>)<br><span class="hljs-comment">##Read in the female liver data set</span><br>femData = read.csv(<span class="hljs-string">&quot;LiverFemale3600.csv&quot;</span>)<br><span class="hljs-comment">## Read in the male liver data set</span><br>maleData = read.csv(<span class="hljs-string">&quot;LiverMale3600.csv&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<h3 id="1-2-Groups-in-List">1.2 Groups in List</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## We work with two sets:</span><br>nSets = <span class="hljs-number">2</span>;<br><span class="hljs-comment">## For easier labeling of plots, create a vector holding descriptive names of the two sets.</span><br>setLabels = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Female liver&quot;</span>, <span class="hljs-string">&quot;Male liver&quot;</span>)<br>shortLabels = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Female&quot;</span>, <span class="hljs-string">&quot;Male&quot;</span>)<br><br><span class="hljs-comment">## Form multi-set expression data: columns starting from 9 contain actual expression data.</span><br>multiExpr = vector(mode = <span class="hljs-string">&quot;list&quot;</span>, <span class="hljs-built_in">length</span> = nSets)<br><br>multiExpr[[<span class="hljs-number">1</span>]] = <span class="hljs-built_in">list</span>(data = as.data.frame(t(femData[-<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)])));<br><span class="hljs-built_in">names</span>(multiExpr[[<span class="hljs-number">1</span>]]$data) = femData$substanceBXH;<br>rownames(multiExpr[[<span class="hljs-number">1</span>]]$data) = <span class="hljs-built_in">names</span>(femData)[-<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)];<br>multiExpr[[<span class="hljs-number">2</span>]] = <span class="hljs-built_in">list</span>(data = as.data.frame(t(maleData[-<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)])));<br><span class="hljs-built_in">names</span>(multiExpr[[<span class="hljs-number">2</span>]]$data) = maleData$substanceBXH;<br>rownames(multiExpr[[<span class="hljs-number">2</span>]]$data) = <span class="hljs-built_in">names</span>(maleData)[-<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)];<br><span class="hljs-comment">## Check that the data has the correct format for many functions operating on multiple sets:</span><br>exprSize = checkSets(multiExpr)<br></code></pre></td></tr></table></figure></div>
<h3 id="1-3-Rudimentary-data-cleaning-and-outlier-removal">1.3 Rudimentary data cleaning and outlier removal</h3>
<p>Check that all genes and samples have sufficiently low numbers of missing values.</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">gsg = goodSamplesGenesMS(multiExpr, verbose = <span class="hljs-number">3</span>);<br>gsg$allOK<br><br><span class="hljs-keyword">if</span> (!gsg$allOK)<br>&#123;<br>  <span class="hljs-comment"># Print information about the removed genes:</span><br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">sum</span>(!gsg$goodGenes) &gt; <span class="hljs-number">0</span>)<br>    printFlush(paste(<span class="hljs-string">&quot;Removing genes:&quot;</span>, paste(<span class="hljs-built_in">names</span>(multiExpr[[<span class="hljs-number">1</span>]]$data)[!gsg$goodGenes],<br>                                              collapse = <span class="hljs-string">&quot;, &quot;</span>)))<br>  <span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:exprSize$nSets)<br>  &#123;<br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">sum</span>(!gsg$goodSamples[[set]]))<br>      printFlush(paste(<span class="hljs-string">&quot;In set&quot;</span>, setLabels[set], <span class="hljs-string">&quot;removing samples&quot;</span>,<br>                       paste(rownames(multiExpr[[set]]$data)[!gsg$goodSamples[[set]]], collapse = <span class="hljs-string">&quot;, &quot;</span>)))<br>    <span class="hljs-comment"># Remove the offending genes and samples</span><br>    multiExpr[[set]]$data = multiExpr[[set]]$data[gsg$goodSamples[[set]], gsg$goodGenes];<br>  &#125;<br>  <span class="hljs-comment"># Update exprSize</span><br>  exprSize = checkSets(multiExpr)<br>&#125;<br></code></pre></td></tr></table></figure></div>
<h3 id="1-4-Cluster">1.4 Cluster</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">sampleTrees = <span class="hljs-built_in">list</span>()<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  sampleTrees[[set]] = hclust(dist(multiExpr[[set]]$data), method = <span class="hljs-string">&quot;average&quot;</span>)<br>&#125;<br><br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>,<span class="hljs-number">1</span>))<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-number">4</span>, <span class="hljs-number">2</span>, <span class="hljs-number">0</span>))<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>  plot(sampleTrees[[set]], main = paste(<span class="hljs-string">&quot;Sample clustering on all genes in&quot;</span>, setLabels[set]),<br>       xlab=<span class="hljs-string">&quot;&quot;</span>, sub=<span class="hljs-string">&quot;&quot;</span>, cex = <span class="hljs-number">0.7</span>)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://s1.ax1x.com/2020/06/07/t2ocjA.png" alt="t2ocjA.png"></p>
<p>By inspection, there seems to be <span style="background:salmon"><strong>one outlier</strong></span> in the female data set, and no obvious outliers in the male set. We<br>
now <span style="background:salmon"><strong>remove the female outlier</strong></span> using a semi-automatic code that only requires a choice of a height cut</p>
<h3 id="1-5-Remove-Outlier">1.5 Remove Outlier</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Choose the &quot;base&quot; cut height for the female data set</span><br>baseHeight = <span class="hljs-number">16</span><br><span class="hljs-comment">## Adjust the cut height for the male data set for the number of samples</span><br>cutHeights = <span class="hljs-built_in">c</span>(<span class="hljs-number">16</span>, <span class="hljs-number">16</span>*exprSize$nSamples[<span class="hljs-number">2</span>]/exprSize$nSamples[<span class="hljs-number">1</span>]);<br><span class="hljs-comment">## Re-plot the dendrograms including the cut lines</span><br>png(file = <span class="hljs-string">&quot;SampleClustering.png&quot;</span>);<br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>,<span class="hljs-number">1</span>))<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-number">4</span>, <span class="hljs-number">2</span>, <span class="hljs-number">0</span>))<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  plot(sampleTrees[[set]], main = paste(<span class="hljs-string">&quot;Sample clustering on all genes in&quot;</span>, setLabels[set]),<br>       xlab=<span class="hljs-string">&quot;&quot;</span>, sub=<span class="hljs-string">&quot;&quot;</span>, cex = <span class="hljs-number">0.7</span>);<br>  abline(h=cutHeights[set], col = <span class="hljs-string">&quot;red&quot;</span>);<br>&#125;<br>dev.off();<br><br><span class="hljs-comment">## Removing outlier</span><br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  <span class="hljs-comment"># Find clusters cut by the line</span><br>  labels = cutreeStatic(sampleTrees[[set]], cutHeight = cutHeights[set])<br>  <span class="hljs-comment"># Keep the largest one (labeled by the number 1)</span><br>  keep = (labels==<span class="hljs-number">1</span>)<br>  multiExpr[[set]]$data = multiExpr[[set]]$data[keep, ]<br>&#125;<br>collectGarbage();<br><span class="hljs-comment">## Check the size of the leftover data</span><br>exprSize = checkSets(multiExpr)<br>exprSize<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/07/RSMCdKeDpl9YqQj.png" alt="SampleClustering"></p>
<h3 id="1-6-Loading-Clinical-Trait-Data">1.6 Loading Clinical Trait Data</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">traitData = read.csv(<span class="hljs-string">&quot;ClinicalTraits.csv&quot;</span>);<br><span class="hljs-comment">## remove columns that hold information we do not need.</span><br>allTraits = traitData[, -<span class="hljs-built_in">c</span>(<span class="hljs-number">31</span>, <span class="hljs-number">16</span>)];<br>allTraits = allTraits[, <span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>, <span class="hljs-number">11</span>:<span class="hljs-number">36</span>) ];<br><span class="hljs-comment">## See how big the traits are and what are the trait and sample names</span><br><span class="hljs-built_in">dim</span>(allTraits)<br><span class="hljs-built_in">names</span>(allTraits)<br>allTraits$Mice<br><span class="hljs-comment">## Form a multi-set structure that will hold the clinical traits.</span><br>Traits = vector(mode=<span class="hljs-string">&quot;list&quot;</span>, <span class="hljs-built_in">length</span> = nSets);<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  setSamples = rownames(multiExpr[[set]]$data);<br>  traitRows = match(setSamples, allTraits$Mice);<br>  Traits[[set]] = <span class="hljs-built_in">list</span>(data = allTraits[traitRows, -<span class="hljs-number">1</span>]);<br>  rownames(Traits[[set]]$data) = allTraits[traitRows, <span class="hljs-number">1</span>];<br>&#125;<br>collectGarbage();<br><span class="hljs-comment">## Define data set dimensions</span><br>nGenes = exprSize$nGenes;<br>nSamples = exprSize$nSamples;<br></code></pre></td></tr></table></figure></div>
<h2 id="2-Network-Construction-and-Module-Detection">2. Network Construction and Module Detection</h2>
<pre class="mermaid">graph LR;
   START(Network construction)
   Net1(one-step network construction)
   Net2(Step-by-step network construction)
   Net3(automatic block-wise network construction)

   START-->|Minimum Effort|Net1;
   START-->|Customized/Alternate Methods|Net2;
   START-->|Large Data Set|Net3;</pre>
<h3 id="2-1-One-Step-Network-Construction-and-Module-Detection">2.1 One-Step Network Construction and Module Detection</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Choose a set of soft-thresholding powers</span><br>powers = <span class="hljs-built_in">c</span>(seq(<span class="hljs-number">4</span>,<span class="hljs-number">10</span>,by=<span class="hljs-number">1</span>), seq(<span class="hljs-number">12</span>,<span class="hljs-number">20</span>, by=<span class="hljs-number">2</span>));<br><span class="hljs-comment">## Initialize a list to hold the results of scale-free analysis</span><br>powerTables = vector(mode = <span class="hljs-string">&quot;list&quot;</span>, <span class="hljs-built_in">length</span> = nSets);<br><span class="hljs-comment">## Call the network topology analysis function for each set in turn</span><br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>  powerTables[[set]] = <span class="hljs-built_in">list</span>(data = pickSoftThreshold(multiExpr[[set]]$data, powerVector=powers,<br>                                                     verbose = <span class="hljs-number">2</span>)[[<span class="hljs-number">2</span>]]);<br>collectGarbage();<br><span class="hljs-comment">## Plot the results:</span><br>colors = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;black&quot;</span>, <span class="hljs-string">&quot;red&quot;</span>)<br><span class="hljs-comment">## Will plot these columns of the returned scale free analysis tables</span><br>plotCols = <span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>,<span class="hljs-number">5</span>,<span class="hljs-number">6</span>,<span class="hljs-number">7</span>)<br>colNames = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Scale Free Topology Model Fit&quot;</span>, <span class="hljs-string">&quot;Mean connectivity&quot;</span>, <span class="hljs-string">&quot;Median connectivity&quot;</span>,<br><span class="hljs-string">&quot;Max connectivity&quot;</span>);<br><span class="hljs-comment">## Get the minima and maxima of the plotted points</span><br>ylim = matrix(<span class="hljs-literal">NA</span>, nrow = <span class="hljs-number">2</span>, ncol = <span class="hljs-number">4</span>);<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  <span class="hljs-keyword">for</span> (col <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:<span class="hljs-built_in">length</span>(plotCols))<br>  &#123;<br>    ylim[<span class="hljs-number">1</span>, col] = <span class="hljs-built_in">min</span>(ylim[<span class="hljs-number">1</span>, col], powerTables[[set]]$data[, plotCols[col]], na.rm = <span class="hljs-literal">TRUE</span>);<br>    ylim[<span class="hljs-number">2</span>, col] = <span class="hljs-built_in">max</span>(ylim[<span class="hljs-number">2</span>, col], powerTables[[set]]$data[, plotCols[col]], na.rm = <span class="hljs-literal">TRUE</span>);<br>  &#125;<br>&#125;<br><span class="hljs-comment">## Plot the quantities in the chosen columns vs. the soft thresholding power</span><br>sizeGrWindow(<span class="hljs-number">8</span>, <span class="hljs-number">6</span>)<br>png(file = <span class="hljs-string">&quot;scaleFreeAnalysis.png&quot;</span>)<br>par(mfcol = <span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>,<span class="hljs-number">2</span>));<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">4.2</span>, <span class="hljs-number">4.2</span> , <span class="hljs-number">2.2</span>, <span class="hljs-number">0.5</span>))<br>cex1 = <span class="hljs-number">0.7</span>;<br><span class="hljs-keyword">for</span> (col <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:<span class="hljs-built_in">length</span>(plotCols)) <span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  <span class="hljs-keyword">if</span> (set==<span class="hljs-number">1</span>)<br>  &#123;<br>    plot(powerTables[[set]]$data[,<span class="hljs-number">1</span>], -<span class="hljs-built_in">sign</span>(powerTables[[set]]$data[,<span class="hljs-number">3</span>])*powerTables[[set]]$data[,<span class="hljs-number">2</span>],<br>         xlab=<span class="hljs-string">&quot;Soft Threshold (power)&quot;</span>,ylab=colNames[col],type=<span class="hljs-string">&quot;n&quot;</span>, ylim = ylim[, col],<br>         main = colNames[col]);<br>    addGrid();<br>  &#125;<br>  <span class="hljs-keyword">if</span> (col==<span class="hljs-number">1</span>)<br>  &#123;<br>    text(powerTables[[set]]$data[,<span class="hljs-number">1</span>], -<span class="hljs-built_in">sign</span>(powerTables[[set]]$data[,<span class="hljs-number">3</span>])*powerTables[[set]]$data[,<span class="hljs-number">2</span>],<br>         labels=powers,cex=cex1,col=colors[set]);<br>  &#125; <span class="hljs-keyword">else</span><br>    text(powerTables[[set]]$data[,<span class="hljs-number">1</span>], powerTables[[set]]$data[,plotCols[col]],<br>         labels=powers,cex=cex1,col=colors[set]);<br>  <span class="hljs-keyword">if</span> (col==<span class="hljs-number">1</span>)<br>  &#123;<br>    legend(<span class="hljs-string">&quot;bottomright&quot;</span>, legend = setLabels, col = colors, pch = <span class="hljs-number">20</span>) ;<br>  &#125; <span class="hljs-keyword">else</span><br>    legend(<span class="hljs-string">&quot;topright&quot;</span>, legend = setLabels, col = colors, pch = <span class="hljs-number">20</span>) ;<br>&#125;<br>dev.off();<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/07/MhcuA8aRYj9yCWg.png" alt="scaleFreeAnalysis"></p>
<h3 id="2-2-Network-construction-and-consensus-module-detection">2.2  Network construction and consensus module detection</h3>
<p><span style="background:salmon">Attention</span>: We have chosen the soft thresholding power <span style="background:salmon">6</span>, minimum module size <span style="background:salmon">30</span>, the module detection sensitivity deepSplit <span style="background:salmon">2</span>, cut height for merging of modules 0.20 (implying that modules whose eigengenes are correlated above 1 − 0.2 = 0.8 will be merged), we requested that the function return numeric module labels rather than color labels, we have effectively turned off reassigning genes based on their module eigengene-based connectivity KME, and we have instructed the code to save the calculated consensus topological overlap.</p>
<p>In this example most of them are left at their default value. We encourage the user to read the help file provided within the package in the R environment and experiment with tweaking the network construction and module detection parameters. The potential reward is, of course, better (biologically more relevant) results of the analysis.</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">net = blockwiseConsensusModules(<br>        multiExpr, power = <span class="hljs-number">6</span>, minModuleSize = <span class="hljs-number">30</span>, deepSplit = <span class="hljs-number">2</span>,<br>        pamRespectsDendro = <span class="hljs-literal">FALSE</span>,<br>        mergeCutHeight = <span class="hljs-number">0.25</span>, numericLabels = <span class="hljs-literal">TRUE</span>,<br>        minKMEtoStay = <span class="hljs-number">0</span>,<br>        saveTOMs = <span class="hljs-literal">TRUE</span>, verbose = <span class="hljs-number">5</span>)<br></code></pre></td></tr></table></figure></div>
<h3 id="2-3-Model-Extract">2.3 Model Extract</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">consMEs = net$multiMEs;<br>moduleLabels = net$colors;<br><span class="hljs-comment">## Convert the numeric labels to color labels</span><br>moduleColors = labels2colors(moduleLabels)<br>consTree = net$dendrograms[[<span class="hljs-number">1</span>]]<br><br>sizeGrWindow(<span class="hljs-number">8</span>,<span class="hljs-number">6</span>);<br>png(file = <span class="hljs-string">&quot;ConsensusDendrogram-auto.png&quot;</span>, wi = <span class="hljs-number">600</span>, he = <span class="hljs-number">340</span>)<br>plotDendroAndColors(consTree, moduleColors,<br>                    <span class="hljs-string">&quot;Module colors&quot;</span>,<br>                    dendroLabels = <span class="hljs-literal">FALSE</span>, hang = <span class="hljs-number">0.03</span>,<br>                    addGuide = <span class="hljs-literal">TRUE</span>, guideHang = <span class="hljs-number">0.05</span>,<br>                    main = <span class="hljs-string">&quot;Consensus gene dendrogram and module colors&quot;</span>)<br><br>dev.off()<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/07/4Mm8FXaLduRPiQ1.png" alt="ConsensusDendrogram-auto"></p>
<h2 id="3-Relating-the-consensus-modules-to-female-set-specific-modules">3. Relating the consensus modules to female set-specific modules</h2>
<h3 id="3-1-datExpr">3.1 datExpr</h3>
<p>(WGCNA Tutorial 1, 1-2)</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">femData = read.csv(<span class="hljs-string">&quot;LiverFemale3600.csv&quot;</span>);<br>datExpr0 = as.data.frame(t(femData[, -<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)]));<br><span class="hljs-built_in">names</span>(datExpr0) = femData$substanceBXH;<br>rownames(datExpr0) = <span class="hljs-built_in">names</span>(femData)[-<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)];<br>gsg = goodSamplesGenes(datExpr0, verbose = <span class="hljs-number">3</span>);<br>gsg$allOK<br><span class="hljs-keyword">if</span> (!gsg$allOK)<br>&#123;<br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">sum</span>(!gsg$goodGenes)&gt;<span class="hljs-number">0</span>)<br>     printFlush(paste(<span class="hljs-string">&quot;Removing genes:&quot;</span>, paste(<span class="hljs-built_in">names</span>(datExpr0)[!gsg$goodGenes], collapse = <span class="hljs-string">&quot;, &quot;</span>)));<br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">sum</span>(!gsg$goodSamples)&gt;<span class="hljs-number">0</span>)<br>     printFlush(paste(<span class="hljs-string">&quot;Removing samples:&quot;</span>, paste(rownames(datExpr0)[!gsg$goodSamples], collapse = <span class="hljs-string">&quot;, &quot;</span>)));<br>  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]<br>&#125;<br>sampleTree = hclust(dist(datExpr0), method = <span class="hljs-string">&quot;average&quot;</span>);<br>clust = cutreeStatic(sampleTree, cutHeight = <span class="hljs-number">15</span>, minSize = <span class="hljs-number">10</span>)<br>keepSamples = (clust==<span class="hljs-number">1</span>)<br>datExpr = datExpr0[keepSamples, ]<br>nGenes = ncol(datExpr)<br>nSamples = nrow(datExpr)<br>traitData = read.csv(<span class="hljs-string">&quot;ClinicalTraits.csv&quot;</span>);<br><span class="hljs-built_in">dim</span>(traitData)<br><span class="hljs-built_in">names</span>(traitData)<br>allTraits = traitData[, -<span class="hljs-built_in">c</span>(<span class="hljs-number">31</span>, <span class="hljs-number">16</span>)];<br>allTraits = allTraits[, <span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>, <span class="hljs-number">11</span>:<span class="hljs-number">36</span>) ];<br><span class="hljs-built_in">dim</span>(allTraits)<br><span class="hljs-built_in">names</span>(allTraits)<br>femaleSamples = rownames(datExpr);<br>traitRows = match(femaleSamples, allTraits$Mice);<br>datTraits = allTraits[traitRows, -<span class="hljs-number">1</span>];<br>rownames(datTraits) = allTraits[traitRows, <span class="hljs-number">1</span>];<br>collectGarbage()<br>sampleTree2 = hclust(dist(datExpr), method = <span class="hljs-string">&quot;average&quot;</span>)<br>traitColors = numbers2colors(datTraits, signed = <span class="hljs-literal">FALSE</span>);<br>powers = <span class="hljs-built_in">c</span>(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">10</span>), seq(from = <span class="hljs-number">12</span>, to=<span class="hljs-number">20</span>, by=<span class="hljs-number">2</span>))<br>sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="hljs-number">5</span>)<br>net = blockwiseModules(datExpr, power = <span class="hljs-number">6</span>,<br>                       TOMType = <span class="hljs-string">&quot;unsigned&quot;</span>, minModuleSize = <span class="hljs-number">30</span>,<br>                       reassignThreshold = <span class="hljs-number">0</span>, mergeCutHeight = <span class="hljs-number">0.25</span>,<br>                       numericLabels = <span class="hljs-literal">TRUE</span>, pamRespectsDendro = <span class="hljs-literal">FALSE</span>,<br>                       saveTOMs = <span class="hljs-literal">TRUE</span>,<br>                       saveTOMFileBase = <span class="hljs-string">&quot;femaleMouseTOM&quot;</span>,<br>                       verbose = <span class="hljs-number">3</span>)<br>moduleLabels = net$colors<br>moduleColors = labels2colors(net$colors)<br>MEs = net$MEs;<br>geneTree = net$dendrograms[[<span class="hljs-number">1</span>]]<br></code></pre></td></tr></table></figure></div>
<h3 id="3-2-Relating-consensus-modules-to-female-set-specific-modules">3.2 Relating consensus modules to female set-specific modules</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">femaleLabels = moduleLabels;<br>femaleColors = moduleColors;<br>femaleTree = geneTree;<br>femaleMEs = orderMEs(MEs, greyName = <span class="hljs-string">&quot;ME0&quot;</span>);<br><br><br><span class="hljs-comment">## Isolate the module labels in the order they appear in ordered module eigengenes</span><br>femModuleLabels = substring(<span class="hljs-built_in">names</span>(femaleMEs), <span class="hljs-number">3</span>)<br>consModuleLabels = substring(<span class="hljs-built_in">names</span>(consMEs[[<span class="hljs-number">1</span>]]$data), <span class="hljs-number">3</span>)<br><span class="hljs-comment">## Convert the numeric module labels to color labels</span><br>femModules = labels2colors(<span class="hljs-built_in">as.numeric</span>(femModuleLabels))<br>consModules = labels2colors(<span class="hljs-built_in">as.numeric</span>(consModuleLabels))<br><span class="hljs-comment">## Numbers of female and consensus modules</span><br>nFemMods = <span class="hljs-built_in">length</span>(femModules)<br>nConsMods = <span class="hljs-built_in">length</span>(consModules)<br><span class="hljs-comment">## Initialize tables of p-values and of the corresponding counts</span><br>pTable = matrix(<span class="hljs-number">0</span>, nrow = nFemMods, ncol = nConsMods);<br>CountTbl = matrix(<span class="hljs-number">0</span>, nrow = nFemMods, ncol = nConsMods);<br><span class="hljs-comment">## Execute all pairwaise comparisons</span><br><span class="hljs-keyword">for</span> (fmod <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nFemMods)<br>  <span class="hljs-keyword">for</span> (cmod <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nConsMods)<br>  &#123;<br>    femMembers = (femaleColors == femModules[fmod]);<br>    consMembers = (moduleColors == consModules[cmod]);<br>    pTable[fmod, cmod] = -log10(fisher.test(femMembers, consMembers, alternative = <span class="hljs-string">&quot;greater&quot;</span>)$p.value);<br>    CountTbl[fmod, cmod] = <span class="hljs-built_in">sum</span>(femaleColors == femModules[fmod] &amp; moduleColors ==<br>                      consModules[cmod])<br>  &#125;<br><br><span class="hljs-comment">## Truncate p values smaller than 10^&#123;-50&#125; to 10^&#123;-50&#125;</span><br>pTable[<span class="hljs-built_in">is.infinite</span>(pTable)] = <span class="hljs-number">1.3</span>*<span class="hljs-built_in">max</span>(pTable[<span class="hljs-built_in">is.finite</span>(pTable)]);<br>pTable[pTable&gt;<span class="hljs-number">50</span> ] = <span class="hljs-number">50</span> ;<br><span class="hljs-comment">## Marginal counts (really module sizes)</span><br>femModTotals = apply(CountTbl, <span class="hljs-number">1</span>, <span class="hljs-built_in">sum</span>)<br>consModTotals = apply(CountTbl, <span class="hljs-number">2</span>, <span class="hljs-built_in">sum</span>)<br><span class="hljs-comment">## Actual plotting</span><br>sizeGrWindow(<span class="hljs-number">10</span>,<span class="hljs-number">7</span> );<br><span class="hljs-comment">##png(file = &quot;ConsensusVsFemaleModules.png&quot;, wi = 500, he = 400);</span><br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>));<br>par(cex = <span class="hljs-number">1.0</span>);<br>par(mar=<span class="hljs-built_in">c</span>(<span class="hljs-number">8</span>, <span class="hljs-number">10.4</span>, <span class="hljs-number">2.7</span>, <span class="hljs-number">1</span>)+<span class="hljs-number">0.3</span>);<br><span class="hljs-comment">## Use function labeledHeatmap to produce the color-coded table with all the trimmings</span><br>labeledHeatmap(Matrix = pTable,<br>               xLabels = paste(<span class="hljs-string">&quot; &quot;</span>, consModules),<br>               yLabels = paste(<span class="hljs-string">&quot; &quot;</span>, femModules),<br>               colorLabels = <span class="hljs-literal">TRUE</span>,<br>               xSymbols = paste(<span class="hljs-string">&quot;Cons &quot;</span>, consModules, <span class="hljs-string">&quot;: &quot;</span>, consModTotals, sep=<span class="hljs-string">&quot;&quot;</span>),<br>               ySymbols = paste(<span class="hljs-string">&quot;Fem &quot;</span>, femModules, <span class="hljs-string">&quot;: &quot;</span>, femModTotals, sep=<span class="hljs-string">&quot;&quot;</span>),<br>               textMatrix = CountTbl,<br>               colors = greenWhiteRed(<span class="hljs-number">100</span>)[<span class="hljs-number">50</span>:<span class="hljs-number">100</span>],<br>               main = <span class="hljs-string">&quot;Correspondence of Female set-specific and Female-Male consensus modules&quot;</span>,<br>               cex.text = <span class="hljs-number">1.0</span>, cex.lab = <span class="hljs-number">1.0</span>, setStdMargins = <span class="hljs-literal">FALSE</span>);<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/9Rw23KCBXcIOMNz.png" alt="ConsensusVsFemaleModules"></p>
<h2 id="4-Consensus-module-to-traits">4. Consensus module to traits</h2>
<h3 id="4-1-Correlation">4.1 Correlation</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">moduleTraitCor = <span class="hljs-built_in">list</span>();<br>moduleTraitPvalue = <span class="hljs-built_in">list</span>();<br><span class="hljs-comment">## Calculate the correlations</span><br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  moduleTraitCor[[set]] = cor(consMEs[[set]]$data, Traits[[set]]$data, use = <span class="hljs-string">&quot;p&quot;</span>);<br>  moduleTraitPvalue[[set]] = corPvalueFisher(moduleTraitCor[[set]], exprSize$nSamples[set]);<br>&#125;<br><br><br><br><br><span class="hljs-comment">## Convert numerical lables to colors for labeling of modules in the plot</span><br>MEColors = labels2colors(<span class="hljs-built_in">as.numeric</span>(substring(<span class="hljs-built_in">names</span>(consMEs[[<span class="hljs-number">1</span>]]$data), <span class="hljs-number">3</span>)));<br>MEColorNames = paste(<span class="hljs-string">&quot;ME&quot;</span>, MEColors, sep=<span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-comment">## Open a suitably sized window (the user should change the window size if necessary)</span><br>sizeGrWindow(<span class="hljs-number">10</span>,<span class="hljs-number">7</span>)<br><span class="hljs-comment">##png(file = &quot;ModuleTraitRelationships-female.png&quot;, wi = 800, he = 560);</span><br><span class="hljs-comment">## Plot the module-trait relationship table for set number 1</span><br>set = <span class="hljs-number">1</span><br>textMatrix =  paste(<span class="hljs-built_in">signif</span>(moduleTraitCor[[set]], <span class="hljs-number">2</span>), <span class="hljs-string">&quot;\n(&quot;</span>,<br>                           <span class="hljs-built_in">signif</span>(moduleTraitPvalue[[set]], <span class="hljs-number">1</span>), <span class="hljs-string">&quot;)&quot;</span>, sep = <span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-built_in">dim</span>(textMatrix) = <span class="hljs-built_in">dim</span>(moduleTraitCor[[set]])<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">6</span>, <span class="hljs-number">8.8</span>, <span class="hljs-number">3</span>, <span class="hljs-number">2.2</span>));<br>labeledHeatmap(Matrix = moduleTraitCor[[set]],<br>               xLabels = <span class="hljs-built_in">names</span>(Traits[[set]]$data),<br>               yLabels = MEColorNames,<br>               ySymbols = MEColorNames,<br>               colorLabels = <span class="hljs-literal">FALSE</span>,<br>               colors = greenWhiteRed(<span class="hljs-number">50</span>),<br>               textMatrix = textMatrix,<br>               setStdMargins = <span class="hljs-literal">FALSE</span>,<br>               cex.text = <span class="hljs-number">0.5</span>,<br>               zlim = <span class="hljs-built_in">c</span>(-<span class="hljs-number">1</span>,<span class="hljs-number">1</span>),<br>               main = paste(<span class="hljs-string">&quot;Module--trait relationships in&quot;</span>, setLabels[set]))<br><span class="hljs-comment">##dev.off();</span><br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/pMdjGFZnW8sbK9y.png" alt="ModuleTraitRelationships-female"></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Plot the module-trait relationship table for set number 2</span><br>set = <span class="hljs-number">2</span><br>textMatrix =  paste(<span class="hljs-built_in">signif</span>(moduleTraitCor[[set]], <span class="hljs-number">2</span>), <span class="hljs-string">&quot;\n(&quot;</span>,<br>                           <span class="hljs-built_in">signif</span>(moduleTraitPvalue[[set]], <span class="hljs-number">1</span>), <span class="hljs-string">&quot;)&quot;</span>, sep = <span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-built_in">dim</span>(textMatrix) = <span class="hljs-built_in">dim</span>(moduleTraitCor[[set]])<br>sizeGrWindow(<span class="hljs-number">10</span>,<span class="hljs-number">7</span>)<br><span class="hljs-comment">##png(file = &quot;ModuleTraitRelationships-male.png&quot;, wi = 800, he = 560);</span><br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">6</span>, <span class="hljs-number">8.8</span>, <span class="hljs-number">3</span>, <span class="hljs-number">2.2</span>));<br>labeledHeatmap(Matrix = moduleTraitCor[[set]],<br>               xLabels = <span class="hljs-built_in">names</span>(Traits[[set]]$data),<br>               yLabels = MEColorNames,<br>               ySymbols = MEColorNames,<br>               colorLabels = <span class="hljs-literal">FALSE</span>,<br>               colors = greenWhiteRed(<span class="hljs-number">50</span>),<br>               textMatrix = textMatrix,<br>               setStdMargins = <span class="hljs-literal">FALSE</span>,<br>               cex.text = <span class="hljs-number">0.5</span>,<br>               zlim = <span class="hljs-built_in">c</span>(-<span class="hljs-number">1</span>,<span class="hljs-number">1</span>),<br>               main = paste(<span class="hljs-string">&quot;Module--trait relationships in&quot;</span>, setLabels[set]))<br>dev.off();<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/WgDs7yQpBbe4CIu.png" alt="ModuleTraitRelationships-male"></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><br><span class="hljs-comment">## Initialize matrices to hold the consensus correlation and p-value</span><br>consensusCor = matrix(<span class="hljs-literal">NA</span>, nrow(moduleTraitCor[[<span class="hljs-number">1</span>]]), ncol(moduleTraitCor[[<span class="hljs-number">1</span>]]));<br>consensusPvalue = matrix(<span class="hljs-literal">NA</span>, nrow(moduleTraitCor[[<span class="hljs-number">1</span>]]), ncol(moduleTraitCor[[<span class="hljs-number">1</span>]]));<br><span class="hljs-comment">## Find consensus negative correlations</span><br>negative = moduleTraitCor[[<span class="hljs-number">1</span>]] &lt; <span class="hljs-number">0</span> &amp; moduleTraitCor[[<span class="hljs-number">2</span>]] &lt; <span class="hljs-number">0</span>;<br>consensusCor[negative] = pmax(moduleTraitCor[[<span class="hljs-number">1</span>]][negative], moduleTraitCor[[<span class="hljs-number">2</span>]][negative]);<br>consensusPvalue[negative] = pmax(moduleTraitPvalue[[<span class="hljs-number">1</span>]][negative], moduleTraitPvalue[[<span class="hljs-number">2</span>]][negative]);<br><span class="hljs-comment">## Find consensus positive correlations</span><br>positive = moduleTraitCor[[<span class="hljs-number">1</span>]] &gt; <span class="hljs-number">0</span> &amp; moduleTraitCor[[<span class="hljs-number">2</span>]] &gt; <span class="hljs-number">0</span>;<br>consensusCor[positive] = pmin(moduleTraitCor[[<span class="hljs-number">1</span>]][positive], moduleTraitCor[[<span class="hljs-number">2</span>]][positive]);<br>consensusPvalue[positive] = pmax(moduleTraitPvalue[[<span class="hljs-number">1</span>]][positive], moduleTraitPvalue[[<span class="hljs-number">2</span>]][positive]);<br><br>textMatrix =  paste(<span class="hljs-built_in">signif</span>(consensusCor, <span class="hljs-number">2</span>), <span class="hljs-string">&quot;\n(&quot;</span>,<br>                           <span class="hljs-built_in">signif</span>(consensusPvalue, <span class="hljs-number">1</span>), <span class="hljs-string">&quot;)&quot;</span>, sep = <span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-built_in">dim</span>(textMatrix) = <span class="hljs-built_in">dim</span>(moduleTraitCor[[set]])<br>sizeGrWindow(<span class="hljs-number">10</span>,<span class="hljs-number">7</span>)<br>png(file = <span class="hljs-string">&quot;ModuleTraitRelationships-consensus.png&quot;</span>, wi = <span class="hljs-number">800</span>, he = <span class="hljs-number">560</span>);<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">6</span>, <span class="hljs-number">8.8</span>, <span class="hljs-number">3</span>, <span class="hljs-number">2.2</span>));<br>labeledHeatmap(Matrix = consensusCor,<br>               xLabels = <span class="hljs-built_in">names</span>(Traits[[set]]$data),<br>               yLabels = MEColorNames,<br>               ySymbols = MEColorNames,<br>               colorLabels = <span class="hljs-literal">FALSE</span>,<br>               colors = greenWhiteRed(<span class="hljs-number">50</span>),<br>               textMatrix = textMatrix,<br>               setStdMargins = <span class="hljs-literal">FALSE</span>,<br>               cex.text = <span class="hljs-number">0.5</span>,<br>               zlim = <span class="hljs-built_in">c</span>(-<span class="hljs-number">1</span>,<span class="hljs-number">1</span>),<br>               main = paste(<span class="hljs-string">&quot;Consensus module--trait relationships across\n&quot;</span>,<br>                            paste(setLabels, collapse = <span class="hljs-string">&quot; and &quot;</span>)))<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/AYwn9aqMlbrsgST.png" alt="ModuleTraitRelationships-consensus"></p>
<h3 id="4-2-Exporting-results-of-the-network-analysis">4.2 Exporting results of the network analysis</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><br>file = gzfile(description = <span class="hljs-string">&quot;GeneAnnotation.csv.gz&quot;</span>);<br>annot = read.csv(file = file);<br><span class="hljs-comment">## Match probes in the data set to the probe IDs in the annotation file</span><br>probes = <span class="hljs-built_in">names</span>(multiExpr[[<span class="hljs-number">1</span>]]$data)<br>probes2annot = match(probes, annot$substanceBXH)<br><br><br>consMEs.unord = multiSetMEs(multiExpr, universalColors = moduleLabels, excludeGrey = <span class="hljs-literal">TRUE</span>)<br>GS = <span class="hljs-built_in">list</span>();<br>kME = <span class="hljs-built_in">list</span>();<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  GS[[set]] = corAndPvalue(multiExpr[[set]]$data, Traits[[set]]$data);<br>  kME[[set]] = corAndPvalue(multiExpr[[set]]$data, consMEs.unord[[set]]$data);<br>&#125;<br><br><br>GS.metaZ = (GS[[<span class="hljs-number">1</span>]]$Z + GS[[<span class="hljs-number">2</span>]]$Z)/<span class="hljs-built_in">sqrt</span>(<span class="hljs-number">2</span>);<br>kME.metaZ = (kME[[<span class="hljs-number">1</span>]]$Z + kME[[<span class="hljs-number">2</span>]]$Z)/<span class="hljs-built_in">sqrt</span>(<span class="hljs-number">2</span>);<br>GS.metaP = <span class="hljs-number">2</span>*pnorm(<span class="hljs-built_in">abs</span>(GS.metaZ), lower.tail = <span class="hljs-literal">FALSE</span>);<br>kME.metaP = <span class="hljs-number">2</span>*pnorm(<span class="hljs-built_in">abs</span>(kME.metaZ), lower.tail = <span class="hljs-literal">FALSE</span>);<br><br>GSmat = rbind(GS[[<span class="hljs-number">1</span>]]$cor, GS[[<span class="hljs-number">2</span>]]$cor, GS[[<span class="hljs-number">1</span>]]$p, GS[[<span class="hljs-number">2</span>]]$p, GS.metaZ, GS.metaP);<br>nTraits = checkSets(Traits)$nGenes<br>traitNames = colnames(Traits[[<span class="hljs-number">1</span>]]$data)<br><span class="hljs-built_in">dim</span>(GSmat) = <span class="hljs-built_in">c</span>(nGenes, <span class="hljs-number">6</span>*nTraits)<br>rownames(GSmat) = probes;<br>colnames(GSmat) = spaste(<br>    <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;GS.set1.&quot;</span>, <span class="hljs-string">&quot;GS.set2.&quot;</span>, <span class="hljs-string">&quot;p.GS.set1.&quot;</span>, <span class="hljs-string">&quot;p.GS.set2.&quot;</span>, <span class="hljs-string">&quot;Z.GS.meta.&quot;</span>, <span class="hljs-string">&quot;p.GS.meta&quot;</span>),<br>    <span class="hljs-built_in">rep</span>(traitNames, <span class="hljs-built_in">rep</span>(<span class="hljs-number">6</span>, nTraits)))<br><span class="hljs-comment">## Same code for kME:</span><br>kMEmat = rbind(kME[[<span class="hljs-number">1</span>]]$cor, kME[[<span class="hljs-number">2</span>]]$cor, kME[[<span class="hljs-number">1</span>]]$p, kME[[<span class="hljs-number">2</span>]]$p, kME.metaZ, kME.metaP);<br>MEnames = colnames(consMEs.unord[[<span class="hljs-number">1</span>]]$data);<br>nMEs = checkSets(consMEs.unord)$nGenes<br><span class="hljs-built_in">dim</span>(kMEmat) = <span class="hljs-built_in">c</span>(nGenes, <span class="hljs-number">6</span>*nMEs)<br>rownames(kMEmat) = probes;<br>colnames(kMEmat) = spaste(<br>    <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;kME.set1.&quot;</span>, <span class="hljs-string">&quot;kME.set2.&quot;</span>, <span class="hljs-string">&quot;p.kME.set1.&quot;</span>, <span class="hljs-string">&quot;p.kME.set2.&quot;</span>, <span class="hljs-string">&quot;Z.kME.meta.&quot;</span>, <span class="hljs-string">&quot;p.kME.meta&quot;</span>),<br>    <span class="hljs-built_in">rep</span>(MEnames, <span class="hljs-built_in">rep</span>(<span class="hljs-number">6</span>, nMEs)))<br><br><br><br>info = data.frame(Probe = probes, GeneSymbol = annot$gene_symbol[probes2annot],<br>             EntrezID = annot$LocusLinkID[probes2annot],<br>             ModuleLabel = moduleLabels,<br>             ModuleColor = labels2colors(moduleLabels),<br>             GSmat,<br>             kMEmat);<br>write.csv(info, file = <span class="hljs-string">&quot;consensusAnalysis-CombinedNetworkResults.csv&quot;</span>,<br>          row.names = <span class="hljs-literal">FALSE</span>, <span class="hljs-built_in">quote</span> = <span class="hljs-literal">FALSE</span>);<br></code></pre></td></tr></table></figure></div>
<h2 id="5-Relationships-among-modules-and-traits">5. Relationships among modules and traits</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><br><span class="hljs-comment">## Create a variable weight that will hold just the body weight of mice in both sets</span><br>weight = vector(mode = <span class="hljs-string">&quot;list&quot;</span>, <span class="hljs-built_in">length</span> = nSets);<br><span class="hljs-keyword">for</span> (set <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nSets)<br>&#123;<br>  weight[[set]] = <span class="hljs-built_in">list</span>(data = as.data.frame(Traits[[set]]$data$weight_g));<br>  <span class="hljs-built_in">names</span>(weight[[set]]$data) = <span class="hljs-string">&quot;weight&quot;</span><br>&#125;<br><span class="hljs-comment">## Recalculate consMEs to give them color names</span><br>consMEsC = multiSetMEs(multiExpr, universalColors = moduleColors);<br><span class="hljs-comment">## We add the weight trait to the eigengenes and order them by consesus hierarchical clustering:</span><br>MET = consensusOrderMEs(addTraitToMEs(consMEsC, weight));<br><br>sizeGrWindow(<span class="hljs-number">8</span>,<span class="hljs-number">10</span>);<br><span class="hljs-comment">##png(file = &quot;EigengeneNetworks.png&quot;, width= 480, height = 600);</span><br>par(cex = <span class="hljs-number">0.9</span>)<br>plotEigengeneNetworks(MET, setLabels, marDendro = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-number">2</span>,<span class="hljs-number">2</span>,<span class="hljs-number">1</span>), marHeatmap = <span class="hljs-built_in">c</span>(<span class="hljs-number">3</span>,<span class="hljs-number">3</span>,<span class="hljs-number">2</span>,<span class="hljs-number">1</span>),<br>                      zlimPreservation = <span class="hljs-built_in">c</span>(<span class="hljs-number">0.5</span>, <span class="hljs-number">1</span>), xLabelsAngle = <span class="hljs-number">90</span>)<br><span class="hljs-comment">## dev.off();</span><br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/dwvPIZEYDp19SJm.png" alt="EigengeneNetworks"></p>
</div><div class="article-licensing box"><div class="licensing-title"><p>WGCNA Tutorial 2</p><p><a href="https://karobben.github.io/2020/06/10/R/WGCNA_2/">https://karobben.github.io/2020/06/10/R/WGCNA_2/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2020-06-10</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2024-01-11</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/Bioinformatics/">Bioinformatics</a><a class="link-muted mr-2" rel="tag" href="/tags/R/">R</a><a class="link-muted mr-2" rel="tag" href="/tags/WGCNA/">WGCNA</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2020/06/10/R/WGCNA/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">WGCNA - 實戰</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2020/06/10/R/nCov-visualization/"><span class="level-item">nCov visualization</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">536</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">151</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">159</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#WGCNA-Tutorial-2"><span class="level-left"><span class="level-item">1</span><span class="level-item">WGCNA Tutorial 2</span></span></a></li><li><a class="level is-mobile" href="#Install"><span class="level-left"><span class="level-item">2</span><span class="level-item">Install</span></span></a></li><li><a class="level is-mobile" href="#1-Data-Preparation"><span class="level-left"><span class="level-item">3</span><span class="level-item">1. Data Preparation</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1-1-Loading-DataSet"><span class="level-left"><span class="level-item">3.1</span><span class="level-item">1.1 Loading DataSet</span></span></a></li><li><a class="level is-mobile" href="#1-2-Groups-in-List"><span class="level-left"><span class="level-item">3.2</span><span class="level-item">1.2 Groups in List</span></span></a></li><li><a class="level is-mobile" href="#1-3-Rudimentary-data-cleaning-and-outlier-removal"><span class="level-left"><span class="level-item">3.3</span><span class="level-item">1.3 Rudimentary data cleaning and outlier removal</span></span></a></li><li><a class="level is-mobile" href="#1-4-Cluster"><span class="level-left"><span class="level-item">3.4</span><span class="level-item">1.4 Cluster</span></span></a></li><li><a class="level is-mobile" href="#1-5-Remove-Outlier"><span class="level-left"><span class="level-item">3.5</span><span class="level-item">1.5 Remove Outlier</span></span></a></li><li><a class="level is-mobile" href="#1-6-Loading-Clinical-Trait-Data"><span class="level-left"><span class="level-item">3.6</span><span class="level-item">1.6 Loading Clinical Trait Data</span></span></a></li></ul></li><li><a class="level is-mobile" href="#2-Network-Construction-and-Module-Detection"><span class="level-left"><span class="level-item">4</span><span class="level-item">2. Network Construction and Module Detection</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#2-1-One-Step-Network-Construction-and-Module-Detection"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">2.1 One-Step Network Construction and Module Detection</span></span></a></li><li><a class="level is-mobile" href="#2-2-Network-construction-and-consensus-module-detection"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">2.2  Network construction and consensus module detection</span></span></a></li><li><a class="level is-mobile" href="#2-3-Model-Extract"><span class="level-left"><span class="level-item">4.3</span><span class="level-item">2.3 Model Extract</span></span></a></li></ul></li><li><a class="level is-mobile" href="#3-Relating-the-consensus-modules-to-female-set-specific-modules"><span class="level-left"><span class="level-item">5</span><span class="level-item">3. Relating the consensus modules to female set-specific modules</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#3-1-datExpr"><span class="level-left"><span class="level-item">5.1</span><span class="level-item">3.1 datExpr</span></span></a></li><li><a class="level is-mobile" href="#3-2-Relating-consensus-modules-to-female-set-specific-modules"><span class="level-left"><span class="level-item">5.2</span><span class="level-item">3.2 Relating consensus modules to female set-specific modules</span></span></a></li></ul></li><li><a class="level is-mobile" href="#4-Consensus-module-to-traits"><span class="level-left"><span class="level-item">6</span><span class="level-item">4. Consensus module to traits</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#4-1-Correlation"><span class="level-left"><span class="level-item">6.1</span><span class="level-item">4.1 Correlation</span></span></a></li><li><a class="level is-mobile" href="#4-2-Exporting-results-of-the-network-analysis"><span class="level-left"><span class="level-item">6.2</span><span class="level-item">4.2 Exporting results of the network analysis</span></span></a></li></ul></li><li><a class="level is-mobile" href="#5-Relationships-among-modules-and-traits"><span class="level-left"><span class="level-item">7</span><span class="level-item">5. Relationships among modules and traits</span></span></a></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/AI/"><span class="tag">AI</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Algorithm/"><span class="tag">Algorithm</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Antibody/"><span class="tag">Antibody</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">79</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">44</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/CLI-Tools/"><span class="tag">CLI Tools</span><span class="tag">15</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data-Science/"><span class="tag">Data Science</span><span class="tag">16</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fasta/"><span class="tag">Fasta</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immunology/"><span class="tag">Immunology</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">28</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">33</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">63</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein/"><span class="tag">Protein</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein-Structure/"><span class="tag">Protein Structure</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyMol/"><span class="tag">PyMol</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">131</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python-Plot/"><span class="tag">Python Plot</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python-Seaborn/"><span class="tag">Python Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Pyvista/"><span class="tag">Pyvista</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">86</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripting/"><span class="tag">Scripting</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sequencing/"><span class="tag">Sequencing</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">48</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/VCF/"><span class="tag">VCF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGS/"><span class="tag">WGS</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dock/"><span class="tag">dock</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/git/"><span class="tag">git</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/protein/"><span class="tag">protein</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://imgur.com/cm7mDjj.png" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">78</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protein-Structure/"><span class="level-start"><span class="level-item">Protein Structure</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">29</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fasta-q/"><span class="level-start"><span class="level-item">Fasta/q</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/WGS/"><span class="level-start"><span class="level-item">WGS</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">41</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">11</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">4</span></span></a><ul><li><a class="level is-mobile" href="/categories/Machine-Learning/LM/"><span class="level-start"><span class="level-item">LM</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Machine-Learning/LM/Protein/"><span class="level-start"><span class="level-item">Protein</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Machine-Learning/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Machine-Learning/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">146</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">15</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Immunity/"><span class="level-start"><span class="level-item">Immunity</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">85</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/AI/"><span class="level-start"><span class="level-item">AI</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">12</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/other/"><span class="level-start"><span class="level-item">other</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immunology/"><span class="level-start"><span class="level-item">Immunology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">56</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">39</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2024 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>