<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>clusterProfiler - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="R package for Bioinformatics; made by Doc. Yu"><meta property="og:type" content="blog"><meta property="og:title" content="clusterProfiler"><meta property="og:url" content="https://karobben.github.io/2020/06/14/R/clusterProfiler/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="R package for Bioinformatics; made by Doc. Yu"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://s1.ax1x.com/2020/06/14/NSYpJP.md.png"><meta property="article:published_time" content="2020-06-14T05:00:00.000Z"><meta property="article:modified_time" content="2024-01-12T00:29:16.315Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="Bioinformatics"><meta property="article:tag" content="R"><meta property="article:tag" content="GO"><meta property="article:tag" content="KEGG"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://s1.ax1x.com/2020/06/14/NSYpJP.md.png"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2020/06/14/R/clusterProfiler/"},"headline":"Karobben","image":["https://s1.ax1x.com/2020/06/14/NSYpJP.md.png"],"datePublished":"2020-06-14T05:00:00.000Z","dateModified":"2024-01-12T00:29:16.315Z","author":{"@type":"Person","name":"Karobben"},"description":"R package for Bioinformatics; made by Doc. Yu"}</script><link rel="canonical" href="https://karobben.github.io/2020/06/14/R/clusterProfiler/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><div class="card-image"><span class="image is-7by3"><img class="fill" src="https://s1.ax1x.com/2020/06/14/NSYpJP.md.png" alt="clusterProfiler"></span><span class="ccopy" style="position: absolute; bottom: 0; right: 0; padding-right:3px; padding-left:4px; background-color: rgba(255, 255, 255, 0.6)"></span></div><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2020-06-14T05:00:00.000Z" title="6/14/2020, 12:00:00 AM">2020-06-14</time></span><span class="level-item">Updated&nbsp;<time dateTime="2024-01-12T00:29:16.315Z" title="1/11/2024, 6:29:16 PM">2024-01-11</time></span><span class="level-item"><a class="link-muted" href="/categories/R/">R</a><span> / </span><a class="link-muted" href="/categories/R/Bio/">Bio</a><span> / </span><a class="link-muted" href="/categories/R/Bio/Anno-Enrich/">Anno/Enrich</a></span><span class="level-item">25 minutes read (About 3689 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">clusterProfiler</h1><div class="content"><h2 id="clusterProfiler">clusterProfiler</h2>
<p>Tutorial: <a target="_blank" rel="noopener" href="https://yulab-smu.github.io/clusterProfiler-book/">yulab</a></p>
<h2 id="Install">Install</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">BiocManager::install(<span class="hljs-string">&quot;clusterProfiler&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<h2 id="Enrichment-analysis-and-GSEA">Enrichment analysis and GSEA</h2>
<blockquote>
<p>Enrichment analysis is a computational method used in bioinformatics to determine whether a given set of genes or proteins is enriched for specific functions, pathways, or biological processes. It involves comparing the input gene set to a reference database, such as Gene Ontology, KEGG, or Reactome, and identifying the over-represented terms using statistical methods.</p>
<p>Gene Set Enrichment Analysis (GSEA) is a type of enrichment analysis that determines whether a particular gene set shows statistically significant differences between two biological states. Unlike traditional enrichment analysis, GSEA considers the entire gene set, rather than individual genes, and evaluates whether the gene set is significantly enriched at the top or bottom of a ranked list of genes based on their differential expression or correlation with a phenotype. GSEA is commonly used in transcriptomics and genomics studies to identify pathways or biological processes that are differentially regulated between disease and control samples.<br>
© ChatGPT</p>
</blockquote>
<h2 id="Go-ontology">Go ontology</h2>
<h3 id="GO-1-Supported-Organism">GO_1. Supported Organism</h3>
<p>For module species which added in <code>OrgDb</code>, we can turn the ID to GO_id;</p>
<p>For other species, you can build your own <code>OrgDb</code> database by following <a target="_blank" rel="noopener" href="https://bioconductor.org/packages/devel/bioc/vignettes/GOSemSim/inst/doc/GOSemSim.html#supported-organisms">GOSemSim</a>.</p>
<p>If genes are already annotated (in <code>data.frame</code> witch gene ID column followed by GO ID), we can use <code>enricher()</code> and <code>geosGO()</code> function to perform over-representation test.</p>
<h3 id="GO-2-Example-for-OrgDb-species">GO_2. Example for <code>OrgDb</code> species</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(clusterProfiler)<br>library(org.Hs.eg.db)<br><br>data(geneList, package=<span class="hljs-string">&quot;DOSE&quot;</span>)<br>gene &lt;- <span class="hljs-built_in">names</span>(geneList)[<span class="hljs-built_in">abs</span>(geneList) &gt; <span class="hljs-number">2</span>]<br>gene.df &lt;- bitr(gene, fromType = <span class="hljs-string">&quot;ENTREZID&quot;</span>,<br>        toType = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;ENSEMBL&quot;</span>, <span class="hljs-string">&quot;SYMBOL&quot;</span>),<br>        OrgDb = org.Hs.eg.db)<br>head(gene.df)<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="APACHE"><figure class="iseeu highlight /apache"><table><tr><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">ENTREZID</span>         ENSEMBL SYMBOL<br><span class="hljs-attribute">1</span>     <span class="hljs-number">4312</span> ENSG<span class="hljs-number">00000196611</span>   MMP<span class="hljs-number">1</span><br><span class="hljs-attribute">2</span>     <span class="hljs-number">8318</span> ENSG<span class="hljs-number">00000093009</span>  CDC<span class="hljs-number">45</span><br><span class="hljs-attribute">3</span>    <span class="hljs-number">10874</span> ENSG<span class="hljs-number">00000109255</span>    NMU<br><span class="hljs-attribute">4</span>    <span class="hljs-number">55143</span> ENSG<span class="hljs-number">00000134690</span>  CDCA<span class="hljs-number">8</span><br><span class="hljs-attribute">5</span>    <span class="hljs-number">55388</span> ENSG<span class="hljs-number">00000065328</span>  MCM<span class="hljs-number">10</span><br><span class="hljs-attribute">6</span>      <span class="hljs-number">991</span> ENSG<span class="hljs-number">00000117399</span>  CDC<span class="hljs-number">20</span><br></code></pre></td></tr></table></figure></div>
<table>
<thead>
<tr>
<th style="text-align:left"></th>
<th style="text-align:left">4312</th>
<th style="text-align:left">8318</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">ACCNUM</td>
<td style="text-align:left">AAA35699</td>
<td style="text-align:left">NP_001139410</td>
</tr>
<tr>
<td style="text-align:left">ENTREZID</td>
<td style="text-align:left">ENSG00000196611</td>
<td style="text-align:left">ENSG00000093009</td>
</tr>
<tr>
<td style="text-align:left">ENSEMBL SYMBOL</td>
<td style="text-align:left">MMP1</td>
<td style="text-align:left">CDC45</td>
</tr>
</tbody>
</table>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">ggo &lt;- groupGO(gene     = gene,<br>               OrgDb    = org.Hs.eg.db,<br>               ont      = <span class="hljs-string">&quot;CC&quot;</span>,<br>               level    = <span class="hljs-number">3</span>,<br>               readable = <span class="hljs-literal">TRUE</span>)<br><br>head(ggo)<br>barplot(ggo, showCategory = <span class="hljs-number">20</span>)<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="DNS"><figure class="iseeu highlight /dns"><table><tr><td class="code"><pre><code class="hljs dns">ID                    Description Count GeneRatio<br>GO:<span class="hljs-number">0005886</span> GO:<span class="hljs-number">0005886</span>                plasma membrane    <span class="hljs-number">55</span>    <span class="hljs-number">55</span>/<span class="hljs-number">207</span><br>GO:<span class="hljs-number">0005628</span> GO:<span class="hljs-number">0005628</span>              prospore membrane     <span class="hljs-number">0</span>     <span class="hljs-number">0</span>/<span class="hljs-number">207</span><br>GO:<span class="hljs-number">0005789</span> GO:<span class="hljs-number">0005789</span> endoplasmic reticulum membrane     <span class="hljs-number">8</span>     <span class="hljs-number">8</span>/<span class="hljs-number">207</span><br>...<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/14/kLpExyf9lHI4FJc.png" alt="132"></p>
<h3 id="GO-3-Over-Representation-Test">GO_3. Over-Representation Test</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">ego &lt;- enrichGO(gene          = gene,<br>                universe      = <span class="hljs-built_in">names</span>(geneList),<br>                OrgDb         = org.Hs.eg.db,<br>                ont           = <span class="hljs-string">&quot;CC&quot;</span>,<br>                pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,<br>                pvalueCutoff  = <span class="hljs-number">0.01</span>,<br>                qvalueCutoff  = <span class="hljs-number">0.05</span>,<br>        readable      = <span class="hljs-literal">TRUE</span>)<br>head(ego)<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">ego2 &lt;- enrichGO(gene         = gene.df$ENSEMBL,<br>                OrgDb         = org.Hs.eg.db,<br>                keyType       = <span class="hljs-string">&#x27;ENSEMBL&#x27;</span>,<br>                ont           = <span class="hljs-string">&quot;CC&quot;</span>,<br>                pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,<br>                pvalueCutoff  = <span class="hljs-number">0.01</span>,<br>                qvalueCutoff  = <span class="hljs-number">0.05</span>)<br><br><span class="hljs-comment">## turn ENSEMBL ID to Samble</span><br>ego2_r &lt;- setReadable(ego2, OrgDb = org.Hs.eg.db)<br><br>dotplot(ego2)<br></code></pre></td></tr></table></figure></div>
<table>
<thead>
<tr>
<th>ego</th>
<th>ego2</th>
</tr>
</thead>
<tbody>
<tr>
<td><img src="https://s1.ax1x.com/2020/06/14/tzt3RK.md.png" alt="tzt3RK.md.png"></td>
<td><img src="https://s1.ax1x.com/2020/06/14/tztDRf.png" alt="tztDRf.png"></td>
</tr>
</tbody>
</table>
<h3 id="GO-4-GO-Gene-Set-Enrichment-Analysis">GO_4 GO Gene Set Enrichment Analysis</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">ego3 &lt;- gseGO(geneList     = geneList,<br>              OrgDb        = org.Hs.eg.db,<br>              ont          = <span class="hljs-string">&quot;CC&quot;</span>,<br>              nPerm        = <span class="hljs-number">1000</span>,<br>              minGSSize    = <span class="hljs-number">100</span>,<br>              maxGSSize    = <span class="hljs-number">500</span>,<br>              pvalueCutoff = <span class="hljs-number">0.05</span>,<br>              verbose      = <span class="hljs-literal">FALSE</span>)<br><br>dotplot(ego3)<br></code></pre></td></tr></table></figure></div>
<pre>
ID                              Description setSize enrichmentScore       NES      pvalue   p.adjust    qvalues rank                   leading_edge
GO:0031012 GO:0031012                     extracellular matrix     427      -0.4868578 -2.138854 0.001231527 0.03095063 0.02171974 1797 tags=37%, list=14%, signal=33%
GO:0099568 GO:0099568                       cytoplasmic region     368      -0.3426037 -1.488559 0.001259446 0.03095063 0.02171974 2613 tags=28%, list=21%, signal=23%
</pre>
<p><img src="https://s1.ax1x.com/2020/06/14/NSMQfg.png" alt="NSMQfg.png"></p>
<h2 id="GO-ontology-for-Non-module-Species">GO ontology for Non-module Species</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><br></code></pre></td></tr></table></figure></div>
<h2 id="KEGG-Enrichment">KEGG Enrichment</h2>
<h3 id="KE-1-Over-Representation-Test">KE_1. Over-Representation Test</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(clusterProfiler)<br>search_kegg_organism(<span class="hljs-string">&#x27;ece&#x27;</span>, by=<span class="hljs-string">&#x27;kegg_code&#x27;</span>)<br><br>ecoli &lt;- search_kegg_organism(<span class="hljs-string">&#x27;Escherichia coli&#x27;</span>, by=<span class="hljs-string">&#x27;scientific_name&#x27;</span>)<br><span class="hljs-built_in">dim</span>(ecoli)<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">data(geneList, package=<span class="hljs-string">&quot;DOSE&quot;</span>)<br>gene &lt;- <span class="hljs-built_in">names</span>(geneList)[<span class="hljs-built_in">abs</span>(geneList) &gt; <span class="hljs-number">2</span>]<br><br>kk &lt;- enrichKEGG(gene         = gene,<br>                 organism     = <span class="hljs-string">&#x27;hsa&#x27;</span>,<br>                 pvalueCutoff = <span class="hljs-number">0.05</span>)<br>head(kk)<br></code></pre></td></tr></table></figure></div>
<pre>
ID                                                   Description GeneRatio  BgRatio       pvalue     p.adjust       qvalue                                             geneID Count
hsa04110 hsa04110                                                    Cell cycle     11/93 124/8031 1.590786e-07 3.213388e-05 3.148083e-05 8318/991/9133/890/983/4085/7272/1111/891/4174/9232    11
hsa04114 hsa04114                                                Oocyte meiosis     10/93 128/8031 1.960649e-06 1.980256e-04 1.940011e-04    991/9133/983/4085/51806/6790/891/9232/3708/5241    10
hsa04218 hsa04218                                           Cellular senescence     10/93 160/8031 1.450991e-05 9.375595e-04 9.185054e-04     2305/4605/9133/890/983/51806/1111/891/776/3708    10
hsa04061 hsa04061 Viral protein interaction with cytokine and cytokine receptor      8/93 100/8031 1.856553e-05 9.375595e-04 9.185054e-04           3627/10563/6373/4283/6362/6355/9547/1524     8
hsa03320 hsa03320                                        PPAR signaling pathway      7/93  77/8031 2.765640e-05 1.117319e-03 1.094611e-03                 4312/9415/9370/5105/2167/3158/5346     7
hsa04914 hsa04914                       Progesterone-mediated oocyte maturation      7/93  99/8031 1.392778e-04 4.689019e-03 4.593724e-03                    9133/890/983/4085/6790/891/5241     7
</pre>
<h3 id="KE-2-Gene-Set-Enrichment-Analysis">KE_2. Gene Set Enrichment Analysis</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">kk2 &lt;- gseKEGG(geneList     = geneList,<br>               organism     = <span class="hljs-string">&#x27;hsa&#x27;</span>,<br>               nPerm        = <span class="hljs-number">1000</span>,<br>               minGSSize    = <span class="hljs-number">120</span>,<br>               pvalueCutoff = <span class="hljs-number">0.05</span>,<br>               verbose      = <span class="hljs-literal">FALSE</span>)<br>head(kk2)<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="APACHE"><figure class="iseeu highlight /apache"><table><tr><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">ID</span>                 Description setSize enrichmentScore       NES      pvalue  p.adjust    qvalues rank                   leading_edge<br><span class="hljs-attribute">hsa04510</span> hsa<span class="hljs-number">04510</span>              Focal adhesion     <span class="hljs-number">190</span>      -<span class="hljs-number">0</span>.<span class="hljs-number">4169068</span> -<span class="hljs-number">1</span>.<span class="hljs-number">697372</span> <span class="hljs-number">0</span>.<span class="hljs-number">001418440</span> <span class="hljs-number">0</span>.<span class="hljs-number">0241065</span> <span class="hljs-number">0</span>.<span class="hljs-number">01552815</span> <span class="hljs-number">2183</span> tags=<span class="hljs-number">27</span>%, list=<span class="hljs-number">17</span>%, signal=<span class="hljs-number">22</span>%<br><span class="hljs-attribute">hsa03013</span> hsa<span class="hljs-number">03013</span>               RNA transport     <span class="hljs-number">131</span>       <span class="hljs-number">0</span>.<span class="hljs-number">4116488</span>  <span class="hljs-number">1</span>.<span class="hljs-number">742019</span> <span class="hljs-number">0</span>.<span class="hljs-number">003067485</span> <span class="hljs-number">0</span>.<span class="hljs-number">0241065</span> <span class="hljs-number">0</span>.<span class="hljs-number">01552815</span> <span class="hljs-number">3383</span> tags=<span class="hljs-number">40</span>%, list=<span class="hljs-number">27</span>%, signal=<span class="hljs-number">29</span>%<br><span class="hljs-attribute">hsa05152</span> hsa<span class="hljs-number">05152</span>                Tuberculosis     <span class="hljs-number">162</span>       <span class="hljs-number">0</span>.<span class="hljs-number">3745153</span>  <span class="hljs-number">1</span>.<span class="hljs-number">628558</span> <span class="hljs-number">0</span>.<span class="hljs-number">003164557</span> <span class="hljs-number">0</span>.<span class="hljs-number">0241065</span> <span class="hljs-number">0</span>.<span class="hljs-number">01552815</span> <span class="hljs-number">2823</span> tags=<span class="hljs-number">34</span>%, list=<span class="hljs-number">23</span>%, signal=<span class="hljs-number">27</span>%<br><span class="hljs-attribute">hsa04218</span> hsa<span class="hljs-number">04218</span>         Cellular senescence     <span class="hljs-number">143</span>       <span class="hljs-number">0</span>.<span class="hljs-number">4153718</span>  <span class="hljs-number">1</span>.<span class="hljs-number">761689</span> <span class="hljs-number">0</span>.<span class="hljs-number">003225806</span> <span class="hljs-number">0</span>.<span class="hljs-number">0241065</span> <span class="hljs-number">0</span>.<span class="hljs-number">01552815</span> <span class="hljs-number">1155</span>  tags=<span class="hljs-number">17</span>%, list=<span class="hljs-number">9</span>%, signal=<span class="hljs-number">16</span>%<br><span class="hljs-attribute">hsa05203</span> hsa<span class="hljs-number">05203</span>        Viral carcinogenesis     <span class="hljs-number">164</span>       <span class="hljs-number">0</span>.<span class="hljs-number">3523856</span>  <span class="hljs-number">1</span>.<span class="hljs-number">535177</span> <span class="hljs-number">0</span>.<span class="hljs-number">003246753</span> <span class="hljs-number">0</span>.<span class="hljs-number">0241065</span> <span class="hljs-number">0</span>.<span class="hljs-number">01552815</span> <span class="hljs-number">3112</span> tags=<span class="hljs-number">35</span>%, list=<span class="hljs-number">25</span>%, signal=<span class="hljs-number">26</span>%<br><span class="hljs-attribute">hsa04062</span> hsa<span class="hljs-number">04062</span> Chemokine signaling pathway     <span class="hljs-number">165</span>       <span class="hljs-number">0</span>.<span class="hljs-number">3754101</span>  <span class="hljs-number">1</span>.<span class="hljs-number">632867</span> <span class="hljs-number">0</span>.<span class="hljs-number">003300330</span> <span class="hljs-number">0</span>.<span class="hljs-number">0241065</span> <span class="hljs-number">0</span>.<span class="hljs-number">01552815</span> <span class="hljs-number">1298</span> tags=<span class="hljs-number">21</span>%, list=<span class="hljs-number">10</span>%, signal=<span class="hljs-number">19</span>%<br></code></pre></td></tr></table></figure></div>
<table>
<thead>
<tr>
<th style="text-align:center">kk</th>
<th style="text-align:center">kk2</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2020/06/14/NSY9Rf.md.png" alt="NSY9Rf.md.png"></td>
<td style="text-align:center"><img src="https://s1.ax1x.com/2020/06/14/NSYpJP.md.png" alt="NSYpJP.md.png"></td>
</tr>
</tbody>
</table>
<h3 id="KE-3-KEGG-Module-over-representation-test">KE_3. KEGG Module over-representation test</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">mkk &lt;- enrichMKEGG(gene = gene,<br>                   organism = <span class="hljs-string">&#x27;hsa&#x27;</span>)<br></code></pre></td></tr></table></figure></div>
<h3 id="KE-4-KEGG-Module-Gene-Set-Enrichment-Analysis">KE_4. KEGG Module Gene Set Enrichment Analysis</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">mkk2 &lt;- gseMKEGG(geneList = geneList,<br>                 organism = <span class="hljs-string">&#x27;hsa&#x27;</span>)<br></code></pre></td></tr></table></figure></div>
<h2 id="A-Pipeline-Example">A Pipeline Example</h2>
<p>Result from: edgeR<br>
species: dme</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(clusterProfiler)<br>library(org.Dm.eg.db)<br>library(pathview)<br>library(reshape2)<br>library(stringr)<br>library(ggplot2)<br>library(enrichplot)<br><br>args = commandArgs(trailingOnly=<span class="hljs-literal">TRUE</span>)<br><br>SampleA= args[<span class="hljs-number">1</span>]<span class="hljs-comment">#&quot;G1F1&quot;</span><br>SampleB= args[<span class="hljs-number">2</span>]<span class="hljs-comment">#&quot;G50F&quot;</span><br>Pvalue = <span class="hljs-number">0.01</span><br>logFC= <span class="hljs-number">2</span><br>C_cut = <span class="hljs-number">10</span><br>Level= <span class="hljs-number">2</span><br><br>Sample_Dir = paste(SampleA,<span class="hljs-string">&quot;_VS_&quot;</span>,SampleB,<span class="hljs-string">&quot;/&quot;</span>,sep=<span class="hljs-string">&quot;&quot;</span>)<br><span class="hljs-keyword">for</span>(i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;GO&quot;</span>, <span class="hljs-string">&quot;KEGG&quot;</span>, <span class="hljs-string">&quot;WIKI&quot;</span>, <span class="hljs-string">&quot;Reactome&quot;</span>))&#123;<br>    dir.create(i)<br>    dir.create(paste(i, Sample_Dir,sep=<span class="hljs-string">&#x27;/&#x27;</span>))<br>&#125;<br><br>Gene_list &lt;- read.table(paste(<span class="hljs-string">&quot;RSEM_transcript/transcript.isoform.counts.matrix.&quot;</span>, SampleA, <span class="hljs-string">&quot;_vs_&quot;</span>, SampleB,<span class="hljs-string">&quot;.edgeR.DE_results&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>))<br><br><br>ENTREZID = bitr(row.names(Gene_list), fromType=<span class="hljs-string">&quot;FLYBASE&quot;</span>, toType=<span class="hljs-string">&quot;ENTREZID&quot;</span>, OrgDb=<span class="hljs-string">&quot;org.Dm.eg.db&quot;</span>)<br>FLYBASECG = bitr(row.names(Gene_list), fromType=<span class="hljs-string">&quot;FLYBASE&quot;</span>, toType=<span class="hljs-string">&quot;FLYBASECG&quot;</span>, OrgDb=<span class="hljs-string">&quot;org.Dm.eg.db&quot;</span>)<br>SYMBOL = bitr(row.names(Gene_list), fromType=<span class="hljs-string">&quot;FLYBASE&quot;</span>, toType=<span class="hljs-string">&quot;SYMBOL&quot;</span>, OrgDb=<span class="hljs-string">&quot;org.Dm.eg.db&quot;</span>)<br><br><br>Gene_list$ENTREZID &lt;- ENTREZID$ENTREZID[match(row.names(Gene_list), ENTREZID$FLYBASE)]<br>Gene_list$FLYBASECG &lt;- FLYBASECG$FLYBASECG[match(row.names(Gene_list), FLYBASECG$FLYBASE)]<br>Gene_list$SYMBOL &lt;- SYMBOL$SYMBOL[match(row.names(Gene_list), SYMBOL$FLYBASE)]<br><br>KE2SY &lt;-<span class="hljs-keyword">function</span>(kk_GSEA, COL)&#123;<br>    <span class="hljs-keyword">if</span>( nrow(kk_GSEA@result) &gt;<span class="hljs-number">0</span>)&#123;<br>        <span class="hljs-keyword">for</span>(i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:nrow(kk_GSEA@result)))&#123;<br>            LIST &lt;- kk_GSEA@result[[COL]][i]<br>            kk_GSEA@result[[COL]][i] &lt;- paste(Gene_list$SYMBOL[match(str_remove(str_split(LIST, <span class="hljs-string">&quot;/&quot;</span>)[[<span class="hljs-number">1</span>]], <span class="hljs-string">&quot;Dmel_&quot;</span>),Gene_list$FLYBASECG )], collapse = <span class="hljs-string">&quot;/&quot;</span>)<br>        &#125;        <br>    &#125;<br>    <span class="hljs-built_in">return</span>(kk_GSEA)<br>&#125;<br><br>EN2SY &lt;-<span class="hljs-keyword">function</span>(WikiP_enrich, COL)&#123;<br>    <span class="hljs-keyword">if</span>(<span class="hljs-built_in">length</span>(WikiP_enrich@result[[COL]])&gt;<span class="hljs-number">0</span>)&#123;<br>        <span class="hljs-keyword">for</span>(i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-built_in">length</span>(WikiP_enrich@result[[COL]])))&#123;<br>            LIST &lt;- WikiP_enrich@result[[COL]][i]<br>            WikiP_enrich@result[[COL]][i] &lt;- paste(Gene_list$SYMBOL[match(str_remove(str_split(LIST, <span class="hljs-string">&quot;/&quot;</span>)[[<span class="hljs-number">1</span>]], <span class="hljs-string">&quot;Dmel_&quot;</span>),Gene_list$ENTREZID )], collapse = <span class="hljs-string">&quot;/&quot;</span>)<br>        &#125;<br>    &#125;<br>    <span class="hljs-built_in">return</span>(WikiP_enrich)<br>&#125;<br><br>ggsave_GO &lt;- <span class="hljs-keyword">function</span>(NAME, LEN)&#123;<br>    BASE = <span class="hljs-number">3.8</span><br>    RATE = <span class="hljs-number">0.125</span><br>    <span class="hljs-keyword">if</span> (BASE + RATE*LEN &lt;= <span class="hljs-number">40</span>)&#123;<br>        W = BASE + RATE*LEN<br>    &#125;<span class="hljs-keyword">else</span>&#123;<br>        W = <span class="hljs-number">40</span><br>    &#125;<br>    ggsave(NAME , w = W, h = <span class="hljs-number">8.35</span>, limitsize = <span class="hljs-literal">FALSE</span> )<br>&#125;<br><br>ggsave_GO_enrich &lt;- <span class="hljs-keyword">function</span>(NAME, LEN)&#123;<br>    BASE = <span class="hljs-number">2</span><br>    RATE = <span class="hljs-number">0.1114</span><br>    <span class="hljs-keyword">if</span> (BASE + RATE*LEN &lt;= <span class="hljs-number">40</span>)&#123;<br>        H = BASE + RATE*LEN<br>    &#125;<span class="hljs-keyword">else</span>&#123;<br>        H = <span class="hljs-number">40</span><br>    &#125;<br>    ggsave(NAME , w = <span class="hljs-number">10</span>, h = H, limitsize = <span class="hljs-literal">FALSE</span>)<br>&#125;<br><br>ridgeplot_save &lt;- <span class="hljs-keyword">function</span>(NAME,LEN)&#123;<br>    BASE = <span class="hljs-number">1.02</span><br>    RATE = <span class="hljs-number">0.2</span><br>    <span class="hljs-keyword">if</span> (BASE + RATE*LEN &lt;= <span class="hljs-number">40</span>)&#123;<br>        H = BASE + RATE*LEN<br>    &#125;<span class="hljs-keyword">else</span>&#123;<br>        H = <span class="hljs-number">40</span><br>    &#125;<br>    ggsave(NAME , w = <span class="hljs-number">10</span>, h = H, limitsize = <span class="hljs-literal">FALSE</span>)<br>&#125;<br><br><span class="hljs-comment">## GO ontology</span><br>TMP &lt;- Gene_list[<span class="hljs-built_in">abs</span>(Gene_list$logFC) &gt;=<span class="hljs-number">2</span>,]<br>TMP &lt;- TMP[TMP$PValue &lt;= Pvalue,]<br><br><br>sig_genes = TMP$ENTREZID<br><br>gooc &lt;- groupGO(gene     = sig_genes,<br>               OrgDb    = org.Dm.eg.db,<br>               ont      = <span class="hljs-string">&quot;CC&quot;</span>,<br>               level    = Level,<br>               readable = <span class="hljs-literal">TRUE</span>)<br>goom &lt;- groupGO(gene     = sig_genes,<br>              OrgDb    = org.Dm.eg.db,<br>              ont      = <span class="hljs-string">&quot;MF&quot;</span>,<br>              level    = Level,<br>              readable = <span class="hljs-literal">TRUE</span>)<br>goob &lt;- groupGO(gene     = sig_genes,<br>             OrgDb    = org.Dm.eg.db,<br>             ont      = <span class="hljs-string">&quot;BP&quot;</span>,<br>             level    = Level,<br>             readable = <span class="hljs-literal">TRUE</span>)<br><br>gooc@result$Group = <span class="hljs-string">&quot;CC&quot;</span><br>goom@result$Group = <span class="hljs-string">&quot;MF&quot;</span><br>goob@result$Group = <span class="hljs-string">&quot;BP&quot;</span><br><br><br>GO_TB &lt;- rbind(gooc@result, goom@result, goob@result)<br>GO_TB &lt;- GO_TB[GO_TB$Count!=<span class="hljs-number">0</span>,]<br>File_name = paste(<span class="hljs-string">&quot;Ontology&quot;</span>, SampleA,SampleB,Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>write.csv(GO_TB, paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br><span class="hljs-comment"># [GO ontology plot]</span><br>ggplot(GO_TB, aes(x=Description, y=Count, fill=Group)) +<br>    geom_bar(stat = <span class="hljs-string">&#x27;identity&#x27;</span>) +<br>    facet_grid(~Group, scales = <span class="hljs-string">&#x27;free&#x27;</span>, space = <span class="hljs-string">&#x27;free&#x27;</span>) + theme_bw() +<br>    theme(axis.text.x = element_text(angle = <span class="hljs-number">270</span>, hjust = <span class="hljs-number">0</span>, vjust = <span class="hljs-number">.5</span>), legend.position = <span class="hljs-string">&#x27;none&#x27;</span>, panel.grid =  element_blank(), strip.background = element_rect(fill = <span class="hljs-string">&#x27;white&#x27;</span>))<br>ggsave_GO(paste(<span class="hljs-string">&quot;GO/&quot;</span>, Sample_Dir, File_name, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(GO_TB))<br><br><span class="hljs-comment">## Overrepresentation</span><br>File_name = paste(<span class="hljs-string">&quot;Enrichment&quot;</span>, SampleA,SampleB,Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br><br><br>ego &lt;- enrichGO(gene          = sig_genes,<br>                universe      = Gene_list$ENTREZID,<br>                OrgDb         = org.Dm.eg.db,<br>                ont           = <span class="hljs-string">&quot;All&quot;</span>,<br>                pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,<br>                pvalueCutoff  = <span class="hljs-number">0.01</span>,<br>                qvalueCutoff  = <span class="hljs-number">0.05</span>,<br>        readable      = <span class="hljs-literal">TRUE</span>)<br>head(ego)<br>SIZE = <span class="hljs-built_in">as.numeric</span>(<span class="hljs-built_in">as.character</span>(as.data.frame(str_split_fixed(ego@result$GeneRatio, <span class="hljs-string">&quot;/&quot;</span>,<span class="hljs-number">2</span>))[[<span class="hljs-number">1</span>]]))/<br><span class="hljs-built_in">as.numeric</span>(<span class="hljs-built_in">as.character</span>(as.data.frame(str_split_fixed(ego@result$BgRatio, <span class="hljs-string">&quot;/&quot;</span>,<span class="hljs-number">2</span>))[[<span class="hljs-number">1</span>]]))<br>ego@result$size = SIZE<br>write.csv(ego, paste(<span class="hljs-string">&quot;GO/&quot;</span>, Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br><span class="hljs-comment"># [GO Enrichment plot]</span><br>ggplot(ego@result, aes(x=Count, y=Description)) +<br>    geom_point(aes(size=size, color=pvalue)) + theme_bw() +<br>    facet_grid(ONTOLOGY~., space = <span class="hljs-string">&#x27;free&#x27;</span>, scales = <span class="hljs-string">&#x27;free&#x27;</span>)<br><br>ggsave_GO_enrich(paste(<span class="hljs-string">&quot;GO/&quot;</span>, Sample_Dir, File_name, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(ego@result))<br><br>geneList &lt;- Gene_list$logFC<br><span class="hljs-built_in">names</span>(geneList) &lt;- Gene_list$ENTREZID<br>geneList &lt;- sort(geneList, decreasing = <span class="hljs-built_in">T</span>)<br><br><span class="hljs-keyword">for</span>(GROUP <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;CC&quot;</span>, <span class="hljs-string">&quot;BP&quot;</span>, <span class="hljs-string">&quot;MF&quot;</span>))&#123;<br>    egocc &lt;- enrichGO(gene          = sig_genes,<br>            universe      = Gene_list$ENTREZID,<br>            OrgDb         = org.Dm.eg.db,<br>            ont           = GROUP,<br>            pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,<br>            pvalueCutoff  = <span class="hljs-number">0.01</span>,<br>            qvalueCutoff  = <span class="hljs-number">0.05</span>,<br>            readable      = <span class="hljs-literal">TRUE</span>)<br><br>    write.csv(egocc, paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br>    <span class="hljs-comment"># [GO Network category]</span><br>    Plot_and_Save &lt;-<span class="hljs-keyword">function</span>()&#123;<br>        goplot(egocc)<br>        ggsave( paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;_category&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ),<br>            w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>    <span class="hljs-comment"># [GO Network Genes]</span><br>        cnetplot(egocc, foldChange=geneList)<br>        ggsave( paste(<span class="hljs-string">&quot;GO/&quot;</span>, Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;_genes&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ),<br>        w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>    &#125;<br>    try(Plot_and_Save(),silent=<span class="hljs-literal">TRUE</span>)<br><br>&#125;<br><br><br><span class="hljs-comment"># GO Gene Set Enrichment Analysis</span><br>File_name = paste(<span class="hljs-string">&quot;GSEA&quot;</span>,Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br><br><span class="hljs-keyword">for</span>(GROUP <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;CC&quot;</span>, <span class="hljs-string">&quot;BP&quot;</span>, <span class="hljs-string">&quot;MF&quot;</span>))&#123;<br>    GOgse_CC &lt;- gseGO(geneList     = geneList,<br>                  OrgDb        = org.Dm.eg.db,<br>                  ont          = GROUP,<br>                  minGSSize    = <span class="hljs-number">100</span>,<br>                  maxGSSize    = <span class="hljs-number">500</span>,<br>                  eps = <span class="hljs-number">1e-10</span>,<br>                  pvalueCutoff = <span class="hljs-number">0.05</span>,<br>                  verbose      = <span class="hljs-literal">FALSE</span>)<br>    write.csv(GOgse_CC, paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br>    ridgeplot(GOgse_CC)<br>    ridgeplot_save(paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_&quot;</span>, GROUP,<span class="hljs-string">&quot;_ridgeplot.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(GOgse_CC@result))<br>    Plot_and_Save &lt;-<span class="hljs-keyword">function</span>()&#123;<br>        goplot(GOgse_CC)<br>        ggsave( paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;_category&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ),<br>        w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>        cnetplot(GOgse_CC, foldChange=geneList)<br>        ggsave( paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;_genes&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ),<br>        w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>    &#125;<br>    try(Plot_and_Save(),silent=<span class="hljs-literal">TRUE</span>)<br><br>    <span class="hljs-comment"># [GO GSEA upsetplot]</span><br>    upsetplot(GOgse_CC)<br>    ggsave( paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, GROUP, <span class="hljs-string">&quot;_upset&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ),<br>        w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>    <span class="hljs-comment"># [GO GSEA Escore]</span><br>    <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:nrow(GOgse_CC@result)))&#123;<br>        gseaplot2(GOgse_CC, geneSetID = i, title = GOgse_CC$Description[i], color = <span class="hljs-string">&#x27;salmon&#x27;</span>, pvalue_table = <span class="hljs-literal">TRUE</span>)<br>        ggsave(paste(<span class="hljs-string">&quot;GO/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, str_split(GOgse_CC@result$ID[i],<span class="hljs-string">&quot;:&quot;</span>)[[<span class="hljs-number">1</span>]][<span class="hljs-number">2</span>], <span class="hljs-string">&quot;_GSEA&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">7.5</span>, h = <span class="hljs-number">5</span>)<br>    &#125;<br>&#125;<br><br><span class="hljs-comment">#########################</span><br><span class="hljs-comment">## KEGG                 #</span><br><span class="hljs-comment">#########################</span><br><br><span class="hljs-comment"># gene set enrichment analysis</span><br>File_name = paste(<span class="hljs-string">&quot;GSEA&quot;</span>,Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>geneList_kk &lt;- Gene_list$logFC<br><span class="hljs-built_in">names</span>(geneList_kk) &lt;- paste(<span class="hljs-string">&quot;Dmel&quot;</span>, Gene_list$FLYBASECG, sep=<span class="hljs-string">&quot;_&quot;</span>)<br>geneList_kk &lt;- sort(geneList_kk, decreasing = <span class="hljs-built_in">T</span>)<br><br>kk_GSEA &lt;- gseKEGG(geneList     = geneList_kk,<br>               organism     = <span class="hljs-string">&#x27;dme&#x27;</span>,<br>               minGSSize    = <span class="hljs-number">10</span>,<br>               pvalueCutoff = <span class="hljs-number">0.05</span>,<br>               verbose      = <span class="hljs-literal">FALSE</span>)<br><br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:nrow(kk_GSEA@result)))&#123;<br>    gseaplot2(kk_GSEA, geneSetID = i, title = kk_GSEA$Description[i], color = <span class="hljs-string">&#x27;salmon&#x27;</span>, pvalue_table = <span class="hljs-literal">TRUE</span>)<br>    ggsave(paste(<span class="hljs-string">&quot;KEGG/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, kk_GSEA@result$ID[i], <span class="hljs-string">&quot;_GSEA&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">7.5</span>, h = <span class="hljs-number">5</span>)<br>&#125;<br><br><br>kk_GSEA2 &lt;- KE2SY(kk_GSEA, <span class="hljs-string">&quot;core_enrichment&quot;</span>)<br><br>write.csv(kk_GSEA2, paste(<span class="hljs-string">&quot;KEGG/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br><br>geneList_smb &lt;- Gene_list$logFC<br><span class="hljs-built_in">names</span>(geneList_smb) &lt;-Gene_list$SYMBOL<br>geneList_smb &lt;- sort(geneList_smb, decreasing = <span class="hljs-built_in">T</span>)<br><br>Plot_and_Save &lt;-<span class="hljs-keyword">function</span>()&#123;<br>    cnetplot(kk_GSEA2, foldChange=geneList_smb)<br>    ggsave( paste(<span class="hljs-string">&quot;KEGG/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_genes&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>&#125;<br>try(Plot_and_Save(),silent=<span class="hljs-literal">TRUE</span>)<br><br><br>sig_genes_kk =  paste(<span class="hljs-string">&quot;Dmel&quot;</span>, TMP$FLYBASECG, sep=<span class="hljs-string">&quot;_&quot;</span>)<br><br><span class="hljs-comment"># over-representation analysis</span><br>File_name = paste(<span class="hljs-string">&quot;Enrichment&quot;</span>, Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>kk_Enrich &lt;- enrichKEGG(gene = sig_genes_kk,<br>                   universe = <span class="hljs-built_in">names</span>(geneList_kk),<br>                   organism = <span class="hljs-string">&#x27;dme&#x27;</span>,<br>                   pvalueCutoff = <span class="hljs-number">0.05</span>)<br>write.csv(kk_Enrich, paste(<span class="hljs-string">&quot;KEGG/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br>ggplot(kk_Enrich@result, aes(x=Count, y=Description)) +<br>    geom_point(aes(size=GeneRatio, color=pvalue)) + theme_bw()<br>ggsave_GO_enrich(paste(<span class="hljs-string">&quot;KEGG/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(kk_Enrich@result))<br><br><br>Plot_and_Save &lt;-<span class="hljs-keyword">function</span>()&#123;<br>    cnetplot(kk_Enrich, foldChange=geneList_kk)<br>    ggsave( paste(<span class="hljs-string">&quot;KEGG/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_genes&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>&#125;<br>try(Plot_and_Save(),silent=<span class="hljs-literal">TRUE</span>)<br><br><br>neList_kk &lt;- Gene_list$logFC<br><span class="hljs-built_in">names</span>(geneList_kk) &lt;-Gene_list$ENTREZID<br>geneList_kk &lt;- sort(geneList_kk, decreasing = <span class="hljs-built_in">T</span>)<br><br>library(<span class="hljs-string">&quot;pathview&quot;</span>)<br><span class="hljs-comment">#dme04624 &lt;- pathview(gene.data  = geneList_kk,</span><br><span class="hljs-comment">#                        pathway.id = &quot;04624&quot;,</span><br><span class="hljs-comment">#                        species    = &quot;dme&quot;,</span><br><span class="hljs-comment">#                        out.prefix = paste(&quot;KEGG/&quot;,Sample_Dir, sep=&quot;&quot;),</span><br><span class="hljs-comment">#                        limit      =2.5)</span><br><br><span class="hljs-comment">## WikiPathways</span><br>File_name = paste(<span class="hljs-string">&quot;Enrichment&quot;</span>, Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>WikiP_enrich &lt;- enrichWP(sig_genes, universe = <span class="hljs-built_in">names</span>(geneList_kk), organism = <span class="hljs-string">&quot;Drosophila melanogaster&quot;</span>)<br>ggplot(WikiP_enrich@result, aes(x=Count, y=Description)) +<br>    geom_point(aes(size=GeneRatio, color=pvalue)) + theme_bw()<br>ggsave_GO_enrich(paste(<span class="hljs-string">&quot;WIKI/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(kk_Enrich@result))<br><br>WikiP_enrich2 &lt;- EN2SY(WikiP_enrich, <span class="hljs-string">&quot;geneID&quot;</span>)<br>write.csv(WikiP_enrich2@result, paste(<span class="hljs-string">&quot;WIKI/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br>Plot_and_Save &lt;-<span class="hljs-keyword">function</span>()&#123;<br>    cnetplot(WikiP_enrich2, foldChange=geneList_smb)<br>    ggsave( paste(<span class="hljs-string">&quot;WIKI/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_genes&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">20</span>, h = <span class="hljs-number">10.9</span> )<br>&#125;<br>try(Plot_and_Save(),silent=<span class="hljs-literal">TRUE</span>)<br><br><br>File_name = paste(<span class="hljs-string">&quot;GSEA&quot;</span>,Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>WikiP_gse &lt;- gseWP(geneList, organism = <span class="hljs-string">&quot;Drosophila melanogaster&quot;</span>)<br>WikiP_gse2 &lt;- EN2SY(WikiP_gse, <span class="hljs-string">&quot;core_enrichment&quot;</span>)<br><br><span class="hljs-keyword">if</span> (nrow(WikiP_gse@result) &gt;<span class="hljs-number">0</span> )&#123;<br>    <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:nrow(WikiP_gse@result)))&#123;<br>        gseaplot2(WikiP_gse, geneSetID = i, title = WikiP_gse$Description[i], color = <span class="hljs-string">&#x27;salmon&#x27;</span>, pvalue_table = <span class="hljs-literal">TRUE</span>)<br>        ggsave(paste(<span class="hljs-string">&quot;WIKI/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, WikiP_gse@result$ID[i], <span class="hljs-string">&quot;_GSEA&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">7.5</span>, h = <span class="hljs-number">5</span>)<br>    &#125;<br>&#125;<br><br>write.csv(WikiP_gse2@result, paste(<span class="hljs-string">&quot;WIKI/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br>Plot_and_Save &lt;-<span class="hljs-keyword">function</span>()&#123;<br>    ridgeplot(WikiP_gse)<br>    ridgeplot_save(paste(<span class="hljs-string">&quot;WIKI/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_&quot;</span>, GROUP,<span class="hljs-string">&quot;_ridgeplot.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(WikiP_gse@result))<br>&#125;<br>try(Plot_and_Save(),silent=<span class="hljs-literal">TRUE</span>)<br><br><br><br><span class="hljs-comment">## Reactome Pathway</span><br>library(ReactomePA)<br>File_name = paste(<span class="hljs-string">&quot;Enrichment&quot;</span>, Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>Reactome_enrich &lt;- enrichPathway(gene= sig_genes, pvalueCutoff = <span class="hljs-number">0.05</span>,<br>    readable=<span class="hljs-literal">TRUE</span>, organism =<span class="hljs-string">&quot;fly&quot;</span>, universe = <span class="hljs-built_in">names</span>(geneList))<br>write.csv(Reactome_enrich, paste(<span class="hljs-string">&quot;Reactome/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br>Reactome_enrich@result &lt;- Reactome_enrich@result[Reactome_enrich@result$pvalue&lt;=<span class="hljs-number">0.05</span>,]<br>ggplot(Reactome_enrich@result, aes(x=Count, y=Description)) +<br>    geom_point(aes(size=GeneRatio, color=pvalue)) + theme_bw()<br>ggsave_GO_enrich(paste(<span class="hljs-string">&quot;Reactome/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(kk_Enrich@result))<br><br><br>File_name = paste(<span class="hljs-string">&quot;GSEA&quot;</span>,Pvalue, logFC, sep=<span class="hljs-string">&quot;_&quot;</span> )<br>Reactome_gse &lt;- gsePathway(geneList,<br>                pvalueCutoff = <span class="hljs-number">0.05</span>,<br>                organism = <span class="hljs-string">&quot;fly&quot;</span>,<br>                pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>)<br><span class="hljs-comment"># [Reactome_gse ridgeplot]</span><br>ridgeplot(Reactome_gse)<br>ridgeplot_save(paste(<span class="hljs-string">&quot;Reactome/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;_&quot;</span>, GROUP,<span class="hljs-string">&quot;_ridgeplot.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), nrow(Reactome_gse@result))<br>write.csv(Reactome_gse, paste(<span class="hljs-string">&quot;Reactome/&quot;</span>,Sample_Dir, File_name, <span class="hljs-string">&quot;.csv&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), row.names=<span class="hljs-built_in">F</span>)<br><br><br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:nrow(Reactome_gse@result)))&#123;<br>    gseaplot2(Reactome_gse, geneSetID = i, title = Reactome_gse$Description[i], color = <span class="hljs-string">&#x27;salmon&#x27;</span>, pvalue_table = <span class="hljs-literal">TRUE</span>)<br>    ggsave(paste(<span class="hljs-string">&quot;Reactome/&quot;</span>,Sample_Dir, File_name,  <span class="hljs-string">&quot;_&quot;</span>, Reactome_gse@result$ID[i], <span class="hljs-string">&quot;_GSEA&quot;</span>, <span class="hljs-string">&quot;.png&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span> ), w = <span class="hljs-number">7.5</span>, h = <span class="hljs-number">5</span>)<br>&#125;<br><br></code></pre></td></tr></table></figure></div>
<p><a target="_blank" rel="noopener" href="https://bioinformaticsbreakdown.com/how-to-gsea/">123</a><br>
<a target="_blank" rel="noopener" href="https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/">123</a></p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/07/x3Zf1g.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">GO ontology plot</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/07/x8d0hT.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">GO Enrichment plot</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/07/x8dqHI.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">GO Network category</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/07/x8dj4f.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">GO Network Genes</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/07/x8wpvQ.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">GO GSEA upsetplot</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/08/x8wibn.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">GO GSEA Escore</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/08/x80lQg.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">pathview</td>
</tr>
<tr>
<td style="text-align:center"><img src="https://s1.ax1x.com/2022/10/08/x8DMrQ.png" alt=""></td>
</tr>
<tr>
<td style="text-align:center">Reactome_gse ridgeplot</td>
</tr>
</tbody>
</table>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">##############</span><br><span class="hljs-comment">####eg</span><br><span class="hljs-comment">#############</span><br><br>eg = bitr(x, fromType=<span class="hljs-string">&quot;SYMBOL&quot;</span>, toType=<span class="hljs-string">&quot;ENTREZID&quot;</span>, OrgDb=<span class="hljs-string">&quot;org.Hs.eg.db&quot;</span>)<br>eg = bitr(x, fromType=<span class="hljs-string">&quot;ENTREZID&quot;</span>, toType=<span class="hljs-string">&quot;SYMBOL&quot;</span>, OrgDb=<span class="hljs-string">&quot;org.Hs.eg.db&quot;</span>)<br>head(eg)<br><br>gene &lt;- eg[[<span class="hljs-number">2</span>]]<br><br><br>barplot(ggio, drop=<span class="hljs-literal">TRUE</span>, showCategory=<span class="hljs-number">122</span>)<br><span class="hljs-comment">#####################################################################</span><br><br><br><br><span class="hljs-comment">###bitr_kegg: converting biological IDs using KEGG API</span><br><br>data(x)<br>hg &lt;- x[[<span class="hljs-number">1</span>]]<br>head(hg)<br>eg2np &lt;- bitr_kegg(hg, fromType=<span class="hljs-string">&#x27;kegg&#x27;</span>, toType=<span class="hljs-string">&#x27;ncbi-proteinid&#x27;</span>, organism=<span class="hljs-string">&#x27;hsa&#x27;</span>)<br>head(eg2np)<br>bitr_kegg(<span class="hljs-string">&quot;Z5100&quot;</span>, fromType=<span class="hljs-string">&quot;kegg&quot;</span>, toType=<span class="hljs-string">&#x27;ncbi-geneid&#x27;</span>, organism=<span class="hljs-string">&#x27;ece&#x27;</span>)<br>bitr_kegg(<span class="hljs-string">&quot;Z5100&quot;</span>, fromType=<span class="hljs-string">&quot;kegg&quot;</span>, toType=<span class="hljs-string">&#x27;ncbi-proteinid&#x27;</span>, organism=<span class="hljs-string">&#x27;ece&#x27;</span>)<br>bitr_kegg(<span class="hljs-string">&quot;Z5100&quot;</span>, fromType=<span class="hljs-string">&quot;kegg&quot;</span>, toType=<span class="hljs-string">&#x27;uniprot&#x27;</span>, organism=<span class="hljs-string">&#x27;ece&#x27;</span>)<br><br><span class="hljs-comment">################################################################################</span><br><span class="hljs-comment">############ GO classification</span><br><span class="hljs-comment">################################################################################</span><br>gene &lt;- <span class="hljs-built_in">names</span>(geneList)[<span class="hljs-built_in">abs</span>(geneList) &gt; <span class="hljs-number">2</span>]<br><br>gobp &lt;- groupGO(gene     = gene,<br>               OrgDb    = org.Hs.eg.db,<br>               ont      = <span class="hljs-string">&quot;BP&quot;</span>,<br>               level    = <span class="hljs-number">2</span>,<br>               readable = <span class="hljs-literal">TRUE</span>)<br><br>head(ggo)<br><br>go &lt;- groupGO(gene = gene, OrgDb = org.Hs.eg.db, ont = <span class="hljs-string">&quot;BP&quot;</span>, level = <span class="hljs-number">2</span>, readable = <span class="hljs-literal">TRUE</span>)<br>barplot(go, drop=<span class="hljs-literal">TRUE</span>, showCategory=<span class="hljs-number">122</span>)<br><span class="hljs-comment">##############################</span><br>GO over-representation test<br><span class="hljs-comment">###############################</span><br>ego1 &lt;- enrichGO(gene =gene, universe= <span class="hljs-built_in">names</span>(geneList), ont = <span class="hljs-string">&quot;CC&quot;</span>,pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,pvalueCutoff  = <span class="hljs-number">0.01</span>,qvalueCutoff = <span class="hljs-number">0.05</span>,readable = <span class="hljs-literal">TRUE</span>)<br>ego2 &lt;- enrichGO(gene =gene, universe= <span class="hljs-built_in">names</span>(geneList), ont = <span class="hljs-string">&quot;BP&quot;</span>,pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,pvalueCutoff  = <span class="hljs-number">0.01</span>,qvalueCutoff = <span class="hljs-number">0.05</span>,readable = <span class="hljs-literal">TRUE</span>)<br>ego3 &lt;- enrichGO(gene =gene, universe= <span class="hljs-built_in">names</span>(geneList), ont = <span class="hljs-string">&quot;MF&quot;</span>,pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,pvalueCutoff  = <span class="hljs-number">0.01</span>,qvalueCutoff = <span class="hljs-number">0.05</span>,readable = <span class="hljs-literal">TRUE</span>)<br><br>head(ego)<br><span class="hljs-comment">###################################</span><br><span class="hljs-comment">##################################</span><br><span class="hljs-comment">##################################</span><br>source(<span class="hljs-string">&quot;http://bioconductor.org/biocLite.R&quot;</span>)<br>library(clusterProfiler)<br>library(org.Hs.eg.db)<br>keytypes(org.Hs.eg.db)<br><br><br><br><br><br>KEGG<br>kk &lt;- enrichKEGG(gene = gene, organism = <span class="hljs-string">&#x27;human&#x27;</span>, pvalueCutoff = <span class="hljs-number">0.05</span>)<br><br>head(kk)<br>barplot(kk, drop=<span class="hljs-literal">TRUE</span>, showCategory=<span class="hljs-number">122</span>)<br><br><br>kk2 &lt;- gseKEGG(geneList = geneList, organism = <span class="hljs-string">&#x27;hsa&#x27;</span>, nPerm = <span class="hljs-number">1000</span>, minGSSize = <span class="hljs-number">120</span>, pvalueCutoff = <span class="hljs-number">100</span>,verbose = <span class="hljs-literal">FALSE</span>)<br>ead(kk2)<br><br><span class="hljs-comment">####</span><br>kk2 &lt;- gseKEGG(geneList = geneList, organism = <span class="hljs-string">&#x27;hsa&#x27;</span>, nPerm = <span class="hljs-number">10</span>, minGSSize = <span class="hljs-number">10</span>, pvalueCutoff = <span class="hljs-number">100</span>,verbose = <span class="hljs-literal">FALSE</span>)<br><br>gseaplot(kk2, geneSetID = <span class="hljs-string">&quot;hsa05200&quot;</span>)<br><br>hsa05205<br>hsa05165<br><span class="hljs-comment">##################################################################</span><br><span class="hljs-comment">#############pathview</span><br><span class="hljs-comment">##################################################################</span><br><br>library(clusterProfiler)<br>library(org.Hs.eg.db)<br>library(<span class="hljs-string">&quot;pathview&quot;</span>)<br><br>hsa05418<br> p  &lt;- pathview(gene.data  = geneList,<br>                     pathway.id = <span class="hljs-string">&quot;hsa04976&quot;</span>,<br>                     species    = <span class="hljs-string">&quot;hsa&quot;</span>,<br>                     limit      = <span class="hljs-built_in">list</span>(gene=<span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(geneList)), cpd=<span class="hljs-number">1</span>))<br><br>pathview(gene.data  = geneList, pathway.id = <span class="hljs-string">&quot;hsa04976&quot;</span>, species = <span class="hljs-string">&quot;hsa&quot;</span>, limit = <span class="hljs-built_in">list</span>(gene=<span class="hljs-built_in">max</span>(<span class="hljs-number">5</span>), cpd=<span class="hljs-number">1</span>))<br>pathview(gene.data  = geneList, pathway.id = i, species = <span class="hljs-string">&quot;hsa&quot;</span>, limit = <span class="hljs-built_in">list</span>(gene=<span class="hljs-built_in">max</span>(<span class="hljs-number">5</span>), cpd=<span class="hljs-number">1</span>))<br><br><br><span class="hljs-comment">####################################################################</span><br><span class="hljs-comment">####geneList</span><br><span class="hljs-comment">####################################################################</span><br><br><br>d = read.csv(your_csv_file)<br><br>d = read.table(<span class="hljs-string">&quot;geneList&quot;</span>)<br><span class="hljs-comment">### assume 1st column is ID</span><br><span class="hljs-comment">### 2nd column is FC</span><br><span class="hljs-comment">### feature 1: numeric vector</span><br><br>geneList = d[,<span class="hljs-number">2</span>]<br><span class="hljs-built_in">names</span>(geneList) = <span class="hljs-built_in">as.character</span>(d[,<span class="hljs-number">1</span>])  <span class="hljs-comment">## feature 2: named vector</span><br>geneList = sort(geneList, decreasing = <span class="hljs-literal">TRUE</span>)   <span class="hljs-comment">## feature 3: decreasing order</span><br>head(geneList)<br><br>eg = bitr(x, fromType=<span class="hljs-string">&quot;ENTREZID&quot;</span>, toType=<span class="hljs-string">&quot;SYMBOL&quot;</span>, annoDb=<span class="hljs-string">&quot;org.Hs.eg.db&quot;</span>)<br><br> p  &lt;- pathview(gene.data  = geneList, pathway.id = <span class="hljs-string">&quot;hsa04668&quot;</span>, species    = <span class="hljs-string">&quot;hsa&quot;</span>, limit      = <span class="hljs-built_in">list</span>(gene=<span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(geneList)), cpd=<span class="hljs-number">1</span>))<br><br><br>gene &lt;- <span class="hljs-built_in">names</span>(geneList)[<span class="hljs-built_in">abs</span>(geneList) &gt; <span class="hljs-number">2</span>]<br>gene.df &lt;- bitr(gene, fromType = <span class="hljs-string">&quot;ENTREZID&quot;</span>,<br>        toType = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;ENSEMBL&quot;</span>, <span class="hljs-string">&quot;SYMBOL&quot;</span>),<br>        OrgDb = org.Hs.eg.db)<br>head(gene.df)<br><br><br><br><br>ego2 &lt;- enrichGO(gene         = gene.df$ENSEMBL,<br>                OrgDb         = org.Hs.eg.db,<br>        keytype       = <span class="hljs-string">&#x27;ENSEMBL&#x27;</span>,<br>                ont           = <span class="hljs-string">&quot;CC&quot;</span>,<br>                pAdjustMethod = <span class="hljs-string">&quot;BH&quot;</span>,<br>                pvalueCutoff  = <span class="hljs-number">0.01</span>,<br>                qvalueCutoff  = <span class="hljs-number">0.05</span>)<br>ego2 &lt;- setReadable(ego2, OrgDb = org.Hs.eg.db)<br><br><br>biocLite(<span class="hljs-string">&quot;topGO&quot;</span>)<br><br><br><br><br>barplot(ggo, drop=<span class="hljs-literal">TRUE</span>, showCategory=<span class="hljs-number">12</span>)<br><br><span class="hljs-comment">###############################################################</span><br><span class="hljs-comment">##</span><br><span class="hljs-comment">###############################################################</span><br>geneList = d[,<span class="hljs-number">2</span>]<br><span class="hljs-built_in">names</span>(geneList) = <span class="hljs-built_in">as.character</span>(d[,<span class="hljs-number">1</span>])  <span class="hljs-comment">## feature 2: named vector</span><br>geneList = sort(geneList, decreasing = <span class="hljs-literal">TRUE</span>)   <span class="hljs-comment">## feature 3: decreasing order</span><br>head(geneList)<br><br>kk2 &lt;- gseKEGG(geneList     = geneList,<br>               organism     = <span class="hljs-string">&#x27;hsa&#x27;</span>,<br>               nPerm        = <span class="hljs-number">1000</span>,<br>               minGSSize    = <span class="hljs-number">120</span>,<br>               pvalueCutoff = <span class="hljs-number">0</span>,<br>               verbose      = <span class="hljs-literal">FALSE</span>)<br>gseaplot(kk2, geneSetID = <span class="hljs-string">&quot;hsa04668&quot;</span>)<br><br><br><span class="hljs-comment">#######################</span><br><span class="hljs-comment">####write the results</span><br><span class="hljs-comment">#######################</span><br><br>write.csv(fortify(kk,showCategory=<span class="hljs-number">120</span>),file=<span class="hljs-string">&quot;kk.matrix&quot;</span>,<span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>,sep=<span class="hljs-string">&#x27;\t&#x27;</span>)<br><br><br><br><br><br><br><br><br><br><span class="hljs-comment">##################################</span><br><br>a &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;3&quot;</span>,<span class="hljs-string">&quot;7&quot;</span>,<span class="hljs-string">&quot;14&quot;</span>,<span class="hljs-string">&quot;21&quot;</span>)<br>b &lt;- read.table(<span class="hljs-string">&quot;single_symol.matrix&quot;</span>)<br><br><span class="hljs-keyword">for</span>(i <span class="hljs-keyword">in</span> a)<br> &#123;<br> x &lt;- read.table(<span class="hljs-string">&quot;3&quot;</span>)<br> probes= x$V1<br> probes1=match(probes,x$V1)<br> probes2=match(probes,b$V1)<br> <span class="hljs-built_in">sum</span>(<span class="hljs-built_in">is.na</span>(probes1))<br> <span class="hljs-built_in">sum</span>(<span class="hljs-built_in">is.na</span>(probes1))<br> List=data.frame(FC=(x$V4[probes1]* -<span class="hljs-number">1</span>),S=b$V2[probes2],Co=b$V1[probes2])<br> List=data.frame(Symbol=List$S, FC=List$FC )<br> A&lt;- List[order(List[,<span class="hljs-number">2</span>],decreasing=<span class="hljs-built_in">F</span>),]<br> write.table(A,file=paste(i,<span class="hljs-string">&quot;-tesst.txt&quot;</span>,sep=<span class="hljs-string">&#x27;&#x27;</span>),sep=<span class="hljs-string">&#x27;\t&#x27;</span>,<span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>,row.names=<span class="hljs-built_in">F</span>)<br> &#125;<br><br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> b[[<span class="hljs-number">1</span>]])<br> &#123;<br>  probes= i<br> probes1= match(probes,b$V1)<br>  <span class="hljs-built_in">c</span>=b$V2[probes1]<br> png(file=paste(i,<span class="hljs-string">&quot;-&quot;</span>,<span class="hljs-built_in">c</span>,<span class="hljs-string">&quot;-GSEA.png&quot;</span>,sep=<span class="hljs-string">&#x27;&#x27;</span>),wi=<span class="hljs-number">400</span>,he=<span class="hljs-number">400</span>)<br> gseaplot(kk2, geneSetID = i,title=paste(<span class="hljs-built_in">c</span>,<span class="hljs-string">&quot;0dpeVs7dpe&quot;</span>))<br> dev.off()<br> &#125;<br><br><br> <span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> b[[<span class="hljs-number">1</span>]])<br> &#123;<br> p  &lt;- pathview(gene.data  = geneList,<br> pathway.id = i,<br>  species    = <span class="hljs-string">&quot;hsa&quot;</span>,<br>                   limit      = <span class="hljs-built_in">list</span>(gene=<span class="hljs-built_in">max</span>(<span class="hljs-built_in">abs</span>(geneList)), cpd=<span class="hljs-number">1</span>))<br> &#125;<br><br><br>geneList=data.frame(List=List$Symbol,FC=List$FC* -<span class="hljs-number">1</span>)<br><br><br> png(file=paste(i,<span class="hljs-string">&quot;-&quot;</span>,title,<span class="hljs-string">&quot;.png&quot;</span>,sep=<span class="hljs-string">&#x27;&#x27;</span>),wi=<span class="hljs-number">900</span>,he=<span class="hljs-number">900</span>)<br> gseaplot(kk2, geneSetID = i,title= <span class="hljs-built_in">T</span>)<br> dev.off()&#125;<br><br><br>a &lt;- read.table(<span class="hljs-string">&quot;list&quot;</span>)<br> x &lt;- a[[<span class="hljs-number">1</span>]]<br>eg = bitr(x, fromType=<span class="hljs-string">&quot;ENTREZID&quot;</span>, toType=<span class="hljs-string">&quot;SYMBOL&quot;</span>, annoDb=<span class="hljs-string">&quot;org.Hs.eg.db&quot;</span>)<br><span class="hljs-built_in">dim</span>(eg)<br>write.table(eg,file=<span class="hljs-string">&quot;&quot;</span>,row.names=<span class="hljs-built_in">F</span>,<span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>,sep=<span class="hljs-string">&#x27;\t&#x27;</span>)<br><br></code></pre></td></tr></table></figure></div>
<h2 id="Error-in-installation">Error in installation</h2>
<pre>
config.status: creating src/Makevars.tmp
config.status: creating src/Makevars
config.status: creating src/config.h
** libs
gfortran -fno-optimize-sibling-calls  -fpic  -g -O2  -c AMD/Source/amd.f -o AMD/Source/amd.o
make: gfortran: No such file or directory
make: *** [/usr/lib64/R/etc/Makeconf:195: AMD/Source/amd.o] Error 127
ERROR: compilation failed for package 'igraph'
* removing '/home/karobben/R/x86_64-pc-linux-gnu-library/4.0/igraph'
During startup - Warning message:
</pre>
<p>problem: lack of gcc-fortran</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="code"><pre><code class="hljs bash">sudo pacman -S gcc-fortran<br></code></pre></td></tr></table></figure></div>
<h2 id="Error-in-using">Error in using</h2>
<pre>
fail to download KEGG data...
Error in download.KEGG.Path(species) : 
  'species' should be one of organisms listed in 'http://www.genome.jp/kegg/catalog/org_list.html'...
Calls: gseKEGG ... prepare_KEGG -> download_KEGG -> download.KEGG.Path
In addition: Warning message:
In download.file(url, method = method, ...) :
  URL 'https://rest.kegg.jp/link/hsa/pathway': status was 'Failure when receiving data from the peer'
Execution halted
</pre>
<p>This is such an annoying issue. The weird thing is that everything works fine when I use the code in the terminal, but as soon as I write the code into a script, it just doesn’t seem to work. The author suggests that the issue could be resolved by updating to the latest version. However, this would mean that I would also need to update either the Bioconductor or R base. I don’t want to update anything, I just want the job to be done. Thankfully, <a target="_blank" rel="noopener" href="https://github.com/YuLab-SMU/clusterProfiler/issues/256">Slohr</a>’s answer in 2022 provided a perfect solution to this problem.</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## https://www.genome.jp/kegg/rest/keggapi.html</span><br><span class="hljs-comment">## kegg_link(&#x27;hsa&#x27;, &#x27;pathway&#x27;)</span><br>ttp_kegg_link &lt;- <span class="hljs-keyword">function</span>(target_db, source_db) &#123;<br>    url &lt;- paste0(<span class="hljs-string">&quot;https://rest.kegg.jp/link/&quot;</span>, target_db, <span class="hljs-string">&quot;/&quot;</span>, source_db, collapse=<span class="hljs-string">&quot;&quot;</span>)<br>    local_mydownload &lt;- <span class="hljs-keyword">function</span> (url, method, quiet = <span class="hljs-literal">TRUE</span>, ...) <br>    &#123;<br>        <span class="hljs-keyword">if</span> (capabilities(<span class="hljs-string">&quot;libcurl&quot;</span>)) &#123;<br>            dl &lt;- tryCatch(utils::download.file(url, quiet = quiet, <br>                method = <span class="hljs-string">&quot;libcurl&quot;</span>, ...), error = <span class="hljs-keyword">function</span>(e) <span class="hljs-literal">NULL</span>)<br>        &#125;<br>        <span class="hljs-keyword">else</span> &#123;<br>            dl &lt;- tryCatch(downloader::download(url, quiet = <span class="hljs-literal">TRUE</span>, <br>                method = method, ...), error = <span class="hljs-keyword">function</span>(e) <span class="hljs-literal">NULL</span>)<br>        &#125;<br>        <span class="hljs-built_in">return</span>(dl)<br>    &#125;<br>    local_kegg_rest &lt;- <span class="hljs-keyword">function</span> (rest_url) <br>    &#123;<br>        message(<span class="hljs-string">&quot;Reading KEGG annotation online:\n&quot;</span>)<br>        f &lt;- tempfile()<br>        dl &lt;- local_mydownload(rest_url, destfile = f)<br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(dl)) &#123;<br>            message(<span class="hljs-string">&quot;fail to download KEGG data...&quot;</span>)<br>            <span class="hljs-built_in">return</span>(<span class="hljs-literal">NULL</span>)<br>        &#125;<br>        content &lt;- readLines(f)<br>        content %&lt;&gt;% strsplit(., <span class="hljs-string">&quot;\t&quot;</span>) %&gt;% do.call(<span class="hljs-string">&quot;rbind&quot;</span>, .)<br>        res &lt;- data.frame(from = content[, <span class="hljs-number">1</span>], to = content[, <span class="hljs-number">2</span>])<br>        <span class="hljs-built_in">return</span>(res)<br>    &#125;<br>    local_kegg_rest(url)<br>&#125;<br><br>ttp_kegg_list &lt;- <span class="hljs-keyword">function</span>(db) &#123;<br>    url &lt;- paste0(<span class="hljs-string">&quot;https://rest.kegg.jp/list/&quot;</span>, db, collapse=<span class="hljs-string">&quot;&quot;</span>)<br>    local_mydownload &lt;- <span class="hljs-keyword">function</span> (url, method, quiet = <span class="hljs-literal">TRUE</span>, ...) <br>    &#123;<br>        <span class="hljs-keyword">if</span> (capabilities(<span class="hljs-string">&quot;libcurl&quot;</span>)) &#123;<br>            dl &lt;- tryCatch(utils::download.file(url, quiet = quiet, <br>                method = <span class="hljs-string">&quot;libcurl&quot;</span>, ...), error = <span class="hljs-keyword">function</span>(e) <span class="hljs-literal">NULL</span>)<br>        &#125;<br>        <span class="hljs-keyword">else</span> &#123;<br>            dl &lt;- tryCatch(downloader::download(url, quiet = <span class="hljs-literal">TRUE</span>, <br>                method = method, ...), error = <span class="hljs-keyword">function</span>(e) <span class="hljs-literal">NULL</span>)<br>        &#125;<br>        <span class="hljs-built_in">return</span>(dl)<br>    &#125;<br>    local_kegg_rest &lt;- <span class="hljs-keyword">function</span> (rest_url) <br>    &#123;<br>        message(<span class="hljs-string">&quot;Reading KEGG annotation online:\n&quot;</span>)<br>        f &lt;- tempfile()<br>        dl &lt;- local_mydownload(rest_url, destfile = f)<br>        <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(dl)) &#123;<br>            message(<span class="hljs-string">&quot;fail to download KEGG data...&quot;</span>)<br>            <span class="hljs-built_in">return</span>(<span class="hljs-literal">NULL</span>)<br>        &#125;<br>        content &lt;- readLines(f)<br>        content %&lt;&gt;% strsplit(., <span class="hljs-string">&quot;\t&quot;</span>) %&gt;% do.call(<span class="hljs-string">&quot;rbind&quot;</span>, .)<br>        res &lt;- data.frame(from = content[, <span class="hljs-number">1</span>], to = content[, <span class="hljs-number">2</span>])<br>        <span class="hljs-built_in">return</span>(res)<br>    &#125;<br>    local_kegg_rest(url)<br>&#125;<br><br>rlang::env_unlock(env = asNamespace(<span class="hljs-string">&#x27;clusterProfiler&#x27;</span>))<br>rlang::env_binding_unlock(env = asNamespace(<span class="hljs-string">&#x27;clusterProfiler&#x27;</span>))<br>assign(<span class="hljs-string">&#x27;kegg_link&#x27;</span>, ttp_kegg_link, envir = asNamespace(<span class="hljs-string">&#x27;clusterProfiler&#x27;</span>))<br>assign(<span class="hljs-string">&#x27;kegg_list&#x27;</span>, ttp_kegg_list, envir = asNamespace(<span class="hljs-string">&#x27;clusterProfiler&#x27;</span>))<br>rlang::env_binding_lock(env = asNamespace(<span class="hljs-string">&#x27;clusterProfiler&#x27;</span>))<br>rlang::env_lock(asNamespace(<span class="hljs-string">&#x27;clusterProfiler&#x27;</span>))<br></code></pre></td></tr></table></figure></div>
</div><div class="article-licensing box"><div class="licensing-title"><p>clusterProfiler</p><p><a href="https://karobben.github.io/2020/06/14/R/clusterProfiler/">https://karobben.github.io/2020/06/14/R/clusterProfiler/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2020-06-14</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2024-01-11</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/Bioinformatics/">Bioinformatics</a><a class="link-muted mr-2" rel="tag" href="/tags/R/">R</a><a class="link-muted mr-2" rel="tag" href="/tags/GO/">GO</a><a class="link-muted mr-2" rel="tag" href="/tags/KEGG/">KEGG</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2020/06/18/R/geom_qq/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">geom_qq</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2020/06/13/Blog/Hexo_gitpage/"><span class="level-item">Hexo with Github</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Studying without reflection leads to underfitting, while reflecting without studying leads to overfitting.</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">559</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">154</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">167</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#clusterProfiler"><span class="level-left"><span class="level-item">1</span><span class="level-item">clusterProfiler</span></span></a></li><li><a class="level is-mobile" href="#Install"><span class="level-left"><span class="level-item">2</span><span class="level-item">Install</span></span></a></li><li><a class="level is-mobile" href="#Enrichment-analysis-and-GSEA"><span class="level-left"><span class="level-item">3</span><span class="level-item">Enrichment analysis and GSEA</span></span></a></li><li><a class="level is-mobile" href="#Go-ontology"><span class="level-left"><span class="level-item">4</span><span class="level-item">Go ontology</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#GO-1-Supported-Organism"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">GO_1. Supported Organism</span></span></a></li><li><a class="level is-mobile" href="#GO-2-Example-for-OrgDb-species"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">GO_2. Example for OrgDb species</span></span></a></li><li><a class="level is-mobile" href="#GO-3-Over-Representation-Test"><span class="level-left"><span class="level-item">4.3</span><span class="level-item">GO_3. Over-Representation Test</span></span></a></li><li><a class="level is-mobile" href="#GO-4-GO-Gene-Set-Enrichment-Analysis"><span class="level-left"><span class="level-item">4.4</span><span class="level-item">GO_4 GO Gene Set Enrichment Analysis</span></span></a></li></ul></li><li><a class="level is-mobile" href="#GO-ontology-for-Non-module-Species"><span class="level-left"><span class="level-item">5</span><span class="level-item">GO ontology for Non-module Species</span></span></a></li><li><a class="level is-mobile" href="#KEGG-Enrichment"><span class="level-left"><span class="level-item">6</span><span class="level-item">KEGG Enrichment</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#KE-1-Over-Representation-Test"><span class="level-left"><span class="level-item">6.1</span><span class="level-item">KE_1. Over-Representation Test</span></span></a></li><li><a class="level is-mobile" href="#KE-2-Gene-Set-Enrichment-Analysis"><span class="level-left"><span class="level-item">6.2</span><span class="level-item">KE_2. Gene Set Enrichment Analysis</span></span></a></li><li><a class="level is-mobile" href="#KE-3-KEGG-Module-over-representation-test"><span class="level-left"><span class="level-item">6.3</span><span class="level-item">KE_3. KEGG Module over-representation test</span></span></a></li><li><a class="level is-mobile" href="#KE-4-KEGG-Module-Gene-Set-Enrichment-Analysis"><span class="level-left"><span class="level-item">6.4</span><span class="level-item">KE_4. KEGG Module Gene Set Enrichment Analysis</span></span></a></li></ul></li><li><a class="level is-mobile" href="#A-Pipeline-Example"><span class="level-left"><span class="level-item">7</span><span class="level-item">A Pipeline Example</span></span></a></li><li><a class="level is-mobile" href="#Error-in-installation"><span class="level-left"><span class="level-item">8</span><span class="level-item">Error in installation</span></span></a></li><li><a class="level is-mobile" href="#Error-in-using"><span class="level-left"><span class="level-item">9</span><span class="level-item">Error in using</span></span></a></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/AI/"><span class="tag">AI</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Algorithm/"><span class="tag">Algorithm</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Antibody/"><span class="tag">Antibody</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochmistry/"><span class="tag">Biochmistry</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">84</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">49</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/CLI-Tools/"><span class="tag">CLI Tools</span><span class="tag">15</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data-Science/"><span class="tag">Data Science</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fasta/"><span class="tag">Fasta</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immunology/"><span class="tag">Immunology</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Influenza/"><span class="tag">Influenza</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">37</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Molecular-Dynamic-Simulation/"><span class="tag">Molecular Dynamic Simulation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">35</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">65</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein/"><span class="tag">Protein</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein-Structure/"><span class="tag">Protein Structure</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyMol/"><span class="tag">PyMol</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">131</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python-Plot/"><span class="tag">Python Plot</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python-Seaborn/"><span class="tag">Python Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Pyvista/"><span class="tag">Pyvista</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">88</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">26</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripting/"><span class="tag">Scripting</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sequencing/"><span class="tag">Sequencing</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">48</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Structure/"><span class="tag">Structure</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/VCF/"><span class="tag">VCF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGS/"><span class="tag">WGS</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/computer-vision/"><span class="tag">computer-vision</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/deeplearning/"><span class="tag">deeplearning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dock/"><span class="tag">dock</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">45</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/git/"><span class="tag">git</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/neuralnetworks/"><span class="tag">neuralnetworks</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/openMM/"><span class="tag">openMM</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/protein/"><span class="tag">protein</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">11</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://imgur.com/cm7mDjj.png" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/AI/"><span class="level-start"><span class="level-item">AI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/AI/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">89</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">83</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">6</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protein-Structure/"><span class="level-start"><span class="level-item">Protein Structure</span></span><span class="level-end"><span class="level-item tag">6</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">11</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">29</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fasta-q/"><span class="level-start"><span class="level-item">Fasta/q</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/WGS/"><span class="level-start"><span class="level-item">WGS</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">41</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">11</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/Machine-Learning/Data-Format/"><span class="level-start"><span class="level-item">Data Format</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Machine-Learning/LM/"><span class="level-start"><span class="level-item">LM</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/Machine-Learning/LM/Protein/"><span class="level-start"><span class="level-item">Protein</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Machine-Learning/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Machine-Learning/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">152</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Immunity/"><span class="level-start"><span class="level-item">Immunity</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">89</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">22</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/AI/"><span class="level-start"><span class="level-item">AI</span></span><span class="level-end"><span class="level-item tag">14</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">12</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/other/"><span class="level-start"><span class="level-item">other</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immunology/"><span class="level-start"><span class="level-item">Immunology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">128</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">86</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">58</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">41</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2025 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>