<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>WGCNA Tutorial 1 - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="WGCNA Tutorial 1"><meta property="og:type" content="blog"><meta property="og:title" content="WGCNA Tutorial 1"><meta property="og:url" content="https://karobben.github.io/2020/06/08/R/WGCNA_1/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="WGCNA Tutorial 1"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://i.loli.net/2020/06/08/TKXgiWr3vBIP4j2.png"><meta property="article:published_time" content="2020-06-08T05:00:00.000Z"><meta property="article:modified_time" content="2023-06-06T21:42:32.150Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="R"><meta property="article:tag" content="Bioinformatics"><meta property="article:tag" content="WGCNA"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://i.loli.net/2020/06/08/TKXgiWr3vBIP4j2.png"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2020/06/08/R/WGCNA_1/"},"headline":"Karobben","image":["https://i.loli.net/2020/06/08/TKXgiWr3vBIP4j2.png"],"datePublished":"2020-06-08T05:00:00.000Z","dateModified":"2023-06-06T21:42:32.150Z","author":{"@type":"Person","name":"Karobben"},"description":"WGCNA Tutorial 1"}</script><link rel="canonical" href="https://karobben.github.io/2020/06/08/R/WGCNA_1/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><div class="card-image"><span class="image is-7by3"><img class="fill" src="https://i.loli.net/2020/06/08/TKXgiWr3vBIP4j2.png" alt="WGCNA Tutorial 1"></span><span class="ccopy" style="position: absolute; bottom: 0; right: 0; padding-right:3px; padding-left:4px; background-color: rgba(255, 255, 255, 0.6)"></span></div><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2020-06-08T05:00:00.000Z" title="6/8/2020, 12:00:00 AM">2020-06-08</time></span><span class="level-item">Updated&nbsp;<time dateTime="2023-06-06T21:42:32.150Z" title="6/6/2023, 4:42:32 PM">2023-06-06</time></span><span class="level-item"><a class="link-muted" href="/categories/R/">R</a><span> / </span><a class="link-muted" href="/categories/R/Bio/">Bio</a><span> / </span><a class="link-muted" href="/categories/R/Bio/WGCNA/">WGCNA</a></span><span class="level-item">15 minutes read (About 2197 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">WGCNA Tutorial 1</h1><div class="content"><h2 id="WGCNA-Tutorial-1">WGCNA Tutorial 1</h2>
<p>raw tutorial: <a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/index.html">Peter</a></p>
<p>Data:</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-Data.zip">Female data</a></li>
<li><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/MaleLiver-Data.zip">Male data</a></li>
</ul>
<h2 id="1-Data-Cleaning">1. Data Cleaning</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Load the WGCNA package</span><br>library(WGCNA);<br><span class="hljs-comment">## The following setting is important, do not omit.</span><br>options(stringsAsFactors = <span class="hljs-literal">FALSE</span>);<br><span class="hljs-comment">##Read in the female liver data set</span><br>femData = read.csv(<span class="hljs-string">&quot;LiverFemale3600.csv&quot;</span>);<br><br>datExpr0 = as.data.frame(t(femData[, -<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)]));<br><span class="hljs-built_in">names</span>(datExpr0) = femData$substanceBXH;<br>rownames(datExpr0) = <span class="hljs-built_in">names</span>(femData)[-<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">8</span>)];<br></code></pre></td></tr></table></figure></div>
<h3 id="1-1-Checking-data">1.1 Checking data</h3>
<p>Checking data for excessive missing values and identification of outlier microarray samples</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">gsg = goodSamplesGenes(datExpr0, verbose = <span class="hljs-number">3</span>);<br>gsg$allOK<br><br><br><span class="hljs-keyword">if</span> (!gsg$allOK)<br>&#123;<br>  <span class="hljs-comment"># Optionally, print the gene and sample names that were removed:</span><br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">sum</span>(!gsg$goodGenes)&gt;<span class="hljs-number">0</span>)<br>     printFlush(paste(<span class="hljs-string">&quot;Removing genes:&quot;</span>, paste(<span class="hljs-built_in">names</span>(datExpr0)[!gsg$goodGenes], collapse = <span class="hljs-string">&quot;, &quot;</span>)));<br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">sum</span>(!gsg$goodSamples)&gt;<span class="hljs-number">0</span>)<br>     printFlush(paste(<span class="hljs-string">&quot;Removing samples:&quot;</span>, paste(rownames(datExpr0)[!gsg$goodSamples], collapse = <span class="hljs-string">&quot;, &quot;</span>)));<br>  <span class="hljs-comment"># Remove the offending genes and samples from the data:</span><br>  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]<br>&#125;<br></code></pre></td></tr></table></figure></div>
<h3 id="1-2-Cluster-the-samples">1.2 Cluster the samples</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">sampleTree = hclust(dist(datExpr0), method = <span class="hljs-string">&quot;average&quot;</span>);<br><span class="hljs-comment">## Plot the sample tree: Open a graphic output window of size 12 by 9 inches</span><br><span class="hljs-comment">## The user should change the dimensions if the window is too large or too small.</span><br>sizeGrWindow(<span class="hljs-number">12</span>,<span class="hljs-number">9</span>)<br><span class="hljs-comment">##png(file = &quot;sampleClustering.png&quot;)</span><br>par(cex = <span class="hljs-number">0.6</span>);<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-number">4</span>,<span class="hljs-number">2</span>,<span class="hljs-number">0</span>))<br>plot(sampleTree, main = <span class="hljs-string">&quot;Sample clustering to detect outliers&quot;</span>, sub=<span class="hljs-string">&quot;&quot;</span>, xlab=<span class="hljs-string">&quot;&quot;</span>, cex.lab = <span class="hljs-number">1.5</span>,<br>     cex.axis = <span class="hljs-number">1.5</span>, cex.main = <span class="hljs-number">2</span>)<br><br><span class="hljs-comment">## Plot a line to show the cut</span><br>abline(h = <span class="hljs-number">15</span>, col = <span class="hljs-string">&quot;red&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/IowWAKgDSJP6lTC.png" alt="sampleClustering"></p>
<p>It appears there is one outlier (sample F2_221, see Fig. 1). One can remove it by hand, or use an automatic approach. Choose a height cut that will remove the offending sample, say 15 (the red line in the plot), and use a branch cut at that height.</p>
<h3 id="1-3-Remove-the-abnormal-sample">1.3 Remove the abnormal sample</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Determine cluster under the line</span><br>clust = cutreeStatic(sampleTree, cutHeight = <span class="hljs-number">15</span>, minSize = <span class="hljs-number">10</span>)<br>table(clust)<br><span class="hljs-comment">## clust 1 contains the samples we want to keep.</span><br>keepSamples = (clust==<span class="hljs-number">1</span>)<br>datExpr = datExpr0[keepSamples, ]<br>nGenes = ncol(datExpr)<br>nSamples = nrow(datExpr)<br><br></code></pre></td></tr></table></figure></div>
<h3 id="1-4-Loading-Clinical-Data">1.4 Loading Clinical Data</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">traitData = read.csv(<span class="hljs-string">&quot;ClinicalTraits.csv&quot;</span>);<br><span class="hljs-built_in">dim</span>(traitData)<br><span class="hljs-built_in">names</span>(traitData)<br><br><span class="hljs-comment">## remove columns that hold information we do not need.</span><br>allTraits = traitData[, -<span class="hljs-built_in">c</span>(<span class="hljs-number">31</span>, <span class="hljs-number">16</span>)];<br>allTraits = allTraits[, <span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>, <span class="hljs-number">11</span>:<span class="hljs-number">36</span>) ];<br><span class="hljs-built_in">dim</span>(allTraits)<br><span class="hljs-built_in">names</span>(allTraits)<br><br><span class="hljs-comment">## Form a data frame analogous to expression data that will hold the clinical traits.</span><br><br>femaleSamples = rownames(datExpr);<br>traitRows = match(femaleSamples, allTraits$Mice);<br>datTraits = allTraits[traitRows, -<span class="hljs-number">1</span>];<br>rownames(datTraits) = allTraits[traitRows, <span class="hljs-number">1</span>];<br><br>collectGarbage()<br><br><span class="hljs-comment">## Re-cluster samples</span><br>sampleTree2 = hclust(dist(datExpr), method = <span class="hljs-string">&quot;average&quot;</span>)<br><span class="hljs-comment">## Convert traits to a color representation: white means low, red means high, grey means missing entry</span><br>traitColors = numbers2colors(datTraits, signed = <span class="hljs-literal">FALSE</span>);<br><span class="hljs-comment">## Plot the sample dendrogram and the colors underneath.</span><br><span class="hljs-comment">## png(&quot;traits_heatmap.png&quot;)</span><br>plotDendroAndColors(sampleTree2, traitColors,<br>                    groupLabels = <span class="hljs-built_in">names</span>(datTraits),<br>                    main = <span class="hljs-string">&quot;Sample dendrogram and trait heatmap&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/QqfwD9e8AroGBLO.png" alt="traits_heatmap"></p>
<h2 id="2-Network-construction-and-module-detection">2. Network construction and module detection</h2>
<p><strong>Automatic one-step detection</strong></p>
<h3 id="2-1-Pick-soft-thresdholding-power">2.1 Pick soft-thresdholding power</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Choose a set of soft-thresholding powers</span><br>powers = <span class="hljs-built_in">c</span>(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">10</span>), seq(from = <span class="hljs-number">12</span>, to=<span class="hljs-number">20</span>, by=<span class="hljs-number">2</span>))<br><span class="hljs-comment">## Call the network topology analysis function</span><br>sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="hljs-number">5</span>)<br><span class="hljs-comment">## Plot the results:</span><br>sizeGrWindow(<span class="hljs-number">9</span>, <span class="hljs-number">5</span>)<br><span class="hljs-comment">## Scale-free topology fit index as a function of the soft-thresholding power</span><br><span class="hljs-comment">## png(&quot;Soft_th.png&quot;,width=500,height=300)</span><br>cex1 = <span class="hljs-number">0.9</span>;<br>par(mfrow = <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>));<br>plot(sft$fitIndices[,<span class="hljs-number">1</span>], -<span class="hljs-built_in">sign</span>(sft$fitIndices[,<span class="hljs-number">3</span>])*sft$fitIndices[,<span class="hljs-number">2</span>],<br>     xlab=<span class="hljs-string">&quot;Soft Threshold (power)&quot;</span>,ylab=<span class="hljs-string">&quot;Scale Free Topology Model Fit,signed R^2&quot;</span>,type=<span class="hljs-string">&quot;n&quot;</span>,<br>     main = paste(<span class="hljs-string">&quot;Scale independence&quot;</span>));<br>text(sft$fitIndices[,<span class="hljs-number">1</span>], -<span class="hljs-built_in">sign</span>(sft$fitIndices[,<span class="hljs-number">3</span>])*sft$fitIndices[,<span class="hljs-number">2</span>],<br>     labels=powers,cex=cex1,col=<span class="hljs-string">&quot;red&quot;</span>);<br><span class="hljs-comment">## this line corresponds to using an R^2 cut-off of h</span><br>abline(h=<span class="hljs-number">0.90</span>,col=<span class="hljs-string">&quot;red&quot;</span>)<br><span class="hljs-comment">## Mean connectivity as a function of the soft-thresholding power</span><br>plot(sft$fitIndices[,<span class="hljs-number">1</span>], sft$fitIndices[,<span class="hljs-number">5</span>],<br>     xlab=<span class="hljs-string">&quot;Soft Threshold (power)&quot;</span>,ylab=<span class="hljs-string">&quot;Mean Connectivity&quot;</span>, type=<span class="hljs-string">&quot;n&quot;</span>,<br>     main = paste(<span class="hljs-string">&quot;Mean connectivity&quot;</span>))<br>text(sft$fitIndices[,<span class="hljs-number">1</span>], sft$fitIndices[,<span class="hljs-number">5</span>], labels=powers, cex=cex1,col=<span class="hljs-string">&quot;red&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/1xQfcvesAq2nCR5.png" alt="Soft_th"></p>
<h3 id="2-2-One-step-network-construction-and-module-detection">2.2 One-step network construction and module detection</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">net = blockwiseModules(datExpr, power = <span class="hljs-number">6</span>,<br>                       TOMType = <span class="hljs-string">&quot;unsigned&quot;</span>, minModuleSize = <span class="hljs-number">30</span>,<br>                       reassignThreshold = <span class="hljs-number">0</span>, mergeCutHeight = <span class="hljs-number">0.25</span>,<br>                       numericLabels = <span class="hljs-literal">TRUE</span>, pamRespectsDendro = <span class="hljs-literal">FALSE</span>,<br>                       saveTOMs = <span class="hljs-literal">TRUE</span>,<br>                       saveTOMFileBase = <span class="hljs-string">&quot;femaleMouseTOM&quot;</span>,<br>                       verbose = <span class="hljs-number">3</span>)<br><br><span class="hljs-comment">## View modules</span><br><span class="hljs-comment">## table(net$colors)</span><br>moduleLabels = net$colors<br>moduleColors = labels2colors(net$colors)<br>MEs = net$MEs;<br>geneTree = net$dendrograms[[<span class="hljs-number">1</span>]]<br><span class="hljs-comment">## open a graphics window</span><br>sizeGrWindow(<span class="hljs-number">12</span>, <span class="hljs-number">9</span>)<br><span class="hljs-comment">## png(&#x27;123.png&#x27;,width=480, height=320)</span><br><span class="hljs-comment">## Convert labels to colors for plotting</span><br>mergedColors = labels2colors(net$colors)<br><span class="hljs-comment">## Plot the dendrogram and the module colors underneath</span><br>plotDendroAndColors(net$dendrograms[[<span class="hljs-number">1</span>]], mergedColors[net$blockGenes[[<span class="hljs-number">1</span>]]],<br>                    <span class="hljs-string">&quot;Module colors&quot;</span>,<br>                    dendroLabels = <span class="hljs-literal">FALSE</span>, hang = <span class="hljs-number">0.03</span>,<br>                    addGuide = <span class="hljs-literal">TRUE</span>, guideHang = <span class="hljs-number">0.05</span>)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/TKXgiWr3vBIP4j2.png" alt="123"></p>
<h2 id="3-Relating-modules-to-external-information-and-identifying-important-genes">3. Relating modules to external information and identifying important genes</h2>
<h3 id="3-1-Clinical-Traits">3.1 Clinical Traits</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Define numbers of genes and samples</span><br>nGenes = ncol(datExpr);<br>nSamples = nrow(datExpr);<br><span class="hljs-comment">## Recalculate MEs with color labels</span><br>MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes<br>MEs = orderMEs(MEs0)<br>moduleTraitCor = cor(MEs, datTraits, use = <span class="hljs-string">&quot;p&quot;</span>);<br>moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);<br><br><br>sizeGrWindow(<span class="hljs-number">10</span>,<span class="hljs-number">6</span>)<br><span class="hljs-comment">## Will display correlations and their p-values</span><br><span class="hljs-comment">## png(&quot;1234.png&quot;,width=1000,height=600)</span><br>textMatrix =  paste(<span class="hljs-built_in">signif</span>(moduleTraitCor, <span class="hljs-number">2</span>), <span class="hljs-string">&quot;\n(&quot;</span>,<br>                           <span class="hljs-built_in">signif</span>(moduleTraitPvalue, <span class="hljs-number">1</span>), <span class="hljs-string">&quot;)&quot;</span>, sep = <span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-built_in">dim</span>(textMatrix) = <span class="hljs-built_in">dim</span>(moduleTraitCor)<br>par(mar = <span class="hljs-built_in">c</span>(<span class="hljs-number">6</span>, <span class="hljs-number">8.5</span>, <span class="hljs-number">3</span>, <span class="hljs-number">3</span>));<br><span class="hljs-comment">## Display the correlation values within a heatmap plot</span><br>labeledHeatmap(Matrix = moduleTraitCor,<br>               xLabels = <span class="hljs-built_in">names</span>(datTraits),<br>               yLabels = <span class="hljs-built_in">names</span>(MEs),<br>               ySymbols = <span class="hljs-built_in">names</span>(MEs),<br>               colorLabels = <span class="hljs-literal">FALSE</span>,<br>               colors = greenWhiteRed(<span class="hljs-number">50</span>),<br>               textMatrix = textMatrix,<br>               setStdMargins = <span class="hljs-literal">FALSE</span>,<br>               cex.text = <span class="hljs-number">0.5</span>,<br>               zlim = <span class="hljs-built_in">c</span>(-<span class="hljs-number">1</span>,<span class="hljs-number">1</span>),<br>               main = paste(<span class="hljs-string">&quot;Module-trait relationships&quot;</span>))<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/VehZjcb2F5GM6vR.png" alt="1234"></p>
<h3 id="3-2-Genes-and-Trait">3.2 Genes and Trait</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Define variable weight containing the weight column of datTrait</span><br>weight = as.data.frame(datTraits$weight_g);<br><span class="hljs-built_in">names</span>(weight) = <span class="hljs-string">&quot;weight&quot;</span><br><span class="hljs-comment">## names (colors) of the modules</span><br>modNames = substring(<span class="hljs-built_in">names</span>(MEs), <span class="hljs-number">3</span>)<br><br>geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = <span class="hljs-string">&quot;p&quot;</span>));<br>MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));<br><br><span class="hljs-built_in">names</span>(geneModuleMembership) = paste(<span class="hljs-string">&quot;MM&quot;</span>, modNames, sep=<span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-built_in">names</span>(MMPvalue) = paste(<span class="hljs-string">&quot;p.MM&quot;</span>, modNames, sep=<span class="hljs-string">&quot;&quot;</span>);<br><br>geneTraitSignificance = as.data.frame(cor(datExpr, weight, use = <span class="hljs-string">&quot;p&quot;</span>));<br>GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));<br><br><span class="hljs-built_in">names</span>(geneTraitSignificance) = paste(<span class="hljs-string">&quot;GS.&quot;</span>, <span class="hljs-built_in">names</span>(weight), sep=<span class="hljs-string">&quot;&quot;</span>);<br><span class="hljs-built_in">names</span>(GSPvalue) = paste(<span class="hljs-string">&quot;p.GS.&quot;</span>, <span class="hljs-built_in">names</span>(weight), sep=<span class="hljs-string">&quot;&quot;</span>);<br><br><br><br>module = <span class="hljs-string">&quot;brown&quot;</span><br>column = match(module, modNames);<br>moduleGenes = moduleColors==module;<br><br>sizeGrWindow(<span class="hljs-number">7</span>, <span class="hljs-number">7</span>);<br><span class="hljs-comment">## png(&quot;1234.png&quot;)</span><br>par(mfrow = <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>));<br>verboseScatterplot(<span class="hljs-built_in">abs</span>(geneModuleMembership[moduleGenes, column]),<br>                   <span class="hljs-built_in">abs</span>(geneTraitSignificance[moduleGenes, <span class="hljs-number">1</span>]),<br>                   xlab = paste(<span class="hljs-string">&quot;Module Membership in&quot;</span>, module, <span class="hljs-string">&quot;module&quot;</span>),<br>                   ylab = <span class="hljs-string">&quot;Gene significance for body weight&quot;</span>,<br>                   main = paste(<span class="hljs-string">&quot;Module membership vs. gene significance\n&quot;</span>),<br>                   cex.main = <span class="hljs-number">1.2</span>, cex.lab = <span class="hljs-number">1.2</span>, cex.axis = <span class="hljs-number">1.2</span>, col = module)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/iWERCpGZ2Nhse7X.png" alt="1234"></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">annot = read.csv(file = <span class="hljs-string">&quot;GeneAnnotation.csv&quot;</span>);<br><span class="hljs-built_in">dim</span>(annot)<br><span class="hljs-built_in">names</span>(annot)<br>probes = <span class="hljs-built_in">names</span>(datExpr)<br>probes2annot = match(probes, annot$substanceBXH)<br><span class="hljs-comment">## The following is the number or probes without annotation:</span><br><span class="hljs-built_in">sum</span>(<span class="hljs-built_in">is.na</span>(probes2annot))<br><span class="hljs-comment">## Should return 0.</span><br><br><span class="hljs-comment">## Create the starting data frame</span><br>geneInfo0 = data.frame(substanceBXH = probes,<br>                      geneSymbol = annot$gene_symbol[probes2annot],<br>                      LocusLinkID = annot$LocusLinkID[probes2annot],<br>                      moduleColor = moduleColors,<br>                      geneTraitSignificance,<br>                      GSPvalue)<br><span class="hljs-comment">## Order modules by their significance for weight</span><br>modOrder = order(-<span class="hljs-built_in">abs</span>(cor(MEs, weight, use = <span class="hljs-string">&quot;p&quot;</span>)));<br><span class="hljs-comment">## Add module membership information in the chosen order</span><br><span class="hljs-keyword">for</span> (mod <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:ncol(geneModuleMembership))<br>&#123;<br>  oldNames = <span class="hljs-built_in">names</span>(geneInfo0)<br>  geneInfo0 = data.frame(geneInfo0, geneModuleMembership[, modOrder[mod]],<br>                         MMPvalue[, modOrder[mod]]);<br>  <span class="hljs-built_in">names</span>(geneInfo0) = <span class="hljs-built_in">c</span>(oldNames, paste(<span class="hljs-string">&quot;MM.&quot;</span>, modNames[modOrder[mod]], sep=<span class="hljs-string">&quot;&quot;</span>),<br>                       paste(<span class="hljs-string">&quot;p.MM.&quot;</span>, modNames[modOrder[mod]], sep=<span class="hljs-string">&quot;&quot;</span>))<br>&#125;<br><span class="hljs-comment">## Order the genes in the geneInfo variable first by module color, then by geneTraitSignificance</span><br>geneOrder = order(geneInfo0$moduleColor, -<span class="hljs-built_in">abs</span>(geneInfo0$GS.weight));<br>geneInfo = geneInfo0[geneOrder, ]<br><br><br></code></pre></td></tr></table></figure></div>
<h2 id="4-GO-Enrichment-of-Modules">4. GO Enrichment of Modules</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Read in the probe annotation</span><br>annot = read.csv(file = <span class="hljs-string">&quot;GeneAnnotation.csv&quot;</span>);<br><span class="hljs-comment">## Match probes in the data set to the probe IDs in the annotation file</span><br>probes = <span class="hljs-built_in">names</span>(datExpr)<br>probes2annot = match(probes, annot$substanceBXH)<br><span class="hljs-comment">## Get the corresponding Locuis Link IDs</span><br>allLLIDs = annot$LocusLinkID[probes2annot];<br><span class="hljs-comment">## $ Choose interesting modules</span><br>intModules = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;brown&quot;</span>, <span class="hljs-string">&quot;red&quot;</span>, <span class="hljs-string">&quot;salmon&quot;</span>)<br><span class="hljs-keyword">for</span> (module <span class="hljs-keyword">in</span> intModules)<br>&#123;<br>  <span class="hljs-comment"># Select module probes</span><br>  modGenes = (moduleColors==module)<br>  <span class="hljs-comment"># Get their entrez ID codes</span><br>  modLLIDs = allLLIDs[modGenes];<br>  <span class="hljs-comment"># Write them into a file</span><br>  fileName = paste(<span class="hljs-string">&quot;LocusLinkIDs-&quot;</span>, module, <span class="hljs-string">&quot;.txt&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>);<br>  write.table(as.data.frame(modLLIDs), file = fileName,<br>              row.names = <span class="hljs-literal">FALSE</span>, col.names = <span class="hljs-literal">FALSE</span>)<br>&#125;<br><span class="hljs-comment">## As background in the enrichment analysis, we will use all probes in the analysis.</span><br>fileName = paste(<span class="hljs-string">&quot;LocusLinkIDs-all.txt&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>);<br>write.table(as.data.frame(allLLIDs), file = fileName,<br>            row.names = <span class="hljs-literal">FALSE</span>, col.names = <span class="hljs-literal">FALSE</span>)<br><br>GOenr = GOenrichmentAnalysis(moduleColors, allLLIDs, organism = <span class="hljs-string">&quot;mouse&quot;</span>, nBestP = <span class="hljs-number">10</span>);<br><br>tab = GOenr$bestPTerms[[<span class="hljs-number">4</span>]]$enrichment<br><br>write.table(tab, file = <span class="hljs-string">&quot;GOEnrichmentTable.csv&quot;</span>, sep = <span class="hljs-string">&quot;,&quot;</span>, <span class="hljs-built_in">quote</span> = <span class="hljs-literal">TRUE</span>, row.names = <span class="hljs-literal">FALSE</span>)<br><br><br>keepCols = <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>, <span class="hljs-number">2</span>, <span class="hljs-number">5</span>, <span class="hljs-number">6</span>, <span class="hljs-number">7</span>, <span class="hljs-number">12</span>, <span class="hljs-number">13</span>);<br>screenTab = tab[, keepCols];<br><span class="hljs-comment">## Round the numeric columns to 2 decimal places:</span><br>numCols = <span class="hljs-built_in">c</span>(<span class="hljs-number">3</span>, <span class="hljs-number">4</span>);<br>screenTab[, numCols] = <span class="hljs-built_in">signif</span>(apply(screenTab[, numCols], <span class="hljs-number">2</span>, <span class="hljs-built_in">as.numeric</span>), <span class="hljs-number">2</span>)<br><span class="hljs-comment">## Truncate the the term name to at most 40 characters</span><br>screenTab[, <span class="hljs-number">7</span>] = substring(screenTab[, <span class="hljs-number">7</span>], <span class="hljs-number">1</span>, <span class="hljs-number">40</span>)<br><span class="hljs-comment">## Shorten the column names:</span><br>colnames(screenTab) = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;module&quot;</span>, <span class="hljs-string">&quot;size&quot;</span>, <span class="hljs-string">&quot;p-val&quot;</span>, <span class="hljs-string">&quot;Bonf&quot;</span>, <span class="hljs-string">&quot;nInTerm&quot;</span>, <span class="hljs-string">&quot;ont&quot;</span>, <span class="hljs-string">&quot;term name&quot;</span>);<br>rownames(screenTab) = <span class="hljs-literal">NULL</span>;<br><span class="hljs-comment">## Set the width of R&#x27;s output. The reader should play with this number to obtain satisfactory output.</span><br>options(width=<span class="hljs-number">95</span>)<br><span class="hljs-comment">## Finally, display the enrichment table:</span><br>screenTab<br></code></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="MIPSASM"><figure class="iseeu highlight /mipsasm"><table><tr><td class="code"><pre><code class="hljs mipsasm">module  size  p-val <span class="hljs-keyword">Bonf </span> nInTerm ont term name<br><span class="hljs-number">1</span> <span class="hljs-keyword">black </span> <span class="hljs-number">166</span> <span class="hljs-number">3</span>.<span class="hljs-number">9</span>e<span class="hljs-number">-04</span> <span class="hljs-number">1</span>.<span class="hljs-number">0</span>e+<span class="hljs-number">00</span>  <span class="hljs-number">4</span>   <span class="hljs-keyword">BP </span> dopamine transport<br><span class="hljs-number">2</span> <span class="hljs-keyword">black </span> <span class="hljs-number">166</span> <span class="hljs-number">6</span>.<span class="hljs-number">5</span>e<span class="hljs-number">-04</span> <span class="hljs-number">1</span>.<span class="hljs-number">0</span>e+<span class="hljs-number">00</span>  <span class="hljs-number">5</span>   <span class="hljs-keyword">BP </span> mRNA transport<br><span class="hljs-number">3</span> <span class="hljs-keyword">black </span> <span class="hljs-number">166</span> <span class="hljs-number">8</span>.<span class="hljs-number">0</span>e<span class="hljs-number">-04</span> <span class="hljs-number">1</span>.<span class="hljs-number">0</span>e+<span class="hljs-number">00</span>  <span class="hljs-number">6</span>   <span class="hljs-keyword">BP </span> RNA transport<br><span class="hljs-number">4</span> <span class="hljs-keyword">black </span> <span class="hljs-number">166</span> <span class="hljs-number">8</span>.<span class="hljs-number">0</span>e<span class="hljs-number">-04</span> <span class="hljs-number">1</span>.<span class="hljs-number">0</span>e+<span class="hljs-number">00</span>  <span class="hljs-number">13</span>  MF  receptor ligand activity<br>...<br></code></pre></td></tr></table></figure></div>
<h2 id="5-Visualization">5. Visualization</h2>
<h3 id="5-1-NetWork-Visualization">5.1 NetWork Visualization</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">nGenes = ncol(datExpr)<br>nSamples = nrow(datExpr)<br><br><span class="hljs-comment">## Calculate topological overlap anew: this could be done more efficiently by saving the TOM</span><br><span class="hljs-comment">## calculated during module detection, but let us do it again here.</span><br>dissTOM = <span class="hljs-number">1</span>-TOMsimilarityFromExpr(datExpr, power = <span class="hljs-number">6</span>);<br><span class="hljs-comment">## Transform dissTOM with a power to make moderately strong connections more visible in the heatmap</span><br>plotTOM = dissTOM^<span class="hljs-number">7</span>;<br><span class="hljs-comment">## Set diagonal to NA for a nicer plot</span><br>diag(plotTOM) = <span class="hljs-literal">NA</span>;<br><span class="hljs-comment">## Call the plot function</span><br>sizeGrWindow(<span class="hljs-number">9</span>,<span class="hljs-number">9</span>)<br>TOMplot(plotTOM, geneTree, moduleColors, main = <span class="hljs-string">&quot;Network heatmap plot, all genes&quot;</span>)<br><br>nSelect = <span class="hljs-number">400</span><br><span class="hljs-comment">## For reproducibility, we set the random seed</span><br>set.seed(<span class="hljs-number">10</span>);<br>select = sample(nGenes, size = nSelect);<br>selectTOM = dissTOM[select, select];<br><span class="hljs-comment">## There&#x27;s no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster.</span><br>selectTree = hclust(as.dist(selectTOM), method = <span class="hljs-string">&quot;average&quot;</span>)<br>selectColors = moduleColors[select];<br><span class="hljs-comment">## Open a graphical window</span><br>sizeGrWindow(<span class="hljs-number">9</span>,<span class="hljs-number">9</span>)<br><span class="hljs-comment">## Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing</span><br><span class="hljs-comment">## the color palette; setting the diagonal to NA also improves the clarity of the plot</span><br><span class="hljs-comment">##png(&#x27;123.png&#x27;)</span><br>plotDiss = selectTOM^<span class="hljs-number">7</span>;<br>diag(plotDiss) = <span class="hljs-literal">NA</span>;<br>TOMplot(plotDiss, selectTree, selectColors, main = <span class="hljs-string">&quot;Network heatmap plot, selected genes&quot;</span>)<br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/wrpXGJj4qQOulim.png" alt="123"></p>
<h3 id="5-2-Visualizing-the-network-of-eigengenes">5.2  Visualizing the network of eigengenes</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Recalculate module eigengenes</span><br>MEs = moduleEigengenes(datExpr, moduleColors)$eigengenes<br><span class="hljs-comment">## Isolate weight from the clinical traits</span><br>weight = as.data.frame(datTraits$weight_g);<br><span class="hljs-built_in">names</span>(weight) = <span class="hljs-string">&quot;weight&quot;</span><br><span class="hljs-comment">## Add the weight to existing module eigengenes</span><br>MET = orderMEs(cbind(MEs, weight))<br><span class="hljs-comment">## Plot the relationships among the eigengenes and the trait</span><br>sizeGrWindow(<span class="hljs-number">5</span>,<span class="hljs-number">7.5</span>);<br><span class="hljs-comment">##png(&#x27;123.png&#x27;,width=400,height=600)</span><br>par(cex = <span class="hljs-number">0.9</span>)<br>plotEigengeneNetworks(MET, <span class="hljs-string">&quot;&quot;</span>, marDendro = <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-number">4</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>), marHeatmap = <span class="hljs-built_in">c</span>(<span class="hljs-number">3</span>,<span class="hljs-number">4</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>), cex.lab = <span class="hljs-number">0.8</span>, xLabelsAngle<br>= <span class="hljs-number">90</span>)<br><br><span class="hljs-comment">## Plot the dendrogram</span><br><span class="hljs-comment">##sizeGrWindow(6,6);</span><br><span class="hljs-comment">##par(cex = 1.0)</span><br><span class="hljs-comment">##plotEigengeneNetworks(MET, &quot;Eigengene dendrogram&quot;, marDendro = c(0,4,2,0),</span><br><span class="hljs-comment">##                      plotHeatmaps = FALSE)</span><br><span class="hljs-comment">## Plot the heatmap matrix (note: this plot will overwrite the dendrogram plot)</span><br><span class="hljs-comment">##par(cex = 1.0)</span><br><span class="hljs-comment">##plotEigengeneNetworks(MET, &quot;Eigengene adjacency heatmap&quot;, marHeatmap = c(3,4,2,2),</span><br><span class="hljs-comment">##                      plotDendrograms = FALSE, xLabelsAngle = 90)</span><br></code></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/08/KcFi5AvfCBhYDJd.png" alt="123"></p>
<h2 id="6-Export-of-networks">6. Export of networks</h2>
<h3 id="6-1-Exporting-to-VisANT">6.1  Exporting to VisANT</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Recalculate topological overlap</span><br>TOM = TOMsimilarityFromExpr(datExpr, power = <span class="hljs-number">6</span>);<br><span class="hljs-comment">## Read in the annotation file</span><br>annot = read.csv(file = <span class="hljs-string">&quot;GeneAnnotation.csv&quot;</span>);<br><span class="hljs-comment">## Select module</span><br>module = <span class="hljs-string">&quot;brown&quot;</span>;<br><span class="hljs-comment">## Select module probes</span><br>probes = <span class="hljs-built_in">names</span>(datExpr)<br>inModule = (moduleColors==module);<br>modProbes = probes[inModule];<br><span class="hljs-comment">## Select the corresponding Topological Overlap</span><br>modTOM = TOM[inModule, inModule];<br><span class="hljs-built_in">dimnames</span>(modTOM) = <span class="hljs-built_in">list</span>(modProbes, modProbes)<br><span class="hljs-comment">## Export the network into an edge list file VisANT can read</span><br>vis = exportNetworkToVisANT(modTOM,<br>  file = paste(<span class="hljs-string">&quot;VisANTInput-&quot;</span>, module, <span class="hljs-string">&quot;.txt&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>),<br>  weighted = <span class="hljs-literal">TRUE</span>,<br>  threshold = <span class="hljs-number">0</span>,<br>  probeToGene = data.frame(annot$substanceBXH, annot$gene_symbol) )<br><br>nTop = <span class="hljs-number">30</span>;<br>IMConn = softConnectivity(datExpr[, modProbes]);<br>top = (rank(-IMConn) &lt;= nTop)<br>vis = exportNetworkToVisANT(modTOM[top, top],<br>  file = paste(<span class="hljs-string">&quot;VisANTInput-&quot;</span>, module, <span class="hljs-string">&quot;-top30.txt&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>),<br>  weighted = <span class="hljs-literal">TRUE</span>,<br>  threshold = <span class="hljs-number">0</span>,<br>  probeToGene = data.frame(annot$substanceBXH, annot$gene_symbol) )<br></code></pre></td></tr></table></figure></div>
<h3 id="6-2-Exporting-to-Cytoscape">6.2 Exporting to Cytoscape</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment">## Recalculate topological overlap if needed</span><br>TOM = TOMsimilarityFromExpr(datExpr, power = <span class="hljs-number">6</span>);<br><span class="hljs-comment">## Read in the annotation file</span><br>annot = read.csv(file = <span class="hljs-string">&quot;GeneAnnotation.csv&quot;</span>);<br><span class="hljs-comment">## Select modules</span><br>modules = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;brown&quot;</span>, <span class="hljs-string">&quot;red&quot;</span>);<br><span class="hljs-comment">## Select module probes</span><br>probes = <span class="hljs-built_in">names</span>(datExpr)<br>inModule = <span class="hljs-built_in">is.finite</span>(match(moduleColors, modules));<br>modProbes = probes[inModule];<br>modGenes = annot$gene_symbol[match(modProbes, annot$substanceBXH)];<br><span class="hljs-comment">## Select the corresponding Topological Overlap</span><br>modTOM = TOM[inModule, inModule];<br><span class="hljs-built_in">dimnames</span>(modTOM) = <span class="hljs-built_in">list</span>(modProbes, modProbes)<br><span class="hljs-comment">## Export the network into edge and node list files Cytoscape can read</span><br>cyt = exportNetworkToCytoscape(modTOM,<br>  edgeFile = paste(<span class="hljs-string">&quot;CytoscapeInput-edges-&quot;</span>, paste(modules, collapse=<span class="hljs-string">&quot;-&quot;</span>), <span class="hljs-string">&quot;.txt&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>),<br>  nodeFile = paste(<span class="hljs-string">&quot;CytoscapeInput-nodes-&quot;</span>, paste(modules, collapse=<span class="hljs-string">&quot;-&quot;</span>), <span class="hljs-string">&quot;.txt&quot;</span>, sep=<span class="hljs-string">&quot;&quot;</span>),<br>  weighted = <span class="hljs-literal">TRUE</span>,<br>  threshold = <span class="hljs-number">0.02</span>,<br>  nodeNames = modProbes,<br>  altNodeNames = modGenes,<br>  nodeAttr = moduleColors[inModule]);<br></code></pre></td></tr></table></figure></div>
<h3 id="6-3-Network3d">6.3 Network3d</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">edges &lt;- read.table(<span class="hljs-string">&quot;CytoscapeInput-edges-brown-red.txt&quot;</span>,header=<span class="hljs-built_in">T</span>)<br>nodes &lt;- read.table(<span class="hljs-string">&quot;CytoscapeInput-nodes-brown-red.txt&quot;</span>,header=<span class="hljs-built_in">T</span>,sep=<span class="hljs-string">&#x27;\t&#x27;</span>)<br><br>colnames(nodes)= <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;nodeName&quot;</span>, <span class="hljs-string">&quot;altName&quot;</span>, <span class="hljs-string">&quot;Group&quot;</span>)<br><br>edges$fromNode = match(edges$fromNode,nodes$nodeName)<br>edges$toNode = match(edges$toNode,nodes$nodeName)<br><br>Index &lt;- read.table(text = paste(unique(edges$fromNode),<span class="hljs-number">0</span>,<span class="hljs-number">0</span>,<span class="hljs-string">&quot;undirected&quot;</span>,<span class="hljs-string">&quot;A&quot;</span>,<span class="hljs-string">&quot;B&quot;</span>,sep=<span class="hljs-string">&#x27;\t&#x27;</span>),sep=<span class="hljs-string">&#x27;\t&#x27;</span>)<br><br>colnames(Index) = colnames(edges)<br>edges  = rbind(Index,edges)<br><br>forceNetwork(Links=edges,<br>        Nodes=nodes,<br>        NodeID=<span class="hljs-string">&quot;nodeName&quot;</span>, <span class="hljs-comment">#指定节点显示的标签</span><br>        Source=<span class="hljs-string">&quot;fromNode&quot;</span>, <span class="hljs-comment">#指定Links文件中的源节点</span><br>        Target=<span class="hljs-string">&quot;toNode&quot;</span>, <span class="hljs-comment">#指定Links文件中的靶节点</span><br>        Value=<span class="hljs-string">&quot;weight&quot;</span>, <span class="hljs-comment">#设定基因间连线的宽度</span><br>        fontSize=<span class="hljs-number">20</span>, <span class="hljs-comment">#设定节点标签的字号，单位为像素</span><br>        Group=<span class="hljs-string">&quot;Group&quot;</span>, <span class="hljs-comment">#对节点进行分组，这里可根据基因的功能进行分组，配置不同颜色</span><br>        opacity=<span class="hljs-number">0.8</span>, <span class="hljs-comment">#指定图像的不透明度</span><br>        zoom=<span class="hljs-literal">TRUE</span>, <span class="hljs-comment">#是否允许图像缩放</span><br>        arrows=<span class="hljs-literal">TRUE</span>, <span class="hljs-comment">#连线是否添加箭头，显示方向</span><br>        opacityNoHover=<span class="hljs-number">0.7</span>, <span class="hljs-comment">#鼠标悬停前，节点标签的不透明度</span><br>        legend=<span class="hljs-literal">TRUE</span>, <span class="hljs-comment">#是否显示图例</span><br>        height=<span class="hljs-number">600</span>, <span class="hljs-comment">#设置图像高度</span><br>        width=<span class="hljs-number">600</span> <span class="hljs-comment">#设置图像宽度</span><br>        <span class="hljs-comment">#Nodesize = &quot;Freq_l&quot;</span><br>        <span class="hljs-comment">#radiusCalculation = &quot;d.nodesize&quot;</span><br>)<br></code></pre></td></tr></table></figure></div>
</div><div class="article-licensing box"><div class="licensing-title"><p>WGCNA Tutorial 1</p><p><a href="https://karobben.github.io/2020/06/08/R/WGCNA_1/">https://karobben.github.io/2020/06/08/R/WGCNA_1/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2020-06-08</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2023-06-06</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/R/">R</a><a class="link-muted mr-2" rel="tag" href="/tags/Bioinformatics/">Bioinformatics</a><a class="link-muted mr-2" rel="tag" href="/tags/WGCNA/">WGCNA</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2020/06/10/R/wordcloud/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">worldcloud</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2020/06/03/Bioinfor/UNIPROT_API/"><span class="level-item">UNIPROT_API</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">514</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">145</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">160</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#WGCNA-Tutorial-1"><span class="level-left"><span class="level-item">1</span><span class="level-item">WGCNA Tutorial 1</span></span></a></li><li><a class="level is-mobile" href="#1-Data-Cleaning"><span class="level-left"><span class="level-item">2</span><span class="level-item">1. Data Cleaning</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#1-1-Checking-data"><span class="level-left"><span class="level-item">2.1</span><span class="level-item">1.1 Checking data</span></span></a></li><li><a class="level is-mobile" href="#1-2-Cluster-the-samples"><span class="level-left"><span class="level-item">2.2</span><span class="level-item">1.2 Cluster the samples</span></span></a></li><li><a class="level is-mobile" href="#1-3-Remove-the-abnormal-sample"><span class="level-left"><span class="level-item">2.3</span><span class="level-item">1.3 Remove the abnormal sample</span></span></a></li><li><a class="level is-mobile" href="#1-4-Loading-Clinical-Data"><span class="level-left"><span class="level-item">2.4</span><span class="level-item">1.4 Loading Clinical Data</span></span></a></li></ul></li><li><a class="level is-mobile" href="#2-Network-construction-and-module-detection"><span class="level-left"><span class="level-item">3</span><span class="level-item">2. Network construction and module detection</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#2-1-Pick-soft-thresdholding-power"><span class="level-left"><span class="level-item">3.1</span><span class="level-item">2.1 Pick soft-thresdholding power</span></span></a></li><li><a class="level is-mobile" href="#2-2-One-step-network-construction-and-module-detection"><span class="level-left"><span class="level-item">3.2</span><span class="level-item">2.2 One-step network construction and module detection</span></span></a></li></ul></li><li><a class="level is-mobile" href="#3-Relating-modules-to-external-information-and-identifying-important-genes"><span class="level-left"><span class="level-item">4</span><span class="level-item">3. Relating modules to external information and identifying important genes</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#3-1-Clinical-Traits"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">3.1 Clinical Traits</span></span></a></li><li><a class="level is-mobile" href="#3-2-Genes-and-Trait"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">3.2 Genes and Trait</span></span></a></li></ul></li><li><a class="level is-mobile" href="#4-GO-Enrichment-of-Modules"><span class="level-left"><span class="level-item">5</span><span class="level-item">4. GO Enrichment of Modules</span></span></a></li><li><a class="level is-mobile" href="#5-Visualization"><span class="level-left"><span class="level-item">6</span><span class="level-item">5. Visualization</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#5-1-NetWork-Visualization"><span class="level-left"><span class="level-item">6.1</span><span class="level-item">5.1 NetWork Visualization</span></span></a></li><li><a class="level is-mobile" href="#5-2-Visualizing-the-network-of-eigengenes"><span class="level-left"><span class="level-item">6.2</span><span class="level-item">5.2  Visualizing the network of eigengenes</span></span></a></li></ul></li><li><a class="level is-mobile" href="#6-Export-of-networks"><span class="level-left"><span class="level-item">7</span><span class="level-item">6. Export of networks</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#6-1-Exporting-to-VisANT"><span class="level-left"><span class="level-item">7.1</span><span class="level-item">6.1  Exporting to VisANT</span></span></a></li><li><a class="level is-mobile" href="#6-2-Exporting-to-Cytoscape"><span class="level-left"><span class="level-item">7.2</span><span class="level-item">6.2 Exporting to Cytoscape</span></span></a></li><li><a class="level is-mobile" href="#6-3-Network3d"><span class="level-left"><span class="level-item">7.3</span><span class="level-item">6.3 Network3d</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Antibody/"><span class="tag">Antibody</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">72</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/CLI-Tools/"><span class="tag">CLI Tools</span><span class="tag">15</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data-Science/"><span class="tag">Data Science</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FASTA-Q/"><span class="tag">FASTA/Q</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fasta/"><span class="tag">Fasta</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immunology/"><span class="tag">Immunology</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG-pathway/"><span class="tag">KEGG pathway</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">15</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matplotlib/"><span class="tag">Matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">27</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">62</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein/"><span class="tag">Protein</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyMol/"><span class="tag">PyMol</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">132</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Pyvista/"><span class="tag">Pyvista</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">86</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripting/"><span class="tag">Scripting</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Seaborn/"><span class="tag">Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SeqKit/"><span class="tag">SeqKit</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sequencing/"><span class="tag">Sequencing</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">47</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI-lib/"><span class="tag">TUI lib</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/VCF/"><span class="tag">VCF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dash/"><span class="tag">dash</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/git/"><span class="tag">git</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/matplotlib/"><span class="tag">matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/vim/"><span class="tag">vim</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://camo.githubusercontent.com/fd83d8c9f8b8dbe85686812e2e186dceec728bbabad2d331aa44e0bfeee8628c/68747470733a2f2f70706f66666963652e6769746875622e696f2f6865786f2d7468656d652d6963617275732f67616c6c6572792f707265766965772e706e673f31" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">74</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">68</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protein-Structure/"><span class="level-start"><span class="level-item">Protein Structure</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/FastQC/"><span class="level-start"><span class="level-item">FastQC</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fastq-QC/"><span class="level-start"><span class="level-item">Fastq QC</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">41</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">11</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">133</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">13</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Immunity/"><span class="level-start"><span class="level-item">Immunity</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">75</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">8</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">12</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immunology/"><span class="level-start"><span class="level-item">Immunology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">56</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">39</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2024 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>