<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>tmp - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="tmp"><meta property="og:type" content="blog"><meta property="og:title" content="tmp"><meta property="og:url" content="https://karobben.github.io/2023/09/13/LearnNotes/tmp/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="tmp"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://karobben.github.io/"><meta property="article:published_time" content="2023-09-13T05:00:00.000Z"><meta property="article:modified_time" content="2023-09-14T01:17:21.410Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="English"><meta property="article:tag" content="Writing"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="/"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2023/09/13/LearnNotes/tmp/"},"headline":"Karobben","image":[],"datePublished":"2023-09-13T05:00:00.000Z","dateModified":"2023-09-14T01:17:21.410Z","author":{"@type":"Person","name":"Karobben"},"description":"tmp"}</script><link rel="canonical" href="https://karobben.github.io/2023/09/13/LearnNotes/tmp/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2023-09-13T05:00:00.000Z" title="9/13/2023, 12:00:00 AM">2023-09-13</time></span><span class="level-item">Updated&nbsp;<time dateTime="2023-09-14T01:17:21.410Z" title="9/13/2023, 8:17:21 PM">2023-09-13</time></span><span class="level-item"><a class="link-muted" href="/categories/Notes/">Notes</a><span> / </span><a class="link-muted" href="/categories/Notes/English/">English</a></span><span class="level-item">28 minutes read (About 4137 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">tmp</h1><div class="content"><h2 id="Abstract">Abstract</h2>
<p>In plants, Long-chain acyl-CoA synthetases (LACSs) <strong>play</strong> key roles in activating fatty acids to fatty acyl-CoA thioesters, which <strong>are</strong> then further <strong>involved</strong> in lipid synthesis and fatty acid catabolism. LACSs <strong>have</strong> been intensively <strong>studied</strong> in Arabidopsis, but its evolutionary relationship in green plants <strong>is</strong> unexplored. In this study, we <strong>performed</strong> a comprehensive genome-wide analysis of the LACS gene family across green plants <strong>followed</strong> by phylogenetic clustering analysis, gene structure determination, detection of conserved motifs, gene expression in tissues and subcellular localization. Our results <strong>identified</strong> LACS genes in 122 plant species including algae, low land plants (i.e., mosses and lycophytes), monocots, and eudicots. In total, 697 sequences <strong>were</strong> identified, and 629 sequences <strong>were</strong> selected because of alignment and some duplication errors. The retrieved amino acid sequences <strong>ranged</strong> from 271 to 1056 residues and <strong>diversified</strong> in intron/exon patterns in different LACSs. Phylogenetic clustering <strong>grouped</strong> LACS gene family into six major clades with distinct potential functions. This classification <strong>is</strong> well supported by examining gene structure and conserved motifs. Also, gene expression analysis and subcellular localization <strong>substantiate</strong> with clade division in the phylogeny, <strong>indicating</strong> that the evolutionary pattern <strong>is</strong> visible in their functionality. Additionally, experimental analysis of lacs2 mutant <strong>validated</strong> that LACS2 <strong>plays</strong> key roles in suberin synthesis. Thus, our study not only <strong>provides</strong> an evolutionary mechanism underlying functional diversification but also <strong>lays</strong> the foundation for further elucidation of the LACS gene family.</p>
<h2 id="Introduction">Introduction</h2>
<p>Long-chain acyl-CoA synthetases (LACSs) <strong>belong</strong> to the carboxyl-CoA ligase superfamily, which <strong>are</strong> also known as acyl-activating enzymes (Lu et al., 2009; Shockey and Browse, 2011). Most of the enzymes in this superfamily <strong>carry out</strong> a two-step mechanism. This two-step ping-pong reaction <strong>associates</strong> the transfer of adenylate from ATP to a fatty acid to form an enzyme-bound acyl-adenylate intermediate and pyrophosphate. Subsequently, this intermediate <strong>is attacked</strong> by a CoA to produce an acyl-CoA and an adenosine monophosphate (AMP) (Shockey and Browse, 2011). LACSs <strong>usually esterify</strong> free long-chain fatty acids (LCFAs) produced by de novo FA synthesis to form fatty acyl-CoA thioesters, which <strong>are</strong> subsequently channeled into many metabolic pathways for producing membrane lipids, storage lipids (triacylglycerols [TAGs]), surface lipids (wax, cutin and suberin) and fatty acid catabolism (Tonon et al., 2005). In addition, some LACSs <strong>are identified</strong> to also function as transporters of fatty acids between different subcellular compartments (Tonon et al., 2005; Jessen et al., 2015).</p>
<p>Arabidopsis <strong>contains</strong> one of the largest known LACS families with nine LACS genes, most of which <strong>were</strong> well characterized (Shockey et al., 2002; Zhao et al., 2019). LACS1, LACS2 and LACS4 <strong>were reported</strong> to be localized in the endoplasmic reticulum (ER) and redundantly <strong>involved</strong> in the synthesis of surface wax and/or cutin (Lu et al., 2009; Zhao et al., 2010; Jessen et al., 2011, 2015). Among these three genes, LACS1 <strong>plays</strong> a major role in modifying very long-chain fatty acids (VLCFAs, C20–C30) for wax synthesis since its mutation <strong>caused</strong> a strong wax deficient glossy phenotype (Lu et al., 2009), whereas a single mutation of either LACS2 or LACS4 only <strong>results</strong> in a slight reduction of wax components. LACS1 also <strong>involves</strong> in cutin synthesis by preferentially modifying C16:0 (Lu et al., 2009). In addition, LACS2 mutation <strong>caused</strong> strong cutin deficiency while it <strong>has</strong> little effects on wax biosynthesis under normal growth conditions, indicating that LACS2 <strong>is mainly involved</strong> in cutin biosynthesis (Schnurr et al., 2004; Lu et al., 2009). Double mutant analysis <strong>indicated</strong> that LACS1 and LACS2 <strong>have</strong> overlapping functions in both plant wax and cutin synthesis (Lu et al., 2009). Recent studies <strong>showed</strong> that LACS2 mutation also <strong>resulted</strong> in reduced wax accumulation under submergence growth conditions, which further <strong>confirmed</strong> that LACS2 <strong>is also involved</strong> in wax synthesis under stress conditions (Xie et al., 2020). Moreover, LACS4 in combination with LACS1 <strong>activates</strong> proper pollen coat formation (Jessen et al., 2011). LACS9 <strong>is</strong> the only LACS gene specifically localized in the outer membrane of the plastid envelope, which in combination with LACS4 <strong>was reported</strong> to be involved in the process of lipid flux from the ER to the plastid for glycerolipid synthesis (Jessen et al., 2015). lacs9 null mutant <strong>does not show</strong> any detectable phenotype (Schnurr et al., 2002), however, double mutant analysis <strong>indicates</strong> that LACS9 <strong>functions</strong> redundantly with either LACS1 or LACS4 in seed TAG biosynthesis (Zhao et al., 2010; Jessen et al., 2015). LACS8, another ER-specific form, also <strong>has</strong> redundant function with that of LACS4 and LACS9, since knockout of LACS8 in the lacs4 lacs9 double mutant background <strong>results</strong> in lethality (Jessen et al., 2015). Different from the above mentioned LACS members, both LACS6 and LACS7 located in peroxisome <strong>has</strong> an overlapping role in β-oxidation (Fulda et al., 2002, 2004). The overall function of LACS3 and LACS5 <strong>is not revealed</strong> till now. As mentioned above, similar to the function of LACS gene families characterized in other Eukaryotes species, Arabidopsis LACS genes not only <strong>vary</strong> in their subcellular locations, expression patterns, and substrates specificities but also <strong>present</strong> a complex network of redundant LACS activities <strong>involved</strong> in different aspects of lipid metabolism (Digel et al., 2009; Zhao et al., 2019).</p>
<p>Even though the LACS genes from Arabidopsis <strong>are</strong> well characterized, however, only a few LACS homologs from other plant species <strong>have been</strong> characterized. GhACS1, an orthologue of the Arabidopsis LACS4 or LACS5, <strong>is</strong> predominantly accumulated in the anther and <strong>plays</strong> an essential role in microsporogenesis in anther development of Cotton (Wang and Li, 2009). GmACSL2, a peroxisomal isoform of LACS from soybean, <strong>facilitates</strong> fatty acid and lipid degradation during seed germination (Yu et al., 2014). HaLACS1 and HaLACS2 of sunflower, with high sequence homology with LACS9 and LACS8 genes of Arabidopsis, <strong>are</strong> highly expressed in developing seeds and <strong>perform</strong> an essential role in oil synthesis of sunflower seeds (Aznar-Moreno et al., 2014), which <strong>is</strong> similar to the function of LACS9 in Arabidopsis. Unlike AtLACS2, which <strong>is</strong> specifically expressed in the epidermis for both cutin and wax synthesis, its orthologue in Brassica napus BnLACS2 <strong>is</strong> predominantly expressed in developing seeds and <strong>involved</strong> in seed oil production (Ding et al., 2020). LuLACS8A <strong>identified</strong> from flax (Linum usitatissimum) <strong>shows</strong> substrate preference towards α-linoleic acid and <strong>contributes</strong> to the biosynthesis of α-linolenic acid enriched in flaxseed oil cooperating with LuDGAT2 (Magwanga et al., 2018).</p>
<h2 id="2-Materials-and-methods">2. Materials and methods</h2>
<h3 id="2-1-LACS-genes-identification-and-their-homologs-in-plants">2.1. LACS genes identification and their homologs in plants</h3>
<p>Despite the above intriguing and significant functions, it <strong>is</strong> still necessary to explore the evolutionary relationship of LACS family since it <strong>can</strong> reveal existing functional data from different lineages and also <strong>provides</strong> some clues to explore the possible functions of those unknown LACS members in different plant species. Thus, we <strong>made</strong> a genome-wide comparative analysis, <strong>constructed</strong> a phylogenetic tree, and <strong>examined</strong> gene structure and gene expression profiling. These studies <strong>will address</strong> these questions: (1) How plants’ LACSs genes <strong>are</strong> classified in various clades based on a phylogenetic approach. (2) Is there any difference in gene structure, composition, and expression patterns of LACSs genes among various plants. (3) How different isoforms of LACSs genes <strong>are</strong> expressed at the subcellular level and in various tissues.</p>
<p>We <strong>performed</strong> an extensive literature survey to search LACSs genes and their isoforms that have already <strong>been</strong> reported and characterized in A. thaliana. Subsequently, we <strong>retrieved</strong> A. thaliana sequences keywords (LACS1 [AT2G47240], LACS2 [AT1G49430], LACS3 [AT1G64400], LACS4 [AT4G23850], LACS5 [AT4G11030], LACS6 [AT3G05970], LACS7 [AT5G27600], LACS8 [AT2G04350], and LACS9 [AT1G77590]) using Basic Local Alignment Tool (BLASTp and TBLASTx) searches in the Phytozome (<a target="_blank" rel="noopener" href="http://www.phytozome.net">www.phytozome.net</a>) and GenomeNet (<a target="_blank" rel="noopener" href="http://www.genome.jp/tools/blast">www.genome.jp/tools/blast</a>) databases. The downloaded sequences <strong>were</strong> compiled into a single FASTA file. Protein classification <strong>was</strong> carried out by utilizing Pfam (<a target="_blank" rel="noopener" href="http://Pfam.sanger.ac.uk/search">Pfam.sanger.ac.uk/search</a>) (Punta et al., 2011) and PANTHER (<a target="_blank" rel="noopener" href="http://www.pantherdb.org/">www.pantherdb.org/</a>) (Mi et al., 2013) classification systems. In our database, all taxa <strong>were</strong> designated by four letters acronym, where the first letter represents the genus and the next three letters indicate the first three letters of the species, for example, A tha assigned to A. thaliana. These acronyms <strong>used</strong> in all analyses <strong>indicated</strong> with the accession numbers (e.g., A_tha_AT1G77590 corresponds to A. thaliana LACS9), but for phylogenetic tree analysis, we <strong>used</strong> the full name of species with their IDs. We <strong>provided</strong> a comprehensive description of accession numbers, protein length, protein domain presence, and intron number <strong>are provided</strong> in supplementary tables S1–S4.</p>
<h3 id="2-2-Alignment-and-phylogeny-inference">2.2. Alignment and phylogeny inference</h3>
<p>DNA and amino acid sequences <strong>were adjusted</strong> through MAFFT (Katoh and Toh, 2010) alignment accessible in CIPRES Science Gateway (CIPRES) (Miller et al., 2010). Changes <strong>were made</strong> manually in BioEdit v.7.0 (Hall et al., 2011), whereas ambiguously aligned positions <strong>were removed</strong> in the final analysis. To select the best model for protein evolution, Smart Model Selection (SMS) <strong>was used</strong> (Lefort et al., 2017), which <strong>showed</strong> JTT + I + G model for protein sequences based on the lowest BIC (Bayesian Information Criterion) score (97,536) in PhyML (Guindon et al., 2010). JModelTest (Posada, 2008) <strong>was selected</strong> for nucleotide evolution, which <strong>revealed</strong> General Time Reversible model with the shape of Gamma distributions plus the proportions of variables sites (GTR + I + G) as the best fit model to reconstruct the phylogenetic relationship of nucleotides. We <strong>reconstruct</strong> ML phylogenetic tree using MrBayes 3.2.7 (Ronquist et al., 2012) in CIPRES with 1,000,000 generations with four Markov chain Monte Carlo (MCMC) and two runs. The reliability of the phylogenetic analysis <strong>was verified</strong> through the construction of an ML tree utilizing FastTree v2.1.8 (Price et al., 2010) with the best fit model. The tree <strong>was visualized</strong> and annotated in iTOL with default sorting (Letunic and Bork, 2019).</p>
<h3 id="2-3-Gene-structure-analysis-and-conserved-motif-detection">2.3. Gene structure analysis and conserved motif detection</h3>
<p>Intron/exon pattern <strong>was determined</strong> using an online gene structure display server (Guo et al., 2007); <a target="_blank" rel="noopener" href="http://gsds.cbi.pku.edu.cn">gsds.cbi.pku.edu.cn</a>). To retrieve gene structure in addition to the phylogenetic tree, a multi FASTA format sequence set based on genomic and CDS for LACS genes of selected species (which <strong>were</strong> common in all clades) <strong>was submitted</strong> to GSDraw. The MEME program (<a target="_blank" rel="noopener" href="http://www.meme-suite.org/meme/tools/meme">www.meme-suite.org/meme/tools/meme</a>) <strong>was used</strong> to find significant functional motifs of LACS (Bailey et al., 2015). We <strong>adjusted</strong> the parameters as: distribution of motifs = any number of repetitions, the maximum number of motifs = 20, and the optimum motif width = 3 to 300 residues. The identified motifs <strong>were further analyzed</strong> by Pfam analysis for protein classification (Punta et al., 2011); <a target="_blank" rel="noopener" href="http://Pfam.sanger.ac.uk/search">Pfam.sanger.ac.uk/search</a>).</p>
<h3 id="2-4-Gene-expression-prediction-and-subcellular-localization">2.4. Gene expression prediction and subcellular localization</h3>
<p>To examine tissue specificity and expression intensity of representative species, GENEVESTIGATOR website (<a target="_blank" rel="noopener" href="http://www.genevestigator.com/gv/">www.genevestigator.com/gv/</a>) (Hruz et al., 2008) <strong>was used</strong>. The hierarchical clustering tool available in GENEVESTIGATOR <strong>was utilized</strong> to carry out this analysis. The highest expression values for genes containing more than one probe set <strong>were considered</strong>. Pearson coefficient <strong>was utilized</strong> to classify gene-wise expression data. We <strong>represented</strong> percent expression of various anatomical regions and developmental stages from LACS and putative LACS genes in heat maps.</p>
<p>To better understand and predict the subcellular localization of different LACS for various functions, we <strong>used</strong> an online website (<a target="_blank" rel="noopener" href="http://bioinfo.usu.edu/Plant-mSubP">bioinfo.usu.edu/Plant-mSubP</a>) (Sahu et al., 2019). The full-length protein sequences <strong>were uploaded</strong>, and different approaches <strong>were used</strong> based on Amino Acid Composition (AAC), Dipeptide Composition (Dipep), Pseudo Amino Acid Composition (PseAAC), and N-Center-C terminal Composition (NCC). This approach successfully predicts 11 single locations (cell membrane, cell wall, plastid, cytoplasm, ER, extracellular, Golgi apparatus, mitochondrion, nucleus, peroxisome, and vacuole) and three significant multi-location proteins (cytoplasm-nucleus, mitochondrion-plastid, and cytoplasm-Golgi apparatus).</p>
<h3 id="2-5-Suberin-analysis">2.5. Suberin analysis</h3>
<p>Roots of 10-day-old seedlings of lacs2-3 (GABI-Kat line 368C02) and Col-0 <strong>grown</strong> on 1/2 MS medium agar plates <strong>were harvested</strong> and immediately <strong>immersed</strong> in hot isopropanol at 80 °C for 10 min. After cooling to room temperature, samples <strong>were stirred</strong> for 1–2 h. Then <strong>centrifuged</strong> the samples for 5 min at 2000 rpm and <strong>discarded</strong> the isopropanol. <strong>Added</strong> the same volume of isopropanol and <strong>stirred</strong> for the whole day. For delipidation, the isopropanol <strong>was then replaced</strong> by CHCl3/CH3OH (2/1 v/v), CHCl3/CH3OH (1/2 v/v), and CH3OH, respectively, and <strong>stirred</strong> the whole day. Then, <strong>centrifuged</strong> the samples for 10 min at 2000 rpm, <strong>removed</strong> solvent, and <strong>dried</strong> it under a hood overnight. The samples <strong>were finally dried</strong> in a vacuum desiccator for two days. Dried tissues <strong>were then incubated</strong> in 5 ml of depolymerization solution (0.9 ml methyl acetate, 1.5 ml sodium methoxide, and 3.6 ml methanol) at 60°C for 2 h. 25 μg methyl heptadecanoate <strong>was used</strong> as an internal standard. After cooling, the fatty acid methyl esters <strong>were extracted</strong> by <strong>adding</strong> methylene dichloride (10 ml) and (0.5 ml) glacial acetic acid to pH 4–5 to the depolymerized samples. The organic layer <strong>was washed</strong> three times with 10 ml saline solution (0.5 M NaCl), <strong>dried</strong> with anhydrous sodium sulfate (6 g), and then <strong>evaporated</strong> under nitrogen gas. Monomers <strong>were derivatized</strong> in 100 μl of pyridine and 100 μl of acetic anhydride at 60 °C for 1 h. After cooling, <strong>evaporated</strong> under nitrogen at room temperature and <strong>redissolved</strong> in 100 μl heptane:toluene (1:1, v/v). The composition and amount of cutin monomers <strong>were analyzed</strong> by Agilent 8860 series GC equipped with an FID and a 30 m, 0.25 mm internal diameter DB-5 capillary column, with helium as the carrier gas. The GC <strong>was programmed</strong> with an initial temperature of 80 °C and <strong>increased</strong> at 15 °C min−1 to 200 °C, then <strong>increased</strong> at 2 °C min−1 to 300 °C, <strong>maintained</strong> for 10 min at 300 °C. Quantification <strong>was based</strong> on uncorrected FID peak areas relative to the internal standard methyl heptadecanoate peak area.</p>
<h2 id="3-Results">3. Results</h2>
<h3 id="3-1-Genome-wide-identification-of-LACS-homologs-sequences-in-plants">3.1. Genome-wide identification of LACS homologs sequences in plants</h3>
<p>Nine genes <strong>are</strong> annotated as LACS in Arabidopsis thaliana. Amino acid sequences <strong>encoding</strong> by these genes <strong>were used</strong> as queries. To <strong>search</strong> LACS homologs genes, a broad survey of fully sequenced genomes <strong>was carried out</strong> in total 122 plant species, which <strong>are obtained</strong> from two websites (<a target="_blank" rel="noopener" href="http://www.genome.jp/">www.genome.jp/</a>(64 plants) and <a target="_blank" rel="noopener" href="http://phytozome.jgi.doe.gov/pz/portal.html">phytozome.jgi.doe.gov/pz/portal.html</a> (106 plants) (Table S2). The whole data set <strong>consists of</strong> Rhodophyta (2), Chlorophyta (13), Embryophyta (3), Tracheophyte (1), and Angiosperm (105). BLASTp and BLASTx <strong>were performed</strong> against these LACS gene and their putative proteins. Searching BLASTp, multiple hits <strong>were selected</strong> for all plants. Although the range <strong>is</strong> indeed 271–1056, most (97%) of sequences <strong>have</strong> between 608 and 737 amino acids. The latter range <strong>is</strong> highly consistent for LACS proteins from numerous non-plant species. We <strong>identified</strong> a total of 697 sequences, while 629 sequences <strong>were considered</strong> due to alignment and duplication errors. The copy number of LACS genes <strong>were</strong> close among all lineages ranging from a single copy <strong>detected</strong> in Rhodophyta, Chlorophyta, Embryophyte, and Ferns to the nine sequences in order Brassicales and Poales (Table S1). To verify the reliability of BLAST results, InterproScan and Pfam analysis <strong>was performed</strong> on all protein sequences while most of them <strong>were classified</strong> into the LACS family. Sequence information <strong>is provided</strong> in Supplementary Table S2.</p>
<h3 id="3-2-Phylogenetic-relationships">3.2. Phylogenetic relationships</h3>
<p>Evolutionary relationships among the plants LACS <strong>were investigated</strong> by reconstructing the phylogenetic tree using protein sequences of A. thaliana and supposed LACS identified by homology searches in 122 species. The phylogenetic analysis of LACS amino acid sequences <strong>appeared</strong> in a well-resolved tree by ML and Bayesian methods, <strong>showing</strong> the formation of six main clades with one algal and ferns sequences (Fig. 1). The clade I &amp; II <strong>include</strong> LACS1 &amp; 2 respectively and some hypothetical sequences. Clade III <strong>consists</strong> of LACS3-5, while LACS6 and 7 <strong>were dominated</strong> in clade IV. Subsequently, clade V <strong>includes</strong> LACS9, and clade VI <strong>contains</strong> LACS8 (Table S2, Figure S2). Some of the algal LACS sequences <strong>revealed</strong> a separate outgroup with different LACS. Clade I <strong>indicated</strong> the most ancient forms as well as a divergence from the land plants.</p>
<p>In all clades, algal, bryophytes, fern (Selaginella moellendorffii), and basal angiosperm (Amborella trichopoda) <strong>were phylogenetically divergent</strong> from land plants. Within clade I, Kalanchoe laxiflora, Kalanchoe fedtschenkoi, Trifolium pratense, Panicum virgatum <strong>contain</strong> more than one sequence. Medicago truncatula, Glycine max, Lotus japonicus in clade II <strong>contains</strong> more than one sequence while in clade III and IV, 27 and 26 species <strong>contain</strong> more than one sequence, respectively. More than one sequence in many clades <strong>possibly presenting</strong> gene duplication events.</p>
<p>Clade III <strong>was</strong> the largest clade <strong>consist</strong> of 120 sequences of LACS3-5 of all angiosperms with 25, 16, 10, 8 Brassicaceae, Fabaceae, Solanaceae, Cucurbitaceae family genes respectively accounting for 18% of all LACS genes. Clade IV <strong>was</strong> the 2nd largest clade <strong>consists</strong> of 115 sequences of LACS6 and 7 having 17 genes from Brassicaceae, 12 genes from Fabaceae, 10 genes from Malvaceae, 8 genes from Euphorbiaceae, and 6 genes from Chlorophyta of three different families (Mamiellaceae, Chlorellaceae, Bathycoccaceae) (Figure S1 and Table S1). Overall, the phylogenetic relationship of LACS genes from all species <strong>reveals</strong> that LACS genes <strong>are</strong> broadly distributed in flowering plants especially in order Brassicales the copy number <strong>was expanded</strong> in most species.</p>
<h3 id="3-3-LACS-gene-structure-and-organization-analysis">3.3. LACS gene structure and organization analysis</h3>
<p>We <strong>examined</strong> the intron/exon organization for each gene to explore the structural diversity of LACS genes as shown in Fig. 2 (details in Table S3). Our results <strong>disclosed</strong> intron number of these genes <strong>ranged</strong> from 5 to 22. Many of the selected LACS retrieved by BLAST searches <strong>contain</strong> more than seven introns except for Manihot esculenta in clade VI. Among LACS clades, the intron number and gene structure <strong>were relatively conserved</strong>. The maximum of the LACS from land plants <strong>is</strong> up to 22 introns, indicating a potential increase of introns in land plant LACS. Species close to each other in the phylogenetic tree <strong>show</strong> similar intron number. For example, Clade IV <strong>had</strong> the maximum number of introns except for one species (Medicago truncatula) <strong>having</strong> ten introns. The maximum number of introns in Clade I-IV <strong>were</strong> 18, while Clade V and VI <strong>had</strong> fewer introns ranging from 5 to 10 (Fig. 2). We <strong>investigated</strong> intron phases across all LACS of representative species and the results <strong>showed</strong> quite variable intron phase pattern across plant LACS. Phase 2 <strong>was</strong> dominant across genes with more than one intron. The intron phase 2,0,0,0,1 <strong>is</strong> significantly conserved across (LACS2-4) that grouped in clades I–III (Fig. 2).</p>
<p>The conserved or consensus sequence in the protein or nucleotide sequence <strong>is known</strong> as a motif (Magwanga et al., 2018). Conserved motifs and gene structural variation <strong>are</strong> potential components for the evolution and diversity of the multigene family. We <strong>examined</strong> the conserved motifs in all studied species by MEME in order to further explore diversification of the LACS gene family. In total, 20 motifs <strong>were identified</strong> in LACS. Motifs 1, 2, 4, 6, 8, 12, 13, 14, 16 as shown in Fig. 3 <strong>were</strong> common in all clades. Most of the motifs <strong>share</strong> all clades, except for motif 19 that <strong>was</strong> only present in Clade III, V, and VI. Among different amino acids conserved in the 20 motifs, Leucine frequency <strong>was</strong> the highest (0.0907), followed by Glycine (0.0833), Valine (0.0798), Lysine (0.0726), and Glutamic acid (0.069), etc., while the minimum frequency <strong>was</strong> 0.0141 for the Tryptophan amino acid. According to InterProScan and GenomeNet annotations, Motifs 1–6, and 8–11 <strong>were</strong> an AMP-binding enzyme domain. A number of specific amino acids residues within the motifs <strong>were</strong> associated with AMP-binding signatures, as reported in previous studies, i.e. for motifs 1 and 4 (Watkins and Ellis, 2012), for motifs 2 and 4 by (Weimar et al., 2002), and for motifs 4 and 19 by (Andersson et al., 2012). These motifs <strong>are</strong> functionally important for LACS proteins, while the remaining conserved motifs <strong>were not</strong> annotated in InterProScan and GenomeNet yet. All these identified motifs <strong>were</strong> part of AMP-dependent synthetase which <strong>play</strong> an important role in lipid biosynthesis and distribution within the cell.</p>
<h3 id="3-4-Evaluation-of-LACS-subcellular-localization">3.4. Evaluation of LACS subcellular localization</h3>
<p>Protein properties <strong>were evaluated</strong> including protein length and their subcellular location (Table S4). The sequence identity of all studied LACS with that of A. thaliana <strong>was compared</strong> in NCBI. The identified sequences <strong>showed</strong> similarity with LACS1-9 sequences mostly ranging from 33% (Volvox carteri) to 99.28% (Arabidopsis helleri in Clade V), indicating a close relationship among all LACS genes. The prediction of subcellular locations <strong>showed</strong> that members of LACS1-5 (Clade 1–3) mostly <strong>found</strong> in the endoplasmic reticulum and cytoplasm. Our results <strong>revealed</strong> isoforms belonging to the same clade <strong>have</strong> different subcellular localization, it <strong>may be likely</strong> due to large and small-scale changes in the structure and functions of genes among species over evolutionary time. This diversification <strong>facilitates</strong> an organism to evolve new genes that <strong>participate</strong> in various pathways occurring in different subcellular localizations (Shockey and Browse, 2011). LACS6 and 7 (Clade IV) <strong>were predicted</strong> to be localized in plastid and peroxisomes. LACS8 (Clade VI) <strong>was mostly reported</strong> to locate in the ER and some of them <strong>were predicted</strong> to locate in the plastid. About 82% of LACS9 (Clade V) <strong>were predicted</strong> to locate in the plastid. 180 out of 629 sequences <strong>were predicted</strong> to locate in the ER and 175 <strong>were localized</strong> in the plastid. The lowest number (6) of sequences <strong>were predicted</strong> to locate in the cell membrane (Table S4; Fig. 4). As the subcellular localization of most Arabidopsis LACS <strong>were experimentally determined</strong>, the prediction <strong>show</strong> in this study with other species <strong>may be due to</strong> gene duplication events but the mechanism underlying functional divergence <strong>are not well understood</strong> due to alteration of single amino acid replacement or divergence.</p>
<h3 id="3-5-Expression-profiling-of-LACS-genes">3.5. Expression profiling of LACS genes</h3>
<p>Certainly! Here’s the paragraph with the verbs highlighted:</p>
<p>The functional roles of each LACS <strong>were analyzed</strong> on GENEVESTIGATOR from the available plant expression data. To assess spatial and temporal patterns of the LACS genes in plant tissue, microarray expression data of A. thaliana (Table 1; Figure S2, S3), G. max (Table 1, Figure S4, S5), and N. tabacum (Table 1; Figure S6, S7) <strong>were utilized</strong>, while for monocots, we <strong>found</strong> data for O. sativa (Table 1; Figure S8, S9) and Z. mays (Table 1; Figure S10, S11). In eudicots, for A. thaliana we <strong>found</strong> all probes in this study. For G. max, four out of 18 genes <strong>found</strong> in BLAST searches <strong>presented</strong> available probes and for N. tabacum we <strong>found</strong> a total of 32 probes, among them 6 <strong>were present</strong> in this study. In monocots, for Z. mays we <strong>identified</strong> 3 available probes, whereas 31 probes for O. sativa in which 8 <strong>were present</strong> in our study (Table 1). We <strong>investigated</strong> 105 anatomical and 10 developmental stages within A. thaliana, whereas 48 anatomical and 5 developmental from G. max, 9 anatomical and 4 developmental from N. tabacum, 31 anatomical and 8 developmental from O. sativa, 57 anatomical and 7 developments from Z. mays.</p>
<p>The highlighted verbs in this paragraph are: were analyzed, were utilized, found, presented, were present, identified, investigated.</p>
<p>Here’s the paragraph with the verbs highlighted:</p>
<p>In silico analysis of gene expression profile <strong>revealed</strong> that all plant LACS genes <strong>present</strong> specific expression patterns throughout different anatomical and developmental stages. Detail of different expression patterns across various tissues and plant developmental stages <strong>is present</strong> in Table 1. We <strong>found</strong> that LACS8 in A. thaliana <strong>are highly expressed</strong> in root and seed coat, whereas LACS9 <strong>is preferentially expressed</strong> in the embryo, endosperm, stamen, and chalazal endosperm. LACS7 <strong>shows high expression</strong> in root and leaf protoplast and endosperm, giant cell, shoot apex, micropylar and chalazal endosperm. Besides, LACS6 <strong>presents in</strong> root epidermis and root cap protoplast, senescence leaf, and leaf protoplast. The A. thaliana LACS5 <strong>was found active</strong> in stamen, pollen, phloem, internode, starch sheath endodermis, micropylar, and cambium cells while, LACS4 <strong>is highly expressed</strong> in the sperm cell, guard cell, root protoplast, senescent leaf, and leaf protoplast suspensor. The LACS3 genes <strong>are predominantly present</strong> in leaf protoplast, vascular tissue, cotyledon, stem, bundle sheath cell, and guard cell, whereas LACS2 <strong>expressed in</strong> inflorescence, primary root, root endodermis, quiescent center cell, flower, pistil, petal and leaf pavement cell. LACS1 of A. thaliana <strong>showed high expression</strong> in the shoot apex, inflorescence, leaf primordia, petal, stamen, guard cell, and pedicel. The Z. mays LACS6 genes included in Clade IV <strong>are predominantly expressed</strong> in sperm cells, plumule, apical cells, tassel, and ear shoot apex. The O. sativa LOC_Os05g04170 included in Clade I <strong>is more expressed</strong> in seedling, leaf, stigma, panicle, shoot, stem, internode, peduncle, and crown. O. sativa LOC_Os03g62850 included in Clade III <strong>is expressed in</strong> panicle and lamina whereas, LOC_Os12g07110 included in Clade V <strong>is eminently present</strong> in the sperm cell, pollen, seedling, leaf, shoot, and blade. The O. sativa LOC_Os01g46750 grouped in Clade III <strong>is highly expressed</strong> in the sperm cell, and root. The N. tabacum LACS9 (Nitab4.5_0001018g0030) gene classified in Clade V and LACS1 (Nitab4.5_0000222g0130) grouped in Clade I <strong>is highly present</strong> in flower, shoot, leaf. LACS6 (Nitab4.5_0000137g0180) included in Clade IV of N. tabacum <strong>is highly expressed</strong> in cell culture, flower, and capsule. The N. tabacum LACS2 (Nitab4.5_0003281g0070) grouped in Clade II <strong>is depicted in</strong> flower, shoot, stalk, and blade. LACS8 (Nitab4.5_0005376g0080) <strong>comprised in</strong> Clade VI <strong>is expressed</strong> in flower, stalk, and root, while LACS4 (Nitab4.5_0002944g0020) included in Clade III, <strong>is present</strong> in cell culture and flower. The G. max LACS4 (Glyma.01G225200) grouped in Clade III and LACS9 (Glyma.04G194200) included in Clade V <strong>are predominantly expressed</strong> gene in parenchyma cell, mesophyll cell, shoot, root hair, testa, and hypocotyl while, LACS1 (Glyma.10G010800) subsumed in Clade I <strong>is expressed</strong> in palisade layer, epidermis, and endothelium. The LACS8 (Glyma.20G060100) of G. max grouped in Clade VI <strong>is present</strong> in blade, parenchyma, cotyledon, root, apical meristem, pericycle and testa.</p>
<p>The highlighted verbs in this paragraph are: revealed, is present, found, are highly expressed, is preferentially expressed, shows, presents in, was found active, is highly expressed, are predominantly present, expressed in, is depicted in, comprised in, is eminently present, is highly present, is depicted, is present, are predominantly expressed, is expressed, and subsumed in.</p>
</div><div class="article-licensing box"><div class="licensing-title"><p>tmp</p><p><a href="https://karobben.github.io/2023/09/13/LearnNotes/tmp/">https://karobben.github.io/2023/09/13/LearnNotes/tmp/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2023-09-13</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2023-09-13</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/English/">English</a><a class="link-muted mr-2" rel="tag" href="/tags/Writing/">Writing</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2023/09/07/LearnNotes/UIUC-AdMC-6/"><span class="level-item">Molecule and Cellular Biology 6</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">487</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">145</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">165</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Bilibili" href="https://space.bilibili.com/393056819"><i class="fas fa-video"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#Abstract"><span class="level-left"><span class="level-item">1</span><span class="level-item">Abstract</span></span></a></li><li><a class="level is-mobile" href="#Introduction"><span class="level-left"><span class="level-item">2</span><span class="level-item">Introduction</span></span></a></li><li><a class="level is-mobile" href="#2-Materials-and-methods"><span class="level-left"><span class="level-item">3</span><span class="level-item">2. Materials and methods</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#2-1-LACS-genes-identification-and-their-homologs-in-plants"><span class="level-left"><span class="level-item">3.1</span><span class="level-item">2.1. LACS genes identification and their homologs in plants</span></span></a></li><li><a class="level is-mobile" href="#2-2-Alignment-and-phylogeny-inference"><span class="level-left"><span class="level-item">3.2</span><span class="level-item">2.2. Alignment and phylogeny inference</span></span></a></li><li><a class="level is-mobile" href="#2-3-Gene-structure-analysis-and-conserved-motif-detection"><span class="level-left"><span class="level-item">3.3</span><span class="level-item">2.3. Gene structure analysis and conserved motif detection</span></span></a></li><li><a class="level is-mobile" href="#2-4-Gene-expression-prediction-and-subcellular-localization"><span class="level-left"><span class="level-item">3.4</span><span class="level-item">2.4. Gene expression prediction and subcellular localization</span></span></a></li><li><a class="level is-mobile" href="#2-5-Suberin-analysis"><span class="level-left"><span class="level-item">3.5</span><span class="level-item">2.5. Suberin analysis</span></span></a></li></ul></li><li><a class="level is-mobile" href="#3-Results"><span class="level-left"><span class="level-item">4</span><span class="level-item">3. Results</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#3-1-Genome-wide-identification-of-LACS-homologs-sequences-in-plants"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">3.1. Genome-wide identification of LACS homologs sequences in plants</span></span></a></li><li><a class="level is-mobile" href="#3-2-Phylogenetic-relationships"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">3.2. Phylogenetic relationships</span></span></a></li><li><a class="level is-mobile" href="#3-3-LACS-gene-structure-and-organization-analysis"><span class="level-left"><span class="level-item">4.3</span><span class="level-item">3.3. LACS gene structure and organization analysis</span></span></a></li><li><a class="level is-mobile" href="#3-4-Evaluation-of-LACS-subcellular-localization"><span class="level-left"><span class="level-item">4.4</span><span class="level-item">3.4. Evaluation of LACS subcellular localization</span></span></a></li><li><a class="level is-mobile" href="#3-5-Expression-profiling-of-LACS-genes"><span class="level-left"><span class="level-item">4.5</span><span class="level-item">3.5. Expression profiling of LACS genes</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/AI/"><span class="tag">AI</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Beginner/"><span class="tag">Beginner</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">61</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">38</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">10</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Desktop/"><span class="tag">Desktop</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitHub/"><span class="tag">GitHub</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immune/"><span class="tag">Immune</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG-pathway/"><span class="tag">KEGG pathway</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matplotlib/"><span class="tag">Matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind/"><span class="tag">Mind</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Multimedia/"><span class="tag">Multimedia</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Odds-Ratio/"><span class="tag">Odds Ratio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">61</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">132</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Quantum-Physics/"><span class="tag">Quantum Physics</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">85</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA/"><span class="tag">RNA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-seq/"><span class="tag">RNA-seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Seaborn/"><span class="tag">Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">47</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI-lib/"><span class="tag">TUI lib</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Ted/"><span class="tag">Ted</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Uniprot/"><span class="tag">Uniprot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/awk/"><span class="tag">awk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/busuanzi/"><span class="tag">busuanzi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dash/"><span class="tag">dash</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dme/"><span class="tag">dme</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/fasta/"><span class="tag">fasta</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flies/"><span class="tag">flies</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">41</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot2/"><span class="tag">ggplot2</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/grep/"><span class="tag">grep</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/live2d/"><span class="tag">live2d</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/math/"><span class="tag">math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/matplotlib/"><span class="tag">matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/wet/"><span class="tag">wet</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://space.bilibili.com/393056819" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">Bilibili</span></span><span class="level-right"><span class="level-item tag">space.bilibili.com</span></span></a></li><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://camo.githubusercontent.com/fd83d8c9f8b8dbe85686812e2e186dceec728bbabad2d331aa44e0bfeee8628c/68747470733a2f2f70706f66666963652e6769746875622e696f2f6865786f2d7468656d652d6963617275732f67616c6c6572792f707265766965772e706e673f31" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">57</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">51</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">22</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/FastQC/"><span class="level-start"><span class="level-item">FastQC</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fastq-QC/"><span class="level-start"><span class="level-item">Fastq QC</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">135</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">11</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">75</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">8</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">6</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Ted/"><span class="level-start"><span class="level-item">Ted</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/flies/"><span class="level-start"><span class="level-item">flies</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immune/"><span class="level-start"><span class="level-item">Immune</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">83</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">55</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">38</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2023 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>