<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>Bio Cell - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="s"><meta property="og:type" content="blog"><meta property="og:title" content="Bio Cell"><meta property="og:url" content="https://karobben.github.io/2024/01/17/LearnNotes/BioCell/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="s"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://virtuallaboratory.colorado.edu/BioFun-Support/labs/Pulse-Chase/graphics/pulse-chase.jpg"><meta property="og:image" content="https://imgur.com/HAUz0uQ.png"><meta property="og:image" content="https://imgur.com/nwUgxkr.png"><meta property="og:image" content="https://imgur.com/ls9R2dL.png"><meta property="og:image" content="https://imgur.com/3i2AulZ.png"><meta property="og:image" content="https://imgur.com/dDsGcAX.png"><meta property="og:image" content="https://imgur.com/WJu1HU2.png"><meta property="og:image" content="https://imgur.com/qxHbfoH.png"><meta property="og:image" content="https://imgur.com/YxKEp4w.png"><meta property="og:image" content="https://imgur.com/wMjlieO.png"><meta property="og:image" content="https://imgur.com/1sCk6f2.png"><meta property="og:image" content="https://imgur.com/p4aOs8L.png"><meta property="og:image" content="https://imgur.com/AAZcCgG.png"><meta property="og:image" content="https://imgur.com/6THdjfF.png"><meta property="og:image" content="https://imgur.com/NwiOu8P.png"><meta property="article:published_time" content="2024-01-17T06:00:00.000Z"><meta property="article:modified_time" content="2023-11-07T04:54:24.805Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="Biology, Data Science, Bioinformatics, R, Python, Linux, ggplot, English"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://virtuallaboratory.colorado.edu/BioFun-Support/labs/Pulse-Chase/graphics/pulse-chase.jpg"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2024/01/17/LearnNotes/BioCell/"},"headline":"Karobben","image":["https://virtuallaboratory.colorado.edu/BioFun-Support/labs/Pulse-Chase/graphics/pulse-chase.jpg","https://imgur.com/HAUz0uQ.png","https://imgur.com/nwUgxkr.png","https://imgur.com/ls9R2dL.png","https://imgur.com/3i2AulZ.png","https://imgur.com/dDsGcAX.png","https://imgur.com/WJu1HU2.png","https://imgur.com/qxHbfoH.png","https://imgur.com/YxKEp4w.png","https://imgur.com/wMjlieO.png","https://imgur.com/1sCk6f2.png","https://imgur.com/p4aOs8L.png","https://imgur.com/AAZcCgG.png","https://imgur.com/6THdjfF.png","https://imgur.com/NwiOu8P.png"],"datePublished":"2024-01-17T06:00:00.000Z","dateModified":"2023-11-07T04:54:24.805Z","author":{"@type":"Person","name":"Karobben"},"description":"s"}</script><link rel="canonical" href="https://karobben.github.io/2024/01/17/LearnNotes/BioCell/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2024-01-17T06:00:00.000Z" title="1/17/2024, 12:00:00 AM">2024-01-17</time></span><span class="level-item">Updated&nbsp;<time dateTime="2023-11-07T04:54:24.805Z" title="11/6/2023, 10:54:24 PM">2023-11-06</time></span><span class="level-item">38 minutes read (About 5664 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">Bio Cell</h1><div class="content"><h2 id="L15-Protein-targeting-to-membranous-organelles">L15: Protein targeting to membranous organelles</h2>
<h3 id="Ways-of-organelles-separation">Ways of organelles separation</h3>
<ol>
<li>Centrifugation allows separation of subcellular components. (<mark>density gradient</mark>)</li>
<li>Scoring enzymatic activities across subcellular fractionation.</li>
<li><mark>Pulse chase analysis</mark>: Pulse labeling of exocrine reveals sites of translation
<ul>
<li>EM autoradiograph after a 7 min chase</li>
</ul>
</li>
<li>Choosing the right cell to study the process of interest.
<ul>
<li>Pancreatic acing secretes massive amounts of protein.</li>
</ul>
</li>
<li>How would you interpret the results?
<ul>
<li>Cartoon summary of how secreted proteins move through the cell ca. 1967.</li>
</ul>
</li>
</ol>
<div class="admonition note"><p class="admonition-title">Pulse Chase Analysis
</p><ol>
<li><strong>Pulse Phase</strong>: exposing cells to a high concentration of a labeled compound</li>
<li><strong>Pulse-chase</strong>: provided with an excess of the unlabeled version of the compound</li>
<li><strong>Analysis</strong>: collected at various time points, and the proteins are analyzed
<img src="https://virtuallaboratory.colorado.edu/BioFun-Support/labs/Pulse-Chase/graphics/pulse-chase.jpg" alt="" /></li>
</ol>
</div>
<h3 id="How-are-proteins-directed-to-the-ER">How are proteins directed to the ER?</h3>
<ol>
<li>
<p>Use a biochemical approach to dissect how proteins enter the secretory pathway at the rough ER.</p>
</li>
<li>
<p>Biochemical dissection of protein translocation into the ER</p>
<ul>
<li>Separation of smooth and rough ER by centrifugation</li>
</ul>
</li>
<li>
<p>Schematic of the mechanism of protein translation and translocation into the ER.</p>
<ul>
<li><img src="https://imgur.com/HAUz0uQ.png" alt=""></li>
</ul>
</li>
<li>
<p>Concept of “necessary” and the concept of “sufficient”</p>
<ul>
<li>If the biochemical feature (e.g., a signal peptide at the N-terminus of a protein) is necessary for a process (in this case, translocation into the ER), then deleting or disrupting that biochemical feature will prevent the process from happening.</li>
<li>If no additional biochemical features are necessary for the process, then that feature alone should mediate the process.</li>
<li>Note that the concept of sufficiency is context dependent.</li>
</ul>
</li>
<li>
<p>Some problems to work on at home</p>
<ul>
<li>How would you test if a signal peptide is necessary for protein translocation into the ER?</li>
<li>How would you test if a signal peptide is sufficient for targeting a protein into the ER in living cells?</li>
<li>How would you test if a signal peptide is sufficient for targeting a protein in rough ER microsomes in vitro?</li>
<li>How could you prove that a protein of interest that encodes a signal peptide was actually inside the vesicle?  Remember, light microscopy does not have the resolution to distinguish if a protein is in the vesicle or on the exterior surface of vesicle.</li>
</ul>
</li>
<li>
<p>Protein targeting continued. How are proteins imported into the nucleus?</p>
</li>
</ol>
<h2 id="L16-Protein-targeting-continued-How-are-proteins-imported-into-the-nucleus">L16 Protein targeting continued.  How are proteins imported into the nucleus?</h2>
<ul>
<li>Many nuclear proteins contain a <strong>nuclear localization sequence</strong> (NLS)</li>
<li><strong>Nuclear import factors</strong> <mark>bind</mark> to the NLS.</li>
<li>Import factors can ferry the NLS containing cargo <strong>across the pore</strong>.</li>
<li>The nucleoplasm (<mark>inner</mark>) contains high concentrations of <strong>Ran-GTP</strong></li>
<li><strong>Ran-GTP binds to the import factor</strong> causing a substantial conformational change that releases the NLS from the import factor</li>
</ul>
<p><img src="https://imgur.com/nwUgxkr.png" alt=""></p>
<h3 id="How-it-works-Nuclear-transport-receptors-and-a-gradient-of-Ran-GTP-drive-transport-of-cargo-across-the-pore">How it works: Nuclear transport receptors and a gradient of Ran-GTP drive transport of cargo across the pore.</h3>
<p><img src="https://imgur.com/ls9R2dL.png" alt=""></p>
<h3 id="How-did-we-figure-that-out">How did we figure that out??</h3>
<p><img src="" alt=""></p>
<p>The nucleus is a conspicuous organelle: Considered by most to be the defining feature of eukaryotic cells.</p>
<p>Subcellular fractionation: Start by homogenizing the cells.</p>
<ul>
<li>Homogenate: The contains by using biochemical/physical force to broken cells suspense/tissue.<br>
Centrifuge the homogenates at different speeds</li>
</ul>
<div class="admonition note"><p class="admonition-title">Nucleus and cytoplasm contain different proteins
</p><ul>
<li>Run a subcellular fractionation to separate nuclei from cytoplasm.</li>
<li>Boil each sample in SDS + reducing agent, like DTT or beta-mercaptoethanol.</li>
<li>Run the samples on a gel and stain with a dye. Coomassie blue is the most common.</li>
<li>Coomassie stains all the proteins, so the image on the right reveals population of proteins present in each fraction.</li>
<li>The intensity of a band is proportional to its abundance - true for coomassie.</li>
<li>What are the implications of this experiment? What questions result from the experiment?
<img src="https://imgur.com/3i2AulZ.png" alt="" /></li>
</ul>
</div>
<h3 id="Two-compartments-with-different-compositions">Two compartments with different compositions.</h3>
<ul>
<li>Reasonable to assume one class of proteins that is completely cytoplasmic, another that is completely nuclear, and a third class that is split between the two compartments.</li>
<li>For example, transcription is in the nucleus and translation is in the cytosol.</li>
</ul>
<p><mark>→</mark> So how do proteins enter the nucleus?  What determines if a molecule, like a protein is nuclear or cytoplasmic?</p>
<h3 id="An-experiment-to-examine-the-fate-of-nuclear-proteins-injected-into-the-cytoplasm">An experiment to examine the fate of nuclear proteins injected into the cytoplasm.</h3>
<ul>
<li>Coat Gold Particles with a purified nuclear protein (<strong>nucleoplasmin</strong>)</li>
</ul>
<div class="admonition note"><p class="admonition-title">Frog oocytes experiment
</p><ol>
<li>Inject the (<strong>nucleoplasmin</strong>) coated gold particles into cytoplasm of frog oocytes.</li>
<li>Inject passivated gold particles into cytoplasm  as a control</li>
<li>Fix the cell. Process for electron microscopy.</li>
</ol>
<ul>
<li><strong>Result</strong>: Thinking like a scientist: Consider the previous results from the Feldherr paper</li>
<li>Gold beads <strong>coated</strong> with a nuclear protein injected into the cytoplasm <strong>enter the nucleus</strong>.  What is the control experiment?  Take the control seriously!</li>
<li>State 2 major hypotheses (conclusions) resulting from the Feldherr experiment</li>
<li>Feldherr, et al, J. Cell Biology, 1984
<img src="https://imgur.com/dDsGcAX.png" alt="Results.  Feldherr, et al, J. Cell Biology, 1984" /></li>
</ul>
</div>
<h3 id="Test-for-a-nuclear-localization-sequence">Test for a nuclear localization sequence.</h3>
<ul>
<li>Use molecular biology to fuse GFP to a known nuclear protein</li>
<li>Do the same to deletion constructs.</li>
<li>Transfect into cells.</li>
<li>Score localization using fluorescence microscopy.</li>
<li><mark>Expected results</mark>:
<ul>
<li>GFP label Nuclear Protein: Nuclear would became GFP positive.</li>
<li>GFP label cytosolic protein: Nuclear would became GFO negative.</li>
</ul>
</li>
</ul>
<h3 id="Test-if-the-NLS-is-sufficient-for-nuclear-localization">Test if the NLS is sufficient for nuclear localization.</h3>
<p>Study question: Given these results, describe the process of nuclear import using the words “necessary” and “sufficient”.</p>
<h3 id="Nuclear-localization-sequences">Nuclear localization sequences</h3>
<p>NLS from T-antigen protein of SV40 Virus.</p>
<table>
<thead>
<tr>
<th style="text-align:left"></th>
<th style="text-align:left"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Wild type</td>
<td style="text-align:left">P<mark>KKKRK</mark>V</td>
<td style="text-align:left">Nucleus</td>
</tr>
<tr>
<td style="text-align:left">Mutant</td>
<td style="text-align:left">PT<mark>KKRK</mark>V</td>
<td style="text-align:left">Cytosol</td>
</tr>
<tr>
<td style="text-align:left">Mutant</td>
<td style="text-align:left">P<mark>KKK</mark>T<mark>K</mark>V</td>
<td style="text-align:left">Cytosol</td>
</tr>
</tbody>
</table>
<ul>
<li>Compare this sequence to other known nuclear proteins in database. ~50% of all nuclear proteins have an NLS of this overall structure,</li>
<li>Meaning a cluster of basic amino acids serve as nuclear localization sequences.</li>
<li>There are different types of NLSs such as monopartite, bipartite, but these details do not matter in this class.</li>
</ul>
<h3 id="Summary-of-understanding-of-nuclear-import-ca-1988">Summary of understanding of nuclear import ca. 1988</h3>
<ol>
<li><strong>Nuclear pores</strong>: Nuclear proteins enter the nucleus through nuclear pores.  Same was demonstrated for nuclear export.</li>
<li><strong>Size limitation</strong>: Proteins ~40 kilodaltons or less can cross the nuclear pore.</li>
<li><strong>NLS overcomes size limits</strong>:Proteins larger than that can only cross the pore if they have an NLS.</li>
<li><strong>NLS</strong>: The NLS is <strong>necessary</strong> and sufficient for localization to the nucleus.</li>
</ol>
<div class="admonition question"><p class="admonition-title">But how? Where do we go from here?
</p><p><strong>NLS is necessary and sufficient for nuclear localization of many, many nuclear proteins. What can we do to uncover the molecular mechanism?</strong></p>
<ul>
<li>Start by thinking about <em>how</em> the NLS manages to <em>get proteins into</em> the nucleus.</li>
<li>The NLS containing protein must go through the pore.  We learned that from the Feldherr experiment.</li>
<li>It could be that the NLS <em>somehow opens the pore, or the NLS binds to the pore</em>.  In this case, the NLS really is sufficient for crossing the pore in the absence of any other factors.</li>
<li>As a hypothesis:  An NLS is sufficient for localizing a protein to the nucleus in the absence of any other factors.</li>
<li>Can we test that hypothesis?</li>
<li>The approach = ==biochemical reconstitution==.</li>
</ul>
</div>
<div class="admonition note"><p class="admonition-title">Biochemical reconstitution
</p><ul>
<li>Biochemistry started with fermentation.</li>
<li>Origins of modern biochemistry.</li>
<li>Living organisms can cause chemical transformations.</li>
<li>The microorganisms themselves contain specific chemicals that are responsible for those transformations.</li>
<li>Therefore, like Buchner said, you can study fermentation just like any other series of chemical reactions.</li>
</ul>
</div>
<h3 id="Biochemical-reconstitution">Biochemical reconstitution</h3>
<p>Perform a reaction or process that normally occurs in cells in a cell free system (in vitro).</p>
<ul>
<li>
<p>Reconstitutions of simple reactions led to our understanding of metabolism.</p>
</li>
<li>
<p>Being able to reconstitute a reaction like the one depicted below allows for</p>
</li>
<li>
<p>a more sophisticated analysis of mechanism.</p>
</li>
<li>
<p>Scientists then became emboldened to try to reconstitute more complex cellular process like DNA replication and transcription.</p>
</li>
</ul>
<h3 id="Biochemical-fractionation-Proteins-can-be-separated-using-chromatography">Biochemical fractionation: Proteins can be separated using chromatography</h3>
<p><img src="https://imgur.com/WJu1HU2.png" alt=""></p>
<ul>
<li>3 common types of chromatography.
<ul>
<li>Ion-exchange chromatography</li>
<li>Gel-filtration chromatography</li>
<li>affinity chromatography</li>
</ul>
</li>
<li>Tracking the activity of an unknown factor</li>
<li>Monitoring the purification of an unknown protein using SDS-PAGE.</li>
</ul>
<p>Identifying the protein responsible for a biological process.</p>
<p>Protein composition of some fractions that contain the enzymatic activity over the course of the purification.</p>
<h3 id="How-to-obtain-the-gene-that-encodes-importin-First-step-obtain-amino-acid-sequence-information">How to obtain the gene that encodes importin? First step, obtain amino acid sequence information.</h3>
<ol>
<li>
<p>Identifying peptide sequences in a protein: Mass spectrometry.</p>
<ul>
<li>Old way: Edman degradation: Used up until ca. ~2000</li>
<li>Modern way: Mass spectrum.
<ul>
<li>These spectra serve as fingerprints for peptides.</li>
<li>Only a subset of all possible peptides can yield this m/z ratio.</li>
<li>A collection of such spectra and the possible peptides that can produce such spectra can be compared to all know protein sequences in the database.</li>
<li>Such analysis yields a likely “hit” which is the identity of the protein.</li>
<li>PS: Nowadays, nearly every gene of the standard “model organisms” used in research has already been cloned.  So, once you know the protein’s identity, you can simply order online from a vendor.</li>
</ul>
</li>
</ul>
</li>
<li>
<p>Obtain the gene that encodes a protein of interest.</p>
<ul>
<li>Second  step. Use degenerate primers (some of its positions have several possible bases) and pcr to amplify the gene of interest from a cDNA  library.</li>
</ul>
</li>
</ol>
<ul>
<li>Each bacteria will contain one (1) plasmid containing one cDNA</li>
<li>Corresponding to each mRNA expressed in the starting tissue.</li>
<li>The entire population of bacteria carrying the different cDNAs is a library.</li>
<li>Then, use PCR to amplify the gene of interest from the library.</li>
</ul>
<div class="admonition note"><p class="admonition-title">Identifying a protein using mass spec.
</p><ul>
<li>Purification led to isolation of the doublet shown in lane 6. (this is for a protein involved in actin polymerization)</li>
<li>The bands were cut out of the gel with a scalpel.</li>
<li>The protein extracted from the gel with solvents.</li>
<li>The protein was digested with trypsin, which cuts the protein after every lysine or arginine</li>
<li>The resulting peptides were analyzed by mass spectrometry to generate spectra like those shown in the cartoon in the previous figure.</li>
<li>The spectra can only be generated by certain amino acid sequences peptides.</li>
<li>Only one protein in the data base has all the peptides found in the analysis. The peptides identified by the mass spec approach are highlighted in the amino acid sequence of the full length protein.</li>
<li>Analysis of the data base shows that this protein was identified in the past and named “CRMP-1”.</li>
</ul>
</div>
<h3 id="Identifying-the-gene-that-encodes-a-protein-of-interest-opens-many-paths-towards-studying-function">Identifying the gene that encodes a protein of interest opens many paths towards studying function.</h3>
<ul>
<li>Gene:
<ul>
<li>Tagged versions to image, e.g., GFP tag.</li>
<li>mRNA expression levels and tissue distribution.</li>
<li>Mutagenesis: deletions and point mutations.</li>
<li>Perturbations, e.g., RNAi or CRISPR or homologous recomb.</li>
</ul>
</li>
<li>Easier expression and purification in heterologous system.
<ul>
<li>Structural analysis (X-ray, EM, NMR)</li>
<li>Quantitative analysis: Kinetics, binding affinities.</li>
<li>Affinity approaches to look for new binding partners</li>
<li>Perturbations - drug screening.</li>
</ul>
</li>
</ul>
<div class="admonition question"><p class="admonition-title">So, can we reconstitute nuclear import?
</p><p>If so, we can test our earlier hypothesis -
AND, it would open the process up to a complete biochemical dissection!</p>
</div>
<h3 id="Reconstituting-nuclear-import">Reconstituting nuclear import</h3>
<h4 id="Permeabilize-cells-with-mild-detergent">Permeabilize cells with mild detergent.</h4>
<ul>
<li><mark>Digitonin chelates cholesterol</mark>: Cholesterol is abundant in the plasma membrane but low in the nuclear membrane.</li>
<li><mark>Trition</mark> is non-ionic detergent that permeabilizes all cellular membranes but leaves the nuclear envelope relatively intact.</li>
<li>Need to remove or permeabilize the plasma membrane to gain biochemical access but also make sure that you didn’t permeabilize the nuclear envelope.</li>
<li>Cells were permeabilized with digitonin (left) or digitonin followed by triton x-100 (right) and stained with antibodies against a nuclear envelope protein (top) or DNA (bottom)<br>
<img src="https://imgur.com/qxHbfoH.png" alt=""></li>
</ul>
<h4 id="Activity-requires-soluble-factors-and-energy">Activity requires soluble factors and energy.</h4>
<p>Reconstituting nuclear import: Depicted as a cartoon.</p>
<ol>
<li>Permeabilize cells.</li>
<li>Add test protein to permeabilized cells.  Image in microscope</li>
</ol>
<p><img src="https://imgur.com/YxKEp4w.png" alt=""></p>
<p>Study question:  Given these results, describe the process of nuclear import using terms necessary and sufficient.</p>
<div class="admonition note"><p class="admonition-title">Reconstituting nuclear import: Restore activity with cytosol.
</p><ul>
<li>What is in cytosol that promotes nuclear import: Can use the in vitro assay and chromatography to find out.</li>
<li>Nuclear import activity splits into two fractions that must be recombined to restore activity.</li>
</ul>
</div>
<h4 id="The-two-steps-of-nuclear-import-targeting-to-the-nuclear-envelope-and-translocation-through-the-nuclear-pore-require-different-cytosolic-factors">The two steps of nuclear import, targeting to the nuclear envelope and translocation through the nuclear pore, require different cytosolic factors</h4>
<p><img src="https://imgur.com/wMjlieO.png" alt=""></p>
<ul>
<li>“We have isolated two cytosolic fractions from Xenopus oocytes that contain all of the activity necessary to support both steps of nuclear import in digitonin-permeabilized mammalian cells: binding at the nuclear envelope and translocation through the nuclear pore. The first cytosolic fraction (fraction A) interacts with an import-competent, but not a mutant, nuclear localization sequence-bearing conjugate and stimulates its accumulation at the nuclear envelope in an ATP-independent fashion. The second cytosolic fraction (fraction B) gives no discernible effect when added alone; but when added either together with fraction A, or after fraction A, stimulates the passage of the conjugate from the outer nuclear envelope to the interior of the nucleus in an ATP-dependent fashion.”</li>
<li>Study questions:
<ul>
<li>What control experiment could you perform to convince a skeptic that the assay depicted above in the cartoon is specific to nuclear import?</li>
<li>Do you think nuclear pores contain NLS binding sites?</li>
<li>The experimental results shown above lead to 2 important predictions about the biochemical function of the active factor in fraction A.  What are the 2 predictions?  Notice that once you have correctly identified the two predicted activities of fraction A, you have basically described two hypotheses regarding the function of factor A which is present in fraction A.</li>
</ul>
</li>
</ul>
<h3 id="Translocation-accumulation-of-nuclear-cargo-into-the-nucleus-requires-the-small-GTPase-Ran">Translocation/accumulation of nuclear cargo into the nucleus requires the small GTPase, Ran.</h3>
<ul>
<li>Identify this factor using Edman degradation (then) or mass spec (now).  Clone the gene using degenerate PCR (then) or order from vendor (now).</li>
<li>Clone the corresponding gene into an expression vector. Express the gene in a heterologous system, e.g., bacteria or insect Sf9 cells.  Purify the recombinant protein.  Test it in the assay.</li>
</ul>
<p>Crude fraction A + <strong>fractionated B</strong> + energy → Crude fraction A + <strong>purified factor B</strong> + energy → Crude fraction A + <mark><strong>recombinant Ran</strong></mark> + energy</p>
<ul>
<li>Ran is a small GTPase:
<ul>
<li>GTPases exist in two states</li>
<li>Intrinsic exchange and hydrolysis rates are slow.</li>
<li>GEFs catalyze the exchange reaction</li>
<li>GAPs catalyze hydrolysis</li>
</ul>
</li>
<li>Ran and nuclear import:
<ul>
<li>Ran GAP is restricted to the cytosol</li>
<li>Ran GEF (RCC1) is confined to the nucleus</li>
</ul>
</li>
</ul>
<h3 id="A-summary-of-where-we-are-now">A summary of where we are now.</h3>
<ul>
<li>Nuclear transport goes through nuclear pores.</li>
<li>Proteins that reside in the nucleus often contain a nuclear localization sequence (NLS) that directs them to the nucleus.</li>
<li>Nuclear import also depends on soluble factors present in the cytosol that direct NLS containing proteins to the nuclear pore and drive their translocation/accumulation in the nucleus.</li>
<li>One of these cytosolic factors is a small GTPase known as Ran.</li>
<li>Ran is not necessary for binding of NLS containing proteins to the nuclear pore, but it is necessary for the NLS cargo translocate/accumulate in the nucleus.</li>
<li>Nuclear transport also requires energy in the form of nucleotide triphosphates.</li>
</ul>
<div class="admonition question"><p class="admonition-title">What is the other factor?
</p></div>
<h3 id="Identification-of-importin-as-a-nuclear-import-factor">Identification of importin as a nuclear import factor.</h3>
<p><img src="https://imgur.com/1sCk6f2.png" alt=""></p>
<p>Thinking about experimental results.: Three things that need to be done with new results.</p>
<ol>
<li>Assess the quality of the data.
<ul>
<li>Does the experiment help address the overall question?</li>
<li>Is the experiment of high quality? Does it have the proper controls?  Are the data interpretable?</li>
</ul>
</li>
<li>Interpret the data. What can you conclude from the experiment?</li>
<li>Act upon the new results.  What needs to be done next?
<ul>
<li>In some cases, your results immediately lead to another experiment that needs to be performed to fully address your hypothesis or before you can draw a definitive conclusion.</li>
<li>In other cases, your results might mark an endpoint for summarizing before moving on to the next question.</li>
</ul>
</li>
</ol>
<h3 id="“Importin”-is-actually-a-heterodimer">“Importin” is actually a heterodimer</h3>
<p>Our cells actually express many other nuclear import factors as well as nuclear export factors that are tasked for moving different cargo across nuclear pores.<br>
Importin alpha/beta was the first discovered.  It is responsible for most of the transport involving proteins that contain an NLS.</p>
<ul>
<li><strong>Importin-alpha</strong> is an adaptor protein that binds To a subset of NLSs, but it alone cannot enter nuclear pores.  However, it can form a complex with importin-beta</li>
<li><strong>Importin-beta</strong> is the actual “nuclear transporter” or “NTR” that can enter the nuclear pore.</li>
</ul>
<p>So, mechanism.  How do nuclear transport receptors (NTRs) like importin promote nuclear import?</p>
<ul>
<li>What is the role of nucleotide hydrolysis?</li>
<li>In vitro reactions allows you to separate reactions.
<ul>
<li>Step 1.  NLS Cargo + importin-a and importin b</li>
<li>Step 2.  Add Ran + Energy (e.g., ATP and GTP)</li>
<li>Therefore, entry into the nuclear pore is independent of Ran and energy (GTP)<br>
However, entering the nuclear interior depends on Ran and GTP.<br>
How does that work???</li>
</ul>
</li>
<li>You can think of nuclear import as rectified diffusion.
<ul>
<li>This complex can diffuse across the nuclear pore</li>
<li>But once inside the nucleus, Ran-GTP binds to importin and releases the cargo</li>
<li>Consequently, the cargo is “stuck” in the nucleus because it cannot diffuse across the pore.</li>
<li>This complex forms in the cytosol where there is no Ran-GTP, only Ran GDP</li>
</ul>
</li>
<li>Which brings us back to the overall model.</li>
</ul>
<h2 id="L17-Membrane-Trafficking-Part-1">L17 Membrane Trafficking Part 1</h2>
<h3 id="Fundamental-reactions-necessary-for-membrane-trafficking">Fundamental reactions necessary for membrane trafficking</h3>
<ul>
<li>Membrane budding and vesicle formation</li>
<li>Cargo selection</li>
<li>Membrane fusion to the target organelle</li>
<li>Show Figure 13-2 Alberts</li>
</ul>
<p>Trafficking routes<br>
Vesicles are enclosed by protein coats</p>
<p><strong>Vesicle formation</strong>: Clathrin coated vesicles</p>
<ul>
<li>Phospholipids help guide coat protein assembly to the correct membrane</li>
<li>Different adaptor molecules selectively recognize different phospho-head groups to help specify location of vesicle formation.</li>
<li>Dynamin drives the final scission event of endocytosis
<ul>
<li>membrane binding and fusing protein.</li>
</ul>
</li>
<li>Other coat proteins assemble through different pathways
<ul>
<li>AP2 was recruit by the signal liek PI(4,5)P<sub>2</sub></li>
</ul>
</li>
<li>COPII coats as an example (ER -&gt; Golgi traffic)
<ul>
<li>COPII-coated vesicles promote anterograde ER-to-Golgi transport of newly synthesized secretory proteins and proteins destined for organelles other than the ER (McCaughey and Stephens, 2018, Miller and Schekman, 2013). COPI vesicles mediate the retrograde transport of escaped ER residents and the recycling of anterograde transmembrane cargo receptors from the ER-Golgi intermediate compartment (ERGIC) and cis-Golgi cisternae toward the ER, as well as in intra-Golgi transport</li>
</ul>
</li>
<li>SNARE proteins drive selective membrane fusion
<ul>
<li>accessory proteins + NSF + ATP promoted the recycle of the SNARE proteins.</li>
</ul>
</li>
</ul>
<h3 id="How-are-vesicles-made">How are vesicles made?</h3>
<p>How are vesicles built?: Roth and Porter experiment - maybe proteins do it.</p>
<ul>
<li>
<p>Images of vesicles in chicken oocytes suggest an overall mechanism for their formation: n.b., these are EM images from fixed cells arranged in a sequence to illustrate a point.  They are not an actual time-lapse series.</p>
</li>
<li>
<p>Exercise #1. What happened to my protein prep and what does it mean?</p>
<ul>
<li>Goal was to purify tubulin from brain cytosol</li>
<li>Microtubules in vitro assembled from purified brain tubulin.</li>
<li>Tubuliin heterodimer of ~ 55 kd on SDS-PAGE</li>
</ul>
</li>
<li>
<p>In the end: Discovery of clathrin</p>
<ul>
<li>Purified from brain, which is expected to have many vesicles.</li>
</ul>
</li>
<li>
<p>Q1: Is this thing a microtubule?  Explain your answer.</p>
</li>
<li>
<p>Q2: Motivate the science? Explain to fellow scientists what you think you have discovered and defend your conclusion with evidence</p>
<ol>
<li>Jamieson and Palade show movement of proteins through the secretory pathway</li>
<li>Roth and Porter show protein density associated with endocytosis</li>
<li>Pearse purifies this protein</li>
</ol>
</li>
<li>
<p>What is the possible connection between these 3 discoveries above?</p>
</li>
<li>
<p>Motivate the science<br>
Earlier work by Palade and colleagues showed that proteins move between membranous organelles, presumably in xxx intermediates, but the underlying yyy mechanisms are not known.  Subsequent work by Roth and Porter showed that xxx is apparently driven by yyy where xxxs are forming.  However, …(fill that in) is not known.  Using differential centrifugation, I have identified a ??? that self assembles into !!!.  (The following characteristics lead me to hypothesize) that…</p>
</li>
</ul>
<div class="admonition question"><p class="admonition-title">What’s next?
</p><ul>
<li>You have discovered a protein that forms cage like structures in vitro.  You named this protein “clathrin”.</li>
<li>You hypothesize that clathrin is important for vesicle formation.</li>
<li>Stated another way, you hypothesize that transport vesicles are surrounded by a protein coat (clathrin) that is important for forming and maintaining vesicles.</li>
<li>If your hypothesis is true, then what does it predict?  Come up with 2-3 predictions (or important questions) as to what needs to be done next.</li>
</ul>
</div>
<ul>
<li>
<p>Cages can be seen in cells.</p>
</li>
<li>
<p>Clathrin cages are built from ‘triskelions’.</p>
<ul>
<li>Clathrin assembly behaves like a condensation polymerization reaction with a critical concentration.</li>
</ul>
</li>
<li>
<p>Self assembly.</p>
</li>
<li>
<p>Structure of a clathrin cage.</p>
<ul>
<li>Multiple methods required to obtain this structure.</li>
<li>Question for further thought:<br>
The full length clathrin protein has not been crystalized.<br>
Why do you think clathrin is so hard to crystalize?</li>
<li>Hemoglobin shown for size comparison.</li>
</ul>
</li>
<li>
<p>Self assembly of viruses.</p>
</li>
<li>
<p>Energetics of self assembly.</p>
</li>
<li>
<p>Viruses tend to be built from compact proteins: Small conformational changes are necessary to accommodate virus geometry.</p>
</li>
<li>
<p>Form follows function.: Louis Sullivan - Architect (e.g., Auditorium Building, Roosevelt University; Holy Trinity Russian Orthodox Cathedral, etc)</p>
<ul>
<li>Depiction of a clathrin triskelion</li>
<li>Depiction of a “generic” ~ 40-50 kilodalton protein</li>
</ul>
</li>
<li>
<p>Viruses: Simple cargo of predetermined, fixed size.</p>
</li>
<li>
<p>Clathrin: Variety of payloads that vary size across 3 orders of magnitude.</p>
</li>
<li>
<p>Clathrin can assemble into cages of differing size by simply adding more hexagons and pentagons.  Conformational flexibility permits adaptation to different sized payloads.</p>
</li>
</ul>
<div class="admonition note"><p class="admonition-title">Summary
</p><ul>
<li>Proteins enter the secretory pathway at the ER via cotranslational translocation in the ER (MCB 150 and see background reading).</li>
<li>Proteins are distributed throughout the pathway by moving in vesicular intermediates.</li>
<li>Building a vesicle from a planar membrane requires energy.</li>
<li>Electron microscopy implies protein coats on forming vesicles.</li>
<li>Clathrin is the most studied coat protein. It self assembles into cages offering a starting point to think about how it promotes vesicle assembly.</li>
</ul>
</div>
<h3 id="How-does-clathrin-promote-vesicle-formation-Does-it-bind-to-membranes">How does clathrin promote vesicle formation?  Does it bind to membranes?</h3>
<p>Clathrin binds membranes through an adaptor<br>
Identification of adaptors (AP1, 2, 3)<br>
Adaptors (AP2) bind <strong>lipids</strong> and recruit clathrin.</p>
<ul>
<li>AP-2 is one of ~20 adaptor molecules used to recruit clathrin to different membranes in different cells under different conditions.</li>
<li>Adaptor complex AP-2 and phospholipid head group in pink. Dotted lines are unstructured.</li>
</ul>
<h3 id="Is-clathrin-necessary-for-vesicle-formation">Is clathrin necessary for vesicle formation?</h3>
<p>Our thoughts: What do we see as the key questions going forwards from the discovery of clathrin?</p>
<p>How might clathrin promote vesicle formation?</p>
<ul>
<li>Clathrin is not an enzyme, but it spontaneously assembles into cages.</li>
</ul>
<p>How might self-assembly drive an unfavorable process?</p>
<ul>
<li>Role of ATP hydrolysis in clathrin cage dynamics<br>
Clathrin coated vesicle formation<br>
Perhaps start with the model figure and then ask students what ATP is for.<br>
Then, ask them to back out of this how clathrin might work.</li>
</ul>
<p>Identifying the location of a protein within the secretory pathway</p>
<h2 id="L19-Membrane-Trafficking-Cargo-selection-with-a-focus-on-endocytosis-and-trafficking-to-lysosomes">L19 Membrane Trafficking - Cargo selection with a focus on endocytosis and trafficking to lysosomes</h2>
<h3 id="Mechanism-of-cholesterol-ester-uptake">Mechanism of cholesterol ester uptake</h3>
<ol>
<li>Purify LDL Particles and label them could use radioactivity like I125 or a fluorophore.</li>
<li>Add increasing concentrations to fibroblasts from unaffected versus affected individuals at 4oC</li>
<li>Wash away excess and plot amount of signal associated with the cells.</li>
</ol>
<p><img src="https://imgur.com/p4aOs8L.png" alt=""></p>
<div class="admonition question"><p class="admonition-title">What is the defect in the affected individual?
</p><ul>
<li><strong>Is LDL binding to cells mediated by a specific receptor</strong>? Explain.
Yes. The binding curve for the unaffected individual saturates. Therefore, it must be mediated by a specific receptor. All available binding sites for LDL on the receptor are eventually occupies by LDL at high LDL concentrations.</li>
<li><strong>What is the defect in the affected individual?</strong>
The affected individual has mutations in each allele of the LDL-receptor. Both mutations affect the ability of the receptor to bind LDL.</li>
</ul>
</div>
<h3 id="Fate-of-LDL-and-cholesterol-esters-after-binding">Fate of LDL and cholesterol esters after binding</h3>
<ol>
<li>Incubate cells at 4°C with labeled LDL containing radioactively labeled cholesterol esters.</li>
<li>Shift to 37°C at times thereafter.</li>
<li>Acid wash the cells and then spin to separate intact cells from acid stripped material.  Note that acid washing does not lyse the cells.</li>
<li>Quantify amount of LDL protein in supernatant and cell pellet.</li>
</ol>
<p><img src="https://imgur.com/AAZcCgG.png" alt=""></p>
<div class="admonition question"><p class="admonition-title">Questions
</p><ul>
<li>Why does LDL association with cells become acid resistant after incubating at 37°C?
Low pH does not lyse the cells. More LDL is pelleting with the cells shortly after warming to 37. Therefore, it must have been internalized.</li>
<li>Why is LDL protein disappearing over time at 37°C?
Once inside the cell, it is eventually degraded.</li>
</ul>
</div>
<p><strong>Another case of LDL uptake in a patient with hypercholesterolemia.</strong></p>
<ul>
<li>Perform the same experiment as before measuring the fate of LDL after warming the cells to 37°C.</li>
<li>Compare cells from an unaffected individual to those of an affected individual.</li>
</ul>
<p><img src="https://imgur.com/6THdjfF.png" alt=""></p>
<p>!!! quest Questions<br>
- <strong>What is NOT wrong with J.D.’s cells?</strong><br>
J.D.’s cells obviously express an LDL receptor that is capable of binding to LDL. Note that normal cells bind ~11 ng LDL/dish of cells. JD’s bind 9 ng/dish, so the binding is basically the same<br>
- <strong>What IS wrong with J.D.’s cells?</strong><br>
Cannot internalize the LDL as indicated by the triangles in the graph.<br>
- <strong>Why might J.D. have extraordinarily high circulating LDL levels?</strong><br>
Because JD’s cells can not internalize the LDL particles containing cholesterol esters. The LDL particles are not cleared efficiently from the blood stream.</p>
<h3 id="Imaging-LDL-Particle-dynamics">Imaging LDL Particle dynamics</h3>
<p><img src="https://imgur.com/NwiOu8P.png" alt=""></p>
<p>Affected individuals could take the LDL and aggrigate it under the cell membranes. But it cann’t take in insie.</p>
<p>Electron Micrographs Showing Representative Stages in the Endocytosis of LDL-Ferritin and Its Subsequent Appearance in the. Lysosome Normal fibroblasts were incubated with 47.5 pg/ml of LDL-ferritin Lysosome Normal fibroblasts were incubated with 47.5 pg/ml of LDL-ferritin for 2 hr at 4”C, washed extensively, and then warmed at 37°C for various times as described in Experimental Procedures. Bar = 1000 A.<br>
(A) A typical coated pit (time at 37”C, 1 min). Magnification 97,000x.<br>
(B) A coated pit being transformed into an endocytic vesicle with LDL-ferritin included (time at 37”C, 1 min). Magnification 81,000x.<br>
© Formation of a coated vesicle. As the plasma membrane begins to fuse to form the vesicle, some of the LDL-ferritin is excluded from the<br>
interior and is left on the surface of the cell (arrow) (time at 37X, 1 min). Magnification 54,500x.<br>
(D) A fully formed coated vesicle that appears to be losing its cytoplasmic coat on one side (arrow)</p>
<h3 id="Summary-of-LDL-cholesterol-ester-uptake">Summary of LDL/cholesterol ester uptake.</h3>
<p>!!! quest Questions<br>
What experiments were done and what results were seen to support the following claims?</p>
<ul>
<li>
<p>“When a cell needs cholesterol it makes membrane receptors for LDL and inserts them into the plasma membrane.” <strong>Via what mechanism do cells insert LDL Receptors into the membrane?</strong></p>
<ul>
<li>The LDL receptor is on the plasma membrane. Therefore, it was originally synthesized in the endoplasmic reticulum and trafficked to the plasma membrane. (The identity of the LDL-receptor is known and it is a single-pass transmembrane protein.)</li>
</ul>
</li>
<li>
<p>“Once in the plasma membrane, the receptors diffuse in clathrin pits that are in the process of forming.  There, an endocytosis signal in the cytoplasmic tail of LDL receptors binds the membrane adaptor AP-2 after its conformation has been locally unlocked by binding to PI(4,5)P2 in the plasma membrane.” <strong>How would you test if PI(4,5)P2 unlocks AP-2 to bind the cytoplasmic tail of AP-2?</strong></p>
<ul>
<li>Notice here that the statement discusses much biochemical detail. Therefore, there must exist biochemical data supporting the claim. You could perform binding assays to test 1) if PI(4,5)P2 binds AP-2 and 2) If AP-2 binding to the cytosolic tail of LDL-R is PI(4,5)P2 dependent. See results on next slide.</li>
</ul>
</li>
</ul>
<h3 id="Molecular-basis-of-familial-hypercholesterolemia">Molecular basis of familial hypercholesterolemia</h3>
<p>!!! Q Question<br>
Which of the following statements are likely, possible explanations for the disease?</p>
<ul>
<li>Mutations in the LDL Receptor ectodomain compromise its ability to bind to LDL.</li>
<li>Mutations in the LDL Receptor cytoplasmic domain compromise its ability to bind to AP-2.</li>
<li>Mutations in AP-2 prevent it from associating with clathrin and limiting efficient formation of clathrin coated vesicles thus inhibiting LDL uptake.</li>
<li>Mutations in a phosphatidyl kinase necessary to make PI(4,5)P2 limit the  amount of LDL uptake because AP-2 binding to LDL-Receptor is limited.</li>
</ul>
<h3 id="Another-example-of-cargo-selection-Lysosomal-enzymes-and-the-Mannose-6-Phosphate-story">Another example of cargo selection.  Lysosomal enzymes and the Mannose-6-Phosphate story.</h3>
<p>Pathways that lead to lysosomal degradation</p>
<ul>
<li>Lysosomes are major sites for intracellular digestion
<ul>
<li>True or false.  Lysosomal enzymes are synthesized in the lysosome.</li>
</ul>
</li>
</ul>
<h3 id="How-are-lysosomal-hydrolyses-targeted-to-the-lysosome">How are lysosomal hydrolyses targeted to the lysosome?</h3>
<p>Mannose-6-Phosphate Receptor sorts hydrolases from the TGN to the lysosome.</p>
<p>Protein glycosylation in the secretory pathway</p>
<ul>
<li>The molecular weight of mature, proteolytically processed N-cadherin (an adhesion molecule expressed on the cell surface) should be ~ 80 kilodaltons but it runs at ~110 kilodaltons on SDS-PAGE.  Why might that be?</li>
<li>(A side note - not about glycosylation). You are studying a spike protein of an enveloped virus. The protein runs at 60 kd on SDS-PAGE when you boil it in the presence of beta-mercaptoethanol but it runs at 180 kd if you do not add the mercaptan. Why might that be?</li>
</ul>
<h3 id="Mannose-6P-is-added-exclusively-to-lysosomal-proteins-in-the-cis-Golgi">Mannose 6P is added exclusively to lysosomal proteins in the cis Golgi</h3>
<ul>
<li>Do you think this reaction is happening on the cytosolic face or lumenal face of the cis Golgi?</li>
<li>You purified GlcNAc phosphotranferase for enriched Golgi stacks.  Sequence analysis of the N-terminus of the proteins shows that it differs from the N-terminus predicted from the cDNA sequence.  Why might the protein in the Golgi be missing the first 20 amino acids?  Is it using an alternative start codon?</li>
</ul>
<p>!!! Q Question<br>
What experiments were done and what results were seen to support the following claims regarding protein targeting to lysosomes?</p>
<ul>
<li>“The M6P receptor binds M6P at pH 6.5-6.7 in the TGN lumen and releases it at pH 6, which is the pH in the lumen of endosomes.”</li>
<li>How do you think they known that M6P binding to M6P-R is pH dependent?</li>
</ul>
<h3 id="Protein-targeting-to-lysosomes">Protein targeting to lysosomes</h3>
<p>Fluorescein fluorescence is largely independent of pH 440 nm but is highly pH dependent at 490 nm.</p>
<p>!!! Q Question<br>
How could use this information to determine the pH of the TGN versus the pH endosomes and lysosomes?</p>
<h3 id="Lumenal-pH-of-intracellular-organelles">Lumenal pH of intracellular organelles</h3>
<ul>
<li>The pH of cytosol is usually around pH 7.4.  Lumenal pH of endosomes is 6 and that of lysosomes is 4.</li>
<li>How do you think cells manage to acidify the lumens of particular organelles?</li>
</ul>
<h3 id="Summary-endocytic-pathway">Summary endocytic pathway</h3>
<p>Internalized cargo meets hydrolytic enzymes in late endosomes</p>
<p>What prevents hydrolytic enzymes destined for late endosomes from degrading the TGN?<br>
What is a zymogen?</p>
<h3 id="Endocytosis-and-desensitization-to-extracellular-signaling">Endocytosis and desensitization to extracellular signaling</h3>
<p>Receptor Downregulation</p>
<ul>
<li>Notice the advantage of quantitative experiments!</li>
<li>Does the cell express fewer receptors on the surface after the initial response?  How would you know?</li>
<li>Activation of a surface receptor often leads to its down regulation by internalization.</li>
</ul>
<p>!!! Q Question<br>
Is internalization of a high affinity signaling molecule enough to turn off the signal?</p>
<ul>
<li>High affinity ligand:receptor interactions can have very slow off rates making it difficult to turn off the signal.</li>
<li>Internalizing the receptor :ligand complex might not be sufficient to turn off the signal.</li>
</ul>
<h3 id="Sequestering-internalized-plasma-membrane-proteins-from-the-cytosol">Sequestering internalized plasma membrane proteins from the cytosol</h3>
<p>The problem of creating a vesicle within a vesicle?</p>
<ul>
<li>Does it take energy to bend a membrane?</li>
<li>How do cells control vesicle formation?</li>
<li>Do you expect cells use the same machinery to generate an intralumenal vesicle as they use to generate a transport vesicle?</li>
</ul>
<h2 id="L20">L20</h2>
<p>some update from the L19</p>
<h3 id="Binding-assays-to-test-if-AP-2-binding-to-LDL-R-cut-tail-is-PI-4-5-P2-dependent">Binding assays to test if AP-2 binding to LDL-R cut-tail is PI(4,5)P2 dependent</h3>
<ul>
<li>
<p>This result would show that PI(4,5)P2 binds specifically to AP-2 and it would give you the dissociation constant.  To perform the experiment, you some assay to distinguish binding from free.  There are many ways to do this.  A couple of options are presented in the next slide.</p>
</li>
<li>
<p>You could repeat the experiment with a different PIP2.  In this example, PI(3,4)P2.  The result shown would indicate that AP-2 strongly prefers to bind (4,5) but not (3,4).</p>
</li>
<li>
<p>Now you know that AP-2 binds to (4,5)P2 but not to (3,4)P2, and you also know the concentration at which (4,5)P2 binding to AP-2 saturates.  Next, measure the binding of the LDL-cytoplasmic tail (which is where the internalization signal in LDL resides) to a fixed amount of AP-2 in the presence of a saturating concentration of (4,5)P2 versus an equimolar concentration of (3,4)P2.  The result shows that LDL-R only binds AP-2 in the presence of the right phosphoinositide.  Hence the statement “an endocytosis signal in the cytoplasmic tail of LDL receptors binds the membrane adaptor AP-2 after its conformation has been locally unlocked by binding to PI(4,5)P2 in the plasma membrane.”</p>
</li>
<li>
<p>Note, you could have also compared binding of AP-2 to LDL-R cytotail in presence of (4,5)P2 versus no added phosphoinositide.  That too would work and suffice.  It’s just more exciting to see it compared to another phosphoinositide which reveals even greater specificity.<br>
From here, you could do a number of experiments to better understand the mechanism and physiological significance of this biochemistry.  You could, for example, perform a deletion analysis to better define the region in LDL-R that binds to AP-2.  You could test the physiological significance of this biochemistry by replacing the WT LDL-R with a mutant version that does not bind to AP-2 and measure the consequences for LDL-R uptake.</p>
</li>
</ul>
<h3 id="Asn-linked-glycosylation-in-ER">Asn-linked glycosylation in ER</h3>
<ul>
<li>This reaction takes place almost as soon as the polypeptide chain enters the ER.</li>
<li>Using bioinformatics, you find that the frequency of Asn-X-Ser/Thr sequences in glycosylated proteins is much lower than it is in non-glycosylated regions.  Therefore, did evolution select for or against the glycosylation motif?  Provide reasonable speculation as to why.</li>
</ul>
<h3 id="Summary-Delivering-lysosomal-hydrolases-to-endosomes">Summary - Delivering lysosomal hydrolases to endosomes</h3>
<p>The M6P-R is a pH dependent receptor that recognizes a unique modification (M6P) that is added exclusively to lysosomal hydrolases.<br>
Imagine the early days of this field.  All you know is that lysosomal hydrolases have the unique modification of M6P.  What would you want to know?  In other words, how would you dissect the problem of lytic enzyme delivery to lysosomes?<br>
What prevents hydrolytic enzymes destined for endosomes from degrading the TGN?<br>
What is a zymogen?</p>
<h3 id="Monoubiquitination-drives-clathrin-mediated-endocytosis-of-EGFR">Monoubiquitination drives clathrin-mediated endocytosis of  EGFR</h3>
<p>Sustained ubiquitination drives the EGFR into MVBs.</p>
</div><div class="article-licensing box"><div class="licensing-title"><p>Bio Cell</p><p><a href="https://karobben.github.io/2024/01/17/LearnNotes/BioCell/">https://karobben.github.io/2024/01/17/LearnNotes/BioCell/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2024-01-17</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2023-11-06</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2024/01/17/LearnNotes/BioChem/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">BioChm</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2023/10/31/LearnNotes/immun/"><span class="level-item">Understanding Antibodies and Phage Display: A Deep Dive</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">500</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">146</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">169</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Bilibili" href="https://space.bilibili.com/393056819"><i class="fas fa-video"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#L15-Protein-targeting-to-membranous-organelles"><span class="level-left"><span class="level-item">1</span><span class="level-item">L15: Protein targeting to membranous organelles</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Ways-of-organelles-separation"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">Ways of organelles separation</span></span></a></li><li><a class="level is-mobile" href="#How-are-proteins-directed-to-the-ER"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">How are proteins directed to the ER?</span></span></a></li></ul></li><li><a class="level is-mobile" href="#L16-Protein-targeting-continued-How-are-proteins-imported-into-the-nucleus"><span class="level-left"><span class="level-item">2</span><span class="level-item">L16 Protein targeting continued.  How are proteins imported into the nucleus?</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#How-it-works-Nuclear-transport-receptors-and-a-gradient-of-Ran-GTP-drive-transport-of-cargo-across-the-pore"><span class="level-left"><span class="level-item">2.1</span><span class="level-item">How it works: Nuclear transport receptors and a gradient of Ran-GTP drive transport of cargo across the pore.</span></span></a></li><li><a class="level is-mobile" href="#How-did-we-figure-that-out"><span class="level-left"><span class="level-item">2.2</span><span class="level-item">How did we figure that out??</span></span></a></li><li><a class="level is-mobile" href="#Two-compartments-with-different-compositions"><span class="level-left"><span class="level-item">2.3</span><span class="level-item">Two compartments with different compositions.</span></span></a></li><li><a class="level is-mobile" href="#An-experiment-to-examine-the-fate-of-nuclear-proteins-injected-into-the-cytoplasm"><span class="level-left"><span class="level-item">2.4</span><span class="level-item">An experiment to examine the fate of nuclear proteins injected into the cytoplasm.</span></span></a></li><li><a class="level is-mobile" href="#Test-for-a-nuclear-localization-sequence"><span class="level-left"><span class="level-item">2.5</span><span class="level-item">Test for a nuclear localization sequence.</span></span></a></li><li><a class="level is-mobile" href="#Test-if-the-NLS-is-sufficient-for-nuclear-localization"><span class="level-left"><span class="level-item">2.6</span><span class="level-item">Test if the NLS is sufficient for nuclear localization.</span></span></a></li><li><a class="level is-mobile" href="#Nuclear-localization-sequences"><span class="level-left"><span class="level-item">2.7</span><span class="level-item">Nuclear localization sequences</span></span></a></li><li><a class="level is-mobile" href="#Summary-of-understanding-of-nuclear-import-ca-1988"><span class="level-left"><span class="level-item">2.8</span><span class="level-item">Summary of understanding of nuclear import ca. 1988</span></span></a></li><li><a class="level is-mobile" href="#Biochemical-reconstitution"><span class="level-left"><span class="level-item">2.9</span><span class="level-item">Biochemical reconstitution</span></span></a></li><li><a class="level is-mobile" href="#Biochemical-fractionation-Proteins-can-be-separated-using-chromatography"><span class="level-left"><span class="level-item">2.10</span><span class="level-item">Biochemical fractionation: Proteins can be separated using chromatography</span></span></a></li><li><a class="level is-mobile" href="#How-to-obtain-the-gene-that-encodes-importin-First-step-obtain-amino-acid-sequence-information"><span class="level-left"><span class="level-item">2.11</span><span class="level-item">How to obtain the gene that encodes importin? First step, obtain amino acid sequence information.</span></span></a></li><li><a class="level is-mobile" href="#Identifying-the-gene-that-encodes-a-protein-of-interest-opens-many-paths-towards-studying-function"><span class="level-left"><span class="level-item">2.12</span><span class="level-item">Identifying the gene that encodes a protein of interest opens many paths towards studying function.</span></span></a></li><li><a class="level is-mobile" href="#Reconstituting-nuclear-import"><span class="level-left"><span class="level-item">2.13</span><span class="level-item">Reconstituting nuclear import</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Permeabilize-cells-with-mild-detergent"><span class="level-left"><span class="level-item">2.13.1</span><span class="level-item">Permeabilize cells with mild detergent.</span></span></a></li><li><a class="level is-mobile" href="#Activity-requires-soluble-factors-and-energy"><span class="level-left"><span class="level-item">2.13.2</span><span class="level-item">Activity requires soluble factors and energy.</span></span></a></li><li><a class="level is-mobile" href="#The-two-steps-of-nuclear-import-targeting-to-the-nuclear-envelope-and-translocation-through-the-nuclear-pore-require-different-cytosolic-factors"><span class="level-left"><span class="level-item">2.13.3</span><span class="level-item">The two steps of nuclear import, targeting to the nuclear envelope and translocation through the nuclear pore, require different cytosolic factors</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Translocation-accumulation-of-nuclear-cargo-into-the-nucleus-requires-the-small-GTPase-Ran"><span class="level-left"><span class="level-item">2.14</span><span class="level-item">Translocation/accumulation of nuclear cargo into the nucleus requires the small GTPase, Ran.</span></span></a></li><li><a class="level is-mobile" href="#A-summary-of-where-we-are-now"><span class="level-left"><span class="level-item">2.15</span><span class="level-item">A summary of where we are now.</span></span></a></li><li><a class="level is-mobile" href="#Identification-of-importin-as-a-nuclear-import-factor"><span class="level-left"><span class="level-item">2.16</span><span class="level-item">Identification of importin as a nuclear import factor.</span></span></a></li><li><a class="level is-mobile" href="#“Importin”-is-actually-a-heterodimer"><span class="level-left"><span class="level-item">2.17</span><span class="level-item">“Importin” is actually a heterodimer</span></span></a></li></ul></li><li><a class="level is-mobile" href="#L17-Membrane-Trafficking-Part-1"><span class="level-left"><span class="level-item">3</span><span class="level-item">L17 Membrane Trafficking Part 1</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Fundamental-reactions-necessary-for-membrane-trafficking"><span class="level-left"><span class="level-item">3.1</span><span class="level-item">Fundamental reactions necessary for membrane trafficking</span></span></a></li><li><a class="level is-mobile" href="#How-are-vesicles-made"><span class="level-left"><span class="level-item">3.2</span><span class="level-item">How are vesicles made?</span></span></a></li><li><a class="level is-mobile" href="#How-does-clathrin-promote-vesicle-formation-Does-it-bind-to-membranes"><span class="level-left"><span class="level-item">3.3</span><span class="level-item">How does clathrin promote vesicle formation?  Does it bind to membranes?</span></span></a></li><li><a class="level is-mobile" href="#Is-clathrin-necessary-for-vesicle-formation"><span class="level-left"><span class="level-item">3.4</span><span class="level-item">Is clathrin necessary for vesicle formation?</span></span></a></li></ul></li><li><a class="level is-mobile" href="#L19-Membrane-Trafficking-Cargo-selection-with-a-focus-on-endocytosis-and-trafficking-to-lysosomes"><span class="level-left"><span class="level-item">4</span><span class="level-item">L19 Membrane Trafficking - Cargo selection with a focus on endocytosis and trafficking to lysosomes</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Mechanism-of-cholesterol-ester-uptake"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">Mechanism of cholesterol ester uptake</span></span></a></li><li><a class="level is-mobile" href="#Fate-of-LDL-and-cholesterol-esters-after-binding"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">Fate of LDL and cholesterol esters after binding</span></span></a></li><li><a class="level is-mobile" href="#Imaging-LDL-Particle-dynamics"><span class="level-left"><span class="level-item">4.3</span><span class="level-item">Imaging LDL Particle dynamics</span></span></a></li><li><a class="level is-mobile" href="#Summary-of-LDL-cholesterol-ester-uptake"><span class="level-left"><span class="level-item">4.4</span><span class="level-item">Summary of LDL/cholesterol ester uptake.</span></span></a></li><li><a class="level is-mobile" href="#Molecular-basis-of-familial-hypercholesterolemia"><span class="level-left"><span class="level-item">4.5</span><span class="level-item">Molecular basis of familial hypercholesterolemia</span></span></a></li><li><a class="level is-mobile" href="#Another-example-of-cargo-selection-Lysosomal-enzymes-and-the-Mannose-6-Phosphate-story"><span class="level-left"><span class="level-item">4.6</span><span class="level-item">Another example of cargo selection.  Lysosomal enzymes and the Mannose-6-Phosphate story.</span></span></a></li><li><a class="level is-mobile" href="#How-are-lysosomal-hydrolyses-targeted-to-the-lysosome"><span class="level-left"><span class="level-item">4.7</span><span class="level-item">How are lysosomal hydrolyses targeted to the lysosome?</span></span></a></li><li><a class="level is-mobile" href="#Mannose-6P-is-added-exclusively-to-lysosomal-proteins-in-the-cis-Golgi"><span class="level-left"><span class="level-item">4.8</span><span class="level-item">Mannose 6P is added exclusively to lysosomal proteins in the cis Golgi</span></span></a></li><li><a class="level is-mobile" href="#Protein-targeting-to-lysosomes"><span class="level-left"><span class="level-item">4.9</span><span class="level-item">Protein targeting to lysosomes</span></span></a></li><li><a class="level is-mobile" href="#Lumenal-pH-of-intracellular-organelles"><span class="level-left"><span class="level-item">4.10</span><span class="level-item">Lumenal pH of intracellular organelles</span></span></a></li><li><a class="level is-mobile" href="#Summary-endocytic-pathway"><span class="level-left"><span class="level-item">4.11</span><span class="level-item">Summary endocytic pathway</span></span></a></li><li><a class="level is-mobile" href="#Endocytosis-and-desensitization-to-extracellular-signaling"><span class="level-left"><span class="level-item">4.12</span><span class="level-item">Endocytosis and desensitization to extracellular signaling</span></span></a></li><li><a class="level is-mobile" href="#Sequestering-internalized-plasma-membrane-proteins-from-the-cytosol"><span class="level-left"><span class="level-item">4.13</span><span class="level-item">Sequestering internalized plasma membrane proteins from the cytosol</span></span></a></li></ul></li><li><a class="level is-mobile" href="#L20"><span class="level-left"><span class="level-item">5</span><span class="level-item">L20</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Binding-assays-to-test-if-AP-2-binding-to-LDL-R-cut-tail-is-PI-4-5-P2-dependent"><span class="level-left"><span class="level-item">5.1</span><span class="level-item">Binding assays to test if AP-2 binding to LDL-R cut-tail is PI(4,5)P2 dependent</span></span></a></li><li><a class="level is-mobile" href="#Asn-linked-glycosylation-in-ER"><span class="level-left"><span class="level-item">5.2</span><span class="level-item">Asn-linked glycosylation in ER</span></span></a></li><li><a class="level is-mobile" href="#Summary-Delivering-lysosomal-hydrolases-to-endosomes"><span class="level-left"><span class="level-item">5.3</span><span class="level-item">Summary - Delivering lysosomal hydrolases to endosomes</span></span></a></li><li><a class="level is-mobile" href="#Monoubiquitination-drives-clathrin-mediated-endocytosis-of-EGFR"><span class="level-left"><span class="level-item">5.4</span><span class="level-item">Monoubiquitination drives clathrin-mediated endocytosis of  EGFR</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/AI/"><span class="tag">AI</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Beginner/"><span class="tag">Beginner</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">69</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">38</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Desktop/"><span class="tag">Desktop</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitHub/"><span class="tag">GitHub</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immune/"><span class="tag">Immune</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG-pathway/"><span class="tag">KEGG pathway</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning-Data-Preprocessing/"><span class="tag">Machine Learning. Data Preprocessing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matplotlib/"><span class="tag">Matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind/"><span class="tag">Mind</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Multimedia/"><span class="tag">Multimedia</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">25</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Odds-Ratio/"><span class="tag">Odds Ratio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">62</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">132</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Quantum-Physics/"><span class="tag">Quantum Physics</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">86</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA/"><span class="tag">RNA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">21</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-seq/"><span class="tag">RNA-seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Seaborn/"><span class="tag">Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">47</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI-lib/"><span class="tag">TUI lib</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Ted/"><span class="tag">Ted</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Uniprot/"><span class="tag">Uniprot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/awk/"><span class="tag">awk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/busuanzi/"><span class="tag">busuanzi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dash/"><span class="tag">dash</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dme/"><span class="tag">dme</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/fasta/"><span class="tag">fasta</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flies/"><span class="tag">flies</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot2/"><span class="tag">ggplot2</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/grep/"><span class="tag">grep</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/live2d/"><span class="tag">live2d</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/math/"><span class="tag">math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/matplotlib/"><span class="tag">matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/wet/"><span class="tag">wet</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://space.bilibili.com/393056819" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">Bilibili</span></span><span class="level-right"><span class="level-item tag">space.bilibili.com</span></span></a></li><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://camo.githubusercontent.com/fd83d8c9f8b8dbe85686812e2e186dceec728bbabad2d331aa44e0bfeee8628c/68747470733a2f2f70706f66666963652e6769746875622e696f2f6865786f2d7468656d652d6963617275732f67616c6c6572792f707265766965772e706e673f31" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">67</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">61</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">8</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">22</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/FastQC/"><span class="level-start"><span class="level-item">FastQC</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fastq-QC/"><span class="level-start"><span class="level-item">Fastq QC</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">136</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">11</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">75</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">8</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">11</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Ted/"><span class="level-start"><span class="level-item">Ted</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/flies/"><span class="level-start"><span class="level-item">flies</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immune/"><span class="level-start"><span class="level-item">Immune</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">56</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">39</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2023 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>