<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>Biopython | Karobben</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="Biopython">
<meta property="og:type" content="website">
<meta property="og:title" content="Biopython">
<meta property="og:url" content="https://karobben.github.io/Python/Biopython.html">
<meta property="og:site_name" content="Karobben">
<meta property="og:description" content="Biopython">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-08-30T09:02:49.277Z">
<meta property="article:modified_time" content="2020-08-30T09:02:49.277Z">
<meta property="article:author" content="Karobben">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml">
  
  
    <link rel="icon" href="img/tag.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

  
<link rel="stylesheet" href="/css/search.css">

  <script type="text/javascript"
   src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>

  <style>
  pre {
    color: salmon;
  }
  </style>
<meta name="generator" content="Hexo 4.2.1"></head>

<script>
  var _hmt = _hmt || [];
  (function() {
    var hm = document.createElement("script");
    hm.src = "https://hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hm, s);
  })();
</script>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header"; style="background:#1B414B;height:150px">
  <!--div id="banner"></div>-->
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Karobben</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">Karobben Blog</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
          <a id="main-nav-toggle" class="nav-icon"></a>
          <!-- Header Section Start -->
  <header id="slider-area">
    <nav class="navbar navbar-expand-md fixed-top scrolling-navbar bg-white">
      <div class="container">
        <div class="collapse navbar-collapse" id="navbarCollapse">
          <ul class="navbar-nav mr-auto w-100 justify-content-end">
              <a class="main-nav-link" href="../index.html">Home</a>
              <a class="main-nav-link" href="../R/R-index.html">R</a>
              <a class="main-nav-link" href="../Python/index.html">Python</a>
              <a class="main-nav-link" href="../Linux/index.html">Linux</a>
              <a class="main-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
              <a class="main-nav-link" href="../LearnNotes/index.html">Notes</a>
              <a class="main-nav-link" href="../Blog/index.html">Others</a>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <!-- Header Section End -->

      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Looking for Something?"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://karobben.github.io"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><div align='right', style='padding-right:160px;padding-top:1px'>
  
  <span class="local-search local-search-google local-search-plugin" style=";position:absolute;z-index:2;">
      <input type="search" placeholder="Looking for Something?" id="local-search-input" class="local-search-input-cls" style="">
      <div id="local-search-result" class="local-search-result-cls"></div>
  </span>


</div>

<article  id="page-" class="article article-type-page" itemscope itemprop="blogPost">
<div id="tipue_search_content" style="display: none"></div>
  <div class="article-meta">
    <a href="/Python/Biopython.html" class="article-date">
  <time datetime="2020-08-30T09:02:49.277Z" itemprop="datePublished">2020-08-30</time>
</a>
    
  </div>
  <div  class="article-inner">
    
    
      <header class="article-header">
        <!--
  
    <h1 class="article-title" itemprop="name">
      Biopython
    </h1>
  
 -->
      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Go To Top Link -->
        <div title="Back to Top" class="back-to-top" style="background:salmon">
  <a href="#"  >
    <i >Top</i>
  </a>
</div>
<div class="title_index">
  <ul id="nav">
   <li>
     Titles
     <ul>
       <!-- Table of Contents -->

  <div id="toc" class="toc-article">
    <strong class="toc-title">Titles</strong>
    <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#"><span class="toc-number">1.</span> <span class="toc-text">Biopython</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#"><span class="toc-number">2.</span> <span class="toc-text">1. Quick Start</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#"><span class="toc-number">3.</span> <span class="toc-text">2. Sequences</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#2-1-reading-FASTA-file-br"><span class="toc-number">3.1.</span> <span class="toc-text">2.1 reading FASTA file
</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-2-Seq-run-as-string"><span class="toc-number">3.2.</span> <span class="toc-text">2.2 Seq run as string</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-4-Slicing-a-sequence"><span class="toc-number">3.3.</span> <span class="toc-text">2.4 Slicing a sequence</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-5-revers-string"><span class="toc-number">3.4.</span> <span class="toc-text">2.5 revers string</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-6-Changing-Font-case"><span class="toc-number">3.5.</span> <span class="toc-text">2.6 Changing Font case</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#"><span class="toc-number">4.</span> <span class="toc-text">3 Bio-information</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#3-1-Revers-Complement"><span class="toc-number">4.1.</span> <span class="toc-text">3.1 Revers Complement</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-2-Translation"><span class="toc-number">4.2.</span> <span class="toc-text">3.2 Translation</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-3-Translation-Tables"><span class="toc-number">4.3.</span> <span class="toc-text">3.3 Translation Tables</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#"><span class="toc-number">5.</span> <span class="toc-text">4. Alignment</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-Echo-a-test-file"><span class="toc-number">5.1.</span> <span class="toc-text">1. Echo a test file</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-Align"><span class="toc-number">5.2.</span> <span class="toc-text">2. Align</span></a></li></ol></li></ol>
  </div>

<!-- Table of Contents Down-->

     </ul>
    </li>
   </ul>
</div>


        <h1>Biopython</h1>
<p><a name="C9IvS"></a></p>
<h1>1. Quick Start</h1>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio.Seq <span class="keyword">import</span> reverse_complement, transcribe, back_transcribe, translate</span><br><span class="line"></span><br><span class="line"><span class="comment"># echo a seq</span></span><br><span class="line">my_string = <span class="string">"GCTGTTATGGGTCGTTGGAAGGGTGGTCGTGCTGCTGGTTAG"</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#Revers complement</span></span><br><span class="line">reverse_complement(my_string)</span><br><span class="line"></span><br><span class="line"><span class="comment"># &#123; 'CTAACCAGCAGCACGACCACCCTTCCAACGACCCATAACAGC' &#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#DNA to RNA</span></span><br><span class="line">transcribe(my_string)</span><br><span class="line"></span><br><span class="line"><span class="comment">#--&gt; &#123; 'GCUGUUAUGGGUCGUUGGAAGGGUGGUCGUGCUGCUGGUUAG' &#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#RNA to DNA</span></span><br><span class="line">back_transcribe(my_string) --&gt; &#123; 'GCTGTTATGGGTCGTTGGAAGGGTGGTCGTGCTGCTGGTTAG' &#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">#DNA to protine</span></span><br><span class="line">translate(my_string) --&gt; &#123; 'AVMGRWKGGRAAG*' &#125;</span><br></pre></td></tr></table></figure>
<p><a name="mNuzg"></a></p>
<h1>2. Sequences</h1>
<p><a name="6mPRo"></a></p>
<h2 id="2-1-reading-FASTA-file-br"><a class="header-anchor" href="#2-1-reading-FASTA-file-br"></a>2.1 reading FASTA file<br /></h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio <span class="keyword">import</span> SeqIO</span><br><span class="line">Seq1=<span class="string">'GSE44995_Reference_assembled_isotig_seq.fna'</span></span><br><span class="line"><span class="keyword">for</span> seq_record <span class="keyword">in</span> SeqIO.parse(Seq1, <span class="string">"fasta"</span>)</span><br><span class="line">    print(seq_record.id)</span><br><span class="line">    print(repr(seq_record.seq))</span><br><span class="line">    print(len(seq_record.seq))</span><br><span class="line"></span><br><span class="line"><span class="comment"># print by fasta formate</span></span><br><span class="line">Seq10=<span class="string">''</span></span><br><span class="line"><span class="keyword">for</span> seq_record <span class="keyword">in</span> SeqIO.parse(Seq1, <span class="string">"fasta"</span>):</span><br><span class="line">  <span class="keyword">if</span> len(seq_record.seq) &lt; <span class="number">100</span>:</span><br><span class="line">    Seq10 += <span class="string">"&gt;"</span>+str(seq_record.id)+<span class="string">"\n"</span></span><br><span class="line">    Seq10 += str(seq_record.seq)+<span class="string">"\n"</span></span><br><span class="line">    print(seq_record.id)</span><br><span class="line">    print(repr(seq_record.seq))</span><br><span class="line">    print(len(seq_record.seq))</span><br></pre></td></tr></table></figure></div>
<p><a name="ZwldO"></a></p>
<h2 id="2-2-Seq-run-as-string"><a class="header-anchor" href="#2-2-Seq-run-as-string"></a>2.2 Seq run as string</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio.Seq <span class="keyword">import</span> Seq</span><br><span class="line"><span class="keyword">from</span> Bio.Alphabet <span class="keyword">import</span> IUPAC</span><br><span class="line">my_seq = Seq(<span class="string">"GATCG"</span>, IUPAC.unambiguous_dna)</span><br><span class="line"><span class="keyword">for</span> index, letter <span class="keyword">in</span> enumerate(my_seq):</span><br><span class="line">  print(<span class="string">"%i %s"</span> % (index, letter))</span><br><span class="line"></span><br><span class="line"><span class="string">'''</span></span><br><span class="line"><span class="string">print result</span></span><br><span class="line"><span class="string">0 G</span></span><br><span class="line"><span class="string">1 A</span></span><br><span class="line"><span class="string">2 T</span></span><br><span class="line"><span class="string">3 C</span></span><br><span class="line"><span class="string">4 G</span></span><br><span class="line"><span class="string">'''</span></span><br><span class="line"></span><br><span class="line">print(my_seq[<span class="number">0</span>]) <span class="comment">#/ print(my_seq[-1])</span></span><br><span class="line">Seq(<span class="string">"AAAA"</span>).count(<span class="string">"AA"</span>)</span><br><span class="line">len(my_seq)</span><br><span class="line"></span><br><span class="line"><span class="comment"># convert to str</span></span><br><span class="line">my_seq = str(my_seq)</span><br></pre></td></tr></table></figure></div>
<br />
<a name="qfmb9"></a>
## 2.3 GC caculate
<br />
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio.Seq <span class="keyword">import</span> Seq</span><br><span class="line"><span class="keyword">from</span> Bio.Alphabet <span class="keyword">import</span> IUPAC</span><br><span class="line"><span class="keyword">from</span> Bio.SeqUtils <span class="keyword">import</span> GC</span><br><span class="line">my_seq = Seq(<span class="string">"GATCGATGGGCCTATATAGGATCGAAAATCGC"</span>, IUPAC.unambiguous_dna)</span><br><span class="line">GC(my_seq)</span><br><span class="line"></span><br><span class="line"><span class="comment">#Result</span></span><br><span class="line"><span class="comment">#46.875</span></span><br></pre></td></tr></table></figure></div>
<p><a name="lvo99"></a></p>
<h2 id="2-4-Slicing-a-sequence"><a class="header-anchor" href="#2-4-Slicing-a-sequence"></a>2.4 Slicing a sequence</h2>
<br />
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">my_seq[<span class="number">4</span>:<span class="number">12</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment">####</span></span><br><span class="line">the first, second <span class="keyword">and</span> third codon positions of this DNA sequence:</span><br><span class="line"><span class="comment">####</span></span><br><span class="line">my_seq[<span class="number">0</span>::<span class="number">3</span>]  <span class="comment"># Seq('GCTGTAGTAAG', IUPACUnambiguousDNA())</span></span><br><span class="line">my_seq[<span class="number">1</span>::<span class="number">3</span>]  <span class="comment"># Seq('AGGCATGCATC', IUPACUnambiguousDNA())</span></span><br><span class="line">my_seq[<span class="number">2</span>::<span class="number">3</span>]  <span class="comment"># Seq('TAGCTAAGAC', IUPACUnambiguousDNA())</span></span><br></pre></td></tr></table></figure></div>
<p><a name="WruAE"></a></p>
<h2 id="2-5-revers-string"><a class="header-anchor" href="#2-5-revers-string"></a>2.5 revers string</h2>
<br />
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">my_seq = my_seq[::<span class="number">-1</span>]</span><br></pre></td></tr></table></figure></div>
<p><a name="ijsZi"></a></p>
<h2 id="2-6-Changing-Font-case"><a class="header-anchor" href="#2-6-Changing-Font-case"></a>2.6 Changing Font case</h2>
<br />
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio.Seq <span class="keyword">import</span> Seq</span><br><span class="line"><span class="keyword">from</span> Bio.Alphabet <span class="keyword">import</span> generic_dna</span><br><span class="line">dna_seq = Seq(<span class="string">"acgtACGT"</span>, generic_dna)</span><br><span class="line">dna_seq.upper() --&gt; &#123;Seq('ACGTACGT', DNAAlphabet())&#125;</span><br><span class="line">dna_seq.lower() --&gt; &#123;Seq('acgtacgt', DNAAlphabet())&#125;</span><br></pre></td></tr></table></figure>
<p><a name="RP0Ys"></a></p>
<h1>3 Bio-information</h1>
<p><a name="tqSMw"></a></p>
<h2 id="3-1-Revers-Complement"><a class="header-anchor" href="#3-1-Revers-Complement"></a>3.1 Revers Complement</h2>
<br />
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">my_seq.reverse_complement()</span><br><span class="line"></span><br><span class="line">Seq(<span class="string">"ACGTCGTAGCTAC"</span>).complement() <span class="comment"># standard example =&gt; output: Seq('TGCAGCATCGATG')</span></span><br><span class="line">Seq(<span class="string">"ACGTCGTAGCTAC"</span>).reverse_complement() <span class="comment"># output =&gt; Seq('GTAGCTACGACGT')</span></span><br></pre></td></tr></table></figure></div>
<p><a name="o4Bl4"></a></p>
<h2 id="3-2-Translation"><a class="header-anchor" href="#3-2-Translation"></a>3.2 Translation</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio.Seq <span class="keyword">import</span> Seq</span><br><span class="line"><span class="keyword">from</span> Bio.Alphabet <span class="keyword">import</span> IUPAC</span><br><span class="line"></span><br><span class="line">Seq(<span class="string">"UUU"</span>, IUPAC.unambiguous_rna).translate() <span class="comment"># for RNA</span></span><br><span class="line">Seq(<span class="string">"TTT"</span>, IUPAC.unambiguous_dna).translate() <span class="comment"># for DNA</span></span><br></pre></td></tr></table></figure></div>
<p><a name="28baJ"></a></p>
<h2 id="3-3-Translation-Tables"><a class="header-anchor" href="#3-3-Translation-Tables"></a>3.3 Translation Tables</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio.Data <span class="keyword">import</span> CodonTable</span><br><span class="line"></span><br><span class="line">standard_table = CodonTable.unambiguous_dna_by_name[<span class="string">'Standard'</span>]</span><br><span class="line">mito_table = CodonTable.unambiguous_dna_by_name[<span class="string">'Vertebrate Mitochondrial'</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment">#same as:</span></span><br><span class="line"></span><br><span class="line">standard_table = CodonTable.unambiguous_dna_by_id[<span class="number">1</span>]</span><br><span class="line">mito_table = CodonTable.unambiguous_dna_by_id[<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line">mito_table.stop_codons  <span class="comment">#--&gt; &#123; ['TAA', 'TAG', 'AGA', 'AGG'] &#125;</span></span><br><span class="line">mito_table.start_codons <span class="comment"># --&gt; &#123; ['ATT', 'ATC', 'ATA', 'ATG', 'GTG'] &#125;</span></span><br><span class="line">mito_table.forward_table[<span class="string">"ACG"</span>] <span class="comment">#--&gt; &#123; 'T' &#125;</span></span><br><span class="line"></span><br><span class="line">print(standard_table)</span><br></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PLAIN"><figure class="iseeu highlight /plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">  |  T      |  C      |  A      |  G      |</span><br><span class="line">--+---------+---------+---------+---------+--</span><br><span class="line">T | TTT F   | TCT S   | TAT Y   | TGT C   | T</span><br><span class="line">T | TTC F   | TCC S   | TAC Y   | TGC C   | C</span><br><span class="line">T | TTA L   | TCA S   | TAA Stop| TGA Stop| A</span><br><span class="line">T | TTG L(s)| TCG S   | TAG Stop| TGG W   | G</span><br><span class="line">--+---------+---------+---------+---------+--</span><br><span class="line">C | CTT L   | CCT P   | CAT H   | CGT R   | T</span><br><span class="line">C | CTC L   | CCC P   | CAC H   | CGC R   | C</span><br><span class="line">C | CTA L   | CCA P   | CAA Q   | CGA R   | A</span><br><span class="line">C | CTG L(s)| CCG P   | CAG Q   | CGG R   | G</span><br><span class="line">--+---------+---------+---------+---------+--</span><br><span class="line">A | ATT I   | ACT T   | AAT N   | AGT S   | T</span><br><span class="line">A | ATC I   | ACC T   | AAC N   | AGC S   | C</span><br><span class="line">A | ATA I   | ACA T   | AAA K   | AGA R   | A</span><br><span class="line">A | ATG M(s)| ACG T   | AAG K   | AGG R   | G</span><br><span class="line">--+---------+---------+---------+---------+--</span><br><span class="line">G | GTT V   | GCT A   | GAT D   | GGT G   | T</span><br><span class="line">G | GTC V   | GCC A   | GAC D   | GGC G   | C</span><br><span class="line">G | GTA V   | GCA A   | GAA E   | GGA G   | A</span><br><span class="line">G | GTG V   | GCG A   | GAG E   | GGG G   | G</span><br><span class="line">--+---------+---------+---------+---------+--</span><br></pre></td></tr></table></figure></div>
<h1>4. Alignment</h1>
<h2 id="1-Echo-a-test-file"><a class="header-anchor" href="#1-Echo-a-test-file"></a>1. Echo a test file</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="BASH"><figure class="iseeu highlight /bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># run in bash</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">"&gt;TRINITY_DN106095_c2_g1_i2</span></span><br><span class="line"><span class="string">MSRIMKVFLFLAVMVCISEAQLHAQCLCPRVRSRISSMTDIREVQIYEATIFCDRMEIVVTNDSGLRYCLNPKLKAVQKLLTAMKPKTSTTARPTVHSSSTGSTNTARM</span></span><br><span class="line"><span class="string">&gt;TRINITY_DN92154_c0_g1_i1</span></span><br><span class="line"><span class="string">DIHVRRRTLTRSKTLGRSTNVNKMKLCILLMLGTLLVLVYGMPPISRDYNTHCRCLQVESRIIPPNSLKSIKLVPEGPHCPDMEVIAGLSNGEKVCLNPRSSWVKKLVNFVLEKQQGGALPKNQGQ"</span> &gt; test.fa</span><br></pre></td></tr></table></figure></div>
<h2 id="2-Align"><a class="header-anchor" href="#2-Align"></a>2. Align</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># run in python</span></span><br><span class="line"><span class="keyword">from</span> Bio <span class="keyword">import</span> pairwise2</span><br><span class="line"><span class="keyword">from</span> Bio.Seq <span class="keyword">import</span> Seq</span><br><span class="line"><span class="keyword">from</span> Bio.pairwise2 <span class="keyword">import</span> format_alignment</span><br><span class="line"><span class="keyword">from</span> Bio.SubsMat <span class="keyword">import</span> MatrixInfo</span><br><span class="line"></span><br><span class="line">seq1 = Seq(<span class="string">"MSRIMKVFLFLAVMVCISEAQLHAQCLCPRVRSRISSMTDIREVQIYEATIFCDRMEIVVTNDSGLRYCLNPKLKAVQKLLTAMKPKTSTTARPTVHSSSTGSTNTARM"</span>)</span><br><span class="line">seq2 = Seq(<span class="string">"DIHVRRRTLTRSKTLGRSTNVNKMKLCILLMLGTLLVLVYGMPPISRDYNTHCRCLQVESRIIPPNSLKSIKLVPEGPHCPDMEVIAGLSNGEKVCLNPRSSWVKKLVNFVLEKQQGGALPKNQGQ"</span>)</span><br><span class="line">alignments = pairwise2.align.globalxx(seq1, seq2)</span><br><span class="line">test_alignments = pairwise2.align.localds(seq1, seq2, MatrixInfo.blosum62, <span class="number">-10</span>, <span class="number">-1</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> alignment <span class="keyword">in</span> alignments:</span><br><span class="line">  print(format_alignment(*alignment))</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> alignment <span class="keyword">in</span> test_alignments:</span><br><span class="line">  print(format_alignment(*alignment))</span><br></pre></td></tr></table></figure></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PLAIN"><figure class="iseeu highlight /plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">SingleLetterAlphabet() alignment with 6 rows and 65 columns</span><br><span class="line">MQNTPAERLPAIIEKAKSKHDINVWLLDRQGRDLLEQRVPAKVA...EGP B7RZ31_9GAMM&#x2F;59-123</span><br><span class="line">AKQRGIAGLEEWLHRLDHSEAIPIFLIDEAGKDLLEREVPADIT...KKP A0A0C3NPG9_9PROT&#x2F;58-119</span><br><span class="line">ARRHGQEYFQQWLERQPKKVKEQVFAVDQFGRELLGRPLPEDMA...KKP A0A143HL37_9GAMM&#x2F;57-121</span><br><span class="line">TRRHGPESFRFWLERQPVEARDRIYAIDRSGAEILDRPIPRGMA...NKP A0A0X3UC67_9GAMM&#x2F;57-121</span><br><span class="line">AINRNTQQLTQDLRAMPNWSLRFVYIVDRNNQDLLKRPLPPGIM...NRK B3PFT7_CELJU&#x2F;62-126</span><br><span class="line">AVNATEREFTERIRTLPHWARRNVFVLDSQGFEIFDRELPSPVA...NRT K4KEM7_SIMAS&#x2F;61-125</span><br></pre></td></tr></table></figure></div>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://karobben.github.io/Python/Biopython.html" data-id="ckcogixm600512bfc6nms1l48" class="article-share-link">Share</a>
      
      
    </footer>
  </div>
  
    
  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    

  
    
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/06/">June 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2020/06/30/tmp/">test</a>
          </li>
        
      </ul>
    </div>
  </div>

  
  <!-- Weather -->
<div id="weather-view-he" ></div>
<script>
WIDGET = {ID: 'HfP68rcshb'};
</script>
<script type="text/javascript" src="https://apip.weatherdt.com/view/static/js/r.js?v=1111"></script>

</aside>

          <!-- Weather Plug -->
        
        <!--git talk-->
        <section id="main">
          <div style='padding-top:10px'>
  <div style="background:white">
    <script src="https://utteranc.es/client.js"
            repo="Karobben/utterances"
            issue-term="pathname"
            theme="preferred-color-scheme"
            crossorigin="anonymous"
            async>
    </script>
  </div>
</div>
<!--git talk-->
  <!-- Link Gitalk 的支持文件  -->
  <!--
    <div>
      <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
      <script src="https://unpkg.com/gitalk@latest/dist/gitalk.min.js"></script>
      <div id="gitalk-container"></div>     <script type="text/javascript">
          var gitalk = new Gitalk({

          // gitalk的主要参数
              clientID: '96c208ade65e80e3aaac',
              clientSecret: '86e9240674a3815b0ddc31dababdb4998d5562f4',
              repo: 'OAuth-Application',
              owner: 'Karobben',
              admin: ['Karobben'],
              id:window.location.pathname,
          });
          gitalk.render('gitalk-container');
      </script>
    </div>
<!-- Gitalk end -->

        </section>
        <!-- Gitalk end -->

      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      <span style="float:right; list-style:none">
        <a style="padding: 10px" href="https://karobben.github.io/">Blog</a>
        <a style="padding: 10px" href="https://github.com/Karobben" target="_blank" rel="noopener">GitHub</a>
        <br>
        <a style="padding: 10px" href="https://space.bilibili.com/393056819" target="_blank" rel="noopener">BiliBili</a>
        <a href="mailto:a591465908@outlook.com">Mail me</a>
      </span>
      &copy; Karobben 2020<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
  <span>
    <p align = "center">
      <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
      <span id="busuanzi_container_site_pv">~<span id="busuanzi_value_site_pv"></span>~</span>
    </p>
  </span>
</footer>

    </div>
    <nav id="mobile-nav">
  <!--`
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  -->
  <a class="mobile-nav-link" href="../index.html">Home</a>
  <a class="mobile-nav-link" href="../R/R-index.html">R</a>
  <a class="mobile-nav-link" href="../Python/index.html">Python</a>
  <a class="mobile-nav-link" href="../Linux/index.html">Linux</a>
  <a class="mobile-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
  <a class="mobile-nav-link" href="../LearnNotes/index.html">Notes</a>
  <a class="mobile-nav-link" href="../Blog/index.html">Others</a>
</nav>

    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




<!--mermaid module-->

  <script src='https://unpkg.com/mermaid@7.1.2/dist/mermaid.min.js'></script>
  <script>
    if (window.mermaid) {
      mermaid.initialize({theme: 'forest'});
    }
  </script>



<!-- Searche Module-->
<script src="/js/search.js"></script>
<script src="/js/load.js"></script>
<!-- 这里可能有一些其他原来文件里的代码，也可以没有 -->

<!-- mathjax config similar to math.stackexchange -->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

  </div>
<script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body>
</html>
