<!doctype html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><title>BioStatistics with R/Python: Block 1 - Karobben</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Karobben"><meta name="msapplication-TileImage" content="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Karobben"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="Normal Distribution; CI calculate; t-table; p-value; T test; Independence and rejection of data; Propagation of errors"><meta property="og:type" content="blog"><meta property="og:title" content="BioStatistics with R/Python: Block 1"><meta property="og:url" content="https://karobben.github.io/2022/01/24/LearnNotes/tulane-biostat-1/"><meta property="og:site_name" content="Karobben"><meta property="og:description" content="Normal Distribution; CI calculate; t-table; p-value; T test; Independence and rejection of data; Propagation of errors"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://spauldinggrp.com/wp-content/uploads/2015/09/statistics-word-cloud.jpg"><meta property="article:published_time" content="2022-01-24T14:23:38.000Z"><meta property="article:modified_time" content="2023-06-06T21:42:32.138Z"><meta property="article:author" content="Karobben"><meta property="article:tag" content="Classes"><meta property="article:tag" content="Tulane Classes"><meta property="article:tag" content="BioStatistics"><meta property="twitter:card" content="summary"><meta property="twitter:image" content="https://spauldinggrp.com/wp-content/uploads/2015/09/statistics-word-cloud.jpg"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://karobben.github.io/2022/01/24/LearnNotes/tulane-biostat-1/"},"headline":"Karobben","image":["https://spauldinggrp.com/wp-content/uploads/2015/09/statistics-word-cloud.jpg"],"datePublished":"2022-01-24T14:23:38.000Z","dateModified":"2023-06-06T21:42:32.138Z","author":{"@type":"Person","name":"Karobben"},"description":"Normal Distribution; CI calculate; t-table; p-value; T test; Independence and rejection of data; Propagation of errors"}</script><link rel="canonical" href="https://karobben.github.io/2022/01/24/LearnNotes/tulane-biostat-1/"><link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml"><link rel="icon" href="https://s3.ax1x.com/2021/02/11/yBZjYQ.png"><link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.12.0/css/all.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><!--!--><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css"><script src="https://www.googletagmanager.com/gtag/js?id=G-0DHWCVQ6K0" async></script><script>window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
    
        gtag('config', 'G-0DHWCVQ6K0');</script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script><!--!--><!--!--><meta name="generator" content="Hexo 5.4.0"></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">Home</a><a class="navbar-item" href="/archives">Archives</a><a class="navbar-item" href="/categories">Categories</a><a class="navbar-item" href="/tags">Tags</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="Catalogue" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="Search" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><div class="card-image"><span class="image is-7by3"><img class="fill" src="https://spauldinggrp.com/wp-content/uploads/2015/09/statistics-word-cloud.jpg" alt="BioStatistics with R/Python: Block 1"></span><span class="ccopy" style="position: absolute; bottom: 0; right: 0; padding-right:3px; padding-left:4px; background-color: rgba(255, 255, 255, 0.6)"><a target="_blank" rel="noopener" href="https://spauldinggrp.com/using-the-stats-worksheet-to-calculate-tracking-error/statistics-word-cloud/">© John D. Simpson, CIPM</a></span></div><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><i class="far fa-eye"></i><span> </span><span id="busuanzi_container_page_pv" style="display: inline;"><span id="busuanzi_value_page_pv">0</span> </span><span class="level-item">Posted&nbsp;<time dateTime="2022-01-24T14:23:38.000Z" title="1/24/2022, 8:23:38 AM">2022-01-24</time></span><span class="level-item">Updated&nbsp;<time dateTime="2023-06-06T21:42:32.138Z" title="6/6/2023, 4:42:32 PM">2023-06-06</time></span><span class="level-item"><a class="link-muted" href="/categories/Notes/">Notes</a><span> / </span><a class="link-muted" href="/categories/Notes/Class/">Class</a><span> / </span><a class="link-muted" href="/categories/Notes/Class/Tulane/">Tulane</a><span> / </span><a class="link-muted" href="/categories/Notes/Class/Tulane/BioStatistics/">BioStatistics</a></span><span class="level-item">27 minutes read (About 4072 words)</span></div></div><h1 class="title is-3 is-size-4-mobile">BioStatistics with R/Python: Block 1</h1><div class="content"><h2 id="Biomedical-Statistics-and-Data-Analysis">Biomedical Statistics and Data Analysis</h2>
<style>
pre {
  background-color:#38393d;
  color: #5fd381;
}
</style>
<h3 id="Normal-Distribution">Normal Distribution</h3>
<p>(PS: If you don’t have any other information, or have too few data points to test the assumption, it is acceptable to assume a Gaussian distribution.)</p>
<p>$$<br>
Pr(X = x|\mu, σ^ 2) = p ( x | \mu , σ^ 2) = \frac{A}{\sigma \sqrt{ 2 \pi}} e^{- \frac{1}{2}(\frac{x-\mu}{\sigma})^ 2}<br>
$$</p>
<p>A = Area under curve<br>
μ = mean<br>
σ = standard deviation</p>
<p><a target="_blank" rel="noopener" href="https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.normaltest.html">Normal distribution codes for python</a></p>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://r-coder.com/wp-content/uploads/2020/10/dnorm-pnorm-qnorm-functions.png" alt="asd.png"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://r-coder.com/normal-distribution-r/">© R CODER</a></td>
</tr>
</tbody>
</table>
<p>In R, there are few functions for calculate normal distribution related values.<br>
Let’s say, the μ = 0, σ = 1, (as the plot above)<br>
than, we have:<br>
the probability of x is <code>dnorm</code>,<br>
the Z is <code>pnorm</code>. The size of blue area<br>
<code>qnorm</code>:  inverse cdf function (Confidence Intervals, CI), the reversed value of the <code>pnorm</code></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">dnorm(<span class="hljs-number">0</span>,   mean=<span class="hljs-number">0</span>, sd=<span class="hljs-number">1</span>)<br>pnorm(<span class="hljs-number">0</span>,   mean=<span class="hljs-number">0</span>, sd=<span class="hljs-number">1</span>)<br>qnorm(<span class="hljs-number">0.5</span>, mean=<span class="hljs-number">0</span>, sd=<span class="hljs-number">1</span>)<br></code></pre></td></tr></table></figure></div>
<pre>
[1] 0.5
[1] 0.3989423
[1] 0
</pre>
<h2 id="CI-calculate-®">CI calculate ®</h2>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">CI &lt;- <span class="hljs-keyword">function</span>(A, alpha)&#123;<br>  alpha = <span class="hljs-number">1</span>- alpha<br>  sample.mean &lt;- mean(A)<br>  sample.n &lt;- <span class="hljs-built_in">length</span>(A)<br>  sample.sd &lt;- sd(A)<br>  sample.se &lt;- sample.sd/<span class="hljs-built_in">sqrt</span>(sample.n)<br>  print(sample.se)<br>  degrees.freedom = sample.n - <span class="hljs-number">1</span><br>  t.score = qt(p=alpha/<span class="hljs-number">2</span>, df=degrees.freedom,lower.tail=<span class="hljs-built_in">F</span>)<br>  print(t.score)<br>  margin.error &lt;- t.score * sample.se<br>  lower.bound &lt;- sample.mean - margin.error<br>  upper.bound &lt;- sample.mean + margin.error<br>  print(<span class="hljs-built_in">c</span>(lower.bound,upper.bound))<br>&#125;<br><br>A &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">12</span>, <span class="hljs-number">3</span>, <span class="hljs-number">4</span>, <span class="hljs-number">2</span>, <span class="hljs-number">42</span>, <span class="hljs-number">4</span>, <span class="hljs-number">42</span>, <span class="hljs-number">524</span>, <span class="hljs-number">5</span>)<br><br>CI(A, <span class="hljs-number">0.90</span>)<br>CI(A, <span class="hljs-number">0.95</span>)<br>CI(A, <span class="hljs-number">0.99</span>)<br></code></pre></td></tr></table></figure></div>
<p><code>rnorm()</code> function for data simulation</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-keyword">from</span> scipy <span class="hljs-keyword">import</span> stats<br><span class="hljs-keyword">import</span> numpy <span class="hljs-keyword">as</span> np<br><br>rng = np.random.default_rng()<br>pts = <span class="hljs-number">1000</span><br>a = rng.normal(<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, size=pts)<br>b = rng.normal(<span class="hljs-number">2</span>, <span class="hljs-number">1</span>, size=pts)<br>x = np.concatenate((a, b))<br>k2, p = stats.normaltest(x)<br>alpha = <span class="hljs-number">1e-3</span><br>print(<span class="hljs-string">&quot;p = &#123;:g&#125;&quot;</span>.<span class="hljs-built_in">format</span>(p))<br>p = <span class="hljs-number">8.4713e-19</span><br><span class="hljs-keyword">if</span> p &lt; alpha:  <span class="hljs-comment"># null hypothesis: x comes from a normal distribution</span><br>    print(<span class="hljs-string">&quot;The null hypothesis can be rejected&quot;</span>)<br><span class="hljs-keyword">else</span>:<br>    print(<span class="hljs-string">&quot;The null hypothesis cannot be rejected&quot;</span>)<br>The null hypothesis can be rejected<br></code></pre></td></tr></table></figure></div>
<p><a target="_blank" rel="noopener" href="https://machinelearningmastery.com/a-gentle-introduction-to-normality-tests-in-python/">codes for python, 2</a></p>
<h3 id="Shapiro-Wilk-Test">Shapiro-Wilk Test</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-comment"># Shapiro-Wilk Test</span><br><span class="hljs-keyword">from</span> numpy.random <span class="hljs-keyword">import</span> seed<br><span class="hljs-keyword">from</span> numpy.random <span class="hljs-keyword">import</span> randn<br><span class="hljs-keyword">from</span> scipy.stats <span class="hljs-keyword">import</span> shapiro<br><span class="hljs-comment"># seed the random number generator</span><br>seed(<span class="hljs-number">1</span>)<br><span class="hljs-comment"># generate univariate observations</span><br>data = <span class="hljs-number">5</span> * randn(<span class="hljs-number">100</span>) + <span class="hljs-number">50</span><br><span class="hljs-comment"># normality test</span><br>stat, p = shapiro(data)<br>print(<span class="hljs-string">&#x27;Statistics=%.3f, p=%.3f&#x27;</span> % (stat, p))<br><span class="hljs-comment"># interpret</span><br>alpha = <span class="hljs-number">0.05</span><br><span class="hljs-keyword">if</span> p &gt; alpha:<br>  print(<span class="hljs-string">&#x27;Sample looks Gaussian (fail to reject H0)&#x27;</span>)<br><span class="hljs-keyword">else</span>:<br>  print(<span class="hljs-string">&#x27;Sample does not look Gaussian (reject H0)&#x27;</span>)<br></code></pre></td></tr></table></figure></div>
<div class="admonition note"><p class="admonition-title">Shapiro in R
</p></div>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">shapiro.test(my_data$len)<br></code></pre></td></tr></table></figure></div>
<pre>
    Shapiro-Wilk normality test
data:  my_data$len
W = 0.96743, p-value = 0.1091
</pre>
<div class="admonition note"><p class="admonition-title">note</p><p>From the output, the p-value &gt; 0.05 implying that the distribution of the data are not significantly different from normal distribution. In other words, we can assume the normality.</p>
</div>
<h3 id="D’Agostino’s-K²-Test-most-widely-and-used-by-GraphPad">D’Agostino’s K² Test (most widely and used by GraphPad)</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-comment"># D&#x27;Agostino and Pearson&#x27;s Test</span><br><span class="hljs-keyword">from</span> numpy.random <span class="hljs-keyword">import</span> seed<br><span class="hljs-keyword">from</span> numpy.random <span class="hljs-keyword">import</span> randn<br><span class="hljs-keyword">from</span> scipy.stats <span class="hljs-keyword">import</span> normaltest<br><span class="hljs-comment"># seed the random number generator</span><br>seed(<span class="hljs-number">1</span>)<br><span class="hljs-comment"># generate univariate observations</span><br>data = <span class="hljs-number">5</span> * randn(<span class="hljs-number">100</span>) + <span class="hljs-number">50</span><br><span class="hljs-comment"># normality test</span><br>stat, p = normaltest(data)<br><span class="hljs-comment"># Stat is k^2</span><br>print(<span class="hljs-string">&#x27;Statistics=%.3f, p=%.3f&#x27;</span> % (stat, p))<br><span class="hljs-comment"># interpret</span><br>alpha = <span class="hljs-number">0.05</span><br><span class="hljs-keyword">if</span> p &gt; alpha:<br>  print(<span class="hljs-string">&#x27;Sample looks Gaussian (fail to reject H0)&#x27;</span>)<br><span class="hljs-keyword">else</span>:<br>  print(<span class="hljs-string">&#x27;Sample does not look Gaussian (reject H0)&#x27;</span>)<br></code></pre></td></tr></table></figure></div>
<h3 id="Anderson-Darling-Test">Anderson-Darling Test</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-comment"># Anderson-Darling Test</span><br><span class="hljs-keyword">from</span> numpy.random <span class="hljs-keyword">import</span> seed<br><span class="hljs-keyword">from</span> numpy.random <span class="hljs-keyword">import</span> randn<br><span class="hljs-keyword">from</span> scipy.stats <span class="hljs-keyword">import</span> anderson<br><span class="hljs-comment"># seed the random number generator</span><br>seed(<span class="hljs-number">1</span>)<br><span class="hljs-comment"># generate univariate observations</span><br>data = <span class="hljs-number">5</span> * randn(<span class="hljs-number">100</span>) + <span class="hljs-number">50</span><br><span class="hljs-comment"># normality test</span><br>result = anderson(data)<br>print(<span class="hljs-string">&#x27;Statistic: %.3f&#x27;</span> % result.statistic)<br>p = <span class="hljs-number">0</span><br><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> <span class="hljs-built_in">range</span>(<span class="hljs-built_in">len</span>(result.critical_values)):<br>  sl, cv = result.significance_level[i], result.critical_values[i]<br>  <span class="hljs-keyword">if</span> result.statistic &lt; result.critical_values[i]:<br>    print(<span class="hljs-string">&#x27;%.3f: %.3f, data looks normal (fail to reject H0)&#x27;</span> % (sl, cv))<br>  <span class="hljs-keyword">else</span>:<br>    print(<span class="hljs-string">&#x27;%.3f: %.3f, data does not look normal (reject H0)&#x27;</span> % (sl, cv))<br></code></pre></td></tr></table></figure></div>
<p>PS: <a target="_blank" rel="noopener" href="https://towardsdatascience.com/normality-tests-in-python-31e04aa4f411">10 Normality Tests in Python (Step-By-Step Guide 2020)</a></p>
<h3 id="SD-and-SEM">SD and SEM</h3>
<p>$$<br>
σ = \sqrt{<br>
\frac {\sum_ N (X_ i - μ)^ 2} { N -1 }<br>
}<br>
$$</p>
<p>$$<br>
sem = \frac{σ}{\sqrt{N}}<br>
$$</p>
<p>SD</p>
<ul>
<li>Describes the width of the parent population as estimated from the scatter in the sample measurements. (For example, when the width of the parent population reflects actual variation of the measured quantity more than in reflects experimental uncertainty)</li>
<li>Does not decrease with more measurements</li>
<li>Should be used when the width of the parent distribution is important to<br>
your audience.</li>
</ul>
<p>SEM</p>
<ul>
<li>Describes the uncertainty in the measured value of the mean</li>
<li>Decreases with more measurements</li>
<li>Should be used when the accuracy in the measured mean is more important to your audience that the width of the parent population. (For example, when most of the width of the parent population is due to experimental uncertainties)</li>
</ul>
<p><mark>Always give the sample size so your reader can convert</mark></p>
<h3 id="Rules-for-expressing-means-and-uncertainties">Rules for expressing means and uncertainties</h3>
<ul>
<li>Express your data with the last significant digit equal to the precision of your measurement.</li>
<li>Plot a histogram to see the distribution. Also test whether your data are distributed in a Gaussian manner.</li>
<li>Present uncertainties as standard deviations if the width of the parent population is important or standard errors if the precision of the measured mean is important. Always define error (SD, SEM) and provide sample number so readers can convert.</li>
<li>Keep in mind that experiments with small sample sizes are much more variable and have higher uncertainties.</li>
</ul>
<h2 id="Lecture-3-The-t-table">Lecture 3 The t-table</h2>
<p>The t distribution is a probability distribution that we can use to express confidence intervals of a mean.</p>
<p>$$<br>
t_ i = \frac{\overline{X_ i} - \mu_ {parent}}{σ/\sqrt{N}}<br>
$$</p>
<p>Interpretation: The deviation of sample mean and parent mean divided by SEM</p>
<h3 id="T-Distributuin">T Distributuin</h3>
<p>For analyze statistical data from small sample.</p>
<p>When measure the small sample Normal distribution, you didn’t get the Gaussine distribution, you get t distribution.</p>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://www.tdistributiontable.com/wp-content/uploads/2020/08/t-table.png" alt="T table"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://www.tdistributiontable.com/">© tdistributiontable.com</a></td>
</tr>
</tbody>
</table>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">dt(x, df, ncp, <span class="hljs-built_in">log</span> = <span class="hljs-literal">FALSE</span>)<br>pt(q, df, ncp, lower.tail = <span class="hljs-literal">TRUE</span>, log.p = <span class="hljs-literal">FALSE</span>)<br>qt(p, df, ncp, lower.tail = <span class="hljs-literal">TRUE</span>, log.p = <span class="hljs-literal">FALSE</span>)<br>rt(n, df, ncp)<br><br><span class="hljs-comment"># two tailed T to P:</span><br><span class="hljs-built_in">T</span> = <span class="hljs-number">2.085963</span><br>N = <span class="hljs-number">20</span><br>pt(<span class="hljs-built_in">T</span>, df=N-<span class="hljs-number">1</span>, lower.tail=<span class="hljs-literal">FALSE</span>)*<span class="hljs-number">2</span><br><span class="hljs-comment"># Two tailed P to T:</span><br>X= <span class="hljs-number">0.05</span><br>N = <span class="hljs-number">20</span><br>qt(X/<span class="hljs-number">2</span>, df=N-<span class="hljs-number">1</span>, lower.tail=<span class="hljs-literal">FALSE</span>)<br></code></pre></td></tr></table></figure></div>
<h3 id="How-to-use-the-T-table">How to use the T table</h3>
<p>In the t table, the index is the degrees of freedom, which is N - 1 (N is the sample size).</p>
<p>For retrieve the rows of the T table:</p>
<ul>
<li>Rows = <mark>N -1</mark>!!! That’s means, when your sample size is 4, you should looking into row 3!</li>
</ul>
<p>T table related examples:<br>
<a target="_blank" rel="noopener" href="https://www.dummies.com/article/academics-the-arts/math/statistics/how-to-use-the-t-table-to-solve-statistics-problems-147282">dummies.com</a></p>
<h3 id="Confidence-Interval-CI-±t-×-SEM">Confidence Interval (CI) = ±t × SEM</h3>
<p>The Confidence Interval (CI) is <strong>the range of possible</strong> means over which there is a specified probability of observing a similarly measured value.<br>
CI is another way of expressing the uncertainty in a mean.</p>
<p>To calculate a confidence interval (CI):</p>
<ol>
<li>Choose what confidence level you want (95% is typical)</li>
<li>p is the probability. Equals 1-confidence level. 95% mean p=0.05</li>
<li>Determine the t-value (t*) from a t-table using dF and p</li>
<li>Calculate CI using the equation above</li>
<li>When you express a CI, make sure the readers knows probability and N</li>
</ol>
<p><strong>Example</strong>:</p>
<blockquote>
<p>You measured 8.72 ± 1.21 (SD, N=7) . What is the 99% CI?</p>
<ol>
<li>Calculate $SE = SD/\sqrt7$ = 0.46</li>
<li>Determine t (for p=0.01 and dF=6) t= 3.71 (t-table)
<ul>
<li>in r:</li>
</ul>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">X= <span class="hljs-number">0.01</span>; N = <span class="hljs-number">7</span><br>qt(X/<span class="hljs-number">2</span>, df = N-<span class="hljs-number">1</span>, lower.tail = <span class="hljs-built_in">F</span>)<br>[<span class="hljs-number">1</span>] <span class="hljs-number">3.707428</span><br></code></pre></td></tr></table></figure></div>
</li>
<li>Calculate CI = t*SE = 3.17*0.48 = 1.78</li>
</ol>
</blockquote>
<h3 id="One-sided-one-tailed-versus-two-sided-confidence-intervals">One sided (one tailed) versus two-sided confidence intervals</h3>
<table>
<thead>
<tr>
<th style="text-align:center"><img src="https://www.analyticsvidhya.com/wp-content/uploads/2015/09/Tailed_Test-850x590.jpg" alt=""></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><a target="_blank" rel="noopener" href="https://www.analyticsvidhya.com/blog/2015/09/hypothesis-testing-explained/">© Sunil Ray; 2015</a></td>
</tr>
</tbody>
</table>
<p>Two tailed probabilities are almost always better than one-tailed probabilities because they are more conservative (less likely to lead you to mistakenly claim a positive correlation) and because they do not require any assumption about the sign of the deviation between the actual and measured value.</p>
<p>For symmetrical distributions (Gaussian, t-distribution, etc) the two-tailed probability is twice the one-tailed probability.</p>
<p>Divide probability in a t-table by 2 to convert two-tailed to one-tailed.<br>
Multiple probability by 2 to covert one-tailed to two-tailed.</p>
<h3 id="The-p-value">The p-value</h3>
<p>The null hypothesis (H<sub>o</sub>) is a hypothetical value of the mean of the parent population. Typically it is the value you would expect to observe if the parameter or conditions of interest have no effect on your measured parameter.</p>
<p>Statistical tests are used to support or disprove the null hypothesis.</p>
<p>The p-value is the probability that you have obtained a difference between your results and the null hypothesis as large as you did observe strictly by chance. In other words, it is the probability of obtaining obtained your results if the null hypothesis was true.</p>
<p><strong>Confidence Intervals and p-values are connected</strong></p>
<p>α is the cutoff that we chose for statistical significance. The ends of the 1-α CI define the values of the null hypothesis that fall exactly at p=α.</p>
<p>A CI defines the range of hypothetical values that have probabilities larger than α, and thus are not significant. Values outside the confidence interval have p-values smaller than α and thus are significant.</p>
<p>$t = \frac{|(H_0 - μ)|}{SE}$<br>
$t = |(H_ 0 -μ)|<em>\sqrt{N}/ sd$<br>
$sd = |(H_ 0 -μ)|</em>\sqrt{N}/t $<br>
$sd = |(0 - )|$</p>
<p>Examples:</p>
<blockquote>
<p>Mesurement: 1.235, 2.1254, 3.8926, 1.0022, 0.9880, 2.9876, 1.4350<br>
Your null hypothesis is that the enzyme expression is 1.0.<br>
Do your data disprove the Null hypothesis (H<sub>0</sub>)?</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">A &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">1.235</span>, <span class="hljs-number">2.1254</span>, <span class="hljs-number">3.8926</span>, <span class="hljs-number">1.0022</span>, <span class="hljs-number">0.9880</span>, <span class="hljs-number">2.9876</span>, <span class="hljs-number">1.4350</span>)<br><span class="hljs-comment"># Step 1: Test the Gaussian?</span><br>shapiro.test(A) <span class="hljs-comment"># p = 0.1528</span><br><span class="hljs-comment"># Step 2: Calculate mean, SD, SE</span><br>mean(A); sd(A); sd(A)/<span class="hljs-built_in">sqrt</span>(<span class="hljs-built_in">length</span>(A))<br>1.952257 <span class="hljs-number">1.116636</span> <span class="hljs-number">0.4220488</span><br><span class="hljs-comment"># Step 3: Calculate a t-value for the difference between the measured mean and the null hypothesis.</span><br><span class="hljs-comment"># t = |(H~0~ - μ)/SE| = (1 – 1.9923)/0.422 = 2.35</span><br><span class="hljs-comment"># Step 4: Look up p-value in a table corresponding to t=2.35 and dF = 6.</span><br><span class="hljs-comment"># Interpolating between values in the Table above, you get p = 0.062.</span><br><span class="hljs-comment"># Instat gives p=0.065 for a hypothetical mean of 1.0</span><br><span class="hljs-built_in">T</span> = <span class="hljs-built_in">abs</span>((<span class="hljs-number">1</span>-mean(A))/ (sd(A)/<span class="hljs-built_in">sqrt</span>(<span class="hljs-built_in">length</span>(A))))<br>N = <span class="hljs-number">7</span><br>pt(<span class="hljs-built_in">T</span>, df=N-<span class="hljs-number">1</span>, lower.tail=<span class="hljs-literal">FALSE</span>)*<span class="hljs-number">2</span><br><span class="hljs-comment"># p = 0.06487952</span><br><span class="hljs-comment"># Or, One step with R:</span><br>t.test(A, mu=<span class="hljs-number">1</span>) <span class="hljs-comment"># p = 0.06488</span><br></code></pre></td></tr></table></figure></div>
<p>At a standard confidence level of 95% (minimum p-value of 5% or 0.05), your data DO NOT disprove the Null hypothesis. <strong>Your measurement is not significantly different from the null hypothesis.</strong></p>
</blockquote>
<p><strong><mark>Assumptions</mark></strong> before t-test:</p>
<ul>
<li>The parent population has a normal (Gaussian) distribution</li>
<li>Your sample is a random sampling of the population</li>
</ul>
<h2 id="Lecture-4-The-p-value">Lecture 4: The p-value</h2>
<p>The p-value is the probability that you have obtained a difference between your results and the null hypothesis as large as you did observe strictly by chance. In other words, due to random sampling.</p>
<p>The p-value is the probability of obtaining your results if the null hypothesis was true.</p>
<h3 id="The-One-Sample-t-test">The One Sample t-test</h3>
<p>The simplest (and most useful) statistical test is the <strong><u>one-sample t-test</u></strong> which compares a <strong>mean</strong> from <strong>set of measurements</strong> to a hypothetical value (the null hypothesis).</p>
<p>The one sample t-test calculates a probability (p-value) that the null hypothesis is true, and that you measured a value as far away from it as yours simply by chance.</p>
<blockquote>
<p>Example data: 7,11,3,5,12,3,0,8<br>
μ = 6.25; σ = 4.0; N = 8; 95% CI = 2.9 to 9.6<br>
<mark>Null Hypothesis is that the value = 0</mark><br>
Our statistical test: Two tailed p-value = 0.0029 (very significant). If the parent<br>
mean is 0, there is a 0.29% chance of measuring a mean of 6.25 ± 4.0 (SD, N=8).<br>
The sample you measured probably does not have a parent mean of 0. The null<br>
hypothesis is disproved.<br>
Now assume that the <mark>Null Hypothesis be that the value = 3</mark><br>
Our statistical test: Two tailed p-value = 0.053 (Not significant at the 95% level)<br>
If the parent mean is 3, there is a 5.3% chance of measuring a mean of 6.25 ± 4.0 by<br>
chance. You have no reason to conclude that the parent mean is not 3. The null<br>
hypothesis is NOT disproved.</p>
</blockquote>
<p>Exp in R:<br>
Ps, I tried running the codes but keep get a sliter different result. The main reason is the mean and sd not match the example data. So, I dicided to change the 0 to 1.</p>
<h4 id="R-codes-for-one-sample-T-test">R codes for one sample T test</h4>
<p><strong>Null Hypothesis is that the value = 0</strong></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">A &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">7</span>,<span class="hljs-number">11</span>,<span class="hljs-number">3</span>,<span class="hljs-number">5</span>,<span class="hljs-number">12</span>,<span class="hljs-number">3</span>,<span class="hljs-number">1</span>,<span class="hljs-number">8</span>)<br>t.test(A)<br></code></pre></td></tr></table></figure></div>
<pre>
  One Sample t-test

  data:  A
  t = 4.4696, df = 7, p-value = 0.002902
  alternative hypothesis: true mean is not equal to 0
  95 percent confidence interval:
  2.943449 9.556551
  sample estimates:
  mean of x
     6.25
</pre>
<p><strong>Null Hypothesis is that the value = 3</strong></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">t.test(A, mu=<span class="hljs-number">3</span>)<br></code></pre></td></tr></table></figure></div>
<pre>
  One Sample t-test

  data:  A
  t = 2.3242, df = 7, p-value = 0.05307
  alternative hypothesis: true mean is not equal to 3
  95 percent confidence interval:
  2.943449 9.556551
  sample estimates:
  mean of x
     6.25
</pre>
<p><strong>Null Hypothesis is that the value = 6</strong></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">t.test(A, mu = <span class="hljs-number">6</span>)<br></code></pre></td></tr></table></figure></div>
<pre>
One Sample t-test

data:  A
t = 0.17878, df = 7, p-value = 0.8632
alternative hypothesis: true mean is not equal to 6
95 percent confidence interval:
2.943449 9.556551
sample estimates:
mean of x
   6.25
</pre>
<h3 id="Two-sample-t-test-unpaired">Two sample t-test: unpaired</h3>
<p>A unpaired experiment is one in which there are no connections between the individual samples measured in the experiments. A “before and after” experiment is unpaired if the identity of the individual samples is not maintained or if both sets of measurements are independent samplings of a parent population.</p>
<p>Example datas and result in R</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">A &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">34</span>, <span class="hljs-number">33</span>, <span class="hljs-number">34</span>, <span class="hljs-number">46</span>, <span class="hljs-number">36</span>, <span class="hljs-number">37</span>, <span class="hljs-number">37</span>, <span class="hljs-number">32</span>, <span class="hljs-number">33</span>, <span class="hljs-number">33</span>)<br>B &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">50</span>, <span class="hljs-number">52</span>, <span class="hljs-number">64</span>, <span class="hljs-number">50</span>, <span class="hljs-number">46</span>, <span class="hljs-number">62</span>, <span class="hljs-number">54</span>, <span class="hljs-number">52</span>, <span class="hljs-number">54</span>, <span class="hljs-number">68</span>)<br>t.test(A, B)<br></code></pre></td></tr></table></figure></div>
<pre>
  Welch Two Sample t-test

  data:  A and B
  t = -7.6291, df = 14.418, p-value = 1.971e-06
  alternative hypothesis: true difference in means is not equal to 0
  95 percent confidence interval:
  -25.22324 -14.17676
  sample estimates:
  mean of x mean of y
     35.5      55.2
</pre>
<p>As we can see, the null hypothesis is accepted. The difference is dromatic.</p>
<p>According to the result above, μ<sub>A</sub> = 35.5, μ<sub>B</sub> = 55.2. In R, we can also using this test to test that the B is larget than A:</p>
<table>
<thead>
<tr>
<th style="text-align:left">Codes</th>
<th style="text-align:left">P value</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= 0, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">1</td>
</tr>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= -10, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">0.999</td>
</tr>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= -20, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">0.4546</td>
</tr>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= -21, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">0.3111</td>
</tr>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= -23, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">0.1107</td>
</tr>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= -24, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">0.05872</td>
</tr>
<tr>
<td style="text-align:left"><code>t.test(A,B, mu= -25, alternative = &quot;greater&quot;)</code></td>
<td style="text-align:left">0.02937</td>
</tr>
</tbody>
</table>
<p>According to the result above, we can “safly” say that <mark><strong>B is greater A than 24</strong></mark>.</p>
<h3 id="Two-sample-t-test-paired">Two sample t-test: paired</h3>
<p>A <strong>paired experiment</strong> is one in which there are multiple measurements of the same sample/subject or on matched (unique) samples/subjects. This can be a <strong>“before and after” experiment</strong>, or an experiments on manually selected matched samples (experiment and control).</p>
<p>Paired sample experiments are more sensitive that unpaired experiments. Paired experiments <strong>DO NOT require</strong> that the samples be Gaussian as long as the <strong>differences (pairwise) are Gaussian</strong>.</p>
<p>We still using the data <code>A, B</code> above</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">t.test(A, B, paired = <span class="hljs-built_in">T</span>)<br></code></pre></td></tr></table></figure></div>
<pre>
Paired t-test

data:  A and B
t = -6.9072, df = 9, p-value = 7.01e-05
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
-26.15189 -13.24811
sample estimates:
mean of the differences
                -19.7
</pre>
<p>Compare the results from one sample t test:</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">t.test(A-B)<br></code></pre></td></tr></table></figure></div>
<pre>
One Sample t-test

data:  A - B
t = -6.9072, df = 9, p-value = 7.01e-05
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
-26.15189 -13.24811
sample estimates:
mean of x
  -19.7
</pre>
<p>We can find that: <strong><mark>The Paired T test is actually running one sample t test with the difference of paired lists</mark></strong></p>
<h3 id="T-test-from-statistic-results">T test from statistic results</h3>
<p>function <code>tsum.test</code> from <code>BSDA</code>:</p>
<pre>
tsum.test(
  mean.x,
  s.x = NULL,
  n.x = NULL,
  mean.y = NULL,
  s.y = NULL,
  n.y = NULL,
  alternative = "two.sided",
  mu = 0,
  var.equal = FALSE,
  conf.level = 0.95
)
</pre>
<p>Instead of give two lists, we given the Mean, Sd, and N (sample size) to do the t-test. (Ps, meam.x; s.x; n.x is for the first sample, mean.y; s.y; n.y are for the second sample)<br>
Codes for R</p>
<table>
<thead>
<tr>
<th style="text-align:left"></th>
<th style="text-align:left">A</th>
<th style="text-align:left">B</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Mean</td>
<td style="text-align:left">0.03</td>
<td style="text-align:left">0.02</td>
</tr>
<tr>
<td style="text-align:left">Sd</td>
<td style="text-align:left">0.05</td>
<td style="text-align:left">0.001</td>
</tr>
<tr>
<td style="text-align:left">N</td>
<td style="text-align:left">18</td>
<td style="text-align:left">3</td>
</tr>
</tbody>
</table>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(BSDA)<br><span class="hljs-comment"># one sample t-test</span><br>tsum.test( <span class="hljs-number">0.03</span>, <span class="hljs-number">0.05</span>, <span class="hljs-number">18</span>)<br><span class="hljs-comment"># two sample t-test for A and B</span><br>tsum.test( <span class="hljs-number">0.03</span>, <span class="hljs-number">0.05</span>, <span class="hljs-number">18</span>, <span class="hljs-number">0.02</span>, <span class="hljs-number">0.001</span>, <span class="hljs-number">3</span>)<br></code></pre></td></tr></table></figure></div>
<h3 id="Non-parametric-tests-for-calculating-confidence-intervals-and-p-values">Non-parametric tests for calculating confidence intervals and p-values</h3>
<p>A non-parametric test is one that does not depend on the assumption that the data have a particular distribution. This makes a non-parametric test weaker than a t-test, because the information contained in the shape of the distribution is not used in a non- Parametric test. But, if the data are strongly <strong>non-Gaussian</strong> and you can not transform The data to make it Gaussian, then you may not have a choice.</p>
<ul>
<li>
<p>one sample versus a null hypothesis</p>
<ul>
<li>Sign Test
<ul>
<li>A sign test is used to decide whether a <strong>binomial distribution</strong> has the equal chance of success and failure.</li>
<li>For number of successes = 5, number of trials = 18, :<br>
<code>binom.test(5, 18) </code><sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup></li>
</ul>
</li>
<li>Mann Whitney test.
<ul>
<li>Two data samples are independent if they come from distinct populations and the samples do not affect each other. Using the Mann-Whitney-Wilcoxon Test, we can decide whether the population distributions are identical without assuming them to follow the <strong>normal distribution</strong>.<br>
<code>wilcox.test(mpg ~ am, data=mtcars)</code><sup class="footnote-ref"><a href="#fn2" id="fnref2">[2]</a></sup></li>
<li>Python<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-keyword">from</span> scipy.stats <span class="hljs-keyword">import</span> mannwhitneyu<br><br>males = [<span class="hljs-number">19</span>, <span class="hljs-number">22</span>, <span class="hljs-number">16</span>, <span class="hljs-number">29</span>, <span class="hljs-number">24</span>]<br>females = [<span class="hljs-number">20</span>, <span class="hljs-number">11</span>, <span class="hljs-number">17</span>, <span class="hljs-number">12</span>]<br>U1, p = mannwhitneyu(males, females)<br>print(p)<br></code></pre></td></tr></table></figure></div>
<pre>
0.055673443266570206
</pre>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p>For unpaired data:</p>
<ul>
<li>Randomization test (See the codes below)
<ul>
<li>It is a test used to analyze the distribution of a set of data to see if it can be described as random (patternless)(<a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Randomness_test">wikipedia</a>)</li>
</ul>
</li>
<li>Mann Whitney test (See the codes above)</li>
</ul>
</li>
<li>
<p>For paired data:</p>
<ul>
<li>Wilcoxon (signed rank) test
<ul>
<li>The Wilcoxon signed-rank test tests the null hypothesis that two related paired samples <strong>come from the same distribution</strong>. In particular, it tests whether the distribution of the differences x - y is symmetric about zero. It is a non-parametric version of the paired T-test.<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-keyword">from</span> scipy.stats <span class="hljs-keyword">import</span> wilcoxon<br>d = [<span class="hljs-number">6</span>, <span class="hljs-number">8</span>, <span class="hljs-number">14</span>, <span class="hljs-number">16</span>, <span class="hljs-number">23</span>, <span class="hljs-number">24</span>, <span class="hljs-number">28</span>, <span class="hljs-number">29</span>, <span class="hljs-number">41</span>, -<span class="hljs-number">48</span>, <span class="hljs-number">49</span>, <span class="hljs-number">56</span>, <span class="hljs-number">60</span>, -<span class="hljs-number">67</span>, <span class="hljs-number">75</span>]<br>w, p = wilcoxon(d)<br>w, p<br></code></pre></td></tr></table></figure></div>
<pre>
(24.0, 0.041259765625)
</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>There are MANY non-parametric tests. These above use nothing more than the actual data collected. The data are scrambled between columns randomly and all the possible outcomes are generated. The p-value is calculated from the probability of an outcome as large or larger than the one observed.</p>
<p>The randomization test uses the actual values. Mann Whitney and Wilcoxon use only the rank of each value (highest, lowest etc) and thus are not sensitive to extreme values.</p>
<h3 id="Randomization-test">Randomization test<sup class="footnote-ref"><a href="#fn3" id="fnref3">[3]</a></sup></h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r"><span class="hljs-comment"># assumes the two samples are in &#x27;x&#x27; and &#x27;y&#x27; and x[i] and y[i] are paired</span><br><span class="hljs-comment"># set up:</span><br>B &lt;- 99999<br>d &lt;- x-y<br>m0 &lt;- mean(d)<br><br><span class="hljs-comment"># perform a one-sample randomization test on d</span><br><span class="hljs-comment"># for the null hypothesis H0: mu_d = 0   vs H1 mu_d != 0  (i.e. two tailed)</span><br><span class="hljs-comment"># here the test statistic is the mean</span><br>rndmdist &lt;- replicate(B,mean((rbinom(<span class="hljs-built_in">length</span>(d),<span class="hljs-number">1</span>,<span class="hljs-number">.5</span>)*<span class="hljs-number">2</span>-<span class="hljs-number">1</span>)*d))<br><br><span class="hljs-comment"># two tailed p-value:</span><br><span class="hljs-built_in">sum</span>( <span class="hljs-built_in">abs</span>(rndmdist) &gt;= <span class="hljs-built_in">abs</span>(m0))/<span class="hljs-built_in">length</span>(rndmdist)<br></code></pre></td></tr></table></figure></div>
<h3 id="T-test-in-python">T test in python</h3>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="PYTHON"><figure class="iseeu highlight /python"><table><tr><td class="code"><pre><code class="hljs python"><span class="hljs-keyword">from</span> scipy <span class="hljs-keyword">import</span> stats<br><br>stats.ttest_ind(set1, set2)<br></code></pre></td></tr></table></figure></div>
<h2 id="Lecture-5-Independence-and-rejection-of-data">Lecture 5 Independence and rejection of data</h2>
<h3 id="Application-of-Chauvenet’s-Criterion">Application of Chauvenet’s Criterion</h3>
<p>PS: you shold really try with the online calculater from <a target="_blank" rel="noopener" href="https://mathcracker.com/chauvenets-criterion-calculator">mathcracker</a>!</p>
<p>According to this website, we need two parameter to find the outliers. One is D<sub>max</sub>, the second is Z-score.<br>
$Pr(Z&gt;D_ {max}) = \frac{1}{4n}$<br>
which in r is: <code>qnorm(1/4/N)</code>, N is the sample size.<br>
So, we can easily define a function like this:</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">chauvenet_test &lt;- <span class="hljs-keyword">function</span>(A)&#123;<br>  Dmax = <span class="hljs-built_in">abs</span>(qnorm(<span class="hljs-number">1</span>/<span class="hljs-number">4</span>/<span class="hljs-built_in">length</span>(A)))<br>  Outer = A[<span class="hljs-built_in">abs</span>(scale(A))&gt;= <span class="hljs-built_in">abs</span>(Dmax)]<br>  Z_out = <span class="hljs-built_in">abs</span>(scale(A))[<span class="hljs-built_in">abs</span>(scale(A))&gt;= <span class="hljs-built_in">abs</span>(Dmax)]<br>  P_val = <span class="hljs-number">1</span>- pnorm(<span class="hljs-built_in">abs</span>(scale(A)), lower.tail = <span class="hljs-built_in">T</span>)<br>  print(paste(<span class="hljs-string">&quot;Dmax:&quot;</span>, Dmax))<br>  print(paste(<span class="hljs-string">&quot;Outliners:&quot;</span>,Outer))<br>  print(paste(<span class="hljs-string">&quot;Zscore:&quot;</span>, Z_out))<br>  print(paste(<span class="hljs-string">&quot;P value:&quot;</span>, P_val))<br>  <span class="hljs-built_in">return</span>(Outer)<br>&#125;<br></code></pre></td></tr></table></figure></div>
<blockquote>
<p>We measure 17, 23, 41, 19, 29, 34, 99 in a new type of experiment. You are suspicious of the measurement of 99. You want to know if you can reject any points.<br>
37.43 ± 28.44(SD, N=7)</p>
<ul>
<li>99 is 2.17 standard deviations from the mean (Z-scores)</li>
<li>The p-value is 0.0152, and 0.0152×7=0.1064<br>
As we calculated below, μ = 37.43, σ = 28.44, Z = 2.17. The probability of Z according from the Z table is 0.015, which means if the cut threshold is 0.05 or 0.03, 99 should be rejected. When the cut threshold is 0.01, then, 99 should be accepted.<br>
On the other hand, the P value of 17 is 0.76, which is acceptable.</li>
</ul>
</blockquote>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">A &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">17</span>, <span class="hljs-number">23</span>, <span class="hljs-number">41</span>, <span class="hljs-number">19</span>, <span class="hljs-number">29</span>, <span class="hljs-number">34</span>, <span class="hljs-number">99</span>)<br>Dmax = <span class="hljs-built_in">abs</span>(qnorm(<span class="hljs-number">1</span>/<span class="hljs-number">4</span>/<span class="hljs-built_in">length</span>(A)))<br>TB = data.frame(Raw= A)<br>TB$Mean &lt;- mean(A)<br>TB$SD &lt;- sd(A)<br>TB$Z_score = scale(A)<br>TB$P_value &lt;- 1- pnorm(q= scale(A), lower.tail=<span class="hljs-literal">TRUE</span>)<br>print(TB$Raw[<span class="hljs-built_in">abs</span>(TB$Z_score)&gt;=Dmax])<br></code></pre></td></tr></table></figure></div>
<table class="dataframe">
<thead>
  <tr><th scope="col">Raw</th><th scope="col">Mean</th><th scope="col">SD</th><th scope="col">Z_score</th><th scope="col">P_value</th></tr>
</thead>
<tbody>
  <tr><td>17</td><td>37.42857</td><td>28.43623</td><td>-0.7183995</td><td>0.76374450</td></tr>
  <tr><td>23</td><td>37.42857</td><td>28.43623</td><td>-0.5074010</td><td>0.69406327</td></tr>
  <tr><td>41</td><td>37.42857</td><td>28.43623</td><td> 0.1255943</td><td>0.45002653</td></tr>
  <tr><td>19</td><td>37.42857</td><td>28.43623</td><td>-0.6480667</td><td>0.74152909</td></tr>
  <tr><td>29</td><td>37.42857</td><td>28.43623</td><td>-0.2964026</td><td>0.61653867</td></tr>
  <tr><td>34</td><td>37.42857</td><td>28.43623</td><td>-0.1205705</td><td>0.54798440</td></tr>
  <tr><td>99</td><td>37.42857</td><td>28.43623</td><td> 2.1652460</td><td>0.01518443</td></tr>
</tbody>
</table>
<h2 id="Another-Example">Another Example</h2>
<p>when the triple experiments of the result is 1, 1, 1000000, then similarly, we can have the table below. It might out of your expectation, but the value 1000000 is actually acceptable.</p>
<table class="dataframe">
<thead>
  <tr><th scope="col">Raw</th><th scope="col">Mean</th><th scope="col">SD</th><th scope="col">Z_score</th><th scope="col">P_value</th></tr>
</thead>
<tbody>
  <tr><td>1</td><td>333334</td><td>577349.7</td><td>-0.5773503</td><td>0.7181486</td></tr>
  <tr><td>1</td><td>333334</td><td>577349.7</td><td>-0.5773503</td><td>0.7181486</td></tr>
  <tr><td>1e+06</td><td>333334</td><td>577349.7</td><td> 1.1547005</td><td>0.1241065</td></tr>
</tbody>
</table>
<h2 id="Lecture-6-Propagation-of-errors">Lecture 6: Propagation of errors</h2>
<h3 id="Weighted-averages">Weighted averages</h3>
<p>$$<br>
\mu_ {weighted} = \frac{\sum(\frac{\mu_ i}{\sigma_ i^2})}{\sum(\frac{1}{\sigma_ i^2})}<br>
$$</p>
<p>$$<br>
\sigma_ {weighted} = \sqrt{<br>
\frac{1}{\sum(\frac{1}{\sigma_ i^2})}<br>
}<br>
$$</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">TB = data.frame(Mean = <span class="hljs-built_in">c</span>(<span class="hljs-number">121</span>, <span class="hljs-number">110</span>, <span class="hljs-number">131</span>), Sd = <span class="hljs-built_in">c</span>(<span class="hljs-number">21</span>, <span class="hljs-number">9</span>, <span class="hljs-number">19</span>), N = <span class="hljs-built_in">c</span>(<span class="hljs-number">9</span>, <span class="hljs-number">13</span>, <span class="hljs-number">11</span>))<br><br><br>Wei_Mean &lt;- <span class="hljs-keyword">function</span>(TB)&#123;<br>  W_mean_t1 = <span class="hljs-number">0</span><br>  W_mean_t2 = <span class="hljs-number">0</span><br>  W_sd = <span class="hljs-number">0</span><br>  <span class="hljs-keyword">for</span>(i <span class="hljs-keyword">in</span> <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>: nrow(TB)))&#123;<br>    Mean = TB[i,<span class="hljs-number">1</span>]<br>    Sd = TB[i,<span class="hljs-number">2</span>]<br>    W_mean_t1 = W_mean_t1 + Mean/(Sd^<span class="hljs-number">2</span>)<br>    W_mean_t2 = W_mean_t2 + <span class="hljs-number">1</span>/(Sd^<span class="hljs-number">2</span>)<br>    W_sd =  <span class="hljs-number">1</span>/(Sd^<span class="hljs-number">2</span>)<br>  &#125;<br>  WM = W_mean_t1/W_mean_t2<br>  WS = <span class="hljs-built_in">sqrt</span>(<span class="hljs-number">1</span>/W_mean_t2)<br>  DF = <span class="hljs-built_in">sum</span>(TB[<span class="hljs-number">3</span>]) - nrow(TB)<br>  SE = WS/<span class="hljs-built_in">sqrt</span>(<span class="hljs-built_in">sum</span>(TB[<span class="hljs-number">3</span>]))<br>  print(paste(<span class="hljs-string">&quot;Weighted Mean:&quot;</span>, WM))<br>  print(paste(<span class="hljs-string">&quot;Weighted Sd:&quot;</span>, WS))<br>  print(paste(<span class="hljs-string">&quot;df:&quot;</span>, DF))<br>  print(paste(<span class="hljs-string">&quot;SE:&quot;</span>, SE))<br>&#125;<br></code></pre></td></tr></table></figure></div>
<pre>
[1] "Weighted Mean: 122.067278328693"
[1] "Weighted Sd: 7.79700223506481"
[1] "df: 30"
[1] "SE: 4.50160133928678"
</pre>
<h3 id="Propagation-of-Errors">Propagation of Errors</h3>
<p>If the calculated number x is a function of measured parameters u and V then the uncertainty in x is given by</p>
<p>$σ^2_x = σ^2_u(\frac{\partial x}{\partial u})^2 + σ^2_v(\frac{\partial x}{\partial u})^2 + 2σ^2_{uV} (\frac{\partial x}{\partial u}) (\frac{\partial x}{\partial V})$</p>
<p>$\frac{\partial x}{\partial u}$is the derivative of x with respect to u.<br>
In other words, dx/du is a measure of how much x changes when u changes. This equation can be applied to any equation in which x is a function of U and v.</p>
<p>The last term in the equation above is the <mark>covariance</mark> which becomes significant only if the uncertainties in the parameters are correlated.<br>
Most of the time, the covariance can be ignored.</p>
<p><a target="_blank" rel="noopener" href="https://www.youtube.com/watch?v=qtaqvPAeEJY">Covariance; youtube</a></p>
<p>$$<br>
σ^2_{uV} = \frac{\sum(u_i - u_{mean})(V_ i - V_ {mean})}{N}<br>
$$</p>
<p>For uncorrelated (random) samples the covariance is zero.</p>
<p><a target="_blank" rel="noopener" href="https://www.youtube.com/watch?v=V0ZRvvHfF0E">Propagation of Errors;  GlacierFilm, LLC. 2013</a></p>
<p>Exp:<br>
13.2m/s ± 0.2m/s<br>
measured velocity: v=13.2 m/s<br>
absolute uncertainty in v: $δv= ± 0.2m/s$<br>
relative uncertainty in v: $\frac{δv}{v}$</p>
<h3 id="Powers">Powers</h3>
<p>$x = αμ^n $<br>
$σ_ x = nx\frac{σ_ μ}{μ}$</p>
<p><strong>Logarithm</strong><br>
$x = α log(μ)$<br>
$σ_ x = α\frac{σ_ μ}{μ}$</p>
<h3 id="Propagation-of-Errors-Calcumation">Propagation of Errors: Calcumation</h3>
<p><strong>X</strong>: is the new number obtained by doing math on measurement (s)<br>
<strong>U</strong> and <strong>V</strong>: are the measured parameters (with uncertainties)</p>
<ul>
<li>
<p>Addition and Subtraction</p>
<ul>
<li>$x=u±V$</li>
<li>$σ_ x = \sqrt{σ_ u ^2 + σ_ V ^2  ± 2σ_ {uV} ^2}$</li>
<li>Ignoring covariance: $σ_ x = \sqrt{σ_ u ^2 + σ_ V ^2}$</li>
</ul>
</li>
<li>
<p>Multiplication:</p>
<ul>
<li>$x = uV$</li>
<li>$\frac{σ_ x}{x} = \sqrt{ \frac{σ_ u}{u}^ 2+ \frac{σ_ V}{V}^ 2 }$</li>
</ul>
</li>
<li>
<p>Division:</p>
<ul>
<li>$x = \frac{u}{V}$</li>
<li>$\frac{σ_ x}{x} = \sqrt{ \frac{σ_ u}{u}^ 2+ \frac{σ_ V}{V}^ 2 }$</li>
</ul>
</li>
</ul>
<h3 id="Degrees-of-freedom-and-calculation-of-standard-errors-for-a-propagated-error">Degrees of freedom and calculation of standard errors for a propagated error.</h3>
<ul>
<li>Propagation of errors is always done using standard deviations.</li>
<li>If N is the same for each of the experiments in your propagated result, then dF for the propagated value is N-1.</li>
<li>If your two variables have different N, use the dF for the smallest N.</li>
</ul>
<p>EXP:</p>
<blockquote>
<p>u= 20.34 ± 0.51 (SD, N=10, dF = 9)<br>
V = 17.81 ± 1.45 (SD, N=10, dF = 9)<br>
When you propagate errors you have</p>
<ul>
<li>$x = u × V = 20.34 × 17.81 = 362.26$</li>
<li>$σ_ x = x\sqrt{ \frac{σ_ u}{u}^ 2+ \frac{σ_ V}{V}^ 2 }$<br>
$σ_ x = 362.26 × \sqrt{ \frac{0.51}{20.34}^ 2+ \frac{1.45}{17.81}^ 2 }$<br>
$σ_ x= 30.86$</li>
</ul>
<p>u×V = 362.26 ± 30.86 (SD, N=10, dF = 9)<br>
SE = SD/N = 9.7589 = 9.8 (proper sig figs)<br>
With proper sig figs the result is 362.3 ± 9.8 (SE, dF = 9)<br>
For the purpose of calculating confidence intervals, p-values etc the result can now be treated as a Mean ± SE with dF =9.<br>
Code in R:</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="code"><pre><code class="hljs r">library(propagate)<br>x &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">20.34</span>, <span class="hljs-number">0.51</span>)<br>y &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">17.81</span>, <span class="hljs-number">1.45</span> )<br>EXPR1 &lt;- <span class="hljs-built_in">expression</span>(x*y)<br>DF1 &lt;- cbind(x,y)<br>propagate(expr = EXPR1, data = DF1)  <br></code></pre></td></tr></table></figure></div>
<pre>
Results from uncertainty propagation:
  Mean.1    Mean.2      sd.1      sd.2      2.5%     97.5%
362.25540 362.25540  30.86000  30.86886 301.75348 422.75732
Results from Monte Carlo simulation:
    Mean        sd    Median       MAD      2.5%     97.5%
362.28108  30.89162 362.04746  30.85789 302.25524 423.44032
</pre>
</blockquote>
<hr class="footnotes-sep">
<section class="footnotes">
<ol class="footnotes-list">
<li id="fn1" class="footnote-item"><p><a target="_blank" rel="noopener" href="http://www.r-tutor.com/elementary-statistics/non-parametric-methods/sign-test">Sign Test; R Tutorial</a> <a href="#fnref1" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn2" class="footnote-item"><p><a target="_blank" rel="noopener" href="https://www.r-tutor.com/elementary-statistics/non-parametric-methods/mann-whitney-wilcoxon-test">Mann-Whitney-Wilcoxon Test; R Tutorial</a> <a href="#fnref2" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn3" class="footnote-item"><p><a target="_blank" rel="noopener" href="https://stats.stackexchange.com/questions/64212/randomisation-permutation-test-for-paired-vectors-in-r">Glen_b; 2013</a> <a href="#fnref3" class="footnote-backref">↩︎</a></p>
</li>
</ol>
</section>
</div><div class="article-licensing box"><div class="licensing-title"><p>BioStatistics with R/Python: Block 1</p><p><a href="https://karobben.github.io/2022/01/24/LearnNotes/tulane-biostat-1/">https://karobben.github.io/2022/01/24/LearnNotes/tulane-biostat-1/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>Author</h6><p>Karobben</p></div></div><div class="level-item is-narrow"><div><h6>Posted on</h6><p>2022-01-24</p></div></div><div class="level-item is-narrow"><div><h6>Updated on</h6><p>2023-06-06</p></div></div><div class="level-item is-narrow"><div><h6>Licensed under</h6><p><a class="icon" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a><a class="icon" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a><a class="icon" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/Classes/">Classes</a><a class="link-muted mr-2" rel="tag" href="/tags/Tulane-Classes/">Tulane Classes</a><a class="link-muted mr-2" rel="tag" href="/tags/BioStatistics/">BioStatistics</a></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/css/share.min.css"><div class="social-share"></div><script src="https://cdn.jsdelivr.net/npm/social-share.js@1.0.16/dist/js/social-share.min.js"></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">Like this article? Support the author with</h3><div class="buttons is-centered"><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>Alipay</span><span class="qrcode"><img src="/alipay.jpg" alt="Alipay"></span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>Wechat</span><span class="qrcode"><img src="/wepay.png" alt="Wechat"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2022/01/31/Linux/popos/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">Pop OS</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2022/01/24/LearnNotes/tulane-genetics-1/"><span class="level-item">Notes for Genetics</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">Comments</h3></div><span styl="max-width: 95%"><script src="https://giscus.app/client.js" data-repo="karobben/utterances" data-repo-id="MDEwOlJlcG9zaXRvcnkyNzUwNjUwODg=" data-category="Announcements" data-category-id="DIC_kwDOEGUpAM4B-wRL" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-theme="light" crossorigin="anonymous" async></script></span></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="https://s1.ax1x.com/2022/09/27/xZnWOU.jpg" alt="Karobben"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">Karobben</p><p class="is-size-6 is-block">Enjoy!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>China</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">Posts</p><a href="/archives"><p class="title">504</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Categories</p><a href="/categories"><p class="title">148</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">Tags</p><a href="/tags"><p class="title">175</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/karobben" target="_blank" rel="noopener">Follow</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/karobben"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/atom.xml"><i class="fas fa-rss"></i></a></div></div><div class="card"><script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5obi4wthjqz&amp;m=0c&amp;c=f03b11&amp;cr1=ffffff&amp;sx=0&amp;cw=ffffff&amp;cb=3472cd" async="async"></script></div></div><div class="card widget is-sticky" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">Catalogue</h3><ul class="menu-list"><li><a class="level is-mobile" href="#Biomedical-Statistics-and-Data-Analysis"><span class="level-left"><span class="level-item">1</span><span class="level-item">Biomedical Statistics and Data Analysis</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Normal-Distribution"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">Normal Distribution</span></span></a></li></ul></li><li><a class="level is-mobile" href="#CI-calculate-®"><span class="level-left"><span class="level-item">2</span><span class="level-item">CI calculate ®</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Shapiro-Wilk-Test"><span class="level-left"><span class="level-item">2.1</span><span class="level-item">Shapiro-Wilk Test</span></span></a></li><li><a class="level is-mobile" href="#D’Agostino’s-K²-Test-most-widely-and-used-by-GraphPad"><span class="level-left"><span class="level-item">2.2</span><span class="level-item">D’Agostino’s K² Test (most widely and used by GraphPad)</span></span></a></li><li><a class="level is-mobile" href="#Anderson-Darling-Test"><span class="level-left"><span class="level-item">2.3</span><span class="level-item">Anderson-Darling Test</span></span></a></li><li><a class="level is-mobile" href="#SD-and-SEM"><span class="level-left"><span class="level-item">2.4</span><span class="level-item">SD and SEM</span></span></a></li><li><a class="level is-mobile" href="#Rules-for-expressing-means-and-uncertainties"><span class="level-left"><span class="level-item">2.5</span><span class="level-item">Rules for expressing means and uncertainties</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Lecture-3-The-t-table"><span class="level-left"><span class="level-item">3</span><span class="level-item">Lecture 3 The t-table</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#T-Distributuin"><span class="level-left"><span class="level-item">3.1</span><span class="level-item">T Distributuin</span></span></a></li><li><a class="level is-mobile" href="#How-to-use-the-T-table"><span class="level-left"><span class="level-item">3.2</span><span class="level-item">How to use the T table</span></span></a></li><li><a class="level is-mobile" href="#Confidence-Interval-CI-±t-×-SEM"><span class="level-left"><span class="level-item">3.3</span><span class="level-item">Confidence Interval (CI) = ±t × SEM</span></span></a></li><li><a class="level is-mobile" href="#One-sided-one-tailed-versus-two-sided-confidence-intervals"><span class="level-left"><span class="level-item">3.4</span><span class="level-item">One sided (one tailed) versus two-sided confidence intervals</span></span></a></li><li><a class="level is-mobile" href="#The-p-value"><span class="level-left"><span class="level-item">3.5</span><span class="level-item">The p-value</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Lecture-4-The-p-value"><span class="level-left"><span class="level-item">4</span><span class="level-item">Lecture 4: The p-value</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#The-One-Sample-t-test"><span class="level-left"><span class="level-item">4.1</span><span class="level-item">The One Sample t-test</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#R-codes-for-one-sample-T-test"><span class="level-left"><span class="level-item">4.1.1</span><span class="level-item">R codes for one sample T test</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Two-sample-t-test-unpaired"><span class="level-left"><span class="level-item">4.2</span><span class="level-item">Two sample t-test: unpaired</span></span></a></li><li><a class="level is-mobile" href="#Two-sample-t-test-paired"><span class="level-left"><span class="level-item">4.3</span><span class="level-item">Two sample t-test: paired</span></span></a></li><li><a class="level is-mobile" href="#T-test-from-statistic-results"><span class="level-left"><span class="level-item">4.4</span><span class="level-item">T test from statistic results</span></span></a></li><li><a class="level is-mobile" href="#Non-parametric-tests-for-calculating-confidence-intervals-and-p-values"><span class="level-left"><span class="level-item">4.5</span><span class="level-item">Non-parametric tests for calculating confidence intervals and p-values</span></span></a></li><li><a class="level is-mobile" href="#Randomization-test"><span class="level-left"><span class="level-item">4.6</span><span class="level-item">Randomization test[3]</span></span></a></li><li><a class="level is-mobile" href="#T-test-in-python"><span class="level-left"><span class="level-item">4.7</span><span class="level-item">T test in python</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Lecture-5-Independence-and-rejection-of-data"><span class="level-left"><span class="level-item">5</span><span class="level-item">Lecture 5 Independence and rejection of data</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Application-of-Chauvenet’s-Criterion"><span class="level-left"><span class="level-item">5.1</span><span class="level-item">Application of Chauvenet’s Criterion</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Another-Example"><span class="level-left"><span class="level-item">6</span><span class="level-item">Another Example</span></span></a></li><li><a class="level is-mobile" href="#Lecture-6-Propagation-of-errors"><span class="level-left"><span class="level-item">7</span><span class="level-item">Lecture 6: Propagation of errors</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Weighted-averages"><span class="level-left"><span class="level-item">7.1</span><span class="level-item">Weighted averages</span></span></a></li><li><a class="level is-mobile" href="#Propagation-of-Errors"><span class="level-left"><span class="level-item">7.2</span><span class="level-item">Propagation of Errors</span></span></a></li><li><a class="level is-mobile" href="#Powers"><span class="level-left"><span class="level-item">7.3</span><span class="level-item">Powers</span></span></a></li><li><a class="level is-mobile" href="#Propagation-of-Errors-Calcumation"><span class="level-left"><span class="level-item">7.4</span><span class="level-item">Propagation of Errors: Calcumation</span></span></a></li><li><a class="level is-mobile" href="#Degrees-of-freedom-and-calculation-of-standard-errors-for-a-propagated-error"><span class="level-left"><span class="level-item">7.5</span><span class="level-item">Degrees of freedom and calculation of standard errors for a propagated error.</span></span></a></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">Tags</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/3D/"><span class="tag">3D</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/AI/"><span class="tag">AI</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/API/"><span class="tag">API</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ATAC-Seq/"><span class="tag">ATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Advanced-Mathematics/"><span class="tag">Advanced Mathematics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Aging/"><span class="tag">Aging</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Align/"><span class="tag">Align</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Android/"><span class="tag">Android</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Audio/"><span class="tag">Audio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Beginner/"><span class="tag">Beginner</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Behavior/"><span class="tag">Behavior</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BiliBili/"><span class="tag">BiliBili</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/BioStatistics/"><span class="tag">BioStatistics</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biochemistry/"><span class="tag">Biochemistry</span><span class="tag">43</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatic/"><span class="tag">Bioinformatic</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Bioinformatics/"><span class="tag">Bioinformatics</span><span class="tag">69</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biology/"><span class="tag">Biology</span><span class="tag">38</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Biopython/"><span class="tag">Biopython</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Book/"><span class="tag">Book</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cancer/"><span class="tag">Cancer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell/"><span class="tag">Cell</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cell-Biology/"><span class="tag">Cell Biology</span><span class="tag">24</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Chip-Seq/"><span class="tag">Chip-Seq</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Classes/"><span class="tag">Classes</span><span class="tag">74</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Cluster/"><span class="tag">Cluster</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Conky/"><span class="tag">Conky</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Crawler/"><span class="tag">Crawler</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Data/"><span class="tag">Data</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Database/"><span class="tag">Database</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deep-Learning/"><span class="tag">Deep Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Deepin/"><span class="tag">Deepin</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Desktop/"><span class="tag">Desktop</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Disk/"><span class="tag">Disk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Distribution/"><span class="tag">Distribution</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Dme/"><span class="tag">Dme</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/English/"><span class="tag">English</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/FTP/"><span class="tag">FTP</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Fish/"><span class="tag">Fish</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GATK/"><span class="tag">GATK</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GO/"><span class="tag">GO</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Genome/"><span class="tag">Genome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitHub/"><span class="tag">GitHub</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GitPage/"><span class="tag">GitPage</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Graphviz/"><span class="tag">Graphviz</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HTML/"><span class="tag">HTML</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo/"><span class="tag">Hexo</span><span class="tag">19</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hexo-Plugin/"><span class="tag">Hexo Plugin</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/IDE/"><span class="tag">IDE</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Image/"><span class="tag">Image</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Immune/"><span class="tag">Immune</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Interoperable-Plot/"><span class="tag">Interoperable Plot</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Japanese/"><span class="tag">Japanese</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Jupyter-Notebook/"><span class="tag">Jupyter Notebook</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG/"><span class="tag">KEGG</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/KEGG-pathway/"><span class="tag">KEGG pathway</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kivy/"><span class="tag">Kivy</span><span class="tag">31</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Latex/"><span class="tag">Latex</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Links/"><span class="tag">Links</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Logistic/"><span class="tag">Logistic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning/"><span class="tag">Machine Learning</span><span class="tag">13</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Machine-Learning-Data-Preprocessing/"><span class="tag">Machine Learning. Data Preprocessing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MarkDown/"><span class="tag">MarkDown</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MateGenome/"><span class="tag">MateGenome</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Math/"><span class="tag">Math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matplotlib/"><span class="tag">Matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Matrix/"><span class="tag">Matrix</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Meta-omics/"><span class="tag">Meta-omics</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabloism/"><span class="tag">Metabloism</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Metabolism/"><span class="tag">Metabolism</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mice/"><span class="tag">Mice</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind/"><span class="tag">Mind</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mind-Plot/"><span class="tag">Mind Plot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Multimedia/"><span class="tag">Multimedia</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Mysql/"><span class="tag">Mysql</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/NGS/"><span class="tag">NGS</span><span class="tag">26</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Network/"><span class="tag">Network</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Neuron/"><span class="tag">Neuron</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Notes/"><span class="tag">Notes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Nutrition-matrix/"><span class="tag">Nutrition matrix</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Odds-Ratio/"><span class="tag">Odds Ratio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/OpenCV/"><span class="tag">OpenCV</span><span class="tag">17</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Others/"><span class="tag">Others</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PCA/"><span class="tag">PCA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PDF/"><span class="tag">PDF</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PacBio/"><span class="tag">PacBio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Paper/"><span class="tag">Paper</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Papers/"><span class="tag">Papers</span><span class="tag">22</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Physics/"><span class="tag">Physics</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Plot/"><span class="tag">Plot</span><span class="tag">62</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protein-Structure/"><span class="tag">Protein Structure</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Proteomic/"><span class="tag">Proteomic</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Protocol/"><span class="tag">Protocol</span><span class="tag">9</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyMol/"><span class="tag">PyMol</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Python/"><span class="tag">Python</span><span class="tag">132</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Pyvista/"><span class="tag">Pyvista</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QC/"><span class="tag">QC</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Quantum-Physics/"><span class="tag">Quantum Physics</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">86</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R-packages/"><span class="tag">R packages</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA/"><span class="tag">RNA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-Seq/"><span class="tag">RNA-Seq</span><span class="tag">21</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNA-seq/"><span class="tag">RNA-seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RNN/"><span class="tag">RNN</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/RasperryPi/"><span class="tag">RasperryPi</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Regression/"><span class="tag">Regression</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Response-Surface-Methodology/"><span class="tag">Response Surface Methodology</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/SNP/"><span class="tag">SNP</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Sanger-Sequencing/"><span class="tag">Sanger Sequencing</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scientific-American/"><span class="tag">Scientific American</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Script/"><span class="tag">Script</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scripts/"><span class="tag">Scripts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Seaborn/"><span class="tag">Seaborn</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Software/"><span class="tag">Software</span><span class="tag">50</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Statistic/"><span class="tag">Statistic</span><span class="tag">14</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Super-Conductor/"><span class="tag">Super Conductor</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Synthetic-Cell/"><span class="tag">Synthetic Cell</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/System/"><span class="tag">System</span><span class="tag">6</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TCGA/"><span class="tag">TCGA</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI/"><span class="tag">TUI</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/TUI-lib/"><span class="tag">TUI lib</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Ted/"><span class="tag">Ted</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tensorflow/"><span class="tag">Tensorflow</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Termux/"><span class="tag">Termux</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tools/"><span class="tag">Tools</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Transfer-Learning/"><span class="tag">Transfer Learning</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Translation/"><span class="tag">Translation</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Trinity/"><span class="tag">Trinity</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tulane-Classes/"><span class="tag">Tulane Classes</span><span class="tag">39</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tutorials/"><span class="tag">Tutorials</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/UIUC-Classes/"><span class="tag">UIUC Classes</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Uniprot/"><span class="tag">Uniprot</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Urwid/"><span class="tag">Urwid</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Video/"><span class="tag">Video</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Vocabulary/"><span class="tag">Vocabulary</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WGCNA/"><span class="tag">WGCNA</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Waque/"><span class="tag">Waque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Web/"><span class="tag">Web</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Wet/"><span class="tag">Wet</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/WordCloud/"><span class="tag">WordCloud</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Writing/"><span class="tag">Writing</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Yuque/"><span class="tag">Yuque</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/abi/"><span class="tag">abi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/antibody/"><span class="tag">antibody</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/api/"><span class="tag">api</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/apt/"><span class="tag">apt</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/awk/"><span class="tag">awk</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bash/"><span class="tag">bash</span><span class="tag">12</span></a></div><div class="control"><a class="tags has-addons" href="/tags/biology/"><span class="tag">biology</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/blast/"><span class="tag">blast</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/bug/"><span class="tag">bug</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/busuanzi/"><span class="tag">busuanzi</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dash/"><span class="tag">dash</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/dme/"><span class="tag">dme</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/email/"><span class="tag">email</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/fasta/"><span class="tag">fasta</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flies/"><span class="tag">flies</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/flybase/"><span class="tag">flybase</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggkaboom/"><span class="tag">ggkaboom</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot/"><span class="tag">ggplot</span><span class="tag">42</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot2/"><span class="tag">ggplot2</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ggplot-splitbar/"><span class="tag">ggplot_splitbar</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/grep/"><span class="tag">grep</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/immunology/"><span class="tag">immunology</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/live2d/"><span class="tag">live2d</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/map/"><span class="tag">map</span><span class="tag">5</span></a></div><div class="control"><a class="tags has-addons" href="/tags/math/"><span class="tag">math</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/matplotlib/"><span class="tag">matplotlib</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/miRNA-Seq/"><span class="tag">miRNA-Seq</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/nCov/"><span class="tag">nCov</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/orc/"><span class="tag">orc</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scATAC-Seq/"><span class="tag">scATAC-Seq</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/scRNA-Seq/"><span class="tag">scRNA-Seq</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/torch/"><span class="tag">torch</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/wet/"><span class="tag">wet</span><span class="tag">1</span></a></div></div></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">Subscribe for updates</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="Subscribe"></div></div></form></div></div></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">Links</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://www.jianshu.com/u/77fc519c6fa2" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">卖萌哥</span></span><span class="level-right"><span class="level-item tag">www.jianshu.com</span></span></a></li></ul></div></div></div><div class="column-right-shadow is-hidden-widescreen is-sticky"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3 is-sticky"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">Recommends</h3><article class="media"><figure class="media-left"><a class="image" href="/2021/02/11/Blog/hexo_icarus/"><img src="https://camo.githubusercontent.com/fd83d8c9f8b8dbe85686812e2e186dceec728bbabad2d331aa44e0bfeee8628c/68747470733a2f2f70706f66666963652e6769746875622e696f2f6865786f2d7468656d652d6963617275732f67616c6c6572792f707265766965772e706e673f31" alt="Hexo theme: icarus| Highly personalize it"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-11T06:00:00.000Z">2021-02-11</time></p><p class="title"><a href="/2021/02/11/Blog/hexo_icarus/">Hexo theme: icarus| Highly personalize it</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Hexo/">Hexo</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/12/16/Python/kivy_Cross-platform-App/"><img src="https://kivy.org/logos/kivy-logo-black-64.png" alt="Kivy: hello world (Cross-platform GUI) for python"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-12-16T06:00:00.000Z">2020-12-16</time></p><p class="title"><a href="/2020/12/16/Python/kivy_Cross-platform-App/">Kivy: hello world (Cross-platform GUI) for python</a></p><p class="categories"><a href="/categories/Python/">Python</a> / <a href="/categories/Python/Kivy/">Kivy</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/26/Blog/waque/"><img src="https://gw.alipayobjects.com/zos/rmsportal/xwaMkpycAdwCBrdgyWiT.png" alt="为什么我要使用瓦雀| markdown 语法大全"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-26T05:00:00.000Z">2020-06-26</time></p><p class="title"><a href="/2020/06/26/Blog/waque/">为什么我要使用瓦雀| markdown 语法大全</a></p><p class="categories"><a href="/categories/others/">others</a> / <a href="/categories/others/Blog/">Blog</a> / <a href="/categories/others/Blog/Yuque/">Yuque</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/12/R/GGplot/"><img src="https://www.rstudio.com/wp-content/uploads/2014/04/ggplot2.png" alt="GGplot | A tutorials for beginner"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-12T05:00:00.000Z">2020-06-12</time></p><p class="title"><a href="/2020/06/12/R/GGplot/">GGplot | A tutorials for beginner</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/GGPLOT/">GGPLOT</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/08/21/R/logistic_curve/"><img src="https://s1.ax1x.com/2022/10/21/x6dNZ9.png" alt="logistic regression in R"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-08-21T05:00:00.000Z">2020-08-21</time></p><p class="title"><a href="/2020/08/21/R/logistic_curve/">logistic regression in R</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Data/">Data</a> / <a href="/categories/R/Data/Statistic/">Statistic</a> / <a href="/categories/R/Data/Statistic/Regression/">Regression</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/29/Bioinfor/Competent_c/"><img src="https://tse3-mm.cn.bing.net/th/id/OIP.pg0lLEEeNeiUp31DPMKtRwHaCY" alt="Calcium Competent Cells"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-29T05:00:00.000Z">2020-06-29</time></p><p class="title"><a href="/2020/06/29/Bioinfor/Competent_c/">Calcium Competent Cells</a></p><p class="categories"><a href="/categories/Biology/">Biology</a> / <a href="/categories/Biology/Wet-Protocol/">Wet Protocol</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2020/06/20/R/pheatmap/"><img src="https://s1.ax1x.com/2020/08/15/dFpkUH.png" alt="Pheatmap"></a></figure><div class="media-content"><p class="date"><time dateTime="2020-06-20T05:00:00.000Z">2020-06-20</time></p><p class="title"><a href="/2020/06/20/R/pheatmap/">Pheatmap</a></p><p class="categories"><a href="/categories/R/">R</a> / <a href="/categories/R/Plot/">Plot</a> / <a href="/categories/R/Plot/Heatmap/">Heatmap</a></p></div></article><article class="media"><figure class="media-left"><a class="image" href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/"><img src="https://cdn.pixabay.com/photo/2016/08/29/06/21/fly-1627592_960_720.jpg" alt="Diet switch| Black soldier fly larvae (BSFL) meal"></a></figure><div class="media-content"><p class="date"><time dateTime="2021-02-21T06:00:00.000Z">2021-02-21</time></p><p class="title"><a href="/2021/02/21/LearnNotes/Paper_BSFlylavea_review/">Diet switch| Black soldier fly larvae (BSFL) meal</a></p><p class="categories"><a href="/categories/Notes/">Notes</a> / <a href="/categories/Notes/Paper/">Paper</a> / <a href="/categories/Notes/Paper/Biology/">Biology</a></p></div></article></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">Categories</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">70</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">64</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Chip-Seq/"><span class="level-start"><span class="level-item">Chip-Seq</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protein-Structure/"><span class="level-start"><span class="level-item">Protein Structure</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Proteomic/"><span class="level-start"><span class="level-item">Proteomic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/"><span class="level-start"><span class="level-item">Protocol</span></span><span class="level-end"><span class="level-item tag">18</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Alternative-Splicing/"><span class="level-start"><span class="level-item">Alternative Splicing</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Annotation/"><span class="level-start"><span class="level-item">Annotation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/GO/"><span class="level-start"><span class="level-item">GO</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/Genome/"><span class="level-start"><span class="level-item">Genome</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/RNA-Seq/"><span class="level-start"><span class="level-item">RNA-Seq</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/SNP/"><span class="level-start"><span class="level-item">SNP</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Protocol/miRNA/"><span class="level-start"><span class="level-item">miRNA</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Single-Cell/"><span class="level-start"><span class="level-item">Single Cell</span></span><span class="level-end"><span class="level-item tag">8</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">22</span></span></a><ul><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Align/"><span class="level-start"><span class="level-item">Align</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Bioconda/"><span class="level-start"><span class="level-item">Bioconda</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/De-nove/"><span class="level-start"><span class="level-item">De nove</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Dna-Methylation/"><span class="level-start"><span class="level-item">Dna Methylation</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Download/"><span class="level-start"><span class="level-item">Download</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/FastQC/"><span class="level-start"><span class="level-item">FastQC</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Fastq-QC/"><span class="level-start"><span class="level-item">Fastq QC</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/List/"><span class="level-start"><span class="level-item">List</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Sam/"><span class="level-start"><span class="level-item">Sam</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Trans/"><span class="level-start"><span class="level-item">Trans</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/Visualization/"><span class="level-start"><span class="level-item">Visualization</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/Software/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Bioinformatics/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Biology/Database/"><span class="level-start"><span class="level-item">Database</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Biology/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Arch/"><span class="level-start"><span class="level-item">Arch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/"><span class="level-start"><span class="level-item">Bash</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/Bash/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/More/"><span class="level-start"><span class="level-item">More</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Bash/awk-grep-sed/"><span class="level-start"><span class="level-item">awk/grep/sed</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/GitHub/"><span class="level-start"><span class="level-item">GitHub</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Monitor-top-Conky/"><span class="level-start"><span class="level-item">Monitor: top\Conky</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/RasperryPi/"><span class="level-start"><span class="level-item">RasperryPi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Software/"><span class="level-start"><span class="level-item">Software</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/System/"><span class="level-start"><span class="level-item">System</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Termux/"><span class="level-start"><span class="level-item">Termux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/Ubuntu/"><span class="level-start"><span class="level-item">Ubuntu</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Linux/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/"><span class="level-start"><span class="level-item">Notes</span></span><span class="level-end"><span class="level-item tag">140</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Biology/Bioinformatic/"><span class="level-start"><span class="level-item">Bioinformatic</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Book/"><span class="level-start"><span class="level-item">Book</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Genetics/"><span class="level-start"><span class="level-item">Genetics</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Immunity/"><span class="level-start"><span class="level-item">Immunity</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Virus/"><span class="level-start"><span class="level-item">Virus</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Biology/Wet-Protocol/"><span class="level-start"><span class="level-item">Wet Protocol</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/"><span class="level-start"><span class="level-item">Class</span></span><span class="level-end"><span class="level-item tag">75</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">25</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Math/"><span class="level-start"><span class="level-item">Math</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/"><span class="level-start"><span class="level-item">Tulane</span></span><span class="level-end"><span class="level-item tag">39</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/BioStatistics/"><span class="level-start"><span class="level-item">BioStatistics</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Bioinformatics/"><span class="level-start"><span class="level-item">Bioinformatics</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/Tulane/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">16</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/"><span class="level-start"><span class="level-item">UIUC</span></span><span class="level-end"><span class="level-item tag">8</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Biochemistry/"><span class="level-start"><span class="level-item">Biochemistry</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Class/UIUC/Cell-Biology/"><span class="level-start"><span class="level-item">Cell Biology</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/English/"><span class="level-start"><span class="level-item">English</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Japanese/"><span class="level-start"><span class="level-item">Japanese</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">23</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Dme/"><span class="level-start"><span class="level-item">Dme</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Paper/Biology/Reading-Notes/"><span class="level-start"><span class="level-item">Reading Notes</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">12</span></span></a><ul><li><a class="level is-mobile" href="/categories/Notes/Statistic/Data-Scientists/"><span class="level-start"><span class="level-item">Data Scientists</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/Distribution/"><span class="level-start"><span class="level-item">Distribution</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/Statistic/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Notes/Ted/"><span class="level-start"><span class="level-item">Ted</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/flies/"><span class="level-start"><span class="level-item">flies</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Notes/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/"><span class="level-start"><span class="level-item">Paper</span></span><span class="level-end"><span class="level-item tag">2</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/"><span class="level-start"><span class="level-item">Biology</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Paper/Biology/Immune/"><span class="level-start"><span class="level-item">Immune</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Paper/Diseases/"><span class="level-start"><span class="level-item">Diseases</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/"><span class="level-start"><span class="level-item">Python</span></span><span class="level-end"><span class="level-item tag">127</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/API/"><span class="level-start"><span class="level-item">API</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Crawler/"><span class="level-start"><span class="level-item">Crawler</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">17</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/Deep-Learning/"><span class="level-start"><span class="level-item">Deep Learning</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Data/Machine-Learning/RNN/"><span class="level-start"><span class="level-item">RNN</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Data/Matrix/"><span class="level-start"><span class="level-item">Matrix</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/GUI/"><span class="level-start"><span class="level-item">GUI</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/GUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/Kivy/"><span class="level-start"><span class="level-item">Kivy</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Kivy/Toolbox/"><span class="level-start"><span class="level-item">Toolbox</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/OpenCV/"><span class="level-start"><span class="level-item">OpenCV</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/"><span class="level-start"><span class="level-item">Scripting</span></span><span class="level-end"><span class="level-item tag">14</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/Scripting/Module/"><span class="level-start"><span class="level-item">Module</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/Scripting/Practice/"><span class="level-start"><span class="level-item">Practice</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/TUI/"><span class="level-start"><span class="level-item">TUI</span></span><span class="level-end"><span class="level-item tag">6</span></span></a><ul><li><a class="level is-mobile" href="/categories/Python/TUI/Urwid/"><span class="level-start"><span class="level-item">Urwid</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/Python/TUI/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/Python/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/"><span class="level-start"><span class="level-item">R</span></span><span class="level-end"><span class="level-item tag">84</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Beginner/"><span class="level-start"><span class="level-item">Beginner</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/"><span class="level-start"><span class="level-item">Bio</span></span><span class="level-end"><span class="level-item tag">10</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Bio/Abi/"><span class="level-start"><span class="level-item">Abi</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Anno-Enrich/"><span class="level-start"><span class="level-item">Anno/Enrich</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/DEG/"><span class="level-start"><span class="level-item">DEG</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/Fasta/"><span class="level-start"><span class="level-item">Fasta</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Bio/WGCNA/"><span class="level-start"><span class="level-item">WGCNA</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Data/"><span class="level-start"><span class="level-item">Data</span></span><span class="level-end"><span class="level-item tag">9</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Data-Clean/"><span class="level-start"><span class="level-item">Data Clean</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Machine-Learning/"><span class="level-start"><span class="level-item">Machine Learning</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Data/Statistic/"><span class="level-start"><span class="level-item">Statistic</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Data/Statistic/Regression/"><span class="level-start"><span class="level-item">Regression</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/R/Image/"><span class="level-start"><span class="level-item">Image</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Pack-your-pack/"><span class="level-start"><span class="level-item">Pack your pack</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/"><span class="level-start"><span class="level-item">Plot</span></span><span class="level-end"><span class="level-item tag">56</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/R/Plot/Blog/Algorithm/"><span class="level-start"><span class="level-item">Algorithm</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/Plot/GGPLOT/"><span class="level-start"><span class="level-item">GGPLOT</span></span><span class="level-end"><span class="level-item tag">39</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Heatmap/"><span class="level-start"><span class="level-item">Heatmap</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/Maps/"><span class="level-start"><span class="level-item">Maps</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/VisuaProtocol/"><span class="level-start"><span class="level-item">VisuaProtocol</span></span><span class="level-end"><span class="level-item tag">5</span></span></a></li><li><a class="level is-mobile" href="/categories/R/Plot/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/R/libraries/"><span class="level-start"><span class="level-item">libraries</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/R/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/"><span class="level-start"><span class="level-item">others</span></span><span class="level-end"><span class="level-item tag">30</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Android/"><span class="level-start"><span class="level-item">Android</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/"><span class="level-start"><span class="level-item">Blog</span></span><span class="level-end"><span class="level-item tag">25</span></span></a><ul><li><a class="level is-mobile" href="/categories/others/Blog/HTML/"><span class="level-start"><span class="level-item">HTML</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Hexo/"><span class="level-start"><span class="level-item">Hexo</span></span><span class="level-end"><span class="level-item tag">17</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Sharing/"><span class="level-start"><span class="level-item">Sharing</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/Yuque/"><span class="level-start"><span class="level-item">Yuque</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/others/Blog/more/"><span class="level-start"><span class="level-item">more</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/others/Tools/"><span class="level-start"><span class="level-item">Tools</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/others/else/"><span class="level-start"><span class="level-item">else</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li></ul></li></ul></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="https://s1.ax1x.com/2022/09/27/xZnL6K.png" alt="Karobben" height="28"></a><script async src="/js/busuanzi.pure.js"></script><span id="busuanzi_container_site_pv">Visitied by <span id="busuanzi_value_site_pv"></span> times</span><p class="is-size-7"><span>&copy; 2023 Karobben</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/Karobben"><i class="fab fa-github"></i></a></p></div></div></div></div><script src="https://cdnjs.cloudflare.com/ajax/libs/viz.js/1.7.1/viz.js"></script></footer><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script><script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script><script>moment.locale("en");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="Back to top" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "This website uses cookies to improve your experience.",
          dismiss: "Got it!",
          allow: "Allow cookies",
          deny: "Decline",
          link: "Learn more",
          policy: "Cookie Policy",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script><script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/x-mathjax-config">MathJax.Hub.Config({
            'HTML-CSS': {
                matchFontHeight: false
            },
            SVG: {
                matchFontHeight: false
            },
            CommonHTML: {
                matchFontHeight: false
            },
            tex2jax: {
                inlineMath: [
                    ['$','$'],
                    ['\\(','\\)']
                ]
            }
        });</script><script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><script src="https://unpkg.com/mermaid@8.8.4/dist/mermaid.min.js"></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container"><input class="searchbox-input" type="text" placeholder="Type something..."></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div></div></div><script src="/js/insight.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadInsight({"contentUrl":"/content.json"}, {"hint":"Type something...","untitled":"(Untitled)","posts":"Posts","pages":"Pages","categories":"Categories","tags":"Tags"});
        });</script><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body></html>