<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>WGCNA - 實戰 | Karobben</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="WGCNA - 實戰">
<meta property="og:type" content="website">
<meta property="og:title" content="WGCNA - 實戰">
<meta property="og:url" content="https://karobben.github.io/R/WGCNA.html">
<meta property="og:site_name" content="Karobben">
<meta property="og:description" content="WGCNA - 實戰">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://i.loli.net/2020/06/10/PXY8AC1jcZiHlBQ.png">
<meta property="og:image" content="https://i.loli.net/2020/06/10/mwHBczOjh4TVFNp.png">
<meta property="og:image" content="https://i.loli.net/2020/06/10/jh13CuI86y9Sfbg.png">
<meta property="og:image" content="https://i.loli.net/2020/06/10/uEx8fr32wglcOIK.png">
<meta property="og:image" content="https://i.loli.net/2020/06/10/HYXmnS3acVgPOyU.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/Ks396OUztmydnu2.png">
<meta property="og:image" content="https://i.loli.net/2020/06/10/6sAcpShBmkZXGyW.png">
<meta property="og:image" content="https://i.loli.net/2020/06/10/jWYKMUZ1o4gqQu5.png">
<meta property="og:image" content="https://i.loli.net/2020/06/08/wrpXGJj4qQOulim.png">
<meta property="og:image" content="https://s1.ax1x.com/2020/06/10/tTvZu9.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/pr5GuX3Z27dHkiz.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/eLtsGyoIViqhdJ7.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/vSlio9atZdUREDe.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/2zYmxMo7ARr3y9L.png">
<meta property="og:image" content="https://s1.ax1x.com/2020/06/11/tHgxTs.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/F95UNu6cLeAD8Jb.png">
<meta property="og:image" content="https://s1.ax1x.com/2020/06/11/tbo3UU.png">
<meta property="og:image" content="https://s1.ax1x.com/2020/06/11/tbo1ET.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/hUMPXSTErxqBI9n.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/hWnYQKlG1HpdqPU.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/PGng6FisKjfvJ48.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/IZJjsXWo2AmlHxU.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/E3GxfJa8T2VNRlk.png">
<meta property="og:image" content="https://i.loli.net/2020/06/11/awg2zCr6LpblocB.png">
<meta property="og:image" content="https://i.loli.net/2020/06/13/1COSfoY6KJGTFil.png">
<meta property="article:published_time" content="2020-07-16T07:13:34.175Z">
<meta property="article:modified_time" content="2020-07-16T07:13:34.175Z">
<meta property="article:author" content="Karobben">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://i.loli.net/2020/06/10/PXY8AC1jcZiHlBQ.png">
  
    <link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml">
  
  
    <link rel="icon" href="img/tag.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

  
<link rel="stylesheet" href="/css/search.css">

  <script type="text/javascript"
   src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>

  <style>
  pre {
    color: salmon;
  }
  </style>
<meta name="generator" content="Hexo 4.2.1"></head>

<script>
  var _hmt = _hmt || [];
  (function() {
    var hm = document.createElement("script");
    hm.src = "https://hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hm, s);
  })();
</script>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header"; style="background:#1B414B;height:150px">
  <!--div id="banner"></div>-->
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Karobben</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">Karobben Blog</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
          <a id="main-nav-toggle" class="nav-icon"></a>
          <!-- Header Section Start -->
  <header id="slider-area">
    <nav class="navbar navbar-expand-md fixed-top scrolling-navbar bg-white">
      <div class="container">
        <div class="collapse navbar-collapse" id="navbarCollapse">
          <ul class="navbar-nav mr-auto w-100 justify-content-end">
              <a class="main-nav-link" href="../index.html">Home</a>
              <a class="main-nav-link" href="../R/R-index.html">R</a>
              <a class="main-nav-link" href="../Python/index.html">Python</a>
              <a class="main-nav-link" href="../Linux/index.html">Linux</a>
              <a class="main-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
              <a class="main-nav-link" href="../LearnNotes/index.html">Notes</a>
              <a class="main-nav-link" href="../Blog/index.html">Others</a>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <!-- Header Section End -->

      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Looking for Something?"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://karobben.github.io"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><div align='right', style='padding-right:160px;padding-top:1px'>
  
  <span class="local-search local-search-google local-search-plugin" style=";position:absolute;z-index:2;">
      <input type="search" placeholder="Looking for Something?" id="local-search-input" class="local-search-input-cls" style="">
      <div id="local-search-result" class="local-search-result-cls"></div>
  </span>


</div>

<article  id="page-" class="article article-type-page" itemscope itemprop="blogPost">
<div id="tipue_search_content" style="display: none"></div>
  <div class="article-meta">
    <a href="/R/WGCNA.html" class="article-date">
  <time datetime="2020-07-16T07:13:34.175Z" itemprop="datePublished">2020-07-16</time>
</a>
    
  </div>
  <div  class="article-inner">
    
    
      <header class="article-header">
        <!--
  
    <h1 class="article-title" itemprop="name">
      WGCNA - 實戰
    </h1>
  
 -->
      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Go To Top Link -->
        <div title="Back to Top" class="back-to-top" style="background:salmon">
  <a href="#"  >
    <i >Top</i>
  </a>
</div>
<div class="title_index">
  <ul id="nav">
   <li>
     Titles
     <ul>
       <!-- Table of Contents -->

  <div id="toc" class="toc-article">
    <strong class="toc-title">Titles</strong>
    <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#WGCNA-實戰"><span class="toc-number">1.</span> <span class="toc-text">WGCNA - 實戰</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#0-數據結構"><span class="toc-number">2.</span> <span class="toc-text">0. 數據結構</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#0-1-Expression-matrix"><span class="toc-number">2.1.</span> <span class="toc-text">0.1 Expression matrix</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#0-2-Traits"><span class="toc-number">2.2.</span> <span class="toc-text">0.2 Traits</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#1-Data-Clean"><span class="toc-number">3.</span> <span class="toc-text">1. Data Clean</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-1-Load-and-Filter"><span class="toc-number">3.1.</span> <span class="toc-text">1.1 Load and Filter</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-2-Clust"><span class="toc-number">3.2.</span> <span class="toc-text">1.2 Clust</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-3-Loading-Traits"><span class="toc-number">3.3.</span> <span class="toc-text">1.3 Loading Traits</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2-Network-construction-and-module-detection"><span class="toc-number">4.</span> <span class="toc-text">2. Network construction and module detection</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#2-1-Pick-soft-thresdholding-power"><span class="toc-number">4.1.</span> <span class="toc-text">2.1 Pick soft-thresdholding power</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#3-刪除不必要reads，再來做一遍"><span class="toc-number">5.</span> <span class="toc-text">3. 刪除不必要reads，再來做一遍</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#4-One-step-NetWork"><span class="toc-number">6.</span> <span class="toc-text">4. One-step NetWork</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#5-Traits"><span class="toc-number">7.</span> <span class="toc-text">5. Traits</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#6-Focus-on-Weight"><span class="toc-number">8.</span> <span class="toc-text">6. Focus on Weight</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#7-NetWork-Visualizing"><span class="toc-number">9.</span> <span class="toc-text">7. NetWork  Visualizing</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#8-數據導出"><span class="toc-number">10.</span> <span class="toc-text">8. 數據導出</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#9-如果-softThread-還是取6"><span class="toc-number">11.</span> <span class="toc-text">9. 如果 softThread 還是取6</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Modules-分佈"><span class="toc-number">11.1.</span> <span class="toc-text">Modules 分佈</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Optimal"><span class="toc-number">12.</span> <span class="toc-text">Optimal</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#A1-Reduce-Input-Transcripts"><span class="toc-number">12.1.</span> <span class="toc-text">A1. Reduce Input Transcripts</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#B-Remove-low-hits-amp-Non-DEGs"><span class="toc-number">13.</span> <span class="toc-text">B. Remove low-hits &amp; Non-DEGs</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#画图"><span class="toc-number">13.0.1.</span> <span class="toc-text">画图</span></a></li></ol></li></ol></li></ol>
  </div>

<!-- Table of Contents Down-->

     </ul>
    </li>
   </ul>
</div>


        <h1 id="WGCNA-實戰"><a href="#WGCNA-實戰" class="headerlink" title="WGCNA - 實戰"></a>WGCNA - 實戰</h1><h1 id="0-數據結構"><a href="#0-數據結構" class="headerlink" title="0. 數據結構"></a>0. 數據結構</h1><h2 id="0-1-Expression-matrix"><a href="#0-1-Expression-matrix" class="headerlink" title="0.1 Expression matrix"></a>0.1 Expression matrix</h2><p>Trinity script TPM result</p>
<table>
<thead>
<tr>
<th align="center">ID1</th>
<th align="center">ID2</th>
<th align="center">Liver_CK</th>
<th align="center">Intest_CK</th>
<th align="center">Muscle_CK</th>
<th align="center">Liver_30</th>
<th align="center">Intest_30</th>
<th align="center">Muscle_30</th>
<th align="center">Liver_75</th>
<th align="center">Intest_75</th>
<th align="center">Muscle_75</th>
</tr>
</thead>
<tbody><tr>
<td align="center">TRINITY_DN100000_c1_g1</td>
<td align="center">TRINITY_DN100000_c1_g1_i1</td>
<td align="center">2.1</td>
<td align="center">4.3</td>
<td align="center">1.32</td>
<td align="center">1.04</td>
<td align="center">4.49</td>
<td align="center">2.85</td>
<td align="center">3.59</td>
<td align="center">4.27</td>
<td align="center">1.7</td>
</tr>
<tr>
<td align="center">TRINITY_DN100001_c0_g1</td>
<td align="center">TRINITY_DN100001_c0_g1_i1</td>
<td align="center">0.45</td>
<td align="center">21.0</td>
<td align="center">0.02</td>
<td align="center">0.15</td>
<td align="center">8.85</td>
<td align="center">4.33</td>
<td align="center">0.18</td>
<td align="center">14.76</td>
<td align="center">0.0</td>
</tr>
<tr>
<td align="center">TRINITY_DN100002_c0_g1</td>
<td align="center">TRINITY_DN100002_c0_g1_i1</td>
<td align="center">8.39</td>
<td align="center">2.01</td>
<td align="center">1.08</td>
<td align="center">11.32</td>
<td align="center">3.2</td>
<td align="center">1.1</td>
<td align="center">7.83</td>
<td align="center">6.23</td>
<td align="center">2.91</td>
</tr>
</tbody></table>
<h2 id="0-2-Traits"><a href="#0-2-Traits" class="headerlink" title="0.2 Traits"></a>0.2 Traits</h2><table>
<thead>
<tr>
<th align="center">ID</th>
<th align="center">Group</th>
<th align="center">Organ</th>
<th align="center">Weight</th>
</tr>
</thead>
<tbody><tr>
<td align="center">Liver_CK</td>
<td align="center">CK</td>
<td align="center">Liver</td>
<td align="center">33.32</td>
</tr>
<tr>
<td align="center">Intest_CK</td>
<td align="center">CK</td>
<td align="center">Intest</td>
<td align="center">33.32</td>
</tr>
<tr>
<td align="center">Muscle_CK</td>
<td align="center">CK</td>
<td align="center">Muscle</td>
<td align="center">33.32</td>
</tr>
<tr>
<td align="center">Liver_30</td>
<td align="center">SPC30</td>
<td align="center">Liver</td>
<td align="center">31.3</td>
</tr>
<tr>
<td align="center">Intest_30</td>
<td align="center">SPC30</td>
<td align="center">Intest</td>
<td align="center">31.3</td>
</tr>
<tr>
<td align="center">…</td>
<td align="center">…</td>
<td align="center">…</td>
<td align="center">…</td>
</tr>
</tbody></table>
<h1 id="1-Data-Clean"><a href="#1-Data-Clean" class="headerlink" title="1. Data Clean"></a>1. Data Clean</h1><h2 id="1-1-Load-and-Filter"><a href="#1-1-Load-and-Filter" class="headerlink" title="1.1 Load and Filter"></a>1.1 Load and Filter</h2><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(WGCNA)</span><br><span class="line"></span><br><span class="line">A &lt;- read.csv(<span class="string">"All_isoform.TPM.matrix.anno.xls"</span>,sep=<span class="string">'\t'</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># rownames:sample; colnames: Transcripts</span></span><br><span class="line">row.names(A) = A[[<span class="number">2</span>]]</span><br><span class="line">A = A[-(<span class="number">1</span>:<span class="number">2</span>)]</span><br><span class="line">datExpr0 = t(A)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Data filter</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr0, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!gsg$allOK)</span><br><span class="line">&#123;</span><br><span class="line">  <span class="comment"># Optionally, print the gene and sample names that were removed:</span></span><br><span class="line">  <span class="keyword">if</span> (sum(!gsg$goodGenes)&gt;<span class="number">0</span>)</span><br><span class="line">     printFlush(paste(<span class="string">"Removing genes:"</span>, paste(names(datExpr0)[!gsg$goodGenes], collapse = <span class="string">", "</span>)));</span><br><span class="line">  <span class="keyword">if</span> (sum(!gsg$goodSamples)&gt;<span class="number">0</span>)</span><br><span class="line">     printFlush(paste(<span class="string">"Removing samples:"</span>, paste(rownames(datExpr0)[!gsg$goodSamples], collapse = <span class="string">", "</span>)));</span><br><span class="line">  <span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line">  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></div>
<h2 id="1-2-Clust"><a href="#1-2-Clust" class="headerlink" title="1.2 Clust"></a>1.2 Clust</h2><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">sampleTree = hclust(dist(datExpr0), method = <span class="string">"average"</span>);</span><br><span class="line"><span class="comment">#png(file = "sampleClustering_T45.png",wi=300, he=300)</span></span><br><span class="line">par(cex = <span class="number">0.6</span>);</span><br><span class="line">par(mar = c(<span class="number">0</span>,<span class="number">4</span>,<span class="number">2</span>,<span class="number">0</span>))</span><br><span class="line">plot(sampleTree, main = <span class="string">"Sample clustering to detect outliers"</span>, sub=<span class="string">""</span>, xlab=<span class="string">""</span>, cex.lab = <span class="number">1.5</span>,</span><br><span class="line">     cex.axis = <span class="number">1.5</span>, cex.main = <span class="number">2</span>)</span><br><span class="line"><span class="comment">#dev.off()</span></span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/PXY8AC1jcZiHlBQ.png" alt="sampleClustering"><br>相對來說， Liver30 異質性有點大， 不過， 根據實驗來說， 應該是正常現象，因此跳過刪除異質性項。</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">datExpr =  datExpr0</span><br><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br></pre></td></tr></table></figure></div>

<h2 id="1-3-Loading-Traits"><a href="#1-3-Loading-Traits" class="headerlink" title="1.3 Loading Traits"></a>1.3 Loading Traits</h2><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">traitData = read.csv(<span class="string">"traits.csv"</span>)</span><br><span class="line"></span><br><span class="line">allTraits = traitData[-c(<span class="number">2</span>,<span class="number">3</span>)]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Samples = rownames(datExpr);</span><br><span class="line">traitRows = match(Samples, allTraits$ID);</span><br><span class="line">datTraits = allTraits[traitRows, -<span class="number">1</span>];</span><br><span class="line">rownames(datTraits) = allTraits[traitRows, <span class="number">1</span>];</span><br><span class="line"></span><br><span class="line">collectGarbage()</span><br><span class="line"></span><br><span class="line"><span class="comment"># Re-cluster samples</span></span><br><span class="line">sampleTree2 = hclust(dist(datExpr), method = <span class="string">"average"</span>)</span><br><span class="line"><span class="comment"># Convert traits to a color representation: white means low, red means high, grey means missing entry</span></span><br><span class="line">traitColors = numbers2colors(datTraits, signed = <span class="literal">FALSE</span>);</span><br><span class="line"><span class="comment"># Plot the sample dendrogram and the colors underneath.</span></span><br><span class="line"><span class="comment"># png("traits_heatmap.png")</span></span><br><span class="line">plotDendroAndColors(sampleTree2, traitColors,</span><br><span class="line">                    groupLabels = names(datTraits),</span><br><span class="line">                    main = <span class="string">"Sample dendrogram and trait heatmap"</span>)</span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/mwHBczOjh4TVFNp.png" alt="traits_heatmap"></p>
<p>ummm, 並不能看出什麼來。只是個普通的熱圖加樹衛而已</p>
<h1 id="2-Network-construction-and-module-detection"><a href="#2-Network-construction-and-module-detection" class="headerlink" title="2. Network construction and module detection"></a>2. Network construction and module detection</h1><h2 id="2-1-Pick-soft-thresdholding-power"><a href="#2-1-Pick-soft-thresdholding-power" class="headerlink" title="2.1 Pick soft-thresdholding power"></a>2.1 Pick soft-thresdholding power</h2><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">powers = c(c(<span class="number">1</span>:<span class="number">10</span>), seq(from = <span class="number">12</span>, to=<span class="number">20</span>, by=<span class="number">2</span>))</span><br><span class="line"><span class="comment"># Call the network topology analysis function</span></span><br><span class="line">sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = <span class="number">5</span>)</span><br><span class="line"><span class="comment"># Plot the results:</span></span><br><span class="line">sizeGrWindow(<span class="number">9</span>, <span class="number">5</span>)</span><br><span class="line"><span class="comment">#png("sorft_Thread.png",wi=630,he=350)</span></span><br><span class="line"><span class="comment"># Scale-free topology fit index as a function of the soft-thresholding power</span></span><br><span class="line"><span class="comment"># png("T45_Soft_th.png",width=500,height=300)</span></span><br><span class="line">cex1 = <span class="number">0.9</span>;</span><br><span class="line">par(mfrow = c(<span class="number">1</span>,<span class="number">2</span>));</span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Scale Free Topology Model Fit,signed R^2"</span>,type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Scale independence"</span>));</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], -sign(sft$fitIndices[,<span class="number">3</span>])*sft$fitIndices[,<span class="number">2</span>],</span><br><span class="line">     labels=powers,cex=cex1,col=<span class="string">"red"</span>);</span><br><span class="line"><span class="comment"># this line corresponds to using an R^2 cut-off of h</span></span><br><span class="line">abline(h=<span class="number">0.90</span>,col=<span class="string">"red"</span>)</span><br><span class="line"><span class="comment"># Mean connectivity as a function of the soft-thresholding power</span></span><br><span class="line">plot(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>],</span><br><span class="line">     xlab=<span class="string">"Soft Threshold (power)"</span>,ylab=<span class="string">"Mean Connectivity"</span>, type=<span class="string">"n"</span>,</span><br><span class="line">     main = paste(<span class="string">"Mean connectivity"</span>))</span><br><span class="line">text(sft$fitIndices[,<span class="number">1</span>], sft$fitIndices[,<span class="number">5</span>], labels=powers, cex=cex1,col=<span class="string">"red"</span>)</span><br><span class="line"><span class="comment">#dev.off()</span></span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/jh13CuI86y9Sfbg.png" alt="sorft_Thread"></p>
<p>ummm… 等了半天，就這。。。這種垃圾數據的話， 還是放棄把。</p>
<h1 id="3-刪除不必要reads，再來做一遍"><a href="#3-刪除不必要reads，再來做一遍" class="headerlink" title="3. 刪除不必要reads，再來做一遍"></a>3. 刪除不必要reads，再來做一遍</h1><p>想法： 刪除低hit的reads， 減少計算量，減少垃圾數據的污染</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">Counts &lt;- read.csv(<span class="string">"/media/ken/Data/Yan/RNA-seq/report/4.exprs//All_isoform.COUNT.matrix.anno.xls"</span>,sep=<span class="string">'\t'</span>)</span><br><span class="line">Counts$Sum = rowSums(Counts[-c(<span class="number">1</span>:<span class="number">2</span>)])</span><br><span class="line">Counts_sub = Counts[which(Counts$Sum&gt;<span class="number">10</span>),]</span><br><span class="line">paste(round((nrow(Counts_sub)/nrow(Counts))*<span class="number">100</span>,<span class="number">2</span>),<span class="string">"%"</span>,sep=<span class="string">""</span>)</span><br><span class="line"></span><br><span class="line">A_sub &lt;- A[match(Counts_sub[[<span class="number">2</span>]],rownames(A)),]</span><br></pre></td></tr></table></figure></div>
<p>然後， 重複上面步驟, 從1.1 <code>datExpr0 = t(A)</code>替換成<code>datExpr0 = t(A_sub)</code>開始<br><img src="https://i.loli.net/2020/06/10/uEx8fr32wglcOIK.png" alt="Soft_th2"><br>鐺鐺鐺鐺鐺! 這次結果就好看多了把！<br>雖然沒有 Tutorial 那樣標緻，但是，至少給了我繼續下去的勇氣。<br>數據不好，就調高閾值把， 更具這條線，大概是要選9了</p>
<h1 id="4-One-step-NetWork"><a href="#4-One-step-NetWork" class="headerlink" title="4. One-step NetWork"></a>4. One-step NetWork</h1><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">net = blockwiseModules(datExpr, power = <span class="number">9</span>,</span><br><span class="line">                       TOMType = <span class="string">"unsigned"</span>, minModuleSize = <span class="number">30</span>,</span><br><span class="line">                       reassignThreshold = <span class="number">0</span>, mergeCutHeight = <span class="number">0.25</span>,</span><br><span class="line">                       numericLabels = <span class="literal">TRUE</span>, pamRespectsDendro = <span class="literal">FALSE</span>,</span><br><span class="line">                       saveTOMs = <span class="literal">TRUE</span>,</span><br><span class="line">                       saveTOMFileBase = <span class="string">"femaleMouseTOM"</span>,</span><br><span class="line">                       verbose = <span class="number">3</span>)</span><br></pre></td></tr></table></figure></div>
<p>這一步等待時間和你電腦計算能力，基因數量成正相關</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">moduleLabels = net$colors</span><br><span class="line">moduleColors = labels2colors(net$colors)</span><br><span class="line">MEs = net$MEs;</span><br><span class="line">geneTree = net$dendrograms[[<span class="number">1</span>]]</span><br><span class="line"><span class="comment"># open a graphics window</span></span><br><span class="line">sizeGrWindow(<span class="number">12</span>, <span class="number">9</span>)</span><br><span class="line"><span class="comment"># png('modules_tree.png',width=480, height=320)</span></span><br><span class="line"><span class="comment"># Convert labels to colors for plotting</span></span><br><span class="line">mergedColors = labels2colors(net$colors)</span><br><span class="line"><span class="comment"># Plot the dendrogram and the module colors underneath</span></span><br><span class="line">plotDendroAndColors(net$dendrograms[[<span class="number">1</span>]], mergedColors[net$blockGenes[[<span class="number">1</span>]]],</span><br><span class="line">                    <span class="string">"Module colors"</span>,</span><br><span class="line">                    dendroLabels = <span class="literal">FALSE</span>, hang = <span class="number">0.03</span>,</span><br><span class="line">                    addGuide = <span class="literal">TRUE</span>, guideHang = <span class="number">0.05</span>)</span><br><span class="line"><span class="comment">#dev.off()</span></span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/HYXmnS3acVgPOyU.png" alt="modules_tree"></p>
<p>ummmm… 模塊似乎有點寬， 分的不均勻- - 感覺很爛的樣子。 要不再， 篩一下？</p>
<p>Modules 統計</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(reshape2)</span><br><span class="line">Color_TB = melt(table(labels2colors(net$colors)))</span><br><span class="line">Color_TB$Var1 = factor(Color_TB$Var1, levels = Color_TB$Var1[order(Color_TB$value,decreasing = <span class="literal">T</span>)])</span><br><span class="line">ggplot(Color_TB) + geom_bar(aes(x=Var1,y=value), stat = <span class="string">'identity'</span>,fill=Color_TB$Var1)+</span><br><span class="line">  theme_light() +theme(axis.text.x = element_text(angle = <span class="number">90</span>,hjust=<span class="number">1</span>))</span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/11/Ks396OUztmydnu2.png" alt="123"></p>
<h1 id="5-Traits"><a href="#5-Traits" class="headerlink" title="5. Traits"></a>5. Traits</h1><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">nGenes = ncol(datExpr);</span><br><span class="line">nSamples = nrow(datExpr);</span><br><span class="line"><span class="comment"># Recalculate MEs with color labels</span></span><br><span class="line">MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes</span><br><span class="line">MEs = orderMEs(MEs0)</span><br><span class="line">moduleTraitCor = cor(MEs, datTraits, use = <span class="string">"p"</span>);</span><br><span class="line">moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">sizeGrWindow(<span class="number">10</span>,<span class="number">6</span>)</span><br><span class="line"><span class="comment"># Will display correlations and their p-values</span></span><br><span class="line"><span class="comment"># png("T45_traits_heatmap.png",width=1000,height=600)</span></span><br><span class="line">textMatrix =  paste(signif(moduleTraitCor, <span class="number">2</span>), <span class="string">"\n("</span>,</span><br><span class="line">                           signif(moduleTraitPvalue, <span class="number">1</span>), <span class="string">")"</span>, sep = <span class="string">""</span>);</span><br><span class="line">dim(textMatrix) = dim(moduleTraitCor)</span><br><span class="line">par(mar = c(<span class="number">6</span>, <span class="number">8.5</span>, <span class="number">3</span>, <span class="number">3</span>));</span><br><span class="line"><span class="comment"># Display the correlation values within a heatmap plot</span></span><br><span class="line">labeledHeatmap(Matrix = moduleTraitCor,</span><br><span class="line">               xLabels = names(datTraits),</span><br><span class="line">               yLabels = names(MEs),</span><br><span class="line">               ySymbols = names(MEs),</span><br><span class="line">               colorLabels = <span class="literal">FALSE</span>,</span><br><span class="line">               colors = greenWhiteRed(<span class="number">50</span>),</span><br><span class="line">               textMatrix = textMatrix,</span><br><span class="line">               setStdMargins = <span class="literal">FALSE</span>,</span><br><span class="line">               cex.text = <span class="number">0.5</span>,</span><br><span class="line">               zlim = c(-<span class="number">1</span>,<span class="number">1</span>),</span><br><span class="line">               main = paste(<span class="string">"Module-trait relationships"</span>))</span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/6sAcpShBmkZXGyW.png" alt="traits_heatmap"><br>ummm, 能看個大概就好2333.還是有點意思的</p>
<h1 id="6-Focus-on-Weight"><a href="#6-Focus-on-Weight" class="headerlink" title="6. Focus on Weight"></a>6. Focus on Weight</h1><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">weight = as.data.frame(datTraits$Weight)</span><br><span class="line">modNames = substring(names(MEs), <span class="number">3</span>)</span><br><span class="line"></span><br><span class="line">geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = <span class="string">"p"</span>));</span><br><span class="line">MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));</span><br><span class="line"></span><br><span class="line">names(geneModuleMembership) = paste(<span class="string">"MM"</span>, modNames, sep=<span class="string">""</span>);</span><br><span class="line">names(MMPvalue) = paste(<span class="string">"p.MM"</span>, modNames, sep=<span class="string">""</span>);</span><br><span class="line"></span><br><span class="line">geneTraitSignificance = as.data.frame(cor(datExpr, weight, use = <span class="string">"p"</span>));</span><br><span class="line">GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));</span><br><span class="line"></span><br><span class="line">names(geneTraitSignificance) = paste(<span class="string">"GS."</span>, names(weight), sep=<span class="string">""</span>);</span><br><span class="line">names(GSPvalue) = paste(<span class="string">"p.GS."</span>, names(weight), sep=<span class="string">""</span>);</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">module = <span class="string">"grey"</span></span><br><span class="line">column = match(module, modNames);</span><br><span class="line">moduleGenes = moduleColors==module;</span><br><span class="line"></span><br><span class="line">sizeGrWindow(<span class="number">7</span>, <span class="number">7</span>);</span><br><span class="line"><span class="comment"># png("grey_weight.png")</span></span><br><span class="line">par(mfrow = c(<span class="number">1</span>,<span class="number">1</span>));</span><br><span class="line">verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),</span><br><span class="line">                   abs(geneTraitSignificance[moduleGenes, <span class="number">1</span>]),</span><br><span class="line">                   xlab = paste(<span class="string">"Module Membership in"</span>, module, <span class="string">"module"</span>),</span><br><span class="line">                   ylab = <span class="string">"Gene significance for body weight"</span>,</span><br><span class="line">                   main = paste(<span class="string">"Module membership vs. gene significance\n"</span>),</span><br><span class="line">                   cex.main = <span class="number">1.2</span>, cex.lab = <span class="number">1.2</span>, cex.axis = <span class="number">1.2</span>, col = module)</span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/10/jWYKMUZ1o4gqQu5.png" alt="grey_weight"><br>ummmm… 有點少， 哈哈哈， 才18個，數了一下。和tutorial 差好多呀</p>
<p>因爲是非模式物種，所以無法直接用tutorial給的方法做註釋。<br>直接調到後面的可視化把。</p>
<h1 id="7-NetWork-Visualizing"><a href="#7-NetWork-Visualizing" class="headerlink" title="7. NetWork  Visualizing"></a>7. NetWork  Visualizing</h1><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">nGenes = ncol(datExpr)</span><br><span class="line">nSamples = nrow(datExpr)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Calculate topological overlap anew: this could be done more efficiently by saving the TOM</span></span><br><span class="line"><span class="comment"># calculated during module detection, but let us do it again here.</span></span><br><span class="line">dissTOM = <span class="number">1</span>-TOMsimilarityFromExpr(datExpr, power = <span class="number">9</span>);</span><br><span class="line"><span class="comment"># Transform dissTOM with a power to make moderately strong connections more visible in the heatmap</span></span><br><span class="line">plotTOM = dissTOM^<span class="number">7</span>;</span><br><span class="line"><span class="comment"># Set diagonal to NA for a nicer plot</span></span><br><span class="line">diag(plotTOM) = <span class="literal">NA</span>;</span><br><span class="line"><span class="comment"># Call the plot function</span></span><br><span class="line">sizeGrWindow(<span class="number">9</span>,<span class="number">9</span>)</span><br><span class="line">TOMplot(plotTOM, geneTree, moduleColors, main = <span class="string">"Network heatmap plot, all genes"</span>)</span><br><span class="line"></span><br><span class="line">nSelect = <span class="number">400</span></span><br><span class="line"><span class="comment"># For reproducibility, we set the random seed</span></span><br><span class="line">set.seed(<span class="number">10</span>);</span><br><span class="line">select = sample(nGenes, size = nSelect);</span><br><span class="line">selectTOM = dissTOM[select, select];</span><br><span class="line"><span class="comment"># There's no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster.</span></span><br><span class="line">selectTree = hclust(as.dist(selectTOM), method = <span class="string">"average"</span>)</span><br><span class="line">selectColors = moduleColors[select];</span><br><span class="line"><span class="comment"># Open a graphical window</span></span><br><span class="line">sizeGrWindow(<span class="number">9</span>,<span class="number">9</span>)</span><br><span class="line"><span class="comment"># Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing</span></span><br><span class="line"><span class="comment"># the color palette; setting the diagonal to NA also improves the clarity of the plot</span></span><br><span class="line"><span class="comment">#png('123.png')</span></span><br><span class="line">plotDiss = selectTOM^<span class="number">7</span>;</span><br><span class="line">diag(plotDiss) = <span class="literal">NA</span>;</span><br><span class="line">TOMplot(plotDiss, selectTree, selectColors, main = <span class="string">"Network heatmap plot, selected genes"</span>)</span><br></pre></td></tr></table></figure></div>
<p>ummm… 好像是內存爆了，直接退出。 試試不用radian看看<br><span style="background:salmon">失敗了- -放棄， 跳過把</span><br>參考圖：<br><img src="https://i.loli.net/2020/06/08/wrpXGJj4qQOulim.png" alt="123"></p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">MEs = moduleEigengenes(datExpr, moduleColors)$eigengenes</span><br><span class="line"><span class="comment"># Isolate weight from the clinical traits</span></span><br><span class="line">weight = as.data.frame(datTraits$Weight);</span><br><span class="line">names(weight) = <span class="string">"weight"</span></span><br><span class="line"><span class="comment"># Add the weight to existing module eigengenes</span></span><br><span class="line">MET = orderMEs(cbind(MEs, weight))</span><br><span class="line"><span class="comment"># Plot the relationships among the eigengenes and the trait</span></span><br><span class="line">sizeGrWindow(<span class="number">5</span>,<span class="number">7.5</span>);</span><br><span class="line"><span class="comment">#png('T45_Tree_weight_module.png',width=400,height=600)</span></span><br><span class="line">par(cex = <span class="number">0.9</span>)</span><br><span class="line">plotEigengeneNetworks(MET, <span class="string">""</span>, marDendro = c(<span class="number">0</span>,<span class="number">4</span>,<span class="number">1</span>,<span class="number">2</span>), marHeatmap = c(<span class="number">3</span>,<span class="number">4</span>,<span class="number">1</span>,<span class="number">2</span>), cex.lab = <span class="number">0.8</span>, xLabelsAngle = <span class="number">90</span>)</span><br></pre></td></tr></table></figure></div>
<p><a href="https://imgchr.com/i/tTvZu9" target="_blank" rel="noopener"><img src="https://s1.ax1x.com/2020/06/10/tTvZu9.png" alt="tTvZu9.png"></a></p>
<p>這個圖還是可以畫的</p>
<h1 id="8-數據導出"><a href="#8-數據導出" class="headerlink" title="8. 數據導出"></a>8. 數據導出</h1><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">modules = c(<span class="string">"darkorange2"</span>, <span class="string">"grey"</span>,<span class="string">"lightcyan"</span>);</span><br><span class="line">inModule = is.finite(match(moduleColors, modules));</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span>(color <span class="keyword">in</span> modules)&#123;</span><br><span class="line">  write.table(colnames(t(net$colors[moduleColors==color])),paste(<span class="string">"module_"</span>,color,<span class="string">'.list'</span>,sep=<span class="string">""</span>) ,row.names=<span class="literal">F</span>,quote=<span class="literal">F</span>,col.names=<span class="literal">F</span>)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">probes = names(datExpr)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Recalculate topological overlap if needed</span></span><br><span class="line">TOM = TOMsimilarityFromExpr(datExpr, power = <span class="number">9</span>);</span><br><span class="line"><span class="comment"># Read in the annotation file</span></span><br><span class="line">annot = read.csv(file = <span class="string">"GeneAnnotation.csv"</span>);</span><br><span class="line"><span class="comment"># Select modules</span></span><br><span class="line"><span class="comment"># Select module probes</span></span><br><span class="line">modProbes = probes[inModule];</span><br><span class="line">modGenes = annot$gene_symbol[match(modProbes, annot$substanceBXH)];</span><br><span class="line"><span class="comment"># Select the corresponding Topological Overlap</span></span><br><span class="line">modTOM = TOM[inModule, inModule];</span><br><span class="line">dimnames(modTOM) = list(modProbes, modProbes)</span><br><span class="line"><span class="comment"># Export the network into edge and node list files Cytoscape can read</span></span><br><span class="line">cyt = exportNetworkToCytoscape(modTOM,</span><br><span class="line">  edgeFile = paste(<span class="string">"CytoscapeInput-edges-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">  nodeFile = paste(<span class="string">"CytoscapeInput-nodes-"</span>, paste(modules, collapse=<span class="string">"-"</span>), <span class="string">".txt"</span>, sep=<span class="string">""</span>),</span><br><span class="line">  weighted = <span class="literal">TRUE</span>,</span><br><span class="line">  threshold = <span class="number">0.02</span>,</span><br><span class="line">  nodeNames = modProbes,</span><br><span class="line">  altNodeNames = modGenes,</span><br><span class="line">  nodeAttr = moduleColors[inModule]);</span><br></pre></td></tr></table></figure></div>
<h1 id="9-如果-softThread-還是取6"><a href="#9-如果-softThread-還是取6" class="headerlink" title="9. 如果 softThread 還是取6"></a>9. 如果 softThread 還是取6</h1><h2 id="Modules-分佈"><a href="#Modules-分佈" class="headerlink" title="Modules 分佈"></a>Modules 分佈</h2><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(reshape2)</span><br><span class="line"><span class="keyword">library</span>(patchwork)</span><br><span class="line">Color_TB = melt(table(labels2colors(net$colors)))</span><br><span class="line">Color_TB$Var1 = factor(Color_TB$Var1, levels = Color_TB$Var1[order(Color_TB$value,decreasing = <span class="literal">T</span>)])</span><br><span class="line">p&lt;- ggplot(Color_TB) + geom_bar(aes(x=Var1,y=value), stat = <span class="string">'identity'</span>,fill=Color_TB$Var1)+</span><br><span class="line">  theme_light() +theme(axis.text.x = element_text(angle = <span class="number">90</span>,hjust=<span class="number">1</span>))</span><br><span class="line"></span><br><span class="line">SplitBar(p,<span class="number">8500</span>,<span class="number">10000</span>,<span class="number">27000</span>,<span class="number">0</span>,c(<span class="number">1</span>,<span class="number">4</span>))</span><br></pre></td></tr></table></figure></div>
<p><span style="background:salmon">畫圖函點擊</span><a href="https://www.yuque.com/liuwenkan/blog/ggplot_splitbar" target="_blank" rel="noopener">SplitBar</a><br><img src="https://i.loli.net/2020/06/11/pr5GuX3Z27dHkiz.png" alt="123"></p>
<p><img src="https://i.loli.net/2020/06/11/eLtsGyoIViqhdJ7.png" alt="S6_modules_tree"></p>
<h1 id="Optimal"><a href="#Optimal" class="headerlink" title="Optimal"></a>Optimal</h1><h2 id="A1-Reduce-Input-Transcripts"><a href="#A1-Reduce-Input-Transcripts" class="headerlink" title="A1. Reduce Input Transcripts"></a>A1. Reduce Input Transcripts</h2><p>這裏， 我們把閾值設定在 45(9*5)</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(WGCNA)</span><br><span class="line"></span><br><span class="line">A &lt;- read.csv(<span class="string">"All_isoform.TPM.matrix.anno.xls"</span>,sep=<span class="string">'\t'</span>)</span><br><span class="line"><span class="comment"># rownames:sample; colnames: Transcripts</span></span><br><span class="line">row.names(A) = A[[<span class="number">2</span>]]</span><br><span class="line">A = A[-(<span class="number">1</span>:<span class="number">2</span>)]</span><br><span class="line"></span><br><span class="line">Counts &lt;- read.csv(<span class="string">"/media/ken/Data/Yan/RNA-seq/report/4.exprs//All_isoform.COUNT.matrix.anno.xls"</span>,sep=<span class="string">'\t'</span>)</span><br><span class="line">Counts$Sum = rowSums(Counts[-c(<span class="number">1</span>:<span class="number">2</span>)])</span><br><span class="line">Counts_sub = Counts[which(Counts$Sum&gt;<span class="number">45</span>),]</span><br><span class="line">paste(round((nrow(Counts_sub)/nrow(Counts))*<span class="number">100</span>,<span class="number">2</span>),<span class="string">"%"</span>,sep=<span class="string">""</span>)</span><br><span class="line"></span><br><span class="line">A_sub &lt;- A[match(Counts_sub[[<span class="number">2</span>]],rownames(A)),]</span><br><span class="line"></span><br><span class="line">datExpr0 = t(A_sub)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Data filter</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr0, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!gsg$allOK)</span><br><span class="line">&#123;</span><br><span class="line">  <span class="comment"># Optionally, print the gene and sample names that were removed:</span></span><br><span class="line">  <span class="keyword">if</span> (sum(!gsg$goodGenes)&gt;<span class="number">0</span>)</span><br><span class="line">     printFlush(paste(<span class="string">"Removing genes:"</span>, paste(names(datExpr0)[!gsg$goodGenes], collapse = <span class="string">", "</span>)));</span><br><span class="line">  <span class="keyword">if</span> (sum(!gsg$goodSamples)&gt;<span class="number">0</span>)</span><br><span class="line">     printFlush(paste(<span class="string">"Removing samples:"</span>, paste(rownames(datExpr0)[!gsg$goodSamples], collapse = <span class="string">", "</span>)));</span><br><span class="line">  <span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line">  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></div>
<p>剩下了<code>&quot;32.96%&quot;</code>的 Transcripts</p>
<p>接下來退回到1.2</p>
<p><img src="https://i.loli.net/2020/06/11/vSlio9atZdUREDe.png" alt="T45_Soft_th"><br>ummmm, 好像還沒之前的好看了- -，</p>
<p>還是取9把，soft</p>
<p>modules 29 個。<br>traits聚類：</p>
<table>
<thead>
<tr>
<th align="center">Cluster1</th>
<th align="center">Cluster2</th>
</tr>
</thead>
<tbody><tr>
<td align="center"><img src="https://i.loli.net/2020/06/11/2zYmxMo7ARr3y9L.png" alt="T45_traits_heatmap"></td>
<td align="center"><a href="https://imgchr.com/i/tHgxTs" target="_blank" rel="noopener"><img src="https://s1.ax1x.com/2020/06/11/tHgxTs.png" alt="tHgxTs.png"></a></td>
</tr>
</tbody></table>
<p>最亮的沒有上次那麼亮，= =相關係數有點低， 不高</p>
<p>modules 分佈情況:</p>
<p><img src="https://i.loli.net/2020/06/11/F95UNu6cLeAD8Jb.png" alt="tmp"></p>
<h1 id="B-Remove-low-hits-amp-Non-DEGs"><a href="#B-Remove-low-hits-amp-Non-DEGs" class="headerlink" title="B. Remove low-hits &amp; Non-DEGs"></a>B. Remove low-hits &amp; Non-DEGs</h1><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(WGCNA)</span><br><span class="line"></span><br><span class="line">A &lt;- read.csv(<span class="string">"All_isoform.TPM.matrix.anno.xls"</span>,sep=<span class="string">'\t'</span>)</span><br><span class="line"><span class="comment"># rownames:sample; colnames: Transcripts</span></span><br><span class="line">row.names(A) = A[[<span class="number">2</span>]]</span><br><span class="line">A = A[-(<span class="number">1</span>:<span class="number">2</span>)]</span><br><span class="line"></span><br><span class="line">Counts &lt;- read.csv(<span class="string">"/media/ken/Data/Yan/RNA-seq/report/4.exprs//All_isoform.COUNT.matrix.anno.xls"</span>,sep=<span class="string">'\t'</span>)</span><br><span class="line">Counts$Sum = rowSums(Counts[-c(<span class="number">1</span>:<span class="number">2</span>)])</span><br><span class="line">Counts_sub = Counts[which(Counts$Sum&gt;<span class="number">9</span>),]</span><br><span class="line">paste(round((nrow(Counts_sub)/nrow(Counts))*<span class="number">100</span>,<span class="number">2</span>),<span class="string">"%"</span>,sep=<span class="string">""</span>)</span><br><span class="line"></span><br><span class="line">A_sub &lt;- A[match(Counts_sub[[<span class="number">2</span>]],rownames(A)),]</span><br><span class="line"></span><br><span class="line">DEG1 &lt;- read.table(<span class="string">"../edgeR_Intest/diffExpr.P1e-3_C2.matrix"</span>,head=<span class="literal">T</span>)</span><br><span class="line">DEG2 &lt;- read.table(<span class="string">"../edgeR_Liver/diffExpr.P1e-3_C2.matrix"</span>,head=<span class="literal">T</span>)</span><br><span class="line">DEG3 &lt;- read.table(<span class="string">"../edgeR_Muscle/diffExpr.P1e-3_C2.matrix"</span>,head=<span class="literal">T</span>)</span><br><span class="line"></span><br><span class="line">List = as.character(unique(DEG1[[<span class="number">1</span>]], DEG2[[<span class="number">1</span>]], DEG3[[<span class="number">1</span>]]))</span><br><span class="line"></span><br><span class="line">A_DEG = na.omit(A_sub[match(row.names(A_sub), List),])</span><br><span class="line"></span><br><span class="line">datExpr0 = t(A_DEG)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Data filter</span></span><br><span class="line">gsg = goodSamplesGenes(datExpr0, verbose = <span class="number">3</span>);</span><br><span class="line">gsg$allOK</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!gsg$allOK)</span><br><span class="line">&#123;</span><br><span class="line">  <span class="comment"># Optionally, print the gene and sample names that were removed:</span></span><br><span class="line">  <span class="keyword">if</span> (sum(!gsg$goodGenes)&gt;<span class="number">0</span>)</span><br><span class="line">     printFlush(paste(<span class="string">"Removing genes:"</span>, paste(names(datExpr0)[!gsg$goodGenes], collapse = <span class="string">", "</span>)));</span><br><span class="line">  <span class="keyword">if</span> (sum(!gsg$goodSamples)&gt;<span class="number">0</span>)</span><br><span class="line">     printFlush(paste(<span class="string">"Removing samples:"</span>, paste(rownames(datExpr0)[!gsg$goodSamples], collapse = <span class="string">", "</span>)));</span><br><span class="line">  <span class="comment"># Remove the offending genes and samples from the data:</span></span><br><span class="line">  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></div>
<p>跳回 1.2<br>ummmm, 這麼篩完以後， 剩下了<code>1105</code>個Transcripts 。。。</p>
<table>
<thead>
<tr>
<th align="center">聚類</th>
<th align="center">Soft圖</th>
</tr>
</thead>
<tbody><tr>
<td align="center">這次聚類結果，就很不一樣了</td>
<td align="center">不過sortpower這個圖，基本還是和前文一樣沒怎麼變。取0.9以上的話， 又得取10了</td>
</tr>
<tr>
<td align="center"><a href="https://imgchr.com/i/tbo3UU" target="_blank" rel="noopener"><img src="https://s1.ax1x.com/2020/06/11/tbo3UU.png" alt="tbo3UU.png"></a></td>
<td align="center"><a href="https://imgchr.com/i/tbo1ET" target="_blank" rel="noopener"><img src="https://s1.ax1x.com/2020/06/11/tbo1ET.png" alt="tbo1ET.png"></a></td>
</tr>
</tbody></table>
<p>還是去9吧。 1000個，秒算完。<br>先看看分組頻率 和樹圖<br>這個圖，我就覺得，看着舒服多了- -唉<br>|聚類|Soft圖|<br>|:—:|:—:|<br>|<img src="https://i.loli.net/2020/06/11/hUMPXSTErxqBI9n.png" alt="DEG_bar">)|<img src="https://i.loli.net/2020/06/11/hWnYQKlG1HpdqPU.png" alt="DEG_tree">|<br>|第一次畫出Tomplot，激動|這個聚類，相關性都不太高呀|<br>|<img src="https://i.loli.net/2020/06/11/PGng6FisKjfvJ48.png" alt="DEG_tomplot">)|<img src="https://i.loli.net/2020/06/11/IZJjsXWo2AmlHxU.png" alt="DEG_Tree_weight_module">|<br><img src="https://i.loli.net/2020/06/11/E3GxfJa8T2VNRlk.png" alt="DEG_traits_heatmap"></p>
<p>相關性確實是太低了。 我們看一下這1000基因的表達趨勢把。</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggplotify)</span><br><span class="line"></span><br><span class="line">A_DEG$Module = labels2colors(net$colors)</span><br><span class="line">A_DEG$ID = row.names(A_DEG)</span><br><span class="line"></span><br><span class="line">TB = melt(A_DEG,id.vars = c(<span class="string">"ID"</span>,<span class="string">"Module"</span>))</span><br><span class="line">TB$variable = factor(TB$variable,levels=sort(as.character(unique((TB$variable)))))</span><br><span class="line"></span><br><span class="line">TB$ID = factor(TB$ID, levels = row.names(A_DEG)[order(A_DEG$Module)])</span><br><span class="line"></span><br><span class="line">p2 &lt;- ggplot( TB, aes( variable, ID)) + theme(panel.grid.major = element_blank())+</span><br><span class="line">    theme(legend.key=element_blank(), axis.text.y = element_blank(),</span><br><span class="line">          axis.title=element_blank(),axis.ticks.y=element_blank(),</span><br><span class="line">          axis.text.x=element_text(angle=<span class="number">90</span>,hjust=<span class="number">1</span>))+</span><br><span class="line">    geom_tile(aes(fill=log(<span class="number">1</span>+value)))+</span><br><span class="line">    scale_fill_gradient2(low=<span class="string">"steelblue"</span>,mid=<span class="string">"white"</span>,high=<span class="string">"red"</span>,midpoint =<span class="number">1</span>)</span><br><span class="line"></span><br><span class="line">A_DEG$ID = factor(A_DEG$ID, levels = row.names(A_DEG)[order(A_DEG$Module)])</span><br><span class="line">p1 &lt;-ggplot(A_DEG[<span class="number">10</span>:<span class="number">11</span>], aes( x=<span class="string">"Module"</span>, y=ID)) + theme(panel.grid.major = element_blank())+</span><br><span class="line">    theme(axis.text.y = element_blank(), legend.position =  <span class="string">'left'</span>,</span><br><span class="line">        axis.title=element_blank(),axis.ticks.y=element_blank(),</span><br><span class="line">        axis.text.x=element_text(angle=<span class="number">90</span>,hjust=<span class="number">1</span>))+</span><br><span class="line">    geom_tile(fill=A_DEG$Module)</span><br><span class="line"></span><br><span class="line">p1+p2 +  plot_layout(design = <span class="string">'ABBBBBBBBBB'</span>)</span><br></pre></td></tr></table></figure></div>
<p>熱圖還是，可以的，就是配色有點麻煩- -<br><img src="https://i.loli.net/2020/06/11/awg2zCr6LpblocB.png" alt="co_heatmap"><br>只管來看， 聚類效果還是OK的把。不過correlation值太低了 = = 有什麼之後再去深究把。</p>
<p>添加樹<br>打包兩個函數</p>
<div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br></pre></td><td class="code"><pre><span class="line">PLOT2 &lt;- <span class="keyword">function</span>(dendr)&#123;</span><br><span class="line">  ggplot() +</span><br><span class="line">  geom_segment(data=segment(dendr), aes(x=x, y=y, xend=xend, yend=yend)) +</span><br><span class="line">  geom_text(data=label(dendr), aes(x, y, label=label, hjust=<span class="number">0</span>),</span><br><span class="line">           size=<span class="number">3</span>) +</span><br><span class="line">  coord_flip() + scale_y_reverse(expand=c(<span class="number">0.2</span>, <span class="number">0</span>)) +</span><br><span class="line">  theme(axis.line.y=element_blank(),</span><br><span class="line">        axis.ticks.y=element_blank(),</span><br><span class="line">        axis.text.y=element_blank(),</span><br><span class="line">        axis.title.y=element_blank(),</span><br><span class="line">        panel.background=element_rect(fill=<span class="string">"white"</span>),</span><br><span class="line">        panel.grid=element_blank())</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">LS_judg&lt;- <span class="keyword">function</span>(TB)&#123;</span><br><span class="line">  <span class="keyword">if</span> (TB[<span class="string">'x'</span>] == TB[<span class="string">'xend'</span>])&#123;</span><br><span class="line">    R = <span class="string">'v'</span></span><br><span class="line">  &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">    R = <span class="string">'h'</span></span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">return</span>(R)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">TreePlot &lt;- <span class="keyword">function</span>(dendr, Width_tree)&#123;</span><br><span class="line">  dendr$labels$x = Width_tree[dendr$labels$x]</span><br><span class="line">  dendr$segments = dendr$segments[order(dendr$segments$x),]</span><br><span class="line">  <span class="comment"># 做個list</span></span><br><span class="line">  List = c()</span><br><span class="line">  <span class="keyword">for</span>(i <span class="keyword">in</span> c(<span class="number">1</span>:length(Width_tree)))&#123;</span><br><span class="line">    List = c(List, which(dendr$segments$x==i))</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="comment"># 調整豎線</span></span><br><span class="line">  <span class="keyword">for</span>( i <span class="keyword">in</span> c(<span class="number">1</span>:length(dendr$labels$label)))&#123;</span><br><span class="line">  Dif_tmp = Width_tree[i]- dendr$segments$x[List[i]]</span><br><span class="line">  dendr$segments$x[List[i]:nrow(dendr$segments)] = dendr$segments$x[List[i]:nrow(dendr$segments)]+Dif_tmp</span><br><span class="line">  dendr$segments$xend[List[i]:nrow(dendr$segments)] = dendr$segments$xend[List[i]:nrow(dendr$segments)]+Dif_tmp</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="comment"># 刪除添加新的橫線</span></span><br><span class="line">  dendr$segments = dendr$segments[dendr$segments$x == dendr$segments$xend,]</span><br><span class="line">  Y_list = dendr$segments$y[duplicated(dendr$segments$y)]</span><br><span class="line">  <span class="keyword">for</span>(i <span class="keyword">in</span> Y_list)&#123;  </span><br><span class="line">  tmp = (dendr$segments$x[dendr$segments$y  == i])</span><br><span class="line">  dendr$segments = rbind(dendr$segments, data.frame(x=tmp[<span class="number">1</span>],y=i,xend=tmp[<span class="number">2</span>],yend=i))</span><br><span class="line">  &#125;</span><br><span class="line">  Seg = dendr$segments</span><br><span class="line">  <span class="keyword">for</span>( i <span class="keyword">in</span> c(nrow(dendr$segments):<span class="number">1</span>))&#123;</span><br><span class="line">    <span class="keyword">if</span>(LS_judg(dendr$segments[i,]) == <span class="string">'h'</span>)&#123;</span><br><span class="line">      tmpy = dendr$segments[i,]$y</span><br><span class="line">      tmpTB = Seg[Seg$yend== tmpy,]</span><br><span class="line">        <span class="keyword">for</span>( ii <span class="keyword">in</span> c(<span class="number">1</span>:nrow(tmpTB)))&#123;</span><br><span class="line">        <span class="keyword">if</span>(LS_judg(tmpTB[ii,]) == <span class="string">'v'</span>)&#123;</span><br><span class="line">          Num = rownames(tmpTB[<span class="number">1</span>,])</span><br><span class="line">          dendr$segments[match(Num, row.names(dendr$segments)),][c(<span class="number">1</span>,<span class="number">3</span>)] = sum(Seg[i,c(<span class="string">'x'</span>,<span class="string">'xend'</span>)])/<span class="number">2</span></span><br><span class="line">        &#125;</span><br><span class="line">      &#125;</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="comment"># 重画横线：</span></span><br><span class="line">  dendr$segments = dendr$segments[dendr$segments$x == dendr$segments$xend,]</span><br><span class="line">  Y_list = dendr$segments$y[duplicated(dendr$segments$y)]</span><br><span class="line">  <span class="keyword">for</span>(i <span class="keyword">in</span> Y_list)&#123;  </span><br><span class="line">  tmp = (dendr$segments$x[dendr$segments$y  == i])</span><br><span class="line">  dendr$segments = rbind(dendr$segments, data.frame(x=tmp[<span class="number">1</span>],y=i,xend=tmp[<span class="number">2</span>],yend=i))</span><br><span class="line">  &#125;</span><br><span class="line"> <span class="keyword">return</span>(dendr)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></div>

<h3 id="画图"><a href="#画图" class="headerlink" title="画图"></a>画图</h3><div class="highlight-wrap"autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" contenteditable="true"data-rel="R"><figure class="iseeu highlight /r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line"><span class="keyword">library</span>(ggdendro)</span><br><span class="line"><span class="keyword">library</span>(reshape2)</span><br><span class="line"><span class="keyword">library</span>(patchwork)</span><br><span class="line"><span class="keyword">library</span>(RColorBrewer)</span><br><span class="line"></span><br><span class="line">colorRampPalette(rev(brewer.pal(n = <span class="number">7</span>,name = <span class="string">"RdYlBu"</span>))) -&gt; cc</span><br><span class="line"></span><br><span class="line">TB_color = (data.frame(net$colors,color=labels2colors(net$colors)))</span><br><span class="line">A_DEG$ID = row.names(A_DEG)</span><br><span class="line">A_DEG$Module = TB_color$color[match(row.names(A_DEG), row.names(TB_color))]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Samples dendrogram</span></span><br><span class="line">hc &lt;- hclust(dist(t(A_DEG[<span class="number">1</span>:<span class="number">9</span>])))</span><br><span class="line">dendr_head &lt;- dendro_data(hc, type=<span class="string">"rectangle"</span>) <span class="comment"># convert for ggplot</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Module dendrogram</span></span><br><span class="line">hc &lt;- hclust(dist(t(MEs)))</span><br><span class="line">dendr &lt;- dendro_data(hc, type=<span class="string">"rectangle"</span>) <span class="comment"># convert for ggplot</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 1. order and list</span></span><br><span class="line">Module_list = as.character(dendr$labels$label[order(dendr$labels$x)])</span><br><span class="line">Module_or = c()</span><br><span class="line"><span class="keyword">for</span>( i <span class="keyword">in</span> strsplit(Module_list,<span class="string">'ME'</span>))&#123;</span><br><span class="line">  Module_or = c(Module_or, i[<span class="number">2</span>])</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">## 2. Axis location</span></span><br><span class="line">TB_mbar = data.frame(table(A_DEG$Module))</span><br><span class="line">TB_mbar = TB_mbar = data.frame(table(A_DEG$Module))</span><br><span class="line">TB_mbar$hei = TB_mbar$Freq[<span class="number">1</span>]/<span class="number">2</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span>(i <span class="keyword">in</span> c(<span class="number">2</span>:nrow(TB_mbar)))&#123;</span><br><span class="line">  TB_mbar$hei[i] =sum(TB_mbar$Freq[<span class="number">1</span>:i-<span class="number">1</span>])+ (TB_mbar$Freq[i]/<span class="number">2</span>)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">Width_tree = TB_mbar$hei</span><br><span class="line"></span><br><span class="line">AA &lt;- TreePlot(dendr, Width_tree)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Module Bar</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## acquiring module list</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># order module</span></span><br><span class="line">A_DEG$Module = factor(A_DEG$Module, levels = Module_or)</span><br><span class="line"><span class="comment"># order ID</span></span><br><span class="line">A_DEG$ID = factor(A_DEG$ID, levels = row.names(A_DEG)[order(A_DEG$Module)])</span><br><span class="line"></span><br><span class="line"><span class="comment"># Transcripts Heatmap</span></span><br><span class="line"></span><br><span class="line">primary_data = as.matrix(A_DEG[<span class="number">1</span>:<span class="number">9</span>])</span><br><span class="line"><span class="comment">#transformations</span></span><br><span class="line">data = log2(primary_data+<span class="number">1</span>)</span><br><span class="line">data = as.matrix(data) <span class="comment"># convert to matrix</span></span><br><span class="line"><span class="comment"># Centering rows</span></span><br><span class="line">data = data.frame(t(scale(t(data), scale=<span class="literal">F</span>)))</span><br><span class="line"></span><br><span class="line">TB = melt(cbind(data,A_DEG[c(<span class="string">"ID"</span>,<span class="string">"Module"</span>)]),id.vars = c(<span class="string">"ID"</span>,<span class="string">"Module"</span>))</span><br><span class="line">TB$ID = factor(TB$ID, levels = row.names(A_DEG)[order(A_DEG$Module)])</span><br><span class="line">TB$variable = factor(TB$variable, levels =levels(dendr_head$labels$label))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">p0 &lt;- ggplot() +</span><br><span class="line">      geom_segment(data=segment(AA), aes(x=x, y=y, xend=xend, yend=yend)) +</span><br><span class="line">      geom_text(data=label(AA), aes(x, y, label=label, hjust=<span class="number">0</span>),</span><br><span class="line">               size=<span class="number">3</span>) +</span><br><span class="line">      coord_flip() + scale_y_reverse(expand=c(<span class="number">0.7</span>, <span class="number">0</span>)) +</span><br><span class="line">      theme(axis.line.y=element_blank(),</span><br><span class="line">            axis.ticks.y=element_blank(),</span><br><span class="line">            axis.text.y=element_blank(),</span><br><span class="line">            axis.title =element_blank(),</span><br><span class="line">            panel.background=element_rect(fill=<span class="string">"white"</span>),</span><br><span class="line">            panel.grid=element_blank())</span><br><span class="line"></span><br><span class="line">p1 &lt;-ggplot(A_DEG[<span class="number">10</span>:<span class="number">11</span>], aes( x=<span class="string">"Module"</span>, y=ID)) +</span><br><span class="line">    theme(panel.grid.major = element_blank())+</span><br><span class="line">    theme(axis.text.y = element_blank(), legend.position =  <span class="string">'left'</span>,</span><br><span class="line">        axis.title=element_blank(),</span><br><span class="line">        axis.ticks=element_blank(),</span><br><span class="line">        panel.background=element_rect(fill=<span class="string">"white"</span>),</span><br><span class="line">        axis.text.x=element_text(angle=<span class="number">90</span>,hjust=<span class="number">1</span>))+</span><br><span class="line">    geom_tile(fill=A_DEG$Module)</span><br><span class="line"></span><br><span class="line">p2 &lt;- ggplot( TB, aes( variable, ID)) +</span><br><span class="line">    theme(panel.grid.major = element_blank())+</span><br><span class="line">    theme(legend.key=element_blank(), axis.text.y = element_blank(),</span><br><span class="line">    axis.title=element_blank(),axis.ticks.y=element_blank(),</span><br><span class="line">    axis.text.x=element_text(angle=<span class="number">90</span>,hjust=<span class="number">1</span>))+</span><br><span class="line">    geom_tile(aes(fill=value))+</span><br><span class="line">    scale_fill_gradientn(colors=cc(<span class="number">100</span>))</span><br><span class="line"></span><br><span class="line">p4 &lt;- ggplot() +</span><br><span class="line">          geom_segment(data=segment(dendr_head), aes(x=x, y=y, xend=xend, yend=yend)) +</span><br><span class="line">          scale_y_reverse(expand=c(<span class="number">0</span>, <span class="number">0</span>)) +</span><br><span class="line">          theme(axis.line.y=element_blank(),</span><br><span class="line">                axis.ticks.x=element_blank(),</span><br><span class="line">                axis.text.x=element_blank(),</span><br><span class="line">                axis.title=element_blank(),</span><br><span class="line">                panel.background=element_rect(fill=<span class="string">"white"</span>),</span><br><span class="line">                panel.grid=element_blank())</span><br><span class="line"></span><br><span class="line">layout_P &lt;- <span class="string">"</span></span><br><span class="line"><span class="string">    AAAABCCCCCC</span></span><br><span class="line"><span class="string">    AAAABCCCCCC</span></span><br><span class="line"><span class="string">    AAAABCCCCCC</span></span><br><span class="line"><span class="string">    AAAABCCCCCC</span></span><br><span class="line"><span class="string">    AAAABCCCCCC</span></span><br><span class="line"><span class="string">    AAAABCCCCCC</span></span><br><span class="line"><span class="string">    #####DDDDDD</span></span><br><span class="line"><span class="string">    "</span></span><br><span class="line"></span><br><span class="line">p0 + p1 + p2 + p4 +  plot_layout(design = layout_P)</span><br></pre></td></tr></table></figure></div>
<p><img src="https://i.loli.net/2020/06/13/1COSfoY6KJGTFil.png" alt="DEG_heatmap"></p>
<hr>
<p>github: <a href="https://github.com/Karobben" target="_blank" rel="noopener">https://github.com/Karobben</a><br>blog: <a href="http://Karobben.github.io">Karobben.github.io</a><br>R 语言画图索引: <a href="https://karobben.github.io/R/R-index.html">https://karobben.github.io/R/R-index.html</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://karobben.github.io/R/WGCNA.html" data-id="ckcogixog006t2bfc8nldf4xg" class="article-share-link">Share</a>
      
      
    </footer>
  </div>
  
    
  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    

  
    
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/06/">June 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2020/06/30/tmp/">test</a>
          </li>
        
      </ul>
    </div>
  </div>

  
  <!-- Weather -->
<div id="weather-view-he" ></div>
<script>
WIDGET = {ID: 'HfP68rcshb'};
</script>
<script type="text/javascript" src="https://apip.weatherdt.com/view/static/js/r.js?v=1111"></script>

</aside>

          <!-- Weather Plug -->
        
        <!--git talk-->
        <section id="main">
          <div style='padding-top:10px'>
  <div style="background:white">
    <script src="https://utteranc.es/client.js"
            repo="Karobben/utterances"
            issue-term="pathname"
            theme="preferred-color-scheme"
            crossorigin="anonymous"
            async>
    </script>
  </div>
</div>
<!--git talk-->
  <!-- Link Gitalk 的支持文件  -->
  <!--
    <div>
      <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
      <script src="https://unpkg.com/gitalk@latest/dist/gitalk.min.js"></script>
      <div id="gitalk-container"></div>     <script type="text/javascript">
          var gitalk = new Gitalk({

          // gitalk的主要参数
              clientID: '96c208ade65e80e3aaac',
              clientSecret: '86e9240674a3815b0ddc31dababdb4998d5562f4',
              repo: 'OAuth-Application',
              owner: 'Karobben',
              admin: ['Karobben'],
              id:window.location.pathname,
          });
          gitalk.render('gitalk-container');
      </script>
    </div>
<!-- Gitalk end -->

        </section>
        <!-- Gitalk end -->

      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      <span style="float:right; list-style:none">
        <a style="padding: 10px" href="https://karobben.github.io/">Blog</a>
        <a style="padding: 10px" href="https://github.com/Karobben" target="_blank" rel="noopener">GitHub</a>
        <br>
        <a style="padding: 10px" href="https://space.bilibili.com/393056819" target="_blank" rel="noopener">BiliBili</a>
        <a href="mailto:a591465908@outlook.com">Mail me</a>
      </span>
      &copy; Karobben 2021<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
  <span>
    <p align = "center">
      <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
      <span id="busuanzi_container_site_pv">~<span id="busuanzi_value_site_pv"></span>~</span>
    </p>
  </span>
</footer>

    </div>
    <nav id="mobile-nav">
  <!--`
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  -->
  <a class="mobile-nav-link" href="../index.html">Home</a>
  <a class="mobile-nav-link" href="../R/R-index.html">R</a>
  <a class="mobile-nav-link" href="../Python/index.html">Python</a>
  <a class="mobile-nav-link" href="../Linux/index.html">Linux</a>
  <a class="mobile-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
  <a class="mobile-nav-link" href="../LearnNotes/index.html">Notes</a>
  <a class="mobile-nav-link" href="../Blog/index.html">Others</a>
</nav>

    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




<!--mermaid module-->

  <script src='https://unpkg.com/mermaid@7.1.2/dist/mermaid.min.js'></script>
  <script>
    if (window.mermaid) {
      mermaid.initialize({theme: 'forest'});
    }
  </script>



<!-- Searche Module-->
<script src="/js/search.js"></script>
<script src="/js/load.js"></script>
<!-- 这里可能有一些其他原来文件里的代码，也可以没有 -->

<!-- mathjax config similar to math.stackexchange -->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

  </div>
<script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body>
</html>
