---
url: paperrnaseqfish2
---

# De novo characterization of the liver transcriptome of javelin goby Synechogobius hasta and analysis of its transcriptomic profile following waterborne copper exposure

Cite: Chen, Q., Luo, Z., Huang, C. et al. De novo characterization of the liver transcriptome of javelin goby Synechogobius hasta and analysis of its transcriptomic profile following waterborne copper exposure. Fish Physiol Biochem 42, 979–994 (2016). https://doi.org/10.1007/s10695-015-0190-2


# General Information
## Statistic information

|Content|Words|Ratio|
|:--:|:--:|:--:|
|Total Words|5424|100%|
|Abstract|234|4.31%|
|Introduction|598|11.03%|
|M&M|1475|27.19%|
|Results|1143|21.07%|
|**Discussion**|1974|36.39%|

## Background
[***Synechogobius hasta***](https://baike.baidu.com/item/%E6%B2%99%E5%85%89%E9%B1%BC/1203814?fromtitle=%E7%9F%9B%E5%B0%BE%E5%A4%8D%E8%99%BE%E8%99%8E%E9%B1%BC&fromid=12760397):
![img](http://a0.att.hudong.com/53/60/01300000274589122785607359638.jpg)
Img from: [link](https://pai-hang-bang.cn/tujie/%E7%9F%9B%E5%B0%BE%E5%A4%8D%E9%B0%95%E8%99%8E%E9%B1%BC.htm)


# Abstract
1. Physiological study of S. hasta exposure to the Cu is studied, but **RNA-Seq is not**. (PS: this experiment is **Unique**)
2. Some **Statistic information** from RNA-seq.
3. Major change of **pathways** are...
4. This experiment is pretty useful.

# Introduction

P1: Cu contamination is harmful. **Comprehensive understanding** of the mechanisms is lack.
P2: Problems caused by Cu exposure are **too complicate to** be explained by few regulatory entities, as a result, a whole gene expression profiling is need.
P3: Though, previous studies find that lipid deposition and metabolic activity were changed,they focus on **few genes only**. Now, it is the first time for people profiling the RNA-Seq of this species.

# M&M
- Fish and Cu exposure
- Sampling and total RNA extraction
- cDNA library preparation and Illumina sequencing
- De novo assembly and function annotation
- Identification of differentially expressed genes
- Real-time quantitative PCR (qPCR) validation
- Statistical analysis

# Results
- Illumina sequencing and de novo transcriptome assembly
- Function annotation and classification of unigenes
- Differentially expressed genes involved in immune system
- Differentially expressed genes involved in apoptosis
- Differentially expressed genes involved in signal transduction
- Validation of differential gene expression by qPCR
> ## Illumina sequencing and de novo transcriptome assembly
> Reads and Assemble result statistic with tables.
>
> ## Function annotation and classification of unigenes
> 1. Annotation software and data base, figures and tables.
> 2. GO ontology: statistic; tables; charts.
> 3. KEGG ontology: statistics and examples
> ## Differentially expressed genes involved in immune system
> Fatty acid metabolism was changed, pathway charts was insert.
> ## Differentially expressed genes involved in immune system
> Changes in some specific pathways with table
> ## Differentially expressed genes involved in apoptosis
> Genes Change in apoptosis pathway with charts
> ## Differentially expressed genes involved in signal transduction
> MAPK and NF<sub>k</sub>B; tables.
> ## Validation of differential gene expression by qPCR
> The expression levels of 25 genes was validate.

# Discussion
- Liver transcriptome of S. hasta generated by RNAseq
- Differentially expressed genes involved in lipid metabolism
- Differentially expressed genes involved in immune system
- Differentially expressed genes involved in apoptosis
- Differentially expressed genes involved in signal transduction

## Liver transcriptome of S. hasta generated by RNAseq
1. Statistic information of De novo result. (PS: you may think this things was mentioned before. Yes, it is, But here, the author made a <span style="color:salmon;"><b>comparing with other species</b></span>. So, it is ok to recite them again)
2. COG is useful, GO is useful, KEGG is useful. (PS: What the hex??? Probably is because this paper was published in 2015. So, it might be done at 2014 or even earlier.)

## Differentially expressed genes involved in lipid metabolism
1. General background information and cited a bunch of recent study(PS: his study).
2. But the RNA-Seq is absent. So, this experiment is useful and meaningful.(PS: Again, recite the lyrics from the Abstract and Introduction)
3. It provide the evidences of mitochondrial damage and inhibition of B-oxidation. (PS: Gosh, finally, we see something new. And end of this section.)

## Differentially expressed genes involved in immune system
1. BG information of immune system and the benifitial of the RNA-Seq. (PS: Suggest to ignore)
2. Interested Pathways and their BG infor.
3. Filled with a list of DEGs, and compared few of them with other studies.
4. suggest that immune reaction were involved in Cu-induced stress.

## Differentially expressed genes involved in apoptosis
1. BG infor for Apoptosis and CASP3.
2. Calpain up-regulated; caspase-3 down-regulated, this was support in other two reports. (PS: <span style="color:salmon;"><b>They spend 236 words to interpret five words</b></span>: Calpain up-regulated; caspase-3 down-regulate)

## Differentially expressed genes involved in signal transduction
1. BG: for MAPK pathway. Evidences from other papers.
Here: c-fos involved in this pathway.
2. Cu could induce ROS to active MAPK to elevate the phosphorylation. (PS: No ending sentence for this paragraph. The last sentence is lack of coherent.)
3. (P2) BG: NF<sub>k</sub>B is an important redox-sensitive TF for immune-response.
Here: NF<sub>k</sub>B down-regulated.
But, there have no consensus on it. So, more experiments are needee.
(PS: This paragraph is mach readable than few previous)
4. BG: important of Ca<sup>2+</sup> and Cu<sup>2+</sup> can effect Ca<sup>2+</sup>
Here: Cu down-regulated genes in the calcium signaling pathway.
The understanding of this process is helpful.
5. Conclusion: This experiment is unique and this experiment has lots of potential contributions...


PS: Finally, I finished this paper. Personally, I'm not suggesting anyone to read this paper really much. Because it is a kind of wasting time if you want to learn how to writing from this passage. You can perceive that too much of non-sense words are filled for words-accumulating. And some sentences are lack of coherent. But, of course, it could be a good counter example because  it is not the worst, it's still a readable paper, it's just not a good model. And I just don't like it.

---
# Related papers:  
Morais S, Pratoomyot J, Taggart JB, Bron JE, Guy DR, Bell JG, Tocher DR (2011b) Genotype-specific responses in Atlantic salmon (Salmo salar) subject to dietary fish oil replacement by vegetable oil: a liver transcriptomic analysis. BMC Genomics 12:255

Morais S, Edvardsen RB, Tocher DR, Bell JG (2012a) Transcriptomic analyses of intestinal gene expression of juvenile Atlantic cod (Gadus morhua) fed diets with Camelina oil as replacement for fish oil. Comp Biochem Physiol B Biochem Mol Biol 161:283–293.  

Tacchi L, Secombes CJ, Bickerdike R, Adler MA, Venegas C, Takle H, Martin SA (2012)  Transcriptomic and physiological responses to fishmeal substitution with plant proteins in formulated feed in farmed Atlantic salmon (Salmo salar). BMC Genomics 13:363

---
# Related Notes:

[Transcriptomic analysis of juvenile Chinese sea bass (Lateolabrax maculatus) anesthetized by MS-222 (tricaine methanesulfonate) and eugenol](paperrnaseqfish1)

[Physiological and transcriptomic responses to fishmeal-based diet and rapeseed meal-based diet in two strains of gibel carp (Carassius gibelio)](paperrnaseqfish3)

[Transcriptomic response to soybean meal-based diets as the frst formulated feed in juvenile yellow perch (Perca favescens)](paperrnaseqfish4)
