<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>WGCNA | Karobben</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="WGCNA">
<meta property="og:type" content="website">
<meta property="og:title" content="WGCNA">
<meta property="og:url" content="https://karobben.github.io/Bioinfor/WGCNA.html">
<meta property="og:site_name" content="Karobben">
<meta property="og:description" content="WGCNA">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-07-16T07:13:33.051Z">
<meta property="article:modified_time" content="2020-07-16T07:13:33.051Z">
<meta property="article:author" content="Karobben">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="Karobben" type="application/atom+xml">
  
  
    <link rel="icon" href="img/tag.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

  
<link rel="stylesheet" href="/css/search.css">

  <script type="text/javascript"
   src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>

  <style>
  pre {
    color: salmon;
  }
  </style>
<meta name="generator" content="Hexo 4.2.1"></head>

<script>
  var _hmt = _hmt || [];
  (function() {
    var hm = document.createElement("script");
    hm.src = "https://hm.baidu.com/hm.js?8442e6ecc87d878298ba7d3bd0a5484c";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hm, s);
  })();
</script>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header"; style="background:#1B414B;height:150px">
  <!--div id="banner"></div>-->
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Karobben</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">Karobben Blog</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
          <a id="main-nav-toggle" class="nav-icon"></a>
          <!-- Header Section Start -->
  <header id="slider-area">
    <nav class="navbar navbar-expand-md fixed-top scrolling-navbar bg-white">
      <div class="container">
        <div class="collapse navbar-collapse" id="navbarCollapse">
          <ul class="navbar-nav mr-auto w-100 justify-content-end">
              <a class="main-nav-link" href="../index.html">Home</a>
              <a class="main-nav-link" href="../R/R-index.html">R</a>
              <a class="main-nav-link" href="../Python/index.html">Python</a>
              <a class="main-nav-link" href="../Linux/index.html">Linux</a>
              <a class="main-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
              <a class="main-nav-link" href="../LearnNotes/index.html">Notes</a>
              <a class="main-nav-link" href="../Blog/index.html">Others</a>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <!-- Header Section End -->

      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Looking for Something?"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://karobben.github.io"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><div align='right', style='padding-right:160px;padding-top:1px'>
  
  <span class="local-search local-search-google local-search-plugin" style=";position:absolute;z-index:2;">
      <input type="search" placeholder="Looking for Something?" id="local-search-input" class="local-search-input-cls" style="">
      <div id="local-search-result" class="local-search-result-cls"></div>
  </span>


</div>

<article  id="page-" class="article article-type-page" itemscope itemprop="blogPost">
<div id="tipue_search_content" style="display: none"></div>
  <div class="article-meta">
    <a href="/Bioinfor/WGCNA.html" class="article-date">
  <time datetime="2020-07-16T07:13:33.051Z" itemprop="datePublished">2020-07-16</time>
</a>
    
  </div>
  <div  class="article-inner">
    
    
      <header class="article-header">
        <!--
  
    <h1 class="article-title" itemprop="name">
      WGCNA
    </h1>
  
 -->
      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Go To Top Link -->
        <div title="Back to Top" class="back-to-top" style="background:salmon">
  <a href="#"  >
    <i >Top</i>
  </a>
</div>
<div class="title_index">
  <ul id="nav">
   <li>
     Titles
     <ul>
       <!-- Table of Contents -->

  <div id="toc" class="toc-article">
    <strong class="toc-title">Titles</strong>
    <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#WGCNA"><span class="toc-number">1.</span> <span class="toc-text">WGCNA</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Lecture-by-Steven-Horvath-2013-UCLA"><span class="toc-number">2.</span> <span class="toc-text">Lecture by Steven Horvath 2013, UCLA</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Unsigned-Network-and-Signed-Network"><span class="toc-number">2.1.</span> <span class="toc-text">Unsigned Network and Signed Network</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Q-amp-A"><span class="toc-number">3.</span> <span class="toc-text">Q&amp;A</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Data-Analysis-Questions"><span class="toc-number">3.1.</span> <span class="toc-text">Data Analysis Questions</span></a></li></ol></li></ol>
  </div>

<!-- Table of Contents Down-->

     </ul>
    </li>
   </ul>
</div>


        <h1 id="WGCNA"><a href="#WGCNA" class="headerlink" title="WGCNA"></a>WGCNA</h1><h1 id="Lecture-by-Steven-Horvath-2013-UCLA"><a href="#Lecture-by-Steven-Horvath-2013-UCLA" class="headerlink" title="Lecture by Steven Horvath 2013, UCLA"></a>Lecture by Steven Horvath 2013, UCLA</h1><p>Though, it was designed for microarray data, but it applied in DNA methylation  data, RNA-seq, miRNA-seq, Peptide Count data…</p>
<ol>
<li>NetWork: the value between notes is either 0 or 1</li>
<li>weighted network: the value is variable between notes (Connection is weighted).</li>
</ol>
<h2 id="Unsigned-Network-and-Signed-Network"><a href="#Unsigned-Network-and-Signed-Network" class="headerlink" title="Unsigned Network and Signed Network"></a>Unsigned Network and Signed Network</h2><ol>
<li>Unsigned network:<ul>
<li>The value is from -1 to 1. (Both -1 and 1 means adjacent is 0, strongly correlated)</li>
<li>Both high value and low (nagetive) value are correlated because of adjacent.</li>
</ul>
</li>
<li>Singed network:<ul>
<li>The correlation value is form 0-1.</li>
</ul>
</li>
</ol>
<p>From unsigned correlation to singed network,</p>
<h1 id="Q-amp-A"><a href="#Q-amp-A" class="headerlink" title="Q&amp;A"></a>Q&amp;A</h1><p>Reference: <a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html" target="_blank" rel="noopener">Peter 2017</a></p>
<h2 id="Data-Analysis-Questions"><a href="#Data-Analysis-Questions" class="headerlink" title="Data Analysis Questions"></a>Data Analysis Questions</h2><ol>
<li><p><strong>How many samples</strong> do I need?<br>At least for <span style="background:salmon"><strong>15 samples</strong></span>. More samples could robust and refined results. Noises may can’t be removed if you have less samples.</p>
<br>
</li>
<li><p>Should I filter <strong>probesets or genes</strong>?</p>
</li>
</ol>
<ul>
<li><span style="background:salmon"><strong>Filtering genes by DEGs</strong></span> is not recommended since it <strong>completely</strong> <span style="background:salmon">invalidates</span> the <strong>scale-free topology assumption</strong>, which is the indicator of the <strong>soft thresholding power</strong>.</li>
<li>On the other hand, filtering gene by DE will lead to a set of correlated genes that will essentially form a single (or few high correlated) module respectively.<br>
</li>
</ul>
<ol start="3">
<li>What <strong>argument</strong> (option) settings are recommended?<br>In general, we attempt to select suitable default which feat multiple applications and also, it is high reproducible.<br>While for new calculations, you can customize argument.</li>
</ol>
<ul>
<li><strong>Signed Network</strong><br>  The choice of <strong><em>Signed</em></strong> or <strong><em>Unsigned</em></strong> network is complicate, but generally, we prefer the <span  style="color:darkred">Signed network</span>.<br>  To construct <em>Signed Networks</em>, we usring:<br>  <code>type = &quot;signed&quot;</code> or <code>type = &quot;signed hybrid&quot;</code> in function such as <code>accuracyMeasures</code>, <code>adjacency</code>, <code>chooseOneHubInEachModule</code>, <code>chooseTopHubInEachModule</code>, <code>nearestNeighborConnectivity</code>, <code>nearestNeighborConnectivityMS</code>, <code>orderBranchesUsingHubGenes</code>, <code>softConnectivity</code>. Some functions useing <code>networkType</code> to select the <code>signed</code>.</li>
<li><strong>Robust correlation</strong><br>  Generaly, we recommend to using default arguments to detect the correlation unless you have enough reason to believe there is no outlier measurment. You can using <code>corFnc</code>, <code>cor</code>, et al. to customizing your own detective. For more, please go to <a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html" target="_blank" rel="noopener">Tutorial</a></li>
</ul>
<ol start="4">
<li><p>Can WGCNA be used to analyze <strong>RNA-Seq data</strong>?<br><strong>Peter</strong>： <span style="background:salmon">Yes</span>. As far as WGCNA is concerned, working with (<strong>properly normalized</strong>) RNA-seq data isn’t really any different from working with (<strong>properly normalized</strong>) microarray data.<br><strong>Suggestion</strong>:</p>
<ul>
<li>Removing <strong>low hits</strong> transcripts.<br>Low counting transcripts tend to reflect noise. (for example, removing all features that have a count of less than say 10 in more than 90% of the samples)</li>
<li><strong>Normalization</strong><br><code>varianceStabilizingTransformation</code> from <em>DESeq2</em> is really useful.<br>RPKM and FPKM is helpful, too.<br>We can also using <code>log2(x+1)</code>.<br><span style="color:salmon"><b>Notions</b></span>:<ol>
<li>Different algorithms have huge impact on the result of expression change, but have limit affect on WGCNA.</li>
<li>Id data comes from different batch, We can use <code>ComBat</code> (Exp: <a href="http://blog.sina.com.cn/s/blog_70a5f5210102wibx.html" target="_blank" rel="noopener">木头的博客</a>) for batch effect removal.</li>
<li>Finally, we usually check quantile scatterplots to make sure there are no <strong>systematic shifts</strong> between samples; if sample quantiles show correlations (which they usually do), quantile normalization can be used to remove this effect.</li>
</ol>
</li>
</ul>
</li>
<li><p>Data heterogeneous<br>Data heterogeneous can effect any statistical analysis. (Skip)</p>
</li>
<li><p><strong>Soft-thresholding power</strong><br>can’t get a good scale-free topology index no matter how high I set the soft-thresholding power.</p>
<ol>
<li>First, the user should ensure that <span style='background:salmon'> variables (probesets, genes etc.) have not been filtered by differential expression</span> with respect to a sample trait.<br>Probability: Checking the clustering tree (<a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-01-dataInput.pdf" target="_blank" rel="noopener">exp</a>); strong clusters in the tree indicates globally different groups of sample. It may caused by batch effects or heterogenous. Carefully adjust the samples before building topology index.</li>
</ol>
<p>If the one causing heterogenous you don’t remove, you can still chosen the soft thresholding power by the number of samples at table below.</p>
</li>
</ol>
<table>
<thead>
<tr>
<th align="center">Number of samples</th>
<th align="center">Unsigned and signed hybrid networks</th>
<th align="center">Signed networks</th>
</tr>
</thead>
<tbody><tr>
<td align="center">Less than 20</td>
<td align="center">9</td>
<td align="center">18</td>
</tr>
<tr>
<td align="center">20-30</td>
<td align="center">8</td>
<td align="center">16</td>
</tr>
<tr>
<td align="center">30-40</td>
<td align="center">7</td>
<td align="center">14</td>
</tr>
<tr>
<td align="center">more than 40</td>
<td align="center">6</td>
<td align="center">12</td>
</tr>
</tbody></table>
<hr>
<p>github: <a href="https://github.com/Karobben" target="_blank" rel="noopener">https://github.com/Karobben</a><br>blog: <a href="http://Karobben.github.io">Karobben.github.io</a><br>R 语言画图索引: <a href="https://karobben.github.io/R/R-index.html">https://karobben.github.io/R/R-index.html</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://karobben.github.io/Bioinfor/WGCNA.html" data-id="ckcogivwz001r2bfchyda1twi" class="article-share-link">Share</a>
      
      
    </footer>
  </div>
  
    
  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    

  
    
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/06/">June 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2020/06/30/tmp/">test</a>
          </li>
        
      </ul>
    </div>
  </div>

  
  <!-- Weather -->
<div id="weather-view-he" ></div>
<script>
WIDGET = {ID: 'HfP68rcshb'};
</script>
<script type="text/javascript" src="https://apip.weatherdt.com/view/static/js/r.js?v=1111"></script>

</aside>

          <!-- Weather Plug -->
        
        <!--git talk-->
        <section id="main">
          <div style='padding-top:10px'>
  <div style="background:white">
    <script src="https://utteranc.es/client.js"
            repo="Karobben/utterances"
            issue-term="pathname"
            theme="preferred-color-scheme"
            crossorigin="anonymous"
            async>
    </script>
  </div>
</div>
<!--git talk-->
  <!-- Link Gitalk 的支持文件  -->
  <!--
    <div>
      <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
      <script src="https://unpkg.com/gitalk@latest/dist/gitalk.min.js"></script>
      <div id="gitalk-container"></div>     <script type="text/javascript">
          var gitalk = new Gitalk({

          // gitalk的主要参数
              clientID: '96c208ade65e80e3aaac',
              clientSecret: '86e9240674a3815b0ddc31dababdb4998d5562f4',
              repo: 'OAuth-Application',
              owner: 'Karobben',
              admin: ['Karobben'],
              id:window.location.pathname,
          });
          gitalk.render('gitalk-container');
      </script>
    </div>
<!-- Gitalk end -->

        </section>
        <!-- Gitalk end -->

      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      <span style="float:right; list-style:none">
        <a style="padding: 10px" href="https://karobben.github.io/">Blog</a>
        <a style="padding: 10px" href="https://github.com/Karobben" target="_blank" rel="noopener">GitHub</a>
        <br>
        <a style="padding: 10px" href="https://space.bilibili.com/393056819" target="_blank" rel="noopener">BiliBili</a>
        <a href="mailto:a591465908@outlook.com">Mail me</a>
      </span>
      &copy; Karobben 2021<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
  <span>
    <p align = "center">
      <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
      <span id="busuanzi_container_site_pv">~<span id="busuanzi_value_site_pv"></span>~</span>
    </p>
  </span>
</footer>

    </div>
    <nav id="mobile-nav">
  <!--`
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  -->
  <a class="mobile-nav-link" href="../index.html">Home</a>
  <a class="mobile-nav-link" href="../R/R-index.html">R</a>
  <a class="mobile-nav-link" href="../Python/index.html">Python</a>
  <a class="mobile-nav-link" href="../Linux/index.html">Linux</a>
  <a class="mobile-nav-link" href="../Bioinfor/index.html">Bio-Protocal</a>
  <a class="mobile-nav-link" href="../LearnNotes/index.html">Notes</a>
  <a class="mobile-nav-link" href="../Blog/index.html">Others</a>
</nav>

    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




<!--mermaid module-->

  <script src='https://unpkg.com/mermaid@7.1.2/dist/mermaid.min.js'></script>
  <script>
    if (window.mermaid) {
      mermaid.initialize({theme: 'forest'});
    }
  </script>



<!-- Searche Module-->
<script src="/js/search.js"></script>
<script src="/js/load.js"></script>
<!-- 这里可能有一些其他原来文件里的代码，也可以没有 -->

<!-- mathjax config similar to math.stackexchange -->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

  </div>
<script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"pluginRootPath":"live2dw/","pluginJsPath":"lib/","pluginModelPath":"assets/","tagMode":false,"log":false,"model":{"jsonPath":"/live2dw/assets/shizuku.model.json"},"display":{"position":"left","width":150,"height":300},"mobile":{"show":true},"react":{"opacity":0.7}});</script></body>
</html>
